{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 163,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# start docker deamon in terminal\n",
    "# !sudo docker login\n",
    "# OR - becuase the jupyther with allow root\n",
    "# !docker login"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copying gs://eila-nextflow-bucket/wdl_data/no-singleton.bed.gz...\n",
      "Skipping attempt to download to filename ending with slash                      \n",
      "(../nf_data/wdl_atac_pipeline/). This typically happens when using\n",
      "gsutil to download from a subdirectory created by the Cloud Console\n",
      "(https://cloud.google.com/console)\n",
      "InvalidUrlError: Invalid destination path: ../nf_data/wdl_atac_pipeline/.\n"
     ]
    }
   ],
   "source": [
    "!gsutil cp gs://eila-nextflow-bucket/wdl_data/no-singleton.bed.gz ../nf_data/wdl_atac_pipeline/ \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "TODO:\n",
    "\n",
    "when running with cloud. \n",
    "if you dont need to keep the execution logs, change the cloud settings to keep it for a short while\n",
    "Check that the cache if you need to set the cache to false (especially in plot process)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/eila/git_igvf/atomic-workflows/nextflow/hai-gcp-genomic-beb9207a40b8.json\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "json_path = os.path.join(os.path.dirname(os.path.dirname(os.getcwd())),\"hai-gcp-genomic-beb9207a40b8.json\")\n",
    "!ls $json_path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Activated service account credentials for: [eila-batch@hai-gcp-genomic.iam.gserviceaccount.com]\n"
     ]
    }
   ],
   "source": [
    "# # in the command line. same env\n",
    "import os\n",
    "json_path = os.path.join(os.path.dirname(os.path.dirname(os.getcwd())),\"hai-gcp-genomic-beb9207a40b8.json\")\n",
    "os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = json_path\n",
    "!gcloud auth activate-service-account eila-batch@hai-gcp-genomic.iam.gserviceaccount.com --key-file=$GOOGLE_APPLICATION_CREDENTIALS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                  Credentialed Accounts\n",
      "ACTIVE  ACCOUNT\n",
      "        886987709752-compute@developer.gserviceaccount.com\n",
      "*       eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\n",
      "\n",
      "To set the active account, run:\n",
      "    $ gcloud config set account `ACCOUNT`\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'hai-gcp-genomic'"
      ]
     },
     "execution_count": 168,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "!gcloud auth list\n",
    "project_name = !gcloud config get-value project\n",
    "project_name = project_name[0]\n",
    "project_name\n",
    "# gcloud config set account\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !gsutil cp gs://eila-nextflow-bucket/work/ATAC/b3/94bc73c6d43de41d17d75cac4fbda1/no-singleton.bed.gz.tbi /home/eila/git_igvf/downloads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "import os\n",
    "os.environ[\"TOWER_ACCESS_TOKEN\"] = 'eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz'\n",
    "os.environ[\"TOWER_WORKSPACE_ID\"]='247507587970818'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz\n",
      "247507587970818\n"
     ]
    }
   ],
   "source": [
    "!echo $TOWER_ACCESS_TOKEN\n",
    "!echo $TOWER_WORKSPACE_ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/bin'"
      ]
     },
     "execution_count": 173,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# !chmod +x ./bin/scrna_qc_metrics.py\n",
    "import os\n",
    "bin_fld = os.path.join(os.getcwd(),\"bin\")\n",
    "bin_fld"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "!chmod +x {os.path.join(bin_fld, \"atac_plot_metadata.R\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"run_tabix.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"atac_generate_merged_barcode_metadata.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"create_sample_filtered_fragment_file.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"compute_tss_enrichment_bulk.py\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"barcode_rank_functions.R\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"plot_insert_size_hist.py\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"compute_tss_enrichment_snapatac2.py\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"add_subpool_to_chromap_output.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"add_subpool_to_summary_file.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"filter_script.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"merge_chromap_logs.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"generate_insert_size_histogram_data.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"calculate_barcode_metadata_stats.sh\")}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# # TODO: For a new user: run these commands with your service account for the execution. I am using eila-batch@...\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/iam.serviceAccountUser\"\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/batch.jobsEditor\"\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/logging.viewer\"\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !nextflow -self-update"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  Version: 23.10.1 build 5891\n",
      "  Created: 12-01-2024 22:01 UTC \n",
      "  System: Linux 5.10.0-27-cloud-amd64\n",
      "  Runtime: Groovy 3.0.19 on OpenJDK 64-Bit Server VM 11.0.1-internal+0-adhoc..src\n",
      "  Encoding: UTF-8 (UTF-8)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!nextflow info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/eila/git_igvf/atomic-workflows/nextflow/nf_data\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "\n",
    "# Get the current working directory\n",
    "current_directory = os.getcwd()\n",
    "\n",
    "# Navigate two folders above\n",
    "two_folders_above = os.path.abspath(os.path.join(current_directory, '..', '..'))\n",
    "\n",
    "nf_data_base_path = os.path.join(two_folders_above,'nf_data')\n",
    "# Print the result\n",
    "print(nf_data_base_path)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 179,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/eila/git_igvf/atomic-workflows/nextflow/nf_data'"
      ]
     },
     "execution_count": 179,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nf_data_base_path\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# nf_data_project=\"test_atac_pipeline\"\n",
    "nf_data_project=\"wdl_atac_pipeline\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/eila/git_igvf/atomic-workflows/nextflow/nf_data/wdl_atac_pipeline/vm_atac_fastqs_files.tsv'"
      ]
     },
     "execution_count": 181,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# LOCAL\n",
    "# os.path.exists(os.path.join(nf_data_base_path,nf_data_project,\"local_atac_fastqs_files.tsv\"))\n",
    "# os.path.join(nf_data_base_path,nf_data_project,\"local_atac_fastqs_files.tsv\")\n",
    "\n",
    "# VM\n",
    "os.path.exists(os.path.join(nf_data_base_path,nf_data_project,\"vm_atac_fastqs_files.tsv\"))\n",
    "os.path.join(nf_data_base_path,nf_data_project,\"vm_atac_fastqs_files.tsv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 182,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "\n",
    "# Write the configuration file\n",
    "with open(\"nextflow_atac.config\", \"w\") as config_file:\n",
    "    config_file.write(f\"\"\"\n",
    "manifest {{\n",
    "  description = 'A nexflow pipeline for fg single cell atac' \n",
    "  mainScript = 'nf_wf_fg_singlecell_atac.nf'\n",
    "}}\n",
    "\n",
    "profiles {{\n",
    "  google_batch_profile {{\n",
    "    workDir = 'gs://eila-nextflow-bucket/work/ATAC'\n",
    "    includeConfig 'conf/docker-google-batch.config'\n",
    "    docker.enabled = true\n",
    "    wave.enabled = true\n",
    "    fusion.enabled = true\n",
    "    process.scratch = false\n",
    "    executor = 'google-batch'\n",
    "    docker.envWhitelist=['ENV_SYNAPSE_TOKEN','TOWER_ACCESS_TOKEN']\n",
    "    google.project = '{project_name}'\n",
    "    google.location = 'us-central1'\n",
    "    google.batch.spot = true\n",
    "    google.batch.serviceAccountEmail='eila-batch@hai-gcp-genomic.iam.gserviceaccount.com'\n",
    "  }}\n",
    "}}\n",
    "\n",
    "// Define ENV_SYNAPSE_TOKEN as a parameter\n",
    "params.FASTQS_SPEC_CH = '{nf_data_base_path}/{nf_data_project}/vm_atac_fastqs_files.tsv'\n",
    "// CHROMAP\n",
    "// 6a\n",
    "params.CHROMAP_IDX='gs://eila-nextflow-bucket/test_atac_pipeline/ref.index'\n",
    "// 6b:\n",
    "// FASTA file\n",
    "params.CHROMAP_READ_LENGTH = 2000\n",
    "params.CHROMAP_BC_ERROR_THRESHOLD = 2\n",
    "params.CHROMAP_BC_PROBABILITY_THRESHOLD = 0.9\n",
    "params.CHROMAP_READ_FORMAT = 'bc:0:-1,r1:0:-1,r2:0:-1'\n",
    "params.CHROMAP_DROP_REPETITIVE_READS = 4\n",
    "params.CHROMAP_QUALITY_THRESHOLD = 0\n",
    "params.CHROMAP_MERGE_LOGS_SCRIPT = \"merge_chromap_logs.sh\"\n",
    "params.CHROMAP_FRAGMENTS_HISTOGRAM_DATA_SCRIPT='generate_insert_size_histogram_data.sh'\n",
    "params.CHROMAP_GENOME_REFERENCE_FASTA='{nf_data_base_path}/{nf_data_project}/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz'\n",
    "\n",
    "// CHROMAP DEBUG: load output files \n",
    "// params.chromap_filter_fragments_tsv ='{nf_data_base_path}/{nf_data_project}/chromap_out_debug/ATAC_BMMC-single-donor.atac.filter.fragments.tsv'\n",
    "// params.barcode_summary_csv ='{nf_data_base_path}/{nf_data_project}/chromap_out_debug/BMMC-single-donor.atac.align.barcode.summary.csv'\n",
    "// params.chromap_alignment_log_out = '{nf_data_base_path}/{nf_data_project}/chromap_out_debug/BMMC-single-donor.atac.align.k4.hg38.log.txt'\n",
    "\n",
    "// CHROMAP WDL DEBUG: load output files \n",
    "params.chromap_fragments_tsv ='{nf_data_base_path}/{nf_data_project}/ATAC_BMMC-single-donor.atac.filter.fragments.tsv'\n",
    "params.barcode_summary_csv ='{nf_data_base_path}/{nf_data_project}/BMMC-single-donor.atac.align.barcode.summary.csv'\n",
    "params.chromap_alignment_log_out = '{nf_data_base_path}/{nf_data_project}/BMMC-single-donor.atac.align.k4.hg38.log.txt'\n",
    "params.bgzip_subpool_chromap_fragments_tsv_out='{nf_data_base_path}/{nf_data_project}/BMMC-single-donor.atac.filter.fragments.hg38.tsv.gz'\n",
    "params.no_singleton_gz='{nf_data_base_path}/{nf_data_project}/no-singleton.bed.gz'\n",
    "// params.tbi_no_singleton_bed_gz_out='{nf_data_base_path}/{nf_data_project}/.tbi'\n",
    "\n",
    "params.TSV_ADD_SUBPOOL_SCRIPT = 'add_subpool_to_chromap_output.sh'\n",
    "params.CSV_ADD_SUBPOOL_SCRIPT = 'add_subpool_to_summary_file.sh'\n",
    "params.FILTER_FRAGMENTS_SCRIPT = 'filter_script.sh'\n",
    "params.ATAC_FRAGMENTS_MIN_FRAG_CUTOFF=10\n",
    "\n",
    "// Other steps\n",
    "\n",
    "params.GENOME_GZ_GTF='http://ftp.ensembl.org/pub/release-109/gtf/homo_sapiens/Homo_sapiens.GRCh38.109.gtf.gz'\n",
    "params.ENV_SYNAPSE_TOKEN = \"nextflow secrets get SYNAPSE\".execute().text.trim()\n",
    "\n",
    "\n",
    "params.MAPQ_THRESHOLD = false\n",
    "params.BARCODE_TRANSLATE = 'path'\n",
    "params.ATAC_FRAGMENTS_CUTOFF = 4\n",
    "params.RUN_TABIX_SCRIPT='run_tabix.sh'\n",
    "params.ATAC_TSS_BASES_FLANK = 2000\n",
    "params.ATAC_TSS_COL_WITH_STRANDS_INFO = 4\n",
    "params.ATAC_TSS_SMOOTHING_WINDOW_SIZE = 20\n",
    "params.ATAC_TSS_REGION_BED_FILE='{nf_data_base_path}/{nf_data_project}/hg38.refGene.TSS.bed'\n",
    "params.ATAC_TSS_BULK_CALCULATION_SCRIPT='compute_tss_enrichment_bulk.py'\n",
    "params.ATAC_TSS_SNAPATAC2_CALCULATION_SCRIPT='compute_tss_enrichment_snapatac2.py'\n",
    "params.ATAC_TSS_SNAPATAC2_GENES_GTF='{nf_data_base_path}/{nf_data_project}/genes.gtf'\n",
    "params.ATAC_TSS_SNAPATAC2_MIN_FRAG_CUTOFF = 10\n",
    "\n",
    "params.ATAC_INSERT_SIZE_PLOT_SCRIPT='plot_insert_size_hist.py'\n",
    "\n",
    "// atac calculate_barcode_metadata_stats\n",
    "params.ATAC_SNAP_METADATA_STATS_SCRIPT='calculate_barcode_metadata_stats.sh'\n",
    "// atac plot barcode metadata\n",
    "params.ATAC_PLOT_METADATA_SCRIPT='atac_plot_metadata.R'\n",
    "params.ATAC_PLOT_METADATA_HELPER_SCRIPT='barcode_rank_functions.R'\n",
    "\n",
    "// debug file\n",
    "// params.SC_ATAC_QC_BARCODE_METADATA_FILE='{nf_data_base_path}/{nf_data_project}/barcode_metadata.txt'\n",
    "params.ATAC_QC_FRAGMENT_CUTOFF=10\n",
    "params.SC_ATAC_QC_BARCODE_OUTPUT_FILE='atac_barcode_metadata_output_file.png'\n",
    "    \n",
    "// params.SC_ATAC_BARCODE_METADATA_SCRIPT='atac_generate_merged_barcode_metadata.sh'\n",
    "// DEBUG: get the data from a file\n",
    "// params.SC_ATAC_BARCODE_METADATA_FILTERED_BARCODE_STATS='{nf_data_base_path}nf_wf_fg_singlecell/filtered_barcode_stats.csv'\n",
    "// params.SC_ATAC_BARCODE_METADATA_TSS_ENRICHMENT_BARCODE_STATS='{nf_data_base_path}nf_wf_fg_singlecell/tss_enrichment_barcode_stats.csv'\n",
    "\n",
    "// params.SC_ATAC_GENERATE_BARCODE_RANK_PLOT_PKR='PKR123'\n",
    "\n",
    "// params.SC_JOINT_CELL_BARCODE_PLOT_SCRTIP_PY='joint_cell_plotting.py'\n",
    "// params.SC_JOINT_CELL_BARCODE_PLOT_SCRTIP_R='joint_cell_plotting_density.R'\n",
    "\n",
    "// DEBUG\n",
    "\n",
    "params.CPUS_TO_USE_TSS=16\n",
    "params.CPUS_TO_USE_CHROMAP_IDX=32\n",
    "\"\"\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 183,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !rm -r /Users/eilaarich-landkof-stanford/Documents/Code/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/work\n",
    "# !rm -r /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/work\n",
    "# !gsutil rm -r gs://eila-nextflow-bucket/work/ATAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !rm .nextflow*\n",
    "\n",
    "# !ls /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/local_atac_fastqs_files.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 185,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "N E X T F L O W  ~  version 23.10.1\n",
      "Launching `nf_wf_fg_singlecell_atac.nf` [adoring_pasteur] DSL2 - revision: 36a818f600\n",
      "/home/eila/git_igvf/atomic-workflows/nextflow/nf_data/wdl_atac_pipeline/vm_atac_fastqs_files.tsv\n",
      "null\n",
      "after genome_fasta_ch\n",
      "[-        ] process > run_zcat                       -\u001b[K\n",
      "[-        ] process > run_add_subpool_to_chromap_... -\u001b[K\n",
      "[-        ] process > run_process_create_temp_sum... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_gzip_on_genes_gtf          -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "\u001b[19A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2rqELz6LM3eIxm\u001b[22;39m\u001b[K\n",
      "[-        ] process > run_zcat                       -\u001b[K\n",
      "[-        ] process > run_add_subpool_to_chromap_... -\u001b[K\n",
      "[-        ] process > run_process_create_temp_sum... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_gzip_on_genes_gtf          -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "\u001b[20A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2rqELz6LM3eIxm\u001b[22;39m\u001b[K\n",
      "[-        ] process > run_zcat                       -\u001b[K\n",
      "[-        ] process > run_add_subpool_to_chromap_... -\u001b[K\n",
      "[-        ] process > run_process_create_temp_sum... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[ea/e47e99] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "\u001b[20A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2rqELz6LM3eIxm\u001b[22;39m\u001b[K\n",
      "[fd/d4a0e5] process > run_zcat                       [100%] 1 of 1, cached: 1\u001b[K\n",
      "[-        ] process > run_add_subpool_to_chromap_... -\u001b[K\n",
      "[-        ] process > run_process_create_temp_sum... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[ea/e47e99] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb 10 01:58 genes.gtf.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb 10 01:58 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/41/45671d165921b1e3a885b4ea55909e/genes.gtf\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "\u001b[28A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2rqELz6LM3eIxm\u001b[22;39m\u001b[K\n",
      "[fd/d4a0e5] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_add_subpool_to_chromap_... -\u001b[K\n",
      "[-        ] process > run_process_create_temp_sum... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[ea/e47e99] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb 10 01:58 genes.gtf.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb 10 01:58 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/41/45671d165921b1e3a885b4ea55909e/genes.gtf\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb 10 01:58 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb 10 01:58 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb 10 01:58 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "\u001b[39A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2rqELz6LM3eIxm\u001b[22;39m\u001b[K\n",
      "[fd/d4a0e5] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_add_subpool_to_chromap_... -\u001b[K\n",
      "[00/e3e10e] process > run_process_create_temp_sum... [100%] 1 of 1, cached: 1\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[ea/e47e99] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb 10 01:58 genes.gtf.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb 10 01:58 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/41/45671d165921b1e3a885b4ea55909e/genes.gtf\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb 10 01:58 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb 10 01:58 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb 10 01:58 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "\u001b[39A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2rqELz6LM3eIxm\u001b[22;39m\u001b[K\n",
      "[fd/d4a0e5] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[d9/629469] process > run_add_subpool_to_chromap_... [100%] 1 of 1, cached: 1\u001b[K\n",
      "[00/e3e10e] process > run_process_create_temp_sum... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[ea/e47e99] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb 10 01:58 genes.gtf.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb 10 01:58 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/41/45671d165921b1e3a885b4ea55909e/genes.gtf\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb 10 01:58 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb 10 01:58 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb 10 01:58 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "start run_process_create_temp_summary_dict\u001b[K\n",
      "input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "------ There is NO conversion list ------\u001b[K\n",
      "temp_summary\u001b[K\n",
      "7021039 temp_summary\u001b[K\n",
      "finished run_process_create_temp_summary_dict\u001b[K\n",
      "\u001b[K\n",
      "\u001b[46A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2rqELz6LM3eIxm\u001b[22;39m\u001b[K\n",
      "[fd/d4a0e5] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[d9/629469] process > run_add_subpool_to_chromap_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[00/e3e10e] process > run_process_create_temp_sum... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[74/0c4d24] process > run_filter_aligned_fragment... [100%] 1 of 1, cached: 1\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[ea/e47e99] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb 10 01:58 genes.gtf.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb 10 01:58 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/41/45671d165921b1e3a885b4ea55909e/genes.gtf\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb 10 01:58 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb 10 01:58 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb 10 01:58 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "start run_process_create_temp_summary_dict\u001b[K\n",
      "input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "------ There is NO conversion list ------\u001b[K\n",
      "temp_summary\u001b[K\n",
      "7021039 temp_summary\u001b[K\n",
      "finished run_process_create_temp_summary_dict\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "Input chromap_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_chromap_output.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "\u001b[K\n",
      "\u001b[54A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2rqELz6LM3eIxm\u001b[22;39m\u001b[K\n",
      "[fd/d4a0e5] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[d9/629469] process > run_add_subpool_to_chromap_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[00/e3e10e] process > run_process_create_temp_sum... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[74/0c4d24] process > run_filter_aligned_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         [  0%] 0 of 1\u001b[K\n",
      "[ea/e47e99] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb 10 01:58 genes.gtf.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb 10 01:58 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/41/45671d165921b1e3a885b4ea55909e/genes.gtf\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb 10 01:58 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb 10 01:58 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb 10 01:58 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "start run_process_create_temp_summary_dict\u001b[K\n",
      "input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "------ There is NO conversion list ------\u001b[K\n",
      "temp_summary\u001b[K\n",
      "7021039 temp_summary\u001b[K\n",
      "finished run_process_create_temp_summary_dict\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "Input chromap_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_chromap_output.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "\u001b[K\n",
      "start run_filter_aligned_fragments\u001b[K\n",
      "input barcodeFragmentsTsvGz is BMMC-single-donor.atac.filter.fragments.hg38.tsv.gz\u001b[K\n",
      "input threshold is 10\u001b[K\n",
      "input temp_summary_dict is temp_summary\u001b[K\n",
      "------ Number of barcodes BEFORE filtering------\u001b[K\n",
      "7021039 temp_summary\u001b[K\n",
      "------ Filtering fragments ------\u001b[K\n",
      "------ Number of barcodes AFTER filtering ------\u001b[K\n",
      "3950969\u001b[K\n",
      "finished run_filter_aligned_fragments\u001b[K\n",
      "\u001b[K\n",
      "\u001b[65A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2rqELz6LM3eIxm\u001b[22;39m\u001b[K\n",
      "[fd/d4a0e5] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[d9/629469] process > run_add_subpool_to_chromap_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[00/e3e10e] process > run_process_create_temp_sum... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[74/0c4d24] process > run_filter_aligned_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         [  0%] 0 of 1\u001b[K\n",
      "[ea/e47e99] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb 10 01:58 genes.gtf.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb 10 01:58 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/41/45671d165921b1e3a885b4ea55909e/genes.gtf\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb 10 01:58 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb 10 01:58 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb 10 01:58 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "start run_process_create_temp_summary_dict\u001b[K\n",
      "input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "------ There is NO conversion list ------\u001b[K\n",
      "temp_summary\u001b[K\n",
      "7021039 temp_summary\u001b[K\n",
      "finished run_process_create_temp_summary_dict\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "Input chromap_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_chromap_output.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "\u001b[K\n",
      "start run_filter_aligned_fragments\u001b[K\n",
      "input barcodeFragmentsTsvGz is BMMC-single-donor.atac.filter.fragments.hg38.tsv.gz\u001b[K\n",
      "input threshold is 10\u001b[K\n",
      "input temp_summary_dict is temp_summary\u001b[K\n",
      "------ Number of barcodes BEFORE filtering------\u001b[K\n",
      "7021039 temp_summary\u001b[K\n",
      "------ Filtering fragments ------\u001b[K\n",
      "------ Number of barcodes AFTER filtering ------\u001b[K\n",
      "3950969\u001b[K\n",
      "finished run_filter_aligned_fragments\u001b[K\n",
      "\u001b[K\n",
      "\u001b[65A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2rqELz6LM3eIxm\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (1)\u001b[K\n",
      "[fd/d4a0e5] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[d9/629469] process > run_add_subpool_to_chromap_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[00/e3e10e] process > run_process_create_temp_sum... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[74/0c4d24] process > run_filter_aligned_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[07/637711] process > run_tabix_no_singleton (1)     [  0%] 0 of 1\u001b[K\n",
      "[ea/e47e99] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb 10 01:58 genes.gtf.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb 10 01:58 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/41/45671d165921b1e3a885b4ea55909e/genes.gtf\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb 10 01:58 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb 10 01:58 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb 10 01:58 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "start run_process_create_temp_summary_dict\u001b[K\n",
      "input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "------ There is NO conversion list ------\u001b[K\n",
      "temp_summary\u001b[K\n",
      "7021039 temp_summary\u001b[K\n",
      "finished run_process_create_temp_summary_dict\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "Input chromap_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_chromap_output.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "\u001b[K\n",
      "start run_filter_aligned_fragments\u001b[K\n",
      "input barcodeFragmentsTsvGz is BMMC-single-donor.atac.filter.fragments.hg38.tsv.gz\u001b[K\n",
      "input threshold is 10\u001b[K\n",
      "input temp_summary_dict is temp_summary\u001b[K\n",
      "------ Number of barcodes BEFORE filtering------\u001b[K\n",
      "7021039 temp_summary\u001b[K\n",
      "------ Filtering fragments ------\u001b[K\n",
      "------ Number of barcodes AFTER filtering ------\u001b[K\n",
      "3950969\u001b[K\n",
      "finished run_filter_aligned_fragments\u001b[K\n",
      "\u001b[K\n",
      "\u001b[66A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2rqELz6LM3eIxm\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (2)\u001b[K\n",
      "[fd/d4a0e5] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[d9/629469] process > run_add_subpool_to_chromap_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[00/e3e10e] process > run_process_create_temp_sum... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[74/0c4d24] process > run_filter_aligned_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[07/637711] process > run_tabix_no_singleton (1)     [  0%] 0 of 1\u001b[K\n",
      "[ea/e47e99] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[59/f41ab9] process > run_calculate_tss_enrichmen... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb 10 01:58 genes.gtf.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb 10 01:58 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/41/45671d165921b1e3a885b4ea55909e/genes.gtf\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb 10 01:58 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb 10 01:58 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb 10 01:58 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "start run_process_create_temp_summary_dict\u001b[K\n",
      "input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "------ There is NO conversion list ------\u001b[K\n",
      "temp_summary\u001b[K\n",
      "7021039 temp_summary\u001b[K\n",
      "finished run_process_create_temp_summary_dict\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "Input chromap_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_chromap_output.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "\u001b[K\n",
      "start run_filter_aligned_fragments\u001b[K\n",
      "input barcodeFragmentsTsvGz is BMMC-single-donor.atac.filter.fragments.hg38.tsv.gz\u001b[K\n",
      "input threshold is 10\u001b[K\n",
      "input temp_summary_dict is temp_summary\u001b[K\n",
      "------ Number of barcodes BEFORE filtering------\u001b[K\n",
      "7021039 temp_summary\u001b[K\n",
      "------ Filtering fragments ------\u001b[K\n",
      "------ Number of barcodes AFTER filtering ------\u001b[K\n",
      "3950969\u001b[K\n",
      "finished run_filter_aligned_fragments\u001b[K\n",
      "\u001b[K\n",
      "\u001b[66A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2rqELz6LM3eIxm\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (2)\u001b[K\n",
      "[fd/d4a0e5] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[d9/629469] process > run_add_subpool_to_chromap_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[00/e3e10e] process > run_process_create_temp_sum... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[74/0c4d24] process > run_filter_aligned_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[07/637711] process > run_tabix_no_singleton (1)     [  0%] 0 of 1\u001b[K\n",
      "[ea/e47e99] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[59/f41ab9] process > run_calculate_tss_enrichmen... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[66A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2rqELz6LM3eIxm\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (2)\u001b[K\n",
      "[fd/d4a0e5] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[d9/629469] process > run_add_subpool_to_chromap_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[00/e3e10e] process > run_process_create_temp_sum... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[74/0c4d24] process > run_filter_aligned_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[07/637711] process > run_tabix_no_singleton (1)     [  0%] 0 of 1\u001b[K\n",
      "[ea/e47e99] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[59/f41ab9] process > run_calculate_tss_enrichmen... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[66A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2rqELz6LM3eIxm\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (2)\u001b[K\n",
      "[fd/d4a0e5] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[d9/629469] process > run_add_subpool_to_chromap_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[00/e3e10e] process > run_process_create_temp_sum... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[74/0c4d24] process > run_filter_aligned_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[07/637711] process > run_tabix_no_singleton (1)     [  0%] 0 of 1\u001b[K\n",
      "[ea/e47e99] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[59/f41ab9] process > run_calculate_tss_enrichmen... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "\u001b[31mERROR ~ Error executing process > 'run_calculate_tss_enrichment_snapatac2 (1)'\u001b[K\n",
      "\u001b[K\n",
      "Caused by:\u001b[K\n",
      "  Process `run_calculate_tss_enrichment_snapatac2 (1)` terminated with an error exit status (1)\u001b[K\n",
      "\u001b[K\n",
      "Command executed:\u001b[K\n",
      "\u001b[K\n",
      "  echo 'start run_calculate_tss_enrichment_snapatac2'\u001b[K\n",
      "  echo 'calculation_script is compute_tss_enrichment_snapatac2.py. check that it is: snapatac2'\u001b[K\n",
      "  echo 'no_singleton_bed_gz: no-singleton.bed.gz'\u001b[K\n",
      "  echo 'genes_gtf_gzip_file_out: genes.gtf.gz'\u001b[K\n",
      "  echo 'min_frag_cutoff: 10'\u001b[K\n",
      "  echo 'prefix: BMMC-single-donor'\u001b[K\n",
      "  echo 'output file: BMMC-single-donor.tss_enrichment_barcode_stats.tsv'\u001b[K\n",
      "  \u001b[K\n",
      "  python3 /usr/local/bin/compute_tss_enrichment_snapatac2.py no-singleton.bed.gz genes.gtf.gz \"BMMC-single-donor.tss_enrichment_barcode_stats.tsv\" 10\u001b[K\n",
      "  echo 'finished run_calculate_tss_enrichment_snapatac2'\u001b[K\n",
      "\u001b[K\n",
      "Command exit status:\u001b[K\n",
      "  1\u001b[K\n",
      "\u001b[K\n",
      "Command output:\u001b[K\n",
      "  start run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "  calculation_script is compute_tss_enrichment_snapatac2.py. check that it is: snapatac2\u001b[K\n",
      "  no_singleton_bed_gz: no-singleton.bed.gz\u001b[K\n",
      "  genes_gtf_gzip_file_out: genes.gtf.gz\u001b[K\n",
      "  min_frag_cutoff: 10\u001b[K\n",
      "  prefix: BMMC-single-donor\u001b[K\n",
      "  output file: BMMC-single-donor.tss_enrichment_barcode_stats.tsv\u001b[K\n",
      "  Loading data from fragment file...\u001b[K\n",
      "\u001b[K\n",
      "Command error:\u001b[K\n",
      "  Traceback (most recent call last):\u001b[K\n",
      "    File \"/usr/local/bin/compute_tss_enrichment_snapatac2.py\", line 47, in <module>\u001b[K\n",
      "      main()\u001b[K\n",
      "    File \"/usr/local/bin/compute_tss_enrichment_snapatac2.py\", line 30, in main\u001b[K\n",
      "      data = snap.pp.import_data(\u001b[K\n",
      "  TypeError: import_data() got an unexpected keyword argument 'chrom_sizes'\u001b[K\n",
      "\u001b[K\n",
      "Work dir:\u001b[K\n",
      "  gs://eila-nextflow-bucket/work/ATAC/59/f41ab91bda34b8bb6baca18bffdc90\u001b[K\n",
      "\u001b[K\n",
      "Tip: you can try to figure out what's wrong by changing to the process work dir and showing the script file named `.command.sh`\u001b[K\n",
      "\u001b[K\n",
      " -- Check '.nextflow.log' file for details\u001b[39m\u001b[K\n",
      "\u001b[56A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2rqELz6LM3eIxm\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (2)\u001b[K\n",
      "[fd/d4a0e5] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[d9/629469] process > run_add_subpool_to_chromap_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[00/e3e10e] process > run_process_create_temp_sum... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[74/0c4d24] process > run_filter_aligned_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[07/637711] process > run_tabix_no_singleton (1)     [  0%] 0 of 1\u001b[K\n",
      "[ea/e47e99] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[59/f41ab9] process > run_calculate_tss_enrichmen... [  0%] 0 of 1 ✔\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "\u001b[31mERROR ~ Error executing process > 'run_calculate_tss_enrichment_snapatac2 (1)'\u001b[K\n",
      "\u001b[K\n",
      "Caused by:\u001b[K\n",
      "  Process `run_calculate_tss_enrichment_snapatac2 (1)` terminated with an error exit status (1)\u001b[K\n",
      "\u001b[K\n",
      "Command executed:\u001b[K\n",
      "\u001b[K\n",
      "  echo 'start run_calculate_tss_enrichment_snapatac2'\u001b[K\n",
      "  echo 'calculation_script is compute_tss_enrichment_snapatac2.py. check that it is: snapatac2'\u001b[K\n",
      "  echo 'no_singleton_bed_gz: no-singleton.bed.gz'\u001b[K\n",
      "  echo 'genes_gtf_gzip_file_out: genes.gtf.gz'\u001b[K\n",
      "  echo 'min_frag_cutoff: 10'\u001b[K\n",
      "  echo 'prefix: BMMC-single-donor'\u001b[K\n",
      "  echo 'output file: BMMC-single-donor.tss_enrichment_barcode_stats.tsv'\u001b[K\n",
      "  \u001b[K\n",
      "  python3 /usr/local/bin/compute_tss_enrichment_snapatac2.py no-singleton.bed.gz genes.gtf.gz \"BMMC-single-donor.tss_enrichment_barcode_stats.tsv\" 10\u001b[K\n",
      "  echo 'finished run_calculate_tss_enrichment_snapatac2'\u001b[K\n",
      "\u001b[K\n",
      "Command exit status:\u001b[K\n",
      "  1\u001b[K\n",
      "\u001b[K\n",
      "Command output:\u001b[K\n",
      "  start run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "  calculation_script is compute_tss_enrichment_snapatac2.py. check that it is: snapatac2\u001b[K\n",
      "  no_singleton_bed_gz: no-singleton.bed.gz\u001b[K\n",
      "  genes_gtf_gzip_file_out: genes.gtf.gz\u001b[K\n",
      "  min_frag_cutoff: 10\u001b[K\n",
      "  prefix: BMMC-single-donor\u001b[K\n",
      "  output file: BMMC-single-donor.tss_enrichment_barcode_stats.tsv\u001b[K\n",
      "  Loading data from fragment file...\u001b[K\n",
      "\u001b[K\n",
      "Command error:\u001b[K\n",
      "  Traceback (most recent call last):\u001b[K\n",
      "    File \"/usr/local/bin/compute_tss_enrichment_snapatac2.py\", line 47, in <module>\u001b[K\n",
      "      main()\u001b[K\n",
      "    File \"/usr/local/bin/compute_tss_enrichment_snapatac2.py\", line 30, in main\u001b[K\n",
      "      data = snap.pp.import_data(\u001b[K\n",
      "  TypeError: import_data() got an unexpected keyword argument 'chrom_sizes'\u001b[K\n",
      "\u001b[K\n",
      "Work dir:\u001b[K\n",
      "  gs://eila-nextflow-bucket/work/ATAC/59/f41ab91bda34b8bb6baca18bffdc90\u001b[K\n",
      "\u001b[K\n",
      "Tip: you can try to figure out what's wrong by changing to the process work dir and showing the script file named `.command.sh`\u001b[K\n",
      "\u001b[K\n",
      " -- Check '.nextflow.log' file for details\u001b[39m\u001b[K\n",
      "\u001b[56A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2rqELz6LM3eIxm\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (2)\u001b[K\n",
      "[fd/d4a0e5] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[d9/629469] process > run_add_subpool_to_chromap_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[00/e3e10e] process > run_process_create_temp_sum... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[74/0c4d24] process > run_filter_aligned_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[07/637711] process > run_tabix_no_singleton (1)     [  0%] 0 of 1\u001b[K\n",
      "[ea/e47e99] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[59/f41ab9] process > run_calculate_tss_enrichmen... [100%] 1 of 1, failed: 1 ✘\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "\u001b[31mERROR ~ Error executing process > 'run_calculate_tss_enrichment_snapatac2 (1)'\u001b[K\n",
      "\u001b[K\n",
      "Caused by:\u001b[K\n",
      "  Process `run_calculate_tss_enrichment_snapatac2 (1)` terminated with an error exit status (1)\u001b[K\n",
      "\u001b[K\n",
      "Command executed:\u001b[K\n",
      "\u001b[K\n",
      "  echo 'start run_calculate_tss_enrichment_snapatac2'\u001b[K\n",
      "  echo 'calculation_script is compute_tss_enrichment_snapatac2.py. check that it is: snapatac2'\u001b[K\n",
      "  echo 'no_singleton_bed_gz: no-singleton.bed.gz'\u001b[K\n",
      "  echo 'genes_gtf_gzip_file_out: genes.gtf.gz'\u001b[K\n",
      "  echo 'min_frag_cutoff: 10'\u001b[K\n",
      "  echo 'prefix: BMMC-single-donor'\u001b[K\n",
      "  echo 'output file: BMMC-single-donor.tss_enrichment_barcode_stats.tsv'\u001b[K\n",
      "  \u001b[K\n",
      "  python3 /usr/local/bin/compute_tss_enrichment_snapatac2.py no-singleton.bed.gz genes.gtf.gz \"BMMC-single-donor.tss_enrichment_barcode_stats.tsv\" 10\u001b[K\n",
      "  echo 'finished run_calculate_tss_enrichment_snapatac2'\u001b[K\n",
      "\u001b[K\n",
      "Command exit status:\u001b[K\n",
      "  1\u001b[K\n",
      "\u001b[K\n",
      "Command output:\u001b[K\n",
      "  start run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "  calculation_script is compute_tss_enrichment_snapatac2.py. check that it is: snapatac2\u001b[K\n",
      "  no_singleton_bed_gz: no-singleton.bed.gz\u001b[K\n",
      "  genes_gtf_gzip_file_out: genes.gtf.gz\u001b[K\n",
      "  min_frag_cutoff: 10\u001b[K\n",
      "  prefix: BMMC-single-donor\u001b[K\n",
      "  output file: BMMC-single-donor.tss_enrichment_barcode_stats.tsv\u001b[K\n",
      "  Loading data from fragment file...\u001b[K\n",
      "\u001b[K\n",
      "Command error:\u001b[K\n",
      "  Traceback (most recent call last):\u001b[K\n",
      "    File \"/usr/local/bin/compute_tss_enrichment_snapatac2.py\", line 47, in <module>\u001b[K\n",
      "      main()\u001b[K\n",
      "    File \"/usr/local/bin/compute_tss_enrichment_snapatac2.py\", line 30, in main\u001b[K\n",
      "      data = snap.pp.import_data(\u001b[K\n",
      "  TypeError: import_data() got an unexpected keyword argument 'chrom_sizes'\u001b[K\n",
      "\u001b[K\n",
      "Work dir:\u001b[K\n",
      "  gs://eila-nextflow-bucket/work/ATAC/59/f41ab91bda34b8bb6baca18bffdc90\u001b[K\n",
      "\u001b[K\n",
      "Tip: you can try to figure out what's wrong by changing to the process work dir and showing the script file named `.command.sh`\u001b[K\n",
      "\u001b[K\n",
      " -- Check '.nextflow.log' file for details\u001b[39m\u001b[K\n",
      "\u001b[56A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2rqELz6LM3eIxm\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (2)\u001b[K\n",
      "[fd/d4a0e5] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[d9/629469] process > run_add_subpool_to_chromap_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[00/e3e10e] process > run_process_create_temp_sum... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[74/0c4d24] process > run_filter_aligned_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[07/637711] process > run_tabix_no_singleton (1)     [  0%] 0 of 1\u001b[K\n",
      "[ea/e47e99] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[59/f41ab9] process > run_calculate_tss_enrichmen... [100%] 1 of 1, failed: 1 ✘\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "\u001b[31mERROR ~ Error executing process > 'run_calculate_tss_enrichment_snapatac2 (1)'\u001b[K\n",
      "\u001b[K\n",
      "Caused by:\u001b[K\n",
      "  Process `run_calculate_tss_enrichment_snapatac2 (1)` terminated with an error exit status (1)\u001b[K\n",
      "\u001b[K\n",
      "Command executed:\u001b[K\n",
      "\u001b[K\n",
      "  echo 'start run_calculate_tss_enrichment_snapatac2'\u001b[K\n",
      "  echo 'calculation_script is compute_tss_enrichment_snapatac2.py. check that it is: snapatac2'\u001b[K\n",
      "  echo 'no_singleton_bed_gz: no-singleton.bed.gz'\u001b[K\n",
      "  echo 'genes_gtf_gzip_file_out: genes.gtf.gz'\u001b[K\n",
      "  echo 'min_frag_cutoff: 10'\u001b[K\n",
      "  echo 'prefix: BMMC-single-donor'\u001b[K\n",
      "  echo 'output file: BMMC-single-donor.tss_enrichment_barcode_stats.tsv'\u001b[K\n",
      "  \u001b[K\n",
      "  python3 /usr/local/bin/compute_tss_enrichment_snapatac2.py no-singleton.bed.gz genes.gtf.gz \"BMMC-single-donor.tss_enrichment_barcode_stats.tsv\" 10\u001b[K\n",
      "  echo 'finished run_calculate_tss_enrichment_snapatac2'\u001b[K\n",
      "\u001b[K\n",
      "Command exit status:\u001b[K\n",
      "  1\u001b[K\n",
      "\u001b[K\n",
      "Command output:\u001b[K\n",
      "  start run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "  calculation_script is compute_tss_enrichment_snapatac2.py. check that it is: snapatac2\u001b[K\n",
      "  no_singleton_bed_gz: no-singleton.bed.gz\u001b[K\n",
      "  genes_gtf_gzip_file_out: genes.gtf.gz\u001b[K\n",
      "  min_frag_cutoff: 10\u001b[K\n",
      "  prefix: BMMC-single-donor\u001b[K\n",
      "  output file: BMMC-single-donor.tss_enrichment_barcode_stats.tsv\u001b[K\n",
      "  Loading data from fragment file...\u001b[K\n",
      "\u001b[K\n",
      "Command error:\u001b[K\n",
      "  Traceback (most recent call last):\u001b[K\n",
      "    File \"/usr/local/bin/compute_tss_enrichment_snapatac2.py\", line 47, in <module>\u001b[K\n",
      "      main()\u001b[K\n",
      "    File \"/usr/local/bin/compute_tss_enrichment_snapatac2.py\", line 30, in main\u001b[K\n",
      "      data = snap.pp.import_data(\u001b[K\n",
      "  TypeError: import_data() got an unexpected keyword argument 'chrom_sizes'\u001b[K\n",
      "\u001b[K\n",
      "Work dir:\u001b[K\n",
      "  gs://eila-nextflow-bucket/work/ATAC/59/f41ab91bda34b8bb6baca18bffdc90\u001b[K\n",
      "\u001b[K\n",
      "Tip: you can try to figure out what's wrong by changing to the process work dir and showing the script file named `.command.sh`\u001b[K\n",
      "\u001b[K\n",
      " -- Check '.nextflow.log' file for details\u001b[39m\u001b[K\n",
      "WARN: Killing running tasks (1)\n",
      "WARN: Tower request field `workflow.errorMessage` exceeds expected size | offending value: `Traceback (most recent call last):\n",
      "  File \"/usr/local/bin/compute_tss_enrichment_snapatac2.py\", line 47, in <module>\n",
      "    main()\n",
      "  File \"/usr/local/bin/compute_tss_enrichment_snapatac2.py\", line 30, in main\n",
      "    data = snap.pp.import_data(\n",
      "TypeError: import_data() got an unexpected keyword argument 'chrom_sizes'`, size: 311 (max: 255)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_atac.config -profile google_batch_profile -process.echo -with-tower -resume \n",
    "# -with-dag -resume\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_vm.config -profile google_batch_profile\n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_vm.config -profile google_batch_profile\n",
    "# !nextflow run nf_\n",
    "# \n",
    "# wf_fg_singlecell_atac.nf -c nextflow_vm_anvil.config -profile google_batch_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_local.config -profile google_batch_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_local.config -profile docker_profile\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 186,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Feb-10 02:14:49.406 [main] DEBUG nextflow.cli.Launcher - $> nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_atac.config -profile google_batch_profile -process.echo -with-tower -resume\n",
      "Feb-10 02:14:49.580 [main] INFO  nextflow.cli.CmdRun - N E X T F L O W  ~  version 23.10.1\n",
      "Feb-10 02:14:49.608 [main] DEBUG nextflow.plugin.PluginsFacade - Setting up plugin manager > mode=prod; embedded=false; plugins-dir=/home/eila/.nextflow/plugins; core-plugins: nf-amazon@2.1.4,nf-azure@1.3.3,nf-cloudcache@0.3.0,nf-codecommit@0.1.5,nf-console@1.0.6,nf-ga4gh@1.1.0,nf-google@1.8.3,nf-tower@1.6.3,nf-wave@1.0.1\n",
      "Feb-10 02:14:49.624 [main] INFO  o.pf4j.DefaultPluginStatusProvider - Enabled plugins: []\n",
      "Feb-10 02:14:49.625 [main] INFO  o.pf4j.DefaultPluginStatusProvider - Disabled plugins: []\n",
      "Feb-10 02:14:49.630 [main] INFO  org.pf4j.DefaultPluginManager - PF4J version 3.4.1 in 'deployment' mode\n",
      "Feb-10 02:14:49.646 [main] INFO  org.pf4j.AbstractPluginManager - No plugins\n",
      "Feb-10 02:14:49.673 [main] DEBUG nextflow.config.ConfigBuilder - User config file: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Feb-10 02:14:49.674 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Feb-10 02:14:49.707 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `google_batch_profile`\n",
      "Feb-10 02:14:52.607 [main] DEBUG nextflow.config.ConfigBuilder - Available config profiles: [google_batch_profile]\n",
      "Feb-10 02:14:52.659 [main] DEBUG nextflow.cli.CmdRun - Applied DSL=2 by global default\n",
      "Feb-10 02:14:52.687 [main] INFO  nextflow.cli.CmdRun - Launching `nf_wf_fg_singlecell_atac.nf` [adoring_pasteur] DSL2 - revision: 36a818f600\n",
      "Feb-10 02:14:52.688 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins default=[nf-google@1.8.3]\n",
      "Feb-10 02:14:52.689 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins resolved requirement=[nf-google@1.8.3, nf-tower@1.6.3, nf-wave@1.0.1]\n",
      "Feb-10 02:14:52.689 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-google version: 1.8.3\n",
      "Feb-10 02:14:52.708 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-google@1.8.3' resolved\n",
      "Feb-10 02:14:52.708 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-google@1.8.3'\n",
      "Feb-10 02:14:52.756 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-google@1.8.3\n",
      "Feb-10 02:14:52.756 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-tower version: 1.6.3\n",
      "Feb-10 02:14:52.759 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-tower@1.6.3' resolved\n",
      "Feb-10 02:14:52.759 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-tower@1.6.3'\n",
      "Feb-10 02:14:52.768 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-tower@1.6.3\n",
      "Feb-10 02:14:52.769 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-wave version: 1.0.1\n",
      "Feb-10 02:14:52.771 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-wave@1.0.1' resolved\n",
      "Feb-10 02:14:52.771 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-wave@1.0.1'\n",
      "Feb-10 02:14:52.779 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-wave@1.0.1\n",
      "Feb-10 02:14:52.798 [main] DEBUG n.secret.LocalSecretsProvider - Secrets store: /home/eila/.nextflow/secrets/store.json\n",
      "Feb-10 02:14:52.803 [main] DEBUG nextflow.secret.SecretsLoader - Discovered secrets providers: [nextflow.secret.LocalSecretsProvider@4e224df5] - activable => nextflow.secret.LocalSecretsProvider@4e224df5\n",
      "Feb-10 02:14:52.891 [main] DEBUG nextflow.Session - Session UUID: 0cbd4d64-b857-4807-b564-11778fa2d750\n",
      "Feb-10 02:14:52.891 [main] DEBUG nextflow.Session - Run name: adoring_pasteur\n",
      "Feb-10 02:14:52.892 [main] DEBUG nextflow.Session - Executor pool size: 8\n",
      "Feb-10 02:14:53.243 [main] DEBUG nextflow.file.FilePorter - File porter settings maxRetries=3; maxTransfers=50; pollTimeout=null\n",
      "Feb-10 02:14:53.249 [main] DEBUG nextflow.util.ThreadPoolBuilder - Creating thread pool 'FileTransfer' minSize=10; maxSize=24; workQueue=LinkedBlockingQueue[10000]; allowCoreThreadTimeout=false\n",
      "Feb-10 02:14:53.255 [main] DEBUG nextflow.config.ConfigBuilder - User config file: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Feb-10 02:14:53.256 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Feb-10 02:14:53.256 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `google_batch_profile`\n",
      "Feb-10 02:14:55.115 [main] DEBUG nextflow.config.ConfigBuilder - Available config profiles: [google_batch_profile]\n",
      "Feb-10 02:14:55.154 [main] DEBUG nextflow.cli.CmdRun - \n",
      "  Version: 23.10.1 build 5891\n",
      "  Created: 12-01-2024 22:01 UTC \n",
      "  System: Linux 5.10.0-27-cloud-amd64\n",
      "  Runtime: Groovy 3.0.19 on OpenJDK 64-Bit Server VM 11.0.1-internal+0-adhoc..src\n",
      "  Encoding: UTF-8 (UTF-8)\n",
      "  Process: 79335@eila-vm [10.128.0.63]\n",
      "  CPUs: 8 - Mem: 29.4 GB (16.3 GB) - Swap: 0 (0)\n",
      "Feb-10 02:14:55.204 [main] DEBUG nextflow.file.FileHelper - Can't check if specified path is NFS (1): gs://eila-nextflow-bucket/work/ATAC\n",
      "\n",
      "Feb-10 02:14:55.205 [main] DEBUG nextflow.Session - Work-dir: gs://eila-nextflow-bucket/work/ATAC [null]\n",
      "Feb-10 02:14:55.240 [main] DEBUG nextflow.executor.ExecutorFactory - Extension executors providers=[GoogleLifeSciencesExecutor, GoogleBatchExecutor]\n",
      "Feb-10 02:14:55.262 [main] DEBUG nextflow.Session - Observer factory: DefaultObserverFactory\n",
      "Feb-10 02:14:55.266 [main] DEBUG nextflow.Session - Observer factory: WaveFactory\n",
      "Feb-10 02:14:55.266 [main] DEBUG io.seqera.wave.plugin.WaveFactory - Detected Fusion enabled -- Enabling bundle project resources -- Disabling upload of remote bin directory\n",
      "Feb-10 02:14:55.295 [main] DEBUG i.s.wave.plugin.config.WaveConfig - Wave strategy not specified - using default: [container, dockerfile, conda, spack]\n",
      "Feb-10 02:14:55.306 [main] DEBUG io.seqera.wave.plugin.WaveClient - Wave config: WaveConfig(enabled:true, endpoint:https://wave.seqera.io, containerConfigUrl:[], tokensCacheMaxDuration:30m, condaOpts:CondaOpts(mambaImage=mambaorg/micromamba:1.5.1; basePackages=conda-forge::procps-ng, commands=null), spackOpts:SpackOpts(basePackages=null, commands=null), strategy:[container, dockerfile, conda, spack], bundleProjectResources:true, buildRepository:null, cacheRepository:null, retryOpts:RetryOpts(delay:450ms, maxDelay:1m 30s, maxAttempts:10, jitter:0.25), httpClientOpts:HttpOpts(), freezeMode:null)\n",
      "Feb-10 02:14:55.435 [main] DEBUG nextflow.Session - Observer factory: TowerFactory\n",
      "Feb-10 02:14:55.529 [main] DEBUG nextflow.cache.CacheFactory - Using Nextflow cache factory: nextflow.cache.DefaultCacheFactory\n",
      "Feb-10 02:14:55.542 [main] DEBUG nextflow.util.CustomThreadPool - Creating default thread pool > poolSize: 9; maxThreads: 1000\n",
      "Feb-10 02:14:55.669 [main] DEBUG nextflow.Session - Session start\n",
      "Feb-10 02:14:55.687 [main] DEBUG io.seqera.tower.plugin.TowerClient - Creating Tower observer -- endpoint=https://api.tower.nf; requestInterval=1s; aliveInterval=1m; maxRetries=5; backOffBase=3; backOffDelay=250\n",
      "Feb-10 02:14:56.693 [main] DEBUG nextflow.script.ScriptRunner - > Launching execution\n",
      "Feb-10 02:14:58.076 [main] INFO  nextflow.script.BaseScript - /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/wdl_atac_pipeline/vm_atac_fastqs_files.tsv\n",
      "Feb-10 02:14:58.077 [main] INFO  nextflow.script.BaseScript - null\n",
      "Feb-10 02:14:58.217 [main] INFO  nextflow.script.BaseScript - after genome_fasta_ch\n",
      "Feb-10 02:14:58.283 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:zcat` matches labels `zcat` for process with name run_zcat\n",
      "Feb-10 02:14:58.290 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-10 02:14:58.290 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-10 02:14:58.293 [main] DEBUG nextflow.executor.Executor - [warm up] executor > google-batch (fusion enabled)\n",
      "Feb-10 02:14:58.297 [main] DEBUG n.processor.TaskPollingMonitor - Creating task monitor for executor 'google-batch' > capacity: 1000; pollInterval: 10s; dumpInterval: 5m \n",
      "Feb-10 02:14:58.300 [main] DEBUG n.processor.TaskPollingMonitor - >>> barrier register (monitor: google-batch)\n",
      "Feb-10 02:14:58.306 [main] DEBUG nextflow.cloud.google.GoogleOpts - Google auth via application credentials file: /home/eila/git_igvf/atomic-workflows/nextflow/hai-gcp-genomic-beb9207a40b8.json\n",
      "Feb-10 02:14:58.312 [main] DEBUG n.c.google.batch.GoogleBatchExecutor - [GOOGLE BATCH] Executor config=BatchConfig[googleOpts=GoogleOpts(projectId:hai-gcp-genomic, credsFile:/home/eila/git_igvf/atomic-workflows/nextflow/hai-gcp-genomic-beb9207a40b8.json, location:us-central1, enableRequesterPaysBuckets:false, httpConnectTimeout:1m, httpReadTimeout:1m, credentials:ServiceAccountCredentials{clientId=111364348320962814542, clientEmail=eila-batch@hai-gcp-genomic.iam.gserviceaccount.com, privateKeyId=beb9207a40b8b2085c7f48c3fcf8c2342180d471, transportFactoryClassName=com.google.auth.oauth2.OAuth2Utils$DefaultHttpTransportFactory, tokenServerUri=https://oauth2.googleapis.com/token, scopes=[https://www.googleapis.com/auth/cloud-platform], defaultScopes=[], serviceAccountUser=null, quotaProjectId=null, lifetime=3600, useJwtAccessWithScope=false, defaultRetriesEnabled=true})\n",
      "Feb-10 02:14:58.332 [main] DEBUG n.c.google.batch.client.BatchClient - [GOOGLE BATCH] Creating service client with config credentials\n",
      "Feb-10 02:14:58.969 [main] INFO  nextflow.script.BaseScript - after run_zcat \n",
      "Feb-10 02:14:58.972 [main] INFO  nextflow.script.BaseScript - after load chrompa mapping debug\n",
      "Feb-10 02:14:58.985 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:pool_fragments` matches labels `pool_fragments` for process with name run_add_subpool_to_chromap_output\n",
      "Feb-10 02:14:58.988 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-10 02:14:58.988 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-10 02:14:58.993 [main] INFO  nextflow.script.BaseScript - before run_add_subpool_prefix_to_summary_file\n",
      "Feb-10 02:14:58.999 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:create_temp_summary_dict` matches labels `create_temp_summary_dict` for process with name run_process_create_temp_summary_dict\n",
      "Feb-10 02:14:59.001 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-10 02:14:59.001 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-10 02:14:59.007 [main] INFO  nextflow.script.BaseScript - after run_add_subpool_prefix_to_summary_file\n",
      "Feb-10 02:14:59.012 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:pool_filter` matches labels `pool_filter` for process with name run_filter_aligned_fragments\n",
      "Feb-10 02:14:59.014 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-10 02:14:59.014 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-10 02:14:59.022 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:tabix_singelston` matches labels `tabix_singelston` for process with name run_tabix_no_singleton\n",
      "Feb-10 02:14:59.023 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-10 02:14:59.024 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-10 02:14:59.027 [main] INFO  nextflow.script.BaseScript - before call run_calculate_tss_enrichment_bulk\n",
      "Feb-10 02:14:59.031 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:gzip` matches labels `gzip` for process with name run_gzip_on_genes_gtf\n",
      "Feb-10 02:14:59.032 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-10 02:14:59.032 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-10 02:14:59.035 [main] INFO  nextflow.script.BaseScript - after run_gzip_on_genes_gtf\n",
      "Feb-10 02:14:59.035 [main] INFO  nextflow.script.BaseScript - before call run_calculate_tss_enrichment_snapatac2\n",
      "Feb-10 02:14:59.039 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:tss_snapatac2` matches labels `tss_snapatac2` for process with name run_calculate_tss_enrichment_snapatac2\n",
      "Feb-10 02:14:59.041 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-10 02:14:59.041 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-10 02:14:59.048 [main] INFO  nextflow.script.BaseScript - after call run_calculate_tss_enrichment_snapatac2\n",
      "Feb-10 02:14:59.048 [main] INFO  nextflow.script.BaseScript - before run_barcode_metadata_stats\n",
      "Feb-10 02:14:59.052 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-10 02:14:59.052 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-10 02:14:59.056 [main] INFO  nextflow.script.BaseScript - after run_barcode_metadata_stats\n",
      "Feb-10 02:14:59.057 [main] DEBUG nextflow.Session - Workflow process names [dsl2]: run_bgzip_on_chromap_fragments_output, run_seqspec_print, run_calculate_tss_enrichment_snapatac2, run_barcode_metadata_stats, run_tabix_chromap, run_calculate_tss_enrichment_bulk, run_generate_insert_size_plot, run_seqspec_test, run_tabix_no_singleton, run_chromap_map_to_idx, run_atac_merge_barcode_metadata, run_filter_aligned_fragments, run_seqspec_modify_rna, run_atac_barcode_rank_plot, run_gzip_on_genes_gtf, run_whitelist_gunzip, run_merge_chromap_logs, run_create_chromap_idx, run_add_subpool_to_chromap_output, run_process_create_temp_summary_dict, run_generate_insert_size_histogram_data, run_zcat, run_seqspec_check, run_seqspec_index_rna_kb, run_seqspec_modify_atac, run_chromap_test, run_tabix_filtered_fragments, run_downloadFiles\n",
      "Feb-10 02:14:59.380 [main] INFO  io.seqera.tower.plugin.TowerClient - Monitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2rqELz6LM3eIxm\n",
      "Feb-10 02:14:59.381 [main] DEBUG nextflow.Session - Igniting dataflow network (16)\n",
      "Feb-10 02:14:59.390 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_zcat\n",
      "Feb-10 02:14:59.391 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_add_subpool_to_chromap_output\n",
      "Feb-10 02:14:59.393 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_process_create_temp_summary_dict\n",
      "Feb-10 02:14:59.393 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_filter_aligned_fragments\n",
      "Feb-10 02:14:59.394 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_tabix_no_singleton\n",
      "Feb-10 02:14:59.397 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_gzip_on_genes_gtf\n",
      "Feb-10 02:14:59.401 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_calculate_tss_enrichment_snapatac2\n",
      "Feb-10 02:14:59.401 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_barcode_metadata_stats\n",
      "Feb-10 02:14:59.410 [main] DEBUG nextflow.script.ScriptRunner - Parsed script files:\n",
      "  Script_e43ef4fe24ad2c21: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_atac_generate_insert_size_plot.nf\n",
      "  Script_fec0f9da3ae38823: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_tss.nf\n",
      "  Script_e00ad044dff0ec57: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_barcode_dictionary_filter.nf\n",
      "  Script_cb3defa7ac67fc21: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nf_wf_fg_singlecell_atac.nf\n",
      "  Script_985c81f2bb4fd175: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_atac_metadata_annoation_and_plot.nf\n",
      "  Script_9d4639c8cd883c70: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_download_url_files.nf\n",
      "  Script_bc539eb5853c51b8: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_gunzip.nf\n",
      "  Script_178b08c698b6775a: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_tabix.nf\n",
      "  Script_ac710cb21b72d60a: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_bgzip.nf\n",
      "  Script_c843d085a2409aa5: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_subpool_summary_dict.nf\n",
      "  Script_e112646bcdb899f0: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_seqspec_utils.nf\n",
      "  Script_8081eedda99eaa43: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_merge_logs.nf\n",
      "  Script_48127624ae4a6640: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_zcat.nf\n",
      "  Script_9abd93a618947195: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_chromap_utils.nf\n",
      "  Script_7f43a272cfe3ae1b: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_generate_merged_barcode_metadata.nf\n",
      "  Script_34d6d3540f17eeb9: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_gzip.nf\n",
      "Feb-10 02:14:59.411 [main] DEBUG nextflow.script.ScriptRunner - > Awaiting termination \n",
      "Feb-10 02:14:59.411 [main] DEBUG nextflow.Session - Session await\n",
      "Feb-10 02:15:00.673 [FileTransfer-1] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/wdl_atac_pipeline/genes.gtf at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/41/45671d165921b1e3a885b4ea55909e/genes.gtf\n",
      "Feb-10 02:15:00.696 [Actor Thread 5] DEBUG i.s.wave.plugin.config.WaveConfig - Wave strategy not specified - using default: [container, dockerfile, conda, spack]\n",
      "Feb-10 02:15:00.697 [Actor Thread 5] DEBUG io.seqera.wave.plugin.WaveClient - Wave config: WaveConfig(enabled:true, endpoint:https://wave.seqera.io, containerConfigUrl:[], tokensCacheMaxDuration:30m, condaOpts:CondaOpts(mambaImage=mambaorg/micromamba:1.5.1; basePackages=conda-forge::procps-ng, commands=null), spackOpts:SpackOpts(basePackages=null, commands=null), strategy:[container, dockerfile, conda, spack], bundleProjectResources:true, buildRepository:null, cacheRepository:null, retryOpts:RetryOpts(delay:450ms, maxDelay:1m 30s, maxAttempts:10, jitter:0.25), httpClientOpts:HttpOpts(), freezeMode:null)\n",
      "Feb-10 02:15:00.710 [Actor Thread 5] DEBUG io.seqera.wave.plugin.WaveClient - Wave assets bundle bin resources: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/bin\n",
      "Feb-10 02:15:00.803 [Actor Thread 5] DEBUG io.seqera.wave.plugin.WaveClient - Wave request container config: https://fusionfs.seqera.io/releases/v2.2-amd64.json\n",
      "Feb-10 02:15:00.919 [FileTransfer-2] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/wdl_atac_pipeline/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\n",
      "Feb-10 02:15:01.001 [Actor Thread 5] DEBUG io.seqera.wave.plugin.WaveClient - Wave container config response: [200] {\n",
      "  \"entrypoint\": [ \"/usr/bin/fusion\" ],\n",
      "  \"layers\": [\n",
      "    {\n",
      "      \"location\": \"https://fusionfs.seqera.io/releases/pkg/2/2/12/fusion-amd64.tar.gz\",\n",
      "      \"gzipDigest\": \"sha256:6ccfd8f431db2874e82e27d61b6d31b95d7dddce7e810c5f8c95438a0d03c066\",\n",
      "      \"gzipSize\": 30262575,\n",
      "      \"tarDigest\": \"sha256:60391bebe0cc5d48b0875854f0c93c0d0adc25b5007fdb1b2efc2a431cfeef6e\",\n",
      "      \"skipHashing\": true\n",
      "    }\n",
      "  ]\n",
      "}\n",
      "\n",
      "Feb-10 02:15:01.109 [Actor Thread 5] DEBUG io.seqera.wave.plugin.WaveClient - Wave request: https://wave.seqera.io/container-token; attempt=1 - request: SubmitContainerTokenRequest(towerAccessToken:eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz, towerRefreshToken:null, towerWorkspaceId:247507587970818, towerEndpoint:https://api.tower.nf, workflowId:null, containerImage:eilalan/utils_env:latest, containerFile:null, containerConfig:ContainerConfig(entrypoint:[/usr/bin/fusion], cmd:null, env:null, workingDir:null, layers:[ContainerLayer[location=data:H4sIAAAAAAAA/+19bXvb...; tarDigest=sha256:955064f0396e4c473c3dcf22b2362650e6fcb5018eefc0b7a68d286ed4e6eca8; gzipDigest=sha256:5f15b4f2d4921bd9b6a36380ff9c6d3512cd9d9983818192c0844710716d1f4f; gzipSize=15987], ContainerLayer[location=https://fusionfs.seqera.io/releases/pkg/2/2/12/fusion-amd64.tar.gz; tarDigest=sha256:60391bebe0cc5d48b0875854f0c93c0d0adc25b5007fdb1b2efc2a431cfeef6e; gzipDigest=sha256:6ccfd8f431db2874e82e27d61b6d31b95d7dddce7e810c5f8c95438a0d03c066; gzipSize=30262575]]), condaFile:null, spackFile:null, containerPlatform:linux/amd64, buildRepository:null, cacheRepository:null, timestamp:2024-02-10T02:15:01.107712Z, fingerprint:4113070522c1d4c0d97161c06457f919, freeze:false, format:null, dryRun:false)\n",
      "Feb-10 02:15:01.781 [Actor Thread 5] DEBUG io.seqera.wave.plugin.WaveClient - Wave response: statusCode=200; body={\"containerToken\":\"7ac708515ba9\",\"targetImage\":\"wave.seqera.io/wt/7ac708515ba9/eilalan/utils_env:latest\",\"expiration\":\"2024-02-11T14:15:01.728715657Z\",\"containerImage\":\"docker.io/eilalan/utils_env:latest\"}\n",
      "Feb-10 02:15:02.477 [Actor Thread 5] INFO  nextflow.processor.TaskProcessor - [ea/e47e99] Cached process > run_gzip_on_genes_gtf (1)\n",
      "Feb-10 02:15:02.625 [Actor Thread 1] INFO  nextflow.processor.TaskProcessor - [fd/d4a0e5] Cached process > run_zcat\n",
      "Feb-10 02:15:04.798 [FileTransfer-4] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/82/1c02896062d7ad9bee48b153038b02/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\n",
      "Feb-10 02:15:04.799 [FileTransfer-3] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/4a/8d5aa2caf4ba953973eb6ceb0209c1/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\n",
      "Feb-10 02:15:04.805 [FileTransfer-5] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/3b/378e6650d13f6cf7227c8199100fca/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\n",
      "Feb-10 02:15:04.811 [FileTransfer-7] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/67/0b839068fabf5a437638652ae82543/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\n",
      "Feb-10 02:15:04.812 [FileTransfer-2] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/test_atac_pipeline/sai_192_whitelist.txt at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/32/f784fb3f3859ff8de353857c4c015f/sai_192_whitelist.txt\n",
      "Feb-10 02:15:04.814 [FileTransfer-9] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/wdl_atac_pipeline/BMMC-single-donor.atac.align.barcode.summary.csv at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/e5/4f78cfd835463ff7b54feb664984f5/BMMC-single-donor.atac.align.barcode.summary.csv\n",
      "Feb-10 02:15:04.815 [FileTransfer-1] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/test_atac_pipeline/na.na at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/89/c1bcbc8e5f5646ebd09ee0ce3faefe/na.na\n",
      "Feb-10 02:15:04.828 [FileTransfer-10] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/de/e2b4899f50b4c0a726cbaade1c86a9/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\n",
      "Feb-10 02:15:04.961 [FileTransfer-4] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/test_atac_pipeline/na1.na at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/ab/3e66243b0d5cb8425a9dcfe650eb5c/na1.na\n",
      "Feb-10 02:15:04.973 [FileTransfer-6] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/9a/495e305984d9974332346f8cc35a95/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\n",
      "Feb-10 02:15:04.973 [FileTransfer-8] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/wdl_atac_pipeline/ATAC_BMMC-single-donor.atac.filter.fragments.tsv at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/ac/117d355dff34bab351800c58734a87/ATAC_BMMC-single-donor.atac.filter.fragments.tsv\n",
      "Feb-10 02:15:06.497 [Actor Thread 9] INFO  nextflow.processor.TaskProcessor - [00/e3e10e] Cached process > run_process_create_temp_summary_dict (1)\n",
      "Feb-10 02:15:06.676 [FileTransfer-3] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/wdl_atac_pipeline/BMMC-single-donor.atac.filter.fragments.hg38.tsv.gz at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/c0/4421a4507e67e2b73fc798231514eb/BMMC-single-donor.atac.filter.fragments.hg38.tsv.gz\n",
      "Feb-10 02:15:06.788 [Actor Thread 3] INFO  nextflow.processor.TaskProcessor - [d9/629469] Cached process > run_add_subpool_to_chromap_output (1)\n",
      "Feb-10 02:15:07.087 [Actor Thread 6] INFO  nextflow.processor.TaskProcessor - [74/0c4d24] Cached process > run_filter_aligned_fragments (1)\n",
      "Feb-10 02:15:07.124 [Actor Thread 9] DEBUG io.seqera.wave.plugin.WaveClient - Wave request: https://wave.seqera.io/container-token; attempt=1 - request: SubmitContainerTokenRequest(towerAccessToken:eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz, towerRefreshToken:null, towerWorkspaceId:247507587970818, towerEndpoint:https://api.tower.nf, workflowId:null, containerImage:eilalan/tss_env:latest, containerFile:null, containerConfig:ContainerConfig(entrypoint:[/usr/bin/fusion], cmd:null, env:null, workingDir:null, layers:[ContainerLayer[location=data:H4sIAAAAAAAA/+19bXvb...; tarDigest=sha256:955064f0396e4c473c3dcf22b2362650e6fcb5018eefc0b7a68d286ed4e6eca8; gzipDigest=sha256:5f15b4f2d4921bd9b6a36380ff9c6d3512cd9d9983818192c0844710716d1f4f; gzipSize=15987], ContainerLayer[location=https://fusionfs.seqera.io/releases/pkg/2/2/12/fusion-amd64.tar.gz; tarDigest=sha256:60391bebe0cc5d48b0875854f0c93c0d0adc25b5007fdb1b2efc2a431cfeef6e; gzipDigest=sha256:6ccfd8f431db2874e82e27d61b6d31b95d7dddce7e810c5f8c95438a0d03c066; gzipSize=30262575]]), condaFile:null, spackFile:null, containerPlatform:linux/amd64, buildRepository:null, cacheRepository:null, timestamp:2024-02-10T02:15:07.123946Z, fingerprint:4d449f0b740081432914bb2fcbd16911, freeze:false, format:null, dryRun:false)\n",
      "Feb-10 02:15:07.436 [Actor Thread 9] DEBUG io.seqera.wave.plugin.WaveClient - Wave response: statusCode=200; body={\"containerToken\":\"4d6005434819\",\"targetImage\":\"wave.seqera.io/wt/4d6005434819/eilalan/tss_env:latest\",\"expiration\":\"2024-02-11T14:15:07.383396897Z\",\"containerImage\":\"docker.io/eilalan/tss_env:latest\"}\n",
      "Feb-10 02:15:07.929 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_tabix_no_singleton (1)` - adding local volume as fusion scratch: DiskResource(request:375 GB, type:local-ssd)\n",
      "Feb-10 02:15:09.383 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_tabix_no_singleton (1)` submitted > job=nf-07637711-1707531307390; uid=nf-07637711-170753-39291601-e475-43640; work-dir=gs://eila-nextflow-bucket/work/ATAC/07/6377118a300a689a715bf2cd8a81dc\n",
      "Feb-10 02:15:09.383 [Task submitter] INFO  nextflow.Session - [07/637711] Submitted process > run_tabix_no_singleton (1)\n",
      "Feb-10 02:15:09.904 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_calculate_tss_enrichment_snapatac2 (1)` - adding local volume as fusion scratch: DiskResource(request:375 GB, type:local-ssd)\n",
      "Feb-10 02:15:10.148 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_calculate_tss_enrichment_snapatac2 (1)` submitted > job=nf-59f41ab9-1707531308792; uid=nf-59f41ab9-170753-d8e275f6-4904-4f200; work-dir=gs://eila-nextflow-bucket/work/ATAC/59/f41ab91bda34b8bb6baca18bffdc90\n",
      "Feb-10 02:15:10.149 [Task submitter] INFO  nextflow.Session - [59/f41ab9] Submitted process > run_calculate_tss_enrichment_snapatac2 (1)\n",
      "Feb-10 02:18:48.385 [Task monitor] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_calculate_tss_enrichment_snapatac2 (1)` - terminated job=nf-59f41ab9-1707531308792; state=FAILED\n",
      "Feb-10 02:18:49.524 [Task monitor] DEBUG n.processor.TaskPollingMonitor - Task completed > TaskHandler[id: 6; name: run_calculate_tss_enrichment_snapatac2 (1); status: COMPLETED; exit: 1; error: -; workDir: gs://eila-nextflow-bucket/work/ATAC/59/f41ab91bda34b8bb6baca18bffdc90]\n",
      "Feb-10 02:18:49.530 [Task monitor] DEBUG nextflow.processor.TaskProcessor - Handling unexpected condition for\n",
      "  task: name=run_calculate_tss_enrichment_snapatac2 (1); work-dir=gs://eila-nextflow-bucket/work/ATAC/59/f41ab91bda34b8bb6baca18bffdc90\n",
      "  error [nextflow.exception.ProcessFailedException]: Process `run_calculate_tss_enrichment_snapatac2 (1)` terminated with an error exit status (1)\n",
      "Feb-10 02:18:50.014 [Task monitor] ERROR nextflow.processor.TaskProcessor - Error executing process > 'run_calculate_tss_enrichment_snapatac2 (1)'\n",
      "\n",
      "Caused by:\n",
      "  Process `run_calculate_tss_enrichment_snapatac2 (1)` terminated with an error exit status (1)\n",
      "\n",
      "Command executed:\n",
      "\n",
      "  echo 'start run_calculate_tss_enrichment_snapatac2'\n",
      "  echo 'calculation_script is compute_tss_enrichment_snapatac2.py. check that it is: snapatac2'\n",
      "  echo 'no_singleton_bed_gz: no-singleton.bed.gz'\n",
      "  echo 'genes_gtf_gzip_file_out: genes.gtf.gz'\n",
      "  echo 'min_frag_cutoff: 10'\n",
      "  echo 'prefix: BMMC-single-donor'\n",
      "  echo 'output file: BMMC-single-donor.tss_enrichment_barcode_stats.tsv'\n",
      "  \n",
      "  python3 /usr/local/bin/compute_tss_enrichment_snapatac2.py no-singleton.bed.gz genes.gtf.gz \"BMMC-single-donor.tss_enrichment_barcode_stats.tsv\" 10\n",
      "  echo 'finished run_calculate_tss_enrichment_snapatac2'\n",
      "\n",
      "Command exit status:\n",
      "  1\n",
      "\n",
      "Command output:\n",
      "  start run_calculate_tss_enrichment_snapatac2\n",
      "  calculation_script is compute_tss_enrichment_snapatac2.py. check that it is: snapatac2\n",
      "  no_singleton_bed_gz: no-singleton.bed.gz\n",
      "  genes_gtf_gzip_file_out: genes.gtf.gz\n",
      "  min_frag_cutoff: 10\n",
      "  prefix: BMMC-single-donor\n",
      "  output file: BMMC-single-donor.tss_enrichment_barcode_stats.tsv\n",
      "  Loading data from fragment file...\n",
      "\n",
      "Command error:\n",
      "  Traceback (most recent call last):\n",
      "    File \"/usr/local/bin/compute_tss_enrichment_snapatac2.py\", line 47, in <module>\n",
      "      main()\n",
      "    File \"/usr/local/bin/compute_tss_enrichment_snapatac2.py\", line 30, in main\n",
      "      data = snap.pp.import_data(\n",
      "  TypeError: import_data() got an unexpected keyword argument 'chrom_sizes'\n",
      "\n",
      "Work dir:\n",
      "  gs://eila-nextflow-bucket/work/ATAC/59/f41ab91bda34b8bb6baca18bffdc90\n",
      "\n",
      "Tip: you can try to figure out what's wrong by changing to the process work dir and showing the script file named `.command.sh`\n",
      "Feb-10 02:18:50.244 [Task monitor] DEBUG nextflow.Session - Session aborted -- Cause: Process `run_calculate_tss_enrichment_snapatac2 (1)` terminated with an error exit status (1)\n",
      "Feb-10 02:18:50.476 [main] DEBUG nextflow.Session - Session await > all processes finished\n",
      "Feb-10 02:18:50.477 [main] DEBUG nextflow.Session - Session await > all barriers passed\n",
      "Feb-10 02:18:50.478 [Task monitor] DEBUG n.processor.TaskPollingMonitor - <<< barrier arrives (monitor: google-batch) - terminating tasks monitor poll loop\n",
      "Feb-10 02:18:50.707 [main] WARN  n.processor.TaskPollingMonitor - Killing running tasks (1)\n",
      "Feb-10 02:18:50.724 [main] DEBUG n.trace.WorkflowStatsObserver - Workflow completed > WorkflowStats[succeededCount=0; failedCount=1; ignoredCount=0; cachedCount=5; pendingCount=0; submittedCount=0; runningCount=0; retriesCount=0; abortedCount=1; succeedDuration=0ms; failedDuration=1.1s; cachedDuration=8m 58s;loadCpus=0; loadMemory=0; peakRunning=2; peakCpus=2; peakMemory=0; ]\n",
      "Feb-10 02:18:50.906 [main] WARN  i.s.tower.plugin.TowerJsonGenerator - Tower request field `workflow.errorMessage` exceeds expected size | offending value: `Traceback (most recent call last):\n",
      "  File \"/usr/local/bin/compute_tss_enrichment_snapatac2.py\", line 47, in <module>\n",
      "    main()\n",
      "  File \"/usr/local/bin/compute_tss_enrichment_snapatac2.py\", line 30, in main\n",
      "    data = snap.pp.import_data(\n",
      "TypeError: import_data() got an unexpected keyword argument 'chrom_sizes'`, size: 311 (max: 255)\n",
      "Feb-10 02:18:51.273 [main] DEBUG nextflow.cache.CacheDB - Closing CacheDB done\n",
      "Feb-10 02:18:51.273 [main] INFO  org.pf4j.AbstractPluginManager - Stop plugin 'nf-wave@1.0.1'\n",
      "Feb-10 02:18:51.274 [main] DEBUG nextflow.plugin.BasePlugin - Plugin stopped nf-wave\n",
      "Feb-10 02:18:51.274 [main] INFO  org.pf4j.AbstractPluginManager - Stop plugin 'nf-tower@1.6.3'\n",
      "Feb-10 02:18:51.274 [main] DEBUG nextflow.plugin.BasePlugin - Plugin stopped nf-tower\n",
      "Feb-10 02:18:51.274 [main] INFO  org.pf4j.AbstractPluginManager - Stop plugin 'nf-google@1.8.3'\n",
      "Feb-10 02:18:51.274 [main] DEBUG nextflow.plugin.BasePlugin - Plugin stopped nf-google\n",
      "Feb-10 02:18:51.294 [main] DEBUG nextflow.script.ScriptRunner - > Execution complete -- Goodbye\n"
     ]
    }
   ],
   "source": [
    "!cat .nextflow.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !cp ../../nf_data/test_atac_pipeline/test_atac_pipeline/* ../../nf_data/test_atac_pipeline/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac\n"
     ]
    }
   ],
   "source": [
    "# !gsutil cp gs://eila-nextflow-bucket/work/ATAC/b3/94bc73c6d43de41d17d75cac4fbda1/no-singleton.bed.gz.tbi /home/eila/git_igvf/downloads/no-singleton.bed.gz.tbi\n",
    "\n",
    "!pwd\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !cp ../../nf_data/test_atac_pipeline/hg38.refGene.TSS.bed ../../nf_data/wdl_atac_pipeline/\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
