{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# start docker deamon in terminal\n",
    "# !sudo docker login\n",
    "# OR - becuase the jupyther with allow root\n",
    "# !docker login"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "TODO:\n",
    "\n",
    "when running with cloud. \n",
    "if you dont need to keep the execution logs, change the cloud settings to keep it for a short while\n",
    "Check that the cache if you need to set the cache to false (especially in plot process)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/hai-gcp-genomic-beb9207a40b8.json\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "json_path = os.path.join(os.path.dirname(os.getcwd()),\"hai-gcp-genomic-beb9207a40b8.json\")\n",
    "!ls $json_path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Activated service account credentials for: [eila-batch@hai-gcp-genomic.iam.gserviceaccount.com]\n"
     ]
    }
   ],
   "source": [
    "# # in the command line. same env\n",
    "import os\n",
    "json_path = os.path.join(os.path.dirname(os.getcwd()),\"hai-gcp-genomic-beb9207a40b8.json\")\n",
    "os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = json_path\n",
    "!gcloud auth activate-service-account eila-batch@hai-gcp-genomic.iam.gserviceaccount.com --key-file=$GOOGLE_APPLICATION_CREDENTIALS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                  Credentialed Accounts\n",
      "ACTIVE  ACCOUNT\n",
      "*       eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\n",
      "        eila@stanford.edu\n",
      "\n",
      "To set the active account, run:\n",
      "    $ gcloud config set account `ACCOUNT`\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'hai-gcp-genomic'"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "!gcloud auth list\n",
    "project_name = !gcloud config get-value project\n",
    "project_name = project_name[0]\n",
    "project_name\n",
    "# gcloud config set account\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "os.environ[\"TOWER_ACCESS_TOKEN\"] = 'eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz'\n",
    "os.environ[\"TOWER_WORKSPACE_ID\"]='247507587970818'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz\n",
      "247507587970818\n"
     ]
    }
   ],
   "source": [
    "!echo $TOWER_ACCESS_TOKEN\n",
    "!echo $TOWER_WORKSPACE_ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/bin'"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# !chmod +x ./bin/scrna_qc_metrics.py\n",
    "import os\n",
    "bin_fld = os.path.join(os.getcwd(),\"bin\")\n",
    "bin_fld"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chmod: cannot access '/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/bin/atac_barcode_and_subpool.sh': No such file or directory\n"
     ]
    }
   ],
   "source": [
    "!chmod +x {os.path.join(bin_fld, \"scrna_plot_atac_qc_metrics.R\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"run_tabix.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"plot_insert_size_hist.py\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"generate_barcode_metadata.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"create_sample_filtered_fragment_file.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"compute_tss_enrichment_bulk.py\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"barcode_rank_functions.R\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"atac_barcode_and_subpool.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"plot_insert_size_hist.py\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"compute_tss_enrichment_snapatac2.py\")}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# # TODO: For a new user: run these commands with your service account for the execution. I am using eila-batch@...\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/iam.serviceAccountUser\"\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/batch.jobsEditor\"\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/logging.viewer\"\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !nextflow -self-update"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  Version: 23.10.1 build 5891\n",
      "  Created: 12-01-2024 22:01 UTC (14:01 PDT)\n",
      "  System: Mac OS X 14.3\n",
      "  Runtime: Groovy 3.0.19 on OpenJDK 64-Bit Server VM 17.0.8+7-LTS\n",
      "  Encoding: UTF-8 (UTF-8)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!nextflow info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "\n",
    "# Get the current working directory\n",
    "current_directory = os.getcwd()\n",
    "\n",
    "# Navigate two folders above\n",
    "two_folders_above = os.path.abspath(os.path.join(current_directory, '..', '..'))\n",
    "\n",
    "nf_data_base_path = os.path.join(two_folders_above,'nf_data')\n",
    "# Print the result\n",
    "print(nf_data_base_path)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data'"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nf_data_base_path\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "nf_data_project=\"test_atac_pipeline\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/local_atac_fastqs_files.tsv'"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.path.exists(os.path.join(nf_data_base_path,nf_data_project,\"local_atac_fastqs_files.tsv\"))\n",
    "os.path.join(nf_data_base_path,nf_data_project,\"local_atac_fastqs_files.tsv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "\n",
    "# Write the configuration file\n",
    "with open(\"nextflow_atac.config\", \"w\") as config_file:\n",
    "    config_file.write(f\"\"\"\n",
    "manifest {{\n",
    "  description = 'A nexflow pipeline for fg single cell atac' \n",
    "  mainScript = 'nf_wf_fg_singlecell_atac.nf'\n",
    "}}\n",
    "\n",
    "profiles {{\n",
    "  google_batch_profile {{\n",
    "    workDir = 'gs://eila-nextflow-bucket/work/ATAC'\n",
    "    includeConfig 'conf/docker-google-batch.config'\n",
    "    docker.enabled = true\n",
    "    wave.enabled = true\n",
    "    fusion.enabled = true\n",
    "    process.scratch = false\n",
    "    executor = 'google-batch'\n",
    "    docker.envWhitelist=['ENV_SYNAPSE_TOKEN','TOWER_ACCESS_TOKEN']\n",
    "    google.project = '{project_name}'\n",
    "    google.location = 'us-central1'\n",
    "    google.batch.spot = true\n",
    "    google.batch.serviceAccountEmail='eila-batch@hai-gcp-genomic.iam.gserviceaccount.com'\n",
    "  }}\n",
    "}}\n",
    "\n",
    "// Define ENV_SYNAPSE_TOKEN as a parameter\n",
    "params.FASTQS_SPEC_CH = '{nf_data_base_path}/{nf_data_project}/local_atac_fastqs_files.tsv'\n",
    "// CHROMAP\n",
    "// 6a\n",
    "params.CHROMAP_IDX='gs://eila-nextflow-bucket/test_atac_pipeline/ref.index'\n",
    "// 6b:\n",
    "// FASTA file\n",
    "params.CHROMAP_READ_LENGTH = 2000\n",
    "params.CHROMAP_BC_ERROR_THRESHOLD = 2\n",
    "params.CHROMAP_BC_PROBABILITY_THRESHOLD = 0.9\n",
    "params.CHROMAP_READ_FORMAT = 'bc:0:-1,r1:0:-1,r2:0:-1'\n",
    "params.CHROMAP_DROP_REPETITIVE_READS = 4\n",
    "params.CHROMAP_QUALITY_THRESHOLD = 0\n",
    "\n",
    "// CHROMAP DEBUG: load output files \n",
    "params.CHROMAP_GENOME_REFERENCE_FASTA='{nf_data_base_path}/{nf_data_project}/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz'\n",
    "params.chromap_filter_fragments_tsv ='{nf_data_base_path}/{nf_data_project}/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.atac.filter.fragments.tsv'\n",
    "params.barcode_summary_csv ='{nf_data_base_path}/{nf_data_project}/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.atac.align.barcode.summary.csv'\n",
    "\n",
    "// Other steps\n",
    "\n",
    "params.GENOME_GZ_GTF='http://ftp.ensembl.org/pub/release-109/gtf/homo_sapiens/Homo_sapiens.GRCh38.109.gtf.gz'\n",
    "params.ENV_SYNAPSE_TOKEN = \"nextflow secrets get SYNAPSE\".execute().text.trim()\n",
    "\n",
    "\n",
    "params.MAPQ_THRESHOLD = false\n",
    "params.BARCODE_TRANSLATE = 'path'\n",
    "params.ATAC_BARCODE_AND_POOL_SCRIPT='atac_barcode_filter_and_subpool.sh'\n",
    "params.ATAC_FRAGMENTS_CUTOFF = 4\n",
    "params.RUN_TABIX_SCRIPT='run_tabix.sh'\n",
    "params.ATAC_TSS_BASES_FLANK = 2000\n",
    "params.ATAC_TSS_COL_WITH_STRANDS_INFO = 4\n",
    "params.ATAC_TSS_SMOOTHING_WINDOW_SIZE = 20\n",
    "params.ATAC_TSS_REGION_BED_FILE='{nf_data_base_path}/{nf_data_project}/hg38.synapse_transcripts.TSS.bed'\n",
    "params.ATAC_TSS_BULK_CALCULATION_SCRIPT='compute_tss_enrichment_bulk.py'\n",
    "params.ATAC_TSS_SNAPATAC2_CALCULATION_SCRIPT='compute_tss_enrichment_snapatac2.py'\n",
    "params.ATAC_TSS_SNAPATAC2_GENES_GTF='{nf_data_base_path}/{nf_data_project}/genes.gtf'\n",
    "params.ATAC_TSS_SNAPATAC2_MIN_FRAG_CUTOFF = 10\n",
    "\n",
    "params.ATAC_INSERT_SIZE_PLOT_SCRIPT='plot_insert_size_hist.py'\n",
    "\n",
    "params.SC_ATAC_QC_PLOT_SCRIPT='scrna_plot_atac_qc_metrics.R'\n",
    "params.SC_ATAC_QC_PLOT_HELPER_SCRIPT='barcode_rank_functions.R'\n",
    "params.SC_ATAC_QC_BARCODE_METADATA_FILE='{nf_data_base_path}/{nf_data_project}/barcode_metadata.txt'\n",
    "params.SC_ATAC_QC_FRAGMENT_CUTOFF=10\n",
    "params.SC_ATAC_QC_BARCODE_OUTPUT_FILE='sc_atac_qc_barcode_output_file.png'\n",
    "    \n",
    "params.SC_ATAC_BARCODE_METADATA_SCRIPT='generate_barcode_metadata.sh'\n",
    "params.SC_ATAC_BARCODE_METADATA_FILTERED_BARCODE_STATS='{nf_data_base_path}nf_wf_fg_singlecell/filtered_barcode_stats.csv'\n",
    "params.SC_ATAC_BARCODE_METADATA_TSS_ENRICHMENT_BARCODE_STATS='{nf_data_base_path}nf_wf_fg_singlecell/tss_enrichment_barcode_stats.csv'\n",
    "\n",
    "params.SC_ATAC_GENERATE_BARCODE_RANK_PLOT_SCRIPT='plot_insert_size_hist.py'\n",
    "params.SC_ATAC_GENERATE_BARCODE_RANK_PLOT_PKR='PKR123'\n",
    "\n",
    "params.SC_JOINT_CELL_BARCODE_PLOT_SCRTIP_PY='joint_cell_plotting.py'\n",
    "params.SC_JOINT_CELL_BARCODE_PLOT_SCRTIP_R='joint_cell_plotting_density.R'\n",
    "\"\"\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !rm -r /Users/eilaarich-landkof-stanford/Documents/Code/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/work\n",
    "# !rm -r /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/work\n",
    "# !gsutil rm -r gs://eila-nextflow-bucket/work/ATAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !rm .nextflow*\n",
    "\n",
    "# !ls /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/local_atac_fastqs_files.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "N E X T F L O W  ~  version 23.10.1\n",
      "Launching `nf_wf_fg_singlecell_atac.nf` [festering_lorenz] DSL2 - revision: 7b9a0433e4\n",
      "/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/local_atac_fastqs_files.tsv\n",
      "eyJ0eXAiOiJKV1QiLCJraWQiOiJXN05OOldMSlQ6SjVSSzpMN1RMOlQ3TDc6M1ZYNjpKRU9VOjY0NFI6VTNJWDo1S1oyOjdaQ0s6RlBUSCIsImFsZyI6IlJTMjU2In0.eyJhY2Nlc3MiOnsic2NvcGUiOlsidmlldyIsImRvd25sb2FkIl0sIm9pZGNfY2xhaW1zIjp7fX0sInRva2VuX3R5cGUiOiJQRVJTT05BTF9BQ0NFU1NfVE9LRU4iLCJpc3MiOiJodHRwczovL3JlcG8tcHJvZC5wcm9kLnNhZ2ViYXNlLm9yZy9hdXRoL3YxIiwiYXVkIjoiMCIsIm5iZiI6MTY5NzgxODkxNiwiaWF0IjoxNjk3ODE4OTE2LCJqdGkiOiI0MTIzIiwic3ViIjoiMzQ3NDU3MSJ9.n0cugYMAVzYZ8mMMc7CjcqqEV7FB-K-oKUDT45SvM6KmbCpb_v3lZ2gg3xXi1tubWH0__wpnbJ7cYrGEtOyt4vXAPkMiEtZFGqS0MFKp_h-mHHH4pnACSQQYEy1zdyqFn9WRE7ipjWg4yYSbq84N7J3gzi8VKmiOAmuIGKaJaS0snkUjtJEVQH-0vZqWs1nKgVRWIZv31Bo8vuDzWpBrXD-zRT_uQW9l_I9fU50o1AqyFhNe_gbty_Z6imb8VT33wMXiDgJh15lG_M70PXU15v3egDXLgw1dGj32kj1nKanfqBVqbMSzgzPAG037p3YH_Q556BLceFBBb5Jaej5JVA\n",
      "after genome_fasta_ch\n",
      "[-        ] process > run_zcat                       -\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip           -\u001b[K\n",
      "[-        ] process > run_chromap_map_to_idx         -\u001b[K\n",
      "[-        ] process > run_bgzip                      -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_logs                 -\u001b[K\n",
      "[-        ] process > run_filter_fragments           -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments   -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "The value na is present in subpoolValues. NO subpool\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix_filtered_fragments for the second time with filtered_fragment_file_out\u001b[K\n",
      "after call run_tabix_filtered_fragments for filtered fragments\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "before call snapatac2\u001b[K\n",
      "after call snapatac2\u001b[K\n",
      "before call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "after call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "before call run_generate_barcode_metadata\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "\u001b[33A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2ArfrV6PV2s7La\u001b[22;39m\u001b[K\n",
      "[-        ] process > run_zcat                       -\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip           -\u001b[K\n",
      "[-        ] process > run_chromap_map_to_idx         -\u001b[K\n",
      "[-        ] process > run_bgzip                      -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_logs                 -\u001b[K\n",
      "[-        ] process > run_filter_fragments           -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments   -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "The value na is present in subpoolValues. NO subpool\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix_filtered_fragments for the second time with filtered_fragment_file_out\u001b[K\n",
      "after call run_tabix_filtered_fragments for filtered fragments\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "before call snapatac2\u001b[K\n",
      "after call snapatac2\u001b[K\n",
      "before call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "after call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "before call run_generate_barcode_metadata\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "\u001b[34A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2ArfrV6PV2s7La\u001b[22;39m\u001b[K\n",
      "[-        ] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip           -\u001b[K\n",
      "[-        ] process > run_chromap_map_to_idx         -\u001b[K\n",
      "[-        ] process > run_bgzip                      -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_logs                 -\u001b[K\n",
      "[-        ] process > run_filter_fragments           -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments   -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "The value na is present in subpoolValues. NO subpool\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix_filtered_fragments for the second time with filtered_fragment_file_out\u001b[K\n",
      "after call run_tabix_filtered_fragments for filtered fragments\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "before call snapatac2\u001b[K\n",
      "after call snapatac2\u001b[K\n",
      "before call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "after call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "before call run_generate_barcode_metadata\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "\u001b[34A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2ArfrV6PV2s7La\u001b[22;39m\u001b[K\n",
      "[-        ] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip           [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_chromap_map_to_idx         -\u001b[K\n",
      "[-        ] process > run_bgzip                      -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_logs                 -\u001b[K\n",
      "[-        ] process > run_filter_fragments           -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments   -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "The value na is present in subpoolValues. NO subpool\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix_filtered_fragments for the second time with filtered_fragment_file_out\u001b[K\n",
      "after call run_tabix_filtered_fragments for filtered fragments\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "before call snapatac2\u001b[K\n",
      "after call snapatac2\u001b[K\n",
      "before call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "after call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "before call run_generate_barcode_metadata\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "\u001b[34A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2ArfrV6PV2s7La\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (1)\u001b[K\n",
      "[a7/10e80c] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip           [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_chromap_map_to_idx         -\u001b[K\n",
      "[-        ] process > run_bgzip                      -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_logs                 -\u001b[K\n",
      "[-        ] process > run_filter_fragments           -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments   -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "The value na is present in subpoolValues. NO subpool\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix_filtered_fragments for the second time with filtered_fragment_file_out\u001b[K\n",
      "after call run_tabix_filtered_fragments for filtered fragments\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "before call snapatac2\u001b[K\n",
      "after call snapatac2\u001b[K\n",
      "before call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "after call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "before call run_generate_barcode_metadata\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "\u001b[35A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2ArfrV6PV2s7La\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (1)\u001b[K\n",
      "[a7/10e80c] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip           [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_chromap_map_to_idx         [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_bgzip                      -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_logs                 -\u001b[K\n",
      "[-        ] process > run_filter_fragments           -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments   -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "The value na is present in subpoolValues. NO subpool\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix_filtered_fragments for the second time with filtered_fragment_file_out\u001b[K\n",
      "after call run_tabix_filtered_fragments for filtered fragments\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "before call snapatac2\u001b[K\n",
      "after call snapatac2\u001b[K\n",
      "before call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "after call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "before call run_generate_barcode_metadata\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "\u001b[35A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2ArfrV6PV2s7La\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (2)\u001b[K\n",
      "[a7/10e80c] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[47/2fe048] process > run_whitelist_gunzip (1)       [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_chromap_map_to_idx         [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_bgzip                      -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_logs                 -\u001b[K\n",
      "[-        ] process > run_filter_fragments           -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments   -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "The value na is present in subpoolValues. NO subpool\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix_filtered_fragments for the second time with filtered_fragment_file_out\u001b[K\n",
      "after call run_tabix_filtered_fragments for filtered fragments\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "before call snapatac2\u001b[K\n",
      "after call snapatac2\u001b[K\n",
      "before call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "after call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "before call run_generate_barcode_metadata\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "\u001b[35A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2ArfrV6PV2s7La\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (3)\u001b[K\n",
      "[a7/10e80c] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[47/2fe048] process > run_whitelist_gunzip (1)       [  0%] 0 of 1\u001b[K\n",
      "[de/254e65] process > run_chromap_map_to_idx (1)     [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_bgzip                      -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_logs                 -\u001b[K\n",
      "[-        ] process > run_filter_fragments           -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments   -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "The value na is present in subpoolValues. NO subpool\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix_filtered_fragments for the second time with filtered_fragment_file_out\u001b[K\n",
      "after call run_tabix_filtered_fragments for filtered fragments\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "before call snapatac2\u001b[K\n",
      "after call snapatac2\u001b[K\n",
      "before call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "after call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "before call run_generate_barcode_metadata\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "\u001b[35A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2ArfrV6PV2s7La\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (3)\u001b[K\n",
      "[a7/10e80c] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[47/2fe048] process > run_whitelist_gunzip (1)       [  0%] 0 of 1\u001b[K\n",
      "[de/254e65] process > run_chromap_map_to_idx (1)     [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_bgzip                      -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_logs                 -\u001b[K\n",
      "[-        ] process > run_filter_fragments           -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments   -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[35A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2ArfrV6PV2s7La\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (3)\u001b[K\n",
      "[a7/10e80c] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[47/2fe048] process > run_whitelist_gunzip (1)       [  0%] 0 of 1\u001b[K\n",
      "[de/254e65] process > run_chromap_map_to_idx (1)     [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_bgzip                      -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_logs                 -\u001b[K\n",
      "[-        ] process > run_filter_fragments           -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments   -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[35A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2ArfrV6PV2s7La\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (3)\u001b[K\n",
      "[a7/10e80c] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[47/2fe048] process > run_whitelist_gunzip (1)       [  0%] 0 of 1\u001b[K\n",
      "[de/254e65] process > run_chromap_map_to_idx (1)     [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_bgzip                      -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_logs                 -\u001b[K\n",
      "[-        ] process > run_filter_fragments           -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments   -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "\u001b[13A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2ArfrV6PV2s7La\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (3)\u001b[K\n",
      "[a7/10e80c] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[47/2fe048] process > run_whitelist_gunzip (1)       [  0%] 0 of 1 ✔\u001b[K\n",
      "[de/254e65] process > run_chromap_map_to_idx (1)     [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_bgzip                      -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_logs                 -\u001b[K\n",
      "[-        ] process > run_filter_fragments           -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments   -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "Start run_whitelist_gunzip\u001b[K\n",
      "Gunzip file is: sai_192_whitelist.txt\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 5\u001b[K\n",
      "lrwxrwxrwx 1 root root 176 Feb  1 05:10 BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/a2/a8d562c8ebabf93e84219ebe3dc6a7/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root 176 Feb  1 05:10 BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/75/c040d60635402464fb2c34c4a2c931/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root 176 Feb  1 05:10 BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/fa/1dcb6c6a83277a3041d83ba66adacc/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root 176 Feb  1 05:10 BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/08/f10141c8271e12dfd004199b55a87c/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root 176 Feb  1 05:10 BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/3f/a2f99633a7a18b1f690aaeaa2af230/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root 176 Feb  1 05:10 BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/e2/1e74f9eb587929aa757342b819ea77/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root 124 Feb  1 05:10 na.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/8b/b521669a71db49f1fb1d2870ff0854/na.na\u001b[K\n",
      "lrwxrwxrwx 1 root root 125 Feb  1 05:10 na1.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/6c/ce8eff486ee7fbc13d0021eac82ef2/na1.na\u001b[K\n",
      "lrwxrwxrwx 1 root root 140 Feb  1 05:10 sai_192_whitelist.txt -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/ed/0c333bf88d827555988fb53f6673cd/sai_192_whitelist.txt\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 172805\u001b[K\n",
      "-r-x--x--x 1 root root 176947200 Feb  1 05:10 sai_192_whitelist\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  1 05:10 BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/a2/a8d562c8ebabf93e84219ebe3dc6a7/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  1 05:10 BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/75/c040d60635402464fb2c34c4a2c931/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  1 05:10 BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/fa/1dcb6c6a83277a3041d83ba66adacc/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  1 05:10 BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/08/f10141c8271e12dfd004199b55a87c/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  1 05:10 BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/3f/a2f99633a7a18b1f690aaeaa2af230/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  1 05:10 BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/e2/1e74f9eb587929aa757342b819ea77/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb  1 05:10 na.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/8b/b521669a71db49f1fb1d2870ff0854/na.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       125 Feb  1 05:10 na1.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/6c/ce8eff486ee7fbc13d0021eac82ef2/na1.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       140 Feb  1 05:10 sai_192_whitelist.txt -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/ed/0c333bf88d827555988fb53f6673cd/sai_192_whitelist.txt\u001b[K\n",
      "Finished run_whitelist_gunzip\u001b[K\n",
      "\u001b[K\n",
      "\u001b[40A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2ArfrV6PV2s7La\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (3)\u001b[K\n",
      "[a7/10e80c] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[47/2fe048] process > run_whitelist_gunzip (1)       [100%] 1 of 1 ✔\u001b[K\n",
      "[de/254e65] process > run_chromap_map_to_idx (1)     [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_bgzip                      -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_logs                 -\u001b[K\n",
      "[-        ] process > run_filter_fragments           -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments   -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "Start run_whitelist_gunzip\u001b[K\n",
      "Gunzip file is: sai_192_whitelist.txt\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 5\u001b[K\n",
      "lrwxrwxrwx 1 root root 176 Feb  1 05:10 BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/a2/a8d562c8ebabf93e84219ebe3dc6a7/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root 176 Feb  1 05:10 BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/75/c040d60635402464fb2c34c4a2c931/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root 176 Feb  1 05:10 BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/fa/1dcb6c6a83277a3041d83ba66adacc/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root 176 Feb  1 05:10 BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/08/f10141c8271e12dfd004199b55a87c/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root 176 Feb  1 05:10 BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/3f/a2f99633a7a18b1f690aaeaa2af230/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root 176 Feb  1 05:10 BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/e2/1e74f9eb587929aa757342b819ea77/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root 124 Feb  1 05:10 na.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/8b/b521669a71db49f1fb1d2870ff0854/na.na\u001b[K\n",
      "lrwxrwxrwx 1 root root 125 Feb  1 05:10 na1.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/6c/ce8eff486ee7fbc13d0021eac82ef2/na1.na\u001b[K\n",
      "lrwxrwxrwx 1 root root 140 Feb  1 05:10 sai_192_whitelist.txt -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/ed/0c333bf88d827555988fb53f6673cd/sai_192_whitelist.txt\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 172805\u001b[K\n",
      "-r-x--x--x 1 root root 176947200 Feb  1 05:10 sai_192_whitelist\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  1 05:10 BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/a2/a8d562c8ebabf93e84219ebe3dc6a7/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  1 05:10 BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/75/c040d60635402464fb2c34c4a2c931/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  1 05:10 BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/fa/1dcb6c6a83277a3041d83ba66adacc/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  1 05:10 BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/08/f10141c8271e12dfd004199b55a87c/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  1 05:10 BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/3f/a2f99633a7a18b1f690aaeaa2af230/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  1 05:10 BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/e2/1e74f9eb587929aa757342b819ea77/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb  1 05:10 na.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/8b/b521669a71db49f1fb1d2870ff0854/na.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       125 Feb  1 05:10 na1.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/6c/ce8eff486ee7fbc13d0021eac82ef2/na1.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       140 Feb  1 05:10 sai_192_whitelist.txt -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/ed/0c333bf88d827555988fb53f6673cd/sai_192_whitelist.txt\u001b[K\n",
      "Finished run_whitelist_gunzip\u001b[K\n",
      "\u001b[K\n",
      "\u001b[40A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2ArfrV6PV2s7La\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (3)\u001b[K\n",
      "[a7/10e80c] process > run_zcat                       [  0%] 0 of 1 ✔\u001b[K\n",
      "[47/2fe048] process > run_whitelist_gunzip (1)       [100%] 1 of 1 ✔\u001b[K\n",
      "[de/254e65] process > run_chromap_map_to_idx (1)     [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_bgzip                      -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_logs                 -\u001b[K\n",
      "[-        ] process > run_filter_fragments           -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments   -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb  1 05:10 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/27/6ed5709030395a942950c5f1d85ff8/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb  1 05:10 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb  1 05:10 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/27/6ed5709030395a942950c5f1d85ff8/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[51A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2ArfrV6PV2s7La\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (3)\u001b[K\n",
      "[a7/10e80c] process > run_zcat                       [100%] 1 of 1 ✔\u001b[K\n",
      "[47/2fe048] process > run_whitelist_gunzip (1)       [100%] 1 of 1 ✔\u001b[K\n",
      "[de/254e65] process > run_chromap_map_to_idx (1)     [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_bgzip                      -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_logs                 -\u001b[K\n",
      "[-        ] process > run_filter_fragments           -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments   -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb  1 05:10 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/27/6ed5709030395a942950c5f1d85ff8/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb  1 05:10 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb  1 05:10 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/27/6ed5709030395a942950c5f1d85ff8/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n"
     ]
    }
   ],
   "source": [
    "!nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_atac.config -profile google_batch_profile -process.echo -with-tower -resume\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_vm.config -profile google_batch_profile\n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_vm.config -profile google_batch_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_vm_anvil.config -profile google_batch_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_local.config -profile google_batch_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_local.config -profile docker_profile\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Jan-31 21:02:05.805 [main] DEBUG nextflow.cli.Launcher - $> nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_atac.config -profile google_batch_profile -process.echo -with-tower -resume\n",
      "Jan-31 21:02:05.888 [main] INFO  nextflow.cli.CmdRun - N E X T F L O W  ~  version 23.10.1\n",
      "Jan-31 21:02:05.901 [main] DEBUG nextflow.plugin.PluginsFacade - Setting up plugin manager > mode=prod; embedded=false; plugins-dir=/Users/eilaarich-landkof-stanford/.nextflow/plugins; core-plugins: nf-amazon@2.1.4,nf-azure@1.3.3,nf-cloudcache@0.3.0,nf-codecommit@0.1.5,nf-console@1.0.6,nf-ga4gh@1.1.0,nf-google@1.8.3,nf-tower@1.6.3,nf-wave@1.0.1\n",
      "Jan-31 21:02:05.908 [main] INFO  o.pf4j.DefaultPluginStatusProvider - Enabled plugins: []\n",
      "Jan-31 21:02:05.909 [main] INFO  o.pf4j.DefaultPluginStatusProvider - Disabled plugins: []\n",
      "Jan-31 21:02:05.911 [main] INFO  org.pf4j.DefaultPluginManager - PF4J version 3.4.1 in 'deployment' mode\n",
      "Jan-31 21:02:05.918 [main] INFO  org.pf4j.AbstractPluginManager - No plugins\n",
      "Jan-31 21:02:05.930 [main] DEBUG nextflow.config.ConfigBuilder - User config file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Jan-31 21:02:05.931 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Jan-31 21:02:05.945 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `google_batch_profile`\n",
      "Jan-31 21:02:07.438 [main] DEBUG nextflow.config.ConfigBuilder - Available config profiles: [google_batch_profile]\n",
      "Jan-31 21:02:07.460 [main] DEBUG nextflow.cli.CmdRun - Applied DSL=2 by global default\n",
      "Jan-31 21:02:07.474 [main] INFO  nextflow.cli.CmdRun - Launching `nf_wf_fg_singlecell_atac.nf` [stoic_kare] DSL2 - revision: 7b9a0433e4\n",
      "Jan-31 21:02:07.475 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins default=[nf-google@1.8.3]\n",
      "Jan-31 21:02:07.475 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins resolved requirement=[nf-google@1.8.3, nf-tower@1.6.3, nf-wave@1.0.1]\n",
      "Jan-31 21:02:07.476 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-google version: 1.8.3\n",
      "Jan-31 21:02:07.489 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-google@1.8.3' resolved\n",
      "Jan-31 21:02:07.489 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-google@1.8.3'\n",
      "Jan-31 21:02:07.523 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-google@1.8.3\n",
      "Jan-31 21:02:07.524 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-tower version: 1.6.3\n",
      "Jan-31 21:02:07.526 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-tower@1.6.3' resolved\n",
      "Jan-31 21:02:07.526 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-tower@1.6.3'\n",
      "Jan-31 21:02:07.531 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-tower@1.6.3\n",
      "Jan-31 21:02:07.531 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-wave version: 1.0.1\n",
      "Jan-31 21:02:07.534 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-wave@1.0.1' resolved\n",
      "Jan-31 21:02:07.534 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-wave@1.0.1'\n",
      "Jan-31 21:02:07.549 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-wave@1.0.1\n",
      "Jan-31 21:02:07.559 [main] DEBUG n.secret.LocalSecretsProvider - Secrets store: /Users/eilaarich-landkof-stanford/.nextflow/secrets/store.json\n",
      "Jan-31 21:02:07.561 [main] DEBUG nextflow.secret.SecretsLoader - Discovered secrets providers: [nextflow.secret.LocalSecretsProvider@1fc713c9] - activable => nextflow.secret.LocalSecretsProvider@1fc713c9\n",
      "Jan-31 21:02:07.628 [main] DEBUG nextflow.Session - Session UUID: 0cbd4d64-b857-4807-b564-11778fa2d750\n",
      "Jan-31 21:02:07.629 [main] DEBUG nextflow.Session - Run name: stoic_kare\n",
      "Jan-31 21:02:07.629 [main] DEBUG nextflow.Session - Executor pool size: 8\n",
      "Jan-31 21:02:07.751 [main] DEBUG nextflow.file.FilePorter - File porter settings maxRetries=3; maxTransfers=50; pollTimeout=null\n",
      "Jan-31 21:02:07.754 [main] DEBUG nextflow.util.ThreadPoolBuilder - Creating thread pool 'FileTransfer' minSize=10; maxSize=24; workQueue=LinkedBlockingQueue[10000]; allowCoreThreadTimeout=false\n",
      "Jan-31 21:02:07.756 [main] DEBUG nextflow.config.ConfigBuilder - User config file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Jan-31 21:02:07.756 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Jan-31 21:02:07.756 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `google_batch_profile`\n",
      "Jan-31 21:02:08.815 [main] DEBUG nextflow.config.ConfigBuilder - Available config profiles: [google_batch_profile]\n",
      "Jan-31 21:02:08.842 [main] DEBUG nextflow.cli.CmdRun - \n",
      "  Version: 23.10.1 build 5891\n",
      "  Created: 12-01-2024 22:01 UTC (14:01 PDT)\n",
      "  System: Mac OS X 14.3\n",
      "  Runtime: Groovy 3.0.19 on OpenJDK 64-Bit Server VM 17.0.8+7-LTS\n",
      "  Encoding: UTF-8 (UTF-8)\n",
      "  Process: 2742@Eilas-MacBook-Air.local [10.0.0.212]\n",
      "  CPUs: 8 - Mem: 16 GB (182 MB) - Swap: 0 (0)\n",
      "Jan-31 21:02:08.871 [main] DEBUG nextflow.Session - Work-dir: gs://eila-nextflow-bucket/work/ATAC [Mac OS X]\n",
      "Jan-31 21:02:08.894 [main] DEBUG nextflow.executor.ExecutorFactory - Extension executors providers=[GoogleLifeSciencesExecutor, GoogleBatchExecutor]\n",
      "Jan-31 21:02:08.902 [main] DEBUG nextflow.Session - Observer factory: DefaultObserverFactory\n",
      "Jan-31 21:02:08.904 [main] DEBUG nextflow.Session - Observer factory: WaveFactory\n",
      "Jan-31 21:02:08.904 [main] DEBUG io.seqera.wave.plugin.WaveFactory - Detected Fusion enabled -- Enabling bundle project resources -- Disabling upload of remote bin directory\n",
      "Jan-31 21:02:08.917 [main] DEBUG i.s.wave.plugin.config.WaveConfig - Wave strategy not specified - using default: [container, dockerfile, conda, spack]\n",
      "Jan-31 21:02:08.923 [main] DEBUG io.seqera.wave.plugin.WaveClient - Wave config: WaveConfig(enabled:true, endpoint:https://wave.seqera.io, containerConfigUrl:[], tokensCacheMaxDuration:30m, condaOpts:CondaOpts(mambaImage=mambaorg/micromamba:1.5.1; basePackages=conda-forge::procps-ng, commands=null), spackOpts:SpackOpts(basePackages=null, commands=null), strategy:[container, dockerfile, conda, spack], bundleProjectResources:true, buildRepository:null, cacheRepository:null, retryOpts:RetryOpts(delay:450ms, maxDelay:1m 30s, maxAttempts:10, jitter:0.25), httpClientOpts:HttpOpts(), freezeMode:null)\n",
      "Jan-31 21:02:09.018 [main] DEBUG nextflow.Session - Observer factory: TowerFactory\n",
      "Jan-31 21:02:09.074 [main] DEBUG nextflow.cache.CacheFactory - Using Nextflow cache factory: nextflow.cache.DefaultCacheFactory\n",
      "Jan-31 21:02:09.081 [main] DEBUG nextflow.util.CustomThreadPool - Creating default thread pool > poolSize: 9; maxThreads: 1000\n",
      "Jan-31 21:02:09.146 [main] DEBUG nextflow.Session - Session start\n",
      "Jan-31 21:02:09.168 [main] DEBUG io.seqera.tower.plugin.TowerClient - Creating Tower observer -- endpoint=https://api.tower.nf; requestInterval=1s; aliveInterval=1m; maxRetries=5; backOffBase=3; backOffDelay=250\n",
      "Jan-31 21:02:10.317 [main] DEBUG nextflow.script.ScriptRunner - > Launching execution\n",
      "Jan-31 21:02:11.485 [main] INFO  nextflow.script.BaseScript - /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/local_atac_fastqs_files.tsv\n",
      "Jan-31 21:02:11.486 [main] INFO  nextflow.script.BaseScript - eyJ0eXAiOiJKV1QiLCJraWQiOiJXN05OOldMSlQ6SjVSSzpMN1RMOlQ3TDc6M1ZYNjpKRU9VOjY0NFI6VTNJWDo1S1oyOjdaQ0s6RlBUSCIsImFsZyI6IlJTMjU2In0.eyJhY2Nlc3MiOnsic2NvcGUiOlsidmlldyIsImRvd25sb2FkIl0sIm9pZGNfY2xhaW1zIjp7fX0sInRva2VuX3R5cGUiOiJQRVJTT05BTF9BQ0NFU1NfVE9LRU4iLCJpc3MiOiJodHRwczovL3JlcG8tcHJvZC5wcm9kLnNhZ2ViYXNlLm9yZy9hdXRoL3YxIiwiYXVkIjoiMCIsIm5iZiI6MTY5NzgxODkxNiwiaWF0IjoxNjk3ODE4OTE2LCJqdGkiOiI0MTIzIiwic3ViIjoiMzQ3NDU3MSJ9.n0cugYMAVzYZ8mMMc7CjcqqEV7FB-K-oKUDT45SvM6KmbCpb_v3lZ2gg3xXi1tubWH0__wpnbJ7cYrGEtOyt4vXAPkMiEtZFGqS0MFKp_h-mHHH4pnACSQQYEy1zdyqFn9WRE7ipjWg4yYSbq84N7J3gzi8VKmiOAmuIGKaJaS0snkUjtJEVQH-0vZqWs1nKgVRWIZv31Bo8vuDzWpBrXD-zRT_uQW9l_I9fU50o1AqyFhNe_gbty_Z6imb8VT33wMXiDgJh15lG_M70PXU15v3egDXLgw1dGj32kj1nKanfqBVqbMSzgzPAG037p3YH_Q556BLceFBBb5Jaej5JVA\n",
      "Jan-31 21:02:11.574 [main] INFO  nextflow.script.BaseScript - after genome_fasta_ch\n",
      "Jan-31 21:02:11.621 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:zcat` matches labels `zcat` for process with name run_zcat\n",
      "Jan-31 21:02:11.625 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Jan-31 21:02:11.625 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Jan-31 21:02:11.627 [main] DEBUG nextflow.executor.Executor - [warm up] executor > google-batch (fusion enabled)\n",
      "Jan-31 21:02:11.631 [main] DEBUG n.processor.TaskPollingMonitor - Creating task monitor for executor 'google-batch' > capacity: 1000; pollInterval: 10s; dumpInterval: 5m \n",
      "Jan-31 21:02:11.633 [main] DEBUG n.processor.TaskPollingMonitor - >>> barrier register (monitor: google-batch)\n",
      "Jan-31 21:02:11.638 [main] DEBUG nextflow.cloud.google.GoogleOpts - Google auth via application credentials file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/hai-gcp-genomic-beb9207a40b8.json\n",
      "Jan-31 21:02:11.644 [main] DEBUG n.c.google.batch.GoogleBatchExecutor - [GOOGLE BATCH] Executor config=BatchConfig[googleOpts=GoogleOpts(projectId:hai-gcp-genomic, credsFile:/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/hai-gcp-genomic-beb9207a40b8.json, location:us-central1, enableRequesterPaysBuckets:false, httpConnectTimeout:1m, httpReadTimeout:1m, credentials:ServiceAccountCredentials{clientId=111364348320962814542, clientEmail=eila-batch@hai-gcp-genomic.iam.gserviceaccount.com, privateKeyId=beb9207a40b8b2085c7f48c3fcf8c2342180d471, transportFactoryClassName=com.google.auth.oauth2.OAuth2Utils$DefaultHttpTransportFactory, tokenServerUri=https://oauth2.googleapis.com/token, scopes=[https://www.googleapis.com/auth/cloud-platform], defaultScopes=[], serviceAccountUser=null, quotaProjectId=null, lifetime=3600, useJwtAccessWithScope=false, defaultRetriesEnabled=true})\n",
      "Jan-31 21:02:11.660 [main] DEBUG n.c.google.batch.client.BatchClient - [GOOGLE BATCH] Creating service client with config credentials\n",
      "Jan-31 21:02:12.913 [main] INFO  nextflow.script.BaseScript - after run_zcat \n",
      "Jan-31 21:02:12.919 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:gunzip` matches labels `gunzip` for process with name run_whitelist_gunzip\n",
      "Jan-31 21:02:12.920 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Jan-31 21:02:12.920 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Jan-31 21:02:12.922 [main] INFO  nextflow.script.BaseScript - after run_whitelist_gunzip\n",
      "Jan-31 21:02:12.926 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:chromap_map_idx` matches labels `chromap_map_idx` for process with name run_chromap_map_to_idx\n",
      "Jan-31 21:02:12.927 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Jan-31 21:02:12.927 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Jan-31 21:02:12.932 [main] INFO  nextflow.script.BaseScript - finished run_chromap_map_to_idx\n",
      "Jan-31 21:02:12.936 [main] INFO  nextflow.script.BaseScript - The value na is present in subpoolValues. NO subpool\n",
      "Jan-31 21:02:12.936 [main] INFO  nextflow.script.BaseScript - before call bgzip\n",
      "Jan-31 21:02:12.937 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:bgzip` matches labels `bgzip` for process with name run_bgzip\n",
      "Jan-31 21:02:12.938 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Jan-31 21:02:12.938 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Jan-31 21:02:12.940 [main] INFO  nextflow.script.BaseScript - after call bgzip\n",
      "Jan-31 21:02:12.940 [main] INFO  nextflow.script.BaseScript - before call run_tabix_chromap with tabix shell script\n",
      "Jan-31 21:02:12.941 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:tabix` matches labels `tabix` for process with name run_tabix_chromap\n",
      "Jan-31 21:02:12.942 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Jan-31 21:02:12.942 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Jan-31 21:02:12.944 [main] INFO  nextflow.script.BaseScript - after call run_tabix_chromap\n",
      "Jan-31 21:02:12.944 [main] INFO  nextflow.script.BaseScript - before call run_merge_logs\n",
      "Jan-31 21:02:12.945 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:merge_log` matches labels `merge_log` for process with name run_merge_logs\n",
      "Jan-31 21:02:12.946 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Jan-31 21:02:12.946 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Jan-31 21:02:12.948 [main] INFO  nextflow.script.BaseScript - after call run_merge_logs\n",
      "Jan-31 21:02:12.949 [main] INFO  nextflow.script.BaseScript - before call run_merge_logs\n",
      "Jan-31 21:02:12.950 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:filter_fragments` matches labels `filter_fragments` for process with name run_filter_fragments\n",
      "Jan-31 21:02:12.951 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Jan-31 21:02:12.952 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Jan-31 21:02:12.955 [main] INFO  nextflow.script.BaseScript - after call run_filter_fragments\n",
      "Jan-31 21:02:12.955 [main] INFO  nextflow.script.BaseScript - before call run_tabix_filtered_fragments for the second time with filtered_fragment_file_out\n",
      "Jan-31 21:02:12.957 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:tabix_after_fragments` matches labels `tabix_after_fragments` for process with name run_tabix_filtered_fragments\n",
      "Jan-31 21:02:12.958 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Jan-31 21:02:12.958 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Jan-31 21:02:12.960 [main] INFO  nextflow.script.BaseScript - after call run_tabix_filtered_fragments for filtered fragments\n",
      "Jan-31 21:02:12.960 [main] INFO  nextflow.script.BaseScript - before call run_calculate_tss_enrichment_bulk\n",
      "Jan-31 21:02:12.962 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:tss_bulk` matches labels `tss_bulk` for process with name run_calculate_tss_enrichment_bulk\n",
      "Jan-31 21:02:12.963 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Jan-31 21:02:12.963 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Jan-31 21:02:12.965 [main] INFO  nextflow.script.BaseScript - after call run_calculate_tss_enrichment_bulk\n",
      "Jan-31 21:02:12.965 [main] INFO  nextflow.script.BaseScript - before call snapatac2\n",
      "Jan-31 21:02:12.966 [main] WARN  nextflow.script.ScriptBinding - Access to undefined parameter `ATAC_TSS_SNAPATAC2_GENES_GTF` -- Initialise it to a default value eg. `params.ATAC_TSS_SNAPATAC2_GENES_GTF = some_value`\n",
      "Jan-31 21:02:12.968 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:tss_snapatac2` matches labels `tss_snapatac2` for process with name run_calculate_tss_enrichment_snapatac2\n",
      "Jan-31 21:02:12.975 [main] DEBUG nextflow.script.ScriptRunner - Parsed script files:\n",
      "  Script_0de7f532c8b3da2c: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_bgzip.nf\n",
      "  Script_87d8d9fb99b69787: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_atac_barcode_rank_plot.nf\n",
      "  Script_eceb0470d0f82fa0: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_gunzip.nf\n",
      "  Script_1dabf5cfd062ce56: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nf_wf_fg_singlecell_atac.nf\n",
      "  Script_85e7960f7cd2325a: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_tss.nf\n",
      "  Script_cc895528cea919f6: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_zcat.nf\n",
      "  Script_9c2ce54dcf0b5c06: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_merge_logs.nf\n",
      "  Script_9d4639c8cd883c70: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_download_url_files.nf\n",
      "  Script_44ead53606f06d49: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_chromap_utils.nf\n",
      "  Script_8cdb03275a5855e0: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_filter_fragments.nf\n",
      "  Script_ea7cdb417e42517f: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_generate_barcode_metadata.nf\n",
      "  Script_e112646bcdb899f0: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_seqspec_utils.nf\n",
      "  Script_b8ed272d33e7871d: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_atac_generate_metadata_annoation.nf\n",
      "  Script_c9985907723aa9e8: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_tabix.nf\n",
      "Jan-31 21:02:12.975 [main] DEBUG nextflow.Session - Session aborted -- Cause: A process input channel evaluates to null -- Invalid declaration `path gtf_file`\n",
      "Jan-31 21:02:12.991 [main] DEBUG nextflow.Session - The following nodes are still active:\n",
      "  [operator] splitCsv\n",
      "  [operator] map\n",
      "  [operator] collect\n",
      "  [operator] collect\n",
      "  [operator] collect\n",
      "\n",
      "Jan-31 21:02:12.997 [Task monitor] DEBUG n.processor.TaskPollingMonitor - <<< barrier arrives (monitor: google-batch) - terminating tasks monitor poll loop\n",
      "Jan-31 21:02:12.997 [main] ERROR nextflow.cli.Launcher - A process input channel evaluates to null -- Invalid declaration `path gtf_file`\n",
      "java.lang.IllegalArgumentException: A process input channel evaluates to null -- Invalid declaration `path gtf_file`\n",
      "\tat java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)\n",
      "\tat java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)\n",
      "\tat java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)\n",
      "\tat java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)\n",
      "\tat java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)\n",
      "\tat org.codehaus.groovy.reflection.CachedConstructor.invoke(CachedConstructor.java:72)\n",
      "\tat org.codehaus.groovy.runtime.callsite.ConstructorSite$ConstructorSiteNoUnwrapNoCoerce.callConstructor(ConstructorSite.java:105)\n",
      "\tat org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallConstructor(CallSiteArray.java:59)\n",
      "\tat org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:263)\n",
      "\tat org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:277)\n",
      "\tat nextflow.script.params.BaseInParam.checkFromNotNull(BaseInParam.groovy:169)\n",
      "\tat jdk.internal.reflect.GeneratedMethodAccessor68.invoke(Unknown Source)\n",
      "\tat java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\n",
      "\tat java.base/java.lang.reflect.Method.invoke(Method.java:568)\n",
      "\tat org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:48)\n",
      "\tat org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:189)\n",
      "\tat org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:57)\n",
      "\tat org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:51)\n",
      "\tat org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:62)\n",
      "\tat org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:185)\n",
      "\tat nextflow.script.params.BaseInParam.setFrom(BaseInParam.groovy:180)\n",
      "\tat nextflow.script.ProcessDef.run(ProcessDef.groovy:179)\n",
      "\tat nextflow.script.BindableDef.invoke_a(BindableDef.groovy:51)\n",
      "\tat nextflow.script.ComponentDef.invoke_o(ComponentDef.groovy:40)\n",
      "\tat nextflow.script.WorkflowBinding.invokeMethod(WorkflowBinding.groovy:102)\n",
      "\tat org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeOnDelegationObjects(ClosureMetaClass.java:408)\n",
      "\tat org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:350)\n",
      "\tat org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.callCurrent(PogoMetaClassSite.java:61)\n",
      "\tat org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:51)\n",
      "\tat org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:171)\n",
      "\tat Script_1dabf5cfd062ce56$_runScript_closure1$_closure2.doCall(Script_1dabf5cfd062ce56:144)\n",
      "\tat java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n",
      "\tat java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)\n",
      "\tat java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\n",
      "\tat java.base/java.lang.reflect.Method.invoke(Method.java:568)\n",
      "\tat org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:107)\n",
      "\tat groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)\n",
      "\tat org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:274)\n",
      "\tat groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1030)\n",
      "\tat groovy.lang.Closure.call(Closure.java:427)\n",
      "\tat groovy.lang.Closure.call(Closure.java:406)\n",
      "\tat nextflow.script.WorkflowDef.run0(WorkflowDef.groovy:204)\n",
      "\tat nextflow.script.WorkflowDef.run(WorkflowDef.groovy:188)\n",
      "\tat nextflow.script.BindableDef.invoke_a(BindableDef.groovy:51)\n",
      "\tat nextflow.script.IterableDef$invoke_a.call(Unknown Source)\n",
      "\tat org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)\n",
      "\tat org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)\n",
      "\tat org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:139)\n",
      "\tat nextflow.script.BaseScript.run0(BaseScript.groovy:183)\n",
      "\tat nextflow.script.BaseScript.run(BaseScript.groovy:192)\n",
      "\tat nextflow.script.ScriptParser.runScript(ScriptParser.groovy:236)\n",
      "\tat nextflow.script.ScriptRunner.run(ScriptRunner.groovy:242)\n",
      "\tat nextflow.script.ScriptRunner.execute(ScriptRunner.groovy:137)\n",
      "\tat nextflow.cli.CmdRun.run(CmdRun.groovy:372)\n",
      "\tat nextflow.cli.Launcher.run(Launcher.groovy:500)\n",
      "\tat nextflow.cli.Launcher.main(Launcher.groovy:672)\n",
      "Jan-31 21:02:12.998 [Actor Thread 8] ERROR nextflow.extension.OperatorImpl - @unknown\n",
      "groovy.lang.MissingMethodException: No signature of method: groovyx.gpars.dataflow.operator.PoisonPill.contains() is applicable for argument types: (String) values: [none]\n",
      "Possible solutions: toString(), toString(), notify()\n",
      "\tat org.codehaus.groovy.runtime.ScriptBytecodeAdapter.unwrap(ScriptBytecodeAdapter.java:70)\n",
      "\tat org.codehaus.groovy.runtime.ScriptBytecodeAdapter.unwrap(ScriptBytecodeAdapter.java:76)\n",
      "\tat org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:50)\n",
      "\tat org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)\n",
      "\tat org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:53)\n",
      "\tat org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:139)\n",
      "\tat nextflow.extension.DataflowHelper$3.afterStop(DataflowHelper.groovy:257)\n",
      "\tat groovyx.gpars.dataflow.operator.DataflowProcessor.fireAfterStop(DataflowProcessor.java:324)\n",
      "\tat groovyx.gpars.dataflow.operator.DataflowProcessorActor.afterStop(DataflowProcessorActor.java:59)\n",
      "\tat java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n",
      "\tat java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)\n",
      "\tat java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\n",
      "\tat java.base/java.lang.reflect.Method.invoke(Method.java:568)\n",
      "\tat org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:107)\n",
      "\tat groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)\n",
      "\tat groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1254)\n",
      "\tat groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1030)\n",
      "\tat org.codehaus.groovy.runtime.InvokerHelper.invokePojoMethod(InvokerHelper.java:1024)\n",
      "\tat org.codehaus.groovy.runtime.InvokerHelper.invokeMethod(InvokerHelper.java:1015)\n",
      "\tat groovyx.gpars.actor.Actor.callDynamic(Actor.java:369)\n",
      "\tat groovyx.gpars.actor.Actor.handleTermination(Actor.java:320)\n",
      "\tat groovyx.gpars.actor.AbstractLoopingActor$1.handleMessage(AbstractLoopingActor.java:86)\n",
      "\tat groovyx.gpars.util.AsyncMessagingCore.run(AsyncMessagingCore.java:132)\n",
      "\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n",
      "\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n",
      "\tat java.base/java.lang.Thread.run(Thread.java:833)\n"
     ]
    }
   ],
   "source": [
    "!cat .nextflow.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !ls ../../nf_data/test_atac_pipeline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !gsutil cp gs://eila-nextflow-bucket/test_atac_pipeline/sai_192_whitelist.txt ../../nf_data/test_atac_pipeline/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
