{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1943,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# start docker deamon in terminal\n",
    "# !sudo docker login\n",
    "# OR - becuase the jupyther with allow root\n",
    "# !docker login"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "TODO:\n",
    "\n",
    "when running with cloud. \n",
    "if you dont need to keep the execution logs, change the cloud settings to keep it for a short while\n",
    "Check that the cache if you need to set the cache to false (especially in plot process)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1944,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1945,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/hai-gcp-genomic-beb9207a40b8.json\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "json_path = os.path.join(os.path.dirname(os.getcwd()),\"hai-gcp-genomic-beb9207a40b8.json\")\n",
    "!ls $json_path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1946,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Activated service account credentials for: [eila-batch@hai-gcp-genomic.iam.gserviceaccount.com]\n"
     ]
    }
   ],
   "source": [
    "# # in the command line. same env\n",
    "import os\n",
    "json_path = os.path.join(os.path.dirname(os.getcwd()),\"hai-gcp-genomic-beb9207a40b8.json\")\n",
    "os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = json_path\n",
    "!gcloud auth activate-service-account eila-batch@hai-gcp-genomic.iam.gserviceaccount.com --key-file=$GOOGLE_APPLICATION_CREDENTIALS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1947,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                  Credentialed Accounts\n",
      "ACTIVE  ACCOUNT\n",
      "*       eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\n",
      "        eila@stanford.edu\n",
      "\n",
      "To set the active account, run:\n",
      "    $ gcloud config set account `ACCOUNT`\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'hai-gcp-genomic'"
      ]
     },
     "execution_count": 1947,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "!gcloud auth list\n",
    "project_name = !gcloud config get-value project\n",
    "project_name = project_name[0]\n",
    "project_name\n",
    "# gcloud config set account\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1948,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "os.environ[\"TOWER_ACCESS_TOKEN\"] = 'eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz'\n",
    "os.environ[\"TOWER_WORKSPACE_ID\"]='247507587970818'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1949,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "247507587970818\n"
     ]
    }
   ],
   "source": [
    "!echo $TOWER_ACCESS_TOKEN\n",
    "!echo $TOWER_WORKSPACE_ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1950,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1951,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/bin'"
      ]
     },
     "execution_count": 1951,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# !chmod +x ./bin/scrna_qc_metrics.py\n",
    "import os\n",
    "bin_fld = os.path.join(os.getcwd(),\"bin\")\n",
    "bin_fld"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1952,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "!chmod +x {os.path.join(bin_fld, \"scrna_plot_atac_qc_metrics.R\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"run_tabix.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"plot_insert_size_hist.py\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"atac_generate_merged_barcode_metadata.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"create_sample_filtered_fragment_file.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"compute_tss_enrichment_bulk.py\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"barcode_rank_functions.R\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"plot_insert_size_hist.py\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"compute_tss_enrichment_snapatac2.py\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"add_subpool_to_fragments_table.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"add_subpool_to_summary_file.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"filter_script.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"merge_chromap_logs.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"generate_insert_size_histogram_data.sh\")}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1953,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# # TODO: For a new user: run these commands with your service account for the execution. I am using eila-batch@...\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/iam.serviceAccountUser\"\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/batch.jobsEditor\"\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/logging.viewer\"\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1954,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !nextflow -self-update"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1955,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  Version: 23.10.1 build 5891\n",
      "  Created: 12-01-2024 22:01 UTC (17:01 EDT)\n",
      "  System: Mac OS X 14.3\n",
      "  Runtime: Groovy 3.0.19 on OpenJDK 64-Bit Server VM 17.0.8+7-LTS\n",
      "  Encoding: UTF-8 (UTF-8)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!nextflow info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1956,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "\n",
    "# Get the current working directory\n",
    "current_directory = os.getcwd()\n",
    "\n",
    "# Navigate two folders above\n",
    "two_folders_above = os.path.abspath(os.path.join(current_directory, '..', '..'))\n",
    "\n",
    "nf_data_base_path = os.path.join(two_folders_above,'nf_data')\n",
    "# Print the result\n",
    "print(nf_data_base_path)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1957,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data'"
      ]
     },
     "execution_count": 1957,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nf_data_base_path\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1958,
   "metadata": {},
   "outputs": [],
   "source": [
    "nf_data_project=\"test_atac_pipeline\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1959,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/local_atac_fastqs_files.tsv'"
      ]
     },
     "execution_count": 1959,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.path.exists(os.path.join(nf_data_base_path,nf_data_project,\"local_atac_fastqs_files.tsv\"))\n",
    "os.path.join(nf_data_base_path,nf_data_project,\"local_atac_fastqs_files.tsv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1960,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "\n",
    "# Write the configuration file\n",
    "with open(\"nextflow_atac.config\", \"w\") as config_file:\n",
    "    config_file.write(f\"\"\"\n",
    "manifest {{\n",
    "  description = 'A nexflow pipeline for fg single cell atac' \n",
    "  mainScript = 'nf_wf_fg_singlecell_atac.nf'\n",
    "}}\n",
    "\n",
    "profiles {{\n",
    "  google_batch_profile {{\n",
    "    workDir = 'gs://eila-nextflow-bucket/work/ATAC'\n",
    "    includeConfig 'conf/docker-google-batch.config'\n",
    "    docker.enabled = true\n",
    "    wave.enabled = true\n",
    "    fusion.enabled = true\n",
    "    process.scratch = false\n",
    "    executor = 'google-batch'\n",
    "    docker.envWhitelist=['ENV_SYNAPSE_TOKEN','TOWER_ACCESS_TOKEN']\n",
    "    google.project = '{project_name}'\n",
    "    google.location = 'us-central1'\n",
    "    google.batch.spot = true\n",
    "    google.batch.serviceAccountEmail='eila-batch@hai-gcp-genomic.iam.gserviceaccount.com'\n",
    "  }}\n",
    "}}\n",
    "\n",
    "// Define ENV_SYNAPSE_TOKEN as a parameter\n",
    "params.FASTQS_SPEC_CH = '{nf_data_base_path}/{nf_data_project}/local_atac_fastqs_files.tsv'\n",
    "// CHROMAP\n",
    "// 6a\n",
    "params.CHROMAP_IDX='gs://eila-nextflow-bucket/test_atac_pipeline/ref.index'\n",
    "// 6b:\n",
    "// FASTA file\n",
    "params.CHROMAP_READ_LENGTH = 2000\n",
    "params.CHROMAP_BC_ERROR_THRESHOLD = 2\n",
    "params.CHROMAP_BC_PROBABILITY_THRESHOLD = 0.9\n",
    "params.CHROMAP_READ_FORMAT = 'bc:0:-1,r1:0:-1,r2:0:-1'\n",
    "params.CHROMAP_DROP_REPETITIVE_READS = 4\n",
    "params.CHROMAP_QUALITY_THRESHOLD = 0\n",
    "params.CHROMAP_MERGE_LOGS_SCRIPT = \"merge_chromap_logs.sh\"\n",
    "params.CHROMAP_FRAGMENTS_HISTOGRAM_DATA_SCRIPT=\"generate_insert_size_histogram_data.sh\"\n",
    "params.CHROMAP_GENOME_REFERENCE_FASTA='{nf_data_base_path}/{nf_data_project}/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz'\n",
    "\n",
    "// CHROMAP DEBUG: load output files \n",
    "params.chromap_filter_fragments_tsv ='{nf_data_base_path}/{nf_data_project}/chromap_out_debug/ATAC_BMMC-single-donor.atac.filter.fragments.tsv'\n",
    "params.barcode_summary_csv ='{nf_data_base_path}/{nf_data_project}/chromap_out_debug/BMMC-single-donor.atac.align.barcode.summary.csv'\n",
    "params.chromap_alignment_log_out = '{nf_data_base_path}/{nf_data_project}/chromap_out_debug/BMMC-single-donor.atac.align.k4.hg38.log.txt'\n",
    "\n",
    "\n",
    "\n",
    "params.TSV_ADD_SUBPOOL_SCRIPT = 'add_subpool_to_fragments_table.sh'\n",
    "params.CSV_ADD_SUBPOOL_SCRIPT = 'add_subpool_to_summary_file.sh'\n",
    "params.FILTER_FRAGMENTS = 'filter_script.sh'\n",
    "params.ATAC_FRAGMENTS_MIN_FRAG_CUTOFF=10\n",
    "\n",
    "// Other steps\n",
    "\n",
    "params.GENOME_GZ_GTF='http://ftp.ensembl.org/pub/release-109/gtf/homo_sapiens/Homo_sapiens.GRCh38.109.gtf.gz'\n",
    "params.ENV_SYNAPSE_TOKEN = \"nextflow secrets get SYNAPSE\".execute().text.trim()\n",
    "\n",
    "\n",
    "params.MAPQ_THRESHOLD = false\n",
    "params.BARCODE_TRANSLATE = 'path'\n",
    "params.ATAC_FRAGMENTS_CUTOFF = 4\n",
    "params.RUN_TABIX_SCRIPT='run_tabix.sh'\n",
    "params.ATAC_TSS_BASES_FLANK = 2000\n",
    "params.ATAC_TSS_COL_WITH_STRANDS_INFO = 4\n",
    "params.ATAC_TSS_SMOOTHING_WINDOW_SIZE = 20\n",
    "params.ATAC_TSS_REGION_BED_FILE='{nf_data_base_path}/{nf_data_project}/hg38.synapse_transcripts.TSS.bed'\n",
    "params.ATAC_TSS_BULK_CALCULATION_SCRIPT='compute_tss_enrichment_bulk.py'\n",
    "params.ATAC_TSS_SNAPATAC2_CALCULATION_SCRIPT='compute_tss_enrichment_snapatac2.py'\n",
    "params.ATAC_TSS_SNAPATAC2_GENES_GTF='{nf_data_base_path}/{nf_data_project}/genes.gtf'\n",
    "params.ATAC_TSS_SNAPATAC2_MIN_FRAG_CUTOFF = 10\n",
    "\n",
    "params.ATAC_INSERT_SIZE_PLOT_SCRIPT='plot_insert_size_hist.py'\n",
    "\n",
    "// params.SC_ATAC_QC_PLOT_SCRIPT='scrna_plot_atac_qc_metrics.R'\n",
    "// params.SC_ATAC_QC_PLOT_HELPER_SCRIPT='barcode_rank_functions.R'\n",
    "// debug file\n",
    "// params.SC_ATAC_QC_BARCODE_METADATA_FILE='{nf_data_base_path}/{nf_data_project}/barcode_metadata.txt'\n",
    "params.SC_ATAC_QC_FRAGMENT_CUTOFF=10\n",
    "params.SC_ATAC_QC_BARCODE_OUTPUT_FILE='sc_atac_qc_barcode_output_file.png'\n",
    "    \n",
    "// params.SC_ATAC_BARCODE_METADATA_SCRIPT='atac_generate_merged_barcode_metadata.sh'\n",
    "// DEBUG: get the data from a file\n",
    "// params.SC_ATAC_BARCODE_METADATA_FILTERED_BARCODE_STATS='{nf_data_base_path}nf_wf_fg_singlecell/filtered_barcode_stats.csv'\n",
    "// params.SC_ATAC_BARCODE_METADATA_TSS_ENRICHMENT_BARCODE_STATS='{nf_data_base_path}nf_wf_fg_singlecell/tss_enrichment_barcode_stats.csv'\n",
    "\n",
    "// params.SC_ATAC_GENERATE_BARCODE_RANK_PLOT_PKR='PKR123'\n",
    "\n",
    "// params.SC_JOINT_CELL_BARCODE_PLOT_SCRTIP_PY='joint_cell_plotting.py'\n",
    "// params.SC_JOINT_CELL_BARCODE_PLOT_SCRTIP_R='joint_cell_plotting_density.R'\n",
    "\"\"\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1961,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !rm -r /Users/eilaarich-landkof-stanford/Documents/Code/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/work\n",
    "# !rm -r /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/work\n",
    "# !gsutil rm -r gs://eila-nextflow-bucket/work/ATAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1962,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !rm .nextflow*\n",
    "\n",
    "# !ls /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/local_atac_fastqs_files.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1963,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "N E X T F L O W  ~  version 23.10.1\n",
      "Launching `nf_wf_fg_singlecell_atac.nf` [hungry_leibniz] DSL2 - revision: d86ddea1b4\n",
      "/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/local_atac_fastqs_files.tsv\n",
      "eyJ0eXAiOiJKV1QiLCJraWQiOiJXN05OOldMSlQ6SjVSSzpMN1RMOlQ3TDc6M1ZYNjpKRU9VOjY0NFI6VTNJWDo1S1oyOjdaQ0s6RlBUSCIsImFsZyI6IlJTMjU2In0.eyJhY2Nlc3MiOnsic2NvcGUiOlsidmlldyIsImRvd25sb2FkIl0sIm9pZGNfY2xhaW1zIjp7fX0sInRva2VuX3R5cGUiOiJQRVJTT05BTF9BQ0NFU1NfVE9LRU4iLCJpc3MiOiJodHRwczovL3JlcG8tcHJvZC5wcm9kLnNhZ2ViYXNlLm9yZy9hdXRoL3YxIiwiYXVkIjoiMCIsIm5iZiI6MTY5NzgxODkxNiwiaWF0IjoxNjk3ODE4OTE2LCJqdGkiOiI0MTIzIiwic3ViIjoiMzQ3NDU3MSJ9.n0cugYMAVzYZ8mMMc7CjcqqEV7FB-K-oKUDT45SvM6KmbCpb_v3lZ2gg3xXi1tubWH0__wpnbJ7cYrGEtOyt4vXAPkMiEtZFGqS0MFKp_h-mHHH4pnACSQQYEy1zdyqFn9WRE7ipjWg4yYSbq84N7J3gzi8VKmiOAmuIGKaJaS0snkUjtJEVQH-0vZqWs1nKgVRWIZv31Bo8vuDzWpBrXD-zRT_uQW9l_I9fU50o1AqyFhNe_gbty_Z6imb8VT33wMXiDgJh15lG_M70PXU15v3egDXLgw1dGj32kj1nKanfqBVqbMSzgzPAG037p3YH_Q556BLceFBBb5Jaej5JVA\n",
      "after genome_fasta_ch\n",
      "[-        ] process > run_zcat                       -\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip           -\u001b[K\n",
      "[-        ] process > run_add_subpool_prefix_to_f... -\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "\u001b[19A\n",
      "[-        ] process > run_zcat                       -\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip           -\u001b[K\n",
      "[-        ] process > run_add_subpool_prefix_to_f... -\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_gzip_on_genes_gtf          -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "\u001b[39A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/22OO1YN5MwXspw\u001b[22;39m\u001b[K\n",
      "[-        ] process > run_zcat                       -\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip           -\u001b[K\n",
      "[-        ] process > run_add_subpool_prefix_to_f... -\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_gzip_on_genes_gtf          -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "\u001b[40A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/22OO1YN5MwXspw\u001b[22;39m\u001b[K\n",
      "[-        ] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip           -\u001b[K\n",
      "[-        ] process > run_add_subpool_prefix_to_f... -\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_gzip_on_genes_gtf          [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "\u001b[40A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/22OO1YN5MwXspw\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (1)\u001b[K\n",
      "[a5/e1fc3c] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip           -\u001b[K\n",
      "[-        ] process > run_add_subpool_prefix_to_f... -\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_gzip_on_genes_gtf          [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "\u001b[41A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/22OO1YN5MwXspw\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (1)\u001b[K\n",
      "[a5/e1fc3c] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip           [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_add_subpool_prefix_to_f... -\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_gzip_on_genes_gtf          [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "\u001b[41A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/22OO1YN5MwXspw\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (1)\u001b[K\n",
      "[a5/e1fc3c] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip           [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_add_subpool_prefix_to_f... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_gzip_on_genes_gtf          [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "\u001b[41A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/22OO1YN5MwXspw\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (2)\u001b[K\n",
      "[a5/e1fc3c] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip           [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_add_subpool_prefix_to_f... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[d1/fb240a] process > run_gzip_on_genes_gtf (1)      [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "\u001b[41A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/22OO1YN5MwXspw\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (3)\u001b[K\n",
      "[a5/e1fc3c] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[8f/b425fe] process > run_whitelist_gunzip (1)       [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_add_subpool_prefix_to_f... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[d1/fb240a] process > run_gzip_on_genes_gtf (1)      [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "\u001b[41A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/22OO1YN5MwXspw\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (4)\u001b[K\n",
      "[a5/e1fc3c] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[8f/b425fe] process > run_whitelist_gunzip (1)       [  0%] 0 of 1\u001b[K\n",
      "[df/b0d418] process > run_add_subpool_prefix_to_f... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[d1/fb240a] process > run_gzip_on_genes_gtf (1)      [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n"
     ]
    }
   ],
   "source": [
    "!nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_atac.config -profile google_batch_profile -process.echo -with-tower -with-dag flowchart.png -resume\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_vm.config -profile google_batch_profile\n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_vm.config -profile google_batch_profile\n",
    "# !nextflow run nf_\n",
    "# \n",
    "# wf_fg_singlecell_atac.nf -c nextflow_vm_anvil.config -profile google_batch_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_local.config -profile google_batch_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_local.config -profile docker_profile\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Feb-05 17:55:50.428 [main] DEBUG nextflow.cli.Launcher - $> nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_atac.config -profile google_batch_profile -process.echo -with-tower -resume\n",
      "Feb-05 17:55:50.576 [main] INFO  nextflow.cli.CmdRun - N E X T F L O W  ~  version 23.10.1\n",
      "Feb-05 17:55:50.593 [main] DEBUG nextflow.plugin.PluginsFacade - Setting up plugin manager > mode=prod; embedded=false; plugins-dir=/Users/eilaarich-landkof-stanford/.nextflow/plugins; core-plugins: nf-amazon@2.1.4,nf-azure@1.3.3,nf-cloudcache@0.3.0,nf-codecommit@0.1.5,nf-console@1.0.6,nf-ga4gh@1.1.0,nf-google@1.8.3,nf-tower@1.6.3,nf-wave@1.0.1\n",
      "Feb-05 17:55:50.600 [main] INFO  o.pf4j.DefaultPluginStatusProvider - Enabled plugins: []\n",
      "Feb-05 17:55:50.601 [main] INFO  o.pf4j.DefaultPluginStatusProvider - Disabled plugins: []\n",
      "Feb-05 17:55:50.603 [main] INFO  org.pf4j.DefaultPluginManager - PF4J version 3.4.1 in 'deployment' mode\n",
      "Feb-05 17:55:50.613 [main] INFO  org.pf4j.AbstractPluginManager - No plugins\n",
      "Feb-05 17:55:50.629 [main] DEBUG nextflow.config.ConfigBuilder - User config file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Feb-05 17:55:50.629 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Feb-05 17:55:50.647 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `google_batch_profile`\n",
      "Feb-05 17:55:52.312 [main] DEBUG nextflow.config.ConfigBuilder - Available config profiles: [google_batch_profile]\n",
      "Feb-05 17:55:52.339 [main] DEBUG nextflow.cli.CmdRun - Applied DSL=2 by global default\n",
      "Feb-05 17:55:52.364 [main] INFO  nextflow.cli.CmdRun - Launching `nf_wf_fg_singlecell_atac.nf` [disturbed_williams] DSL2 - revision: 478211dda1\n",
      "Feb-05 17:55:52.364 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins default=[nf-google@1.8.3]\n",
      "Feb-05 17:55:52.365 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins resolved requirement=[nf-google@1.8.3, nf-tower@1.6.3, nf-wave@1.0.1]\n",
      "Feb-05 17:55:52.365 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-google version: 1.8.3\n",
      "Feb-05 17:55:52.379 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-google@1.8.3' resolved\n",
      "Feb-05 17:55:52.379 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-google@1.8.3'\n",
      "Feb-05 17:55:52.417 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-google@1.8.3\n",
      "Feb-05 17:55:52.417 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-tower version: 1.6.3\n",
      "Feb-05 17:55:52.419 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-tower@1.6.3' resolved\n",
      "Feb-05 17:55:52.419 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-tower@1.6.3'\n",
      "Feb-05 17:55:52.425 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-tower@1.6.3\n",
      "Feb-05 17:55:52.425 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-wave version: 1.0.1\n",
      "Feb-05 17:55:52.428 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-wave@1.0.1' resolved\n",
      "Feb-05 17:55:52.429 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-wave@1.0.1'\n",
      "Feb-05 17:55:52.436 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-wave@1.0.1\n",
      "Feb-05 17:55:52.447 [main] DEBUG n.secret.LocalSecretsProvider - Secrets store: /Users/eilaarich-landkof-stanford/.nextflow/secrets/store.json\n",
      "Feb-05 17:55:52.450 [main] DEBUG nextflow.secret.SecretsLoader - Discovered secrets providers: [nextflow.secret.LocalSecretsProvider@79518e00] - activable => nextflow.secret.LocalSecretsProvider@79518e00\n",
      "Feb-05 17:55:52.508 [main] DEBUG nextflow.Session - Session UUID: ab09407f-2f8c-43e1-855a-210d7286a110\n",
      "Feb-05 17:55:52.508 [main] DEBUG nextflow.Session - Run name: disturbed_williams\n",
      "Feb-05 17:55:52.509 [main] DEBUG nextflow.Session - Executor pool size: 8\n",
      "Feb-05 17:55:52.642 [main] DEBUG nextflow.file.FilePorter - File porter settings maxRetries=3; maxTransfers=50; pollTimeout=null\n",
      "Feb-05 17:55:52.644 [main] DEBUG nextflow.util.ThreadPoolBuilder - Creating thread pool 'FileTransfer' minSize=10; maxSize=24; workQueue=LinkedBlockingQueue[10000]; allowCoreThreadTimeout=false\n",
      "Feb-05 17:55:52.647 [main] DEBUG nextflow.config.ConfigBuilder - User config file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Feb-05 17:55:52.647 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Feb-05 17:55:52.648 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `google_batch_profile`\n",
      "Feb-05 17:55:53.751 [main] DEBUG nextflow.config.ConfigBuilder - Available config profiles: [google_batch_profile]\n",
      "Feb-05 17:55:53.786 [main] DEBUG nextflow.cli.CmdRun - \n",
      "  Version: 23.10.1 build 5891\n",
      "  Created: 12-01-2024 22:01 UTC (17:01 EDT)\n",
      "  System: Mac OS X 14.3\n",
      "  Runtime: Groovy 3.0.19 on OpenJDK 64-Bit Server VM 17.0.8+7-LTS\n",
      "  Encoding: UTF-8 (UTF-8)\n",
      "  Process: 47688@Eilas-Air.fios-router.home [192.168.1.48]\n",
      "  CPUs: 8 - Mem: 16 GB (179 MB) - Swap: 3 GB (1.1 GB)\n",
      "Feb-05 17:55:53.831 [main] DEBUG nextflow.Session - Work-dir: gs://eila-nextflow-bucket/work/ATAC [Mac OS X]\n",
      "Feb-05 17:55:53.855 [main] DEBUG nextflow.executor.ExecutorFactory - Extension executors providers=[GoogleLifeSciencesExecutor, GoogleBatchExecutor]\n",
      "Feb-05 17:55:53.865 [main] DEBUG nextflow.Session - Observer factory: DefaultObserverFactory\n",
      "Feb-05 17:55:53.867 [main] DEBUG nextflow.Session - Observer factory: WaveFactory\n",
      "Feb-05 17:55:53.867 [main] DEBUG io.seqera.wave.plugin.WaveFactory - Detected Fusion enabled -- Enabling bundle project resources -- Disabling upload of remote bin directory\n",
      "Feb-05 17:55:53.882 [main] DEBUG i.s.wave.plugin.config.WaveConfig - Wave strategy not specified - using default: [container, dockerfile, conda, spack]\n",
      "Feb-05 17:55:53.888 [main] DEBUG io.seqera.wave.plugin.WaveClient - Wave config: WaveConfig(enabled:true, endpoint:https://wave.seqera.io, containerConfigUrl:[], tokensCacheMaxDuration:30m, condaOpts:CondaOpts(mambaImage=mambaorg/micromamba:1.5.1; basePackages=conda-forge::procps-ng, commands=null), spackOpts:SpackOpts(basePackages=null, commands=null), strategy:[container, dockerfile, conda, spack], bundleProjectResources:true, buildRepository:null, cacheRepository:null, retryOpts:RetryOpts(delay:450ms, maxDelay:1m 30s, maxAttempts:10, jitter:0.25), httpClientOpts:HttpOpts(), freezeMode:null)\n",
      "Feb-05 17:55:54.025 [main] DEBUG nextflow.Session - Observer factory: TowerFactory\n",
      "Feb-05 17:55:54.083 [main] DEBUG nextflow.cache.CacheFactory - Using Nextflow cache factory: nextflow.cache.DefaultCacheFactory\n",
      "Feb-05 17:55:54.094 [main] DEBUG nextflow.util.CustomThreadPool - Creating default thread pool > poolSize: 9; maxThreads: 1000\n",
      "Feb-05 17:55:54.178 [main] DEBUG nextflow.Session - Session start\n",
      "Feb-05 17:55:54.192 [main] DEBUG io.seqera.tower.plugin.TowerClient - Creating Tower observer -- endpoint=https://api.tower.nf; requestInterval=1s; aliveInterval=1m; maxRetries=5; backOffBase=3; backOffDelay=250\n",
      "Feb-05 17:55:55.080 [main] DEBUG nextflow.script.ScriptRunner - > Launching execution\n",
      "Feb-05 17:55:56.207 [main] INFO  nextflow.script.BaseScript - /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/local_atac_fastqs_files.tsv\n",
      "Feb-05 17:55:56.208 [main] INFO  nextflow.script.BaseScript - eyJ0eXAiOiJKV1QiLCJraWQiOiJXN05OOldMSlQ6SjVSSzpMN1RMOlQ3TDc6M1ZYNjpKRU9VOjY0NFI6VTNJWDo1S1oyOjdaQ0s6RlBUSCIsImFsZyI6IlJTMjU2In0.eyJhY2Nlc3MiOnsic2NvcGUiOlsidmlldyIsImRvd25sb2FkIl0sIm9pZGNfY2xhaW1zIjp7fX0sInRva2VuX3R5cGUiOiJQRVJTT05BTF9BQ0NFU1NfVE9LRU4iLCJpc3MiOiJodHRwczovL3JlcG8tcHJvZC5wcm9kLnNhZ2ViYXNlLm9yZy9hdXRoL3YxIiwiYXVkIjoiMCIsIm5iZiI6MTY5NzgxODkxNiwiaWF0IjoxNjk3ODE4OTE2LCJqdGkiOiI0MTIzIiwic3ViIjoiMzQ3NDU3MSJ9.n0cugYMAVzYZ8mMMc7CjcqqEV7FB-K-oKUDT45SvM6KmbCpb_v3lZ2gg3xXi1tubWH0__wpnbJ7cYrGEtOyt4vXAPkMiEtZFGqS0MFKp_h-mHHH4pnACSQQYEy1zdyqFn9WRE7ipjWg4yYSbq84N7J3gzi8VKmiOAmuIGKaJaS0snkUjtJEVQH-0vZqWs1nKgVRWIZv31Bo8vuDzWpBrXD-zRT_uQW9l_I9fU50o1AqyFhNe_gbty_Z6imb8VT33wMXiDgJh15lG_M70PXU15v3egDXLgw1dGj32kj1nKanfqBVqbMSzgzPAG037p3YH_Q556BLceFBBb5Jaej5JVA\n",
      "Feb-05 17:55:56.273 [main] INFO  nextflow.script.BaseScript - after genome_fasta_ch\n",
      "Feb-05 17:55:56.311 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:zcat` matches labels `zcat` for process with name run_zcat\n",
      "Feb-05 17:55:56.314 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-05 17:55:56.314 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-05 17:55:56.316 [main] DEBUG nextflow.executor.Executor - [warm up] executor > google-batch (fusion enabled)\n",
      "Feb-05 17:55:56.320 [main] DEBUG n.processor.TaskPollingMonitor - Creating task monitor for executor 'google-batch' > capacity: 1000; pollInterval: 10s; dumpInterval: 5m \n",
      "Feb-05 17:55:56.321 [main] DEBUG n.processor.TaskPollingMonitor - >>> barrier register (monitor: google-batch)\n",
      "Feb-05 17:55:56.325 [main] DEBUG nextflow.cloud.google.GoogleOpts - Google auth via application credentials file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/hai-gcp-genomic-beb9207a40b8.json\n",
      "Feb-05 17:55:56.329 [main] DEBUG n.c.google.batch.GoogleBatchExecutor - [GOOGLE BATCH] Executor config=BatchConfig[googleOpts=GoogleOpts(projectId:hai-gcp-genomic, credsFile:/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/hai-gcp-genomic-beb9207a40b8.json, location:us-central1, enableRequesterPaysBuckets:false, httpConnectTimeout:1m, httpReadTimeout:1m, credentials:ServiceAccountCredentials{clientId=111364348320962814542, clientEmail=eila-batch@hai-gcp-genomic.iam.gserviceaccount.com, privateKeyId=beb9207a40b8b2085c7f48c3fcf8c2342180d471, transportFactoryClassName=com.google.auth.oauth2.OAuth2Utils$DefaultHttpTransportFactory, tokenServerUri=https://oauth2.googleapis.com/token, scopes=[https://www.googleapis.com/auth/cloud-platform], defaultScopes=[], serviceAccountUser=null, quotaProjectId=null, lifetime=3600, useJwtAccessWithScope=false, defaultRetriesEnabled=true})\n",
      "Feb-05 17:55:56.345 [main] DEBUG n.c.google.batch.client.BatchClient - [GOOGLE BATCH] Creating service client with config credentials\n",
      "Feb-05 17:55:57.014 [main] INFO  nextflow.script.BaseScript - after run_zcat \n",
      "Feb-05 17:55:57.022 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:gunzip` matches labels `gunzip` for process with name run_whitelist_gunzip\n",
      "Feb-05 17:55:57.023 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-05 17:55:57.023 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-05 17:55:57.026 [main] INFO  nextflow.script.BaseScript - after run_whitelist_gunzip\n",
      "Feb-05 17:55:57.027 [main] INFO  nextflow.script.BaseScript - after load chrompa mapping debug\n",
      "Feb-05 17:55:57.030 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:pool_fragments` matches labels `pool_fragments` for process with name run_add_subpool_prefix_to_fragment_table\n",
      "Feb-05 17:55:57.032 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-05 17:55:57.032 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-05 17:55:57.035 [main] INFO  nextflow.script.BaseScript - before call bgzip\n",
      "Feb-05 17:55:57.036 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:bgzip` matches labels `bgzip` for process with name run_bgzip_on_chromap_fragments_output\n",
      "Feb-05 17:55:57.037 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-05 17:55:57.037 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-05 17:55:57.039 [main] INFO  nextflow.script.BaseScript - after call bgzip\n",
      "Feb-05 17:55:57.039 [main] INFO  nextflow.script.BaseScript - before call run_tabix_chromap with tabix shell script\n",
      "Feb-05 17:55:57.041 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:tabix` matches labels `tabix` for process with name run_tabix_chromap\n",
      "Feb-05 17:55:57.042 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-05 17:55:57.042 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-05 17:55:57.043 [main] INFO  nextflow.script.BaseScript - after call run_tabix_chromap\n",
      "Feb-05 17:55:57.043 [main] INFO  nextflow.script.BaseScript - before call run_merge_chromap_logs\n",
      "Feb-05 17:55:57.045 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:merge_log` matches labels `merge_log` for process with name run_merge_chromap_logs\n",
      "Feb-05 17:55:57.046 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-05 17:55:57.046 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-05 17:55:57.049 [main] INFO  nextflow.script.BaseScript - after call run_merge_chromap_logs\n",
      "Feb-05 17:55:57.144 [main] INFO  nextflow.script.BaseScript - before run_add_subpool_prefix_to_summary_file\n",
      "Feb-05 17:55:57.147 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:pool_dictionary` matches labels `pool_dictionary` for process with name run_process_conversion_to_barcode\n",
      "Feb-05 17:55:57.147 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-05 17:55:57.147 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-05 17:55:57.150 [main] INFO  nextflow.script.BaseScript - after run_add_subpool_prefix_to_summary_file\n",
      "Feb-05 17:55:57.152 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:pool_filter` matches labels `pool_filter` for process with name run_filter_aligned_fragments\n",
      "Feb-05 17:55:57.153 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-05 17:55:57.153 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-05 17:55:57.157 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-05 17:55:57.157 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-05 17:55:57.159 [main] INFO  nextflow.script.BaseScript - before call run_calculate_tss_enrichment_bulk\n",
      "Feb-05 17:55:57.162 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:tss_bulk` matches labels `tss_bulk` for process with name run_calculate_tss_enrichment_bulk\n",
      "Feb-05 17:55:57.162 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-05 17:55:57.162 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-05 17:55:57.164 [main] INFO  nextflow.script.BaseScript - after call run_calculate_tss_enrichment_bulk\n",
      "Feb-05 17:55:57.165 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:gzip` matches labels `gzip` for process with name run_gzip_on_genes_gtf\n",
      "Feb-05 17:55:57.166 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-05 17:55:57.166 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-05 17:55:57.168 [main] INFO  nextflow.script.BaseScript - after run_gzip_on_genes_gtf\n",
      "Feb-05 17:55:57.168 [main] INFO  nextflow.script.BaseScript - before call run_calculate_tss_enrichment_snapatac2\n",
      "Feb-05 17:55:57.170 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:tss_snapatac2` matches labels `tss_snapatac2` for process with name run_calculate_tss_enrichment_snapatac2\n",
      "Feb-05 17:55:57.171 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-05 17:55:57.171 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-05 17:55:57.173 [main] INFO  nextflow.script.BaseScript - after call run_calculate_tss_enrichment_snapatac2\n",
      "Feb-05 17:55:57.173 [main] INFO  nextflow.script.BaseScript - before run_generate_insert_size_histogram_data\n",
      "Feb-05 17:55:57.174 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:zcat_insert` matches labels `zcat_insert` for process with name run_generate_insert_size_histogram_data\n",
      "Feb-05 17:55:57.175 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-05 17:55:57.175 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-05 17:55:57.176 [main] INFO  nextflow.script.BaseScript - after run_generate_insert_size_histogram_data\n",
      "Feb-05 17:55:57.176 [main] INFO  nextflow.script.BaseScript - before call run_generate_insert_size_plot\n",
      "Feb-05 17:55:57.177 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-05 17:55:57.177 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-05 17:55:57.179 [main] INFO  nextflow.script.BaseScript - after call run_generate_insert_size_plot\n",
      "Feb-05 17:55:57.179 [main] INFO  nextflow.script.BaseScript - before run_barcode_metadata_stats\n",
      "Feb-05 17:55:57.181 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-05 17:55:57.181 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-05 17:55:57.183 [main] INFO  nextflow.script.BaseScript - after run_barcode_metadata_stats\n",
      "Feb-05 17:55:57.184 [main] INFO  nextflow.script.BaseScript - before call run_generate_barcode_rank_plot\n",
      "Feb-05 17:55:57.184 [main] DEBUG nextflow.Session - Workflow process names [dsl2]: run_bgzip_on_chromap_fragments_output, run_seqspec_print, run_calculate_tss_enrichment_snapatac2, run_process_conversion_to_barcode, run_barcode_metadata_stats, run_tabix_chromap, run_calculate_tss_enrichment_bulk, run_generate_insert_size_plot, run_seqspec_test, run_tabix_no_singleton, run_chromap_map_to_idx, run_atac_merge_barcode_metadata, run_filter_aligned_fragments, run_seqspec_modify_rna, run_atac_barcode_rank_plot, run_gzip_on_genes_gtf, run_whitelist_gunzip, run_merge_chromap_logs, run_create_chromap_idx, run_generate_insert_size_histogram_data, run_zcat, run_seqspec_check, run_seqspec_index_rna_kb, run_chromap_test, run_seqspec_modify_atac, run_tabix_filtered_fragments, run_add_subpool_prefix_to_fragment_table, run_downloadFiles\n",
      "Feb-05 17:55:57.568 [main] INFO  io.seqera.tower.plugin.TowerClient - Monitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/5h2wtcbCQGrHJy\n",
      "Feb-05 17:55:57.569 [main] DEBUG nextflow.Session - Igniting dataflow network (24)\n",
      "Feb-05 17:55:57.576 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_zcat\n",
      "Feb-05 17:55:57.577 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_whitelist_gunzip\n",
      "Feb-05 17:55:57.578 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_add_subpool_prefix_to_fragment_table\n",
      "Feb-05 17:55:57.578 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_bgzip_on_chromap_fragments_output\n",
      "Feb-05 17:55:57.578 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_tabix_chromap\n",
      "Feb-05 17:55:57.578 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_merge_chromap_logs\n",
      "Feb-05 17:55:57.579 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_process_conversion_to_barcode\n",
      "Feb-05 17:55:57.579 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_filter_aligned_fragments\n",
      "Feb-05 17:55:57.579 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_tabix_no_singleton\n",
      "Feb-05 17:55:57.579 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_calculate_tss_enrichment_bulk\n",
      "Feb-05 17:55:57.579 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_gzip_on_genes_gtf\n",
      "Feb-05 17:55:57.579 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_calculate_tss_enrichment_snapatac2\n",
      "Feb-05 17:55:57.580 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_generate_insert_size_histogram_data\n",
      "Feb-05 17:55:57.580 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_generate_insert_size_plot\n",
      "Feb-05 17:55:57.580 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_barcode_metadata_stats\n",
      "Feb-05 17:55:57.581 [main] DEBUG nextflow.script.ScriptRunner - Parsed script files:\n",
      "  Script_30732e62620d0e07: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_tss.nf\n",
      "  Script_900e20e0f3b521d7: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_atac_generate_insert_size_plot.nf\n",
      "  Script_67e45395c241a45f: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_chromap_utils.nf\n",
      "  Script_9d4639c8cd883c70: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_download_url_files.nf\n",
      "  Script_bc539eb5853c51b8: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_gunzip.nf\n",
      "  Script_6d8835b1ac7eadd3: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_barcode_subpool.nf\n",
      "  Script_280f31ae5ae5ed4d: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_tabix.nf\n",
      "  Script_5df81d0ae421efa0: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_bgzip.nf\n",
      "  Script_e112646bcdb899f0: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_seqspec_utils.nf\n",
      "  Script_8081eedda99eaa43: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_merge_logs.nf\n",
      "  Script_48127624ae4a6640: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_zcat.nf\n",
      "  Script_103349f59593da4c: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_atac_generate_metadata_annoation.nf\n",
      "  Script_7f43a272cfe3ae1b: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_generate_merged_barcode_metadata.nf\n",
      "  Script_be6b2aa7f6b3d9f7: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nf_wf_fg_singlecell_atac.nf\n",
      "  Script_34d6d3540f17eeb9: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_gzip.nf\n",
      "Feb-05 17:55:57.581 [main] DEBUG nextflow.script.ScriptRunner - > Awaiting termination \n",
      "Feb-05 17:55:57.581 [main] DEBUG nextflow.Session - Session await\n",
      "Feb-05 17:55:58.832 [FileTransfer-1] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\n",
      "Feb-05 17:55:58.851 [Actor Thread 3] DEBUG i.s.wave.plugin.config.WaveConfig - Wave strategy not specified - using default: [container, dockerfile, conda, spack]\n",
      "Feb-05 17:55:58.852 [Actor Thread 3] DEBUG io.seqera.wave.plugin.WaveClient - Wave config: WaveConfig(enabled:true, endpoint:https://wave.seqera.io, containerConfigUrl:[], tokensCacheMaxDuration:30m, condaOpts:CondaOpts(mambaImage=mambaorg/micromamba:1.5.1; basePackages=conda-forge::procps-ng, commands=null), spackOpts:SpackOpts(basePackages=null, commands=null), strategy:[container, dockerfile, conda, spack], bundleProjectResources:true, buildRepository:null, cacheRepository:null, retryOpts:RetryOpts(delay:450ms, maxDelay:1m 30s, maxAttempts:10, jitter:0.25), httpClientOpts:HttpOpts(), freezeMode:null)\n",
      "Feb-05 17:55:58.859 [Actor Thread 3] DEBUG io.seqera.wave.plugin.WaveClient - Wave assets bundle bin resources: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/bin\n",
      "Feb-05 17:55:59.147 [Actor Thread 3] DEBUG io.seqera.wave.plugin.WaveClient - Wave request container config: https://fusionfs.seqera.io/releases/v2.2-amd64.json\n",
      "Feb-05 17:55:59.225 [FileTransfer-2] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/genes.gtf at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/eb/6412854cc6c199859cdfbe280f86c9/genes.gtf\n",
      "Feb-05 17:55:59.465 [Actor Thread 3] DEBUG io.seqera.wave.plugin.WaveClient - Wave container config response: [200] {\n",
      "  \"entrypoint\": [ \"/usr/bin/fusion\" ],\n",
      "  \"layers\": [\n",
      "    {\n",
      "      \"location\": \"https://fusionfs.seqera.io/releases/pkg/2/2/12/fusion-amd64.tar.gz\",\n",
      "      \"gzipDigest\": \"sha256:6ccfd8f431db2874e82e27d61b6d31b95d7dddce7e810c5f8c95438a0d03c066\",\n",
      "      \"gzipSize\": 30262575,\n",
      "      \"tarDigest\": \"sha256:60391bebe0cc5d48b0875854f0c93c0d0adc25b5007fdb1b2efc2a431cfeef6e\",\n",
      "      \"skipHashing\": true\n",
      "    }\n",
      "  ]\n",
      "}\n",
      "\n",
      "Feb-05 17:55:59.531 [Actor Thread 3] DEBUG io.seqera.wave.plugin.WaveClient - Wave request: https://wave.seqera.io/container-token; attempt=1 - request: SubmitContainerTokenRequest(towerAccessToken:eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz, towerRefreshToken:null, towerWorkspaceId:247507587970818, towerEndpoint:https://api.tower.nf, workflowId:null, containerImage:eilalan/utils_env:latest, containerFile:null, containerConfig:ContainerConfig(entrypoint:[/usr/bin/fusion], cmd:null, env:null, workingDir:null, layers:[ContainerLayer[location=data:H4sIAAAAAAAA/+19bXvb...; tarDigest=sha256:4400c6f035b4477f402328f733d2822cd0ebde3fd5fce4ae9ffe1baa6901cd2f; gzipDigest=sha256:84e9be039f2abdab76c5bfc554aac8b9eca9fafca04189074161ba46fb842008; gzipSize=11650], ContainerLayer[location=https://fusionfs.seqera.io/releases/pkg/2/2/12/fusion-amd64.tar.gz; tarDigest=sha256:60391bebe0cc5d48b0875854f0c93c0d0adc25b5007fdb1b2efc2a431cfeef6e; gzipDigest=sha256:6ccfd8f431db2874e82e27d61b6d31b95d7dddce7e810c5f8c95438a0d03c066; gzipSize=30262575]]), condaFile:null, spackFile:null, containerPlatform:linux/amd64, buildRepository:null, cacheRepository:null, timestamp:2024-02-05T17:55:59.530532-05:00, fingerprint:877b4461e46488589d4dbe24853979a8, freeze:false, format:null, dryRun:false)\n",
      "Feb-05 17:56:00.300 [Actor Thread 3] DEBUG io.seqera.wave.plugin.WaveClient - Wave response: statusCode=200; body={\"containerToken\":\"d66f42b1607b\",\"targetImage\":\"wave.seqera.io/wt/d66f42b1607b/eilalan/utils_env:latest\",\"expiration\":\"2024-02-07T10:56:00.348141406Z\",\"containerImage\":\"docker.io/eilalan/utils_env:latest\"}\n",
      "Feb-05 17:56:01.099 [FileTransfer-3] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/6f/e3ee295a7f5364c39365c48e8bd0b0/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\n",
      "Feb-05 17:56:01.127 [Actor Thread 3] INFO  nextflow.processor.TaskProcessor - [43/5c5a8d] Cached process > run_zcat\n",
      "Feb-05 17:56:01.147 [Actor Thread 1] INFO  nextflow.processor.TaskProcessor - [12/2cbbfd] Cached process > run_gzip_on_genes_gtf (1)\n",
      "Feb-05 17:56:01.164 [FileTransfer-3] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/chromap_out_debug/BMMC-single-donor.atac.align.barcode.summary.csv at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/83/230e2c5cb965e17c572d46066ce07f/BMMC-single-donor.atac.align.barcode.summary.csv\n",
      "Feb-05 17:56:01.164 [FileTransfer-6] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/dc/65201d13e12dffd8779bbc935604a6/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\n",
      "Feb-05 17:56:01.168 [FileTransfer-4] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/c5/cc9149d8d21bb423337888a9dc77ea/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\n",
      "Feb-05 17:56:01.176 [FileTransfer-2] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/chromap_out_debug/ATAC_BMMC-single-donor.atac.filter.fragments.tsv at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/12/a7aa6d9e76c7234da48d36951f96c6/ATAC_BMMC-single-donor.atac.filter.fragments.tsv\n",
      "Feb-05 17:56:01.182 [FileTransfer-9] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/na.na at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/f4/bab8ffeff41eecffb2ddcf013b4121/na.na\n",
      "Feb-05 17:56:01.191 [FileTransfer-1] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/na1.na at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/38/b73884d031cf00d6089d843ba0f965/na1.na\n",
      "Feb-05 17:56:01.191 [FileTransfer-5] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/4a/7b5d74b3a4903977d7316c5f02ef84/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\n",
      "Feb-05 17:56:01.203 [FileTransfer-7] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/d5/12b6921f983d2326fa1e91570a96e7/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\n",
      "Feb-05 17:56:01.203 [FileTransfer-10] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/sai_192_whitelist.txt at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/09/7ba70bbeac00bcd6547ad0c09655c7/sai_192_whitelist.txt\n",
      "Feb-05 17:56:01.207 [FileTransfer-8] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/2d/0e59fd7f9797eaf684de4abc52eb46/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\n",
      "Feb-05 17:56:02.337 [Actor Thread 10] INFO  nextflow.processor.TaskProcessor - [07/478896] Cached process > run_whitelist_gunzip (1)\n",
      "Feb-05 17:56:02.576 [Actor Thread 11] INFO  nextflow.processor.TaskProcessor - [59/e8237c] Cached process > run_add_subpool_prefix_to_fragment_table (1)\n",
      "Feb-05 17:56:02.802 [FileTransfer-3] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/chromap_out_debug/BMMC-single-donor.atac.align.k4.hg38.log.txt at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/ad/37cb0426d11a8313b27a0af2761113/BMMC-single-donor.atac.align.k4.hg38.log.txt\n",
      "Feb-05 17:56:03.467 [Actor Thread 3] INFO  nextflow.processor.TaskProcessor - [e8/f6c1d8] Cached process > run_merge_chromap_logs (1)\n",
      "Feb-05 17:56:04.063 [Actor Thread 1] INFO  nextflow.processor.TaskProcessor - [92/d0df15] Cached process > run_process_conversion_to_barcode (1)\n",
      "Feb-05 17:56:05.395 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_bgzip_on_chromap_fragments_output (1)` submitted > job=nf-acabd7bd-1707173763155; uid=nf-acabd7bd-170717-44d3a3e2-d583-48c50; work-dir=gs://eila-nextflow-bucket/work/ATAC/ac/abd7bdd68375327217692386d6f45c\n",
      "Feb-05 17:56:05.395 [Task submitter] INFO  nextflow.Session - [ac/abd7bd] Submitted process > run_bgzip_on_chromap_fragments_output (1)\n",
      "Feb-05 17:56:06.266 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_filter_aligned_fragments (1)` submitted > job=nf-12c29476-1707173764394; uid=nf-12c29476-170717-761f4639-34e6-4bed0; work-dir=gs://eila-nextflow-bucket/work/ATAC/12/c294769ad7cece527e3b4f9ff535e8\n",
      "Feb-05 17:56:06.267 [Task submitter] INFO  nextflow.Session - [12/c29476] Submitted process > run_filter_aligned_fragments (1)\n",
      "Feb-05 17:58:06.749 [Task monitor] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_filter_aligned_fragments (1)` - terminated job=nf-12c29476-1707173764394; state=SUCCEEDED\n",
      "Feb-05 17:58:06.959 [Task monitor] DEBUG n.processor.TaskPollingMonitor - Task completed > TaskHandler[id: 8; name: run_filter_aligned_fragments (1); status: COMPLETED; exit: 0; error: -; workDir: gs://eila-nextflow-bucket/work/ATAC/12/c294769ad7cece527e3b4f9ff535e8]\n",
      "Feb-05 17:58:07.264 [Task submitter] WARN  n.executor.BashWrapperBuilder - Task runtime metrics are not reported when using macOS without a container engine\n",
      "Feb-05 17:58:07.644 [Task submitter] DEBUG nextflow.processor.TaskProcessor - Handling unexpected condition for\n",
      "  task: name=run_tabix_no_singleton (1); work-dir=gs://eila-nextflow-bucket/work/ATAC/bc/b8e623412f5383621dbe1927db9f37\n",
      "  error [nextflow.exception.ProcessUnrecoverableException]: Process `run_tabix_no_singleton (1)` failed because the container image was not specified\n",
      "Feb-05 17:58:07.651 [Task submitter] ERROR nextflow.processor.TaskProcessor - Error executing process > 'run_tabix_no_singleton (1)'\n",
      "\n",
      "Caused by:\n",
      "  Process `run_tabix_no_singleton (1)` failed because the container image was not specified\n",
      "\n",
      "Command executed:\n",
      "\n",
      "  echo 'start run_tabix_no_singleton'\n",
      "  echo 'input no_singelton_gz is no-singleton.bed.gz'\n",
      "  /usr/local/bin/run_tabix.sh no-singleton.bed.gz\n",
      "  ls\n",
      "  echo 'finished run_tabix_no_singleton'\n",
      "\n",
      "Command exit status:\n",
      "  -\n",
      "\n",
      "Command output:\n",
      "  (empty)\n",
      "\n",
      "Work dir:\n",
      "  gs://eila-nextflow-bucket/work/ATAC/bc/b8e623412f5383621dbe1927db9f37\n",
      "\n",
      "Tip: you can try to figure out what's wrong by changing to the process work dir and showing the script file named `.command.sh`\n",
      "Feb-05 17:58:07.652 [Task submitter] DEBUG nextflow.Session - Session aborted -- Cause: Process `run_tabix_no_singleton (1)` failed because the container image was not specified\n",
      "Feb-05 17:58:07.662 [Task submitter] DEBUG nextflow.Session - The following nodes are still active:\n",
      "[process] run_tabix_chromap\n",
      "  status=ACTIVE\n",
      "  port 0: (value) bound ; channel: tabix_script\n",
      "  port 1: (queue) OPEN  ; channel: tabix_fragments\n",
      "  port 2: (cntrl) -     ; channel: $\n",
      "\n",
      "[process] run_calculate_tss_enrichment_bulk\n",
      "  status=ACTIVE\n",
      "  port 0: (queue) OPEN  ; channel: -\n",
      "  port 1: (value) bound ; channel: calculation_script\n",
      "  port 2: (queue) OPEN  ; channel: fragments\n",
      "  port 3: (queue) OPEN  ; channel: tbi_fragments\n",
      "  port 4: (value) bound ; channel: regions_file\n",
      "  port 5: (value) bound ; channel: tss_bases_flank\n",
      "  port 6: (value) bound ; channel: tss_col_strand_info\n",
      "  port 7: (value) bound ; channel: smoothing_window_size\n",
      "  port 8: (value) bound ; channel: CPUS_TO_USE\n",
      "  port 9: (cntrl) -     ; channel: $\n",
      "\n",
      "[process] run_calculate_tss_enrichment_snapatac2\n",
      "  status=ACTIVE\n",
      "  port 0: (queue) OPEN  ; channel: -\n",
      "  port 1: (value) bound ; channel: calculation_script\n",
      "  port 2: (queue) OPEN  ; channel: tbi_fragments\n",
      "  port 3: (queue) OPEN  ; channel: genes_gtf_gzip_file_out\n",
      "  port 4: (value) bound ; channel: min_frag_cutoff\n",
      "  port 5: (cntrl) -     ; channel: $\n",
      "\n",
      "[process] run_generate_insert_size_histogram_data\n",
      "  status=ACTIVE\n",
      "  port 0: (value) bound ; channel: script_name\n",
      "  port 1: (queue) OPEN  ; channel: in_chromap_bzip_fragments_tsv\n",
      "  port 2: (cntrl) -     ; channel: $\n",
      "\n",
      "[process] run_generate_insert_size_plot\n",
      "  status=ACTIVE\n",
      "  port 0: (value) bound ; channel: plot_script\n",
      "  port 1: (queue) OPEN  ; channel: histogram_file\n",
      "  port 2: (value) bound ; channel: pkr_id\n",
      "  port 3: (cntrl) -     ; channel: $\n",
      "\n",
      "[process] run_barcode_metadata_stats\n",
      "  status=ACTIVE\n",
      "  port 0: (queue) OPEN  ; channel: temp_dict_conversion\n",
      "  port 1: (queue) OPEN  ; channel: snapatac_tss_fragments_stats_out\n",
      "  port 2: (cntrl) -     ; channel: $\n",
      "\n",
      "Feb-05 17:58:07.662 [Task monitor] DEBUG n.processor.TaskPollingMonitor - <<< barrier arrives (monitor: google-batch) - terminating tasks monitor poll loop\n",
      "Feb-05 17:58:07.663 [main] DEBUG nextflow.Session - Session await > all processes finished\n",
      "Feb-05 17:58:07.663 [main] DEBUG nextflow.Session - Session await > all barriers passed\n",
      "Feb-05 17:58:07.667 [main] WARN  n.processor.TaskPollingMonitor - Killing running tasks (1)\n",
      "Feb-05 17:58:07.674 [main] DEBUG n.trace.WorkflowStatsObserver - Workflow completed > WorkflowStats[succeededCount=1; failedCount=1; ignoredCount=0; cachedCount=6; pendingCount=1; submittedCount=1; runningCount=-2; retriesCount=0; abortedCount=1; succeedDuration=3.9s; failedDuration=0ms; cachedDuration=1m 43s;loadCpus=-2; loadMemory=0; peakRunning=1; peakCpus=1; peakMemory=0; ]\n",
      "Feb-05 17:58:08.313 [main] DEBUG nextflow.cache.CacheDB - Closing CacheDB done\n",
      "Feb-05 17:58:08.313 [main] INFO  org.pf4j.AbstractPluginManager - Stop plugin 'nf-wave@1.0.1'\n",
      "Feb-05 17:58:08.314 [main] DEBUG nextflow.plugin.BasePlugin - Plugin stopped nf-wave\n",
      "Feb-05 17:58:08.314 [main] INFO  org.pf4j.AbstractPluginManager - Stop plugin 'nf-tower@1.6.3'\n",
      "Feb-05 17:58:08.314 [main] DEBUG nextflow.plugin.BasePlugin - Plugin stopped nf-tower\n",
      "Feb-05 17:58:08.314 [main] INFO  org.pf4j.AbstractPluginManager - Stop plugin 'nf-google@1.8.3'\n",
      "Feb-05 17:58:08.314 [main] DEBUG nextflow.plugin.BasePlugin - Plugin stopped nf-google\n",
      "Feb-05 17:58:08.377 [main] DEBUG nextflow.script.ScriptRunner - > Execution complete -- Goodbye\n"
     ]
    }
   ],
   "source": [
    "!cat .nextflow.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !ls ../../nf_data/test_atac_pipeline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !gsutil cp gs://eila-nextflow-bucket/test_atac_pipeline/sai_192_whitelist.txt ../../nf_data/test_atac_pipeline/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
