{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# start docker deamon in terminal\n",
    "# !sudo docker login\n",
    "# OR - becuase the jupyther with allow root\n",
    "# !docker login"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "TODO:\n",
    "\n",
    "when running with cloud. \n",
    "if you dont need to keep the execution logs, change the cloud settings to keep it for a short while\n",
    "Check that the cache if you need to set the cache to false (especially in plot process)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/eila/git_igvf/atomic-workflows/nextflow/hai-gcp-genomic-beb9207a40b8.json\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "json_path = os.path.join(os.path.dirname(os.path.dirname(os.getcwd())),\"hai-gcp-genomic-beb9207a40b8.json\")\n",
    "!ls $json_path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Activated service account credentials for: [eila-batch@hai-gcp-genomic.iam.gserviceaccount.com]\n"
     ]
    }
   ],
   "source": [
    "# # in the command line. same env\n",
    "import os\n",
    "json_path = os.path.join(os.path.dirname(os.path.dirname(os.getcwd())),\"hai-gcp-genomic-beb9207a40b8.json\")\n",
    "os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = json_path\n",
    "!gcloud auth activate-service-account eila-batch@hai-gcp-genomic.iam.gserviceaccount.com --key-file=$GOOGLE_APPLICATION_CREDENTIALS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                  Credentialed Accounts\n",
      "ACTIVE  ACCOUNT\n",
      "        886987709752-compute@developer.gserviceaccount.com\n",
      "*       eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\n",
      "\n",
      "To set the active account, run:\n",
      "    $ gcloud config set account `ACCOUNT`\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'hai-gcp-genomic'"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "!gcloud auth list\n",
    "project_name = !gcloud config get-value project\n",
    "project_name = project_name[0]\n",
    "project_name\n",
    "# gcloud config set account\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !gsutil cp gs://eila-nextflow-bucket/work/ATAC/b3/94bc73c6d43de41d17d75cac4fbda1/no-singleton.bed.gz.tbi /home/eila/git_igvf/downloads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "import os\n",
    "os.environ[\"TOWER_ACCESS_TOKEN\"] = 'eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz'\n",
    "os.environ[\"TOWER_WORKSPACE_ID\"]='247507587970818'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz\n",
      "247507587970818\n"
     ]
    }
   ],
   "source": [
    "!echo $TOWER_ACCESS_TOKEN\n",
    "!echo $TOWER_WORKSPACE_ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/bin'"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# !chmod +x ./bin/scrna_qc_metrics.py\n",
    "import os\n",
    "bin_fld = os.path.join(os.getcwd(),\"bin\")\n",
    "bin_fld"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "!chmod +x {os.path.join(bin_fld, \"atac_generate_metadata_barcode_rank_plot.R\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"run_tabix.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"atac_generate_merged_barcode_metadata.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"create_sample_filtered_fragment_file.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"compute_tss_enrichment_bulk.py\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"barcode_rank_functions.R\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"plot_insert_size_hist.py\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"compute_tss_enrichment_snapatac2.py\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"add_subpool_to_chromap_output.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"add_subpool_to_summary_file.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"filter_fragments_gz_sort_bgzip_script.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"merge_chromap_logs.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"generate_insert_size_histogram_data.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"calculate_barcode_metadata_stats.sh\")}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# # TODO: For a new user: run these commands with your service account for the execution. I am using eila-batch@...\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/iam.serviceAccountUser\"\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/batch.jobsEditor\"\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/logging.viewer\"\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !nextflow -self-update"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  Version: 23.10.1 build 5891\n",
      "  Created: 12-01-2024 22:01 UTC \n",
      "  System: Linux 5.10.0-27-cloud-amd64\n",
      "  Runtime: Groovy 3.0.19 on OpenJDK 64-Bit Server VM 11.0.1-internal+0-adhoc..src\n",
      "  Encoding: UTF-8 (UTF-8)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!nextflow info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/eila/git_igvf/atomic-workflows/nextflow/nf_data\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "\n",
    "# Get the current working directory\n",
    "current_directory = os.getcwd()\n",
    "\n",
    "# Navigate two folders above\n",
    "two_folders_above = os.path.abspath(os.path.join(current_directory, '..', '..'))\n",
    "\n",
    "nf_data_base_path = os.path.join(two_folders_above,'nf_data')\n",
    "# Print the result\n",
    "print(nf_data_base_path)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/eila/git_igvf/atomic-workflows/nextflow/nf_data'"
      ]
     },
     "execution_count": 108,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nf_data_base_path\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# nf_data_project=\"test_atac_pipeline\"\n",
    "nf_data_project=\"wdl_atac_pipeline\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/eila/git_igvf/atomic-workflows/nextflow/nf_data/wdl_atac_pipeline/vm_atac_fastqs_files.tsv'"
      ]
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# LOCAL\n",
    "# os.path.exists(os.path.join(nf_data_base_path,nf_data_project,\"local_atac_fastqs_files.tsv\"))\n",
    "# os.path.join(nf_data_base_path,nf_data_project,\"local_atac_fastqs_files.tsv\")\n",
    "\n",
    "# VM\n",
    "os.path.exists(os.path.join(nf_data_base_path,nf_data_project,\"vm_atac_fastqs_files.tsv\"))\n",
    "os.path.join(nf_data_base_path,nf_data_project,\"vm_atac_fastqs_files.tsv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "\n",
    "# Write the configuration file\n",
    "with open(\"nextflow_atac.config\", \"w\") as config_file:\n",
    "    config_file.write(f\"\"\"\n",
    "manifest {{\n",
    "  description = 'A nexflow pipeline for fg single cell atac' \n",
    "  mainScript = 'nf_wf_fg_singlecell_atac.nf'\n",
    "}}\n",
    "\n",
    "profiles {{\n",
    "  google_batch_profile {{\n",
    "    workDir = 'gs://eila-nextflow-bucket/work/ATAC'\n",
    "    includeConfig 'conf/docker-google-batch.config'\n",
    "    docker.enabled = true\n",
    "    wave.enabled = true\n",
    "    fusion.enabled = true\n",
    "    process.scratch = false\n",
    "    executor = 'google-batch'\n",
    "    docker.envWhitelist=['ENV_SYNAPSE_TOKEN','TOWER_ACCESS_TOKEN']\n",
    "    google.project = '{project_name}'\n",
    "    google.location = 'us-central1'\n",
    "    google.batch.spot = true\n",
    "    google.batch.serviceAccountEmail='eila-batch@hai-gcp-genomic.iam.gserviceaccount.com'\n",
    "  }}\n",
    "}}\n",
    "\n",
    "// Define ENV_SYNAPSE_TOKEN as a parameter\n",
    "params.FASTQS_SPEC_CH = '{nf_data_base_path}/{nf_data_project}/vm_atac_fastqs_files.tsv'\n",
    "// CHROMAP\n",
    "// 6a\n",
    "params.CHROMAP_IDX='gs://eila-nextflow-bucket/test_atac_pipeline/ref.index'\n",
    "// 6b:\n",
    "// FASTA file\n",
    "params.CHROMAP_READ_LENGTH = 2000\n",
    "params.CHROMAP_BC_ERROR_THRESHOLD = 2\n",
    "params.CHROMAP_BC_PROBABILITY_THRESHOLD = 0.9\n",
    "params.CHROMAP_READ_FORMAT = 'bc:0:-1,r1:0:-1,r2:0:-1'\n",
    "params.CHROMAP_DROP_REPETITIVE_READS = 4\n",
    "params.CHROMAP_QUALITY_THRESHOLD = 0\n",
    "params.CHROMAP_MERGE_LOGS_SCRIPT = \"merge_chromap_logs.sh\"\n",
    "params.CHROMAP_FRAGMENTS_HISTOGRAM_DATA_SCRIPT='generate_insert_size_histogram_data.sh'\n",
    "params.CHROMAP_GENOME_REFERENCE_FASTA='{nf_data_base_path}/{nf_data_project}/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz'\n",
    "\n",
    "// CHROMAP DEBUG: load output files \n",
    "// params.chromap_filter_fragments_tsv ='{nf_data_base_path}/{nf_data_project}/chromap_out_debug/ATAC_BMMC-single-donor.atac.filter.fragments.tsv'\n",
    "// params.barcode_summary_csv ='{nf_data_base_path}/{nf_data_project}/chromap_out_debug/BMMC-single-donor.atac.align.barcode.summary.csv'\n",
    "// params.chromap_alignment_log_out = '{nf_data_base_path}/{nf_data_project}/chromap_out_debug/BMMC-single-donor.atac.align.k4.hg38.log.txt'\n",
    "\n",
    "// CHROMAP WDL DEBUG: load output files \n",
    "params.chromap_fragments_tsv ='{nf_data_base_path}/{nf_data_project}/ATAC_BMMC-single-donor.atac.filter.fragments.tsv'\n",
    "params.barcode_summary_csv ='{nf_data_base_path}/{nf_data_project}/BMMC-single-donor.atac.align.barcode.summary.csv'\n",
    "params.chromap_alignment_log_out = '{nf_data_base_path}/{nf_data_project}/BMMC-single-donor.atac.align.k4.hg38.log.txt'\n",
    "params.bgzip_subpool_chromap_fragments_tsv_out='{nf_data_base_path}/{nf_data_project}/BMMC-single-donor.atac.filter.fragments.hg38.tsv.gz'\n",
    "params.no_singleton_gz='{nf_data_base_path}/{nf_data_project}/no-singleton.bed.gz'\n",
    "// params.tbi_no_singleton_bed_gz_out='{nf_data_base_path}/{nf_data_project}/.tbi'\n",
    "\n",
    "params.TSV_ADD_SUBPOOL_SCRIPT = 'add_subpool_to_chromap_output.sh'\n",
    "params.CSV_ADD_SUBPOOL_SCRIPT = 'add_subpool_to_summary_file.sh'\n",
    "params.FILTER_FRAGMENTS_SCRIPT = 'filter_fragments_gz_sort_bgzip_script.sh'\n",
    "params.ATAC_FRAGMENTS_MIN_FRAG_CUTOFF=10\n",
    "\n",
    "// Other steps\n",
    "\n",
    "params.GENOME_GZ_GTF='http://ftp.ensembl.org/pub/release-109/gtf/homo_sapiens/Homo_sapiens.GRCh38.109.gtf.gz'\n",
    "params.ENV_SYNAPSE_TOKEN = \"nextflow secrets get SYNAPSE\".execute().text.trim()\n",
    "\n",
    "\n",
    "params.MAPQ_THRESHOLD = false\n",
    "params.BARCODE_TRANSLATE = 'path'\n",
    "params.ATAC_FRAGMENTS_CUTOFF = 4\n",
    "params.RUN_TABIX_SCRIPT='run_tabix.sh'\n",
    "params.ATAC_TSS_BASES_FLANK = 2000\n",
    "params.ATAC_TSS_COL_WITH_STRANDS_INFO = 4\n",
    "params.ATAC_TSS_SMOOTHING_WINDOW_SIZE = 20\n",
    "params.ATAC_TSS_REGION_BED_FILE='{nf_data_base_path}/{nf_data_project}/hg38.refGene.TSS.bed'\n",
    "params.ATAC_TSS_BULK_CALCULATION_SCRIPT='compute_tss_enrichment_bulk.py'\n",
    "params.ATAC_TSS_SNAPATAC2_CALCULATION_SCRIPT='compute_tss_enrichment_snapatac2.py'\n",
    "params.ATAC_TSS_SNAPATAC2_GENES_GTF='{nf_data_base_path}/{nf_data_project}/genes.gtf'\n",
    "params.ATAC_TSS_SNAPATAC2_MIN_FRAG_CUTOFF = 10\n",
    "\n",
    "params.ATAC_INSERT_SIZE_PLOT_SCRIPT='plot_insert_size_hist.py'\n",
    "params.pkr_id=\"BMM_PKR_ID\"\n",
    "\n",
    "// atac calculate_barcode_metadata_stats\n",
    "params.ATAC_SNAP_METADATA_STATS_SCRIPT='calculate_barcode_metadata_stats.sh'\n",
    "// atac plot barcode metadata\n",
    "params.ATAC_PLOT_METADATA_SCRIPT='atac_generate_metadata_barcode_rank_plot.R'\n",
    "params.ATAC_PLOT_METADATA_HELPER_SCRIPT='barcode_rank_functions.R'\n",
    "\n",
    "// debug file\n",
    "// params.SC_ATAC_QC_BARCODE_METADATA_FILE='{nf_data_base_path}/{nf_data_project}/barcode_metadata.txt'\n",
    "params.ATAC_QC_FRAGMENT_CUTOFF=10\n",
    "params.SC_ATAC_QC_BARCODE_OUTPUT_FILE='atac_barcode_metadata_output_file.png'\n",
    "    \n",
    "// params.SC_ATAC_BARCODE_METADATA_SCRIPT='atac_generate_merged_barcode_metadata.sh'\n",
    "// DEBUG: get the data from a file\n",
    "// params.SC_ATAC_BARCODE_METADATA_FILTERED_BARCODE_STATS='{nf_data_base_path}nf_wf_fg_singlecell/filtered_barcode_stats.csv'\n",
    "// params.SC_ATAC_BARCODE_METADATA_TSS_ENRICHMENT_BARCODE_STATS='{nf_data_base_path}nf_wf_fg_singlecell/tss_enrichment_barcode_stats.csv'\n",
    "\n",
    "// params.SC_ATAC_GENERATE_BARCODE_RANK_PLOT_PKR='PKR123'\n",
    "\n",
    "// params.SC_JOINT_CELL_BARCODE_PLOT_SCRTIP_PY='joint_cell_plotting.py'\n",
    "// params.SC_JOINT_CELL_BARCODE_PLOT_SCRTIP_R='joint_cell_plotting_density.R'\n",
    "\n",
    "// DEBUG\n",
    "\n",
    "params.CPUS_TO_USE_TSS=16\n",
    "params.CPUS_TO_USE_CHROMAP_IDX=32\n",
    "\"\"\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !rm -r /Users/eilaarich-landkof-stanford/Documents/Code/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/work\n",
    "# !rm -r /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/work\n",
    "# !gsutil rm -r gs://eila-nextflow-bucket/work/ATAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !rm .nextflow*\n",
    "\n",
    "# !ls /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/local_atac_fastqs_files.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "N E X T F L O W  ~  version 23.10.1\n",
      "Launching `nf_wf_fg_singlecell_atac.nf` [insane_woese] DSL2 - revision: 74ff183a6a\n",
      "/home/eila/git_igvf/atomic-workflows/nextflow/nf_data/wdl_atac_pipeline/vm_atac_fastqs_files.tsv\n",
      "null\n",
      "after genome_fasta_ch\n",
      "[-        ] process > run_zcat                       -\u001b[K\n",
      "[-        ] process > run_add_subpool_to_chromap_... -\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_process_create_temp_sum... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragment... -\u001b[K\n",
      "[-        ] process > run_gzip_on_genes_gtf          -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "[-        ] process > run_atac_barcode_rank_plot     -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call \u001b[K\n",
      "after call run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "before run_process_create_temp_summary_dict\u001b[K\n",
      "after run_process_create_temp_summary_dict\u001b[K\n",
      "before run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "\u001b[34A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2v39mmFiJZmXWq\u001b[22;39m\u001b[K\n",
      "[-        ] process > run_zcat                       -\u001b[K\n",
      "[-        ] process > run_add_subpool_to_chromap_... -\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_process_create_temp_sum... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragment... -\u001b[K\n",
      "[-        ] process > run_gzip_on_genes_gtf          -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "[-        ] process > run_atac_barcode_rank_plot     -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call \u001b[K\n",
      "after call run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "before run_process_create_temp_summary_dict\u001b[K\n",
      "after run_process_create_temp_summary_dict\u001b[K\n",
      "before run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "\u001b[35A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2v39mmFiJZmXWq\u001b[22;39m\u001b[K\n",
      "[33/c79639] process > run_zcat                       [100%] 1 of 1, cached: 1\u001b[K\n",
      "[-        ] process > run_add_subpool_to_chromap_... -\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_process_create_temp_sum... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragment... -\u001b[K\n",
      "[aa/adbebc] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "[-        ] process > run_atac_barcode_rank_plot     -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call \u001b[K\n",
      "after call run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "before run_process_create_temp_summary_dict\u001b[K\n",
      "after run_process_create_temp_summary_dict\u001b[K\n",
      "before run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "\u001b[35A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2v39mmFiJZmXWq\u001b[22;39m\u001b[K\n",
      "[33/c79639] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_add_subpool_to_chromap_... -\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_process_create_temp_sum... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragment... -\u001b[K\n",
      "[aa/adbebc] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "[-        ] process > run_atac_barcode_rank_plot     -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call \u001b[K\n",
      "after call run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "before run_process_create_temp_summary_dict\u001b[K\n",
      "after run_process_create_temp_summary_dict\u001b[K\n",
      "before run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb 14 01:34 genes.gtf.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb 14 01:34 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/41/45671d165921b1e3a885b4ea55909e/genes.gtf\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "\u001b[54A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2v39mmFiJZmXWq\u001b[22;39m\u001b[K\n",
      "[33/c79639] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[9f/40830d] process > run_add_subpool_to_chromap_... [100%] 1 of 1, cached: 1\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_process_create_temp_sum... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragment... -\u001b[K\n",
      "[aa/adbebc] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "[-        ] process > run_atac_barcode_rank_plot     -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call \u001b[K\n",
      "after call run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "before run_process_create_temp_summary_dict\u001b[K\n",
      "after run_process_create_temp_summary_dict\u001b[K\n",
      "before run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb 14 01:34 genes.gtf.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb 14 01:34 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/41/45671d165921b1e3a885b4ea55909e/genes.gtf\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "\u001b[54A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2v39mmFiJZmXWq\u001b[22;39m\u001b[K\n",
      "[33/c79639] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[9f/40830d] process > run_add_subpool_to_chromap_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_process_create_temp_sum... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragment... -\u001b[K\n",
      "[aa/adbebc] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "[-        ] process > run_atac_barcode_rank_plot     -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call \u001b[K\n",
      "after call run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "before run_process_create_temp_summary_dict\u001b[K\n",
      "after run_process_create_temp_summary_dict\u001b[K\n",
      "before run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb 14 01:34 genes.gtf.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb 14 01:34 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/41/45671d165921b1e3a885b4ea55909e/genes.gtf\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "Input chromap_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_chromap_output.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "\u001b[K\n",
      "\u001b[62A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2v39mmFiJZmXWq\u001b[22;39m\u001b[K\n",
      "[33/c79639] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[9f/40830d] process > run_add_subpool_to_chromap_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3e/470034] process > run_bgzip_on_chromap_fragme... [100%] 1 of 1, cached: 1\u001b[K\n",
      "[-        ] process > run_process_create_temp_sum... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragment... -\u001b[K\n",
      "[aa/adbebc] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "[-        ] process > run_atac_barcode_rank_plot     -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call \u001b[K\n",
      "after call run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "before run_process_create_temp_summary_dict\u001b[K\n",
      "after run_process_create_temp_summary_dict\u001b[K\n",
      "before run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb 14 01:34 genes.gtf.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb 14 01:34 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/41/45671d165921b1e3a885b4ea55909e/genes.gtf\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "Input chromap_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_chromap_output.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "\u001b[K\n",
      "\u001b[62A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2v39mmFiJZmXWq\u001b[22;39m\u001b[K\n",
      "[33/c79639] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[9f/40830d] process > run_add_subpool_to_chromap_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3e/470034] process > run_bgzip_on_chromap_fragme... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[d3/c0aca2] process > run_process_create_temp_sum... [100%] 1 of 1, cached: 1\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragment... -\u001b[K\n",
      "[aa/adbebc] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "[-        ] process > run_atac_barcode_rank_plot     -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call \u001b[K\n",
      "after call run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "before run_process_create_temp_summary_dict\u001b[K\n",
      "after run_process_create_temp_summary_dict\u001b[K\n",
      "before run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb 14 01:34 genes.gtf.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb 14 01:34 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/41/45671d165921b1e3a885b4ea55909e/genes.gtf\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "Input chromap_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_chromap_output.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "\u001b[K\n",
      "Start run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "Fragment file is: ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "total 458900\u001b[K\n",
      "-rw-r--r-- 1 root root 469912630 Feb 14 02:24 ATAC_BMMC-single-donor.atac.filter.fragments.tsv.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb 14 02:24 ATAC_BMMC-single-donor.atac.filter.fragments.tsv -> /fusion/gs/eila-nextflow-bucket/work/ATAC/9f/40830dbfeebcb71ec69808fa3bfd6c/ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Finished run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "\u001b[K\n",
      "\u001b[69A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2v39mmFiJZmXWq\u001b[22;39m\u001b[K\n",
      "[33/c79639] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[9f/40830d] process > run_add_subpool_to_chromap_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3e/470034] process > run_bgzip_on_chromap_fragme... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[d3/c0aca2] process > run_process_create_temp_sum... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragment... -\u001b[K\n",
      "[aa/adbebc] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "[-        ] process > run_atac_barcode_rank_plot     -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call \u001b[K\n",
      "after call run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "before run_process_create_temp_summary_dict\u001b[K\n",
      "after run_process_create_temp_summary_dict\u001b[K\n",
      "before run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb 14 01:34 genes.gtf.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb 14 01:34 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/41/45671d165921b1e3a885b4ea55909e/genes.gtf\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "Input chromap_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_chromap_output.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "\u001b[K\n",
      "Start run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "Fragment file is: ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "total 458900\u001b[K\n",
      "-rw-r--r-- 1 root root 469912630 Feb 14 02:24 ATAC_BMMC-single-donor.atac.filter.fragments.tsv.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb 14 02:24 ATAC_BMMC-single-donor.atac.filter.fragments.tsv -> /fusion/gs/eila-nextflow-bucket/work/ATAC/9f/40830dbfeebcb71ec69808fa3bfd6c/ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Finished run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "\u001b[K\n",
      "start run_process_create_temp_summary_dict\u001b[K\n",
      "input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "------ There is NO conversion list ------\u001b[K\n",
      "temp_summary\u001b[K\n",
      "7021039 temp_summary\u001b[K\n",
      "finished run_process_create_temp_summary_dict\u001b[K\n",
      "\u001b[K\n",
      "\u001b[76A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2v39mmFiJZmXWq\u001b[22;39m\u001b[K\n",
      "[33/c79639] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[9f/40830d] process > run_add_subpool_to_chromap_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3e/470034] process > run_bgzip_on_chromap_fragme... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[d3/c0aca2] process > run_process_create_temp_sum... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[47/cceb9e] process > run_filter_aligned_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[aa/adbebc] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "[-        ] process > run_atac_barcode_rank_plot     -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[2d/d3c262] process > run_merge_chromap_logs (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call \u001b[K\n",
      "after call run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "before run_process_create_temp_summary_dict\u001b[K\n",
      "after run_process_create_temp_summary_dict\u001b[K\n",
      "before run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb 14 01:34 genes.gtf.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb 14 01:34 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/41/45671d165921b1e3a885b4ea55909e/genes.gtf\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "Input chromap_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_chromap_output.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "\u001b[K\n",
      "Start run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "Fragment file is: ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "total 458900\u001b[K\n",
      "-rw-r--r-- 1 root root 469912630 Feb 14 02:24 ATAC_BMMC-single-donor.atac.filter.fragments.tsv.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb 14 02:24 ATAC_BMMC-single-donor.atac.filter.fragments.tsv -> /fusion/gs/eila-nextflow-bucket/work/ATAC/9f/40830dbfeebcb71ec69808fa3bfd6c/ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Finished run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "\u001b[K\n",
      "start run_process_create_temp_summary_dict\u001b[K\n",
      "input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "------ There is NO conversion list ------\u001b[K\n",
      "temp_summary\u001b[K\n",
      "7021039 temp_summary\u001b[K\n",
      "finished run_process_create_temp_summary_dict\u001b[K\n",
      "\u001b[K\n",
      "start run_merge_logs\u001b[K\n",
      "input script_name is merge_chromap_logs.sh\u001b[K\n",
      "input chromap_alignment_log is BMMC-single-donor.atac.align.k4.hg38.log.txt\u001b[K\n",
      "input chromap_barcode_log is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "start run_merge_chromap_logs shell script\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "********1***********\u001b[K\n",
      "********2***********\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "uni-mappings,260650161\u001b[K\n",
      "multi-mappings,14666503\u001b[K\n",
      "total,275316664\u001b[K\n",
      "********3***********\u001b[K\n",
      "finished run_merge_chromap_logs shell script\u001b[K\n",
      "finished run_merge_logs\u001b[K\n",
      "\u001b[K\n",
      "start run_filter_aligned_fragments_sort_gz\u001b[K\n",
      "input barcodeFragmentsTsvGz is ATAC_BMMC-single-donor.atac.filter.fragments.tsv.gz\u001b[K\n",
      "input threshold is 10\u001b[K\n",
      "input temp_summary_dict is temp_summary\u001b[K\n",
      "------ Number of barcodes BEFORE filtering------\u001b[K\n",
      "7021039 temp_summary\u001b[K\n",
      "------ Filtering fragments ------\u001b[K\n",
      "------ Number of barcodes AFTER filtering ------\u001b[K\n",
      "3950969\u001b[K\n",
      "finished run_filter_aligned_fragments_sort_gz\u001b[K\n",
      "\u001b[K\n",
      "\u001b[123A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2v39mmFiJZmXWq\u001b[22;39m\u001b[K\n",
      "[33/c79639] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[9f/40830d] process > run_add_subpool_to_chromap_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3e/470034] process > run_bgzip_on_chromap_fragme... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[d3/c0aca2] process > run_process_create_temp_sum... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[47/cceb9e] process > run_filter_aligned_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[aa/adbebc] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "[-        ] process > run_atac_barcode_rank_plot     -\u001b[K\n",
      "[d8/bdb929] process > run_generate_insert_size_hi... [100%] 1 of 1, cached: 1\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[2d/d3c262] process > run_merge_chromap_logs (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call \u001b[K\n",
      "after call run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "before run_process_create_temp_summary_dict\u001b[K\n",
      "after run_process_create_temp_summary_dict\u001b[K\n",
      "before run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb 14 01:34 genes.gtf.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb 14 01:34 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/41/45671d165921b1e3a885b4ea55909e/genes.gtf\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "Input chromap_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_chromap_output.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "\u001b[K\n",
      "Start run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "Fragment file is: ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "total 458900\u001b[K\n",
      "-rw-r--r-- 1 root root 469912630 Feb 14 02:24 ATAC_BMMC-single-donor.atac.filter.fragments.tsv.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb 14 02:24 ATAC_BMMC-single-donor.atac.filter.fragments.tsv -> /fusion/gs/eila-nextflow-bucket/work/ATAC/9f/40830dbfeebcb71ec69808fa3bfd6c/ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Finished run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "\u001b[K\n",
      "start run_process_create_temp_summary_dict\u001b[K\n",
      "input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "------ There is NO conversion list ------\u001b[K\n",
      "temp_summary\u001b[K\n",
      "7021039 temp_summary\u001b[K\n",
      "finished run_process_create_temp_summary_dict\u001b[K\n",
      "\u001b[K\n",
      "start run_merge_logs\u001b[K\n",
      "input script_name is merge_chromap_logs.sh\u001b[K\n",
      "input chromap_alignment_log is BMMC-single-donor.atac.align.k4.hg38.log.txt\u001b[K\n",
      "input chromap_barcode_log is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "start run_merge_chromap_logs shell script\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "********1***********\u001b[K\n",
      "********2***********\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "uni-mappings,260650161\u001b[K\n",
      "multi-mappings,14666503\u001b[K\n",
      "total,275316664\u001b[K\n",
      "********3***********\u001b[K\n",
      "finished run_merge_chromap_logs shell script\u001b[K\n",
      "finished run_merge_logs\u001b[K\n",
      "\u001b[K\n",
      "start run_filter_aligned_fragments_sort_gz\u001b[K\n",
      "input barcodeFragmentsTsvGz is ATAC_BMMC-single-donor.atac.filter.fragments.tsv.gz\u001b[K\n",
      "input threshold is 10\u001b[K\n",
      "input temp_summary_dict is temp_summary\u001b[K\n",
      "------ Number of barcodes BEFORE filtering------\u001b[K\n",
      "7021039 temp_summary\u001b[K\n",
      "------ Filtering fragments ------\u001b[K\n",
      "------ Number of barcodes AFTER filtering ------\u001b[K\n",
      "3950969\u001b[K\n",
      "finished run_filter_aligned_fragments_sort_gz\u001b[K\n",
      "\u001b[K\n",
      "\u001b[123A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2v39mmFiJZmXWq\u001b[22;39m\u001b[K\n",
      "[33/c79639] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[9f/40830d] process > run_add_subpool_to_chromap_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3e/470034] process > run_bgzip_on_chromap_fragme... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[d3/c0aca2] process > run_process_create_temp_sum... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[47/cceb9e] process > run_filter_aligned_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[aa/adbebc] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "[-        ] process > run_atac_barcode_rank_plot     -\u001b[K\n",
      "[d8/bdb929] process > run_generate_insert_size_hi... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[2d/d3c262] process > run_merge_chromap_logs (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call \u001b[K\n",
      "after call run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "before run_process_create_temp_summary_dict\u001b[K\n",
      "after run_process_create_temp_summary_dict\u001b[K\n",
      "before run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb 14 01:34 genes.gtf.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb 14 01:34 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/41/45671d165921b1e3a885b4ea55909e/genes.gtf\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "Input chromap_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_chromap_output.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "\u001b[K\n",
      "Start run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "Fragment file is: ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "total 458900\u001b[K\n",
      "-rw-r--r-- 1 root root 469912630 Feb 14 02:24 ATAC_BMMC-single-donor.atac.filter.fragments.tsv.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb 14 02:24 ATAC_BMMC-single-donor.atac.filter.fragments.tsv -> /fusion/gs/eila-nextflow-bucket/work/ATAC/9f/40830dbfeebcb71ec69808fa3bfd6c/ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Finished run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "\u001b[K\n",
      "start run_process_create_temp_summary_dict\u001b[K\n",
      "input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "------ There is NO conversion list ------\u001b[K\n",
      "temp_summary\u001b[K\n",
      "7021039 temp_summary\u001b[K\n",
      "finished run_process_create_temp_summary_dict\u001b[K\n",
      "\u001b[K\n",
      "start run_merge_logs\u001b[K\n",
      "input script_name is merge_chromap_logs.sh\u001b[K\n",
      "input chromap_alignment_log is BMMC-single-donor.atac.align.k4.hg38.log.txt\u001b[K\n",
      "input chromap_barcode_log is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "start run_merge_chromap_logs shell script\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "********1***********\u001b[K\n",
      "********2***********\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "uni-mappings,260650161\u001b[K\n",
      "multi-mappings,14666503\u001b[K\n",
      "total,275316664\u001b[K\n",
      "********3***********\u001b[K\n",
      "finished run_merge_chromap_logs shell script\u001b[K\n",
      "finished run_merge_logs\u001b[K\n",
      "\u001b[K\n",
      "start run_filter_aligned_fragments_sort_gz\u001b[K\n",
      "input barcodeFragmentsTsvGz is ATAC_BMMC-single-donor.atac.filter.fragments.tsv.gz\u001b[K\n",
      "input threshold is 10\u001b[K\n",
      "input temp_summary_dict is temp_summary\u001b[K\n",
      "------ Number of barcodes BEFORE filtering------\u001b[K\n",
      "7021039 temp_summary\u001b[K\n",
      "------ Filtering fragments ------\u001b[K\n",
      "------ Number of barcodes AFTER filtering ------\u001b[K\n",
      "3950969\u001b[K\n",
      "finished run_filter_aligned_fragments_sort_gz\u001b[K\n",
      "\u001b[K\n",
      "start run_generate_insert_size_histogram_data\u001b[K\n",
      "script_name is generate_insert_size_histogram_data.sh\u001b[K\n",
      "in_chromap_bzip_fragments_tsv is no-singleton.bed.gz\u001b[K\n",
      "output_log_file: no-singleton.bed.hist.log.txt\u001b[K\n",
      "input_file: no-singleton.bed.gz\u001b[K\n",
      "Head of input_file:\u001b[K\n",
      "chr1\t10006\t10266\tCAGGAATCGAATCTGTGTACATCT\t1\u001b[K\n",
      "chr1\t10006\t10512\tAGTCATGGATCTAGTTTATCAAGG\t1\u001b[K\n",
      "chr1\t10007\t10113\tTACTTCGGTAGAACACGGATTAGG\t1\u001b[K\n",
      "chr1\t10007\t10113\tTCAGATTCCTAATATGTCATTGAG\t1\u001b[K\n",
      "chr1\t10007\t10114\tTGCGTGAATCAGGAGGACAGTGGT\t1\u001b[K\n",
      "chr1\t10007\t10132\tTGATACGTTAGACGGATCCTCAAT\t1\u001b[K\n",
      "chr1\t10007\t10151\tGAGGAGAGGCTCCTTGCCGGCGTT\t1\u001b[K\n",
      "chr1\t10007\t10157\tCCAACCGCGACCTTAGTGCATAGT\t1\u001b[K\n",
      "chr1\t10007\t10253\tGGCGGATATTCGCACCCGATGTTT\t1\u001b[K\n",
      "chr1\t10007\t10553\tTATCAAGGTGACCACTGGTGAGTT\t1\u001b[K\n",
      "finished run_generate_insert_size_histogram_data\u001b[K\n",
      "\u001b[K\n",
      "\u001b[141A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2v39mmFiJZmXWq\u001b[22;39m\u001b[K\n",
      "[33/c79639] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[9f/40830d] process > run_add_subpool_to_chromap_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3e/470034] process > run_bgzip_on_chromap_fragme... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[d3/c0aca2] process > run_process_create_temp_sum... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[47/cceb9e] process > run_filter_aligned_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[aa/adbebc] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "[-        ] process > run_atac_barcode_rank_plot     -\u001b[K\n",
      "[d8/bdb929] process > run_generate_insert_size_hi... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  [  0%] 0 of 1\u001b[K\n",
      "[2d/d3c262] process > run_merge_chromap_logs (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call \u001b[K\n",
      "after call run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "before run_process_create_temp_summary_dict\u001b[K\n",
      "after run_process_create_temp_summary_dict\u001b[K\n",
      "before run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb 14 01:34 genes.gtf.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb 14 01:34 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/41/45671d165921b1e3a885b4ea55909e/genes.gtf\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "Input chromap_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_chromap_output.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "\u001b[K\n",
      "Start run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "Fragment file is: ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "total 458900\u001b[K\n",
      "-rw-r--r-- 1 root root 469912630 Feb 14 02:24 ATAC_BMMC-single-donor.atac.filter.fragments.tsv.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb 14 02:24 ATAC_BMMC-single-donor.atac.filter.fragments.tsv -> /fusion/gs/eila-nextflow-bucket/work/ATAC/9f/40830dbfeebcb71ec69808fa3bfd6c/ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Finished run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "\u001b[K\n",
      "start run_process_create_temp_summary_dict\u001b[K\n",
      "input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "------ There is NO conversion list ------\u001b[K\n",
      "temp_summary\u001b[K\n",
      "7021039 temp_summary\u001b[K\n",
      "finished run_process_create_temp_summary_dict\u001b[K\n",
      "\u001b[K\n",
      "start run_merge_logs\u001b[K\n",
      "input script_name is merge_chromap_logs.sh\u001b[K\n",
      "input chromap_alignment_log is BMMC-single-donor.atac.align.k4.hg38.log.txt\u001b[K\n",
      "input chromap_barcode_log is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "start run_merge_chromap_logs shell script\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "********1***********\u001b[K\n",
      "********2***********\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "uni-mappings,260650161\u001b[K\n",
      "multi-mappings,14666503\u001b[K\n",
      "total,275316664\u001b[K\n",
      "********3***********\u001b[K\n",
      "finished run_merge_chromap_logs shell script\u001b[K\n",
      "finished run_merge_logs\u001b[K\n",
      "\u001b[K\n",
      "start run_filter_aligned_fragments_sort_gz\u001b[K\n",
      "input barcodeFragmentsTsvGz is ATAC_BMMC-single-donor.atac.filter.fragments.tsv.gz\u001b[K\n",
      "input threshold is 10\u001b[K\n",
      "input temp_summary_dict is temp_summary\u001b[K\n",
      "------ Number of barcodes BEFORE filtering------\u001b[K\n",
      "7021039 temp_summary\u001b[K\n",
      "------ Filtering fragments ------\u001b[K\n",
      "------ Number of barcodes AFTER filtering ------\u001b[K\n",
      "3950969\u001b[K\n",
      "finished run_filter_aligned_fragments_sort_gz\u001b[K\n",
      "\u001b[K\n",
      "start run_generate_insert_size_histogram_data\u001b[K\n",
      "script_name is generate_insert_size_histogram_data.sh\u001b[K\n",
      "in_chromap_bzip_fragments_tsv is no-singleton.bed.gz\u001b[K\n",
      "output_log_file: no-singleton.bed.hist.log.txt\u001b[K\n",
      "input_file: no-singleton.bed.gz\u001b[K\n",
      "Head of input_file:\u001b[K\n",
      "chr1\t10006\t10266\tCAGGAATCGAATCTGTGTACATCT\t1\u001b[K\n",
      "chr1\t10006\t10512\tAGTCATGGATCTAGTTTATCAAGG\t1\u001b[K\n",
      "chr1\t10007\t10113\tTACTTCGGTAGAACACGGATTAGG\t1\u001b[K\n",
      "chr1\t10007\t10113\tTCAGATTCCTAATATGTCATTGAG\t1\u001b[K\n",
      "chr1\t10007\t10114\tTGCGTGAATCAGGAGGACAGTGGT\t1\u001b[K\n",
      "chr1\t10007\t10132\tTGATACGTTAGACGGATCCTCAAT\t1\u001b[K\n",
      "chr1\t10007\t10151\tGAGGAGAGGCTCCTTGCCGGCGTT\t1\u001b[K\n",
      "chr1\t10007\t10157\tCCAACCGCGACCTTAGTGCATAGT\t1\u001b[K\n",
      "chr1\t10007\t10253\tGGCGGATATTCGCACCCGATGTTT\t1\u001b[K\n",
      "chr1\t10007\t10553\tTATCAAGGTGACCACTGGTGAGTT\t1\u001b[K\n",
      "finished run_generate_insert_size_histogram_data\u001b[K\n",
      "\u001b[K\n",
      "\u001b[141A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2v39mmFiJZmXWq\u001b[22;39m\u001b[K\n",
      "[33/c79639] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[9f/40830d] process > run_add_subpool_to_chromap_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3e/470034] process > run_bgzip_on_chromap_fragme... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[d3/c0aca2] process > run_process_create_temp_sum... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[47/cceb9e] process > run_filter_aligned_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[aa/adbebc] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[7b/53117e] process > run_calculate_tss_enrichmen... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "[-        ] process > run_atac_barcode_rank_plot     -\u001b[K\n",
      "[d8/bdb929] process > run_generate_insert_size_hi... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  [  0%] 0 of 1\u001b[K\n",
      "[2d/d3c262] process > run_merge_chromap_logs (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call \u001b[K\n",
      "after call run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "before run_process_create_temp_summary_dict\u001b[K\n",
      "after run_process_create_temp_summary_dict\u001b[K\n",
      "before run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb 14 01:34 genes.gtf.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb 14 01:34 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/41/45671d165921b1e3a885b4ea55909e/genes.gtf\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "Input chromap_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_chromap_output.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "\u001b[K\n",
      "Start run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "Fragment file is: ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "total 458900\u001b[K\n",
      "-rw-r--r-- 1 root root 469912630 Feb 14 02:24 ATAC_BMMC-single-donor.atac.filter.fragments.tsv.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb 14 02:24 ATAC_BMMC-single-donor.atac.filter.fragments.tsv -> /fusion/gs/eila-nextflow-bucket/work/ATAC/9f/40830dbfeebcb71ec69808fa3bfd6c/ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Finished run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "\u001b[K\n",
      "start run_process_create_temp_summary_dict\u001b[K\n",
      "input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "------ There is NO conversion list ------\u001b[K\n",
      "temp_summary\u001b[K\n",
      "7021039 temp_summary\u001b[K\n",
      "finished run_process_create_temp_summary_dict\u001b[K\n",
      "\u001b[K\n",
      "start run_merge_logs\u001b[K\n",
      "input script_name is merge_chromap_logs.sh\u001b[K\n",
      "input chromap_alignment_log is BMMC-single-donor.atac.align.k4.hg38.log.txt\u001b[K\n",
      "input chromap_barcode_log is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "start run_merge_chromap_logs shell script\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "********1***********\u001b[K\n",
      "********2***********\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "uni-mappings,260650161\u001b[K\n",
      "multi-mappings,14666503\u001b[K\n",
      "total,275316664\u001b[K\n",
      "********3***********\u001b[K\n",
      "finished run_merge_chromap_logs shell script\u001b[K\n",
      "finished run_merge_logs\u001b[K\n",
      "\u001b[K\n",
      "start run_filter_aligned_fragments_sort_gz\u001b[K\n",
      "input barcodeFragmentsTsvGz is ATAC_BMMC-single-donor.atac.filter.fragments.tsv.gz\u001b[K\n",
      "input threshold is 10\u001b[K\n",
      "input temp_summary_dict is temp_summary\u001b[K\n",
      "------ Number of barcodes BEFORE filtering------\u001b[K\n",
      "7021039 temp_summary\u001b[K\n",
      "------ Filtering fragments ------\u001b[K\n",
      "------ Number of barcodes AFTER filtering ------\u001b[K\n",
      "3950969\u001b[K\n",
      "finished run_filter_aligned_fragments_sort_gz\u001b[K\n",
      "\u001b[K\n",
      "start run_generate_insert_size_histogram_data\u001b[K\n",
      "script_name is generate_insert_size_histogram_data.sh\u001b[K\n",
      "in_chromap_bzip_fragments_tsv is no-singleton.bed.gz\u001b[K\n",
      "output_log_file: no-singleton.bed.hist.log.txt\u001b[K\n",
      "input_file: no-singleton.bed.gz\u001b[K\n",
      "Head of input_file:\u001b[K\n",
      "chr1\t10006\t10266\tCAGGAATCGAATCTGTGTACATCT\t1\u001b[K\n",
      "chr1\t10006\t10512\tAGTCATGGATCTAGTTTATCAAGG\t1\u001b[K\n",
      "chr1\t10007\t10113\tTACTTCGGTAGAACACGGATTAGG\t1\u001b[K\n",
      "chr1\t10007\t10113\tTCAGATTCCTAATATGTCATTGAG\t1\u001b[K\n",
      "chr1\t10007\t10114\tTGCGTGAATCAGGAGGACAGTGGT\t1\u001b[K\n",
      "chr1\t10007\t10132\tTGATACGTTAGACGGATCCTCAAT\t1\u001b[K\n",
      "chr1\t10007\t10151\tGAGGAGAGGCTCCTTGCCGGCGTT\t1\u001b[K\n",
      "chr1\t10007\t10157\tCCAACCGCGACCTTAGTGCATAGT\t1\u001b[K\n",
      "chr1\t10007\t10253\tGGCGGATATTCGCACCCGATGTTT\t1\u001b[K\n",
      "chr1\t10007\t10553\tTATCAAGGTGACCACTGGTGAGTT\t1\u001b[K\n",
      "finished run_generate_insert_size_histogram_data\u001b[K\n",
      "\u001b[K\n",
      "start run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "calculation_script is compute_tss_enrichment_snapatac2.py. check that it is: snapatac2\u001b[K\n",
      "no_singleton_bed_gz: no-singleton.bed.gz\u001b[K\n",
      "genes_gtf_gzip_file_out: genes.gtf.gz\u001b[K\n",
      "min_frag_cutoff: 10\u001b[K\n",
      "prefix: BMMC-single-donor\u001b[K\n",
      "output file: BMMC-single-donor.tss_enrichment_barcode_stats.tsv\u001b[K\n",
      "SnapATAC version: 2.5.0\u001b[K\n",
      "no_singleton_bed_file: no-singleton.bed.gz\u001b[K\n",
      "genes_gtf_file: genes.gtf.gz\u001b[K\n",
      "output_file: BMMC-single-donor.tss_enrichment_barcode_stats.tsv\u001b[K\n",
      "min_frag_cutoff: 10\u001b[K\n",
      "Loading data from fragment file...\u001b[K\n",
      "Calculating TSSE...\u001b[K\n",
      "Results saved to BMMC-single-donor.tss_enrichment_barcode_stats.tsv\u001b[K\n",
      "finished run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "\u001b[K\n",
      "\u001b[158A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2v39mmFiJZmXWq\u001b[22;39m\u001b[K\n",
      "[33/c79639] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[9f/40830d] process > run_add_subpool_to_chromap_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3e/470034] process > run_bgzip_on_chromap_fragme... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[d3/c0aca2] process > run_process_create_temp_sum... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[47/cceb9e] process > run_filter_aligned_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[aa/adbebc] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[7b/53117e] process > run_calculate_tss_enrichmen... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[06/5b09cc] process > run_barcode_metadata_stats (1) [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_atac_barcode_rank_plot     -\u001b[K\n",
      "[d8/bdb929] process > run_generate_insert_size_hi... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  [  0%] 0 of 1\u001b[K\n",
      "[2d/d3c262] process > run_merge_chromap_logs (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call \u001b[K\n",
      "after call run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "before run_process_create_temp_summary_dict\u001b[K\n",
      "after run_process_create_temp_summary_dict\u001b[K\n",
      "before run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb 14 01:34 genes.gtf.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb 14 01:34 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/41/45671d165921b1e3a885b4ea55909e/genes.gtf\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "Input chromap_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_chromap_output.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "\u001b[K\n",
      "Start run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "Fragment file is: ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "total 458900\u001b[K\n",
      "-rw-r--r-- 1 root root 469912630 Feb 14 02:24 ATAC_BMMC-single-donor.atac.filter.fragments.tsv.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb 14 02:24 ATAC_BMMC-single-donor.atac.filter.fragments.tsv -> /fusion/gs/eila-nextflow-bucket/work/ATAC/9f/40830dbfeebcb71ec69808fa3bfd6c/ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Finished run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "\u001b[K\n",
      "start run_process_create_temp_summary_dict\u001b[K\n",
      "input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "------ There is NO conversion list ------\u001b[K\n",
      "temp_summary\u001b[K\n",
      "7021039 temp_summary\u001b[K\n",
      "finished run_process_create_temp_summary_dict\u001b[K\n",
      "\u001b[K\n",
      "start run_merge_logs\u001b[K\n",
      "input script_name is merge_chromap_logs.sh\u001b[K\n",
      "input chromap_alignment_log is BMMC-single-donor.atac.align.k4.hg38.log.txt\u001b[K\n",
      "input chromap_barcode_log is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "start run_merge_chromap_logs shell script\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "********1***********\u001b[K\n",
      "********2***********\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "uni-mappings,260650161\u001b[K\n",
      "multi-mappings,14666503\u001b[K\n",
      "total,275316664\u001b[K\n",
      "********3***********\u001b[K\n",
      "finished run_merge_chromap_logs shell script\u001b[K\n",
      "finished run_merge_logs\u001b[K\n",
      "\u001b[K\n",
      "start run_filter_aligned_fragments_sort_gz\u001b[K\n",
      "input barcodeFragmentsTsvGz is ATAC_BMMC-single-donor.atac.filter.fragments.tsv.gz\u001b[K\n",
      "input threshold is 10\u001b[K\n",
      "input temp_summary_dict is temp_summary\u001b[K\n",
      "------ Number of barcodes BEFORE filtering------\u001b[K\n",
      "7021039 temp_summary\u001b[K\n",
      "------ Filtering fragments ------\u001b[K\n",
      "------ Number of barcodes AFTER filtering ------\u001b[K\n",
      "3950969\u001b[K\n",
      "finished run_filter_aligned_fragments_sort_gz\u001b[K\n",
      "\u001b[K\n",
      "start run_generate_insert_size_histogram_data\u001b[K\n",
      "script_name is generate_insert_size_histogram_data.sh\u001b[K\n",
      "in_chromap_bzip_fragments_tsv is no-singleton.bed.gz\u001b[K\n",
      "output_log_file: no-singleton.bed.hist.log.txt\u001b[K\n",
      "input_file: no-singleton.bed.gz\u001b[K\n",
      "Head of input_file:\u001b[K\n",
      "chr1\t10006\t10266\tCAGGAATCGAATCTGTGTACATCT\t1\u001b[K\n",
      "chr1\t10006\t10512\tAGTCATGGATCTAGTTTATCAAGG\t1\u001b[K\n",
      "chr1\t10007\t10113\tTACTTCGGTAGAACACGGATTAGG\t1\u001b[K\n",
      "chr1\t10007\t10113\tTCAGATTCCTAATATGTCATTGAG\t1\u001b[K\n",
      "chr1\t10007\t10114\tTGCGTGAATCAGGAGGACAGTGGT\t1\u001b[K\n",
      "chr1\t10007\t10132\tTGATACGTTAGACGGATCCTCAAT\t1\u001b[K\n",
      "chr1\t10007\t10151\tGAGGAGAGGCTCCTTGCCGGCGTT\t1\u001b[K\n",
      "chr1\t10007\t10157\tCCAACCGCGACCTTAGTGCATAGT\t1\u001b[K\n",
      "chr1\t10007\t10253\tGGCGGATATTCGCACCCGATGTTT\t1\u001b[K\n",
      "chr1\t10007\t10553\tTATCAAGGTGACCACTGGTGAGTT\t1\u001b[K\n",
      "finished run_generate_insert_size_histogram_data\u001b[K\n",
      "\u001b[K\n",
      "start run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "calculation_script is compute_tss_enrichment_snapatac2.py. check that it is: snapatac2\u001b[K\n",
      "no_singleton_bed_gz: no-singleton.bed.gz\u001b[K\n",
      "genes_gtf_gzip_file_out: genes.gtf.gz\u001b[K\n",
      "min_frag_cutoff: 10\u001b[K\n",
      "prefix: BMMC-single-donor\u001b[K\n",
      "output file: BMMC-single-donor.tss_enrichment_barcode_stats.tsv\u001b[K\n",
      "SnapATAC version: 2.5.0\u001b[K\n",
      "no_singleton_bed_file: no-singleton.bed.gz\u001b[K\n",
      "genes_gtf_file: genes.gtf.gz\u001b[K\n",
      "output_file: BMMC-single-donor.tss_enrichment_barcode_stats.tsv\u001b[K\n",
      "min_frag_cutoff: 10\u001b[K\n",
      "Loading data from fragment file...\u001b[K\n",
      "Calculating TSSE...\u001b[K\n",
      "Results saved to BMMC-single-donor.tss_enrichment_barcode_stats.tsv\u001b[K\n",
      "finished run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "\u001b[K\n",
      "------ START: Run Barcode Metadata Stats Script ------\u001b[K\n",
      "Input temp_dict_conversion: temp_summary\u001b[K\n",
      "Input tss_enrichment_barcode_stats: BMMC-single-donor.tss_enrichment_barcode_stats.tsv\u001b[K\n",
      "Output tmp_barcode_stats: tmp_barcode_stats.tsv\u001b[K\n",
      "Output barcodes_passing_threshold: barcodes_passing_threshold.txt\u001b[K\n",
      "------ START: Output tmp_barcode_stats ------\u001b[K\n",
      "Column1\tColumn2\tunique\tpct_dup\tpct_unmapped\u001b[K\n",
      "AACCTTCTAACCTTCTGACGGATT\t11\t6.24026\t0.0331992\t0.413223\u001b[K\n",
      "AACCTTCTAACCTTCTGTGTCCTT\t277\t239.651\t0.00101365\t0.133956\u001b[K\n",
      "AACCTTCTAACCTTCTTAACCAGT\t218\t204.674\t0.00129791\t0.0599091\u001b[K\n",
      "AACCTTCTAAGTTCCAACAGTGGT\t14\t10.4118\t0.0118632\t0.247372\u001b[K\n",
      "AACCTTCTAAGTTCCAATCCGGAT\t10\t8.03763\t0.0279098\t0.17316\u001b[K\n",
      "AACCTTCTAAGTTCCACGATGTTT\t147\t122.337\t0.00213141\t0.165999\u001b[K\n",
      "AACCTTCTAAGTTCCAGCCTCGGT\t104\t93.6458\t0.00206521\t0.0976964\u001b[K\n",
      "AACCTTCTAAGTTCCAGGATTCTC\t51\t46.0795\t0.00383552\t0.0930016\u001b[K\n",
      "AACCTTCTAATCTCGCGAGCCAAT\t204\t189.949\t0.00153314\t0.0674471\u001b[K\n",
      "------ END: Output tmp_barcode_stats ------\u001b[K\n",
      "------ START: Output barcodes_passing_threshold ------\u001b[K\n",
      "barcode\u001b[K\n",
      "AACCTTCTAACCTTCTGACGGATT\u001b[K\n",
      "AACCTTCTAACCTTCTGTGTCCTT\u001b[K\n",
      "AACCTTCTAACCTTCTTAACCAGT\u001b[K\n",
      "AACCTTCTAAGTTCCAACAGTGGT\u001b[K\n",
      "AACCTTCTAAGTTCCAATCCGGAT\u001b[K\n",
      "AACCTTCTAAGTTCCACGATGTTT\u001b[K\n",
      "AACCTTCTAAGTTCCAGCCTCGGT\u001b[K\n",
      "AACCTTCTAAGTTCCAGGATTCTC\u001b[K\n",
      "AACCTTCTAATCTCGCGAGCCAAT\u001b[K\n",
      "------ END: Output barcodes_passing_threshold ------\u001b[K\n",
      "------ FINISH: Run Barcode Metadata Stats Script ------\u001b[K\n",
      "\u001b[K\n",
      "\u001b[189A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2v39mmFiJZmXWq\u001b[22;39m\u001b[K\n",
      "[33/c79639] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[9f/40830d] process > run_add_subpool_to_chromap_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3e/470034] process > run_bgzip_on_chromap_fragme... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[d3/c0aca2] process > run_process_create_temp_sum... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[47/cceb9e] process > run_filter_aligned_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[aa/adbebc] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[7b/53117e] process > run_calculate_tss_enrichmen... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[06/5b09cc] process > run_barcode_metadata_stats (1) [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[bd/22c6af] process > run_atac_barcode_rank_plot (1) [100%] 1 of 1, cached: 1\u001b[K\n",
      "[d8/bdb929] process > run_generate_insert_size_hi... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  [  0%] 0 of 1\u001b[K\n",
      "[2d/d3c262] process > run_merge_chromap_logs (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call \u001b[K\n",
      "after call run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "before run_process_create_temp_summary_dict\u001b[K\n",
      "after run_process_create_temp_summary_dict\u001b[K\n",
      "before run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb 14 01:34 genes.gtf.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb 14 01:34 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/41/45671d165921b1e3a885b4ea55909e/genes.gtf\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "Input chromap_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_chromap_output.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "\u001b[K\n",
      "Start run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "Fragment file is: ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "total 458900\u001b[K\n",
      "-rw-r--r-- 1 root root 469912630 Feb 14 02:24 ATAC_BMMC-single-donor.atac.filter.fragments.tsv.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb 14 02:24 ATAC_BMMC-single-donor.atac.filter.fragments.tsv -> /fusion/gs/eila-nextflow-bucket/work/ATAC/9f/40830dbfeebcb71ec69808fa3bfd6c/ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Finished run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "\u001b[K\n",
      "start run_process_create_temp_summary_dict\u001b[K\n",
      "input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "------ There is NO conversion list ------\u001b[K\n",
      "temp_summary\u001b[K\n",
      "7021039 temp_summary\u001b[K\n",
      "finished run_process_create_temp_summary_dict\u001b[K\n",
      "\u001b[K\n",
      "start run_merge_logs\u001b[K\n",
      "input script_name is merge_chromap_logs.sh\u001b[K\n",
      "input chromap_alignment_log is BMMC-single-donor.atac.align.k4.hg38.log.txt\u001b[K\n",
      "input chromap_barcode_log is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "start run_merge_chromap_logs shell script\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "********1***********\u001b[K\n",
      "********2***********\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "uni-mappings,260650161\u001b[K\n",
      "multi-mappings,14666503\u001b[K\n",
      "total,275316664\u001b[K\n",
      "********3***********\u001b[K\n",
      "finished run_merge_chromap_logs shell script\u001b[K\n",
      "finished run_merge_logs\u001b[K\n",
      "\u001b[K\n",
      "start run_filter_aligned_fragments_sort_gz\u001b[K\n",
      "input barcodeFragmentsTsvGz is ATAC_BMMC-single-donor.atac.filter.fragments.tsv.gz\u001b[K\n",
      "input threshold is 10\u001b[K\n",
      "input temp_summary_dict is temp_summary\u001b[K\n",
      "------ Number of barcodes BEFORE filtering------\u001b[K\n",
      "7021039 temp_summary\u001b[K\n",
      "------ Filtering fragments ------\u001b[K\n",
      "------ Number of barcodes AFTER filtering ------\u001b[K\n",
      "3950969\u001b[K\n",
      "finished run_filter_aligned_fragments_sort_gz\u001b[K\n",
      "\u001b[K\n",
      "start run_generate_insert_size_histogram_data\u001b[K\n",
      "script_name is generate_insert_size_histogram_data.sh\u001b[K\n",
      "in_chromap_bzip_fragments_tsv is no-singleton.bed.gz\u001b[K\n",
      "output_log_file: no-singleton.bed.hist.log.txt\u001b[K\n",
      "input_file: no-singleton.bed.gz\u001b[K\n",
      "Head of input_file:\u001b[K\n",
      "chr1\t10006\t10266\tCAGGAATCGAATCTGTGTACATCT\t1\u001b[K\n",
      "chr1\t10006\t10512\tAGTCATGGATCTAGTTTATCAAGG\t1\u001b[K\n",
      "chr1\t10007\t10113\tTACTTCGGTAGAACACGGATTAGG\t1\u001b[K\n",
      "chr1\t10007\t10113\tTCAGATTCCTAATATGTCATTGAG\t1\u001b[K\n",
      "chr1\t10007\t10114\tTGCGTGAATCAGGAGGACAGTGGT\t1\u001b[K\n",
      "chr1\t10007\t10132\tTGATACGTTAGACGGATCCTCAAT\t1\u001b[K\n",
      "chr1\t10007\t10151\tGAGGAGAGGCTCCTTGCCGGCGTT\t1\u001b[K\n",
      "chr1\t10007\t10157\tCCAACCGCGACCTTAGTGCATAGT\t1\u001b[K\n",
      "chr1\t10007\t10253\tGGCGGATATTCGCACCCGATGTTT\t1\u001b[K\n",
      "chr1\t10007\t10553\tTATCAAGGTGACCACTGGTGAGTT\t1\u001b[K\n",
      "finished run_generate_insert_size_histogram_data\u001b[K\n",
      "\u001b[K\n",
      "start run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "calculation_script is compute_tss_enrichment_snapatac2.py. check that it is: snapatac2\u001b[K\n",
      "no_singleton_bed_gz: no-singleton.bed.gz\u001b[K\n",
      "genes_gtf_gzip_file_out: genes.gtf.gz\u001b[K\n",
      "min_frag_cutoff: 10\u001b[K\n",
      "prefix: BMMC-single-donor\u001b[K\n",
      "output file: BMMC-single-donor.tss_enrichment_barcode_stats.tsv\u001b[K\n",
      "SnapATAC version: 2.5.0\u001b[K\n",
      "no_singleton_bed_file: no-singleton.bed.gz\u001b[K\n",
      "genes_gtf_file: genes.gtf.gz\u001b[K\n",
      "output_file: BMMC-single-donor.tss_enrichment_barcode_stats.tsv\u001b[K\n",
      "min_frag_cutoff: 10\u001b[K\n",
      "Loading data from fragment file...\u001b[K\n",
      "Calculating TSSE...\u001b[K\n",
      "Results saved to BMMC-single-donor.tss_enrichment_barcode_stats.tsv\u001b[K\n",
      "finished run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "\u001b[K\n",
      "------ START: Run Barcode Metadata Stats Script ------\u001b[K\n",
      "Input temp_dict_conversion: temp_summary\u001b[K\n",
      "Input tss_enrichment_barcode_stats: BMMC-single-donor.tss_enrichment_barcode_stats.tsv\u001b[K\n",
      "Output tmp_barcode_stats: tmp_barcode_stats.tsv\u001b[K\n",
      "Output barcodes_passing_threshold: barcodes_passing_threshold.txt\u001b[K\n",
      "------ START: Output tmp_barcode_stats ------\u001b[K\n",
      "Column1\tColumn2\tunique\tpct_dup\tpct_unmapped\u001b[K\n",
      "AACCTTCTAACCTTCTGACGGATT\t11\t6.24026\t0.0331992\t0.413223\u001b[K\n",
      "AACCTTCTAACCTTCTGTGTCCTT\t277\t239.651\t0.00101365\t0.133956\u001b[K\n",
      "AACCTTCTAACCTTCTTAACCAGT\t218\t204.674\t0.00129791\t0.0599091\u001b[K\n",
      "AACCTTCTAAGTTCCAACAGTGGT\t14\t10.4118\t0.0118632\t0.247372\u001b[K\n",
      "AACCTTCTAAGTTCCAATCCGGAT\t10\t8.03763\t0.0279098\t0.17316\u001b[K\n",
      "AACCTTCTAAGTTCCACGATGTTT\t147\t122.337\t0.00213141\t0.165999\u001b[K\n",
      "AACCTTCTAAGTTCCAGCCTCGGT\t104\t93.6458\t0.00206521\t0.0976964\u001b[K\n",
      "AACCTTCTAAGTTCCAGGATTCTC\t51\t46.0795\t0.00383552\t0.0930016\u001b[K\n",
      "AACCTTCTAATCTCGCGAGCCAAT\t204\t189.949\t0.00153314\t0.0674471\u001b[K\n",
      "------ END: Output tmp_barcode_stats ------\u001b[K\n",
      "------ START: Output barcodes_passing_threshold ------\u001b[K\n",
      "barcode\u001b[K\n",
      "AACCTTCTAACCTTCTGACGGATT\u001b[K\n",
      "AACCTTCTAACCTTCTGTGTCCTT\u001b[K\n",
      "AACCTTCTAACCTTCTTAACCAGT\u001b[K\n",
      "AACCTTCTAAGTTCCAACAGTGGT\u001b[K\n",
      "AACCTTCTAAGTTCCAATCCGGAT\u001b[K\n",
      "AACCTTCTAAGTTCCACGATGTTT\u001b[K\n",
      "AACCTTCTAAGTTCCAGCCTCGGT\u001b[K\n",
      "AACCTTCTAAGTTCCAGGATTCTC\u001b[K\n",
      "AACCTTCTAATCTCGCGAGCCAAT\u001b[K\n",
      "------ END: Output barcodes_passing_threshold ------\u001b[K\n",
      "------ FINISH: Run Barcode Metadata Stats Script ------\u001b[K\n",
      "\u001b[K\n",
      "\u001b[189A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2v39mmFiJZmXWq\u001b[22;39m\u001b[K\n",
      "[33/c79639] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[9f/40830d] process > run_add_subpool_to_chromap_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3e/470034] process > run_bgzip_on_chromap_fragme... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[d3/c0aca2] process > run_process_create_temp_sum... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[47/cceb9e] process > run_filter_aligned_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[aa/adbebc] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[7b/53117e] process > run_calculate_tss_enrichmen... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[06/5b09cc] process > run_barcode_metadata_stats (1) [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[bd/22c6af] process > run_atac_barcode_rank_plot (1) [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[d8/bdb929] process > run_generate_insert_size_hi... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  [  0%] 0 of 1\u001b[K\n",
      "[2d/d3c262] process > run_merge_chromap_logs (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call \u001b[K\n",
      "after call run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "before run_process_create_temp_summary_dict\u001b[K\n",
      "after run_process_create_temp_summary_dict\u001b[K\n",
      "before run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb 14 01:34 genes.gtf.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb 14 01:34 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/41/45671d165921b1e3a885b4ea55909e/genes.gtf\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "Input chromap_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_chromap_output.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "\u001b[K\n",
      "Start run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "Fragment file is: ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "total 458900\u001b[K\n",
      "-rw-r--r-- 1 root root 469912630 Feb 14 02:24 ATAC_BMMC-single-donor.atac.filter.fragments.tsv.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb 14 02:24 ATAC_BMMC-single-donor.atac.filter.fragments.tsv -> /fusion/gs/eila-nextflow-bucket/work/ATAC/9f/40830dbfeebcb71ec69808fa3bfd6c/ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Finished run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "\u001b[K\n",
      "start run_process_create_temp_summary_dict\u001b[K\n",
      "input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "------ There is NO conversion list ------\u001b[K\n",
      "temp_summary\u001b[K\n",
      "7021039 temp_summary\u001b[K\n",
      "finished run_process_create_temp_summary_dict\u001b[K\n",
      "\u001b[K\n",
      "start run_merge_logs\u001b[K\n",
      "input script_name is merge_chromap_logs.sh\u001b[K\n",
      "input chromap_alignment_log is BMMC-single-donor.atac.align.k4.hg38.log.txt\u001b[K\n",
      "input chromap_barcode_log is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "start run_merge_chromap_logs shell script\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "********1***********\u001b[K\n",
      "********2***********\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "uni-mappings,260650161\u001b[K\n",
      "multi-mappings,14666503\u001b[K\n",
      "total,275316664\u001b[K\n",
      "********3***********\u001b[K\n",
      "finished run_merge_chromap_logs shell script\u001b[K\n",
      "finished run_merge_logs\u001b[K\n",
      "\u001b[K\n",
      "start run_filter_aligned_fragments_sort_gz\u001b[K\n",
      "input barcodeFragmentsTsvGz is ATAC_BMMC-single-donor.atac.filter.fragments.tsv.gz\u001b[K\n",
      "input threshold is 10\u001b[K\n",
      "input temp_summary_dict is temp_summary\u001b[K\n",
      "------ Number of barcodes BEFORE filtering------\u001b[K\n",
      "7021039 temp_summary\u001b[K\n",
      "------ Filtering fragments ------\u001b[K\n",
      "------ Number of barcodes AFTER filtering ------\u001b[K\n",
      "3950969\u001b[K\n",
      "finished run_filter_aligned_fragments_sort_gz\u001b[K\n",
      "\u001b[K\n",
      "start run_generate_insert_size_histogram_data\u001b[K\n",
      "script_name is generate_insert_size_histogram_data.sh\u001b[K\n",
      "in_chromap_bzip_fragments_tsv is no-singleton.bed.gz\u001b[K\n",
      "output_log_file: no-singleton.bed.hist.log.txt\u001b[K\n",
      "input_file: no-singleton.bed.gz\u001b[K\n",
      "Head of input_file:\u001b[K\n",
      "chr1\t10006\t10266\tCAGGAATCGAATCTGTGTACATCT\t1\u001b[K\n",
      "chr1\t10006\t10512\tAGTCATGGATCTAGTTTATCAAGG\t1\u001b[K\n",
      "chr1\t10007\t10113\tTACTTCGGTAGAACACGGATTAGG\t1\u001b[K\n",
      "chr1\t10007\t10113\tTCAGATTCCTAATATGTCATTGAG\t1\u001b[K\n",
      "chr1\t10007\t10114\tTGCGTGAATCAGGAGGACAGTGGT\t1\u001b[K\n",
      "chr1\t10007\t10132\tTGATACGTTAGACGGATCCTCAAT\t1\u001b[K\n",
      "chr1\t10007\t10151\tGAGGAGAGGCTCCTTGCCGGCGTT\t1\u001b[K\n",
      "chr1\t10007\t10157\tCCAACCGCGACCTTAGTGCATAGT\t1\u001b[K\n",
      "chr1\t10007\t10253\tGGCGGATATTCGCACCCGATGTTT\t1\u001b[K\n",
      "chr1\t10007\t10553\tTATCAAGGTGACCACTGGTGAGTT\t1\u001b[K\n",
      "finished run_generate_insert_size_histogram_data\u001b[K\n",
      "\u001b[K\n",
      "start run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "calculation_script is compute_tss_enrichment_snapatac2.py. check that it is: snapatac2\u001b[K\n",
      "no_singleton_bed_gz: no-singleton.bed.gz\u001b[K\n",
      "genes_gtf_gzip_file_out: genes.gtf.gz\u001b[K\n",
      "min_frag_cutoff: 10\u001b[K\n",
      "prefix: BMMC-single-donor\u001b[K\n",
      "output file: BMMC-single-donor.tss_enrichment_barcode_stats.tsv\u001b[K\n",
      "SnapATAC version: 2.5.0\u001b[K\n",
      "no_singleton_bed_file: no-singleton.bed.gz\u001b[K\n",
      "genes_gtf_file: genes.gtf.gz\u001b[K\n",
      "output_file: BMMC-single-donor.tss_enrichment_barcode_stats.tsv\u001b[K\n",
      "min_frag_cutoff: 10\u001b[K\n",
      "Loading data from fragment file...\u001b[K\n",
      "Calculating TSSE...\u001b[K\n",
      "Results saved to BMMC-single-donor.tss_enrichment_barcode_stats.tsv\u001b[K\n",
      "finished run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "\u001b[K\n",
      "------ START: Run Barcode Metadata Stats Script ------\u001b[K\n",
      "Input temp_dict_conversion: temp_summary\u001b[K\n",
      "Input tss_enrichment_barcode_stats: BMMC-single-donor.tss_enrichment_barcode_stats.tsv\u001b[K\n",
      "Output tmp_barcode_stats: tmp_barcode_stats.tsv\u001b[K\n",
      "Output barcodes_passing_threshold: barcodes_passing_threshold.txt\u001b[K\n",
      "------ START: Output tmp_barcode_stats ------\u001b[K\n",
      "Column1\tColumn2\tunique\tpct_dup\tpct_unmapped\u001b[K\n",
      "AACCTTCTAACCTTCTGACGGATT\t11\t6.24026\t0.0331992\t0.413223\u001b[K\n",
      "AACCTTCTAACCTTCTGTGTCCTT\t277\t239.651\t0.00101365\t0.133956\u001b[K\n",
      "AACCTTCTAACCTTCTTAACCAGT\t218\t204.674\t0.00129791\t0.0599091\u001b[K\n",
      "AACCTTCTAAGTTCCAACAGTGGT\t14\t10.4118\t0.0118632\t0.247372\u001b[K\n",
      "AACCTTCTAAGTTCCAATCCGGAT\t10\t8.03763\t0.0279098\t0.17316\u001b[K\n",
      "AACCTTCTAAGTTCCACGATGTTT\t147\t122.337\t0.00213141\t0.165999\u001b[K\n",
      "AACCTTCTAAGTTCCAGCCTCGGT\t104\t93.6458\t0.00206521\t0.0976964\u001b[K\n",
      "AACCTTCTAAGTTCCAGGATTCTC\t51\t46.0795\t0.00383552\t0.0930016\u001b[K\n",
      "AACCTTCTAATCTCGCGAGCCAAT\t204\t189.949\t0.00153314\t0.0674471\u001b[K\n",
      "------ END: Output tmp_barcode_stats ------\u001b[K\n",
      "------ START: Output barcodes_passing_threshold ------\u001b[K\n",
      "barcode\u001b[K\n",
      "AACCTTCTAACCTTCTGACGGATT\u001b[K\n",
      "AACCTTCTAACCTTCTGTGTCCTT\u001b[K\n",
      "AACCTTCTAACCTTCTTAACCAGT\u001b[K\n",
      "AACCTTCTAAGTTCCAACAGTGGT\u001b[K\n",
      "AACCTTCTAAGTTCCAATCCGGAT\u001b[K\n",
      "AACCTTCTAAGTTCCACGATGTTT\u001b[K\n",
      "AACCTTCTAAGTTCCAGCCTCGGT\u001b[K\n",
      "AACCTTCTAAGTTCCAGGATTCTC\u001b[K\n",
      "AACCTTCTAATCTCGCGAGCCAAT\u001b[K\n",
      "------ END: Output barcodes_passing_threshold ------\u001b[K\n",
      "------ FINISH: Run Barcode Metadata Stats Script ------\u001b[K\n",
      "\u001b[K\n",
      "Starting run_scrna_atac_plot_qc_metrics...\u001b[K\n",
      "R QC ATAC Plot Script: atac_generate_metadata_barcode_rank_plot.R\u001b[K\n",
      "R QC ATAC Plot Helper Script: barcode_rank_functions.R\u001b[K\n",
      "barcode_metadata_file: tmp_barcode_stats.tsv\u001b[K\n",
      "fragment_cutoff: 10\u001b[K\n",
      "fragment_rank_plot_file_output: /work/ATAC/7b/53117e8073042ba5730ffb766ef157/BMMC-single-donor.tss_enrichment_barcode_stats.tsv\u001b[K\n",
      "Script path: /usr/local/bin/atac_generate_metadata_barcode_rank_plot.R\u001b[K\n",
      "ls /usr/local/bin/atac_generate_metadata_barcode_rank_plot.R\u001b[K\n",
      "Running the following conda command...\u001b[K\n",
      "conda run -n rna_atac_plot_qc_env Rscript /usr/local/bin/atac_generate_metadata_barcode_rank_plot.R barcode_rank_functions.R tmp_barcode_stats.tsv 10 BMMC-single-donor.tss_enrichment_barcode_stats\u001b[K\n",
      "[1] \"atac_plot_metadata: r_qc_plot_helper_script is: barcode_rank_functions.R\"\u001b[K\n",
      "[1] \"At source_file_in_path\"\u001b[K\n",
      "Found file: /usr/local/bin/barcode_rank_functions.R \u001b[K\n",
      "[1] \"&&&&&&&&&&&source barcode_rank_functions.R\"\u001b[K\n",
      "[1] \"atac_plot_metadata: barcode_metadata_file is: tmp_barcode_stats.tsv\"\u001b[K\n",
      "[1] \"atac_plot_metadata: fragment_cutoff is: 10\"\u001b[K\n",
      "[1] \"atac_plot_metadata: fragment_rank_plot_file is: BMMC-single-donor.tss_enrichment_barcode_stats.png\"\u001b[K\n",
      "[1] \"atac_plot_metadata: Reading barcode metadata from file: tmp_barcode_stats.tsv\"\u001b[K\n",
      "[1] \"atac_plot_metadata: Reading barcode metadata...\"\u001b[K\n",
      "[1] \"atac_plot_metadata: Barcode metadata read successfully.\"\u001b[K\n",
      "[1] \"atac_plot_metadata: Head of barcode_metadata data:\"\u001b[K\n",
      "                   Column1 Column2    unique    pct_dup pct_unmapped\u001b[K\n",
      "1 AACCTTCTAACCTTCTGACGGATT      11   6.24026 0.03319920    0.4132230\u001b[K\n",
      "2 AACCTTCTAACCTTCTGTGTCCTT     277 239.65100 0.00101365    0.1339560\u001b[K\n",
      "3 AACCTTCTAACCTTCTTAACCAGT     218 204.67400 0.00129791    0.0599091\u001b[K\n",
      "4 AACCTTCTAAGTTCCAACAGTGGT      14  10.41180 0.01186320    0.2473720\u001b[K\n",
      "5 AACCTTCTAAGTTCCAATCCGGAT      10   8.03763 0.02790980    0.1731600\u001b[K\n",
      "6 AACCTTCTAAGTTCCACGATGTTT     147 122.33700 0.00213141    0.1659990\u001b[K\n",
      "[1] \"atac_plot_metadata: Processing fragment data...\"\u001b[K\n",
      "[1] \"atac_plot_metadata: Fragment data processed successfully.\"\u001b[K\n",
      "[1] \"atac_plot_metadata: Finding elbow/knee points...\"\u001b[K\n",
      "[1] \"Starting get_elbow_knee_points method.\"\u001b[K\n",
      "[1] \"Starting elbow_knee_finder method.\"\u001b[K\n",
      "[1] \"Starting elbow_knee_finder method.\"\u001b[K\n",
      "[1] \"Starting get_vectors method.\"\u001b[K\n",
      "[1] \"Returning original vectors\"\u001b[K\n",
      "[1] \"atac_plot_metadata: Elbow/knee points found successfully.\"\u001b[K\n",
      "[1] \"atac_plot_metadata: Generating fragment barcode rank plots...\"\u001b[K\n",
      "[1] \"atac_plot_metadata: Plots generated successfully.\"\u001b[K\n",
      "\u001b[K\n",
      "Finished run_scrna_atac_plot_qc_metrics.\u001b[K\n",
      "\u001b[K\n",
      "\u001b[231A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2v39mmFiJZmXWq\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (1)\u001b[K\n",
      "[33/c79639] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[9f/40830d] process > run_add_subpool_to_chromap_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3e/470034] process > run_bgzip_on_chromap_fragme... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[d3/c0aca2] process > run_process_create_temp_sum... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[47/cceb9e] process > run_filter_aligned_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[aa/adbebc] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[7b/53117e] process > run_calculate_tss_enrichmen... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[06/5b09cc] process > run_barcode_metadata_stats (1) [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[bd/22c6af] process > run_atac_barcode_rank_plot (1) [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[d8/bdb929] process > run_generate_insert_size_hi... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[ac/f5bb73] process > run_generate_insert_size_pl... [  0%] 0 of 1\u001b[K\n",
      "[2d/d3c262] process > run_merge_chromap_logs (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call \u001b[K\n",
      "after call run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "before run_process_create_temp_summary_dict\u001b[K\n",
      "after run_process_create_temp_summary_dict\u001b[K\n",
      "before run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_filter_aligned_fragments_gz\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb 14 01:35 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb 14 01:34 genes.gtf.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb 14 01:34 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/41/45671d165921b1e3a885b4ea55909e/genes.gtf\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "Input chromap_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_chromap_output.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_subpool_to_chrompap_output ------\u001b[K\n",
      "\u001b[K\n",
      "Start run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "Fragment file is: ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "total 458900\u001b[K\n",
      "-rw-r--r-- 1 root root 469912630 Feb 14 02:24 ATAC_BMMC-single-donor.atac.filter.fragments.tsv.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb 14 02:24 ATAC_BMMC-single-donor.atac.filter.fragments.tsv -> /fusion/gs/eila-nextflow-bucket/work/ATAC/9f/40830dbfeebcb71ec69808fa3bfd6c/ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Finished run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "\u001b[K\n",
      "start run_process_create_temp_summary_dict\u001b[K\n",
      "input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "------ There is NO conversion list ------\u001b[K\n",
      "temp_summary\u001b[K\n",
      "7021039 temp_summary\u001b[K\n",
      "finished run_process_create_temp_summary_dict\u001b[K\n",
      "\u001b[K\n",
      "start run_merge_logs\u001b[K\n",
      "input script_name is merge_chromap_logs.sh\u001b[K\n",
      "input chromap_alignment_log is BMMC-single-donor.atac.align.k4.hg38.log.txt\u001b[K\n",
      "input chromap_barcode_log is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "start run_merge_chromap_logs shell script\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "********1***********\u001b[K\n",
      "********2***********\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "uni-mappings,260650161\u001b[K\n",
      "multi-mappings,14666503\u001b[K\n",
      "total,275316664\u001b[K\n",
      "********3***********\u001b[K\n",
      "finished run_merge_chromap_logs shell script\u001b[K\n",
      "finished run_merge_logs\u001b[K\n",
      "\u001b[K\n",
      "start run_filter_aligned_fragments_sort_gz\u001b[K\n",
      "input barcodeFragmentsTsvGz is ATAC_BMMC-single-donor.atac.filter.fragments.tsv.gz\u001b[K\n",
      "input threshold is 10\u001b[K\n",
      "input temp_summary_dict is temp_summary\u001b[K\n",
      "------ Number of barcodes BEFORE filtering------\u001b[K\n",
      "7021039 temp_summary\u001b[K\n",
      "------ Filtering fragments ------\u001b[K\n",
      "------ Number of barcodes AFTER filtering ------\u001b[K\n",
      "3950969\u001b[K\n",
      "finished run_filter_aligned_fragments_sort_gz\u001b[K\n",
      "\u001b[K\n",
      "start run_generate_insert_size_histogram_data\u001b[K\n",
      "script_name is generate_insert_size_histogram_data.sh\u001b[K\n",
      "in_chromap_bzip_fragments_tsv is no-singleton.bed.gz\u001b[K\n",
      "output_log_file: no-singleton.bed.hist.log.txt\u001b[K\n",
      "input_file: no-singleton.bed.gz\u001b[K\n",
      "Head of input_file:\u001b[K\n",
      "chr1\t10006\t10266\tCAGGAATCGAATCTGTGTACATCT\t1\u001b[K\n",
      "chr1\t10006\t10512\tAGTCATGGATCTAGTTTATCAAGG\t1\u001b[K\n",
      "chr1\t10007\t10113\tTACTTCGGTAGAACACGGATTAGG\t1\u001b[K\n",
      "chr1\t10007\t10113\tTCAGATTCCTAATATGTCATTGAG\t1\u001b[K\n",
      "chr1\t10007\t10114\tTGCGTGAATCAGGAGGACAGTGGT\t1\u001b[K\n",
      "chr1\t10007\t10132\tTGATACGTTAGACGGATCCTCAAT\t1\u001b[K\n",
      "chr1\t10007\t10151\tGAGGAGAGGCTCCTTGCCGGCGTT\t1\u001b[K\n",
      "chr1\t10007\t10157\tCCAACCGCGACCTTAGTGCATAGT\t1\u001b[K\n",
      "chr1\t10007\t10253\tGGCGGATATTCGCACCCGATGTTT\t1\u001b[K\n",
      "chr1\t10007\t10553\tTATCAAGGTGACCACTGGTGAGTT\t1\u001b[K\n",
      "finished run_generate_insert_size_histogram_data\u001b[K\n",
      "\u001b[K\n",
      "start run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "calculation_script is compute_tss_enrichment_snapatac2.py. check that it is: snapatac2\u001b[K\n",
      "no_singleton_bed_gz: no-singleton.bed.gz\u001b[K\n",
      "genes_gtf_gzip_file_out: genes.gtf.gz\u001b[K\n",
      "min_frag_cutoff: 10\u001b[K\n",
      "prefix: BMMC-single-donor\u001b[K\n",
      "output file: BMMC-single-donor.tss_enrichment_barcode_stats.tsv\u001b[K\n",
      "SnapATAC version: 2.5.0\u001b[K\n",
      "no_singleton_bed_file: no-singleton.bed.gz\u001b[K\n",
      "genes_gtf_file: genes.gtf.gz\u001b[K\n",
      "output_file: BMMC-single-donor.tss_enrichment_barcode_stats.tsv\u001b[K\n",
      "min_frag_cutoff: 10\u001b[K\n",
      "Loading data from fragment file...\u001b[K\n",
      "Calculating TSSE...\u001b[K\n",
      "Results saved to BMMC-single-donor.tss_enrichment_barcode_stats.tsv\u001b[K\n",
      "finished run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "\u001b[K\n",
      "------ START: Run Barcode Metadata Stats Script ------\u001b[K\n",
      "Input temp_dict_conversion: temp_summary\u001b[K\n",
      "Input tss_enrichment_barcode_stats: BMMC-single-donor.tss_enrichment_barcode_stats.tsv\u001b[K\n",
      "Output tmp_barcode_stats: tmp_barcode_stats.tsv\u001b[K\n",
      "Output barcodes_passing_threshold: barcodes_passing_threshold.txt\u001b[K\n",
      "------ START: Output tmp_barcode_stats ------\u001b[K\n",
      "Column1\tColumn2\tunique\tpct_dup\tpct_unmapped\u001b[K\n",
      "AACCTTCTAACCTTCTGACGGATT\t11\t6.24026\t0.0331992\t0.413223\u001b[K\n",
      "AACCTTCTAACCTTCTGTGTCCTT\t277\t239.651\t0.00101365\t0.133956\u001b[K\n",
      "AACCTTCTAACCTTCTTAACCAGT\t218\t204.674\t0.00129791\t0.0599091\u001b[K\n",
      "AACCTTCTAAGTTCCAACAGTGGT\t14\t10.4118\t0.0118632\t0.247372\u001b[K\n",
      "AACCTTCTAAGTTCCAATCCGGAT\t10\t8.03763\t0.0279098\t0.17316\u001b[K\n",
      "AACCTTCTAAGTTCCACGATGTTT\t147\t122.337\t0.00213141\t0.165999\u001b[K\n",
      "AACCTTCTAAGTTCCAGCCTCGGT\t104\t93.6458\t0.00206521\t0.0976964\u001b[K\n",
      "AACCTTCTAAGTTCCAGGATTCTC\t51\t46.0795\t0.00383552\t0.0930016\u001b[K\n",
      "AACCTTCTAATCTCGCGAGCCAAT\t204\t189.949\t0.00153314\t0.0674471\u001b[K\n",
      "------ END: Output tmp_barcode_stats ------\u001b[K\n",
      "------ START: Output barcodes_passing_threshold ------\u001b[K\n",
      "barcode\u001b[K\n",
      "AACCTTCTAACCTTCTGACGGATT\u001b[K\n",
      "AACCTTCTAACCTTCTGTGTCCTT\u001b[K\n",
      "AACCTTCTAACCTTCTTAACCAGT\u001b[K\n",
      "AACCTTCTAAGTTCCAACAGTGGT\u001b[K\n",
      "AACCTTCTAAGTTCCAATCCGGAT\u001b[K\n",
      "AACCTTCTAAGTTCCACGATGTTT\u001b[K\n",
      "AACCTTCTAAGTTCCAGCCTCGGT\u001b[K\n",
      "AACCTTCTAAGTTCCAGGATTCTC\u001b[K\n",
      "AACCTTCTAATCTCGCGAGCCAAT\u001b[K\n",
      "------ END: Output barcodes_passing_threshold ------\u001b[K\n",
      "------ FINISH: Run Barcode Metadata Stats Script ------\u001b[K\n",
      "\u001b[K\n",
      "Starting run_scrna_atac_plot_qc_metrics...\u001b[K\n",
      "R QC ATAC Plot Script: atac_generate_metadata_barcode_rank_plot.R\u001b[K\n",
      "R QC ATAC Plot Helper Script: barcode_rank_functions.R\u001b[K\n",
      "barcode_metadata_file: tmp_barcode_stats.tsv\u001b[K\n",
      "fragment_cutoff: 10\u001b[K\n",
      "fragment_rank_plot_file_output: /work/ATAC/7b/53117e8073042ba5730ffb766ef157/BMMC-single-donor.tss_enrichment_barcode_stats.tsv\u001b[K\n",
      "Script path: /usr/local/bin/atac_generate_metadata_barcode_rank_plot.R\u001b[K\n",
      "ls /usr/local/bin/atac_generate_metadata_barcode_rank_plot.R\u001b[K\n",
      "Running the following conda command...\u001b[K\n",
      "conda run -n rna_atac_plot_qc_env Rscript /usr/local/bin/atac_generate_metadata_barcode_rank_plot.R barcode_rank_functions.R tmp_barcode_stats.tsv 10 BMMC-single-donor.tss_enrichment_barcode_stats\u001b[K\n",
      "[1] \"atac_plot_metadata: r_qc_plot_helper_script is: barcode_rank_functions.R\"\u001b[K\n",
      "[1] \"At source_file_in_path\"\u001b[K\n",
      "Found file: /usr/local/bin/barcode_rank_functions.R \u001b[K\n",
      "[1] \"&&&&&&&&&&&source barcode_rank_functions.R\"\u001b[K\n",
      "[1] \"atac_plot_metadata: barcode_metadata_file is: tmp_barcode_stats.tsv\"\u001b[K\n",
      "[1] \"atac_plot_metadata: fragment_cutoff is: 10\"\u001b[K\n",
      "[1] \"atac_plot_metadata: fragment_rank_plot_file is: BMMC-single-donor.tss_enrichment_barcode_stats.png\"\u001b[K\n",
      "[1] \"atac_plot_metadata: Reading barcode metadata from file: tmp_barcode_stats.tsv\"\u001b[K\n",
      "[1] \"atac_plot_metadata: Reading barcode metadata...\"\u001b[K\n",
      "[1] \"atac_plot_metadata: Barcode metadata read successfully.\"\u001b[K\n",
      "[1] \"atac_plot_metadata: Head of barcode_metadata data:\"\u001b[K\n",
      "                   Column1 Column2    unique    pct_dup pct_unmapped\u001b[K\n",
      "1 AACCTTCTAACCTTCTGACGGATT      11   6.24026 0.03319920    0.4132230\u001b[K\n",
      "2 AACCTTCTAACCTTCTGTGTCCTT     277 239.65100 0.00101365    0.1339560\u001b[K\n",
      "3 AACCTTCTAACCTTCTTAACCAGT     218 204.67400 0.00129791    0.0599091\u001b[K\n",
      "4 AACCTTCTAAGTTCCAACAGTGGT      14  10.41180 0.01186320    0.2473720\u001b[K\n",
      "5 AACCTTCTAAGTTCCAATCCGGAT      10   8.03763 0.02790980    0.1731600\u001b[K\n",
      "6 AACCTTCTAAGTTCCACGATGTTT     147 122.33700 0.00213141    0.1659990\u001b[K\n",
      "[1] \"atac_plot_metadata: Processing fragment data...\"\u001b[K\n",
      "[1] \"atac_plot_metadata: Fragment data processed successfully.\"\u001b[K\n",
      "[1] \"atac_plot_metadata: Finding elbow/knee points...\"\u001b[K\n",
      "[1] \"Starting get_elbow_knee_points method.\"\u001b[K\n",
      "[1] \"Starting elbow_knee_finder method.\"\u001b[K\n",
      "[1] \"Starting elbow_knee_finder method.\"\u001b[K\n",
      "[1] \"Starting get_vectors method.\"\u001b[K\n",
      "[1] \"Returning original vectors\"\u001b[K\n",
      "[1] \"atac_plot_metadata: Elbow/knee points found successfully.\"\u001b[K\n",
      "[1] \"atac_plot_metadata: Generating fragment barcode rank plots...\"\u001b[K\n",
      "[1] \"atac_plot_metadata: Plots generated successfully.\"\u001b[K\n",
      "\u001b[K\n",
      "Finished run_scrna_atac_plot_qc_metrics.\u001b[K\n",
      "\u001b[K\n",
      "\u001b[232A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2v39mmFiJZmXWq\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (1)\u001b[K\n",
      "[33/c79639] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[9f/40830d] process > run_add_subpool_to_chromap_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3e/470034] process > run_bgzip_on_chromap_fragme... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[d3/c0aca2] process > run_process_create_temp_sum... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[47/cceb9e] process > run_filter_aligned_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[aa/adbebc] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[7b/53117e] process > run_calculate_tss_enrichmen... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[06/5b09cc] process > run_barcode_metadata_stats (1) [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[bd/22c6af] process > run_atac_barcode_rank_plot (1) [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[d8/bdb929] process > run_generate_insert_size_hi... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[ac/f5bb73] process > run_generate_insert_size_pl... [  0%] 0 of 1\u001b[K\n",
      "[2d/d3c262] process > run_merge_chromap_logs (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[232A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2v39mmFiJZmXWq\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (1)\u001b[K\n",
      "[33/c79639] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[9f/40830d] process > run_add_subpool_to_chromap_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3e/470034] process > run_bgzip_on_chromap_fragme... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[d3/c0aca2] process > run_process_create_temp_sum... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[47/cceb9e] process > run_filter_aligned_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[aa/adbebc] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[7b/53117e] process > run_calculate_tss_enrichmen... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[06/5b09cc] process > run_barcode_metadata_stats (1) [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[bd/22c6af] process > run_atac_barcode_rank_plot (1) [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[d8/bdb929] process > run_generate_insert_size_hi... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[ac/f5bb73] process > run_generate_insert_size_pl... [100%] 1 of 1 ✔\u001b[K\n",
      "[2d/d3c262] process > run_merge_chromap_logs (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "start run_generate_insert_size_plot\u001b[K\n",
      "plot_script is plot_insert_size_hist.py\u001b[K\n",
      "histogram_file: no-singleton.bed.hist.log.txt\u001b[K\n",
      "pkr_id: BMM_PKR_ID\u001b[K\n",
      "Starting histogram plotting script\u001b[K\n",
      "Received arguments: ['/usr/local/bin/plot_insert_size_hist.py', '--histogram_file', 'no-singleton.bed.hist.log.txt', '--out_file', 'no-singleton.bed.hist.log.png', '--pkr', 'BMM_PKR_ID']\u001b[K\n",
      "Received arguments:\u001b[K\n",
      "histogram_file: no-singleton.bed.hist.log.txt\u001b[K\n",
      "pkr: BMM_PKR_ID\u001b[K\n",
      "out_file: no-singleton.bed.hist.log.png\u001b[K\n",
      "Finished plotting\u001b[K\n",
      "Formatting metadata...\u001b[K\n",
      "Displaying the head of the metadata file...\u001b[K\n",
      "Counting the fields in the metadata...\u001b[K\n",
      "finished run_generate_insert_size_plot\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[32;1mCompleted at: 14-Feb-2024 17:45:13\n",
      "Duration    : 2m 53s\n",
      "CPU hours   : 0.2 (98.4% cached)\n",
      "Succeeded   : 1\n",
      "Cached      : 11\n",
      "\u001b[22;39m\u001b[K\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_atac.config -profile google_batch_profile -process.echo -with-tower -resume \n",
    "# -with-dag -resume\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_vm.config -profile google_batch_profile\n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_vm.config -profile google_batch_profile\n",
    "# !nextflow run nf_\n",
    "# \n",
    "# wf_fg_singlecell_atac.nf -c nextflow_vm_anvil.config -profile google_batch_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_local.config -profile google_batch_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_local.config -profile docker_profile\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Feb-14 17:42:13.515 [main] DEBUG nextflow.cli.Launcher - $> nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_atac.config -profile google_batch_profile -process.echo -with-tower -resume\n",
      "Feb-14 17:42:13.787 [main] INFO  nextflow.cli.CmdRun - N E X T F L O W  ~  version 23.10.1\n",
      "Feb-14 17:42:13.818 [main] DEBUG nextflow.plugin.PluginsFacade - Setting up plugin manager > mode=prod; embedded=false; plugins-dir=/home/eila/.nextflow/plugins; core-plugins: nf-amazon@2.1.4,nf-azure@1.3.3,nf-cloudcache@0.3.0,nf-codecommit@0.1.5,nf-console@1.0.6,nf-ga4gh@1.1.0,nf-google@1.8.3,nf-tower@1.6.3,nf-wave@1.0.1\n",
      "Feb-14 17:42:13.835 [main] INFO  o.pf4j.DefaultPluginStatusProvider - Enabled plugins: []\n",
      "Feb-14 17:42:13.836 [main] INFO  o.pf4j.DefaultPluginStatusProvider - Disabled plugins: []\n",
      "Feb-14 17:42:13.840 [main] INFO  org.pf4j.DefaultPluginManager - PF4J version 3.4.1 in 'deployment' mode\n",
      "Feb-14 17:42:13.858 [main] INFO  org.pf4j.AbstractPluginManager - No plugins\n",
      "Feb-14 17:42:13.889 [main] DEBUG nextflow.config.ConfigBuilder - User config file: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Feb-14 17:42:13.890 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Feb-14 17:42:13.933 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `google_batch_profile`\n",
      "Feb-14 17:42:16.779 [main] DEBUG nextflow.config.ConfigBuilder - Available config profiles: [google_batch_profile]\n",
      "Feb-14 17:42:16.843 [main] DEBUG nextflow.cli.CmdRun - Applied DSL=2 by global default\n",
      "Feb-14 17:42:16.872 [main] INFO  nextflow.cli.CmdRun - Launching `nf_wf_fg_singlecell_atac.nf` [insane_woese] DSL2 - revision: 74ff183a6a\n",
      "Feb-14 17:42:16.873 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins default=[nf-google@1.8.3]\n",
      "Feb-14 17:42:16.874 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins resolved requirement=[nf-google@1.8.3, nf-tower@1.6.3, nf-wave@1.0.1]\n",
      "Feb-14 17:42:16.875 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-google version: 1.8.3\n",
      "Feb-14 17:42:16.891 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-google@1.8.3' resolved\n",
      "Feb-14 17:42:16.892 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-google@1.8.3'\n",
      "Feb-14 17:42:16.939 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-google@1.8.3\n",
      "Feb-14 17:42:16.939 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-tower version: 1.6.3\n",
      "Feb-14 17:42:16.941 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-tower@1.6.3' resolved\n",
      "Feb-14 17:42:16.941 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-tower@1.6.3'\n",
      "Feb-14 17:42:16.952 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-tower@1.6.3\n",
      "Feb-14 17:42:16.952 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-wave version: 1.0.1\n",
      "Feb-14 17:42:16.955 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-wave@1.0.1' resolved\n",
      "Feb-14 17:42:16.955 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-wave@1.0.1'\n",
      "Feb-14 17:42:16.963 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-wave@1.0.1\n",
      "Feb-14 17:42:16.984 [main] DEBUG n.secret.LocalSecretsProvider - Secrets store: /home/eila/.nextflow/secrets/store.json\n",
      "Feb-14 17:42:16.989 [main] DEBUG nextflow.secret.SecretsLoader - Discovered secrets providers: [nextflow.secret.LocalSecretsProvider@52227eb2] - activable => nextflow.secret.LocalSecretsProvider@52227eb2\n",
      "Feb-14 17:42:17.067 [main] DEBUG nextflow.Session - Session UUID: 0cbd4d64-b857-4807-b564-11778fa2d750\n",
      "Feb-14 17:42:17.067 [main] DEBUG nextflow.Session - Run name: insane_woese\n",
      "Feb-14 17:42:17.067 [main] DEBUG nextflow.Session - Executor pool size: 8\n",
      "Feb-14 17:42:17.350 [main] DEBUG nextflow.file.FilePorter - File porter settings maxRetries=3; maxTransfers=50; pollTimeout=null\n",
      "Feb-14 17:42:17.357 [main] DEBUG nextflow.util.ThreadPoolBuilder - Creating thread pool 'FileTransfer' minSize=10; maxSize=24; workQueue=LinkedBlockingQueue[10000]; allowCoreThreadTimeout=false\n",
      "Feb-14 17:42:17.364 [main] DEBUG nextflow.config.ConfigBuilder - User config file: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Feb-14 17:42:17.365 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Feb-14 17:42:17.366 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `google_batch_profile`\n",
      "Feb-14 17:42:19.173 [main] DEBUG nextflow.config.ConfigBuilder - Available config profiles: [google_batch_profile]\n",
      "Feb-14 17:42:19.211 [main] DEBUG nextflow.cli.CmdRun - \n",
      "  Version: 23.10.1 build 5891\n",
      "  Created: 12-01-2024 22:01 UTC \n",
      "  System: Linux 5.10.0-27-cloud-amd64\n",
      "  Runtime: Groovy 3.0.19 on OpenJDK 64-Bit Server VM 11.0.1-internal+0-adhoc..src\n",
      "  Encoding: UTF-8 (UTF-8)\n",
      "  Process: 203981@eila-vm [10.128.0.63]\n",
      "  CPUs: 8 - Mem: 29.4 GB (6.2 GB) - Swap: 0 (0)\n",
      "Feb-14 17:42:19.281 [main] DEBUG nextflow.file.FileHelper - Can't check if specified path is NFS (1): gs://eila-nextflow-bucket/work/ATAC\n",
      "\n",
      "Feb-14 17:42:19.281 [main] DEBUG nextflow.Session - Work-dir: gs://eila-nextflow-bucket/work/ATAC [null]\n",
      "Feb-14 17:42:19.317 [main] DEBUG nextflow.executor.ExecutorFactory - Extension executors providers=[GoogleLifeSciencesExecutor, GoogleBatchExecutor]\n",
      "Feb-14 17:42:19.335 [main] DEBUG nextflow.Session - Observer factory: DefaultObserverFactory\n",
      "Feb-14 17:42:19.338 [main] DEBUG nextflow.Session - Observer factory: WaveFactory\n",
      "Feb-14 17:42:19.339 [main] DEBUG io.seqera.wave.plugin.WaveFactory - Detected Fusion enabled -- Enabling bundle project resources -- Disabling upload of remote bin directory\n",
      "Feb-14 17:42:19.367 [main] DEBUG i.s.wave.plugin.config.WaveConfig - Wave strategy not specified - using default: [container, dockerfile, conda, spack]\n",
      "Feb-14 17:42:19.383 [main] DEBUG io.seqera.wave.plugin.WaveClient - Wave config: WaveConfig(enabled:true, endpoint:https://wave.seqera.io, containerConfigUrl:[], tokensCacheMaxDuration:30m, condaOpts:CondaOpts(mambaImage=mambaorg/micromamba:1.5.1; basePackages=conda-forge::procps-ng, commands=null), spackOpts:SpackOpts(basePackages=null, commands=null), strategy:[container, dockerfile, conda, spack], bundleProjectResources:true, buildRepository:null, cacheRepository:null, retryOpts:RetryOpts(delay:450ms, maxDelay:1m 30s, maxAttempts:10, jitter:0.25), httpClientOpts:HttpOpts(), freezeMode:null)\n",
      "Feb-14 17:42:19.524 [main] DEBUG nextflow.Session - Observer factory: TowerFactory\n",
      "Feb-14 17:42:19.615 [main] DEBUG nextflow.cache.CacheFactory - Using Nextflow cache factory: nextflow.cache.DefaultCacheFactory\n",
      "Feb-14 17:42:19.634 [main] DEBUG nextflow.util.CustomThreadPool - Creating default thread pool > poolSize: 9; maxThreads: 1000\n",
      "Feb-14 17:42:19.752 [main] DEBUG nextflow.Session - Session start\n",
      "Feb-14 17:42:19.768 [main] DEBUG io.seqera.tower.plugin.TowerClient - Creating Tower observer -- endpoint=https://api.tower.nf; requestInterval=1s; aliveInterval=1m; maxRetries=5; backOffBase=3; backOffDelay=250\n",
      "Feb-14 17:42:20.769 [main] DEBUG nextflow.script.ScriptRunner - > Launching execution\n",
      "Feb-14 17:42:22.283 [main] INFO  nextflow.script.BaseScript - /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/wdl_atac_pipeline/vm_atac_fastqs_files.tsv\n",
      "Feb-14 17:42:22.284 [main] INFO  nextflow.script.BaseScript - null\n",
      "Feb-14 17:42:22.426 [main] INFO  nextflow.script.BaseScript - after genome_fasta_ch\n",
      "Feb-14 17:42:22.489 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:zcat` matches labels `zcat` for process with name run_zcat\n",
      "Feb-14 17:42:22.498 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-14 17:42:22.498 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-14 17:42:22.501 [main] DEBUG nextflow.executor.Executor - [warm up] executor > google-batch (fusion enabled)\n",
      "Feb-14 17:42:22.506 [main] DEBUG n.processor.TaskPollingMonitor - Creating task monitor for executor 'google-batch' > capacity: 1000; pollInterval: 10s; dumpInterval: 5m \n",
      "Feb-14 17:42:22.508 [main] DEBUG n.processor.TaskPollingMonitor - >>> barrier register (monitor: google-batch)\n",
      "Feb-14 17:42:22.515 [main] DEBUG nextflow.cloud.google.GoogleOpts - Google auth via application credentials file: /home/eila/git_igvf/atomic-workflows/nextflow/hai-gcp-genomic-beb9207a40b8.json\n",
      "Feb-14 17:42:22.522 [main] DEBUG n.c.google.batch.GoogleBatchExecutor - [GOOGLE BATCH] Executor config=BatchConfig[googleOpts=GoogleOpts(projectId:hai-gcp-genomic, credsFile:/home/eila/git_igvf/atomic-workflows/nextflow/hai-gcp-genomic-beb9207a40b8.json, location:us-central1, enableRequesterPaysBuckets:false, httpConnectTimeout:1m, httpReadTimeout:1m, credentials:ServiceAccountCredentials{clientId=111364348320962814542, clientEmail=eila-batch@hai-gcp-genomic.iam.gserviceaccount.com, privateKeyId=beb9207a40b8b2085c7f48c3fcf8c2342180d471, transportFactoryClassName=com.google.auth.oauth2.OAuth2Utils$DefaultHttpTransportFactory, tokenServerUri=https://oauth2.googleapis.com/token, scopes=[https://www.googleapis.com/auth/cloud-platform], defaultScopes=[], serviceAccountUser=null, quotaProjectId=null, lifetime=3600, useJwtAccessWithScope=false, defaultRetriesEnabled=true})\n",
      "Feb-14 17:42:22.544 [main] DEBUG n.c.google.batch.client.BatchClient - [GOOGLE BATCH] Creating service client with config credentials\n",
      "Feb-14 17:42:23.268 [main] INFO  nextflow.script.BaseScript - after run_zcat \n",
      "Feb-14 17:42:23.271 [main] INFO  nextflow.script.BaseScript - after load chrompa mapping debug\n",
      "Feb-14 17:42:23.282 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:pool_fragments` matches labels `pool_fragments` for process with name run_add_subpool_to_chromap_output\n",
      "Feb-14 17:42:23.284 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-14 17:42:23.285 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-14 17:42:23.288 [main] INFO  nextflow.script.BaseScript - before call \n",
      "Feb-14 17:42:23.291 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:bgzip_chromap` matches labels `bgzip_chromap` for process with name run_bgzip_on_chromap_fragments_output\n",
      "Feb-14 17:42:23.292 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-14 17:42:23.292 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-14 17:42:23.295 [main] INFO  nextflow.script.BaseScript - after call run_bgzip_on_chromap_fragments_output\n",
      "Feb-14 17:42:23.295 [main] INFO  nextflow.script.BaseScript - before run_process_create_temp_summary_dict\n",
      "Feb-14 17:42:23.300 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:create_temp_summary_dict` matches labels `create_temp_summary_dict` for process with name run_process_create_temp_summary_dict\n",
      "Feb-14 17:42:23.301 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-14 17:42:23.302 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-14 17:42:23.306 [main] INFO  nextflow.script.BaseScript - after run_process_create_temp_summary_dict\n",
      "Feb-14 17:42:23.306 [main] INFO  nextflow.script.BaseScript - before run_filter_aligned_fragments_gz\n",
      "Feb-14 17:42:23.308 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:filter_dictionary_sort` matches labels `filter_dictionary_sort` for process with name run_filter_aligned_fragments_gz\n",
      "Feb-14 17:42:23.310 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-14 17:42:23.310 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-14 17:42:23.315 [main] INFO  nextflow.script.BaseScript - after run_filter_aligned_fragments_gz\n",
      "Feb-14 17:42:23.318 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:gzip` matches labels `gzip` for process with name run_gzip_on_genes_gtf\n",
      "Feb-14 17:42:23.319 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-14 17:42:23.319 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-14 17:42:23.321 [main] INFO  nextflow.script.BaseScript - after run_gzip_on_genes_gtf\n",
      "Feb-14 17:42:23.321 [main] INFO  nextflow.script.BaseScript - before call run_calculate_tss_enrichment_snapatac2\n",
      "Feb-14 17:42:23.324 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:tss_snapatac2` matches labels `tss_snapatac2` for process with name run_calculate_tss_enrichment_snapatac2\n",
      "Feb-14 17:42:23.325 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-14 17:42:23.326 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-14 17:42:23.330 [main] INFO  nextflow.script.BaseScript - after call run_calculate_tss_enrichment_snapatac2\n",
      "Feb-14 17:42:23.331 [main] INFO  nextflow.script.BaseScript - before run_barcode_metadata_stats\n",
      "Feb-14 17:42:23.333 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:generate_barcode_metadata` matches labels `generate_barcode_metadata` for process with name run_barcode_metadata_stats\n",
      "Feb-14 17:42:23.334 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-14 17:42:23.334 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-14 17:42:23.337 [main] INFO  nextflow.script.BaseScript - after run_barcode_metadata_stats\n",
      "Feb-14 17:42:23.337 [main] INFO  nextflow.script.BaseScript - before call run_atac_barcode_rank_plot\n",
      "Feb-14 17:42:23.339 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:atac_barcode_rank_plot` matches labels `atac_barcode_rank_plot` for process with name run_atac_barcode_rank_plot\n",
      "Feb-14 17:42:23.341 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-14 17:42:23.341 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-14 17:42:23.344 [main] INFO  nextflow.script.BaseScript - after run_atac_barcode_rank_plot\n",
      "Feb-14 17:42:23.345 [main] INFO  nextflow.script.BaseScript - before run_generate_insert_size_histogram_data\n",
      "Feb-14 17:42:23.347 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:zcat_insert` matches labels `zcat_insert` for process with name run_generate_insert_size_histogram_data\n",
      "Feb-14 17:42:23.348 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-14 17:42:23.348 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-14 17:42:23.350 [main] INFO  nextflow.script.BaseScript - after run_generate_insert_size_histogram_data\n",
      "Feb-14 17:42:23.350 [main] INFO  nextflow.script.BaseScript - before call run_generate_insert_size_plot\n",
      "Feb-14 17:42:23.353 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:atac_insert_plot` matches labels `atac_insert_plot` for process with name run_generate_insert_size_plot\n",
      "Feb-14 17:42:23.353 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-14 17:42:23.353 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-14 17:42:23.355 [main] INFO  nextflow.script.BaseScript - after call run_generate_insert_size_plot\n",
      "Feb-14 17:42:23.356 [main] INFO  nextflow.script.BaseScript - before call run_merge_chromap_logs\n",
      "Feb-14 17:42:23.359 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:merge_log` matches labels `merge_log` for process with name run_merge_chromap_logs\n",
      "Feb-14 17:42:23.360 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-14 17:42:23.361 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-14 17:42:23.365 [main] INFO  nextflow.script.BaseScript - after call run_merge_chromap_logs\n",
      "Feb-14 17:42:23.366 [main] DEBUG nextflow.Session - Workflow process names [dsl2]: run_bgzip_on_chromap_fragments_output, run_seqspec_print, run_calculate_tss_enrichment_snapatac2, run_barcode_metadata_stats, run_calculate_tss_enrichment_bulk, run_generate_insert_size_plot, run_seqspec_test, run_chromap_map_to_idx, run_atac_merge_barcode_metadata, run_filter_aligned_fragments, run_seqspec_modify_rna, run_atac_barcode_rank_plot, run_gzip_on_genes_gtf, run_tabix_on_sorted_bgzip_chromap, run_sort_on_chromap_fragments_output, run_whitelist_gunzip, run_merge_chromap_logs, run_create_chromap_idx, run_add_subpool_to_chromap_output, run_process_create_temp_summary_dict, run_generate_insert_size_histogram_data, run_seqspec_check, run_zcat, run_seqspec_index_rna_kb, run_seqspec_modify_atac, run_chromap_test, run_filter_aligned_fragments_gz, run_downloadFiles, run_tabix_filtered_fragments_no_singleton\n",
      "Feb-14 17:42:23.707 [main] INFO  io.seqera.tower.plugin.TowerClient - Monitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2v39mmFiJZmXWq\n",
      "Feb-14 17:42:23.708 [main] DEBUG nextflow.Session - Igniting dataflow network (18)\n",
      "Feb-14 17:42:23.718 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_zcat\n",
      "Feb-14 17:42:23.720 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_add_subpool_to_chromap_output\n",
      "Feb-14 17:42:23.721 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_bgzip_on_chromap_fragments_output\n",
      "Feb-14 17:42:23.722 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_process_create_temp_summary_dict\n",
      "Feb-14 17:42:23.723 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_filter_aligned_fragments_gz\n",
      "Feb-14 17:42:23.731 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_gzip_on_genes_gtf\n",
      "Feb-14 17:42:23.731 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_calculate_tss_enrichment_snapatac2\n",
      "Feb-14 17:42:23.732 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_barcode_metadata_stats\n",
      "Feb-14 17:42:23.733 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_atac_barcode_rank_plot\n",
      "Feb-14 17:42:23.733 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_generate_insert_size_histogram_data\n",
      "Feb-14 17:42:23.733 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_generate_insert_size_plot\n",
      "Feb-14 17:42:23.733 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_merge_chromap_logs\n",
      "Feb-14 17:42:23.745 [main] DEBUG nextflow.script.ScriptRunner - Parsed script files:\n",
      "  Script_7b6cdd1ba6b82474: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_atac_metadata_annoation_and_plot.nf\n",
      "  Script_522c85b74472fe49: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_tss.nf\n",
      "  Script_2c943e88731b0547: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_atac_generate_insert_size_plot.nf\n",
      "  Script_5d15b86ab801da8d: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nf_wf_fg_singlecell_atac.nf\n",
      "  Script_9d4639c8cd883c70: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_download_url_files.nf\n",
      "  Script_bc539eb5853c51b8: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_gunzip.nf\n",
      "  Script_ac710cb21b72d60a: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_bgzip.nf\n",
      "  Script_52c0773071085951: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_sort_fragments_file.nf\n",
      "  Script_c843d085a2409aa5: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_subpool_summary_dict.nf\n",
      "  Script_13475a81f0466813: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_filter_by_dictionary_barcode.nf\n",
      "  Script_e112646bcdb899f0: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_seqspec_utils.nf\n",
      "  Script_586db11f47634c7d: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_tabix.nf\n",
      "  Script_8081eedda99eaa43: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_merge_logs.nf\n",
      "  Script_48127624ae4a6640: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_zcat.nf\n",
      "  Script_9abd93a618947195: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_chromap_utils.nf\n",
      "  Script_7f43a272cfe3ae1b: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_generate_merged_barcode_metadata.nf\n",
      "  Script_34d6d3540f17eeb9: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_gzip.nf\n",
      "Feb-14 17:42:23.745 [main] DEBUG nextflow.script.ScriptRunner - > Awaiting termination \n",
      "Feb-14 17:42:23.745 [main] DEBUG nextflow.Session - Session await\n",
      "Feb-14 17:42:25.240 [FileTransfer-2] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/wdl_atac_pipeline/genes.gtf at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/41/45671d165921b1e3a885b4ea55909e/genes.gtf\n",
      "Feb-14 17:42:25.259 [Actor Thread 7] DEBUG i.s.wave.plugin.config.WaveConfig - Wave strategy not specified - using default: [container, dockerfile, conda, spack]\n",
      "Feb-14 17:42:25.259 [Actor Thread 7] DEBUG io.seqera.wave.plugin.WaveClient - Wave config: WaveConfig(enabled:true, endpoint:https://wave.seqera.io, containerConfigUrl:[], tokensCacheMaxDuration:30m, condaOpts:CondaOpts(mambaImage=mambaorg/micromamba:1.5.1; basePackages=conda-forge::procps-ng, commands=null), spackOpts:SpackOpts(basePackages=null, commands=null), strategy:[container, dockerfile, conda, spack], bundleProjectResources:true, buildRepository:null, cacheRepository:null, retryOpts:RetryOpts(delay:450ms, maxDelay:1m 30s, maxAttempts:10, jitter:0.25), httpClientOpts:HttpOpts(), freezeMode:null)\n",
      "Feb-14 17:42:25.274 [Actor Thread 7] DEBUG io.seqera.wave.plugin.WaveClient - Wave assets bundle bin resources: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/bin\n",
      "Feb-14 17:42:25.274 [FileTransfer-1] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/wdl_atac_pipeline/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/f2/e60f03e75862ee5ca99088aa1ad7b5/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\n",
      "Feb-14 17:42:25.355 [Actor Thread 7] DEBUG io.seqera.wave.plugin.WaveClient - Wave request container config: https://fusionfs.seqera.io/releases/v2.2-amd64.json\n",
      "Feb-14 17:42:25.663 [Actor Thread 7] DEBUG io.seqera.wave.plugin.WaveClient - Wave container config response: [200] {\n",
      "  \"entrypoint\": [ \"/usr/bin/fusion\" ],\n",
      "  \"layers\": [\n",
      "    {\n",
      "      \"location\": \"https://fusionfs.seqera.io/releases/pkg/2/2/12/fusion-amd64.tar.gz\",\n",
      "      \"gzipDigest\": \"sha256:6ccfd8f431db2874e82e27d61b6d31b95d7dddce7e810c5f8c95438a0d03c066\",\n",
      "      \"gzipSize\": 30262575,\n",
      "      \"tarDigest\": \"sha256:60391bebe0cc5d48b0875854f0c93c0d0adc25b5007fdb1b2efc2a431cfeef6e\",\n",
      "      \"skipHashing\": true\n",
      "    }\n",
      "  ]\n",
      "}\n",
      "\n",
      "Feb-14 17:42:25.780 [Actor Thread 7] DEBUG io.seqera.wave.plugin.WaveClient - Wave request: https://wave.seqera.io/container-token; attempt=1 - request: SubmitContainerTokenRequest(towerAccessToken:eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz, towerRefreshToken:null, towerWorkspaceId:247507587970818, towerEndpoint:https://api.tower.nf, workflowId:null, containerImage:eilalan/utils_env:latest, containerFile:null, containerConfig:ContainerConfig(entrypoint:[/usr/bin/fusion], cmd:null, env:null, workingDir:null, layers:[ContainerLayer[location=data:H4sIAAAAAAAA/+19a3fb...; tarDigest=sha256:80d921201f4aef927ace1748698dbad799c5e85e5434fd2ae18f8fd344ef46ff; gzipDigest=sha256:43aba159b562fdbf35ebef096fde61f37aee30bdbeac72b28be6ca02f6fd4c63; gzipSize=20679], ContainerLayer[location=https://fusionfs.seqera.io/releases/pkg/2/2/12/fusion-amd64.tar.gz; tarDigest=sha256:60391bebe0cc5d48b0875854f0c93c0d0adc25b5007fdb1b2efc2a431cfeef6e; gzipDigest=sha256:6ccfd8f431db2874e82e27d61b6d31b95d7dddce7e810c5f8c95438a0d03c066; gzipSize=30262575]]), condaFile:null, spackFile:null, containerPlatform:linux/amd64, buildRepository:null, cacheRepository:null, timestamp:2024-02-14T17:42:25.778771Z, fingerprint:aaaebe51614877061782b4a048c44f7c, freeze:false, format:null, dryRun:false)\n",
      "Feb-14 17:42:26.544 [Actor Thread 7] DEBUG io.seqera.wave.plugin.WaveClient - Wave response: statusCode=200; body={\"containerToken\":\"eefe1b8e0e21\",\"targetImage\":\"wave.seqera.io/wt/eefe1b8e0e21/eilalan/utils_env:latest\",\"expiration\":\"2024-02-16T05:42:26.489021903Z\",\"containerImage\":\"docker.io/eilalan/utils_env:latest\"}\n",
      "Feb-14 17:42:27.103 [Actor Thread 3] INFO  nextflow.processor.TaskProcessor - [33/c79639] Cached process > run_zcat\n",
      "Feb-14 17:42:27.104 [Actor Thread 7] INFO  nextflow.processor.TaskProcessor - [aa/adbebc] Cached process > run_gzip_on_genes_gtf (1)\n",
      "Feb-14 17:42:29.226 [FileTransfer-4] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/82/1c02896062d7ad9bee48b153038b02/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\n",
      "Feb-14 17:42:29.237 [FileTransfer-3] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/4a/8d5aa2caf4ba953973eb6ceb0209c1/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\n",
      "Feb-14 17:42:29.239 [FileTransfer-6] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/9a/495e305984d9974332346f8cc35a95/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\n",
      "Feb-14 17:42:29.241 [FileTransfer-8] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/de/e2b4899f50b4c0a726cbaade1c86a9/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\n",
      "Feb-14 17:42:29.241 [FileTransfer-5] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/3b/378e6650d13f6cf7227c8199100fca/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\n",
      "Feb-14 17:42:29.243 [FileTransfer-7] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/67/0b839068fabf5a437638652ae82543/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\n",
      "Feb-14 17:42:29.246 [FileTransfer-2] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/test_atac_pipeline/na1.na at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/ab/3e66243b0d5cb8425a9dcfe650eb5c/na1.na\n",
      "Feb-14 17:42:29.248 [FileTransfer-9] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/test_atac_pipeline/na.na at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/89/c1bcbc8e5f5646ebd09ee0ce3faefe/na.na\n",
      "Feb-14 17:42:29.250 [FileTransfer-10] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/test_atac_pipeline/sai_192_whitelist.txt at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/32/f784fb3f3859ff8de353857c4c015f/sai_192_whitelist.txt\n",
      "Feb-14 17:42:29.256 [FileTransfer-1] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/wdl_atac_pipeline/BMMC-single-donor.atac.align.barcode.summary.csv at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/e5/4f78cfd835463ff7b54feb664984f5/BMMC-single-donor.atac.align.barcode.summary.csv\n",
      "Feb-14 17:42:29.280 [FileTransfer-4] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/wdl_atac_pipeline/ATAC_BMMC-single-donor.atac.filter.fragments.tsv at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/ac/117d355dff34bab351800c58734a87/ATAC_BMMC-single-donor.atac.filter.fragments.tsv\n",
      "Feb-14 17:42:30.278 [Actor Thread 10] INFO  nextflow.processor.TaskProcessor - [9f/40830d] Cached process > run_add_subpool_to_chromap_output (1)\n",
      "Feb-14 17:42:30.562 [FileTransfer-3] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/wdl_atac_pipeline/BMMC-single-donor.atac.align.k4.hg38.log.txt at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/a0/35e0e425bfac37ba2323d9e6e33d19/BMMC-single-donor.atac.align.k4.hg38.log.txt\n",
      "Feb-14 17:42:30.721 [Actor Thread 8] INFO  nextflow.processor.TaskProcessor - [3e/470034] Cached process > run_bgzip_on_chromap_fragments_output (1)\n",
      "Feb-14 17:42:30.794 [Actor Thread 9] INFO  nextflow.processor.TaskProcessor - [d3/c0aca2] Cached process > run_process_create_temp_summary_dict (1)\n",
      "Feb-14 17:42:31.143 [Actor Thread 3] INFO  nextflow.processor.TaskProcessor - [2d/d3c262] Cached process > run_merge_chromap_logs (1)\n",
      "Feb-14 17:42:31.209 [Actor Thread 5] INFO  nextflow.processor.TaskProcessor - [47/cceb9e] Cached process > run_filter_aligned_fragments_gz (1)\n",
      "Feb-14 17:42:31.237 [Actor Thread 8] DEBUG io.seqera.wave.plugin.WaveClient - Wave request: https://wave.seqera.io/container-token; attempt=1 - request: SubmitContainerTokenRequest(towerAccessToken:eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz, towerRefreshToken:null, towerWorkspaceId:247507587970818, towerEndpoint:https://api.tower.nf, workflowId:null, containerImage:eilalan/tss_env:latest, containerFile:null, containerConfig:ContainerConfig(entrypoint:[/usr/bin/fusion], cmd:null, env:null, workingDir:null, layers:[ContainerLayer[location=data:H4sIAAAAAAAA/+19a3fb...; tarDigest=sha256:80d921201f4aef927ace1748698dbad799c5e85e5434fd2ae18f8fd344ef46ff; gzipDigest=sha256:43aba159b562fdbf35ebef096fde61f37aee30bdbeac72b28be6ca02f6fd4c63; gzipSize=20679], ContainerLayer[location=https://fusionfs.seqera.io/releases/pkg/2/2/12/fusion-amd64.tar.gz; tarDigest=sha256:60391bebe0cc5d48b0875854f0c93c0d0adc25b5007fdb1b2efc2a431cfeef6e; gzipDigest=sha256:6ccfd8f431db2874e82e27d61b6d31b95d7dddce7e810c5f8c95438a0d03c066; gzipSize=30262575]]), condaFile:null, spackFile:null, containerPlatform:linux/amd64, buildRepository:null, cacheRepository:null, timestamp:2024-02-14T17:42:31.236764Z, fingerprint:e23552a7d275b52da524c56c5a458517, freeze:false, format:null, dryRun:false)\n",
      "Feb-14 17:42:31.642 [Actor Thread 8] DEBUG io.seqera.wave.plugin.WaveClient - Wave response: statusCode=200; body={\"containerToken\":\"79c7f1b56a0b\",\"targetImage\":\"wave.seqera.io/wt/79c7f1b56a0b/eilalan/tss_env:latest\",\"expiration\":\"2024-02-16T05:42:31.589662103Z\",\"containerImage\":\"docker.io/eilalan/tss_env:latest\"}\n",
      "Feb-14 17:42:31.946 [Actor Thread 11] INFO  nextflow.processor.TaskProcessor - [d8/bdb929] Cached process > run_generate_insert_size_histogram_data (1)\n",
      "Feb-14 17:42:31.973 [Actor Thread 10] DEBUG io.seqera.wave.plugin.WaveClient - Wave request: https://wave.seqera.io/container-token; attempt=1 - request: SubmitContainerTokenRequest(towerAccessToken:eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz, towerRefreshToken:null, towerWorkspaceId:247507587970818, towerEndpoint:https://api.tower.nf, workflowId:null, containerImage:eilalan/rank_plot_env:latest, containerFile:null, containerConfig:ContainerConfig(entrypoint:[/usr/bin/fusion], cmd:null, env:null, workingDir:null, layers:[ContainerLayer[location=data:H4sIAAAAAAAA/+19a3fb...; tarDigest=sha256:80d921201f4aef927ace1748698dbad799c5e85e5434fd2ae18f8fd344ef46ff; gzipDigest=sha256:43aba159b562fdbf35ebef096fde61f37aee30bdbeac72b28be6ca02f6fd4c63; gzipSize=20679], ContainerLayer[location=https://fusionfs.seqera.io/releases/pkg/2/2/12/fusion-amd64.tar.gz; tarDigest=sha256:60391bebe0cc5d48b0875854f0c93c0d0adc25b5007fdb1b2efc2a431cfeef6e; gzipDigest=sha256:6ccfd8f431db2874e82e27d61b6d31b95d7dddce7e810c5f8c95438a0d03c066; gzipSize=30262575]]), condaFile:null, spackFile:null, containerPlatform:linux/amd64, buildRepository:null, cacheRepository:null, timestamp:2024-02-14T17:42:31.972108Z, fingerprint:11a41d650161d353b181c0933e7ff06b, freeze:false, format:null, dryRun:false)\n",
      "Feb-14 17:42:32.175 [Actor Thread 10] DEBUG io.seqera.wave.plugin.WaveClient - Wave response: statusCode=200; body={\"containerToken\":\"4cb8b4aa9045\",\"targetImage\":\"wave.seqera.io/wt/4cb8b4aa9045/eilalan/rank_plot_env:latest\",\"expiration\":\"2024-02-16T05:42:32.123192055Z\",\"containerImage\":\"docker.io/eilalan/rank_plot_env:latest\"}\n",
      "Feb-14 17:42:32.574 [Actor Thread 8] INFO  nextflow.processor.TaskProcessor - [7b/53117e] Cached process > run_calculate_tss_enrichment_snapatac2 (1)\n",
      "Feb-14 17:42:33.040 [Actor Thread 6] INFO  nextflow.processor.TaskProcessor - [06/5b09cc] Cached process > run_barcode_metadata_stats (1)\n",
      "Feb-14 17:42:33.066 [Actor Thread 5] DEBUG io.seqera.wave.plugin.WaveClient - Wave request: https://wave.seqera.io/container-token; attempt=1 - request: SubmitContainerTokenRequest(towerAccessToken:eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz, towerRefreshToken:null, towerWorkspaceId:247507587970818, towerEndpoint:https://api.tower.nf, workflowId:null, containerImage:eilalan/rna_atac_plot_qc_env:latest, containerFile:null, containerConfig:ContainerConfig(entrypoint:[/usr/bin/fusion], cmd:null, env:null, workingDir:null, layers:[ContainerLayer[location=data:H4sIAAAAAAAA/+19a3fb...; tarDigest=sha256:80d921201f4aef927ace1748698dbad799c5e85e5434fd2ae18f8fd344ef46ff; gzipDigest=sha256:43aba159b562fdbf35ebef096fde61f37aee30bdbeac72b28be6ca02f6fd4c63; gzipSize=20679], ContainerLayer[location=https://fusionfs.seqera.io/releases/pkg/2/2/12/fusion-amd64.tar.gz; tarDigest=sha256:60391bebe0cc5d48b0875854f0c93c0d0adc25b5007fdb1b2efc2a431cfeef6e; gzipDigest=sha256:6ccfd8f431db2874e82e27d61b6d31b95d7dddce7e810c5f8c95438a0d03c066; gzipSize=30262575]]), condaFile:null, spackFile:null, containerPlatform:linux/amd64, buildRepository:null, cacheRepository:null, timestamp:2024-02-14T17:42:33.065504Z, fingerprint:9b378069a4fefd01b26d9c740397d48d, freeze:false, format:null, dryRun:false)\n",
      "Feb-14 17:42:33.259 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_generate_insert_size_plot (1)` - adding local volume as fusion scratch: DiskResource(request:375 GB, type:local-ssd)\n",
      "Feb-14 17:42:33.267 [Actor Thread 5] DEBUG io.seqera.wave.plugin.WaveClient - Wave response: statusCode=200; body={\"containerToken\":\"e8b8f218417d\",\"targetImage\":\"wave.seqera.io/wt/e8b8f218417d/eilalan/rna_atac_plot_qc_env:latest\",\"expiration\":\"2024-02-16T05:42:33.215594382Z\",\"containerImage\":\"docker.io/eilalan/rna_atac_plot_qc_env:latest\"}\n",
      "Feb-14 17:42:34.006 [Actor Thread 5] INFO  nextflow.processor.TaskProcessor - [bd/22c6af] Cached process > run_atac_barcode_rank_plot (1)\n",
      "Feb-14 17:42:34.677 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_generate_insert_size_plot (1)` submitted > job=nf-acf5bb73-1707932552531; uid=nf-acf5bb73-170793-799cc901-146b-41d90; work-dir=gs://eila-nextflow-bucket/work/ATAC/ac/f5bb73f8a977b08be7a306cc7cd1ff\n",
      "Feb-14 17:42:34.677 [Task submitter] INFO  nextflow.Session - [ac/f5bb73] Submitted process > run_generate_insert_size_plot (1)\n",
      "Feb-14 17:45:12.562 [Task monitor] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_generate_insert_size_plot (1)` - terminated job=nf-acf5bb73-1707932552531; state=SUCCEEDED\n",
      "Feb-14 17:45:12.706 [Task monitor] DEBUG n.processor.TaskPollingMonitor - Task completed > TaskHandler[id: 10; name: run_generate_insert_size_plot (1); status: COMPLETED; exit: 0; error: -; workDir: gs://eila-nextflow-bucket/work/ATAC/ac/f5bb73f8a977b08be7a306cc7cd1ff]\n",
      "Feb-14 17:45:12.958 [main] DEBUG nextflow.Session - Session await > all processes finished\n",
      "Feb-14 17:45:13.089 [Task monitor] DEBUG n.processor.TaskPollingMonitor - <<< barrier arrives (monitor: google-batch) - terminating tasks monitor poll loop\n",
      "Feb-14 17:45:13.089 [main] DEBUG nextflow.Session - Session await > all barriers passed\n",
      "Feb-14 17:45:13.096 [main] DEBUG n.trace.WorkflowStatsObserver - Workflow completed > WorkflowStats[succeededCount=1; failedCount=0; ignoredCount=0; cachedCount=11; pendingCount=0; submittedCount=0; runningCount=0; retriesCount=0; abortedCount=0; succeedDuration=9.7s; failedDuration=0ms; cachedDuration=9m 41s;loadCpus=0; loadMemory=0; peakRunning=1; peakCpus=1; peakMemory=0; ]\n",
      "Feb-14 17:45:14.659 [main] DEBUG nextflow.cache.CacheDB - Closing CacheDB done\n",
      "Feb-14 17:45:14.660 [main] INFO  org.pf4j.AbstractPluginManager - Stop plugin 'nf-wave@1.0.1'\n",
      "Feb-14 17:45:14.660 [main] DEBUG nextflow.plugin.BasePlugin - Plugin stopped nf-wave\n",
      "Feb-14 17:45:14.660 [main] INFO  org.pf4j.AbstractPluginManager - Stop plugin 'nf-tower@1.6.3'\n",
      "Feb-14 17:45:14.660 [main] DEBUG nextflow.plugin.BasePlugin - Plugin stopped nf-tower\n",
      "Feb-14 17:45:14.660 [main] INFO  org.pf4j.AbstractPluginManager - Stop plugin 'nf-google@1.8.3'\n",
      "Feb-14 17:45:14.660 [main] DEBUG nextflow.plugin.BasePlugin - Plugin stopped nf-google\n",
      "Feb-14 17:45:14.684 [main] DEBUG nextflow.util.ThreadPoolManager - Thread pool 'FileTransfer' shutdown completed (hard=false)\n",
      "Feb-14 17:45:14.685 [main] DEBUG nextflow.script.ScriptRunner - > Execution complete -- Goodbye\n"
     ]
    }
   ],
   "source": [
    "!cat .nextflow.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
