{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2463,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# start docker deamon in terminal\n",
    "# !sudo docker login\n",
    "# OR - becuase the jupyther with allow root\n",
    "# !docker login"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "TODO:\n",
    "\n",
    "when running with cloud. \n",
    "if you dont need to keep the execution logs, change the cloud settings to keep it for a short while\n",
    "Check that the cache if you need to set the cache to false (especially in plot process)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2464,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2465,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/hai-gcp-genomic-beb9207a40b8.json\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "json_path = os.path.join(os.path.dirname(os.getcwd()),\"hai-gcp-genomic-beb9207a40b8.json\")\n",
    "!ls $json_path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2466,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Activated service account credentials for: [eila-batch@hai-gcp-genomic.iam.gserviceaccount.com]\n"
     ]
    }
   ],
   "source": [
    "# # in the command line. same env\n",
    "import os\n",
    "json_path = os.path.join(os.path.dirname(os.getcwd()),\"hai-gcp-genomic-beb9207a40b8.json\")\n",
    "os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = json_path\n",
    "!gcloud auth activate-service-account eila-batch@hai-gcp-genomic.iam.gserviceaccount.com --key-file=$GOOGLE_APPLICATION_CREDENTIALS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2467,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                  Credentialed Accounts\n",
      "ACTIVE  ACCOUNT\n",
      "*       eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\n",
      "        eila@stanford.edu\n",
      "\n",
      "To set the active account, run:\n",
      "    $ gcloud config set account `ACCOUNT`\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'hai-gcp-genomic'"
      ]
     },
     "execution_count": 2467,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "!gcloud auth list\n",
    "project_name = !gcloud config get-value project\n",
    "project_name = project_name[0]\n",
    "project_name\n",
    "# gcloud config set account\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2468,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "os.environ[\"TOWER_ACCESS_TOKEN\"] = 'eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz'\n",
    "os.environ[\"TOWER_WORKSPACE_ID\"]='247507587970818'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2469,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz\n",
      "247507587970818\n"
     ]
    }
   ],
   "source": [
    "!echo $TOWER_ACCESS_TOKEN\n",
    "!echo $TOWER_WORKSPACE_ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2470,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2471,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/bin'"
      ]
     },
     "execution_count": 2471,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# !chmod +x ./bin/scrna_qc_metrics.py\n",
    "import os\n",
    "bin_fld = os.path.join(os.getcwd(),\"bin\")\n",
    "bin_fld"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2472,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "!chmod +x {os.path.join(bin_fld, \"scrna_plot_atac_qc_metrics.R\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"run_tabix.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"plot_insert_size_hist.py\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"atac_generate_merged_barcode_metadata.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"create_sample_filtered_fragment_file.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"compute_tss_enrichment_bulk.py\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"barcode_rank_functions.R\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"plot_insert_size_hist.py\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"compute_tss_enrichment_snapatac2.py\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"add_subpool_to_fragments_table.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"add_subpool_to_summary_file.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"filter_script.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"merge_chromap_logs.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"generate_insert_size_histogram_data.sh\")}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2473,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# # TODO: For a new user: run these commands with your service account for the execution. I am using eila-batch@...\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/iam.serviceAccountUser\"\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/batch.jobsEditor\"\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/logging.viewer\"\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2474,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !nextflow -self-update"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2475,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  Version: 23.10.1 build 5891\n",
      "  Created: 12-01-2024 22:01 UTC (17:01 EDT)\n",
      "  System: Mac OS X 14.3\n",
      "  Runtime: Groovy 3.0.19 on OpenJDK 64-Bit Server VM 17.0.8+7-LTS\n",
      "  Encoding: UTF-8 (UTF-8)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!nextflow info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2476,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "\n",
    "# Get the current working directory\n",
    "current_directory = os.getcwd()\n",
    "\n",
    "# Navigate two folders above\n",
    "two_folders_above = os.path.abspath(os.path.join(current_directory, '..', '..'))\n",
    "\n",
    "nf_data_base_path = os.path.join(two_folders_above,'nf_data')\n",
    "# Print the result\n",
    "print(nf_data_base_path)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2477,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data'"
      ]
     },
     "execution_count": 2477,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nf_data_base_path\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2478,
   "metadata": {},
   "outputs": [],
   "source": [
    "nf_data_project=\"test_atac_pipeline\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2479,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/local_atac_fastqs_files.tsv'"
      ]
     },
     "execution_count": 2479,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.path.exists(os.path.join(nf_data_base_path,nf_data_project,\"local_atac_fastqs_files.tsv\"))\n",
    "os.path.join(nf_data_base_path,nf_data_project,\"local_atac_fastqs_files.tsv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2480,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "\n",
    "# Write the configuration file\n",
    "with open(\"nextflow_atac.config\", \"w\") as config_file:\n",
    "    config_file.write(f\"\"\"\n",
    "manifest {{\n",
    "  description = 'A nexflow pipeline for fg single cell atac' \n",
    "  mainScript = 'nf_wf_fg_singlecell_atac.nf'\n",
    "}}\n",
    "\n",
    "profiles {{\n",
    "  google_batch_profile {{\n",
    "    workDir = 'gs://eila-nextflow-bucket/work/ATAC'\n",
    "    includeConfig 'conf/docker-google-batch.config'\n",
    "    docker.enabled = true\n",
    "    wave.enabled = true\n",
    "    fusion.enabled = true\n",
    "    process.scratch = false\n",
    "    executor = 'google-batch'\n",
    "    docker.envWhitelist=['ENV_SYNAPSE_TOKEN','TOWER_ACCESS_TOKEN']\n",
    "    google.project = '{project_name}'\n",
    "    google.location = 'us-central1'\n",
    "    google.batch.spot = true\n",
    "    google.batch.serviceAccountEmail='eila-batch@hai-gcp-genomic.iam.gserviceaccount.com'\n",
    "  }}\n",
    "}}\n",
    "\n",
    "// Define ENV_SYNAPSE_TOKEN as a parameter\n",
    "params.FASTQS_SPEC_CH = '{nf_data_base_path}/{nf_data_project}/local_atac_fastqs_files.tsv'\n",
    "// CHROMAP\n",
    "// 6a\n",
    "params.CHROMAP_IDX='gs://eila-nextflow-bucket/test_atac_pipeline/ref.index'\n",
    "// 6b:\n",
    "// FASTA file\n",
    "params.CHROMAP_READ_LENGTH = 2000\n",
    "params.CHROMAP_BC_ERROR_THRESHOLD = 2\n",
    "params.CHROMAP_BC_PROBABILITY_THRESHOLD = 0.9\n",
    "params.CHROMAP_READ_FORMAT = 'bc:0:-1,r1:0:-1,r2:0:-1'\n",
    "params.CHROMAP_DROP_REPETITIVE_READS = 4\n",
    "params.CHROMAP_QUALITY_THRESHOLD = 0\n",
    "params.CHROMAP_MERGE_LOGS_SCRIPT = \"merge_chromap_logs.sh\"\n",
    "params.CHROMAP_FRAGMENTS_HISTOGRAM_DATA_SCRIPT=\"generate_insert_size_histogram_data.sh\"\n",
    "params.CHROMAP_GENOME_REFERENCE_FASTA='{nf_data_base_path}/{nf_data_project}/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz'\n",
    "\n",
    "// CHROMAP DEBUG: load output files \n",
    "params.chromap_filter_fragments_tsv ='{nf_data_base_path}/{nf_data_project}/chromap_out_debug/ATAC_BMMC-single-donor.atac.filter.fragments.tsv'\n",
    "params.barcode_summary_csv ='{nf_data_base_path}/{nf_data_project}/chromap_out_debug/BMMC-single-donor.atac.align.barcode.summary.csv'\n",
    "params.chromap_alignment_log_out = '{nf_data_base_path}/{nf_data_project}/chromap_out_debug/BMMC-single-donor.atac.align.k4.hg38.log.txt'\n",
    "\n",
    "\n",
    "\n",
    "params.TSV_ADD_SUBPOOL_SCRIPT = 'add_subpool_to_fragments_table.sh'\n",
    "params.CSV_ADD_SUBPOOL_SCRIPT = 'add_subpool_to_summary_file.sh'\n",
    "params.FILTER_FRAGMENTS_SCRIPT = 'filter_script.sh'\n",
    "params.ATAC_FRAGMENTS_MIN_FRAG_CUTOFF=10\n",
    "\n",
    "// Other steps\n",
    "\n",
    "params.GENOME_GZ_GTF='http://ftp.ensembl.org/pub/release-109/gtf/homo_sapiens/Homo_sapiens.GRCh38.109.gtf.gz'\n",
    "params.ENV_SYNAPSE_TOKEN = \"nextflow secrets get SYNAPSE\".execute().text.trim()\n",
    "\n",
    "\n",
    "params.MAPQ_THRESHOLD = false\n",
    "params.BARCODE_TRANSLATE = 'path'\n",
    "params.ATAC_FRAGMENTS_CUTOFF = 4\n",
    "params.RUN_TABIX_SCRIPT='run_tabix.sh'\n",
    "params.ATAC_TSS_BASES_FLANK = 2000\n",
    "params.ATAC_TSS_COL_WITH_STRANDS_INFO = 4\n",
    "params.ATAC_TSS_SMOOTHING_WINDOW_SIZE = 20\n",
    "params.ATAC_TSS_REGION_BED_FILE='{nf_data_base_path}/{nf_data_project}/hg38.synapse_transcripts.TSS.bed'\n",
    "params.ATAC_TSS_BULK_CALCULATION_SCRIPT='compute_tss_enrichment_bulk.py'\n",
    "params.ATAC_TSS_SNAPATAC2_CALCULATION_SCRIPT='compute_tss_enrichment_snapatac2.py'\n",
    "params.ATAC_TSS_SNAPATAC2_GENES_GTF='{nf_data_base_path}/{nf_data_project}/genes.gtf'\n",
    "params.ATAC_TSS_SNAPATAC2_MIN_FRAG_CUTOFF = 10\n",
    "\n",
    "params.ATAC_INSERT_SIZE_PLOT_SCRIPT='plot_insert_size_hist.py'\n",
    "\n",
    "// params.SC_ATAC_QC_PLOT_SCRIPT='scrna_plot_atac_qc_metrics.R'\n",
    "// params.SC_ATAC_QC_PLOT_HELPER_SCRIPT='barcode_rank_functions.R'\n",
    "// debug file\n",
    "// params.SC_ATAC_QC_BARCODE_METADATA_FILE='{nf_data_base_path}/{nf_data_project}/barcode_metadata.txt'\n",
    "params.SC_ATAC_QC_FRAGMENT_CUTOFF=10\n",
    "params.SC_ATAC_QC_BARCODE_OUTPUT_FILE='sc_atac_qc_barcode_output_file.png'\n",
    "    \n",
    "// params.SC_ATAC_BARCODE_METADATA_SCRIPT='atac_generate_merged_barcode_metadata.sh'\n",
    "// DEBUG: get the data from a file\n",
    "// params.SC_ATAC_BARCODE_METADATA_FILTERED_BARCODE_STATS='{nf_data_base_path}nf_wf_fg_singlecell/filtered_barcode_stats.csv'\n",
    "// params.SC_ATAC_BARCODE_METADATA_TSS_ENRICHMENT_BARCODE_STATS='{nf_data_base_path}nf_wf_fg_singlecell/tss_enrichment_barcode_stats.csv'\n",
    "\n",
    "// params.SC_ATAC_GENERATE_BARCODE_RANK_PLOT_PKR='PKR123'\n",
    "\n",
    "// params.SC_JOINT_CELL_BARCODE_PLOT_SCRTIP_PY='joint_cell_plotting.py'\n",
    "// params.SC_JOINT_CELL_BARCODE_PLOT_SCRTIP_R='joint_cell_plotting_density.R'\n",
    "\"\"\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2481,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !rm -r /Users/eilaarich-landkof-stanford/Documents/Code/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/work\n",
    "# !rm -r /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/work\n",
    "# !gsutil rm -r gs://eila-nextflow-bucket/work/ATAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2482,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !rm .nextflow*\n",
    "\n",
    "# !ls /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/local_atac_fastqs_files.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2483,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "N E X T F L O W  ~  version 23.10.1\n",
      "Launching `nf_wf_fg_singlecell_atac.nf` [sharp_leakey] DSL2 - revision: a05d84b688\n",
      "/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/local_atac_fastqs_files.tsv\n",
      "eyJ0eXAiOiJKV1QiLCJraWQiOiJXN05OOldMSlQ6SjVSSzpMN1RMOlQ3TDc6M1ZYNjpKRU9VOjY0NFI6VTNJWDo1S1oyOjdaQ0s6RlBUSCIsImFsZyI6IlJTMjU2In0.eyJhY2Nlc3MiOnsic2NvcGUiOlsidmlldyIsImRvd25sb2FkIl0sIm9pZGNfY2xhaW1zIjp7fX0sInRva2VuX3R5cGUiOiJQRVJTT05BTF9BQ0NFU1NfVE9LRU4iLCJpc3MiOiJodHRwczovL3JlcG8tcHJvZC5wcm9kLnNhZ2ViYXNlLm9yZy9hdXRoL3YxIiwiYXVkIjoiMCIsIm5iZiI6MTY5NzgxODkxNiwiaWF0IjoxNjk3ODE4OTE2LCJqdGkiOiI0MTIzIiwic3ViIjoiMzQ3NDU3MSJ9.n0cugYMAVzYZ8mMMc7CjcqqEV7FB-K-oKUDT45SvM6KmbCpb_v3lZ2gg3xXi1tubWH0__wpnbJ7cYrGEtOyt4vXAPkMiEtZFGqS0MFKp_h-mHHH4pnACSQQYEy1zdyqFn9WRE7ipjWg4yYSbq84N7J3gzi8VKmiOAmuIGKaJaS0snkUjtJEVQH-0vZqWs1nKgVRWIZv31Bo8vuDzWpBrXD-zRT_uQW9l_I9fU50o1AqyFhNe_gbty_Z6imb8VT33wMXiDgJh15lG_M70PXU15v3egDXLgw1dGj32kj1nKanfqBVqbMSzgzPAG037p3YH_Q556BLceFBBb5Jaej5JVA\n",
      "after genome_fasta_ch\n",
      "[-        ] process > run_zcat                       -\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip           -\u001b[K\n",
      "[-        ] process > run_add_subpool_prefix_to_f... -\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_gzip_on_genes_gtf          -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "\u001b[39A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "[-        ] process > run_zcat                       -\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip           -\u001b[K\n",
      "[-        ] process > run_add_subpool_prefix_to_f... -\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_gzip_on_genes_gtf          -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "\u001b[40A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "[-        ] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip           -\u001b[K\n",
      "[-        ] process > run_add_subpool_prefix_to_f... -\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_gzip_on_genes_gtf          [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "\u001b[40A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (1)\u001b[K\n",
      "[-        ] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip           [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_add_subpool_prefix_to_f... -\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "\u001b[41A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (1)\u001b[K\n",
      "[-        ] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip           [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_add_subpool_prefix_to_f... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "\u001b[41A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (2)\u001b[K\n",
      "[21/b2f514] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip           [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_add_subpool_prefix_to_f... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "\u001b[41A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (3)\u001b[K\n",
      "[21/b2f514] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[bf/49817f] process > run_whitelist_gunzip (1)       [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_add_subpool_prefix_to_f... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "\u001b[41A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (4)\u001b[K\n",
      "[21/b2f514] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[bf/49817f] process > run_whitelist_gunzip (1)       [  0%] 0 of 1\u001b[K\n",
      "[29/b7b785] process > run_add_subpool_prefix_to_f... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "\u001b[41A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (4)\u001b[K\n",
      "[21/b2f514] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[bf/49817f] process > run_whitelist_gunzip (1)       [  0%] 0 of 1\u001b[K\n",
      "[29/b7b785] process > run_add_subpool_prefix_to_f... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[41A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (4)\u001b[K\n",
      "[21/b2f514] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[bf/49817f] process > run_whitelist_gunzip (1)       [100%] 1 of 1 ✔\u001b[K\n",
      "[29/b7b785] process > run_add_subpool_prefix_to_f... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "Start run_whitelist_gunzip\u001b[K\n",
      "Gunzip file is: sai_192_whitelist.txt\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 172805\u001b[K\n",
      "-r-x--x--x 1 root root 176947200 Feb  6 04:52 sai_192_whitelist\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/d5/12b6921f983d2326fa1e91570a96e7/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/6f/e3ee295a7f5364c39365c48e8bd0b0/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/4a/7b5d74b3a4903977d7316c5f02ef84/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/2d/0e59fd7f9797eaf684de4abc52eb46/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/c5/cc9149d8d21bb423337888a9dc77ea/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/dc/65201d13e12dffd8779bbc935604a6/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb  6 04:52 na.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/f4/bab8ffeff41eecffb2ddcf013b4121/na.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       125 Feb  6 04:52 na1.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/38/b73884d031cf00d6089d843ba0f965/na1.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       140 Feb  6 04:52 sai_192_whitelist.txt -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/09/7ba70bbeac00bcd6547ad0c09655c7/sai_192_whitelist.txt\u001b[K\n",
      "Finished run_whitelist_gunzip\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[41A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (4)\u001b[K\n",
      "[21/b2f514] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[bf/49817f] process > run_whitelist_gunzip (1)       [100%] 1 of 1 ✔\u001b[K\n",
      "[29/b7b785] process > run_add_subpool_prefix_to_f... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "Start run_whitelist_gunzip\u001b[K\n",
      "Gunzip file is: sai_192_whitelist.txt\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 172805\u001b[K\n",
      "-r-x--x--x 1 root root 176947200 Feb  6 04:52 sai_192_whitelist\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/d5/12b6921f983d2326fa1e91570a96e7/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/6f/e3ee295a7f5364c39365c48e8bd0b0/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/4a/7b5d74b3a4903977d7316c5f02ef84/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/2d/0e59fd7f9797eaf684de4abc52eb46/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/c5/cc9149d8d21bb423337888a9dc77ea/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/dc/65201d13e12dffd8779bbc935604a6/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb  6 04:52 na.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/f4/bab8ffeff41eecffb2ddcf013b4121/na.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       125 Feb  6 04:52 na1.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/38/b73884d031cf00d6089d843ba0f965/na1.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       140 Feb  6 04:52 sai_192_whitelist.txt -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/09/7ba70bbeac00bcd6547ad0c09655c7/sai_192_whitelist.txt\u001b[K\n",
      "Finished run_whitelist_gunzip\u001b[K\n",
      "\u001b[K\n",
      "\u001b[35A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (4)\u001b[K\n",
      "[21/b2f514] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[bf/49817f] process > run_whitelist_gunzip (1)       [100%] 1 of 1 ✔\u001b[K\n",
      "[29/b7b785] process > run_add_subpool_prefix_to_f... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "Start run_whitelist_gunzip\u001b[K\n",
      "Gunzip file is: sai_192_whitelist.txt\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 172805\u001b[K\n",
      "-r-x--x--x 1 root root 176947200 Feb  6 04:52 sai_192_whitelist\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/d5/12b6921f983d2326fa1e91570a96e7/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/6f/e3ee295a7f5364c39365c48e8bd0b0/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/4a/7b5d74b3a4903977d7316c5f02ef84/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/2d/0e59fd7f9797eaf684de4abc52eb46/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/c5/cc9149d8d21bb423337888a9dc77ea/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/dc/65201d13e12dffd8779bbc935604a6/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb  6 04:52 na.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/f4/bab8ffeff41eecffb2ddcf013b4121/na.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       125 Feb  6 04:52 na1.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/38/b73884d031cf00d6089d843ba0f965/na1.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       140 Feb  6 04:52 sai_192_whitelist.txt -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/09/7ba70bbeac00bcd6547ad0c09655c7/sai_192_whitelist.txt\u001b[K\n",
      "Finished run_whitelist_gunzip\u001b[K\n",
      "\u001b[K\n",
      "\u001b[35A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (4)\u001b[K\n",
      "[21/b2f514] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[bf/49817f] process > run_whitelist_gunzip (1)       [100%] 1 of 1 ✔\u001b[K\n",
      "[29/b7b785] process > run_add_subpool_prefix_to_f... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "Start run_whitelist_gunzip\u001b[K\n",
      "Gunzip file is: sai_192_whitelist.txt\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 172805\u001b[K\n",
      "-r-x--x--x 1 root root 176947200 Feb  6 04:52 sai_192_whitelist\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/d5/12b6921f983d2326fa1e91570a96e7/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/6f/e3ee295a7f5364c39365c48e8bd0b0/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/4a/7b5d74b3a4903977d7316c5f02ef84/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/2d/0e59fd7f9797eaf684de4abc52eb46/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/c5/cc9149d8d21bb423337888a9dc77ea/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/dc/65201d13e12dffd8779bbc935604a6/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb  6 04:52 na.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/f4/bab8ffeff41eecffb2ddcf013b4121/na.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       125 Feb  6 04:52 na1.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/38/b73884d031cf00d6089d843ba0f965/na1.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       140 Feb  6 04:52 sai_192_whitelist.txt -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/09/7ba70bbeac00bcd6547ad0c09655c7/sai_192_whitelist.txt\u001b[K\n",
      "Finished run_whitelist_gunzip\u001b[K\n",
      "\u001b[K\n",
      "\u001b[35A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (4)\u001b[K\n",
      "[21/b2f514] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[bf/49817f] process > run_whitelist_gunzip (1)       [100%] 1 of 1 ✔\u001b[K\n",
      "[29/b7b785] process > run_add_subpool_prefix_to_f... [  0%] 0 of 1 ✔\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "Start run_whitelist_gunzip\u001b[K\n",
      "Gunzip file is: sai_192_whitelist.txt\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 172805\u001b[K\n",
      "-r-x--x--x 1 root root 176947200 Feb  6 04:52 sai_192_whitelist\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/d5/12b6921f983d2326fa1e91570a96e7/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/6f/e3ee295a7f5364c39365c48e8bd0b0/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/4a/7b5d74b3a4903977d7316c5f02ef84/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/2d/0e59fd7f9797eaf684de4abc52eb46/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/c5/cc9149d8d21bb423337888a9dc77ea/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/dc/65201d13e12dffd8779bbc935604a6/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb  6 04:52 na.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/f4/bab8ffeff41eecffb2ddcf013b4121/na.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       125 Feb  6 04:52 na1.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/38/b73884d031cf00d6089d843ba0f965/na1.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       140 Feb  6 04:52 sai_192_whitelist.txt -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/09/7ba70bbeac00bcd6547ad0c09655c7/sai_192_whitelist.txt\u001b[K\n",
      "Finished run_whitelist_gunzip\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "Input chromap_filter_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_fragments_table.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "\u001b[K\n",
      "\u001b[43A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (4)\u001b[K\n",
      "[21/b2f514] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[bf/49817f] process > run_whitelist_gunzip (1)       [100%] 1 of 1 ✔\u001b[K\n",
      "[29/b7b785] process > run_add_subpool_prefix_to_f... [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "Start run_whitelist_gunzip\u001b[K\n",
      "Gunzip file is: sai_192_whitelist.txt\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 172805\u001b[K\n",
      "-r-x--x--x 1 root root 176947200 Feb  6 04:52 sai_192_whitelist\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/d5/12b6921f983d2326fa1e91570a96e7/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/6f/e3ee295a7f5364c39365c48e8bd0b0/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/4a/7b5d74b3a4903977d7316c5f02ef84/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/2d/0e59fd7f9797eaf684de4abc52eb46/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/c5/cc9149d8d21bb423337888a9dc77ea/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/dc/65201d13e12dffd8779bbc935604a6/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb  6 04:52 na.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/f4/bab8ffeff41eecffb2ddcf013b4121/na.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       125 Feb  6 04:52 na1.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/38/b73884d031cf00d6089d843ba0f965/na1.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       140 Feb  6 04:52 sai_192_whitelist.txt -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/09/7ba70bbeac00bcd6547ad0c09655c7/sai_192_whitelist.txt\u001b[K\n",
      "Finished run_whitelist_gunzip\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "Input chromap_filter_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_fragments_table.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "\u001b[K\n",
      "\u001b[43A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (4)\u001b[K\n",
      "[21/b2f514] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[bf/49817f] process > run_whitelist_gunzip (1)       [100%] 1 of 1 ✔\u001b[K\n",
      "[29/b7b785] process > run_add_subpool_prefix_to_f... [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "Start run_whitelist_gunzip\u001b[K\n",
      "Gunzip file is: sai_192_whitelist.txt\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 172805\u001b[K\n",
      "-r-x--x--x 1 root root 176947200 Feb  6 04:52 sai_192_whitelist\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/d5/12b6921f983d2326fa1e91570a96e7/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/6f/e3ee295a7f5364c39365c48e8bd0b0/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/4a/7b5d74b3a4903977d7316c5f02ef84/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/2d/0e59fd7f9797eaf684de4abc52eb46/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/c5/cc9149d8d21bb423337888a9dc77ea/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/dc/65201d13e12dffd8779bbc935604a6/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb  6 04:52 na.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/f4/bab8ffeff41eecffb2ddcf013b4121/na.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       125 Feb  6 04:52 na1.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/38/b73884d031cf00d6089d843ba0f965/na1.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       140 Feb  6 04:52 sai_192_whitelist.txt -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/09/7ba70bbeac00bcd6547ad0c09655c7/sai_192_whitelist.txt\u001b[K\n",
      "Finished run_whitelist_gunzip\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "Input chromap_filter_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_fragments_table.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "\u001b[K\n",
      "\u001b[43A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (5)\u001b[K\n",
      "[21/b2f514] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[bf/49817f] process > run_whitelist_gunzip (1)       [100%] 1 of 1 ✔\u001b[K\n",
      "[29/b7b785] process > run_add_subpool_prefix_to_f... [100%] 1 of 1 ✔\u001b[K\n",
      "[cb/0b8d42] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "Start run_whitelist_gunzip\u001b[K\n",
      "Gunzip file is: sai_192_whitelist.txt\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 172805\u001b[K\n",
      "-r-x--x--x 1 root root 176947200 Feb  6 04:52 sai_192_whitelist\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/d5/12b6921f983d2326fa1e91570a96e7/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/6f/e3ee295a7f5364c39365c48e8bd0b0/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/4a/7b5d74b3a4903977d7316c5f02ef84/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/2d/0e59fd7f9797eaf684de4abc52eb46/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/c5/cc9149d8d21bb423337888a9dc77ea/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/dc/65201d13e12dffd8779bbc935604a6/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb  6 04:52 na.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/f4/bab8ffeff41eecffb2ddcf013b4121/na.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       125 Feb  6 04:52 na1.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/38/b73884d031cf00d6089d843ba0f965/na1.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       140 Feb  6 04:52 sai_192_whitelist.txt -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/09/7ba70bbeac00bcd6547ad0c09655c7/sai_192_whitelist.txt\u001b[K\n",
      "Finished run_whitelist_gunzip\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "Input chromap_filter_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_fragments_table.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "\u001b[K\n",
      "\u001b[43A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (5)\u001b[K\n",
      "[21/b2f514] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[bf/49817f] process > run_whitelist_gunzip (1)       [100%] 1 of 1 ✔\u001b[K\n",
      "[29/b7b785] process > run_add_subpool_prefix_to_f... [100%] 1 of 1 ✔\u001b[K\n",
      "[cb/0b8d42] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "Start run_whitelist_gunzip\u001b[K\n",
      "Gunzip file is: sai_192_whitelist.txt\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 172805\u001b[K\n",
      "-r-x--x--x 1 root root 176947200 Feb  6 04:52 sai_192_whitelist\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/d5/12b6921f983d2326fa1e91570a96e7/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/6f/e3ee295a7f5364c39365c48e8bd0b0/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/4a/7b5d74b3a4903977d7316c5f02ef84/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/2d/0e59fd7f9797eaf684de4abc52eb46/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/c5/cc9149d8d21bb423337888a9dc77ea/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/dc/65201d13e12dffd8779bbc935604a6/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb  6 04:52 na.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/f4/bab8ffeff41eecffb2ddcf013b4121/na.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       125 Feb  6 04:52 na1.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/38/b73884d031cf00d6089d843ba0f965/na1.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       140 Feb  6 04:52 sai_192_whitelist.txt -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/09/7ba70bbeac00bcd6547ad0c09655c7/sai_192_whitelist.txt\u001b[K\n",
      "Finished run_whitelist_gunzip\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "Input chromap_filter_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_fragments_table.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "\u001b[K\n",
      "\u001b[43A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (6)\u001b[K\n",
      "[21/b2f514] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[bf/49817f] process > run_whitelist_gunzip (1)       [100%] 1 of 1 ✔\u001b[K\n",
      "[29/b7b785] process > run_add_subpool_prefix_to_f... [100%] 1 of 1 ✔\u001b[K\n",
      "[cb/0b8d42] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[30/49232d] process > run_merge_chromap_logs (1)     [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "Start run_whitelist_gunzip\u001b[K\n",
      "Gunzip file is: sai_192_whitelist.txt\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 172805\u001b[K\n",
      "-r-x--x--x 1 root root 176947200 Feb  6 04:52 sai_192_whitelist\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/d5/12b6921f983d2326fa1e91570a96e7/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/6f/e3ee295a7f5364c39365c48e8bd0b0/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/4a/7b5d74b3a4903977d7316c5f02ef84/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/2d/0e59fd7f9797eaf684de4abc52eb46/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/c5/cc9149d8d21bb423337888a9dc77ea/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/dc/65201d13e12dffd8779bbc935604a6/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb  6 04:52 na.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/f4/bab8ffeff41eecffb2ddcf013b4121/na.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       125 Feb  6 04:52 na1.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/38/b73884d031cf00d6089d843ba0f965/na1.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       140 Feb  6 04:52 sai_192_whitelist.txt -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/09/7ba70bbeac00bcd6547ad0c09655c7/sai_192_whitelist.txt\u001b[K\n",
      "Finished run_whitelist_gunzip\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "Input chromap_filter_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_fragments_table.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "\u001b[K\n",
      "\u001b[43A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (7)\u001b[K\n",
      "[21/b2f514] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[bf/49817f] process > run_whitelist_gunzip (1)       [100%] 1 of 1 ✔\u001b[K\n",
      "[29/b7b785] process > run_add_subpool_prefix_to_f... [100%] 1 of 1 ✔\u001b[K\n",
      "[cb/0b8d42] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[30/49232d] process > run_merge_chromap_logs (1)     [  0%] 0 of 1\u001b[K\n",
      "[09/1f1698] process > run_process_conversion_to_b... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "Start run_whitelist_gunzip\u001b[K\n",
      "Gunzip file is: sai_192_whitelist.txt\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 172805\u001b[K\n",
      "-r-x--x--x 1 root root 176947200 Feb  6 04:52 sai_192_whitelist\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/d5/12b6921f983d2326fa1e91570a96e7/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/6f/e3ee295a7f5364c39365c48e8bd0b0/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/4a/7b5d74b3a4903977d7316c5f02ef84/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/2d/0e59fd7f9797eaf684de4abc52eb46/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/c5/cc9149d8d21bb423337888a9dc77ea/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/dc/65201d13e12dffd8779bbc935604a6/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb  6 04:52 na.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/f4/bab8ffeff41eecffb2ddcf013b4121/na.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       125 Feb  6 04:52 na1.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/38/b73884d031cf00d6089d843ba0f965/na1.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       140 Feb  6 04:52 sai_192_whitelist.txt -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/09/7ba70bbeac00bcd6547ad0c09655c7/sai_192_whitelist.txt\u001b[K\n",
      "Finished run_whitelist_gunzip\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "Input chromap_filter_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_fragments_table.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "\u001b[K\n",
      "\u001b[43A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (7)\u001b[K\n",
      "[21/b2f514] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[bf/49817f] process > run_whitelist_gunzip (1)       [100%] 1 of 1 ✔\u001b[K\n",
      "[29/b7b785] process > run_add_subpool_prefix_to_f... [100%] 1 of 1 ✔\u001b[K\n",
      "[cb/0b8d42] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[30/49232d] process > run_merge_chromap_logs (1)     [  0%] 0 of 1\u001b[K\n",
      "[09/1f1698] process > run_process_conversion_to_b... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [  0%] 0 of 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "Start run_whitelist_gunzip\u001b[K\n",
      "Gunzip file is: sai_192_whitelist.txt\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 172805\u001b[K\n",
      "-r-x--x--x 1 root root 176947200 Feb  6 04:52 sai_192_whitelist\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/d5/12b6921f983d2326fa1e91570a96e7/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/6f/e3ee295a7f5364c39365c48e8bd0b0/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/4a/7b5d74b3a4903977d7316c5f02ef84/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/2d/0e59fd7f9797eaf684de4abc52eb46/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/c5/cc9149d8d21bb423337888a9dc77ea/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/dc/65201d13e12dffd8779bbc935604a6/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb  6 04:52 na.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/f4/bab8ffeff41eecffb2ddcf013b4121/na.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       125 Feb  6 04:52 na1.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/38/b73884d031cf00d6089d843ba0f965/na1.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       140 Feb  6 04:52 sai_192_whitelist.txt -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/09/7ba70bbeac00bcd6547ad0c09655c7/sai_192_whitelist.txt\u001b[K\n",
      "Finished run_whitelist_gunzip\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "Input chromap_filter_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_fragments_table.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb  6 04:52 genes.gtf.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb  6 04:52 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/eb/6412854cc6c199859cdfbe280f86c9/genes.gtf\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "\u001b[51A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (7)\u001b[K\n",
      "[21/b2f514] process > run_zcat                       [  0%] 0 of 1\u001b[K\n",
      "[bf/49817f] process > run_whitelist_gunzip (1)       [100%] 1 of 1 ✔\u001b[K\n",
      "[29/b7b785] process > run_add_subpool_prefix_to_f... [100%] 1 of 1 ✔\u001b[K\n",
      "[cb/0b8d42] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[30/49232d] process > run_merge_chromap_logs (1)     [  0%] 0 of 1\u001b[K\n",
      "[09/1f1698] process > run_process_conversion_to_b... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "Start run_whitelist_gunzip\u001b[K\n",
      "Gunzip file is: sai_192_whitelist.txt\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 172805\u001b[K\n",
      "-r-x--x--x 1 root root 176947200 Feb  6 04:52 sai_192_whitelist\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/d5/12b6921f983d2326fa1e91570a96e7/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/6f/e3ee295a7f5364c39365c48e8bd0b0/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/4a/7b5d74b3a4903977d7316c5f02ef84/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/2d/0e59fd7f9797eaf684de4abc52eb46/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/c5/cc9149d8d21bb423337888a9dc77ea/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  6 04:52 BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/dc/65201d13e12dffd8779bbc935604a6/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb  6 04:52 na.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/f4/bab8ffeff41eecffb2ddcf013b4121/na.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       125 Feb  6 04:52 na1.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/38/b73884d031cf00d6089d843ba0f965/na1.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       140 Feb  6 04:52 sai_192_whitelist.txt -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/09/7ba70bbeac00bcd6547ad0c09655c7/sai_192_whitelist.txt\u001b[K\n",
      "Finished run_whitelist_gunzip\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "Input chromap_filter_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_fragments_table.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb  6 04:52 genes.gtf.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb  6 04:52 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/eb/6412854cc6c199859cdfbe280f86c9/genes.gtf\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "\u001b[51A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (7)\u001b[K\n",
      "[21/b2f514] process > run_zcat                       [  0%] 0 of 1 ✔\u001b[K\n",
      "[bf/49817f] process > run_whitelist_gunzip (1)       [100%] 1 of 1 ✔\u001b[K\n",
      "[29/b7b785] process > run_add_subpool_prefix_to_f... [100%] 1 of 1 ✔\u001b[K\n",
      "[cb/0b8d42] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[30/49232d] process > run_merge_chromap_logs (1)     [  0%] 0 of 1\u001b[K\n",
      "[09/1f1698] process > run_process_conversion_to_b... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb  6 04:52 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb  6 04:52 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb  6 04:52 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[62A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (7)\u001b[K\n",
      "[21/b2f514] process > run_zcat                       [100%] 1 of 1 ✔\u001b[K\n",
      "[bf/49817f] process > run_whitelist_gunzip (1)       [100%] 1 of 1 ✔\u001b[K\n",
      "[29/b7b785] process > run_add_subpool_prefix_to_f... [100%] 1 of 1 ✔\u001b[K\n",
      "[cb/0b8d42] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[30/49232d] process > run_merge_chromap_logs (1)     [  0%] 0 of 1\u001b[K\n",
      "[09/1f1698] process > run_process_conversion_to_b... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb  6 04:52 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb  6 04:52 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb  6 04:52 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[62A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (7)\u001b[K\n",
      "[21/b2f514] process > run_zcat                       [100%] 1 of 1 ✔\u001b[K\n",
      "[bf/49817f] process > run_whitelist_gunzip (1)       [100%] 1 of 1 ✔\u001b[K\n",
      "[29/b7b785] process > run_add_subpool_prefix_to_f... [100%] 1 of 1 ✔\u001b[K\n",
      "[cb/0b8d42] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[30/49232d] process > run_merge_chromap_logs (1)     [  0%] 0 of 1\u001b[K\n",
      "[09/1f1698] process > run_process_conversion_to_b... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[29A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (7)\u001b[K\n",
      "[21/b2f514] process > run_zcat                       [100%] 1 of 1 ✔\u001b[K\n",
      "[bf/49817f] process > run_whitelist_gunzip (1)       [100%] 1 of 1 ✔\u001b[K\n",
      "[29/b7b785] process > run_add_subpool_prefix_to_f... [100%] 1 of 1 ✔\u001b[K\n",
      "[cb/0b8d42] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[30/49232d] process > run_merge_chromap_logs (1)     [  0%] 0 of 1\u001b[K\n",
      "[09/1f1698] process > run_process_conversion_to_b... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[29A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (7)\u001b[K\n",
      "[21/b2f514] process > run_zcat                       [100%] 1 of 1 ✔\u001b[K\n",
      "[bf/49817f] process > run_whitelist_gunzip (1)       [100%] 1 of 1 ✔\u001b[K\n",
      "[29/b7b785] process > run_add_subpool_prefix_to_f... [100%] 1 of 1 ✔\u001b[K\n",
      "[cb/0b8d42] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[30/49232d] process > run_merge_chromap_logs (1)     [100%] 1 of 1 ✔\u001b[K\n",
      "[09/1f1698] process > run_process_conversion_to_b... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "start run_merge_logs\u001b[K\n",
      "input script_name is merge_chromap_logs.sh\u001b[K\n",
      "input chromap_alignment_log is BMMC-single-donor.atac.align.k4.hg38.log.txt\u001b[K\n",
      "input chromap_barcode_log is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "start run_merge_chromap_logs shell script\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "********1***********\u001b[K\n",
      "********2***********\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "uni-mappings,260650161\u001b[K\n",
      "multi-mappings,14666503\u001b[K\n",
      "total,275316664\u001b[K\n",
      "********3***********\u001b[K\n",
      "finished run_merge_chromap_logs shell script\u001b[K\n",
      "finished run_merge_logs\u001b[K\n",
      "\u001b[K\n",
      "\u001b[54A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (7)\u001b[K\n",
      "[21/b2f514] process > run_zcat                       [100%] 1 of 1 ✔\u001b[K\n",
      "[bf/49817f] process > run_whitelist_gunzip (1)       [100%] 1 of 1 ✔\u001b[K\n",
      "[29/b7b785] process > run_add_subpool_prefix_to_f... [100%] 1 of 1 ✔\u001b[K\n",
      "[cb/0b8d42] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[30/49232d] process > run_merge_chromap_logs (1)     [100%] 1 of 1 ✔\u001b[K\n",
      "[09/1f1698] process > run_process_conversion_to_b... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "start run_merge_logs\u001b[K\n",
      "input script_name is merge_chromap_logs.sh\u001b[K\n",
      "input chromap_alignment_log is BMMC-single-donor.atac.align.k4.hg38.log.txt\u001b[K\n",
      "input chromap_barcode_log is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "start run_merge_chromap_logs shell script\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "********1***********\u001b[K\n",
      "********2***********\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "uni-mappings,260650161\u001b[K\n",
      "multi-mappings,14666503\u001b[K\n",
      "total,275316664\u001b[K\n",
      "********3***********\u001b[K\n",
      "finished run_merge_chromap_logs shell script\u001b[K\n",
      "finished run_merge_logs\u001b[K\n",
      "\u001b[K\n",
      "\u001b[54A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (7)\u001b[K\n",
      "[21/b2f514] process > run_zcat                       [100%] 1 of 1 ✔\u001b[K\n",
      "[bf/49817f] process > run_whitelist_gunzip (1)       [100%] 1 of 1 ✔\u001b[K\n",
      "[29/b7b785] process > run_add_subpool_prefix_to_f... [100%] 1 of 1 ✔\u001b[K\n",
      "[cb/0b8d42] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[30/49232d] process > run_merge_chromap_logs (1)     [100%] 1 of 1 ✔\u001b[K\n",
      "[09/1f1698] process > run_process_conversion_to_b... [  0%] 0 of 1 ✔\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "start run_merge_logs\u001b[K\n",
      "input script_name is merge_chromap_logs.sh\u001b[K\n",
      "input chromap_alignment_log is BMMC-single-donor.atac.align.k4.hg38.log.txt\u001b[K\n",
      "input chromap_barcode_log is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "start run_merge_chromap_logs shell script\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "********1***********\u001b[K\n",
      "********2***********\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "uni-mappings,260650161\u001b[K\n",
      "multi-mappings,14666503\u001b[K\n",
      "total,275316664\u001b[K\n",
      "********3***********\u001b[K\n",
      "finished run_merge_chromap_logs shell script\u001b[K\n",
      "finished run_merge_logs\u001b[K\n",
      "\u001b[K\n",
      "start run_process_conversion_to_barcode\u001b[K\n",
      "input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "------ There is NO conversion list ------\u001b[K\n",
      "temp_conversion\u001b[K\n",
      "7021039 temp_conversion\u001b[K\n",
      "finished run_process_conversion_to_barcode\u001b[K\n",
      "\u001b[K\n",
      "\u001b[61A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (7)\u001b[K\n",
      "[21/b2f514] process > run_zcat                       [100%] 1 of 1 ✔\u001b[K\n",
      "[bf/49817f] process > run_whitelist_gunzip (1)       [100%] 1 of 1 ✔\u001b[K\n",
      "[29/b7b785] process > run_add_subpool_prefix_to_f... [100%] 1 of 1 ✔\u001b[K\n",
      "[cb/0b8d42] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[30/49232d] process > run_merge_chromap_logs (1)     [100%] 1 of 1 ✔\u001b[K\n",
      "[09/1f1698] process > run_process_conversion_to_b... [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "start run_merge_logs\u001b[K\n",
      "input script_name is merge_chromap_logs.sh\u001b[K\n",
      "input chromap_alignment_log is BMMC-single-donor.atac.align.k4.hg38.log.txt\u001b[K\n",
      "input chromap_barcode_log is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "start run_merge_chromap_logs shell script\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "********1***********\u001b[K\n",
      "********2***********\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "uni-mappings,260650161\u001b[K\n",
      "multi-mappings,14666503\u001b[K\n",
      "total,275316664\u001b[K\n",
      "********3***********\u001b[K\n",
      "finished run_merge_chromap_logs shell script\u001b[K\n",
      "finished run_merge_logs\u001b[K\n",
      "\u001b[K\n",
      "start run_process_conversion_to_barcode\u001b[K\n",
      "input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "------ There is NO conversion list ------\u001b[K\n",
      "temp_conversion\u001b[K\n",
      "7021039 temp_conversion\u001b[K\n",
      "finished run_process_conversion_to_barcode\u001b[K\n",
      "\u001b[K\n",
      "\u001b[61A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (7)\u001b[K\n",
      "[21/b2f514] process > run_zcat                       [100%] 1 of 1 ✔\u001b[K\n",
      "[bf/49817f] process > run_whitelist_gunzip (1)       [100%] 1 of 1 ✔\u001b[K\n",
      "[29/b7b785] process > run_add_subpool_prefix_to_f... [100%] 1 of 1 ✔\u001b[K\n",
      "[cb/0b8d42] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[30/49232d] process > run_merge_chromap_logs (1)     [100%] 1 of 1 ✔\u001b[K\n",
      "[09/1f1698] process > run_process_conversion_to_b... [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "start run_merge_logs\u001b[K\n",
      "input script_name is merge_chromap_logs.sh\u001b[K\n",
      "input chromap_alignment_log is BMMC-single-donor.atac.align.k4.hg38.log.txt\u001b[K\n",
      "input chromap_barcode_log is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "start run_merge_chromap_logs shell script\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "********1***********\u001b[K\n",
      "********2***********\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "uni-mappings,260650161\u001b[K\n",
      "multi-mappings,14666503\u001b[K\n",
      "total,275316664\u001b[K\n",
      "********3***********\u001b[K\n",
      "finished run_merge_chromap_logs shell script\u001b[K\n",
      "finished run_merge_logs\u001b[K\n",
      "\u001b[K\n",
      "start run_process_conversion_to_barcode\u001b[K\n",
      "input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "------ There is NO conversion list ------\u001b[K\n",
      "temp_conversion\u001b[K\n",
      "7021039 temp_conversion\u001b[K\n",
      "finished run_process_conversion_to_barcode\u001b[K\n",
      "\u001b[K\n",
      "\u001b[61A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (8)\u001b[K\n",
      "[21/b2f514] process > run_zcat                       [100%] 1 of 1 ✔\u001b[K\n",
      "[bf/49817f] process > run_whitelist_gunzip (1)       [100%] 1 of 1 ✔\u001b[K\n",
      "[29/b7b785] process > run_add_subpool_prefix_to_f... [100%] 1 of 1 ✔\u001b[K\n",
      "[cb/0b8d42] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[30/49232d] process > run_merge_chromap_logs (1)     [100%] 1 of 1 ✔\u001b[K\n",
      "[09/1f1698] process > run_process_conversion_to_b... [100%] 1 of 1 ✔\u001b[K\n",
      "[85/78bb28] process > run_filter_aligned_fragment... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "start run_merge_logs\u001b[K\n",
      "input script_name is merge_chromap_logs.sh\u001b[K\n",
      "input chromap_alignment_log is BMMC-single-donor.atac.align.k4.hg38.log.txt\u001b[K\n",
      "input chromap_barcode_log is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "start run_merge_chromap_logs shell script\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "********1***********\u001b[K\n",
      "********2***********\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "uni-mappings,260650161\u001b[K\n",
      "multi-mappings,14666503\u001b[K\n",
      "total,275316664\u001b[K\n",
      "********3***********\u001b[K\n",
      "finished run_merge_chromap_logs shell script\u001b[K\n",
      "finished run_merge_logs\u001b[K\n",
      "\u001b[K\n",
      "start run_process_conversion_to_barcode\u001b[K\n",
      "input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "------ There is NO conversion list ------\u001b[K\n",
      "temp_conversion\u001b[K\n",
      "7021039 temp_conversion\u001b[K\n",
      "finished run_process_conversion_to_barcode\u001b[K\n",
      "\u001b[K\n",
      "\u001b[61A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (8)\u001b[K\n",
      "[21/b2f514] process > run_zcat                       [100%] 1 of 1 ✔\u001b[K\n",
      "[bf/49817f] process > run_whitelist_gunzip (1)       [100%] 1 of 1 ✔\u001b[K\n",
      "[29/b7b785] process > run_add_subpool_prefix_to_f... [100%] 1 of 1 ✔\u001b[K\n",
      "[cb/0b8d42] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[30/49232d] process > run_merge_chromap_logs (1)     [100%] 1 of 1 ✔\u001b[K\n",
      "[09/1f1698] process > run_process_conversion_to_b... [100%] 1 of 1 ✔\u001b[K\n",
      "[85/78bb28] process > run_filter_aligned_fragment... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[61A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (8)\u001b[K\n",
      "[21/b2f514] process > run_zcat                       [100%] 1 of 1 ✔\u001b[K\n",
      "[bf/49817f] process > run_whitelist_gunzip (1)       [100%] 1 of 1 ✔\u001b[K\n",
      "[29/b7b785] process > run_add_subpool_prefix_to_f... [100%] 1 of 1 ✔\u001b[K\n",
      "[cb/0b8d42] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[30/49232d] process > run_merge_chromap_logs (1)     [100%] 1 of 1 ✔\u001b[K\n",
      "[09/1f1698] process > run_process_conversion_to_b... [100%] 1 of 1 ✔\u001b[K\n",
      "[85/78bb28] process > run_filter_aligned_fragment... [  0%] 0 of 1 ✔\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "start run_filter_aligned_fragments\u001b[K\n",
      "input barcodeFragmentsTsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "input threshold is 10\u001b[K\n",
      "input temp_dict_conversion is temp_conversion\u001b[K\n",
      "------ Number of barcodes BEFORE filtering------\u001b[K\n",
      "7021039 temp_conversion\u001b[K\n",
      "------ Filtering fragments ------\u001b[K\n",
      "--1--\u001b[K\n",
      "--2--\u001b[K\n",
      "--3--\u001b[K\n",
      "------ Lines matching filtering condition ------\u001b[K\n",
      "GTTAGGAGGCGCCTAGTATGCCAG,19,3,1,0\u001b[K\n",
      "ACAGTGGTTGCATAGTGCGCCTAG,15,1,1,0\u001b[K\n",
      "GTTAGCCTGACCTTAGATTGCATA,20,2,0,0\u001b[K\n",
      "TCTCGGTTTAAGTTCGGTTGTCGG,13,2,0,0\u001b[K\n",
      "GACGGATTTACTTCGGGTGTCCTT,28,4,3,0\u001b[K\n",
      "GTTAGCCTGCGCCTAGTATGCCAG,20,3,0,0\u001b[K\n",
      "GACGGATTGTTGTCGGGTGTCCTT,48,9,11,0\u001b[K\n",
      "TTGTTCCATGGTTGTTTCTCTTCA,17,1,1,0\u001b[K\n",
      "CGATGTTTTTGGTATGATGCTAAC,32,5,3,0\u001b[K\n",
      "TCTCTTCATAAGTTCGGTTGTCGG,174,2,135,0\u001b[K\n",
      "------ Number of lines in temp_dict_conversion BEFORE filtering ------\u001b[K\n",
      "7021039 temp_conversion\u001b[K\n",
      "------ 3 - Number of lines in intermediate file AFTER 2nd filtering ------\u001b[K\n",
      "3950969 tmp_file\u001b[K\n",
      "------ Number of lines in zcat no-singleton.bed.gz AFTER filtering and compression ------\u001b[K\n",
      "3950969\u001b[K\n",
      "nam of output_file_name is: no-singleton.bed.gz\u001b[K\n",
      "------ Additional command: Counting lines in temp_filtered_file ------\u001b[K\n",
      "3950969\u001b[K\n",
      "finished run_filter_aligned_fragments\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[61A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (8)\u001b[K\n",
      "[21/b2f514] process > run_zcat                       [100%] 1 of 1 ✔\u001b[K\n",
      "[bf/49817f] process > run_whitelist_gunzip (1)       [100%] 1 of 1 ✔\u001b[K\n",
      "[29/b7b785] process > run_add_subpool_prefix_to_f... [100%] 1 of 1 ✔\u001b[K\n",
      "[cb/0b8d42] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[30/49232d] process > run_merge_chromap_logs (1)     [100%] 1 of 1 ✔\u001b[K\n",
      "[09/1f1698] process > run_process_conversion_to_b... [100%] 1 of 1 ✔\u001b[K\n",
      "[85/78bb28] process > run_filter_aligned_fragment... [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "start run_filter_aligned_fragments\u001b[K\n",
      "input barcodeFragmentsTsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "input threshold is 10\u001b[K\n",
      "input temp_dict_conversion is temp_conversion\u001b[K\n",
      "------ Number of barcodes BEFORE filtering------\u001b[K\n",
      "7021039 temp_conversion\u001b[K\n",
      "------ Filtering fragments ------\u001b[K\n",
      "--1--\u001b[K\n",
      "--2--\u001b[K\n",
      "--3--\u001b[K\n",
      "------ Lines matching filtering condition ------\u001b[K\n",
      "GTTAGGAGGCGCCTAGTATGCCAG,19,3,1,0\u001b[K\n",
      "ACAGTGGTTGCATAGTGCGCCTAG,15,1,1,0\u001b[K\n",
      "GTTAGCCTGACCTTAGATTGCATA,20,2,0,0\u001b[K\n",
      "TCTCGGTTTAAGTTCGGTTGTCGG,13,2,0,0\u001b[K\n",
      "GACGGATTTACTTCGGGTGTCCTT,28,4,3,0\u001b[K\n",
      "GTTAGCCTGCGCCTAGTATGCCAG,20,3,0,0\u001b[K\n",
      "GACGGATTGTTGTCGGGTGTCCTT,48,9,11,0\u001b[K\n",
      "TTGTTCCATGGTTGTTTCTCTTCA,17,1,1,0\u001b[K\n",
      "CGATGTTTTTGGTATGATGCTAAC,32,5,3,0\u001b[K\n",
      "TCTCTTCATAAGTTCGGTTGTCGG,174,2,135,0\u001b[K\n",
      "------ Number of lines in temp_dict_conversion BEFORE filtering ------\u001b[K\n",
      "7021039 temp_conversion\u001b[K\n",
      "------ 3 - Number of lines in intermediate file AFTER 2nd filtering ------\u001b[K\n",
      "3950969 tmp_file\u001b[K\n",
      "------ Number of lines in zcat no-singleton.bed.gz AFTER filtering and compression ------\u001b[K\n",
      "3950969\u001b[K\n",
      "nam of output_file_name is: no-singleton.bed.gz\u001b[K\n",
      "------ Additional command: Counting lines in temp_filtered_file ------\u001b[K\n",
      "3950969\u001b[K\n",
      "finished run_filter_aligned_fragments\u001b[K\n",
      "\u001b[K\n",
      "\u001b[50A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (9)\u001b[K\n",
      "[21/b2f514] process > run_zcat                       [100%] 1 of 1 ✔\u001b[K\n",
      "[bf/49817f] process > run_whitelist_gunzip (1)       [100%] 1 of 1 ✔\u001b[K\n",
      "[29/b7b785] process > run_add_subpool_prefix_to_f... [100%] 1 of 1 ✔\u001b[K\n",
      "[cb/0b8d42] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[30/49232d] process > run_merge_chromap_logs (1)     [100%] 1 of 1 ✔\u001b[K\n",
      "[09/1f1698] process > run_process_conversion_to_b... [100%] 1 of 1 ✔\u001b[K\n",
      "[85/78bb28] process > run_filter_aligned_fragment... [100%] 1 of 1 ✔\u001b[K\n",
      "[13/79788c] process > run_tabix_no_singleton (1)     [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "start run_filter_aligned_fragments\u001b[K\n",
      "input barcodeFragmentsTsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "input threshold is 10\u001b[K\n",
      "input temp_dict_conversion is temp_conversion\u001b[K\n",
      "------ Number of barcodes BEFORE filtering------\u001b[K\n",
      "7021039 temp_conversion\u001b[K\n",
      "------ Filtering fragments ------\u001b[K\n",
      "--1--\u001b[K\n",
      "--2--\u001b[K\n",
      "--3--\u001b[K\n",
      "------ Lines matching filtering condition ------\u001b[K\n",
      "GTTAGGAGGCGCCTAGTATGCCAG,19,3,1,0\u001b[K\n",
      "ACAGTGGTTGCATAGTGCGCCTAG,15,1,1,0\u001b[K\n",
      "GTTAGCCTGACCTTAGATTGCATA,20,2,0,0\u001b[K\n",
      "TCTCGGTTTAAGTTCGGTTGTCGG,13,2,0,0\u001b[K\n",
      "GACGGATTTACTTCGGGTGTCCTT,28,4,3,0\u001b[K\n",
      "GTTAGCCTGCGCCTAGTATGCCAG,20,3,0,0\u001b[K\n",
      "GACGGATTGTTGTCGGGTGTCCTT,48,9,11,0\u001b[K\n",
      "TTGTTCCATGGTTGTTTCTCTTCA,17,1,1,0\u001b[K\n",
      "CGATGTTTTTGGTATGATGCTAAC,32,5,3,0\u001b[K\n",
      "TCTCTTCATAAGTTCGGTTGTCGG,174,2,135,0\u001b[K\n",
      "------ Number of lines in temp_dict_conversion BEFORE filtering ------\u001b[K\n",
      "7021039 temp_conversion\u001b[K\n",
      "------ 3 - Number of lines in intermediate file AFTER 2nd filtering ------\u001b[K\n",
      "3950969 tmp_file\u001b[K\n",
      "------ Number of lines in zcat no-singleton.bed.gz AFTER filtering and compression ------\u001b[K\n",
      "3950969\u001b[K\n",
      "nam of output_file_name is: no-singleton.bed.gz\u001b[K\n",
      "------ Additional command: Counting lines in temp_filtered_file ------\u001b[K\n",
      "3950969\u001b[K\n",
      "finished run_filter_aligned_fragments\u001b[K\n",
      "\u001b[K\n",
      "\u001b[50A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (9)\u001b[K\n",
      "[21/b2f514] process > run_zcat                       [100%] 1 of 1 ✔\u001b[K\n",
      "[bf/49817f] process > run_whitelist_gunzip (1)       [100%] 1 of 1 ✔\u001b[K\n",
      "[29/b7b785] process > run_add_subpool_prefix_to_f... [100%] 1 of 1 ✔\u001b[K\n",
      "[cb/0b8d42] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[30/49232d] process > run_merge_chromap_logs (1)     [100%] 1 of 1 ✔\u001b[K\n",
      "[09/1f1698] process > run_process_conversion_to_b... [100%] 1 of 1 ✔\u001b[K\n",
      "[85/78bb28] process > run_filter_aligned_fragment... [100%] 1 of 1 ✔\u001b[K\n",
      "[13/79788c] process > run_tabix_no_singleton (1)     [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[50A\n",
      "WARN: Killing running tasks (1)\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (9)\u001b[K\n",
      "[21/b2f514] process > run_zcat                       [100%] 1 of 1 ✔\u001b[K\n",
      "[bf/49817f] process > run_whitelist_gunzip (1)       [100%] 1 of 1 ✔\u001b[K\n",
      "[29/b7b785] process > run_add_subpool_prefix_to_f... [100%] 1 of 1 ✔\u001b[K\n",
      "[cb/0b8d42] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[30/49232d] process > run_merge_chromap_logs (1)     [100%] 1 of 1 ✔\u001b[K\n",
      "[09/1f1698] process > run_process_conversion_to_b... [100%] 1 of 1 ✔\u001b[K\n",
      "[85/78bb28] process > run_filter_aligned_fragment... [100%] 1 of 1 ✔\u001b[K\n",
      "[13/79788c] process > run_tabix_no_singleton (1)     [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "\u001b[31mERROR ~ Error executing process > 'run_generate_insert_size_histogram_data (1)'\u001b[K\n",
      "\u001b[K\n",
      "Caused by:\u001b[K\n",
      "  Not a valid path value: 'generate_insert_size_histogram_data.sh'\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "Tip: when you have fixed the problem you can continue the execution adding the option `-resume` to the run command line\u001b[K\n",
      "\u001b[K\n",
      " -- Check '.nextflow.log' file for details\u001b[39m\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[50A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (9)\u001b[K\n",
      "[21/b2f514] process > run_zcat                       [100%] 1 of 1 ✔\u001b[K\n",
      "[bf/49817f] process > run_whitelist_gunzip (1)       [100%] 1 of 1 ✔\u001b[K\n",
      "[29/b7b785] process > run_add_subpool_prefix_to_f... [100%] 1 of 1 ✔\u001b[K\n",
      "[cb/0b8d42] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1 ✔\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[30/49232d] process > run_merge_chromap_logs (1)     [100%] 1 of 1 ✔\u001b[K\n",
      "[09/1f1698] process > run_process_conversion_to_b... [100%] 1 of 1 ✔\u001b[K\n",
      "[85/78bb28] process > run_filter_aligned_fragment... [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton (1)     -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[cd/bca4ab] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "[-        ] process > run_barcode_metadata_stats     -\u001b[K\n",
      "Start run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "Fragment file is: ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "total 2887679\u001b[K\n",
      "-rw-r--r-- 1 root root 2956982667 Feb  6 04:54 ATAC_BMMC-single-donor.atac.filter.fragments.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root        124 Feb  6 04:54 ATAC_BMMC-single-donor.atac.filter.fragments.tsv -> /fusion/gs/eila-nextflow-bucket/work/ATAC/29/b7b785aaaa075ee3bc1375da8789da/ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Finished run_bgzip_on_chromap_fragments_output\u001b[K\n",
      "\u001b[K\n",
      "\u001b[31mERROR ~ Error executing process > 'run_generate_insert_size_histogram_data (1)'\u001b[K\n",
      "\u001b[K\n",
      "Caused by:\u001b[K\n",
      "  Not a valid path value: 'generate_insert_size_histogram_data.sh'\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "Tip: when you have fixed the problem you can continue the execution adding the option `-resume` to the run command line\u001b[K\n",
      "\u001b[K\n",
      " -- Check '.nextflow.log' file for details\u001b[39m\u001b[K\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_atac.config -profile google_batch_profile -process.echo -with-tower -with-dag -resume\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_vm.config -profile google_batch_profile\n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_vm.config -profile google_batch_profile\n",
    "# !nextflow run nf_\n",
    "# \n",
    "# wf_fg_singlecell_atac.nf -c nextflow_vm_anvil.config -profile google_batch_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_local.config -profile google_batch_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_local.config -profile docker_profile\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2484,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Feb-05 23:50:05.408 [main] DEBUG nextflow.cli.Launcher - $> nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_atac.config -profile google_batch_profile -process.echo -with-tower -with-dag -resume\n",
      "Feb-05 23:50:05.640 [main] INFO  nextflow.cli.CmdRun - N E X T F L O W  ~  version 23.10.1\n",
      "Feb-05 23:50:05.656 [main] DEBUG nextflow.plugin.PluginsFacade - Setting up plugin manager > mode=prod; embedded=false; plugins-dir=/Users/eilaarich-landkof-stanford/.nextflow/plugins; core-plugins: nf-amazon@2.1.4,nf-azure@1.3.3,nf-cloudcache@0.3.0,nf-codecommit@0.1.5,nf-console@1.0.6,nf-ga4gh@1.1.0,nf-google@1.8.3,nf-tower@1.6.3,nf-wave@1.0.1\n",
      "Feb-05 23:50:05.664 [main] INFO  o.pf4j.DefaultPluginStatusProvider - Enabled plugins: []\n",
      "Feb-05 23:50:05.665 [main] INFO  o.pf4j.DefaultPluginStatusProvider - Disabled plugins: []\n",
      "Feb-05 23:50:05.668 [main] INFO  org.pf4j.DefaultPluginManager - PF4J version 3.4.1 in 'deployment' mode\n",
      "Feb-05 23:50:05.679 [main] INFO  org.pf4j.AbstractPluginManager - No plugins\n",
      "Feb-05 23:50:05.695 [main] DEBUG nextflow.config.ConfigBuilder - User config file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Feb-05 23:50:05.695 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Feb-05 23:50:05.712 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `google_batch_profile`\n",
      "Feb-05 23:50:07.604 [main] DEBUG nextflow.config.ConfigBuilder - Available config profiles: [google_batch_profile]\n",
      "Feb-05 23:50:07.659 [main] DEBUG nextflow.cli.CmdRun - Applied DSL=2 by global default\n",
      "Feb-05 23:50:07.684 [main] INFO  nextflow.cli.CmdRun - Launching `nf_wf_fg_singlecell_atac.nf` [sharp_leakey] DSL2 - revision: a05d84b688\n",
      "Feb-05 23:50:07.685 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins default=[nf-google@1.8.3]\n",
      "Feb-05 23:50:07.686 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins resolved requirement=[nf-google@1.8.3, nf-tower@1.6.3, nf-wave@1.0.1]\n",
      "Feb-05 23:50:07.686 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-google version: 1.8.3\n",
      "Feb-05 23:50:07.717 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-google@1.8.3' resolved\n",
      "Feb-05 23:50:07.717 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-google@1.8.3'\n",
      "Feb-05 23:50:07.783 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-google@1.8.3\n",
      "Feb-05 23:50:07.784 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-tower version: 1.6.3\n",
      "Feb-05 23:50:07.787 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-tower@1.6.3' resolved\n",
      "Feb-05 23:50:07.787 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-tower@1.6.3'\n",
      "Feb-05 23:50:07.793 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-tower@1.6.3\n",
      "Feb-05 23:50:07.793 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-wave version: 1.0.1\n",
      "Feb-05 23:50:07.799 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-wave@1.0.1' resolved\n",
      "Feb-05 23:50:07.799 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-wave@1.0.1'\n",
      "Feb-05 23:50:07.807 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-wave@1.0.1\n",
      "Feb-05 23:50:07.819 [main] DEBUG n.secret.LocalSecretsProvider - Secrets store: /Users/eilaarich-landkof-stanford/.nextflow/secrets/store.json\n",
      "Feb-05 23:50:07.821 [main] DEBUG nextflow.secret.SecretsLoader - Discovered secrets providers: [nextflow.secret.LocalSecretsProvider@210d2a6c] - activable => nextflow.secret.LocalSecretsProvider@210d2a6c\n",
      "Feb-05 23:50:07.888 [main] DEBUG nextflow.Session - Session UUID: ab09407f-2f8c-43e1-855a-210d7286a110\n",
      "Feb-05 23:50:07.888 [main] DEBUG nextflow.Session - Run name: sharp_leakey\n",
      "Feb-05 23:50:07.888 [main] DEBUG nextflow.Session - Executor pool size: 8\n",
      "Feb-05 23:50:08.031 [main] DEBUG nextflow.file.FilePorter - File porter settings maxRetries=3; maxTransfers=50; pollTimeout=null\n",
      "Feb-05 23:50:08.034 [main] DEBUG nextflow.util.ThreadPoolBuilder - Creating thread pool 'FileTransfer' minSize=10; maxSize=24; workQueue=LinkedBlockingQueue[10000]; allowCoreThreadTimeout=false\n",
      "Feb-05 23:50:08.038 [main] DEBUG nextflow.config.ConfigBuilder - User config file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Feb-05 23:50:08.038 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Feb-05 23:50:08.038 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `google_batch_profile`\n",
      "Feb-05 23:50:09.176 [main] DEBUG nextflow.config.ConfigBuilder - Available config profiles: [google_batch_profile]\n",
      "Feb-05 23:50:09.215 [main] DEBUG nextflow.cli.CmdRun - \n",
      "  Version: 23.10.1 build 5891\n",
      "  Created: 12-01-2024 22:01 UTC (17:01 EDT)\n",
      "  System: Mac OS X 14.3\n",
      "  Runtime: Groovy 3.0.19 on OpenJDK 64-Bit Server VM 17.0.8+7-LTS\n",
      "  Encoding: UTF-8 (UTF-8)\n",
      "  Process: 69734@Eilas-Air.fios-router.home [192.168.1.48]\n",
      "  CPUs: 8 - Mem: 16 GB (250.3 MB) - Swap: 3 GB (1.4 GB)\n",
      "Feb-05 23:50:09.246 [main] DEBUG nextflow.Session - Work-dir: gs://eila-nextflow-bucket/work/ATAC [Mac OS X]\n",
      "Feb-05 23:50:09.268 [main] DEBUG nextflow.executor.ExecutorFactory - Extension executors providers=[GoogleLifeSciencesExecutor, GoogleBatchExecutor]\n",
      "Feb-05 23:50:09.276 [main] DEBUG nextflow.Session - Observer factory: DefaultObserverFactory\n",
      "Feb-05 23:50:09.280 [main] DEBUG nextflow.Session - Observer factory: WaveFactory\n",
      "Feb-05 23:50:09.280 [main] DEBUG io.seqera.wave.plugin.WaveFactory - Detected Fusion enabled -- Enabling bundle project resources -- Disabling upload of remote bin directory\n",
      "Feb-05 23:50:09.293 [main] DEBUG i.s.wave.plugin.config.WaveConfig - Wave strategy not specified - using default: [container, dockerfile, conda, spack]\n",
      "Feb-05 23:50:09.297 [main] DEBUG io.seqera.wave.plugin.WaveClient - Wave config: WaveConfig(enabled:true, endpoint:https://wave.seqera.io, containerConfigUrl:[], tokensCacheMaxDuration:30m, condaOpts:CondaOpts(mambaImage=mambaorg/micromamba:1.5.1; basePackages=conda-forge::procps-ng, commands=null), spackOpts:SpackOpts(basePackages=null, commands=null), strategy:[container, dockerfile, conda, spack], bundleProjectResources:true, buildRepository:null, cacheRepository:null, retryOpts:RetryOpts(delay:450ms, maxDelay:1m 30s, maxAttempts:10, jitter:0.25), httpClientOpts:HttpOpts(), freezeMode:null)\n",
      "Feb-05 23:50:09.394 [main] DEBUG nextflow.Session - Observer factory: TowerFactory\n",
      "Feb-05 23:50:09.430 [main] DEBUG nextflow.cache.CacheFactory - Using Nextflow cache factory: nextflow.cache.DefaultCacheFactory\n",
      "Feb-05 23:50:09.445 [main] DEBUG nextflow.util.CustomThreadPool - Creating default thread pool > poolSize: 9; maxThreads: 1000\n",
      "Feb-05 23:50:09.524 [main] DEBUG nextflow.Session - Session start\n",
      "Feb-05 23:50:09.548 [main] DEBUG io.seqera.tower.plugin.TowerClient - Creating Tower observer -- endpoint=https://api.tower.nf; requestInterval=1s; aliveInterval=1m; maxRetries=5; backOffBase=3; backOffDelay=250\n",
      "Feb-05 23:50:11.616 [main] DEBUG nextflow.script.ScriptRunner - > Launching execution\n",
      "Feb-05 23:50:12.404 [main] INFO  nextflow.script.BaseScript - /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/local_atac_fastqs_files.tsv\n",
      "Feb-05 23:50:12.404 [main] INFO  nextflow.script.BaseScript - eyJ0eXAiOiJKV1QiLCJraWQiOiJXN05OOldMSlQ6SjVSSzpMN1RMOlQ3TDc6M1ZYNjpKRU9VOjY0NFI6VTNJWDo1S1oyOjdaQ0s6RlBUSCIsImFsZyI6IlJTMjU2In0.eyJhY2Nlc3MiOnsic2NvcGUiOlsidmlldyIsImRvd25sb2FkIl0sIm9pZGNfY2xhaW1zIjp7fX0sInRva2VuX3R5cGUiOiJQRVJTT05BTF9BQ0NFU1NfVE9LRU4iLCJpc3MiOiJodHRwczovL3JlcG8tcHJvZC5wcm9kLnNhZ2ViYXNlLm9yZy9hdXRoL3YxIiwiYXVkIjoiMCIsIm5iZiI6MTY5NzgxODkxNiwiaWF0IjoxNjk3ODE4OTE2LCJqdGkiOiI0MTIzIiwic3ViIjoiMzQ3NDU3MSJ9.n0cugYMAVzYZ8mMMc7CjcqqEV7FB-K-oKUDT45SvM6KmbCpb_v3lZ2gg3xXi1tubWH0__wpnbJ7cYrGEtOyt4vXAPkMiEtZFGqS0MFKp_h-mHHH4pnACSQQYEy1zdyqFn9WRE7ipjWg4yYSbq84N7J3gzi8VKmiOAmuIGKaJaS0snkUjtJEVQH-0vZqWs1nKgVRWIZv31Bo8vuDzWpBrXD-zRT_uQW9l_I9fU50o1AqyFhNe_gbty_Z6imb8VT33wMXiDgJh15lG_M70PXU15v3egDXLgw1dGj32kj1nKanfqBVqbMSzgzPAG037p3YH_Q556BLceFBBb5Jaej5JVA\n",
      "Feb-05 23:50:12.448 [main] INFO  nextflow.script.BaseScript - after genome_fasta_ch\n",
      "Feb-05 23:50:12.476 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:zcat` matches labels `zcat` for process with name run_zcat\n",
      "Feb-05 23:50:12.479 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-05 23:50:12.479 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-05 23:50:12.480 [main] DEBUG nextflow.executor.Executor - [warm up] executor > google-batch (fusion enabled)\n",
      "Feb-05 23:50:12.484 [main] DEBUG n.processor.TaskPollingMonitor - Creating task monitor for executor 'google-batch' > capacity: 1000; pollInterval: 10s; dumpInterval: 5m \n",
      "Feb-05 23:50:12.485 [main] DEBUG n.processor.TaskPollingMonitor - >>> barrier register (monitor: google-batch)\n",
      "Feb-05 23:50:12.488 [main] DEBUG nextflow.cloud.google.GoogleOpts - Google auth via application credentials file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/hai-gcp-genomic-beb9207a40b8.json\n",
      "Feb-05 23:50:12.491 [main] DEBUG n.c.google.batch.GoogleBatchExecutor - [GOOGLE BATCH] Executor config=BatchConfig[googleOpts=GoogleOpts(projectId:hai-gcp-genomic, credsFile:/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/hai-gcp-genomic-beb9207a40b8.json, location:us-central1, enableRequesterPaysBuckets:false, httpConnectTimeout:1m, httpReadTimeout:1m, credentials:ServiceAccountCredentials{clientId=111364348320962814542, clientEmail=eila-batch@hai-gcp-genomic.iam.gserviceaccount.com, privateKeyId=beb9207a40b8b2085c7f48c3fcf8c2342180d471, transportFactoryClassName=com.google.auth.oauth2.OAuth2Utils$DefaultHttpTransportFactory, tokenServerUri=https://oauth2.googleapis.com/token, scopes=[https://www.googleapis.com/auth/cloud-platform], defaultScopes=[], serviceAccountUser=null, quotaProjectId=null, lifetime=3600, useJwtAccessWithScope=false, defaultRetriesEnabled=true})\n",
      "Feb-05 23:50:12.507 [main] DEBUG n.c.google.batch.client.BatchClient - [GOOGLE BATCH] Creating service client with config credentials\n",
      "Feb-05 23:50:13.060 [main] INFO  nextflow.script.BaseScript - after run_zcat \n",
      "Feb-05 23:50:13.066 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:gunzip` matches labels `gunzip` for process with name run_whitelist_gunzip\n",
      "Feb-05 23:50:13.067 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-05 23:50:13.067 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-05 23:50:13.069 [main] INFO  nextflow.script.BaseScript - after run_whitelist_gunzip\n",
      "Feb-05 23:50:13.070 [main] INFO  nextflow.script.BaseScript - after load chrompa mapping debug\n",
      "Feb-05 23:50:13.072 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:pool_fragments` matches labels `pool_fragments` for process with name run_add_subpool_prefix_to_fragment_table\n",
      "Feb-05 23:50:13.073 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-05 23:50:13.073 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-05 23:50:13.075 [main] INFO  nextflow.script.BaseScript - before call bgzip\n",
      "Feb-05 23:50:13.076 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:bgzip` matches labels `bgzip` for process with name run_bgzip_on_chromap_fragments_output\n",
      "Feb-05 23:50:13.077 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-05 23:50:13.077 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-05 23:50:13.078 [main] INFO  nextflow.script.BaseScript - after call bgzip\n",
      "Feb-05 23:50:13.078 [main] INFO  nextflow.script.BaseScript - before call run_tabix_chromap with tabix shell script\n",
      "Feb-05 23:50:13.079 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:tabix` matches labels `tabix` for process with name run_tabix_chromap\n",
      "Feb-05 23:50:13.080 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-05 23:50:13.080 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-05 23:50:13.081 [main] INFO  nextflow.script.BaseScript - after call run_tabix_chromap\n",
      "Feb-05 23:50:13.081 [main] INFO  nextflow.script.BaseScript - before call run_merge_chromap_logs\n",
      "Feb-05 23:50:13.082 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:merge_log` matches labels `merge_log` for process with name run_merge_chromap_logs\n",
      "Feb-05 23:50:13.083 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-05 23:50:13.083 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-05 23:50:13.085 [main] INFO  nextflow.script.BaseScript - after call run_merge_chromap_logs\n",
      "Feb-05 23:50:13.085 [main] INFO  nextflow.script.BaseScript - before run_add_subpool_prefix_to_summary_file\n",
      "Feb-05 23:50:13.086 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:pool_dictionary` matches labels `pool_dictionary` for process with name run_process_conversion_to_barcode\n",
      "Feb-05 23:50:13.086 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-05 23:50:13.087 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-05 23:50:13.088 [main] INFO  nextflow.script.BaseScript - after run_add_subpool_prefix_to_summary_file\n",
      "Feb-05 23:50:13.089 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:pool_filter` matches labels `pool_filter` for process with name run_filter_aligned_fragments\n",
      "Feb-05 23:50:13.090 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-05 23:50:13.090 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-05 23:50:13.093 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:tabix_singelston` matches labels `tabix_singelston` for process with name run_tabix_no_singleton\n",
      "Feb-05 23:50:13.094 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-05 23:50:13.094 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-05 23:50:13.095 [main] INFO  nextflow.script.BaseScript - before call run_calculate_tss_enrichment_bulk\n",
      "Feb-05 23:50:13.096 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:tss_bulk` matches labels `tss_bulk` for process with name run_calculate_tss_enrichment_bulk\n",
      "Feb-05 23:50:13.097 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-05 23:50:13.097 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-05 23:50:13.098 [main] INFO  nextflow.script.BaseScript - after call run_calculate_tss_enrichment_bulk\n",
      "Feb-05 23:50:13.100 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:gzip` matches labels `gzip` for process with name run_gzip_on_genes_gtf\n",
      "Feb-05 23:50:13.100 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-05 23:50:13.100 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-05 23:50:13.101 [main] INFO  nextflow.script.BaseScript - after run_gzip_on_genes_gtf\n",
      "Feb-05 23:50:13.102 [main] INFO  nextflow.script.BaseScript - before call run_calculate_tss_enrichment_snapatac2\n",
      "Feb-05 23:50:13.103 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:tss_snapatac2` matches labels `tss_snapatac2` for process with name run_calculate_tss_enrichment_snapatac2\n",
      "Feb-05 23:50:13.104 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-05 23:50:13.104 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-05 23:50:13.105 [main] INFO  nextflow.script.BaseScript - after call run_calculate_tss_enrichment_snapatac2\n",
      "Feb-05 23:50:13.106 [main] INFO  nextflow.script.BaseScript - before run_generate_insert_size_histogram_data\n",
      "Feb-05 23:50:13.107 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:zcat_insert` matches labels `zcat_insert` for process with name run_generate_insert_size_histogram_data\n",
      "Feb-05 23:50:13.108 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-05 23:50:13.108 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-05 23:50:13.109 [main] INFO  nextflow.script.BaseScript - after run_generate_insert_size_histogram_data\n",
      "Feb-05 23:50:13.109 [main] INFO  nextflow.script.BaseScript - before call run_generate_insert_size_plot\n",
      "Feb-05 23:50:13.110 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-05 23:50:13.110 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-05 23:50:13.112 [main] INFO  nextflow.script.BaseScript - after call run_generate_insert_size_plot\n",
      "Feb-05 23:50:13.112 [main] INFO  nextflow.script.BaseScript - before run_barcode_metadata_stats\n",
      "Feb-05 23:50:13.113 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-05 23:50:13.114 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-05 23:50:13.115 [main] INFO  nextflow.script.BaseScript - after run_barcode_metadata_stats\n",
      "Feb-05 23:50:13.115 [main] INFO  nextflow.script.BaseScript - before call run_generate_barcode_rank_plot\n",
      "Feb-05 23:50:13.116 [main] DEBUG nextflow.Session - Workflow process names [dsl2]: run_bgzip_on_chromap_fragments_output, run_seqspec_print, run_calculate_tss_enrichment_snapatac2, run_process_conversion_to_barcode, run_barcode_metadata_stats, run_tabix_chromap, run_calculate_tss_enrichment_bulk, run_generate_insert_size_plot, run_seqspec_test, run_tabix_no_singleton, run_chromap_map_to_idx, run_atac_merge_barcode_metadata, run_filter_aligned_fragments, run_seqspec_modify_rna, run_atac_barcode_rank_plot, run_gzip_on_genes_gtf, run_whitelist_gunzip, run_merge_chromap_logs, run_create_chromap_idx, run_generate_insert_size_histogram_data, run_zcat, run_seqspec_check, run_seqspec_index_rna_kb, run_seqspec_modify_atac, run_chromap_test, run_tabix_filtered_fragments, run_add_subpool_prefix_to_fragment_table, run_downloadFiles\n",
      "Feb-05 23:50:13.559 [main] INFO  io.seqera.tower.plugin.TowerClient - Monitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/3JABk9TjqtQYRd\n",
      "Feb-05 23:50:13.559 [main] DEBUG nextflow.Session - Igniting dataflow network (24)\n",
      "Feb-05 23:50:13.564 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_zcat\n",
      "Feb-05 23:50:13.564 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_whitelist_gunzip\n",
      "Feb-05 23:50:13.565 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_add_subpool_prefix_to_fragment_table\n",
      "Feb-05 23:50:13.565 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_bgzip_on_chromap_fragments_output\n",
      "Feb-05 23:50:13.565 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_tabix_chromap\n",
      "Feb-05 23:50:13.565 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_merge_chromap_logs\n",
      "Feb-05 23:50:13.565 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_process_conversion_to_barcode\n",
      "Feb-05 23:50:13.566 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_filter_aligned_fragments\n",
      "Feb-05 23:50:13.566 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_tabix_no_singleton\n",
      "Feb-05 23:50:13.566 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_calculate_tss_enrichment_bulk\n",
      "Feb-05 23:50:13.566 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_gzip_on_genes_gtf\n",
      "Feb-05 23:50:13.567 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_calculate_tss_enrichment_snapatac2\n",
      "Feb-05 23:50:13.567 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_generate_insert_size_histogram_data\n",
      "Feb-05 23:50:13.567 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_generate_insert_size_plot\n",
      "Feb-05 23:50:13.567 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_barcode_metadata_stats\n",
      "Feb-05 23:50:13.568 [main] DEBUG nextflow.script.ScriptRunner - Parsed script files:\n",
      "  Script_30732e62620d0e07: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_tss.nf\n",
      "  Script_900e20e0f3b521d7: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_atac_generate_insert_size_plot.nf\n",
      "  Script_8584831ea472e1e1: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_barcode_subpool.nf\n",
      "  Script_67e45395c241a45f: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_chromap_utils.nf\n",
      "  Script_9d4639c8cd883c70: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_download_url_files.nf\n",
      "  Script_bc539eb5853c51b8: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_gunzip.nf\n",
      "  Script_280f31ae5ae5ed4d: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_tabix.nf\n",
      "  Script_5df81d0ae421efa0: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_bgzip.nf\n",
      "  Script_e112646bcdb899f0: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_seqspec_utils.nf\n",
      "  Script_8081eedda99eaa43: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_merge_logs.nf\n",
      "  Script_e5c462e2b7b505cd: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nf_wf_fg_singlecell_atac.nf\n",
      "  Script_103349f59593da4c: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_atac_generate_metadata_annoation.nf\n",
      "  Script_48127624ae4a6640: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_zcat.nf\n",
      "  Script_7f43a272cfe3ae1b: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_generate_merged_barcode_metadata.nf\n",
      "  Script_34d6d3540f17eeb9: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_gzip.nf\n",
      "Feb-05 23:50:13.568 [main] DEBUG nextflow.script.ScriptRunner - > Awaiting termination \n",
      "Feb-05 23:50:13.568 [main] DEBUG nextflow.Session - Session await\n",
      "Feb-05 23:50:14.865 [FileTransfer-1] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\n",
      "Feb-05 23:50:14.875 [Actor Thread 6] DEBUG i.s.wave.plugin.config.WaveConfig - Wave strategy not specified - using default: [container, dockerfile, conda, spack]\n",
      "Feb-05 23:50:14.875 [Actor Thread 6] DEBUG io.seqera.wave.plugin.WaveClient - Wave config: WaveConfig(enabled:true, endpoint:https://wave.seqera.io, containerConfigUrl:[], tokensCacheMaxDuration:30m, condaOpts:CondaOpts(mambaImage=mambaorg/micromamba:1.5.1; basePackages=conda-forge::procps-ng, commands=null), spackOpts:SpackOpts(basePackages=null, commands=null), strategy:[container, dockerfile, conda, spack], bundleProjectResources:true, buildRepository:null, cacheRepository:null, retryOpts:RetryOpts(delay:450ms, maxDelay:1m 30s, maxAttempts:10, jitter:0.25), httpClientOpts:HttpOpts(), freezeMode:null)\n",
      "Feb-05 23:50:14.879 [Actor Thread 6] DEBUG io.seqera.wave.plugin.WaveClient - Wave assets bundle bin resources: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/bin\n",
      "Feb-05 23:50:14.949 [FileTransfer-2] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/genes.gtf at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/eb/6412854cc6c199859cdfbe280f86c9/genes.gtf\n",
      "Feb-05 23:50:15.011 [Actor Thread 6] DEBUG io.seqera.wave.plugin.WaveClient - Wave request container config: https://fusionfs.seqera.io/releases/v2.2-amd64.json\n",
      "Feb-05 23:50:15.265 [Actor Thread 6] DEBUG io.seqera.wave.plugin.WaveClient - Wave container config response: [200] {\n",
      "  \"entrypoint\": [ \"/usr/bin/fusion\" ],\n",
      "  \"layers\": [\n",
      "    {\n",
      "      \"location\": \"https://fusionfs.seqera.io/releases/pkg/2/2/12/fusion-amd64.tar.gz\",\n",
      "      \"gzipDigest\": \"sha256:6ccfd8f431db2874e82e27d61b6d31b95d7dddce7e810c5f8c95438a0d03c066\",\n",
      "      \"gzipSize\": 30262575,\n",
      "      \"tarDigest\": \"sha256:60391bebe0cc5d48b0875854f0c93c0d0adc25b5007fdb1b2efc2a431cfeef6e\",\n",
      "      \"skipHashing\": true\n",
      "    }\n",
      "  ]\n",
      "}\n",
      "\n",
      "Feb-05 23:50:15.313 [Actor Thread 2] DEBUG io.seqera.wave.plugin.WaveClient - Wave request: https://wave.seqera.io/container-token; attempt=1 - request: SubmitContainerTokenRequest(towerAccessToken:eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz, towerRefreshToken:null, towerWorkspaceId:247507587970818, towerEndpoint:https://api.tower.nf, workflowId:null, containerImage:eilalan/utils_env:latest, containerFile:null, containerConfig:ContainerConfig(entrypoint:[/usr/bin/fusion], cmd:null, env:null, workingDir:null, layers:[ContainerLayer[location=data:H4sIAAAAAAAA/+197X/b...; tarDigest=sha256:7d25d71a43afd432a4768f9b3cea676f18b157cb5ee27be33af026e49314e37e; gzipDigest=sha256:7d45259f3651182da56a100dbbf8690d65a6b4ec26573686fd40800b29867435; gzipSize=12115], ContainerLayer[location=https://fusionfs.seqera.io/releases/pkg/2/2/12/fusion-amd64.tar.gz; tarDigest=sha256:60391bebe0cc5d48b0875854f0c93c0d0adc25b5007fdb1b2efc2a431cfeef6e; gzipDigest=sha256:6ccfd8f431db2874e82e27d61b6d31b95d7dddce7e810c5f8c95438a0d03c066; gzipSize=30262575]]), condaFile:null, spackFile:null, containerPlatform:linux/amd64, buildRepository:null, cacheRepository:null, timestamp:2024-02-05T23:50:15.312909-05:00, fingerprint:d8e31a4bb001195f40b75eb2fbd5c9ac, freeze:false, format:null, dryRun:false)\n",
      "Feb-05 23:50:16.003 [Actor Thread 2] DEBUG io.seqera.wave.plugin.WaveClient - Wave response: statusCode=200; body={\"containerToken\":\"acd26b5206f6\",\"targetImage\":\"wave.seqera.io/wt/acd26b5206f6/eilalan/utils_env:latest\",\"expiration\":\"2024-02-07T16:50:16.014017776Z\",\"containerImage\":\"docker.io/eilalan/utils_env:latest\"}\n",
      "Feb-05 23:50:17.494 [FileTransfer-6] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/dc/65201d13e12dffd8779bbc935604a6/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\n",
      "Feb-05 23:50:17.507 [FileTransfer-3] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/6f/e3ee295a7f5364c39365c48e8bd0b0/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\n",
      "Feb-05 23:50:17.511 [FileTransfer-4] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/c5/cc9149d8d21bb423337888a9dc77ea/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\n",
      "Feb-05 23:50:17.574 [FileTransfer-7] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/d5/12b6921f983d2326fa1e91570a96e7/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\n",
      "Feb-05 23:50:17.574 [FileTransfer-1] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/na1.na at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/38/b73884d031cf00d6089d843ba0f965/na1.na\n",
      "Feb-05 23:50:17.574 [FileTransfer-9] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/na.na at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/f4/bab8ffeff41eecffb2ddcf013b4121/na.na\n",
      "Feb-05 23:50:17.578 [FileTransfer-2] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/chromap_out_debug/ATAC_BMMC-single-donor.atac.filter.fragments.tsv at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/12/a7aa6d9e76c7234da48d36951f96c6/ATAC_BMMC-single-donor.atac.filter.fragments.tsv\n",
      "Feb-05 23:50:17.578 [FileTransfer-10] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/sai_192_whitelist.txt at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/09/7ba70bbeac00bcd6547ad0c09655c7/sai_192_whitelist.txt\n",
      "Feb-05 23:50:17.580 [FileTransfer-8] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/2d/0e59fd7f9797eaf684de4abc52eb46/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\n",
      "Feb-05 23:50:17.585 [FileTransfer-5] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/4a/7b5d74b3a4903977d7316c5f02ef84/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\n",
      "Feb-05 23:50:17.587 [FileTransfer-6] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/chromap_out_debug/BMMC-single-donor.atac.align.barcode.summary.csv at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/83/230e2c5cb965e17c572d46066ce07f/BMMC-single-donor.atac.align.barcode.summary.csv\n",
      "Feb-05 23:50:18.295 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_gzip_on_genes_gtf (1)` submitted > job=nf-cdbca4ab-1707195016277; uid=nf-cdbca4ab-170719-399feb14-b546-40f70; work-dir=gs://eila-nextflow-bucket/work/ATAC/cd/bca4ab0c4ebc99a09381bbd6695e48\n",
      "Feb-05 23:50:18.295 [Task submitter] INFO  nextflow.Session - [cd/bca4ab] Submitted process > run_gzip_on_genes_gtf (1)\n",
      "Feb-05 23:50:18.720 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_zcat` - adding local volume as fusion scratch: DiskResource(request:375 GB, type:local-ssd)\n",
      "Feb-05 23:50:19.074 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_zcat` submitted > job=nf-21b2f514-1707195016324; uid=nf-21b2f514-170719-5cf900fc-2dfa-4e080; work-dir=gs://eila-nextflow-bucket/work/ATAC/21/b2f514dbef62ee570adb96cfc81667\n",
      "Feb-05 23:50:19.075 [Task submitter] INFO  nextflow.Session - [21/b2f514] Submitted process > run_zcat\n",
      "Feb-05 23:50:19.510 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_whitelist_gunzip (1)` - adding local volume as fusion scratch: DiskResource(request:375 GB, type:local-ssd)\n",
      "Feb-05 23:50:19.820 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_whitelist_gunzip (1)` submitted > job=nf-bf49817f-1707195018436; uid=nf-bf49817f-170719-bc070fe0-6f29-4a220; work-dir=gs://eila-nextflow-bucket/work/ATAC/bf/49817f0460a8586c3ad604c4ae8019\n",
      "Feb-05 23:50:19.820 [Task submitter] INFO  nextflow.Session - [bf/49817f] Submitted process > run_whitelist_gunzip (1)\n",
      "Feb-05 23:50:20.240 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_add_subpool_prefix_to_fragment_table (1)` - adding local volume as fusion scratch: DiskResource(request:375 GB, type:local-ssd)\n",
      "Feb-05 23:50:20.668 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_add_subpool_prefix_to_fragment_table (1)` submitted > job=nf-29b7b785-1707195018634; uid=nf-29b7b785-170719-73118873-e3c8-4c9c0; work-dir=gs://eila-nextflow-bucket/work/ATAC/29/b7b785aaaa075ee3bc1375da8789da\n",
      "Feb-05 23:50:20.668 [Task submitter] INFO  nextflow.Session - [29/b7b785] Submitted process > run_add_subpool_prefix_to_fragment_table (1)\n",
      "Feb-05 23:52:33.309 [Task monitor] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_whitelist_gunzip (1)` - terminated job=nf-bf49817f-1707195018436; state=SUCCEEDED\n",
      "Feb-05 23:52:33.570 [Task monitor] DEBUG n.processor.TaskPollingMonitor - Task completed > TaskHandler[id: 3; name: run_whitelist_gunzip (1); status: COMPLETED; exit: 0; error: -; workDir: gs://eila-nextflow-bucket/work/ATAC/bf/49817f0460a8586c3ad604c4ae8019]\n",
      "Feb-05 23:52:52.989 [Task monitor] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_add_subpool_prefix_to_fragment_table (1)` - terminated job=nf-29b7b785-1707195018634; state=SUCCEEDED\n",
      "Feb-05 23:52:53.201 [Task monitor] DEBUG n.processor.TaskPollingMonitor - Task completed > TaskHandler[id: 4; name: run_add_subpool_prefix_to_fragment_table (1); status: COMPLETED; exit: 0; error: -; workDir: gs://eila-nextflow-bucket/work/ATAC/29/b7b785aaaa075ee3bc1375da8789da]\n",
      "Feb-05 23:52:53.374 [Actor Thread 6] DEBUG io.seqera.wave.plugin.WaveClient - Wave request container config: https://fusionfs.seqera.io/releases/v2.2-amd64.json\n",
      "Feb-05 23:52:53.436 [Actor Thread 6] DEBUG io.seqera.wave.plugin.WaveClient - Wave container config response: [200] {\n",
      "  \"entrypoint\": [ \"/usr/bin/fusion\" ],\n",
      "  \"layers\": [\n",
      "    {\n",
      "      \"location\": \"https://fusionfs.seqera.io/releases/pkg/2/2/12/fusion-amd64.tar.gz\",\n",
      "      \"gzipDigest\": \"sha256:6ccfd8f431db2874e82e27d61b6d31b95d7dddce7e810c5f8c95438a0d03c066\",\n",
      "      \"gzipSize\": 30262575,\n",
      "      \"tarDigest\": \"sha256:60391bebe0cc5d48b0875854f0c93c0d0adc25b5007fdb1b2efc2a431cfeef6e\",\n",
      "      \"skipHashing\": true\n",
      "    }\n",
      "  ]\n",
      "}\n",
      "\n",
      "Feb-05 23:52:53.591 [FileTransfer-3] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/chromap_out_debug/BMMC-single-donor.atac.align.k4.hg38.log.txt at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/ad/37cb0426d11a8313b27a0af2761113/BMMC-single-donor.atac.align.k4.hg38.log.txt\n",
      "Feb-05 23:52:54.523 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_bgzip_on_chromap_fragments_output (1)` submitted > job=nf-cb0b8d42-1707195173722; uid=nf-cb0b8d42-170719-1837b1f6-978c-4bba0; work-dir=gs://eila-nextflow-bucket/work/ATAC/cb/0b8d42509cce9c31994602d91e9489\n",
      "Feb-05 23:52:54.523 [Task submitter] INFO  nextflow.Session - [cb/0b8d42] Submitted process > run_bgzip_on_chromap_fragments_output (1)\n",
      "Feb-05 23:52:55.143 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_merge_chromap_logs (1)` - adding local volume as fusion scratch: DiskResource(request:375 GB, type:local-ssd)\n",
      "Feb-05 23:52:55.511 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_merge_chromap_logs (1)` submitted > job=nf-3049232d-1707195173926; uid=nf-3049232d-170719-c7446a47-9911-4a7a0; work-dir=gs://eila-nextflow-bucket/work/ATAC/30/49232d3945e4ccc9a6d9edbbc4969b\n",
      "Feb-05 23:52:55.512 [Task submitter] INFO  nextflow.Session - [30/49232d] Submitted process > run_merge_chromap_logs (1)\n",
      "Feb-05 23:52:55.964 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_process_conversion_to_barcode (1)` - adding local volume as fusion scratch: DiskResource(request:375 GB, type:local-ssd)\n",
      "Feb-05 23:52:56.243 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_process_conversion_to_barcode (1)` submitted > job=nf-091f1698-1707195174595; uid=nf-091f1698-170719-309fa42b-2a62-49900; work-dir=gs://eila-nextflow-bucket/work/ATAC/09/1f169862916d3ba4c2194592ef4dcb\n",
      "Feb-05 23:52:56.244 [Task submitter] INFO  nextflow.Session - [09/1f1698] Submitted process > run_process_conversion_to_barcode (1)\n",
      "Feb-05 23:53:02.758 [Task monitor] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_gzip_on_genes_gtf (1)` - terminated job=nf-cdbca4ab-1707195016277; state=SUCCEEDED\n",
      "Feb-05 23:53:03.577 [Task monitor] DEBUG n.processor.TaskPollingMonitor - Task completed > TaskHandler[id: 2; name: run_gzip_on_genes_gtf (1); status: COMPLETED; exit: 0; error: -; workDir: gs://eila-nextflow-bucket/work/ATAC/cd/bca4ab0c4ebc99a09381bbd6695e48]\n",
      "Feb-05 23:53:42.750 [Task monitor] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_zcat` - terminated job=nf-21b2f514-1707195016324; state=SUCCEEDED\n",
      "Feb-05 23:53:43.058 [Task monitor] DEBUG n.processor.TaskPollingMonitor - Task completed > TaskHandler[id: 1; name: run_zcat; status: COMPLETED; exit: 0; error: -; workDir: gs://eila-nextflow-bucket/work/ATAC/21/b2f514dbef62ee570adb96cfc81667]\n",
      "Feb-05 23:55:12.887 [Task monitor] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_merge_chromap_logs (1)` - terminated job=nf-3049232d-1707195173926; state=SUCCEEDED\n",
      "Feb-05 23:55:13.105 [Task monitor] DEBUG n.processor.TaskPollingMonitor - Task completed > TaskHandler[id: 7; name: run_merge_chromap_logs (1); status: COMPLETED; exit: 0; error: -; workDir: gs://eila-nextflow-bucket/work/ATAC/30/49232d3945e4ccc9a6d9edbbc4969b]\n",
      "Feb-05 23:55:13.677 [Task monitor] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_process_conversion_to_barcode (1)` - terminated job=nf-091f1698-1707195174595; state=SUCCEEDED\n",
      "Feb-05 23:55:13.820 [Task monitor] DEBUG n.processor.TaskPollingMonitor - Task completed > TaskHandler[id: 6; name: run_process_conversion_to_barcode (1); status: COMPLETED; exit: 0; error: -; workDir: gs://eila-nextflow-bucket/work/ATAC/09/1f169862916d3ba4c2194592ef4dcb]\n",
      "Feb-05 23:55:13.951 [Actor Thread 5] DEBUG io.seqera.wave.plugin.WaveClient - Wave request container config: https://fusionfs.seqera.io/releases/v2.2-amd64.json\n",
      "Feb-05 23:55:14.034 [Actor Thread 5] DEBUG io.seqera.wave.plugin.WaveClient - Wave container config response: [200] {\n",
      "  \"entrypoint\": [ \"/usr/bin/fusion\" ],\n",
      "  \"layers\": [\n",
      "    {\n",
      "      \"location\": \"https://fusionfs.seqera.io/releases/pkg/2/2/12/fusion-amd64.tar.gz\",\n",
      "      \"gzipDigest\": \"sha256:6ccfd8f431db2874e82e27d61b6d31b95d7dddce7e810c5f8c95438a0d03c066\",\n",
      "      \"gzipSize\": 30262575,\n",
      "      \"tarDigest\": \"sha256:60391bebe0cc5d48b0875854f0c93c0d0adc25b5007fdb1b2efc2a431cfeef6e\",\n",
      "      \"skipHashing\": true\n",
      "    }\n",
      "  ]\n",
      "}\n",
      "\n",
      "Feb-05 23:55:15.234 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_filter_aligned_fragments (1)` submitted > job=nf-8578bb28-1707195314457; uid=nf-8578bb28-170719-40b8e539-0719-40030; work-dir=gs://eila-nextflow-bucket/work/ATAC/85/78bb28746928104b61b1a4d12a67a8\n",
      "Feb-05 23:55:15.235 [Task submitter] INFO  nextflow.Session - [85/78bb28] Submitted process > run_filter_aligned_fragments (1)\n",
      "Feb-05 23:55:22.605 [Task monitor] DEBUG n.processor.TaskPollingMonitor - !! executor google-batch > tasks to be completed: 2 -- submitted tasks are shown below\n",
      "~> TaskHandler[id: 5; name: run_bgzip_on_chromap_fragments_output (1); status: RUNNING; exit: -; error: -; workDir: gs://eila-nextflow-bucket/work/ATAC/cb/0b8d42509cce9c31994602d91e9489]\n",
      "~> TaskHandler[id: 8; name: run_filter_aligned_fragments (1); status: SUBMITTED; exit: -; error: -; workDir: gs://eila-nextflow-bucket/work/ATAC/85/78bb28746928104b61b1a4d12a67a8]\n",
      "Feb-05 23:57:52.882 [Task monitor] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_filter_aligned_fragments (1)` - terminated job=nf-8578bb28-1707195314457; state=SUCCEEDED\n",
      "Feb-05 23:57:53.121 [Task monitor] DEBUG n.processor.TaskPollingMonitor - Task completed > TaskHandler[id: 8; name: run_filter_aligned_fragments (1); status: COMPLETED; exit: 0; error: -; workDir: gs://eila-nextflow-bucket/work/ATAC/85/78bb28746928104b61b1a4d12a67a8]\n",
      "Feb-05 23:57:54.110 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_tabix_no_singleton (1)` - adding local volume as fusion scratch: DiskResource(request:375 GB, type:local-ssd)\n",
      "Feb-05 23:57:54.549 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_tabix_no_singleton (1)` submitted > job=nf-1379788c-1707195473630; uid=nf-1379788c-170719-c95b95a3-2736-446a0; work-dir=gs://eila-nextflow-bucket/work/ATAC/13/79788c0e396136029f5800e5795f8c\n",
      "Feb-05 23:57:54.550 [Task submitter] INFO  nextflow.Session - [13/79788c] Submitted process > run_tabix_no_singleton (1)\n",
      "Feb-06 00:00:22.697 [Task monitor] DEBUG n.processor.TaskPollingMonitor - !! executor google-batch > tasks to be completed: 2 -- submitted tasks are shown below\n",
      "~> TaskHandler[id: 5; name: run_bgzip_on_chromap_fragments_output (1); status: RUNNING; exit: -; error: -; workDir: gs://eila-nextflow-bucket/work/ATAC/cb/0b8d42509cce9c31994602d91e9489]\n",
      "~> TaskHandler[id: 9; name: run_tabix_no_singleton (1); status: SUBMITTED; exit: -; error: -; workDir: gs://eila-nextflow-bucket/work/ATAC/13/79788c0e396136029f5800e5795f8c]\n",
      "Feb-06 00:05:22.800 [Task monitor] DEBUG n.processor.TaskPollingMonitor - !! executor google-batch > tasks to be completed: 2 -- submitted tasks are shown below\n",
      "~> TaskHandler[id: 5; name: run_bgzip_on_chromap_fragments_output (1); status: RUNNING; exit: -; error: -; workDir: gs://eila-nextflow-bucket/work/ATAC/cb/0b8d42509cce9c31994602d91e9489]\n",
      "~> TaskHandler[id: 9; name: run_tabix_no_singleton (1); status: RUNNING; exit: -; error: -; workDir: gs://eila-nextflow-bucket/work/ATAC/13/79788c0e396136029f5800e5795f8c]\n",
      "Feb-06 00:05:52.950 [Task monitor] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_bgzip_on_chromap_fragments_output (1)` - terminated job=nf-cb0b8d42-1707195173722; state=SUCCEEDED\n",
      "Feb-06 00:05:53.220 [Task monitor] DEBUG n.processor.TaskPollingMonitor - Task completed > TaskHandler[id: 5; name: run_bgzip_on_chromap_fragments_output (1); status: COMPLETED; exit: 0; error: -; workDir: gs://eila-nextflow-bucket/work/ATAC/cb/0b8d42509cce9c31994602d91e9489]\n",
      "Feb-06 00:05:53.295 [Actor Thread 6] DEBUG nextflow.processor.TaskProcessor - Handling unexpected condition for\n",
      "  task: name=run_generate_insert_size_histogram_data (1); work-dir=null\n",
      "  error [nextflow.exception.ProcessUnrecoverableException]: Not a valid path value: 'generate_insert_size_histogram_data.sh'\n",
      "Feb-06 00:05:53.304 [Actor Thread 6] ERROR nextflow.processor.TaskProcessor - Error executing process > 'run_generate_insert_size_histogram_data (1)'\n",
      "\n",
      "Caused by:\n",
      "  Not a valid path value: 'generate_insert_size_histogram_data.sh'\n",
      "\n",
      "\n",
      "Tip: when you have fixed the problem you can continue the execution adding the option `-resume` to the run command line\n",
      "Feb-06 00:05:53.306 [Actor Thread 6] DEBUG nextflow.Session - Session aborted -- Cause: Not a valid path value: 'generate_insert_size_histogram_data.sh'\n",
      "Feb-06 00:05:53.319 [Actor Thread 6] DEBUG nextflow.Session - The following nodes are still active:\n",
      "[process] run_tabix_chromap\n",
      "  status=ACTIVE\n",
      "  port 0: (value) bound ; channel: tabix_script\n",
      "  port 1: (queue) OPEN  ; channel: tabix_fragments\n",
      "  port 2: (cntrl) -     ; channel: $\n",
      "\n",
      "[process] run_calculate_tss_enrichment_bulk\n",
      "  status=ACTIVE\n",
      "  port 0: (queue) OPEN  ; channel: -\n",
      "  port 1: (value) bound ; channel: calculation_script\n",
      "  port 2: (queue) OPEN  ; channel: fragments\n",
      "  port 3: (queue) OPEN  ; channel: tbi_fragments\n",
      "  port 4: (value) bound ; channel: regions_file\n",
      "  port 5: (value) bound ; channel: tss_bases_flank\n",
      "  port 6: (value) bound ; channel: tss_col_strand_info\n",
      "  port 7: (value) bound ; channel: smoothing_window_size\n",
      "  port 8: (value) bound ; channel: CPUS_TO_USE\n",
      "  port 9: (cntrl) -     ; channel: $\n",
      "\n",
      "[process] run_calculate_tss_enrichment_snapatac2\n",
      "  status=ACTIVE\n",
      "  port 0: (queue) OPEN  ; channel: -\n",
      "  port 1: (value) bound ; channel: calculation_script\n",
      "  port 2: (queue) OPEN  ; channel: tbi_fragments\n",
      "  port 3: (queue) OPEN  ; channel: genes_gtf_gzip_file_out\n",
      "  port 4: (value) bound ; channel: min_frag_cutoff\n",
      "  port 5: (cntrl) -     ; channel: $\n",
      "\n",
      "[process] run_generate_insert_size_histogram_data\n",
      "  status=ACTIVE\n",
      "  port 0: (value) bound ; channel: script_name\n",
      "  port 1: (queue) OPEN  ; channel: in_chromap_bzip_fragments_tsv\n",
      "  port 2: (cntrl) -     ; channel: $\n",
      "\n",
      "[process] run_generate_insert_size_plot\n",
      "  status=ACTIVE\n",
      "  port 0: (value) bound ; channel: plot_script\n",
      "  port 1: (queue) OPEN  ; channel: histogram_file\n",
      "  port 2: (value) bound ; channel: pkr_id\n",
      "  port 3: (cntrl) -     ; channel: $\n",
      "\n",
      "[process] run_barcode_metadata_stats\n",
      "  status=ACTIVE\n",
      "  port 0: (queue) OPEN  ; channel: temp_dict_conversion\n",
      "  port 1: (queue) OPEN  ; channel: snapatac_tss_fragments_stats_out\n",
      "  port 2: (cntrl) -     ; channel: $\n",
      "\n",
      "Feb-06 00:05:53.320 [main] DEBUG nextflow.Session - Session await > all processes finished\n",
      "Feb-06 00:05:53.320 [main] DEBUG nextflow.Session - Session await > all barriers passed\n",
      "Feb-06 00:05:53.325 [main] WARN  n.processor.TaskPollingMonitor - Killing running tasks (1)\n",
      "Feb-06 00:05:53.340 [main] DEBUG n.trace.WorkflowStatsObserver - Workflow completed > WorkflowStats[succeededCount=7; failedCount=0; ignoredCount=0; cachedCount=0; pendingCount=0; submittedCount=0; runningCount=1; retriesCount=0; abortedCount=1; succeedDuration=2m 18s; failedDuration=0ms; cachedDuration=0ms;loadCpus=1; loadMemory=0; peakRunning=3; peakCpus=3; peakMemory=0; ]\n",
      "Feb-06 00:05:53.397 [Actor Thread 4] DEBUG io.seqera.wave.plugin.WaveClient - Wave request container config: https://fusionfs.seqera.io/releases/v2.2-amd64.json\n",
      "Feb-06 00:05:53.497 [Actor Thread 4] DEBUG io.seqera.wave.plugin.WaveClient - Wave container config response: [200] {\n",
      "  \"entrypoint\": [ \"/usr/bin/fusion\" ],\n",
      "  \"layers\": [\n",
      "    {\n",
      "      \"location\": \"https://fusionfs.seqera.io/releases/pkg/2/2/12/fusion-amd64.tar.gz\",\n",
      "      \"gzipDigest\": \"sha256:6ccfd8f431db2874e82e27d61b6d31b95d7dddce7e810c5f8c95438a0d03c066\",\n",
      "      \"gzipSize\": 30262575,\n",
      "      \"tarDigest\": \"sha256:60391bebe0cc5d48b0875854f0c93c0d0adc25b5007fdb1b2efc2a431cfeef6e\",\n",
      "      \"skipHashing\": true\n",
      "    }\n",
      "  ]\n",
      "}\n",
      "\n",
      "Feb-06 00:05:53.502 [Actor Thread 4] DEBUG nextflow.processor.TaskProcessor - Handling unexpected condition for\n",
      "  task: name=run_tabix_chromap (1); work-dir=null\n",
      "  error [java.lang.reflect.UndeclaredThrowableException]: java.lang.reflect.UndeclaredThrowableException\n",
      "Feb-06 00:05:53.828 [Task monitor] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_tabix_no_singleton (1)` - terminated job=nf-1379788c-1707195473630; state=DELETION_IN_PROGRESS\n",
      "Feb-06 00:05:53.881 [Task monitor] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Cannot read exit status for task: `run_tabix_no_singleton (1)` - gs://eila-nextflow-bucket/work/ATAC/13/79788c0e396136029f5800e5795f8c/.exitcode\n",
      "Feb-06 00:05:53.882 [Task monitor] DEBUG n.processor.TaskPollingMonitor - Task completed > TaskHandler[id: 9; name: run_tabix_no_singleton (1); status: COMPLETED; exit: -; error: -; workDir: gs://eila-nextflow-bucket/work/ATAC/13/79788c0e396136029f5800e5795f8c]\n",
      "Feb-06 00:05:53.883 [Task monitor] DEBUG nextflow.processor.TaskProcessor - Handling unexpected condition for\n",
      "  task: name=run_tabix_no_singleton (1); work-dir=gs://eila-nextflow-bucket/work/ATAC/13/79788c0e396136029f5800e5795f8c\n",
      "  error [nextflow.exception.ProcessFailedException]: Process `run_tabix_no_singleton (1)` terminated for an unknown reason -- Likely it has been terminated by the external system\n",
      "Feb-06 00:05:54.008 [main] DEBUG nextflow.cache.CacheDB - Closing CacheDB done\n",
      "Feb-06 00:05:54.008 [main] INFO  org.pf4j.AbstractPluginManager - Stop plugin 'nf-wave@1.0.1'\n",
      "Feb-06 00:05:54.008 [main] DEBUG nextflow.plugin.BasePlugin - Plugin stopped nf-wave\n",
      "Feb-06 00:05:54.008 [main] INFO  org.pf4j.AbstractPluginManager - Stop plugin 'nf-tower@1.6.3'\n",
      "Feb-06 00:05:54.008 [main] DEBUG nextflow.plugin.BasePlugin - Plugin stopped nf-tower\n",
      "Feb-06 00:05:54.008 [main] INFO  org.pf4j.AbstractPluginManager - Stop plugin 'nf-google@1.8.3'\n",
      "Feb-06 00:05:54.008 [main] DEBUG nextflow.plugin.BasePlugin - Plugin stopped nf-google\n",
      "Feb-06 00:05:54.027 [Task monitor] DEBUG n.processor.TaskPollingMonitor - <<< barrier arrives (monitor: google-batch) - terminating tasks monitor poll loop\n",
      "Feb-06 00:05:54.058 [main] DEBUG nextflow.script.ScriptRunner - > Execution complete -- Goodbye\n"
     ]
    }
   ],
   "source": [
    "!cat .nextflow.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 2485,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2486,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2487,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !ls ../../nf_data/test_atac_pipeline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2488,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !gsutil cp gs://eila-nextflow-bucket/test_atac_pipeline/sai_192_whitelist.txt ../../nf_data/test_atac_pipeline/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
