{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 190,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# start docker deamon in terminal\n",
    "# !sudo docker login\n",
    "# OR - becuase the jupyther with allow root\n",
    "# !docker login"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "TODO:\n",
    "\n",
    "when running with cloud. \n",
    "if you dont need to keep the execution logs, change the cloud settings to keep it for a short while\n",
    "Check that the cache if you need to set the cache to false (especially in plot process)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 191,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 192,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# # in the command line. same env\n",
    "# export GOOGLE_APPLICATION_CREDENTIALS=\"/home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/hai-gcp-genomic-beb9207a40b8.json.json\"\n",
    "# gcloud auth activate-service-account eila-batch@hai-gcp-genomic.iam.gserviceaccount.com --key-file=$GOOGLE_APPLICATION_CREDENTIALS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  Credentialed Accounts\n",
      "ACTIVE  ACCOUNT\n",
      "*       eila@stanford.edu\n",
      "\n",
      "To set the active account, run:\n",
      "    $ gcloud config set account `ACCOUNT`\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'hai-gcp-genomic'"
      ]
     },
     "execution_count": 193,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "!gcloud auth list\n",
    "project_name = !gcloud config get-value project\n",
    "project_name = project_name[0]\n",
    "project_name\n",
    "# gcloud config set account\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 194,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "os.environ[\"TOWER_ACCESS_TOKEN\"] = 'eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz'\n",
    "os.environ[\"TOWER_WORKSPACE_ID\"]='247507587970818'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz\n",
      "247507587970818\n"
     ]
    }
   ],
   "source": [
    "!echo $TOWER_ACCESS_TOKEN\n",
    "!echo $TOWER_WORKSPACE_ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 196,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/bin'"
      ]
     },
     "execution_count": 197,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# !chmod +x ./bin/scrna_qc_metrics.py\n",
    "import os\n",
    "bin_fld = os.path.join(os.getcwd(),\"bin\")\n",
    "bin_fld"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "!chmod +x {os.path.join(bin_fld, \"scrna_plot_atac_qc_metrics.R\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"run_tabix.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"plot_insert_size_hist.py\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"generate_barcode_metadata.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"create_sample_filtered_fragment_file.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"compute_tss_script.py\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"barcode_rank_functions.R\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"atac_barcode_and_subpool.sh\")}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# # TODO: For a new user: run these commands with your service account for the execution. I am using eila-batch@...\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/iam.serviceAccountUser\"\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/batch.jobsEditor\"\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/logging.viewer\"\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !nextflow -self-update"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  Version: 23.10.1 build 5891\n",
      "  Created: 12-01-2024 22:01 UTC (14:01 PDT)\n",
      "  System: Mac OS X 14.2.1\n",
      "  Runtime: Groovy 3.0.19 on OpenJDK 64-Bit Server VM 17.0.8+7-LTS\n",
      "  Encoding: UTF-8 (UTF-8)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!nextflow info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 202,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "\n",
    "# Get the current working directory\n",
    "current_directory = os.getcwd()\n",
    "\n",
    "# Navigate two folders above\n",
    "two_folders_above = os.path.abspath(os.path.join(current_directory, '..', '..'))\n",
    "\n",
    "nf_data_base_path = os.path.join(two_folders_above,'nf_data')\n",
    "# Print the result\n",
    "print(nf_data_base_path)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 203,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data'"
      ]
     },
     "execution_count": 203,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nf_data_base_path\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 204,
   "metadata": {},
   "outputs": [],
   "source": [
    "nf_data_project=\"test_atac_pipeline\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 205,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/local_atac_fastqs_files.tsv'"
      ]
     },
     "execution_count": 205,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.path.exists(os.path.join(nf_data_base_path,nf_data_project,\"local_atac_fastqs_files.tsv\"))\n",
    "os.path.join(nf_data_base_path,nf_data_project,\"local_atac_fastqs_files.tsv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 206,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "\n",
    "# Write the configuration file\n",
    "with open(\"nextflow_atac.config\", \"w\") as config_file:\n",
    "    config_file.write(f\"\"\"\n",
    "manifest {{\n",
    "  description = 'A nexflow pipeline for fg single cell atac' \n",
    "  mainScript = 'nf_wf_fg_singlecell_atac.nf'\n",
    "}}\n",
    "\n",
    "profiles {{\n",
    "  google_batch_profile {{\n",
    "    workDir = 'gs://eila-nextflow-bucket/work/ATAC'\n",
    "    includeConfig 'conf/docker-google-batch.config'\n",
    "    docker.enabled = true\n",
    "    wave.enabled = true\n",
    "    fusion.enabled = true\n",
    "    process.scratch = false\n",
    "    executor = 'google-batch'\n",
    "    docker.envWhitelist=['ENV_SYNAPSE_TOKEN','TOWER_ACCESS_TOKEN']\n",
    "    google.project = '{project_name}'\n",
    "    google.location = 'us-central1'\n",
    "    google.batch.spot = true\n",
    "    google.batch.serviceAccountEmail='eila-batch@hai-gcp-genomic.iam.gserviceaccount.com'\n",
    "  }}\n",
    "}}\n",
    "\n",
    "// Define ENV_SYNAPSE_TOKEN as a parameter\n",
    "params.FASTQS_SPEC_CH = '{nf_data_base_path}/{nf_data_project}/local_atac_fastqs_files.tsv'\n",
    "// CHROMAP\n",
    "// 6a\n",
    "params.CHROMAP_IDX='gs://eila-nextflow-bucket/test_atac_pipeline/ref.index'\n",
    "// 6b:\n",
    "// FASTA file\n",
    "params.CHROMAP_READ_LENGTH = 2000\n",
    "params.CHROMAP_BC_ERROR_THRESHOLD = 2\n",
    "params.CHROMAP_BC_PROBABILITY_THRESHOLD = 0.9\n",
    "params.CHROMAP_READ_FORMAT = 'bc:0:-1,r1:0:-1,r2:0:-1'\n",
    "params.CHROMAP_DROP_REPETITIVE_READS = 4\n",
    "params.CHROMAP_QUALITY_THRESHOLD = 0\n",
    "\n",
    "// CHROMAP DEBUG: load output files \n",
    "params.CHROMAP_GENOME_REFERENCE_FASTA='{nf_data_base_path}/{nf_data_project}/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz'\n",
    "params.chromap_filter_fragments_tsv ='{nf_data_base_path}/{nf_data_project}/barcode_summary_csv.tsv'\n",
    "params.barcode_summary_csv ='{nf_data_base_path}/{nf_data_project}/barcode_summary_csv.csv'\n",
    "\n",
    "// Other steps\n",
    "\n",
    "params.GENOME_GZ_GTF='http://ftp.ensembl.org/pub/release-109/gtf/homo_sapiens/Homo_sapiens.GRCh38.109.gtf.gz'\n",
    "params.ENV_SYNAPSE_TOKEN = \"nextflow secrets get SYNAPSE\".execute().text.trim()\n",
    "\n",
    "\n",
    "params.MAPQ_THRESHOLD = false\n",
    "params.BARCODE_TRANSLATE = 'path'\n",
    "//barcode_conversion_dict_file\n",
    "params.BARCODE_CONVERSION_DICT_FILE = '{nf_data_base_path}/nf_wf_fg_singlecell/na.na'\n",
    "params.SUBPOOL='  '\n",
    "params.ATAC_BARCODE_AND_POOL='atac_barcode_and_subpool.sh'\n",
    "params.ATAC_FRAGMENTS_CUTOFF = 4\n",
    "params.RUN_TABIX_SCRIPT='run_tabix.sh'\n",
    "params.ATAC_TSS_BASES_FLANK = 2000\n",
    "params.ATAC_TSS_COL_WITH_STRANDS_INFO = 4\n",
    "params.ATAC_TSS_SMOOTHING_WINDOW_SIZE = 20\n",
    "params.ATAC_TSS_REGION_BED_FILE='{nf_data_base_path}nf_wf_fg_singlecell/tss_regions.bed'\n",
    "params.ATAC_TSS_CALCULATION_SCRIPT='compute_tss_script.py'\n",
    "\n",
    "\n",
    "params.SC_ATAC_QC_PLOT_SCRIPT='scrna_plot_atac_qc_metrics.R'\n",
    "params.SC_ATAC_QC_PLOT_HELPER_SCRIPT='barcode_rank_functions.R'\n",
    "params.SC_ATAC_QC_BARCODE_METADATA_FILE='{nf_data_base_path}nf_wf_fg_singlecell/barcode_metadata.txt'\n",
    "params.SC_ATAC_QC_FRAGMENT_CUTOFF=4\n",
    "params.SC_ATAC_QC_BARCODE_OUTPUT_FILE='sc_atac_qc_barcode_output_file.png'\n",
    "    \n",
    "params.SC_ATAC_BARCODE_METADATA_SCRIPT='generate_barcode_metadata.sh'\n",
    "params.SC_ATAC_BARCODE_METADATA_FILTERED_BARCODE_STATS='{nf_data_base_path}nf_wf_fg_singlecell/filtered_barcode_stats.csv'\n",
    "params.SC_ATAC_BARCODE_METADATA_TSS_ENRICHMENT_BARCODE_STATS='{nf_data_base_path}nf_wf_fg_singlecell/tss_enrichment_barcode_stats.csv'\n",
    "\n",
    "params.SC_ATAC_GENERATE_BARCODE_RANK_PLOT_SCRIPT='plot_insert_size_hist.py'\n",
    "params.SC_ATAC_GENERATE_BARCODE_RANK_PLOT_PKR='PKR123'\n",
    "\n",
    "params.SC_JOINT_CELL_BARCODE_PLOT_SCRTIP_PY='joint_cell_plotting.py'\n",
    "params.SC_JOINT_CELL_BARCODE_PLOT_SCRTIP_R='joint_cell_plotting_density.R'\n",
    "\"\"\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !rm -r /Users/eilaarich-landkof-stanford/Documents/Code/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/work\n",
    "# !rm -r /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/work\n",
    "# !gsutil rm -r gs://eila-nextflow-bucket/work/ATAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !rm .nextflow*\n",
    "\n",
    "# !ls /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/local_atac_fastqs_files.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 209,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "N E X T F L O W  ~  version 23.10.1\n",
      "Launching `nf_wf_fg_singlecell_atac.nf` [thirsty_brazil] DSL2 - revision: 6e47c48f91\n",
      "Jan 24, 2024 12:59:48 PM com.google.auth.oauth2.DefaultCredentialsProvider warnAboutProblematicCredentials\n",
      "WARNING: Your application has authenticated using end user credentials from Google Cloud SDK. We recommend that most server applications use service accounts instead. If your application continues to use end user credentials from Cloud SDK, you might receive a \"quota exceeded\" or \"API not enabled\" error. For more information about service accounts, see https://cloud.google.com/docs/authentication/.\n",
      "\u001b[33mWARN: `params.tss_bases_flank` is defined multiple times -- Assignments following the first are ignored\u001b[39m\u001b[K\n",
      "/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/local_atac_fastqs_files.tsv\n",
      "eyJ0eXAiOiJKV1QiLCJraWQiOiJXN05OOldMSlQ6SjVSSzpMN1RMOlQ3TDc6M1ZYNjpKRU9VOjY0NFI6VTNJWDo1S1oyOjdaQ0s6RlBUSCIsImFsZyI6IlJTMjU2In0.eyJhY2Nlc3MiOnsic2NvcGUiOlsidmlldyIsImRvd25sb2FkIl0sIm9pZGNfY2xhaW1zIjp7fX0sInRva2VuX3R5cGUiOiJQRVJTT05BTF9BQ0NFU1NfVE9LRU4iLCJpc3MiOiJodHRwczovL3JlcG8tcHJvZC5wcm9kLnNhZ2ViYXNlLm9yZy9hdXRoL3YxIiwiYXVkIjoiMCIsIm5iZiI6MTY5NzgxODkxNiwiaWF0IjoxNjk3ODE4OTE2LCJqdGkiOiI0MTIzIiwic3ViIjoiMzQ3NDU3MSJ9.n0cugYMAVzYZ8mMMc7CjcqqEV7FB-K-oKUDT45SvM6KmbCpb_v3lZ2gg3xXi1tubWH0__wpnbJ7cYrGEtOyt4vXAPkMiEtZFGqS0MFKp_h-mHHH4pnACSQQYEy1zdyqFn9WRE7ipjWg4yYSbq84N7J3gzi8VKmiOAmuIGKaJaS0snkUjtJEVQH-0vZqWs1nKgVRWIZv31Bo8vuDzWpBrXD-zRT_uQW9l_I9fU50o1AqyFhNe_gbty_Z6imb8VT33wMXiDgJh15lG_M70PXU15v3egDXLgw1dGj32kj1nKanfqBVqbMSzgzPAG037p3YH_Q556BLceFBBb5Jaej5JVA\n",
      "after genome_fasta_ch\n",
      "[-        ] process > run_zcat               -\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip   -\u001b[K\n",
      "[-        ] process > run_chromap_map_to_idx -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_filter_fragments\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "before call run_calculate_tss_enrichment\u001b[K\n",
      "after regions_ch\u001b[K\n",
      "after call run_calculate_tss_enrichment\u001b[K\n",
      "before call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "after call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "before call run_generate_barcode_metadata\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "\u001b[18A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/uTtaw2f7i3Go4\u001b[22;39m\u001b[K\n",
      "[-        ] process > run_zcat               -\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip   -\u001b[K\n",
      "[-        ] process > run_chromap_map_to_idx -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_filter_fragments\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "before call run_calculate_tss_enrichment\u001b[K\n",
      "after regions_ch\u001b[K\n",
      "after call run_calculate_tss_enrichment\u001b[K\n",
      "before call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "after call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "before call run_generate_barcode_metadata\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "\u001b[19A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/uTtaw2f7i3Go4\u001b[22;39m\u001b[K\n",
      "[-        ] process > run_zcat               -\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip   -\u001b[K\n",
      "[-        ] process > run_chromap_map_to_idx -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_filter_fragments\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "before call run_calculate_tss_enrichment\u001b[K\n",
      "after regions_ch\u001b[K\n",
      "after call run_calculate_tss_enrichment\u001b[K\n",
      "before call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "after call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "before call run_generate_barcode_metadata\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "\u001b[20A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/uTtaw2f7i3Go4\u001b[22;39m\u001b[K\n",
      "[c4/48e77e] process > run_zcat               [100%] 1 of 1, cached: 1\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip   -\u001b[K\n",
      "[-        ] process > run_chromap_map_to_idx -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_filter_fragments\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "before call run_calculate_tss_enrichment\u001b[K\n",
      "after regions_ch\u001b[K\n",
      "after call run_calculate_tss_enrichment\u001b[K\n",
      "before call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "after call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "before call run_generate_barcode_metadata\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "\u001b[20A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/uTtaw2f7i3Go4\u001b[22;39m\u001b[K\n",
      "[c4/48e77e] process > run_zcat               [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip   -\u001b[K\n",
      "[-        ] process > run_chromap_map_to_idx -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_filter_fragments\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "before call run_calculate_tss_enrichment\u001b[K\n",
      "after regions_ch\u001b[K\n",
      "after call run_calculate_tss_enrichment\u001b[K\n",
      "before call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "after call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "before call run_generate_barcode_metadata\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Jan 24 20:39 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/08/c38215e764bb1dd1b00fab8d2a89d4/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Jan 24 20:39 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Jan 24 20:39 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/08/c38215e764bb1dd1b00fab8d2a89d4/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "\u001b[31A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/uTtaw2f7i3Go4\u001b[22;39m\u001b[K\n",
      "[c4/48e77e] process > run_zcat               [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip   -\u001b[K\n",
      "[-        ] process > run_chromap_map_to_idx -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_filter_fragments\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "before call run_calculate_tss_enrichment\u001b[K\n",
      "after regions_ch\u001b[K\n",
      "after call run_calculate_tss_enrichment\u001b[K\n",
      "before call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "after call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "before call run_generate_barcode_metadata\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Jan 24 20:39 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/08/c38215e764bb1dd1b00fab8d2a89d4/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Jan 24 20:39 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Jan 24 20:39 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/08/c38215e764bb1dd1b00fab8d2a89d4/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "\u001b[32A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/uTtaw2f7i3Go4\u001b[22;39m\u001b[K\n",
      "[c4/48e77e] process > run_zcat               [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip   -\u001b[K\n",
      "[-        ] process > run_chromap_map_to_idx -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_filter_fragments\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "before call run_calculate_tss_enrichment\u001b[K\n",
      "after regions_ch\u001b[K\n",
      "after call run_calculate_tss_enrichment\u001b[K\n",
      "before call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "after call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "before call run_generate_barcode_metadata\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Jan 24 20:39 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/08/c38215e764bb1dd1b00fab8d2a89d4/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Jan 24 20:39 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Jan 24 20:39 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/08/c38215e764bb1dd1b00fab8d2a89d4/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "\u001b[33A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/uTtaw2f7i3Go4\u001b[22;39m\u001b[K\n",
      "[c4/48e77e] process > run_zcat               [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip   -\u001b[K\n",
      "[-        ] process > run_chromap_map_to_idx -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_filter_fragments\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "before call run_calculate_tss_enrichment\u001b[K\n",
      "after regions_ch\u001b[K\n",
      "after call run_calculate_tss_enrichment\u001b[K\n",
      "before call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "after call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "before call run_generate_barcode_metadata\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Jan 24 20:39 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/08/c38215e764bb1dd1b00fab8d2a89d4/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Jan 24 20:39 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Jan 24 20:39 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/08/c38215e764bb1dd1b00fab8d2a89d4/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "\u001b[34A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/uTtaw2f7i3Go4\u001b[22;39m\u001b[K\n",
      "[c4/48e77e] process > run_zcat               [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip   -\u001b[K\n",
      "[-        ] process > run_chromap_map_to_idx -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_filter_fragments\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "before call run_calculate_tss_enrichment\u001b[K\n",
      "after regions_ch\u001b[K\n",
      "after call run_calculate_tss_enrichment\u001b[K\n",
      "before call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "after call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "before call run_generate_barcode_metadata\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Jan 24 20:39 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/08/c38215e764bb1dd1b00fab8d2a89d4/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Jan 24 20:39 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Jan 24 20:39 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/08/c38215e764bb1dd1b00fab8d2a89d4/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "\u001b[35A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/uTtaw2f7i3Go4\u001b[22;39m\u001b[K\n",
      "[c4/48e77e] process > run_zcat               [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip   -\u001b[K\n",
      "[-        ] process > run_chromap_map_to_idx -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_filter_fragments\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "before call run_calculate_tss_enrichment\u001b[K\n",
      "after regions_ch\u001b[K\n",
      "after call run_calculate_tss_enrichment\u001b[K\n",
      "before call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "after call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "before call run_generate_barcode_metadata\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Jan 24 20:39 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/08/c38215e764bb1dd1b00fab8d2a89d4/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Jan 24 20:39 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Jan 24 20:39 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/08/c38215e764bb1dd1b00fab8d2a89d4/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\u001b[K\n"
     ]
    }
   ],
   "source": [
    "!nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_atac.config -profile google_batch_profile -with-tower -resume\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_vm.config -profile google_batch_profile\n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_vm.config -profile google_batch_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_vm_anvil.config -profile google_batch_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_local.config -profile google_batch_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_local.config -profile docker_profile\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Jan-24 12:52:35.604 [main] DEBUG nextflow.cli.Launcher - $> nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_atac.config -profile google_batch_profile -with-tower -resume\n",
      "Jan-24 12:52:35.668 [main] INFO  nextflow.cli.CmdRun - N E X T F L O W  ~  version 23.10.1\n",
      "Jan-24 12:52:35.679 [main] DEBUG nextflow.plugin.PluginsFacade - Setting up plugin manager > mode=prod; embedded=false; plugins-dir=/Users/eilaarich-landkof-stanford/.nextflow/plugins; core-plugins: nf-amazon@2.1.4,nf-azure@1.3.3,nf-cloudcache@0.3.0,nf-codecommit@0.1.5,nf-console@1.0.6,nf-ga4gh@1.1.0,nf-google@1.8.3,nf-tower@1.6.3,nf-wave@1.0.1\n",
      "Jan-24 12:52:35.685 [main] INFO  o.pf4j.DefaultPluginStatusProvider - Enabled plugins: []\n",
      "Jan-24 12:52:35.685 [main] INFO  o.pf4j.DefaultPluginStatusProvider - Disabled plugins: []\n",
      "Jan-24 12:52:35.686 [main] INFO  org.pf4j.DefaultPluginManager - PF4J version 3.4.1 in 'deployment' mode\n",
      "Jan-24 12:52:35.691 [main] INFO  org.pf4j.AbstractPluginManager - No plugins\n",
      "Jan-24 12:52:35.701 [main] DEBUG nextflow.config.ConfigBuilder - User config file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Jan-24 12:52:35.701 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Jan-24 12:52:35.712 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `google_batch_profile`\n",
      "Jan-24 12:52:36.737 [main] DEBUG nextflow.config.ConfigBuilder - Available config profiles: [google_batch_profile]\n",
      "Jan-24 12:52:36.757 [main] DEBUG nextflow.cli.CmdRun - Applied DSL=2 by global default\n",
      "Jan-24 12:52:36.767 [main] INFO  nextflow.cli.CmdRun - Launching `nf_wf_fg_singlecell_atac.nf` [hopeful_jang] DSL2 - revision: 6e47c48f91\n",
      "Jan-24 12:52:36.768 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins default=[nf-google@1.8.3]\n",
      "Jan-24 12:52:36.768 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins resolved requirement=[nf-google@1.8.3, nf-tower@1.6.3, nf-wave@1.0.1]\n",
      "Jan-24 12:52:36.768 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-google version: 1.8.3\n",
      "Jan-24 12:52:36.778 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-google@1.8.3' resolved\n",
      "Jan-24 12:52:36.778 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-google@1.8.3'\n",
      "Jan-24 12:52:36.826 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-google@1.8.3\n",
      "Jan-24 12:52:36.826 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-tower version: 1.6.3\n",
      "Jan-24 12:52:36.827 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-tower@1.6.3' resolved\n",
      "Jan-24 12:52:36.827 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-tower@1.6.3'\n",
      "Jan-24 12:52:36.832 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-tower@1.6.3\n",
      "Jan-24 12:52:36.832 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-wave version: 1.0.1\n",
      "Jan-24 12:52:36.835 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-wave@1.0.1' resolved\n",
      "Jan-24 12:52:36.835 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-wave@1.0.1'\n",
      "Jan-24 12:52:36.841 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-wave@1.0.1\n",
      "Jan-24 12:52:36.855 [main] DEBUG n.secret.LocalSecretsProvider - Secrets store: /Users/eilaarich-landkof-stanford/.nextflow/secrets/store.json\n",
      "Jan-24 12:52:36.856 [main] DEBUG nextflow.secret.SecretsLoader - Discovered secrets providers: [nextflow.secret.LocalSecretsProvider@633cc6b5] - activable => nextflow.secret.LocalSecretsProvider@633cc6b5\n",
      "Jan-24 12:52:36.898 [main] DEBUG nextflow.Session - Session UUID: 0cbd4d64-b857-4807-b564-11778fa2d750\n",
      "Jan-24 12:52:36.898 [main] DEBUG nextflow.Session - Run name: hopeful_jang\n",
      "Jan-24 12:52:36.898 [main] DEBUG nextflow.Session - Executor pool size: 8\n",
      "Jan-24 12:52:36.989 [main] DEBUG nextflow.file.FilePorter - File porter settings maxRetries=3; maxTransfers=50; pollTimeout=null\n",
      "Jan-24 12:52:36.991 [main] DEBUG nextflow.util.ThreadPoolBuilder - Creating thread pool 'FileTransfer' minSize=10; maxSize=24; workQueue=LinkedBlockingQueue[10000]; allowCoreThreadTimeout=false\n",
      "Jan-24 12:52:36.992 [main] DEBUG nextflow.config.ConfigBuilder - User config file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Jan-24 12:52:36.992 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Jan-24 12:52:36.993 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `google_batch_profile`\n",
      "Jan-24 12:52:37.673 [main] DEBUG nextflow.config.ConfigBuilder - Available config profiles: [google_batch_profile]\n",
      "Jan-24 12:52:37.852 [main] DEBUG nextflow.cli.CmdRun - \n",
      "  Version: 23.10.1 build 5891\n",
      "  Created: 12-01-2024 22:01 UTC (14:01 PDT)\n",
      "  System: Mac OS X 14.2.1\n",
      "  Runtime: Groovy 3.0.19 on OpenJDK 64-Bit Server VM 17.0.8+7-LTS\n",
      "  Encoding: UTF-8 (UTF-8)\n",
      "  Process: 22531@Eilas-MacBook-Air.local [10.0.0.212]\n",
      "  CPUs: 8 - Mem: 16 GB (163 MB) - Swap: 2 GB (829.8 MB)\n",
      "Jan-24 12:52:37.872 [main] DEBUG nextflow.Session - Work-dir: gs://eila-nextflow-bucket/work/ATAC [Mac OS X]\n",
      "Jan-24 12:52:37.889 [main] DEBUG nextflow.executor.ExecutorFactory - Extension executors providers=[GoogleLifeSciencesExecutor, GoogleBatchExecutor]\n",
      "Jan-24 12:52:37.899 [main] DEBUG nextflow.Session - Observer factory: DefaultObserverFactory\n",
      "Jan-24 12:52:37.900 [main] DEBUG nextflow.Session - Observer factory: WaveFactory\n",
      "Jan-24 12:52:37.900 [main] DEBUG io.seqera.wave.plugin.WaveFactory - Detected Fusion enabled -- Enabling bundle project resources -- Disabling upload of remote bin directory\n",
      "Jan-24 12:52:37.910 [main] DEBUG i.s.wave.plugin.config.WaveConfig - Wave strategy not specified - using default: [container, dockerfile, conda, spack]\n",
      "Jan-24 12:52:37.913 [main] DEBUG io.seqera.wave.plugin.WaveClient - Wave config: WaveConfig(enabled:true, endpoint:https://wave.seqera.io, containerConfigUrl:[], tokensCacheMaxDuration:30m, condaOpts:CondaOpts(mambaImage=mambaorg/micromamba:1.5.1; basePackages=conda-forge::procps-ng, commands=null), spackOpts:SpackOpts(basePackages=null, commands=null), strategy:[container, dockerfile, conda, spack], bundleProjectResources:true, buildRepository:null, cacheRepository:null, retryOpts:RetryOpts(delay:450ms, maxDelay:1m 30s, maxAttempts:10, jitter:0.25), httpClientOpts:HttpOpts(), freezeMode:null)\n",
      "Jan-24 12:52:37.985 [main] DEBUG nextflow.Session - Observer factory: TowerFactory\n",
      "Jan-24 12:52:38.015 [main] DEBUG nextflow.cache.CacheFactory - Using Nextflow cache factory: nextflow.cache.DefaultCacheFactory\n",
      "Jan-24 12:52:38.019 [main] DEBUG nextflow.util.CustomThreadPool - Creating default thread pool > poolSize: 9; maxThreads: 1000\n",
      "Jan-24 12:52:38.068 [main] DEBUG nextflow.Session - Session start\n",
      "Jan-24 12:52:38.073 [main] DEBUG io.seqera.tower.plugin.TowerClient - Creating Tower observer -- endpoint=https://api.tower.nf; requestInterval=1s; aliveInterval=1m; maxRetries=5; backOffBase=3; backOffDelay=250\n",
      "Jan-24 12:52:39.284 [main] DEBUG nextflow.script.ScriptRunner - > Launching execution\n",
      "Jan-24 12:52:39.603 [main] WARN  nextflow.script.ScriptBinding - `params.tss_bases_flank` is defined multiple times -- Assignments following the first are ignored\n",
      "Jan-24 12:52:39.730 [main] INFO  nextflow.script.BaseScript - /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/local_atac_fastqs_files.tsv\n",
      "Jan-24 12:52:39.731 [main] INFO  nextflow.script.BaseScript - eyJ0eXAiOiJKV1QiLCJraWQiOiJXN05OOldMSlQ6SjVSSzpMN1RMOlQ3TDc6M1ZYNjpKRU9VOjY0NFI6VTNJWDo1S1oyOjdaQ0s6RlBUSCIsImFsZyI6IlJTMjU2In0.eyJhY2Nlc3MiOnsic2NvcGUiOlsidmlldyIsImRvd25sb2FkIl0sIm9pZGNfY2xhaW1zIjp7fX0sInRva2VuX3R5cGUiOiJQRVJTT05BTF9BQ0NFU1NfVE9LRU4iLCJpc3MiOiJodHRwczovL3JlcG8tcHJvZC5wcm9kLnNhZ2ViYXNlLm9yZy9hdXRoL3YxIiwiYXVkIjoiMCIsIm5iZiI6MTY5NzgxODkxNiwiaWF0IjoxNjk3ODE4OTE2LCJqdGkiOiI0MTIzIiwic3ViIjoiMzQ3NDU3MSJ9.n0cugYMAVzYZ8mMMc7CjcqqEV7FB-K-oKUDT45SvM6KmbCpb_v3lZ2gg3xXi1tubWH0__wpnbJ7cYrGEtOyt4vXAPkMiEtZFGqS0MFKp_h-mHHH4pnACSQQYEy1zdyqFn9WRE7ipjWg4yYSbq84N7J3gzi8VKmiOAmuIGKaJaS0snkUjtJEVQH-0vZqWs1nKgVRWIZv31Bo8vuDzWpBrXD-zRT_uQW9l_I9fU50o1AqyFhNe_gbty_Z6imb8VT33wMXiDgJh15lG_M70PXU15v3egDXLgw1dGj32kj1nKanfqBVqbMSzgzPAG037p3YH_Q556BLceFBBb5Jaej5JVA\n",
      "Jan-24 12:52:39.765 [main] INFO  nextflow.script.BaseScript - after genome_fasta_ch\n",
      "Jan-24 12:52:39.797 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:zcat` matches labels `zcat` for process with name run_zcat\n",
      "Jan-24 12:52:39.801 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Jan-24 12:52:39.802 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Jan-24 12:52:39.803 [main] DEBUG nextflow.executor.Executor - [warm up] executor > google-batch (fusion enabled)\n",
      "Jan-24 12:52:39.806 [main] DEBUG n.processor.TaskPollingMonitor - Creating task monitor for executor 'google-batch' > capacity: 1000; pollInterval: 10s; dumpInterval: 5m \n",
      "Jan-24 12:52:39.807 [main] DEBUG n.processor.TaskPollingMonitor - >>> barrier register (monitor: google-batch)\n",
      "Jan-24 12:52:39.809 [main] DEBUG nextflow.cloud.google.GoogleOpts - Google auth via application DEFAULT\n",
      "Jan-24 12:52:39.811 [main] DEBUG n.c.google.batch.GoogleBatchExecutor - [GOOGLE BATCH] Executor config=BatchConfig[googleOpts=GoogleOpts(projectId:hai-gcp-genomic, credsFile:null, location:us-central1, enableRequesterPaysBuckets:false, httpConnectTimeout:1m, httpReadTimeout:1m, credentials:UserCredentials{requestMetadata=null, temporaryAccess=null, clientId=764086051850-6qr4p6gpi6hn506pt8ejuq83di341hur.apps.googleusercontent.com, refreshToken=1//06o-o0zDxXSs1CgYIARAAGAYSNwF-L9IrUI26O_biiC7xbALX-c2sD4FAkhuU8h_Vtb27_cEuZ6it1oqhLF2G_Za95PIBVrUBVrY, tokenServerUri=https://oauth2.googleapis.com/token, transportFactoryClassName=com.google.auth.oauth2.OAuth2Utils$DefaultHttpTransportFactory, quotaProjectId=hai-gcp-genomic})\n",
      "Jan-24 12:52:39.820 [main] DEBUG n.c.google.batch.client.BatchClient - [GOOGLE BATCH] Creating service client with config credentials\n",
      "Jan-24 12:52:40.339 [main] INFO  nextflow.script.BaseScript - after run_zcat \n",
      "Jan-24 12:52:40.342 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:gunzip` matches labels `gunzip` for process with name run_whitelist_gunzip\n",
      "Jan-24 12:52:40.342 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Jan-24 12:52:40.342 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Jan-24 12:52:40.343 [main] INFO  nextflow.script.BaseScript - after run_whitelist_gunzip\n",
      "Jan-24 12:52:40.345 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:chromap_map_idx` matches labels `chromap_map_idx` for process with name run_chromap_map_to_idx\n",
      "Jan-24 12:52:40.346 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Jan-24 12:52:40.346 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Jan-24 12:52:40.347 [main] INFO  nextflow.script.BaseScript - finished run_chromap_map_to_idx\n",
      "Jan-24 12:52:40.347 [main] INFO  nextflow.script.BaseScript - after call bgzip\n",
      "Jan-24 12:52:40.347 [main] INFO  nextflow.script.BaseScript - before call run_filter_fragments\n",
      "Jan-24 12:52:40.347 [main] INFO  nextflow.script.BaseScript - after call run_filter_fragments\n",
      "Jan-24 12:52:40.347 [main] INFO  nextflow.script.BaseScript - before call run_tabix for the second time with filtered_fragment_file_out\n",
      "Jan-24 12:52:40.347 [main] INFO  nextflow.script.BaseScript - before call run_calculate_tss_enrichment\n",
      "Jan-24 12:52:40.348 [main] INFO  nextflow.script.BaseScript - after regions_ch\n",
      "Jan-24 12:52:40.348 [main] INFO  nextflow.script.BaseScript - after call run_calculate_tss_enrichment\n",
      "Jan-24 12:52:40.348 [main] INFO  nextflow.script.BaseScript - before call run_scrna_atac_plot_qc_metrics\n",
      "Jan-24 12:52:40.348 [main] INFO  nextflow.script.BaseScript - after call run_scrna_atac_plot_qc_metrics\n",
      "Jan-24 12:52:40.348 [main] INFO  nextflow.script.BaseScript - before call run_generate_barcode_metadata\n",
      "Jan-24 12:52:40.348 [main] INFO  nextflow.script.BaseScript - before call run_generate_barcode_rank_plot\n",
      "Jan-24 12:52:40.348 [main] DEBUG nextflow.Session - Workflow process names [dsl2]: run_merge_logs, run_seqspec_modify_rna, run_seqspec_print, run_atac_barcode_rank_plot, run_bgzip, run_calculate_tss_enrichment, run_atac_barcode_metadata, run_tabix, run_filter_fragments, run_whitelist_gunzip, run_create_chromap_idx, run_zcat, run_seqspec_check, run_seqspec_test, run_seqspec_index_rna_kb, run_scrna_atac_plot_qc_metrics, run_seqspec_modify_atac, run_chromap_test, run_tabix_filtered_fragments, run_downloadFiles, run_chromap_map_to_idx\n",
      "Jan-24 12:52:40.750 [main] INFO  io.seqera.tower.plugin.TowerClient - Monitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/1mFOpOCaKoRZPe\n",
      "Jan-24 12:52:40.751 [main] DEBUG nextflow.Session - Igniting dataflow network (6)\n",
      "Jan-24 12:52:40.757 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_zcat\n",
      "Jan-24 12:52:40.757 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_whitelist_gunzip\n",
      "Jan-24 12:52:40.757 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_chromap_map_to_idx\n",
      "Jan-24 12:52:40.758 [main] DEBUG nextflow.script.ScriptRunner - Parsed script files:\n",
      "  Script_0de7f532c8b3da2c: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_bgzip.nf\n",
      "  Script_87d8d9fb99b69787: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_atac_barcode_rank_plot.nf\n",
      "  Script_ed5bff89020dcf2d: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_chromap_utils.nf\n",
      "  Script_1aaf2ed009725448: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_filter_fragments.nf\n",
      "  Script_cc895528cea919f6: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_zcat.nf\n",
      "  Script_9c2ce54dcf0b5c06: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_merge_logs.nf\n",
      "  Script_9d4639c8cd883c70: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_download_url_files.nf\n",
      "  Script_7664153e6cd8ad6c: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_tss.nf\n",
      "  Script_ea7cdb417e42517f: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_generate_barcode_metadata.nf\n",
      "  Script_e112646bcdb899f0: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_seqspec_utils.nf\n",
      "  Script_ea6dd3976b47313b: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nf_wf_fg_singlecell_atac.nf\n",
      "  Script_3d78df279f932afb: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_atac_qc_plots.nf\n",
      "  Script_63bd9e7e1461f2d4: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_gunzip.nf\n",
      "  Script_caef5739a3ac2dec: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_tabix.nf\n",
      "Jan-24 12:52:40.759 [main] DEBUG nextflow.script.ScriptRunner - > Awaiting termination \n",
      "Jan-24 12:52:40.759 [main] DEBUG nextflow.Session - Session await\n",
      "Jan-24 12:52:42.399 [FileTransfer-1] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/08/c38215e764bb1dd1b00fab8d2a89d4/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\n",
      "Jan-24 12:52:42.408 [Actor Thread 3] DEBUG i.s.wave.plugin.config.WaveConfig - Wave strategy not specified - using default: [container, dockerfile, conda, spack]\n",
      "Jan-24 12:52:42.409 [Actor Thread 3] DEBUG io.seqera.wave.plugin.WaveClient - Wave config: WaveConfig(enabled:true, endpoint:https://wave.seqera.io, containerConfigUrl:[], tokensCacheMaxDuration:30m, condaOpts:CondaOpts(mambaImage=mambaorg/micromamba:1.5.1; basePackages=conda-forge::procps-ng, commands=null), spackOpts:SpackOpts(basePackages=null, commands=null), strategy:[container, dockerfile, conda, spack], bundleProjectResources:true, buildRepository:null, cacheRepository:null, retryOpts:RetryOpts(delay:450ms, maxDelay:1m 30s, maxAttempts:10, jitter:0.25), httpClientOpts:HttpOpts(), freezeMode:null)\n",
      "Jan-24 12:52:42.412 [Actor Thread 3] DEBUG io.seqera.wave.plugin.WaveClient - Wave assets bundle bin resources: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/bin\n",
      "Jan-24 12:52:42.429 [Actor Thread 3] DEBUG io.seqera.wave.plugin.WaveClient - Wave request container config: https://fusionfs.seqera.io/releases/v2.2-amd64.json\n",
      "Jan-24 12:52:42.615 [Actor Thread 3] DEBUG io.seqera.wave.plugin.WaveClient - Wave container config response: [200] {\n",
      "  \"entrypoint\": [ \"/usr/bin/fusion\" ],\n",
      "  \"layers\": [\n",
      "    {\n",
      "      \"location\": \"https://fusionfs.seqera.io/releases/pkg/2/2/11/fusion-amd64.tar.gz\",\n",
      "      \"gzipDigest\": \"sha256:b1664eb81e6049524db659a1225570507eea75267a1a55b1e2a85edded9e773d\",\n",
      "      \"gzipSize\": 30262205,\n",
      "      \"tarDigest\": \"sha256:5e126d27b3c3b4ffac9a4b2c94bd2d15fabdceeddcdf9b711154fc03d8a88d47\",\n",
      "      \"skipHashing\": true\n",
      "    }\n",
      "  ]\n",
      "}\n",
      "\n",
      "Jan-24 12:52:42.640 [Actor Thread 3] DEBUG io.seqera.wave.plugin.WaveClient - Wave request: https://wave.seqera.io/container-token; attempt=1 - request: SubmitContainerTokenRequest(towerAccessToken:eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz, towerRefreshToken:null, towerWorkspaceId:247507587970818, towerEndpoint:https://api.tower.nf, workflowId:null, containerImage:docker.io/eilalan/utils_env:latest, containerFile:null, containerConfig:ContainerConfig(entrypoint:[/usr/bin/fusion], cmd:null, env:null, workingDir:null, layers:[ContainerLayer[location=data:H4sIAAAAAAAA/+09/Xfb...; tarDigest=sha256:80a2a1dfd91cd69973586dccdd4d2ce0e3ee7ddea6c62956e056335093291bb2; gzipDigest=sha256:46ed86bb24fdc046d6e46c813531e5049e4386eb4052d116d7d71b7ac8bf97b4; gzipSize=7566], ContainerLayer[location=https://fusionfs.seqera.io/releases/pkg/2/2/11/fusion-amd64.tar.gz; tarDigest=sha256:5e126d27b3c3b4ffac9a4b2c94bd2d15fabdceeddcdf9b711154fc03d8a88d47; gzipDigest=sha256:b1664eb81e6049524db659a1225570507eea75267a1a55b1e2a85edded9e773d; gzipSize=30262205]]), condaFile:null, spackFile:null, containerPlatform:linux/amd64, buildRepository:null, cacheRepository:null, timestamp:2024-01-24T12:52:42.639800-08:00, fingerprint:916a2ec5ef6599914c35fa7fd0112902, freeze:false, format:null, dryRun:false)\n",
      "Jan-24 12:52:43.612 [Actor Thread 3] DEBUG io.seqera.wave.plugin.WaveClient - Wave response: statusCode=200; body={\"containerToken\":\"5b16368115e8\",\"targetImage\":\"wave.seqera.io/wt/5b16368115e8/eilalan/utils_env:latest\",\"expiration\":\"2024-01-26T08:52:43.601193830Z\",\"containerImage\":\"docker.io/eilalan/utils_env:latest\"}\n",
      "Jan-24 12:52:44.153 [FileTransfer-2] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/na.na at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/8b/b521669a71db49f1fb1d2870ff0854/na.na\n",
      "Jan-24 12:52:44.219 [FileTransfer-8] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/na6.na at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/71/44705762e55f12bcce7c7f81f8db66/na6.na\n",
      "Jan-24 12:52:44.219 [FileTransfer-5] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/na3.na at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/82/e88e18dc38d16a4d861dd11dead3e3/na3.na\n",
      "Jan-24 12:52:44.219 [FileTransfer-6] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/na4.na at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/c8/f74a7cdf760508436ae991a018a30a/na4.na\n",
      "Jan-24 12:52:44.219 [FileTransfer-7] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/na5.na at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/45/ae806881ee78a1e18e11fa7dae53ad/na5.na\n",
      "Jan-24 12:52:44.221 [FileTransfer-9] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/sai_192_whitelist.txt at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/49/466f9021bd066634c6b7271f80a7a7/sai_192_whitelist.txt\n",
      "Jan-24 12:52:44.221 [FileTransfer-3] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/na1.na at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/6c/ce8eff486ee7fbc13d0021eac82ef2/na1.na\n",
      "Jan-24 12:52:44.222 [FileTransfer-4] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/na2.na at gs://eila-nextflow-bucket/work/ATAC/stage-0cbd4d64-b857-4807-b564-11778fa2d750/c5/86aab1bdad5a114b18d2b8dab9fb39/na2.na\n",
      "Jan-24 12:52:44.625 [Actor Thread 3] INFO  nextflow.processor.TaskProcessor - [c4/48e77e] Cached process > run_zcat\n",
      "Jan-24 12:52:45.969 [Actor Thread 7] INFO  nextflow.processor.TaskProcessor - [9a/07af04] Cached process > run_whitelist_gunzip (1)\n",
      "Jan-24 12:52:45.995 [Actor Thread 3] DEBUG io.seqera.wave.plugin.WaveClient - Wave request: https://wave.seqera.io/container-token; attempt=1 - request: SubmitContainerTokenRequest(towerAccessToken:eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz, towerRefreshToken:null, towerWorkspaceId:247507587970818, towerEndpoint:https://api.tower.nf, workflowId:null, containerImage:eilalan/chromap_env:latest, containerFile:null, containerConfig:ContainerConfig(entrypoint:[/usr/bin/fusion], cmd:null, env:null, workingDir:null, layers:[ContainerLayer[location=data:H4sIAAAAAAAA/+09/Xfb...; tarDigest=sha256:80a2a1dfd91cd69973586dccdd4d2ce0e3ee7ddea6c62956e056335093291bb2; gzipDigest=sha256:46ed86bb24fdc046d6e46c813531e5049e4386eb4052d116d7d71b7ac8bf97b4; gzipSize=7566], ContainerLayer[location=https://fusionfs.seqera.io/releases/pkg/2/2/11/fusion-amd64.tar.gz; tarDigest=sha256:5e126d27b3c3b4ffac9a4b2c94bd2d15fabdceeddcdf9b711154fc03d8a88d47; gzipDigest=sha256:b1664eb81e6049524db659a1225570507eea75267a1a55b1e2a85edded9e773d; gzipSize=30262205]]), condaFile:null, spackFile:null, containerPlatform:linux/amd64, buildRepository:null, cacheRepository:null, timestamp:2024-01-24T12:52:45.995178-08:00, fingerprint:aa128c2f7ee90fea00adfa693aa9cfc6, freeze:false, format:null, dryRun:false)\n",
      "Jan-24 12:52:46.282 [Actor Thread 3] DEBUG io.seqera.wave.plugin.WaveClient - Wave response: statusCode=200; body={\"containerToken\":\"5493c4e388e5\",\"targetImage\":\"wave.seqera.io/wt/5493c4e388e5/eilalan/chromap_env:latest\",\"expiration\":\"2024-01-26T08:52:46.273604923Z\",\"containerImage\":\"docker.io/eilalan/chromap_env:latest\"}\n",
      "Jan-24 12:52:48.992 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_chromap_map_to_idx (1)` - adding local volume as fusion scratch: DiskResource(request:375 GB, type:local-ssd)\n",
      "Jan-24 12:52:51.491 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_chromap_map_to_idx (1)` submitted > job=nf-d8e8a678-1706129568176; uid=nf-d8e8a678-170612-ad6090c2-1a14-4fb40; work-dir=gs://eila-nextflow-bucket/work/ATAC/d8/e8a678b3e424473ec756122b5d802a\n",
      "Jan-24 12:52:51.491 [Task submitter] INFO  nextflow.Session - [d8/e8a678] Submitted process > run_chromap_map_to_idx (1)\n",
      "Jan-24 12:55:10.336 [Task monitor] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_chromap_map_to_idx (1)` - terminated job=nf-d8e8a678-1706129568176; state=FAILED\n",
      "Jan-24 12:55:12.581 [Task monitor] DEBUG n.processor.TaskPollingMonitor - Task completed > TaskHandler[id: 3; name: run_chromap_map_to_idx (1); status: COMPLETED; exit: 134; error: -; workDir: gs://eila-nextflow-bucket/work/ATAC/d8/e8a678b3e424473ec756122b5d802a]\n",
      "Jan-24 12:55:12.585 [Task monitor] DEBUG nextflow.processor.TaskProcessor - Handling unexpected condition for\n",
      "  task: name=run_chromap_map_to_idx (1); work-dir=gs://eila-nextflow-bucket/work/ATAC/d8/e8a678b3e424473ec756122b5d802a\n",
      "  error [nextflow.exception.ProcessFailedException]: Process `run_chromap_map_to_idx (1)` terminated with an error exit status (134)\n",
      "Jan-24 12:55:13.284 [Task monitor] ERROR nextflow.processor.TaskProcessor - Error executing process > 'run_chromap_map_to_idx (1)'\n",
      "\n",
      "Caused by:\n",
      "  Process `run_chromap_map_to_idx (1)` terminated with an error exit status (134)\n",
      "\n",
      "Command executed:\n",
      "\n",
      "  echo run_chromap_map_to_idx\n",
      "  echo 'fastq1 is na.na'\n",
      "  echo 'fastq2 is na1.na'\n",
      "  echo 'fastq3 is na2.na'\n",
      "  echo 'fastq4 is na3.na'\n",
      "  echo 'barcode1_fastq is na4.na'\n",
      "  echo 'barcode2_fastq is na5.na'\n",
      "  echo 'genome_fasta is GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz'\n",
      "  echo 'genome_chromap_idx is ref.index'\n",
      "  echo 'genome_fasta_zcat_out is GCA_000001405.15_GRCh38_no_alt_analysis_set.fna'\n",
      "  echo 'barcode_whitelist_inclusion_list is sai_192_whitelist'\n",
      "  echo 'CHROMAP_READ_LENGTH is 2000'\n",
      "  echo 'CHROMAP_BC_ERROR_THRESHOLD is 2'\n",
      "  echo 'CHROMAP_BC_PROBABILITY_THRESHOLD is 0.9'\n",
      "  echo 'CHROMAP_READ_FORMAT is bc:0:-1,r1:0:-1,r2:0:-1'\n",
      "  echo 'CHROMAP_DROP_REPETITIVE_READS is 4'\n",
      "  echo 'CHROMAP_QUALITY_THRESHOLD is 0'\n",
      "  echo 'task.cpus is 1. should be 32. TODO: check what is the issue'\n",
      "  # Run chromap command\n",
      "  chromap -x chromap_index/index --trim-adapters --remove-pcr-duplicates --remove-pcr-duplicates-at-cell-level --Tn5-shift --low-mem --BED          -l 2000 --bc-error-threshold 2 --bc-probability-threshold 0.9         --read-format bc:0:-1,r1:0:-1,r2:0:-1 --drop-repetitive-reads 4         -x chromap_index/index -r GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -q 0 -t 1         -1 na.na,na1.na         -2 na2.na,na3.na         -b na4.na,na5.na         --barcode-whitelist sai_192_whitelist                  -o na.atac.filter.fragments.tsv         --summary na.atac.align.barcode.summary.csv\n",
      "  \n",
      "  # Create placeholder output files\n",
      "  # touch 'na.atac.filter.fragments.tsv'\n",
      "  # touch 'na.atac.align.barcode.summary.csv'\n",
      "  echo run_chromap_map_to_idx\n",
      "\n",
      "Command exit status:\n",
      "  134\n",
      "\n",
      "Command output:\n",
      "  run_chromap_map_to_idx\n",
      "  fastq1 is na.na\n",
      "  fastq2 is na1.na\n",
      "  fastq3 is na2.na\n",
      "  fastq4 is na3.na\n",
      "  barcode1_fastq is na4.na\n",
      "  barcode2_fastq is na5.na\n",
      "  genome_fasta is GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\n",
      "  genome_chromap_idx is ref.index\n",
      "  genome_fasta_zcat_out is GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\n",
      "  barcode_whitelist_inclusion_list is sai_192_whitelist\n",
      "  CHROMAP_READ_LENGTH is 2000\n",
      "  CHROMAP_BC_ERROR_THRESHOLD is 2\n",
      "  CHROMAP_BC_PROBABILITY_THRESHOLD is 0.9\n",
      "  CHROMAP_READ_FORMAT is bc:0:-1,r1:0:-1,r2:0:-1\n",
      "  CHROMAP_DROP_REPETITIVE_READS is 4\n",
      "  CHROMAP_QUALITY_THRESHOLD is 0\n",
      "  task.cpus is 1. should be 32. TODO: check what is the issue\n",
      "\n",
      "Command error:\n",
      "  Start to map reads.\n",
      "  na.na\n",
      "  na1.na\n",
      "  na2.na\n",
      "  na3.na\n",
      "  na4.na\n",
      "  na5.na\n",
      "  Parameters: error threshold: 8, min-num-seeds: 2, max-seed-frequency: 500,1000, max-num-best-mappings: 1, max-insert-size: 2000, MAPQ-threshold: 0, min-read-length: 30, bc-error-threshold: 2, bc-probability-threshold: 0.90\n",
      "  Number of threads: 1\n",
      "  Analyze single-cell data.\n",
      "  Will try to remove adapters on 3'.\n",
      "  Will remove PCR duplicates after mapping.\n",
      "  Will remove PCR duplicates at cell level.\n",
      "  Won't allocate multi-mappings after mapping.\n",
      "  Only output unique mappings after mapping.\n",
      "  Only output mappings of which barcodes are in whitelist.\n",
      "  Perform Tn5 shift.\n",
      "  Output mappings in BED/BEDPE format.\n",
      "  Reference file: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\n",
      "  Index file: chromap_index/index\n",
      "  1th read 1 file: na.na\n",
      "  2th read 1 file: na1.na\n",
      "  1th read 2 file: na2.na\n",
      "  2th read 2 file: na3.na\n",
      "  1th cell barcode file: na4.na\n",
      "  2th cell barcode file: na5.na\n",
      "  Cell barcode whitelist file: sai_192_whitelist\n",
      "  Output file: na.atac.filter.fragments.tsv\n",
      "  Loaded all sequences successfully in 24.15s, number of sequences: 195, number of bases: 3099922541.\n",
      "  chromap: src/index.cc:135: void chromap::Index::Load(): Assertion `index_file != nullptr' failed.\n",
      "  /fusion/gs/eila-nextflow-bucket/work/ATAC/d8/e8a678b3e424473ec756122b5d802a/.command.sh: line 21:   117 Aborted                 (core dumped) chromap -x chromap_index/index --trim-adapters --remove-pcr-duplicates --remove-pcr-duplicates-at-cell-level --Tn5-shift --low-mem --BED -l 2000 --bc-error-threshold 2 --bc-probability-threshold 0.9 --read-format bc:0:-1,r1:0:-1,r2:0:-1 --drop-repetitive-reads 4 -x chromap_index/index -r GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -q 0 -t 1 -1 na.na,na1.na -2 na2.na,na3.na -b na4.na,na5.na --barcode-whitelist sai_192_whitelist -o na.atac.filter.fragments.tsv --summary na.atac.align.barcode.summary.csv\n",
      "\n",
      "Work dir:\n",
      "  gs://eila-nextflow-bucket/work/ATAC/d8/e8a678b3e424473ec756122b5d802a\n",
      "\n",
      "Tip: you can replicate the issue by changing to the process work dir and entering the command `bash .command.run`\n",
      "Jan-24 12:55:13.286 [main] DEBUG nextflow.Session - Session await > all processes finished\n",
      "Jan-24 12:55:13.508 [Task monitor] DEBUG nextflow.Session - Session aborted -- Cause: Process `run_chromap_map_to_idx (1)` terminated with an error exit status (134)\n",
      "Jan-24 12:55:13.807 [main] DEBUG nextflow.Session - Session await > all barriers passed\n",
      "Jan-24 12:55:13.807 [Task monitor] DEBUG n.processor.TaskPollingMonitor - <<< barrier arrives (monitor: google-batch) - terminating tasks monitor poll loop\n",
      "Jan-24 12:55:14.056 [main] DEBUG n.trace.WorkflowStatsObserver - Workflow completed > WorkflowStats[succeededCount=0; failedCount=1; ignoredCount=0; cachedCount=2; pendingCount=0; submittedCount=0; runningCount=0; retriesCount=0; abortedCount=0; succeedDuration=0ms; failedDuration=22.2s; cachedDuration=55.9s;loadCpus=0; loadMemory=0; peakRunning=1; peakCpus=1; peakMemory=0; ]\n",
      "Jan-24 12:55:14.187 [main] WARN  i.s.tower.plugin.TowerJsonGenerator - Tower request field `workflow.errorMessage` exceeds expected size | offending value: `Start to map reads.\n",
      "na.na\n",
      "na1.na\n",
      "na2.na\n",
      "na3.na\n",
      "na4.na\n",
      "na5.na\n",
      "Parameters: error threshold: 8, min-num-seeds: 2, max-seed-frequency: 500,1000, max-num-best-mappings: 1, max-insert-size: 2000, MAPQ-threshold: 0, min-read-length: 30, bc-error-threshold: 2, bc-probability-threshold: 0.90\n",
      "Number of threads: 1\n",
      "Analyze single-cell data.\n",
      "Will try to remove adapters on 3'.\n",
      "Will remove PCR duplicates after mapping.\n",
      "Will remove PCR duplicates at cell level.\n",
      "Won't allocate multi-mappings after mapping.\n",
      "Only output unique mappings after mapping.\n",
      "Only output mappings of which barcodes are in whitelist.\n",
      "Perform Tn5 shift.\n",
      "Output mappings in BED/BEDPE format.\n",
      "Reference file: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\n",
      "Index file: chromap_index/index\n",
      "1th read 1 file: na.na\n",
      "2th read 1 file: na1.na\n",
      "1th read 2 file: na2.na\n",
      "2th read 2 file: na3.na\n",
      "1th cell barcode file: na4.na\n",
      "2th cell barcode file: na5.na\n",
      "Cell barcode whitelist file: sai_192_whitelist\n",
      "Output file: na.atac.filter.fragments.tsv\n",
      "Loaded all sequences successfully in 24.15s, number of sequences: 195, number of bases: 3099922541.\n",
      "chromap: src/index.cc:135: void chromap::Index::Load(): Assertion `index_file != nullptr' failed.\n",
      "/fusion/gs/eila-nextflow-bucket/work/ATAC/d8/e8a678b3e424473ec756122b5d802a/.command.sh: line 21:   117 Aborted                 (core dumped) chromap -x chromap_index/index --trim-adapters --remove-pcr-duplicates --remove-pcr-duplicates-at-cell-level --Tn5-shift --low-mem --BED -l 2000 --bc-error-threshold 2 --bc-probability-threshold 0.9 --read-format bc:0:-1,r1:0:-1,r2:0:-1 --drop-repetitive-reads 4 -x chromap_index/index -r GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -q 0 -t 1 -1 na.na,na1.na -2 na2.na,na3.na -b na4.na,na5.na --barcode-whitelist sai_192_whitelist -o na.atac.filter.fragments.tsv --summary na.atac.align.barcode.summary.csv`, size: 1847 (max: 255)\n",
      "Jan-24 12:55:14.608 [main] DEBUG nextflow.cache.CacheDB - Closing CacheDB done\n",
      "Jan-24 12:55:14.609 [main] INFO  org.pf4j.AbstractPluginManager - Stop plugin 'nf-wave@1.0.1'\n",
      "Jan-24 12:55:14.609 [main] DEBUG nextflow.plugin.BasePlugin - Plugin stopped nf-wave\n",
      "Jan-24 12:55:14.609 [main] INFO  org.pf4j.AbstractPluginManager - Stop plugin 'nf-tower@1.6.3'\n",
      "Jan-24 12:55:14.609 [main] DEBUG nextflow.plugin.BasePlugin - Plugin stopped nf-tower\n",
      "Jan-24 12:55:14.609 [main] INFO  org.pf4j.AbstractPluginManager - Stop plugin 'nf-google@1.8.3'\n",
      "Jan-24 12:55:14.609 [main] DEBUG nextflow.plugin.BasePlugin - Plugin stopped nf-google\n",
      "Jan-24 12:55:14.655 [main] DEBUG nextflow.script.ScriptRunner - > Execution complete -- Goodbye\n"
     ]
    }
   ],
   "source": [
    "!cat .nextflow.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
