{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3392,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# start docker deamon in terminal\n",
    "# !sudo docker login\n",
    "# OR - becuase the jupyther with allow root\n",
    "# !docker login"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "TODO:\n",
    "\n",
    "when running with cloud. \n",
    "if you dont need to keep the execution logs, change the cloud settings to keep it for a short while\n",
    "Check that the cache if you need to set the cache to false (especially in plot process)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3393,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3394,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/hai-gcp-genomic-beb9207a40b8.json\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "json_path = os.path.join(os.path.dirname(os.getcwd()),\"hai-gcp-genomic-beb9207a40b8.json\")\n",
    "!ls $json_path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3395,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Activated service account credentials for: [eila-batch@hai-gcp-genomic.iam.gserviceaccount.com]\n"
     ]
    }
   ],
   "source": [
    "# # in the command line. same env\n",
    "import os\n",
    "json_path = os.path.join(os.path.dirname(os.getcwd()),\"hai-gcp-genomic-beb9207a40b8.json\")\n",
    "os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = json_path\n",
    "!gcloud auth activate-service-account eila-batch@hai-gcp-genomic.iam.gserviceaccount.com --key-file=$GOOGLE_APPLICATION_CREDENTIALS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3396,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                  Credentialed Accounts\n",
      "ACTIVE  ACCOUNT\n",
      "*       eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\n",
      "        eila@stanford.edu\n",
      "\n",
      "To set the active account, run:\n",
      "    $ gcloud config set account `ACCOUNT`\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'hai-gcp-genomic'"
      ]
     },
     "execution_count": 3396,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "!gcloud auth list\n",
    "project_name = !gcloud config get-value project\n",
    "project_name = project_name[0]\n",
    "project_name\n",
    "# gcloud config set account\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3421,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ServiceException: 401 Anonymous caller does not have storage.objects.list access to the Google Cloud Storage bucket. Permission 'storage.objects.list' denied on resource (or it may not exist).\n"
     ]
    }
   ],
   "source": [
    "!gsutil ls gs://eila-nextflow-bucket"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3398,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "os.environ[\"TOWER_ACCESS_TOKEN\"] = 'eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz'\n",
    "os.environ[\"TOWER_WORKSPACE_ID\"]='247507587970818'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3399,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz\n",
      "247507587970818\n"
     ]
    }
   ],
   "source": [
    "!echo $TOWER_ACCESS_TOKEN\n",
    "!echo $TOWER_WORKSPACE_ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3400,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3401,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/bin'"
      ]
     },
     "execution_count": 3401,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# !chmod +x ./bin/scrna_qc_metrics.py\n",
    "import os\n",
    "bin_fld = os.path.join(os.getcwd(),\"bin\")\n",
    "bin_fld"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3402,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "!chmod +x {os.path.join(bin_fld, \"atac_plot_metadata.R\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"run_tabix.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"plot_insert_size_hist.py\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"atac_generate_merged_barcode_metadata.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"create_sample_filtered_fragment_file.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"compute_tss_enrichment_bulk.py\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"barcode_rank_functions.R\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"plot_insert_size_hist.py\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"compute_tss_enrichment_snapatac2.py\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"add_subpool_to_fragments_table.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"add_subpool_to_summary_file.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"filter_script.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"merge_chromap_logs.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"generate_insert_size_histogram_data.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"calculate_barcode_metadata_stats.sh\")}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3403,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# # TODO: For a new user: run these commands with your service account for the execution. I am using eila-batch@...\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/iam.serviceAccountUser\"\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/batch.jobsEditor\"\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/logging.viewer\"\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3404,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !nextflow -self-update"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3405,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  Version: 23.10.1 build 5891\n",
      "  Created: 12-01-2024 22:01 UTC (17:01 EDT)\n",
      "  System: Mac OS X 14.3\n",
      "  Runtime: Groovy 3.0.19 on OpenJDK 64-Bit Server VM 17.0.8+7-LTS\n",
      "  Encoding: UTF-8 (UTF-8)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!nextflow info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3406,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "\n",
    "# Get the current working directory\n",
    "current_directory = os.getcwd()\n",
    "\n",
    "# Navigate two folders above\n",
    "two_folders_above = os.path.abspath(os.path.join(current_directory, '..', '..'))\n",
    "\n",
    "nf_data_base_path = os.path.join(two_folders_above,'nf_data')\n",
    "# Print the result\n",
    "print(nf_data_base_path)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3407,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data'"
      ]
     },
     "execution_count": 3407,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nf_data_base_path\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3408,
   "metadata": {},
   "outputs": [],
   "source": [
    "nf_data_project=\"test_atac_pipeline\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3409,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/local_atac_fastqs_files.tsv'"
      ]
     },
     "execution_count": 3409,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.path.exists(os.path.join(nf_data_base_path,nf_data_project,\"local_atac_fastqs_files.tsv\"))\n",
    "os.path.join(nf_data_base_path,nf_data_project,\"local_atac_fastqs_files.tsv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3410,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "\n",
    "# Write the configuration file\n",
    "with open(\"nextflow_atac.config\", \"w\") as config_file:\n",
    "    config_file.write(f\"\"\"\n",
    "manifest {{\n",
    "  description = 'A nexflow pipeline for fg single cell atac' \n",
    "  mainScript = 'nf_wf_fg_singlecell_atac.nf'\n",
    "}}\n",
    "\n",
    "profiles {{\n",
    "  google_batch_profile {{\n",
    "    workDir = 'gs://eila-nextflow-bucket/work/ATAC'\n",
    "    includeConfig 'conf/docker-google-batch.config'\n",
    "    docker.enabled = true\n",
    "    wave.enabled = true\n",
    "    fusion.enabled = true\n",
    "    process.scratch = false\n",
    "    executor = 'google-batch'\n",
    "    docker.envWhitelist=['ENV_SYNAPSE_TOKEN','TOWER_ACCESS_TOKEN']\n",
    "    google.project = '{project_name}'\n",
    "    google.location = 'us-central1'\n",
    "    google.batch.spot = true\n",
    "    google.batch.serviceAccountEmail='eila-batch@hai-gcp-genomic.iam.gserviceaccount.com'\n",
    "  }}\n",
    "}}\n",
    "\n",
    "// Define ENV_SYNAPSE_TOKEN as a parameter\n",
    "params.FASTQS_SPEC_CH = '{nf_data_base_path}/{nf_data_project}/local_atac_fastqs_files.tsv'\n",
    "// CHROMAP\n",
    "// 6a\n",
    "params.CHROMAP_IDX='gs://eila-nextflow-bucket/test_atac_pipeline/ref.index'\n",
    "// 6b:\n",
    "// FASTA file\n",
    "params.CHROMAP_READ_LENGTH = 2000\n",
    "params.CHROMAP_BC_ERROR_THRESHOLD = 2\n",
    "params.CHROMAP_BC_PROBABILITY_THRESHOLD = 0.9\n",
    "params.CHROMAP_READ_FORMAT = 'bc:0:-1,r1:0:-1,r2:0:-1'\n",
    "params.CHROMAP_DROP_REPETITIVE_READS = 4\n",
    "params.CHROMAP_QUALITY_THRESHOLD = 0\n",
    "params.CHROMAP_MERGE_LOGS_SCRIPT = \"merge_chromap_logs.sh\"\n",
    "params.CHROMAP_FRAGMENTS_HISTOGRAM_DATA_SCRIPT='generate_insert_size_histogram_data.sh'\n",
    "params.CHROMAP_GENOME_REFERENCE_FASTA='{nf_data_base_path}/{nf_data_project}/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz'\n",
    "\n",
    "// CHROMAP DEBUG: load output files \n",
    "params.chromap_filter_fragments_tsv ='{nf_data_base_path}/{nf_data_project}/chromap_out_debug/ATAC_BMMC-single-donor.atac.filter.fragments.tsv'\n",
    "params.barcode_summary_csv ='{nf_data_base_path}/{nf_data_project}/chromap_out_debug/BMMC-single-donor.atac.align.barcode.summary.csv'\n",
    "params.chromap_alignment_log_out = '{nf_data_base_path}/{nf_data_project}/chromap_out_debug/BMMC-single-donor.atac.align.k4.hg38.log.txt'\n",
    "\n",
    "\n",
    "\n",
    "params.TSV_ADD_SUBPOOL_SCRIPT = 'add_subpool_to_fragments_table.sh'\n",
    "params.CSV_ADD_SUBPOOL_SCRIPT = 'add_subpool_to_summary_file.sh'\n",
    "params.FILTER_FRAGMENTS_SCRIPT = 'filter_script.sh'\n",
    "params.ATAC_FRAGMENTS_MIN_FRAG_CUTOFF=10\n",
    "\n",
    "// Other steps\n",
    "\n",
    "params.GENOME_GZ_GTF='http://ftp.ensembl.org/pub/release-109/gtf/homo_sapiens/Homo_sapiens.GRCh38.109.gtf.gz'\n",
    "params.ENV_SYNAPSE_TOKEN = \"nextflow secrets get SYNAPSE\".execute().text.trim()\n",
    "\n",
    "\n",
    "params.MAPQ_THRESHOLD = false\n",
    "params.BARCODE_TRANSLATE = 'path'\n",
    "params.ATAC_FRAGMENTS_CUTOFF = 4\n",
    "params.RUN_TABIX_SCRIPT='run_tabix.sh'\n",
    "params.ATAC_TSS_BASES_FLANK = 2000\n",
    "params.ATAC_TSS_COL_WITH_STRANDS_INFO = 4\n",
    "params.ATAC_TSS_SMOOTHING_WINDOW_SIZE = 20\n",
    "params.ATAC_TSS_REGION_BED_FILE='{nf_data_base_path}/{nf_data_project}/hg38.synapse_transcripts.TSS.bed'\n",
    "params.ATAC_TSS_BULK_CALCULATION_SCRIPT='compute_tss_enrichment_bulk.py'\n",
    "params.ATAC_TSS_SNAPATAC2_CALCULATION_SCRIPT='compute_tss_enrichment_snapatac2.py'\n",
    "params.ATAC_TSS_SNAPATAC2_GENES_GTF='{nf_data_base_path}/{nf_data_project}/genes.gtf'\n",
    "params.ATAC_TSS_SNAPATAC2_MIN_FRAG_CUTOFF = 10\n",
    "\n",
    "params.ATAC_INSERT_SIZE_PLOT_SCRIPT='plot_insert_size_hist.py'\n",
    "\n",
    "// atac calculate_barcode_metadata_stats\n",
    "params.ATAC_SNAP_METADATA_STATS_SCRIPT='calculate_barcode_metadata_stats.sh'\n",
    "// atac plot barcode metadata\n",
    "params.ATAC_PLOT_METADATA_SCRIPT='atac_plot_metadata.R'\n",
    "params.ATAC_PLOT_METADATA_HELPER_SCRIPT='barcode_rank_functions.R'\n",
    "\n",
    "// debug file\n",
    "// params.SC_ATAC_QC_BARCODE_METADATA_FILE='{nf_data_base_path}/{nf_data_project}/barcode_metadata.txt'\n",
    "params.ATAC_QC_FRAGMENT_CUTOFF=10\n",
    "params.SC_ATAC_QC_BARCODE_OUTPUT_FILE='atac_barcode_metadata_output_file.png'\n",
    "    \n",
    "// params.SC_ATAC_BARCODE_METADATA_SCRIPT='atac_generate_merged_barcode_metadata.sh'\n",
    "// DEBUG: get the data from a file\n",
    "// params.SC_ATAC_BARCODE_METADATA_FILTERED_BARCODE_STATS='{nf_data_base_path}nf_wf_fg_singlecell/filtered_barcode_stats.csv'\n",
    "// params.SC_ATAC_BARCODE_METADATA_TSS_ENRICHMENT_BARCODE_STATS='{nf_data_base_path}nf_wf_fg_singlecell/tss_enrichment_barcode_stats.csv'\n",
    "\n",
    "// params.SC_ATAC_GENERATE_BARCODE_RANK_PLOT_PKR='PKR123'\n",
    "\n",
    "// params.SC_JOINT_CELL_BARCODE_PLOT_SCRTIP_PY='joint_cell_plotting.py'\n",
    "// params.SC_JOINT_CELL_BARCODE_PLOT_SCRTIP_R='joint_cell_plotting_density.R'\n",
    "\"\"\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3411,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !rm -r /Users/eilaarich-landkof-stanford/Documents/Code/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/work\n",
    "# !rm -r /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/work\n",
    "# !gsutil rm -r gs://eila-nextflow-bucket/work/ATAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3412,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !rm .nextflow*\n",
    "\n",
    "# !ls /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/local_atac_fastqs_files.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3413,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "N E X T F L O W  ~  version 23.10.1\n",
      "Launching `nf_wf_fg_singlecell_atac.nf` [nauseous_maxwell] DSL2 - revision: 16a69dd583\n",
      "/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/local_atac_fastqs_files.tsv\n",
      "eyJ0eXAiOiJKV1QiLCJraWQiOiJXN05OOldMSlQ6SjVSSzpMN1RMOlQ3TDc6M1ZYNjpKRU9VOjY0NFI6VTNJWDo1S1oyOjdaQ0s6RlBUSCIsImFsZyI6IlJTMjU2In0.eyJhY2Nlc3MiOnsic2NvcGUiOlsidmlldyIsImRvd25sb2FkIl0sIm9pZGNfY2xhaW1zIjp7fX0sInRva2VuX3R5cGUiOiJQRVJTT05BTF9BQ0NFU1NfVE9LRU4iLCJpc3MiOiJodHRwczovL3JlcG8tcHJvZC5wcm9kLnNhZ2ViYXNlLm9yZy9hdXRoL3YxIiwiYXVkIjoiMCIsIm5iZiI6MTY5NzgxODkxNiwiaWF0IjoxNjk3ODE4OTE2LCJqdGkiOiI0MTIzIiwic3ViIjoiMzQ3NDU3MSJ9.n0cugYMAVzYZ8mMMc7CjcqqEV7FB-K-oKUDT45SvM6KmbCpb_v3lZ2gg3xXi1tubWH0__wpnbJ7cYrGEtOyt4vXAPkMiEtZFGqS0MFKp_h-mHHH4pnACSQQYEy1zdyqFn9WRE7ipjWg4yYSbq84N7J3gzi8VKmiOAmuIGKaJaS0snkUjtJEVQH-0vZqWs1nKgVRWIZv31Bo8vuDzWpBrXD-zRT_uQW9l_I9fU50o1AqyFhNe_gbty_Z6imb8VT33wMXiDgJh15lG_M70PXU15v3egDXLgw1dGj32kj1nKanfqBVqbMSzgzPAG037p3YH_Q556BLceFBBb5Jaej5JVA\n",
      "after genome_fasta_ch\n",
      "[-        ] process > run_zcat                       -\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip           -\u001b[K\n",
      "[-        ] process > run_add_subpool_prefix_to_f... -\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_gzip_on_genes_gtf          -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "\u001b[38A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/4YUdmqfnbjPozx\u001b[22;39m\u001b[K\n",
      "[-        ] process > run_zcat                       -\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip           -\u001b[K\n",
      "[-        ] process > run_add_subpool_prefix_to_f... -\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_gzip_on_genes_gtf          -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "\u001b[39A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/4YUdmqfnbjPozx\u001b[22;39m\u001b[K\n",
      "[3f/f09a4d] process > run_zcat                       [100%] 1 of 1, cached: 1\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip           -\u001b[K\n",
      "[-        ] process > run_add_subpool_prefix_to_f... -\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[-        ] process > run_gzip_on_genes_gtf          -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "\u001b[39A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/4YUdmqfnbjPozx\u001b[22;39m\u001b[K\n",
      "[3f/f09a4d] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip           -\u001b[K\n",
      "[-        ] process > run_add_subpool_prefix_to_f... -\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[67/a1b7db] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb  7 20:07 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/eb/6412854cc6c199859cdfbe280f86c9/genes.gtf\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb  7 20:07 genes.gtf.gz\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "\u001b[58A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/4YUdmqfnbjPozx\u001b[22;39m\u001b[K\n",
      "[3f/f09a4d] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip           -\u001b[K\n",
      "[-        ] process > run_add_subpool_prefix_to_f... -\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[67/a1b7db] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb  7 20:07 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/eb/6412854cc6c199859cdfbe280f86c9/genes.gtf\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb  7 20:07 genes.gtf.gz\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "\u001b[58A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/4YUdmqfnbjPozx\u001b[22;39m\u001b[K\n",
      "[3f/f09a4d] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[15/0ac619] process > run_whitelist_gunzip (1)       [100%] 1 of 1, cached: 1\u001b[K\n",
      "[-        ] process > run_add_subpool_prefix_to_f... -\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[67/a1b7db] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb  7 20:07 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/eb/6412854cc6c199859cdfbe280f86c9/genes.gtf\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb  7 20:07 genes.gtf.gz\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "\u001b[58A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/4YUdmqfnbjPozx\u001b[22;39m\u001b[K\n",
      "[3f/f09a4d] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[15/0ac619] process > run_whitelist_gunzip (1)       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_add_subpool_prefix_to_f... -\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[67/a1b7db] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb  7 20:07 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/eb/6412854cc6c199859cdfbe280f86c9/genes.gtf\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb  7 20:07 genes.gtf.gz\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "Start run_whitelist_gunzip\u001b[K\n",
      "Gunzip file is: sai_192_whitelist.txt\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 172805\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/d5/12b6921f983d2326fa1e91570a96e7/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/6f/e3ee295a7f5364c39365c48e8bd0b0/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/4a/7b5d74b3a4903977d7316c5f02ef84/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/2d/0e59fd7f9797eaf684de4abc52eb46/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/c5/cc9149d8d21bb423337888a9dc77ea/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/dc/65201d13e12dffd8779bbc935604a6/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb  7 20:08 na.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/f4/bab8ffeff41eecffb2ddcf013b4121/na.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       125 Feb  7 20:08 na1.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/38/b73884d031cf00d6089d843ba0f965/na1.na\u001b[K\n",
      "-r-x--x--x 1 root root 176947200 Feb  7 20:08 sai_192_whitelist\u001b[K\n",
      "lrwxrwxrwx 1 root root       140 Feb  7 20:08 sai_192_whitelist.txt -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/09/7ba70bbeac00bcd6547ad0c09655c7/sai_192_whitelist.txt\u001b[K\n",
      "Finished run_whitelist_gunzip\u001b[K\n",
      "\u001b[K\n",
      "\u001b[75A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/4YUdmqfnbjPozx\u001b[22;39m\u001b[K\n",
      "[3f/f09a4d] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[15/0ac619] process > run_whitelist_gunzip (1)       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[59/d22a1d] process > run_add_subpool_prefix_to_f... [100%] 1 of 1, cached: 1\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[67/a1b7db] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb  7 20:07 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/eb/6412854cc6c199859cdfbe280f86c9/genes.gtf\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb  7 20:07 genes.gtf.gz\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "Start run_whitelist_gunzip\u001b[K\n",
      "Gunzip file is: sai_192_whitelist.txt\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 172805\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/d5/12b6921f983d2326fa1e91570a96e7/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/6f/e3ee295a7f5364c39365c48e8bd0b0/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/4a/7b5d74b3a4903977d7316c5f02ef84/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/2d/0e59fd7f9797eaf684de4abc52eb46/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/c5/cc9149d8d21bb423337888a9dc77ea/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/dc/65201d13e12dffd8779bbc935604a6/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb  7 20:08 na.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/f4/bab8ffeff41eecffb2ddcf013b4121/na.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       125 Feb  7 20:08 na1.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/38/b73884d031cf00d6089d843ba0f965/na1.na\u001b[K\n",
      "-r-x--x--x 1 root root 176947200 Feb  7 20:08 sai_192_whitelist\u001b[K\n",
      "lrwxrwxrwx 1 root root       140 Feb  7 20:08 sai_192_whitelist.txt -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/09/7ba70bbeac00bcd6547ad0c09655c7/sai_192_whitelist.txt\u001b[K\n",
      "Finished run_whitelist_gunzip\u001b[K\n",
      "\u001b[K\n",
      "\u001b[75A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/4YUdmqfnbjPozx\u001b[22;39m\u001b[K\n",
      "[3f/f09a4d] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[15/0ac619] process > run_whitelist_gunzip (1)       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[59/d22a1d] process > run_add_subpool_prefix_to_f... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... -\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[67/a1b7db] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb  7 20:07 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/eb/6412854cc6c199859cdfbe280f86c9/genes.gtf\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb  7 20:07 genes.gtf.gz\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "Start run_whitelist_gunzip\u001b[K\n",
      "Gunzip file is: sai_192_whitelist.txt\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 172805\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/d5/12b6921f983d2326fa1e91570a96e7/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/6f/e3ee295a7f5364c39365c48e8bd0b0/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/4a/7b5d74b3a4903977d7316c5f02ef84/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/2d/0e59fd7f9797eaf684de4abc52eb46/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/c5/cc9149d8d21bb423337888a9dc77ea/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/dc/65201d13e12dffd8779bbc935604a6/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb  7 20:08 na.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/f4/bab8ffeff41eecffb2ddcf013b4121/na.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       125 Feb  7 20:08 na1.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/38/b73884d031cf00d6089d843ba0f965/na1.na\u001b[K\n",
      "-r-x--x--x 1 root root 176947200 Feb  7 20:08 sai_192_whitelist\u001b[K\n",
      "lrwxrwxrwx 1 root root       140 Feb  7 20:08 sai_192_whitelist.txt -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/09/7ba70bbeac00bcd6547ad0c09655c7/sai_192_whitelist.txt\u001b[K\n",
      "Finished run_whitelist_gunzip\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "Input chromap_filter_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_fragments_table.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "\u001b[K\n",
      "\u001b[83A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/4YUdmqfnbjPozx\u001b[22;39m\u001b[K\n",
      "[3f/f09a4d] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[15/0ac619] process > run_whitelist_gunzip (1)       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[59/d22a1d] process > run_add_subpool_prefix_to_f... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[-        ] process > run_merge_chromap_logs         -\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[67/a1b7db] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb  7 20:07 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/eb/6412854cc6c199859cdfbe280f86c9/genes.gtf\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb  7 20:07 genes.gtf.gz\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "Start run_whitelist_gunzip\u001b[K\n",
      "Gunzip file is: sai_192_whitelist.txt\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 172805\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/d5/12b6921f983d2326fa1e91570a96e7/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/6f/e3ee295a7f5364c39365c48e8bd0b0/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/4a/7b5d74b3a4903977d7316c5f02ef84/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/2d/0e59fd7f9797eaf684de4abc52eb46/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/c5/cc9149d8d21bb423337888a9dc77ea/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/dc/65201d13e12dffd8779bbc935604a6/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb  7 20:08 na.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/f4/bab8ffeff41eecffb2ddcf013b4121/na.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       125 Feb  7 20:08 na1.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/38/b73884d031cf00d6089d843ba0f965/na1.na\u001b[K\n",
      "-r-x--x--x 1 root root 176947200 Feb  7 20:08 sai_192_whitelist\u001b[K\n",
      "lrwxrwxrwx 1 root root       140 Feb  7 20:08 sai_192_whitelist.txt -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/09/7ba70bbeac00bcd6547ad0c09655c7/sai_192_whitelist.txt\u001b[K\n",
      "Finished run_whitelist_gunzip\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "Input chromap_filter_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_fragments_table.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "\u001b[K\n",
      "\u001b[83A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/4YUdmqfnbjPozx\u001b[22;39m\u001b[K\n",
      "[3f/f09a4d] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[15/0ac619] process > run_whitelist_gunzip (1)       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[59/d22a1d] process > run_add_subpool_prefix_to_f... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[a9/95c478] process > run_merge_chromap_logs (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_process_conversion_to_b... -\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[67/a1b7db] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb  7 20:07 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/eb/6412854cc6c199859cdfbe280f86c9/genes.gtf\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb  7 20:07 genes.gtf.gz\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "Start run_whitelist_gunzip\u001b[K\n",
      "Gunzip file is: sai_192_whitelist.txt\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 172805\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/d5/12b6921f983d2326fa1e91570a96e7/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/6f/e3ee295a7f5364c39365c48e8bd0b0/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/4a/7b5d74b3a4903977d7316c5f02ef84/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/2d/0e59fd7f9797eaf684de4abc52eb46/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/c5/cc9149d8d21bb423337888a9dc77ea/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/dc/65201d13e12dffd8779bbc935604a6/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb  7 20:08 na.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/f4/bab8ffeff41eecffb2ddcf013b4121/na.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       125 Feb  7 20:08 na1.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/38/b73884d031cf00d6089d843ba0f965/na1.na\u001b[K\n",
      "-r-x--x--x 1 root root 176947200 Feb  7 20:08 sai_192_whitelist\u001b[K\n",
      "lrwxrwxrwx 1 root root       140 Feb  7 20:08 sai_192_whitelist.txt -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/09/7ba70bbeac00bcd6547ad0c09655c7/sai_192_whitelist.txt\u001b[K\n",
      "Finished run_whitelist_gunzip\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "Input chromap_filter_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_fragments_table.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "\u001b[K\n",
      "start run_merge_logs\u001b[K\n",
      "input script_name is merge_chromap_logs.sh\u001b[K\n",
      "input chromap_alignment_log is BMMC-single-donor.atac.align.k4.hg38.log.txt\u001b[K\n",
      "input chromap_barcode_log is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "start run_merge_chromap_logs shell script\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "********1***********\u001b[K\n",
      "********2***********\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "uni-mappings,260650161\u001b[K\n",
      "multi-mappings,14666503\u001b[K\n",
      "total,275316664\u001b[K\n",
      "********3***********\u001b[K\n",
      "finished run_merge_chromap_logs shell script\u001b[K\n",
      "finished run_merge_logs\u001b[K\n",
      "\u001b[K\n",
      "\u001b[119A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/4YUdmqfnbjPozx\u001b[22;39m\u001b[K\n",
      "[3f/f09a4d] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[15/0ac619] process > run_whitelist_gunzip (1)       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[59/d22a1d] process > run_add_subpool_prefix_to_f... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[a9/95c478] process > run_merge_chromap_logs (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3c/fca9ef] process > run_process_conversion_to_b... [100%] 1 of 1, cached: 1\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[67/a1b7db] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb  7 20:07 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/eb/6412854cc6c199859cdfbe280f86c9/genes.gtf\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb  7 20:07 genes.gtf.gz\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "Start run_whitelist_gunzip\u001b[K\n",
      "Gunzip file is: sai_192_whitelist.txt\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 172805\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/d5/12b6921f983d2326fa1e91570a96e7/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/6f/e3ee295a7f5364c39365c48e8bd0b0/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/4a/7b5d74b3a4903977d7316c5f02ef84/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/2d/0e59fd7f9797eaf684de4abc52eb46/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/c5/cc9149d8d21bb423337888a9dc77ea/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/dc/65201d13e12dffd8779bbc935604a6/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb  7 20:08 na.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/f4/bab8ffeff41eecffb2ddcf013b4121/na.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       125 Feb  7 20:08 na1.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/38/b73884d031cf00d6089d843ba0f965/na1.na\u001b[K\n",
      "-r-x--x--x 1 root root 176947200 Feb  7 20:08 sai_192_whitelist\u001b[K\n",
      "lrwxrwxrwx 1 root root       140 Feb  7 20:08 sai_192_whitelist.txt -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/09/7ba70bbeac00bcd6547ad0c09655c7/sai_192_whitelist.txt\u001b[K\n",
      "Finished run_whitelist_gunzip\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "Input chromap_filter_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_fragments_table.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "\u001b[K\n",
      "start run_merge_logs\u001b[K\n",
      "input script_name is merge_chromap_logs.sh\u001b[K\n",
      "input chromap_alignment_log is BMMC-single-donor.atac.align.k4.hg38.log.txt\u001b[K\n",
      "input chromap_barcode_log is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "start run_merge_chromap_logs shell script\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "********1***********\u001b[K\n",
      "********2***********\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "uni-mappings,260650161\u001b[K\n",
      "multi-mappings,14666503\u001b[K\n",
      "total,275316664\u001b[K\n",
      "********3***********\u001b[K\n",
      "finished run_merge_chromap_logs shell script\u001b[K\n",
      "finished run_merge_logs\u001b[K\n",
      "\u001b[K\n",
      "\u001b[119A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/4YUdmqfnbjPozx\u001b[22;39m\u001b[K\n",
      "[3f/f09a4d] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[15/0ac619] process > run_whitelist_gunzip (1)       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[59/d22a1d] process > run_add_subpool_prefix_to_f... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[a9/95c478] process > run_merge_chromap_logs (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3c/fca9ef] process > run_process_conversion_to_b... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_filter_aligned_fragments   -\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[67/a1b7db] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb  7 20:07 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/eb/6412854cc6c199859cdfbe280f86c9/genes.gtf\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb  7 20:07 genes.gtf.gz\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "Start run_whitelist_gunzip\u001b[K\n",
      "Gunzip file is: sai_192_whitelist.txt\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 172805\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/d5/12b6921f983d2326fa1e91570a96e7/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/6f/e3ee295a7f5364c39365c48e8bd0b0/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/4a/7b5d74b3a4903977d7316c5f02ef84/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/2d/0e59fd7f9797eaf684de4abc52eb46/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/c5/cc9149d8d21bb423337888a9dc77ea/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/dc/65201d13e12dffd8779bbc935604a6/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb  7 20:08 na.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/f4/bab8ffeff41eecffb2ddcf013b4121/na.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       125 Feb  7 20:08 na1.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/38/b73884d031cf00d6089d843ba0f965/na1.na\u001b[K\n",
      "-r-x--x--x 1 root root 176947200 Feb  7 20:08 sai_192_whitelist\u001b[K\n",
      "lrwxrwxrwx 1 root root       140 Feb  7 20:08 sai_192_whitelist.txt -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/09/7ba70bbeac00bcd6547ad0c09655c7/sai_192_whitelist.txt\u001b[K\n",
      "Finished run_whitelist_gunzip\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "Input chromap_filter_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_fragments_table.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "\u001b[K\n",
      "start run_merge_logs\u001b[K\n",
      "input script_name is merge_chromap_logs.sh\u001b[K\n",
      "input chromap_alignment_log is BMMC-single-donor.atac.align.k4.hg38.log.txt\u001b[K\n",
      "input chromap_barcode_log is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "start run_merge_chromap_logs shell script\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "********1***********\u001b[K\n",
      "********2***********\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "uni-mappings,260650161\u001b[K\n",
      "multi-mappings,14666503\u001b[K\n",
      "total,275316664\u001b[K\n",
      "********3***********\u001b[K\n",
      "finished run_merge_chromap_logs shell script\u001b[K\n",
      "finished run_merge_logs\u001b[K\n",
      "\u001b[K\n",
      "start run_process_conversion_to_barcode\u001b[K\n",
      "input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "------ There is NO conversion list ------\u001b[K\n",
      "temp_conversion\u001b[K\n",
      "7021039 temp_conversion\u001b[K\n",
      "finished run_process_conversion_to_barcode\u001b[K\n",
      "\u001b[K\n",
      "\u001b[126A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/4YUdmqfnbjPozx\u001b[22;39m\u001b[K\n",
      "[3f/f09a4d] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[15/0ac619] process > run_whitelist_gunzip (1)       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[59/d22a1d] process > run_add_subpool_prefix_to_f... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[a9/95c478] process > run_merge_chromap_logs (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3c/fca9ef] process > run_process_conversion_to_b... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3b/05be46] process > run_filter_aligned_fragment... [100%] 1 of 1, cached: 1\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[67/a1b7db] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb  7 20:07 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/eb/6412854cc6c199859cdfbe280f86c9/genes.gtf\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb  7 20:07 genes.gtf.gz\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "Start run_whitelist_gunzip\u001b[K\n",
      "Gunzip file is: sai_192_whitelist.txt\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 172805\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/d5/12b6921f983d2326fa1e91570a96e7/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/6f/e3ee295a7f5364c39365c48e8bd0b0/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/4a/7b5d74b3a4903977d7316c5f02ef84/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/2d/0e59fd7f9797eaf684de4abc52eb46/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/c5/cc9149d8d21bb423337888a9dc77ea/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/dc/65201d13e12dffd8779bbc935604a6/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb  7 20:08 na.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/f4/bab8ffeff41eecffb2ddcf013b4121/na.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       125 Feb  7 20:08 na1.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/38/b73884d031cf00d6089d843ba0f965/na1.na\u001b[K\n",
      "-r-x--x--x 1 root root 176947200 Feb  7 20:08 sai_192_whitelist\u001b[K\n",
      "lrwxrwxrwx 1 root root       140 Feb  7 20:08 sai_192_whitelist.txt -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/09/7ba70bbeac00bcd6547ad0c09655c7/sai_192_whitelist.txt\u001b[K\n",
      "Finished run_whitelist_gunzip\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "Input chromap_filter_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_fragments_table.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "\u001b[K\n",
      "start run_merge_logs\u001b[K\n",
      "input script_name is merge_chromap_logs.sh\u001b[K\n",
      "input chromap_alignment_log is BMMC-single-donor.atac.align.k4.hg38.log.txt\u001b[K\n",
      "input chromap_barcode_log is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "start run_merge_chromap_logs shell script\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "********1***********\u001b[K\n",
      "********2***********\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "uni-mappings,260650161\u001b[K\n",
      "multi-mappings,14666503\u001b[K\n",
      "total,275316664\u001b[K\n",
      "********3***********\u001b[K\n",
      "finished run_merge_chromap_logs shell script\u001b[K\n",
      "finished run_merge_logs\u001b[K\n",
      "\u001b[K\n",
      "start run_process_conversion_to_barcode\u001b[K\n",
      "input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "------ There is NO conversion list ------\u001b[K\n",
      "temp_conversion\u001b[K\n",
      "7021039 temp_conversion\u001b[K\n",
      "finished run_process_conversion_to_barcode\u001b[K\n",
      "\u001b[K\n",
      "\u001b[126A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/4YUdmqfnbjPozx\u001b[22;39m\u001b[K\n",
      "[3f/f09a4d] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[15/0ac619] process > run_whitelist_gunzip (1)       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[59/d22a1d] process > run_add_subpool_prefix_to_f... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[a9/95c478] process > run_merge_chromap_logs (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3c/fca9ef] process > run_process_conversion_to_b... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3b/05be46] process > run_filter_aligned_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_tabix_no_singleton         -\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[67/a1b7db] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb  7 20:07 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/eb/6412854cc6c199859cdfbe280f86c9/genes.gtf\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb  7 20:07 genes.gtf.gz\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "Start run_whitelist_gunzip\u001b[K\n",
      "Gunzip file is: sai_192_whitelist.txt\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 172805\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/d5/12b6921f983d2326fa1e91570a96e7/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/6f/e3ee295a7f5364c39365c48e8bd0b0/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/4a/7b5d74b3a4903977d7316c5f02ef84/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/2d/0e59fd7f9797eaf684de4abc52eb46/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/c5/cc9149d8d21bb423337888a9dc77ea/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/dc/65201d13e12dffd8779bbc935604a6/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb  7 20:08 na.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/f4/bab8ffeff41eecffb2ddcf013b4121/na.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       125 Feb  7 20:08 na1.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/38/b73884d031cf00d6089d843ba0f965/na1.na\u001b[K\n",
      "-r-x--x--x 1 root root 176947200 Feb  7 20:08 sai_192_whitelist\u001b[K\n",
      "lrwxrwxrwx 1 root root       140 Feb  7 20:08 sai_192_whitelist.txt -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/09/7ba70bbeac00bcd6547ad0c09655c7/sai_192_whitelist.txt\u001b[K\n",
      "Finished run_whitelist_gunzip\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "Input chromap_filter_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_fragments_table.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "\u001b[K\n",
      "start run_merge_logs\u001b[K\n",
      "input script_name is merge_chromap_logs.sh\u001b[K\n",
      "input chromap_alignment_log is BMMC-single-donor.atac.align.k4.hg38.log.txt\u001b[K\n",
      "input chromap_barcode_log is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "start run_merge_chromap_logs shell script\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "********1***********\u001b[K\n",
      "********2***********\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "uni-mappings,260650161\u001b[K\n",
      "multi-mappings,14666503\u001b[K\n",
      "total,275316664\u001b[K\n",
      "********3***********\u001b[K\n",
      "finished run_merge_chromap_logs shell script\u001b[K\n",
      "finished run_merge_logs\u001b[K\n",
      "\u001b[K\n",
      "start run_process_conversion_to_barcode\u001b[K\n",
      "input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "------ There is NO conversion list ------\u001b[K\n",
      "temp_conversion\u001b[K\n",
      "7021039 temp_conversion\u001b[K\n",
      "finished run_process_conversion_to_barcode\u001b[K\n",
      "\u001b[K\n",
      "start run_filter_aligned_fragments\u001b[K\n",
      "input barcodeFragmentsTsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "input threshold is 10\u001b[K\n",
      "input temp_dict_conversion is temp_conversion\u001b[K\n",
      "------ Number of barcodes BEFORE filtering------\u001b[K\n",
      "7021039 temp_conversion\u001b[K\n",
      "------ Filtering fragments ------\u001b[K\n",
      "------ 3 - Number of lines in intermediate file AFTER 2nd filtering ------\u001b[K\n",
      "3950969 tmp_file\u001b[K\n",
      "------ bgzip of temp_filtered_file and creating output_file_name------\u001b[K\n",
      "------ Number of lines in zcat no-singleton.bed.gz AFTER filtering and compression ------\u001b[K\n",
      "3950969\u001b[K\n",
      "name of output_file_name is: no-singleton.bed.gz\u001b[K\n",
      "------ Additional command: Counting lines in temp_filtered_file ------\u001b[K\n",
      "3950969\u001b[K\n",
      "finished run_filter_aligned_fragments\u001b[K\n",
      "\u001b[K\n",
      "\u001b[143A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/4YUdmqfnbjPozx\u001b[22;39m\u001b[K\n",
      "[3f/f09a4d] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[15/0ac619] process > run_whitelist_gunzip (1)       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[59/d22a1d] process > run_add_subpool_prefix_to_f... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[a9/95c478] process > run_merge_chromap_logs (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3c/fca9ef] process > run_process_conversion_to_b... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3b/05be46] process > run_filter_aligned_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[b3/94bc73] process > run_tabix_no_singleton (1)     [100%] 1 of 1, cached: 1\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[67/a1b7db] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb  7 20:07 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/eb/6412854cc6c199859cdfbe280f86c9/genes.gtf\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb  7 20:07 genes.gtf.gz\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "Start run_whitelist_gunzip\u001b[K\n",
      "Gunzip file is: sai_192_whitelist.txt\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 172805\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/d5/12b6921f983d2326fa1e91570a96e7/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/6f/e3ee295a7f5364c39365c48e8bd0b0/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/4a/7b5d74b3a4903977d7316c5f02ef84/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/2d/0e59fd7f9797eaf684de4abc52eb46/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/c5/cc9149d8d21bb423337888a9dc77ea/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/dc/65201d13e12dffd8779bbc935604a6/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb  7 20:08 na.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/f4/bab8ffeff41eecffb2ddcf013b4121/na.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       125 Feb  7 20:08 na1.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/38/b73884d031cf00d6089d843ba0f965/na1.na\u001b[K\n",
      "-r-x--x--x 1 root root 176947200 Feb  7 20:08 sai_192_whitelist\u001b[K\n",
      "lrwxrwxrwx 1 root root       140 Feb  7 20:08 sai_192_whitelist.txt -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/09/7ba70bbeac00bcd6547ad0c09655c7/sai_192_whitelist.txt\u001b[K\n",
      "Finished run_whitelist_gunzip\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "Input chromap_filter_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_fragments_table.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "\u001b[K\n",
      "start run_merge_logs\u001b[K\n",
      "input script_name is merge_chromap_logs.sh\u001b[K\n",
      "input chromap_alignment_log is BMMC-single-donor.atac.align.k4.hg38.log.txt\u001b[K\n",
      "input chromap_barcode_log is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "start run_merge_chromap_logs shell script\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "********1***********\u001b[K\n",
      "********2***********\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "uni-mappings,260650161\u001b[K\n",
      "multi-mappings,14666503\u001b[K\n",
      "total,275316664\u001b[K\n",
      "********3***********\u001b[K\n",
      "finished run_merge_chromap_logs shell script\u001b[K\n",
      "finished run_merge_logs\u001b[K\n",
      "\u001b[K\n",
      "start run_process_conversion_to_barcode\u001b[K\n",
      "input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "------ There is NO conversion list ------\u001b[K\n",
      "temp_conversion\u001b[K\n",
      "7021039 temp_conversion\u001b[K\n",
      "finished run_process_conversion_to_barcode\u001b[K\n",
      "\u001b[K\n",
      "start run_filter_aligned_fragments\u001b[K\n",
      "input barcodeFragmentsTsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "input threshold is 10\u001b[K\n",
      "input temp_dict_conversion is temp_conversion\u001b[K\n",
      "------ Number of barcodes BEFORE filtering------\u001b[K\n",
      "7021039 temp_conversion\u001b[K\n",
      "------ Filtering fragments ------\u001b[K\n",
      "------ 3 - Number of lines in intermediate file AFTER 2nd filtering ------\u001b[K\n",
      "3950969 tmp_file\u001b[K\n",
      "------ bgzip of temp_filtered_file and creating output_file_name------\u001b[K\n",
      "------ Number of lines in zcat no-singleton.bed.gz AFTER filtering and compression ------\u001b[K\n",
      "3950969\u001b[K\n",
      "name of output_file_name is: no-singleton.bed.gz\u001b[K\n",
      "------ Additional command: Counting lines in temp_filtered_file ------\u001b[K\n",
      "3950969\u001b[K\n",
      "finished run_filter_aligned_fragments\u001b[K\n",
      "\u001b[K\n",
      "\u001b[143A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/4YUdmqfnbjPozx\u001b[22;39m\u001b[K\n",
      "[3f/f09a4d] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[15/0ac619] process > run_whitelist_gunzip (1)       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[59/d22a1d] process > run_add_subpool_prefix_to_f... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[a9/95c478] process > run_merge_chromap_logs (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3c/fca9ef] process > run_process_conversion_to_b... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3b/05be46] process > run_filter_aligned_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[b3/94bc73] process > run_tabix_no_singleton (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[67/a1b7db] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb  7 20:07 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/eb/6412854cc6c199859cdfbe280f86c9/genes.gtf\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb  7 20:07 genes.gtf.gz\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "Start run_whitelist_gunzip\u001b[K\n",
      "Gunzip file is: sai_192_whitelist.txt\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 172805\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/d5/12b6921f983d2326fa1e91570a96e7/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/6f/e3ee295a7f5364c39365c48e8bd0b0/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/4a/7b5d74b3a4903977d7316c5f02ef84/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/2d/0e59fd7f9797eaf684de4abc52eb46/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/c5/cc9149d8d21bb423337888a9dc77ea/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/dc/65201d13e12dffd8779bbc935604a6/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb  7 20:08 na.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/f4/bab8ffeff41eecffb2ddcf013b4121/na.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       125 Feb  7 20:08 na1.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/38/b73884d031cf00d6089d843ba0f965/na1.na\u001b[K\n",
      "-r-x--x--x 1 root root 176947200 Feb  7 20:08 sai_192_whitelist\u001b[K\n",
      "lrwxrwxrwx 1 root root       140 Feb  7 20:08 sai_192_whitelist.txt -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/09/7ba70bbeac00bcd6547ad0c09655c7/sai_192_whitelist.txt\u001b[K\n",
      "Finished run_whitelist_gunzip\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "Input chromap_filter_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_fragments_table.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "\u001b[K\n",
      "start run_merge_logs\u001b[K\n",
      "input script_name is merge_chromap_logs.sh\u001b[K\n",
      "input chromap_alignment_log is BMMC-single-donor.atac.align.k4.hg38.log.txt\u001b[K\n",
      "input chromap_barcode_log is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "start run_merge_chromap_logs shell script\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "********1***********\u001b[K\n",
      "********2***********\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "uni-mappings,260650161\u001b[K\n",
      "multi-mappings,14666503\u001b[K\n",
      "total,275316664\u001b[K\n",
      "********3***********\u001b[K\n",
      "finished run_merge_chromap_logs shell script\u001b[K\n",
      "finished run_merge_logs\u001b[K\n",
      "\u001b[K\n",
      "start run_process_conversion_to_barcode\u001b[K\n",
      "input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "------ There is NO conversion list ------\u001b[K\n",
      "temp_conversion\u001b[K\n",
      "7021039 temp_conversion\u001b[K\n",
      "finished run_process_conversion_to_barcode\u001b[K\n",
      "\u001b[K\n",
      "start run_filter_aligned_fragments\u001b[K\n",
      "input barcodeFragmentsTsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "input threshold is 10\u001b[K\n",
      "input temp_dict_conversion is temp_conversion\u001b[K\n",
      "------ Number of barcodes BEFORE filtering------\u001b[K\n",
      "7021039 temp_conversion\u001b[K\n",
      "------ Filtering fragments ------\u001b[K\n",
      "------ 3 - Number of lines in intermediate file AFTER 2nd filtering ------\u001b[K\n",
      "3950969 tmp_file\u001b[K\n",
      "------ bgzip of temp_filtered_file and creating output_file_name------\u001b[K\n",
      "------ Number of lines in zcat no-singleton.bed.gz AFTER filtering and compression ------\u001b[K\n",
      "3950969\u001b[K\n",
      "name of output_file_name is: no-singleton.bed.gz\u001b[K\n",
      "------ Additional command: Counting lines in temp_filtered_file ------\u001b[K\n",
      "3950969\u001b[K\n",
      "finished run_filter_aligned_fragments\u001b[K\n",
      "\u001b[K\n",
      "start run_tabix_no_singleton\u001b[K\n",
      "input no_singelton_gz is no-singleton.bed.gz\u001b[K\n",
      "start run_tabix\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 95 Feb  7 20:15 no-singleton.bed.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/3b/05be46c60026121474ee6b3fdc470a/no-singleton.bed.gz\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 95 Feb  7 20:15 no-singleton.bed.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/3b/05be46c60026121474ee6b3fdc470a/no-singleton.bed.gz\u001b[K\n",
      "-rw-r--r-- 1 root root 75 Feb  7 20:16 no-singleton.bed.gz.tbi\u001b[K\n",
      "finished run_tabix\u001b[K\n",
      "no-singleton.bed.gz\u001b[K\n",
      "no-singleton.bed.gz.tbi\u001b[K\n",
      "finished run_tabix_no_singleton\u001b[K\n",
      "\u001b[K\n",
      "\u001b[156A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/4YUdmqfnbjPozx\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (1)\u001b[K\n",
      "[3f/f09a4d] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[15/0ac619] process > run_whitelist_gunzip (1)       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[59/d22a1d] process > run_add_subpool_prefix_to_f... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[70/cf6256] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[a9/95c478] process > run_merge_chromap_logs (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3c/fca9ef] process > run_process_conversion_to_b... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3b/05be46] process > run_filter_aligned_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[b3/94bc73] process > run_tabix_no_singleton (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... -\u001b[K\n",
      "[67/a1b7db] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb  7 20:07 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/eb/6412854cc6c199859cdfbe280f86c9/genes.gtf\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb  7 20:07 genes.gtf.gz\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "Start run_whitelist_gunzip\u001b[K\n",
      "Gunzip file is: sai_192_whitelist.txt\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 172805\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/d5/12b6921f983d2326fa1e91570a96e7/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/6f/e3ee295a7f5364c39365c48e8bd0b0/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/4a/7b5d74b3a4903977d7316c5f02ef84/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/2d/0e59fd7f9797eaf684de4abc52eb46/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/c5/cc9149d8d21bb423337888a9dc77ea/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/dc/65201d13e12dffd8779bbc935604a6/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb  7 20:08 na.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/f4/bab8ffeff41eecffb2ddcf013b4121/na.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       125 Feb  7 20:08 na1.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/38/b73884d031cf00d6089d843ba0f965/na1.na\u001b[K\n",
      "-r-x--x--x 1 root root 176947200 Feb  7 20:08 sai_192_whitelist\u001b[K\n",
      "lrwxrwxrwx 1 root root       140 Feb  7 20:08 sai_192_whitelist.txt -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/09/7ba70bbeac00bcd6547ad0c09655c7/sai_192_whitelist.txt\u001b[K\n",
      "Finished run_whitelist_gunzip\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "Input chromap_filter_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_fragments_table.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "\u001b[K\n",
      "start run_merge_logs\u001b[K\n",
      "input script_name is merge_chromap_logs.sh\u001b[K\n",
      "input chromap_alignment_log is BMMC-single-donor.atac.align.k4.hg38.log.txt\u001b[K\n",
      "input chromap_barcode_log is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "start run_merge_chromap_logs shell script\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "********1***********\u001b[K\n",
      "********2***********\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "uni-mappings,260650161\u001b[K\n",
      "multi-mappings,14666503\u001b[K\n",
      "total,275316664\u001b[K\n",
      "********3***********\u001b[K\n",
      "finished run_merge_chromap_logs shell script\u001b[K\n",
      "finished run_merge_logs\u001b[K\n",
      "\u001b[K\n",
      "start run_process_conversion_to_barcode\u001b[K\n",
      "input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "------ There is NO conversion list ------\u001b[K\n",
      "temp_conversion\u001b[K\n",
      "7021039 temp_conversion\u001b[K\n",
      "finished run_process_conversion_to_barcode\u001b[K\n",
      "\u001b[K\n",
      "start run_filter_aligned_fragments\u001b[K\n",
      "input barcodeFragmentsTsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "input threshold is 10\u001b[K\n",
      "input temp_dict_conversion is temp_conversion\u001b[K\n",
      "------ Number of barcodes BEFORE filtering------\u001b[K\n",
      "7021039 temp_conversion\u001b[K\n",
      "------ Filtering fragments ------\u001b[K\n",
      "------ 3 - Number of lines in intermediate file AFTER 2nd filtering ------\u001b[K\n",
      "3950969 tmp_file\u001b[K\n",
      "------ bgzip of temp_filtered_file and creating output_file_name------\u001b[K\n",
      "------ Number of lines in zcat no-singleton.bed.gz AFTER filtering and compression ------\u001b[K\n",
      "3950969\u001b[K\n",
      "name of output_file_name is: no-singleton.bed.gz\u001b[K\n",
      "------ Additional command: Counting lines in temp_filtered_file ------\u001b[K\n",
      "3950969\u001b[K\n",
      "finished run_filter_aligned_fragments\u001b[K\n",
      "\u001b[K\n",
      "start run_tabix_no_singleton\u001b[K\n",
      "input no_singelton_gz is no-singleton.bed.gz\u001b[K\n",
      "start run_tabix\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 95 Feb  7 20:15 no-singleton.bed.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/3b/05be46c60026121474ee6b3fdc470a/no-singleton.bed.gz\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 95 Feb  7 20:15 no-singleton.bed.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/3b/05be46c60026121474ee6b3fdc470a/no-singleton.bed.gz\u001b[K\n",
      "-rw-r--r-- 1 root root 75 Feb  7 20:16 no-singleton.bed.gz.tbi\u001b[K\n",
      "finished run_tabix\u001b[K\n",
      "no-singleton.bed.gz\u001b[K\n",
      "no-singleton.bed.gz.tbi\u001b[K\n",
      "finished run_tabix_no_singleton\u001b[K\n",
      "\u001b[K\n",
      "\u001b[157A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/4YUdmqfnbjPozx\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (1)\u001b[K\n",
      "[3f/f09a4d] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[15/0ac619] process > run_whitelist_gunzip (1)       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[59/d22a1d] process > run_add_subpool_prefix_to_f... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[70/cf6256] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[a9/95c478] process > run_merge_chromap_logs (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3c/fca9ef] process > run_process_conversion_to_b... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3b/05be46] process > run_filter_aligned_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[b3/94bc73] process > run_tabix_no_singleton (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_calculate_tss_enrichmen... [  0%] 0 of 1\u001b[K\n",
      "[67/a1b7db] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb  7 20:07 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/eb/6412854cc6c199859cdfbe280f86c9/genes.gtf\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb  7 20:07 genes.gtf.gz\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "Start run_whitelist_gunzip\u001b[K\n",
      "Gunzip file is: sai_192_whitelist.txt\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 172805\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/d5/12b6921f983d2326fa1e91570a96e7/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/6f/e3ee295a7f5364c39365c48e8bd0b0/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/4a/7b5d74b3a4903977d7316c5f02ef84/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/2d/0e59fd7f9797eaf684de4abc52eb46/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/c5/cc9149d8d21bb423337888a9dc77ea/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/dc/65201d13e12dffd8779bbc935604a6/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb  7 20:08 na.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/f4/bab8ffeff41eecffb2ddcf013b4121/na.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       125 Feb  7 20:08 na1.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/38/b73884d031cf00d6089d843ba0f965/na1.na\u001b[K\n",
      "-r-x--x--x 1 root root 176947200 Feb  7 20:08 sai_192_whitelist\u001b[K\n",
      "lrwxrwxrwx 1 root root       140 Feb  7 20:08 sai_192_whitelist.txt -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/09/7ba70bbeac00bcd6547ad0c09655c7/sai_192_whitelist.txt\u001b[K\n",
      "Finished run_whitelist_gunzip\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "Input chromap_filter_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_fragments_table.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "\u001b[K\n",
      "start run_merge_logs\u001b[K\n",
      "input script_name is merge_chromap_logs.sh\u001b[K\n",
      "input chromap_alignment_log is BMMC-single-donor.atac.align.k4.hg38.log.txt\u001b[K\n",
      "input chromap_barcode_log is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "start run_merge_chromap_logs shell script\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "********1***********\u001b[K\n",
      "********2***********\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "uni-mappings,260650161\u001b[K\n",
      "multi-mappings,14666503\u001b[K\n",
      "total,275316664\u001b[K\n",
      "********3***********\u001b[K\n",
      "finished run_merge_chromap_logs shell script\u001b[K\n",
      "finished run_merge_logs\u001b[K\n",
      "\u001b[K\n",
      "start run_process_conversion_to_barcode\u001b[K\n",
      "input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "------ There is NO conversion list ------\u001b[K\n",
      "temp_conversion\u001b[K\n",
      "7021039 temp_conversion\u001b[K\n",
      "finished run_process_conversion_to_barcode\u001b[K\n",
      "\u001b[K\n",
      "start run_filter_aligned_fragments\u001b[K\n",
      "input barcodeFragmentsTsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "input threshold is 10\u001b[K\n",
      "input temp_dict_conversion is temp_conversion\u001b[K\n",
      "------ Number of barcodes BEFORE filtering------\u001b[K\n",
      "7021039 temp_conversion\u001b[K\n",
      "------ Filtering fragments ------\u001b[K\n",
      "------ 3 - Number of lines in intermediate file AFTER 2nd filtering ------\u001b[K\n",
      "3950969 tmp_file\u001b[K\n",
      "------ bgzip of temp_filtered_file and creating output_file_name------\u001b[K\n",
      "------ Number of lines in zcat no-singleton.bed.gz AFTER filtering and compression ------\u001b[K\n",
      "3950969\u001b[K\n",
      "name of output_file_name is: no-singleton.bed.gz\u001b[K\n",
      "------ Additional command: Counting lines in temp_filtered_file ------\u001b[K\n",
      "3950969\u001b[K\n",
      "finished run_filter_aligned_fragments\u001b[K\n",
      "\u001b[K\n",
      "start run_tabix_no_singleton\u001b[K\n",
      "input no_singelton_gz is no-singleton.bed.gz\u001b[K\n",
      "start run_tabix\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 95 Feb  7 20:15 no-singleton.bed.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/3b/05be46c60026121474ee6b3fdc470a/no-singleton.bed.gz\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 95 Feb  7 20:15 no-singleton.bed.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/3b/05be46c60026121474ee6b3fdc470a/no-singleton.bed.gz\u001b[K\n",
      "-rw-r--r-- 1 root root 75 Feb  7 20:16 no-singleton.bed.gz.tbi\u001b[K\n",
      "finished run_tabix\u001b[K\n",
      "no-singleton.bed.gz\u001b[K\n",
      "no-singleton.bed.gz.tbi\u001b[K\n",
      "finished run_tabix_no_singleton\u001b[K\n",
      "\u001b[K\n",
      "\u001b[157A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/4YUdmqfnbjPozx\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (2)\u001b[K\n",
      "[3f/f09a4d] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[15/0ac619] process > run_whitelist_gunzip (1)       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[59/d22a1d] process > run_add_subpool_prefix_to_f... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[70/cf6256] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[a9/95c478] process > run_merge_chromap_logs (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3c/fca9ef] process > run_process_conversion_to_b... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3b/05be46] process > run_filter_aligned_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[b3/94bc73] process > run_tabix_no_singleton (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[8a/8f19f8] process > run_calculate_tss_enrichmen... [  0%] 0 of 1\u001b[K\n",
      "[67/a1b7db] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "after load chrompa mapping debug\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix_chromap with tabix shell script\u001b[K\n",
      "after call run_tabix_chromap\u001b[K\n",
      "before call run_merge_chromap_logs\u001b[K\n",
      "after call run_merge_chromap_logs\u001b[K\n",
      "before run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "after run_add_subpool_prefix_to_summary_file\u001b[K\n",
      "before call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after call run_calculate_tss_enrichment_bulk\u001b[K\n",
      "after run_gzip_on_genes_gtf\u001b[K\n",
      "before call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "after call run_calculate_tss_enrichment_snapatac2\u001b[K\n",
      "before run_generate_insert_size_histogram_data\u001b[K\n",
      "after run_generate_insert_size_histogram_data\u001b[K\n",
      "before call run_generate_insert_size_plot\u001b[K\n",
      "after call run_generate_insert_size_plot\u001b[K\n",
      "before run_barcode_metadata_stats\u001b[K\n",
      "after run_barcode_metadata_stats\u001b[K\n",
      "before call run_atac_barcode_rank_plot\u001b[K\n",
      "after run_atac_barcode_rank_plot\u001b[K\n",
      "Start run_zcat\u001b[K\n",
      "Zcat file is: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 169 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 3070539\u001b[K\n",
      "-rw-r--r-- 1 root root 3144230986 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna\u001b[K\n",
      "lrwxrwxrwx 1 root root        169 Feb  7 20:08 GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Finished run_zcat\u001b[K\n",
      "\u001b[K\n",
      "Start run_gzip\u001b[K\n",
      "Gzip file is: genes.gtf.gz\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 55875\u001b[K\n",
      "lrwxrwxrwx 1 root root      128 Feb  7 20:07 genes.gtf -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/eb/6412854cc6c199859cdfbe280f86c9/genes.gtf\u001b[K\n",
      "-rw-r--r-- 1 root root 57215195 Feb  7 20:07 genes.gtf.gz\u001b[K\n",
      "Finished run_gzip\u001b[K\n",
      "\u001b[K\n",
      "Start run_whitelist_gunzip\u001b[K\n",
      "Gunzip file is: sai_192_whitelist.txt\u001b[K\n",
      "Files before processing:\u001b[K\n",
      "Files after processing:\u001b[K\n",
      "total 172805\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/d5/12b6921f983d2326fa1e91570a96e7/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/6f/e3ee295a7f5364c39365c48e8bd0b0/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/4a/7b5d74b3a4903977d7316c5f02ef84/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/2d/0e59fd7f9797eaf684de4abc52eb46/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/c5/cc9149d8d21bb423337888a9dc77ea/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       176 Feb  7 20:08 BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/dc/65201d13e12dffd8779bbc935604a6/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "lrwxrwxrwx 1 root root       124 Feb  7 20:08 na.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/f4/bab8ffeff41eecffb2ddcf013b4121/na.na\u001b[K\n",
      "lrwxrwxrwx 1 root root       125 Feb  7 20:08 na1.na -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/38/b73884d031cf00d6089d843ba0f965/na1.na\u001b[K\n",
      "-r-x--x--x 1 root root 176947200 Feb  7 20:08 sai_192_whitelist\u001b[K\n",
      "lrwxrwxrwx 1 root root       140 Feb  7 20:08 sai_192_whitelist.txt -> /fusion/gs/eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/09/7ba70bbeac00bcd6547ad0c09655c7/sai_192_whitelist.txt\u001b[K\n",
      "Finished run_whitelist_gunzip\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "Input chromap_filter_fragments_tsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "Input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "Input subpool is none\u001b[K\n",
      "Input subpool_script is add_subpool_to_fragments_table.sh\u001b[K\n",
      "No subpool specified. Skipping subpool addition.\u001b[K\n",
      "------ Finished run_add_pool_prefix_to_fragment_table ------\u001b[K\n",
      "\u001b[K\n",
      "start run_merge_logs\u001b[K\n",
      "input script_name is merge_chromap_logs.sh\u001b[K\n",
      "input chromap_alignment_log is BMMC-single-donor.atac.align.k4.hg38.log.txt\u001b[K\n",
      "input chromap_barcode_log is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "start run_merge_chromap_logs shell script\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "********1***********\u001b[K\n",
      "********2***********\u001b[K\n",
      "Number_of_reads,770560126\u001b[K\n",
      "Number_of_mapped_reads,717898494\u001b[K\n",
      "Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "Number_of_candidates,3413147335\u001b[K\n",
      "Number_of_mappings,717898494\u001b[K\n",
      "Number_of_uni-mappings,660005178\u001b[K\n",
      "Number_of_multi-mappings,57893316\u001b[K\n",
      "Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "Number_of_corrected_barcodes,0\u001b[K\n",
      "Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "uni-mappings,260650161\u001b[K\n",
      "multi-mappings,14666503\u001b[K\n",
      "total,275316664\u001b[K\n",
      "********3***********\u001b[K\n",
      "finished run_merge_chromap_logs shell script\u001b[K\n",
      "finished run_merge_logs\u001b[K\n",
      "\u001b[K\n",
      "start run_process_conversion_to_barcode\u001b[K\n",
      "input barcode_summary_csv is BMMC-single-donor.atac.align.barcode.summary.csv\u001b[K\n",
      "------ There is NO conversion list ------\u001b[K\n",
      "temp_conversion\u001b[K\n",
      "7021039 temp_conversion\u001b[K\n",
      "finished run_process_conversion_to_barcode\u001b[K\n",
      "\u001b[K\n",
      "start run_filter_aligned_fragments\u001b[K\n",
      "input barcodeFragmentsTsv is ATAC_BMMC-single-donor.atac.filter.fragments.tsv\u001b[K\n",
      "input threshold is 10\u001b[K\n",
      "input temp_dict_conversion is temp_conversion\u001b[K\n",
      "------ Number of barcodes BEFORE filtering------\u001b[K\n",
      "7021039 temp_conversion\u001b[K\n",
      "------ Filtering fragments ------\u001b[K\n",
      "------ 3 - Number of lines in intermediate file AFTER 2nd filtering ------\u001b[K\n",
      "3950969 tmp_file\u001b[K\n",
      "------ bgzip of temp_filtered_file and creating output_file_name------\u001b[K\n",
      "------ Number of lines in zcat no-singleton.bed.gz AFTER filtering and compression ------\u001b[K\n",
      "3950969\u001b[K\n",
      "name of output_file_name is: no-singleton.bed.gz\u001b[K\n",
      "------ Additional command: Counting lines in temp_filtered_file ------\u001b[K\n",
      "3950969\u001b[K\n",
      "finished run_filter_aligned_fragments\u001b[K\n",
      "\u001b[K\n",
      "start run_tabix_no_singleton\u001b[K\n",
      "input no_singelton_gz is no-singleton.bed.gz\u001b[K\n",
      "start run_tabix\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 95 Feb  7 20:15 no-singleton.bed.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/3b/05be46c60026121474ee6b3fdc470a/no-singleton.bed.gz\u001b[K\n",
      "total 1\u001b[K\n",
      "lrwxrwxrwx 1 root root 95 Feb  7 20:15 no-singleton.bed.gz -> /fusion/gs/eila-nextflow-bucket/work/ATAC/3b/05be46c60026121474ee6b3fdc470a/no-singleton.bed.gz\u001b[K\n",
      "-rw-r--r-- 1 root root 75 Feb  7 20:16 no-singleton.bed.gz.tbi\u001b[K\n",
      "finished run_tabix\u001b[K\n",
      "no-singleton.bed.gz\u001b[K\n",
      "no-singleton.bed.gz.tbi\u001b[K\n",
      "finished run_tabix_no_singleton\u001b[K\n",
      "\u001b[K\n",
      "\u001b[157A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/4YUdmqfnbjPozx\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (2)\u001b[K\n",
      "[3f/f09a4d] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[15/0ac619] process > run_whitelist_gunzip (1)       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[59/d22a1d] process > run_add_subpool_prefix_to_f... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[70/cf6256] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[a9/95c478] process > run_merge_chromap_logs (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3c/fca9ef] process > run_process_conversion_to_b... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3b/05be46] process > run_filter_aligned_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[b3/94bc73] process > run_tabix_no_singleton (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[8a/8f19f8] process > run_calculate_tss_enrichmen... [  0%] 0 of 1\u001b[K\n",
      "[67/a1b7db] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[157A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/4YUdmqfnbjPozx\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (2)\u001b[K\n",
      "[3f/f09a4d] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[15/0ac619] process > run_whitelist_gunzip (1)       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[59/d22a1d] process > run_add_subpool_prefix_to_f... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[70/cf6256] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[a9/95c478] process > run_merge_chromap_logs (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3c/fca9ef] process > run_process_conversion_to_b... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3b/05be46] process > run_filter_aligned_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[b3/94bc73] process > run_tabix_no_singleton (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[8a/8f19f8] process > run_calculate_tss_enrichmen... [  0%] 0 of 1\u001b[K\n",
      "[67/a1b7db] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[157A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/4YUdmqfnbjPozx\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (2)\u001b[K\n",
      "[3f/f09a4d] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[15/0ac619] process > run_whitelist_gunzip (1)       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[59/d22a1d] process > run_add_subpool_prefix_to_f... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[70/cf6256] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[a9/95c478] process > run_merge_chromap_logs (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3c/fca9ef] process > run_process_conversion_to_b... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3b/05be46] process > run_filter_aligned_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[b3/94bc73] process > run_tabix_no_singleton (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[8a/8f19f8] process > run_calculate_tss_enrichmen... [  0%] 0 of 1\u001b[K\n",
      "[67/a1b7db] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "\u001b[31mERROR ~ Error executing process > 'run_calculate_tss_enrichment_bulk (1)'\u001b[K\n",
      "\u001b[K\n",
      "Caused by:\u001b[K\n",
      "  Process `run_calculate_tss_enrichment_bulk (1)` terminated with an error exit status (1)\u001b[K\n",
      "\u001b[K\n",
      "Command executed:\u001b[K\n",
      "\u001b[K\n",
      "  echo 'start run_calculate_tss_enrichment'\u001b[K\n",
      "  echo 'calculation_script is compute_tss_enrichment_bulk.py. check that it is: compute_tss_script.py'\u001b[K\n",
      "  echo 'tbi_fragments: no-singleton.bed.gz.tbi'\u001b[K\n",
      "  echo 'tss_bases_flank: 2000'\u001b[K\n",
      "  echo 'tss_col_strand_info: 4'\u001b[K\n",
      "  echo 'smoothing_window_size: 20'\u001b[K\n",
      "  python3 /usr/local/bin/compute_tss_enrichment_bulk.py no-singleton.bed.gz.tbi -e 2000 -s 4 -p 16 --prefix BMMC-single-donor --regions hg38.synapse_transcripts.TSS.bed\u001b[K\n",
      "  echo 'finished run_calculate_tss_enrichment'\u001b[K\n",
      "\u001b[K\n",
      "Command exit status:\u001b[K\n",
      "  1\u001b[K\n",
      "\u001b[K\n",
      "Command output:\u001b[K\n",
      "  start run_calculate_tss_enrichment\u001b[K\n",
      "  calculation_script is compute_tss_enrichment_bulk.py. check that it is: compute_tss_script.py\u001b[K\n",
      "  tbi_fragments: no-singleton.bed.gz.tbi\u001b[K\n",
      "  tss_bases_flank: 2000\u001b[K\n",
      "  tss_col_strand_info: 4\u001b[K\n",
      "  smoothing_window_size: 20\u001b[K\n",
      "  Command-line arguments: Namespace(e=2000, p=16, prefix='BMMC-single-donor', regions='hg38.synapse_transcripts.TSS.bed', s=4, tabix='no-singleton.bed.gz.tbi', window=20)\u001b[K\n",
      "\u001b[K\n",
      "Command error:\u001b[K\n",
      "  Traceback (most recent call last):\u001b[K\n",
      "    File \"/usr/local/bin/compute_tss_enrichment_bulk.py\", line 108, in <module>\u001b[K\n",
      "      count_args = _prepare_args_for_counting(args.tabix, regions_list, args.e, args.p)\u001b[K\n",
      "  NameError: name '_prepare_args_for_counting' is not defined\u001b[K\n",
      "\u001b[K\n",
      "Work dir:\u001b[K\n",
      "  gs://eila-nextflow-bucket/work/ATAC/8a/8f19f86c2ac97bd9ca2b62ae1266a2\u001b[K\n",
      "\u001b[K\n",
      "Tip: you can replicate the issue by changing to the process work dir and entering the command `bash .command.run`\u001b[K\n",
      "\u001b[K\n",
      " -- Check '.nextflow.log' file for details\u001b[39m\u001b[K\n",
      "\u001b[56A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/4YUdmqfnbjPozx\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (2)\u001b[K\n",
      "[3f/f09a4d] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[15/0ac619] process > run_whitelist_gunzip (1)       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[59/d22a1d] process > run_add_subpool_prefix_to_f... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[70/cf6256] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[a9/95c478] process > run_merge_chromap_logs (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3c/fca9ef] process > run_process_conversion_to_b... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3b/05be46] process > run_filter_aligned_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[b3/94bc73] process > run_tabix_no_singleton (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[8a/8f19f8] process > run_calculate_tss_enrichmen... [100%] 1 of 1, failed: 1 ✘\u001b[K\n",
      "[67/a1b7db] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "\u001b[31mERROR ~ Error executing process > 'run_calculate_tss_enrichment_bulk (1)'\u001b[K\n",
      "\u001b[K\n",
      "Caused by:\u001b[K\n",
      "  Process `run_calculate_tss_enrichment_bulk (1)` terminated with an error exit status (1)\u001b[K\n",
      "\u001b[K\n",
      "Command executed:\u001b[K\n",
      "\u001b[K\n",
      "  echo 'start run_calculate_tss_enrichment'\u001b[K\n",
      "  echo 'calculation_script is compute_tss_enrichment_bulk.py. check that it is: compute_tss_script.py'\u001b[K\n",
      "  echo 'tbi_fragments: no-singleton.bed.gz.tbi'\u001b[K\n",
      "  echo 'tss_bases_flank: 2000'\u001b[K\n",
      "  echo 'tss_col_strand_info: 4'\u001b[K\n",
      "  echo 'smoothing_window_size: 20'\u001b[K\n",
      "  python3 /usr/local/bin/compute_tss_enrichment_bulk.py no-singleton.bed.gz.tbi -e 2000 -s 4 -p 16 --prefix BMMC-single-donor --regions hg38.synapse_transcripts.TSS.bed\u001b[K\n",
      "  echo 'finished run_calculate_tss_enrichment'\u001b[K\n",
      "\u001b[K\n",
      "Command exit status:\u001b[K\n",
      "  1\u001b[K\n",
      "\u001b[K\n",
      "Command output:\u001b[K\n",
      "  start run_calculate_tss_enrichment\u001b[K\n",
      "  calculation_script is compute_tss_enrichment_bulk.py. check that it is: compute_tss_script.py\u001b[K\n",
      "  tbi_fragments: no-singleton.bed.gz.tbi\u001b[K\n",
      "  tss_bases_flank: 2000\u001b[K\n",
      "  tss_col_strand_info: 4\u001b[K\n",
      "  smoothing_window_size: 20\u001b[K\n",
      "  Command-line arguments: Namespace(e=2000, p=16, prefix='BMMC-single-donor', regions='hg38.synapse_transcripts.TSS.bed', s=4, tabix='no-singleton.bed.gz.tbi', window=20)\u001b[K\n",
      "\u001b[K\n",
      "Command error:\u001b[K\n",
      "  Traceback (most recent call last):\u001b[K\n",
      "    File \"/usr/local/bin/compute_tss_enrichment_bulk.py\", line 108, in <module>\u001b[K\n",
      "      count_args = _prepare_args_for_counting(args.tabix, regions_list, args.e, args.p)\u001b[K\n",
      "  NameError: name '_prepare_args_for_counting' is not defined\u001b[K\n",
      "\u001b[K\n",
      "Work dir:\u001b[K\n",
      "  gs://eila-nextflow-bucket/work/ATAC/8a/8f19f86c2ac97bd9ca2b62ae1266a2\u001b[K\n",
      "\u001b[K\n",
      "Tip: you can replicate the issue by changing to the process work dir and entering the command `bash .command.run`\u001b[K\n",
      "\u001b[K\n",
      " -- Check '.nextflow.log' file for details\u001b[39m\u001b[K\n",
      "\u001b[56A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/4YUdmqfnbjPozx\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (2)\u001b[K\n",
      "[3f/f09a4d] process > run_zcat                       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[15/0ac619] process > run_whitelist_gunzip (1)       [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[59/d22a1d] process > run_add_subpool_prefix_to_f... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[70/cf6256] process > run_bgzip_on_chromap_fragme... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_tabix_chromap              -\u001b[K\n",
      "[a9/95c478] process > run_merge_chromap_logs (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3c/fca9ef] process > run_process_conversion_to_b... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[3b/05be46] process > run_filter_aligned_fragment... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[b3/94bc73] process > run_tabix_no_singleton (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[8a/8f19f8] process > run_calculate_tss_enrichmen... [100%] 1 of 1, failed: 1 ✘\u001b[K\n",
      "[67/a1b7db] process > run_gzip_on_genes_gtf (1)      [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_generate_insert_size_hi... -\u001b[K\n",
      "[-        ] process > run_generate_insert_size_plot  -\u001b[K\n",
      "\u001b[31mERROR ~ Error executing process > 'run_calculate_tss_enrichment_bulk (1)'\u001b[K\n",
      "\u001b[K\n",
      "Caused by:\u001b[K\n",
      "  Process `run_calculate_tss_enrichment_bulk (1)` terminated with an error exit status (1)\u001b[K\n",
      "\u001b[K\n",
      "Command executed:\u001b[K\n",
      "\u001b[K\n",
      "  echo 'start run_calculate_tss_enrichment'\u001b[K\n",
      "  echo 'calculation_script is compute_tss_enrichment_bulk.py. check that it is: compute_tss_script.py'\u001b[K\n",
      "  echo 'tbi_fragments: no-singleton.bed.gz.tbi'\u001b[K\n",
      "  echo 'tss_bases_flank: 2000'\u001b[K\n",
      "  echo 'tss_col_strand_info: 4'\u001b[K\n",
      "  echo 'smoothing_window_size: 20'\u001b[K\n",
      "  python3 /usr/local/bin/compute_tss_enrichment_bulk.py no-singleton.bed.gz.tbi -e 2000 -s 4 -p 16 --prefix BMMC-single-donor --regions hg38.synapse_transcripts.TSS.bed\u001b[K\n",
      "  echo 'finished run_calculate_tss_enrichment'\u001b[K\n",
      "\u001b[K\n",
      "Command exit status:\u001b[K\n",
      "  1\u001b[K\n",
      "\u001b[K\n",
      "Command output:\u001b[K\n",
      "  start run_calculate_tss_enrichment\u001b[K\n",
      "  calculation_script is compute_tss_enrichment_bulk.py. check that it is: compute_tss_script.py\u001b[K\n",
      "  tbi_fragments: no-singleton.bed.gz.tbi\u001b[K\n",
      "  tss_bases_flank: 2000\u001b[K\n",
      "  tss_col_strand_info: 4\u001b[K\n",
      "  smoothing_window_size: 20\u001b[K\n",
      "  Command-line arguments: Namespace(e=2000, p=16, prefix='BMMC-single-donor', regions='hg38.synapse_transcripts.TSS.bed', s=4, tabix='no-singleton.bed.gz.tbi', window=20)\u001b[K\n",
      "\u001b[K\n",
      "Command error:\u001b[K\n",
      "  Traceback (most recent call last):\u001b[K\n",
      "    File \"/usr/local/bin/compute_tss_enrichment_bulk.py\", line 108, in <module>\u001b[K\n",
      "      count_args = _prepare_args_for_counting(args.tabix, regions_list, args.e, args.p)\u001b[K\n",
      "  NameError: name '_prepare_args_for_counting' is not defined\u001b[K\n",
      "\u001b[K\n",
      "Work dir:\u001b[K\n",
      "  gs://eila-nextflow-bucket/work/ATAC/8a/8f19f86c2ac97bd9ca2b62ae1266a2\u001b[K\n",
      "\u001b[K\n",
      "Tip: you can replicate the issue by changing to the process work dir and entering the command `bash .command.run`\u001b[K\n",
      "\u001b[K\n",
      " -- Check '.nextflow.log' file for details\u001b[39m\u001b[K\n",
      "WARN: Killing running tasks (1)\n",
      "WARN: Tower request field `workflow.errorMessage` exceeds expected size | offending value: `Traceback (most recent call last):\n",
      "  File \"/usr/local/bin/compute_tss_enrichment_bulk.py\", line 108, in <module>\n",
      "    count_args = _prepare_args_for_counting(args.tabix, regions_list, args.e, args.p)\n",
      "NameError: name '_prepare_args_for_counting' is not defined`, size: 258 (max: 255)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_atac.config -profile google_batch_profile -process.echo -with-tower -with-dag -resume\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_vm.config -profile google_batch_profile\n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_vm.config -profile google_batch_profile\n",
    "# !nextflow run nf_\n",
    "# \n",
    "# wf_fg_singlecell_atac.nf -c nextflow_vm_anvil.config -profile google_batch_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_local.config -profile google_batch_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_local.config -profile docker_profile\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3414,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Feb-07 15:45:42.785 [main] DEBUG nextflow.cli.Launcher - $> nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_atac.config -profile google_batch_profile -process.echo -with-tower -with-dag -resume\n",
      "Feb-07 15:45:42.935 [main] INFO  nextflow.cli.CmdRun - N E X T F L O W  ~  version 23.10.1\n",
      "Feb-07 15:45:42.950 [main] DEBUG nextflow.plugin.PluginsFacade - Setting up plugin manager > mode=prod; embedded=false; plugins-dir=/Users/eilaarich-landkof-stanford/.nextflow/plugins; core-plugins: nf-amazon@2.1.4,nf-azure@1.3.3,nf-cloudcache@0.3.0,nf-codecommit@0.1.5,nf-console@1.0.6,nf-ga4gh@1.1.0,nf-google@1.8.3,nf-tower@1.6.3,nf-wave@1.0.1\n",
      "Feb-07 15:45:42.957 [main] INFO  o.pf4j.DefaultPluginStatusProvider - Enabled plugins: []\n",
      "Feb-07 15:45:42.958 [main] INFO  o.pf4j.DefaultPluginStatusProvider - Disabled plugins: []\n",
      "Feb-07 15:45:42.960 [main] INFO  org.pf4j.DefaultPluginManager - PF4J version 3.4.1 in 'deployment' mode\n",
      "Feb-07 15:45:42.967 [main] INFO  org.pf4j.AbstractPluginManager - No plugins\n",
      "Feb-07 15:45:42.983 [main] DEBUG nextflow.config.ConfigBuilder - User config file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Feb-07 15:45:42.983 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Feb-07 15:45:42.997 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `google_batch_profile`\n",
      "Feb-07 15:45:44.553 [main] DEBUG nextflow.config.ConfigBuilder - Available config profiles: [google_batch_profile]\n",
      "Feb-07 15:45:44.589 [main] DEBUG nextflow.cli.CmdRun - Applied DSL=2 by global default\n",
      "Feb-07 15:45:44.615 [main] INFO  nextflow.cli.CmdRun - Launching `nf_wf_fg_singlecell_atac.nf` [nauseous_maxwell] DSL2 - revision: 16a69dd583\n",
      "Feb-07 15:45:44.615 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins default=[nf-google@1.8.3]\n",
      "Feb-07 15:45:44.616 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins resolved requirement=[nf-google@1.8.3, nf-tower@1.6.3, nf-wave@1.0.1]\n",
      "Feb-07 15:45:44.616 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-google version: 1.8.3\n",
      "Feb-07 15:45:44.633 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-google@1.8.3' resolved\n",
      "Feb-07 15:45:44.633 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-google@1.8.3'\n",
      "Feb-07 15:45:44.713 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-google@1.8.3\n",
      "Feb-07 15:45:44.714 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-tower version: 1.6.3\n",
      "Feb-07 15:45:44.716 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-tower@1.6.3' resolved\n",
      "Feb-07 15:45:44.717 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-tower@1.6.3'\n",
      "Feb-07 15:45:44.722 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-tower@1.6.3\n",
      "Feb-07 15:45:44.722 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-wave version: 1.0.1\n",
      "Feb-07 15:45:44.728 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-wave@1.0.1' resolved\n",
      "Feb-07 15:45:44.728 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-wave@1.0.1'\n",
      "Feb-07 15:45:44.735 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-wave@1.0.1\n",
      "Feb-07 15:45:44.749 [main] DEBUG n.secret.LocalSecretsProvider - Secrets store: /Users/eilaarich-landkof-stanford/.nextflow/secrets/store.json\n",
      "Feb-07 15:45:44.751 [main] DEBUG nextflow.secret.SecretsLoader - Discovered secrets providers: [nextflow.secret.LocalSecretsProvider@66236a0a] - activable => nextflow.secret.LocalSecretsProvider@66236a0a\n",
      "Feb-07 15:45:44.821 [main] DEBUG nextflow.Session - Session UUID: ab09407f-2f8c-43e1-855a-210d7286a110\n",
      "Feb-07 15:45:44.821 [main] DEBUG nextflow.Session - Run name: nauseous_maxwell\n",
      "Feb-07 15:45:44.821 [main] DEBUG nextflow.Session - Executor pool size: 8\n",
      "Feb-07 15:45:45.095 [main] DEBUG nextflow.file.FilePorter - File porter settings maxRetries=3; maxTransfers=50; pollTimeout=null\n",
      "Feb-07 15:45:45.099 [main] DEBUG nextflow.util.ThreadPoolBuilder - Creating thread pool 'FileTransfer' minSize=10; maxSize=24; workQueue=LinkedBlockingQueue[10000]; allowCoreThreadTimeout=false\n",
      "Feb-07 15:45:45.104 [main] DEBUG nextflow.config.ConfigBuilder - User config file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Feb-07 15:45:45.105 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Feb-07 15:45:45.105 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `google_batch_profile`\n",
      "Feb-07 15:45:46.240 [main] DEBUG nextflow.config.ConfigBuilder - Available config profiles: [google_batch_profile]\n",
      "Feb-07 15:45:46.430 [main] DEBUG nextflow.cli.CmdRun - \n",
      "  Version: 23.10.1 build 5891\n",
      "  Created: 12-01-2024 22:01 UTC (17:01 EDT)\n",
      "  System: Mac OS X 14.3\n",
      "  Runtime: Groovy 3.0.19 on OpenJDK 64-Bit Server VM 17.0.8+7-LTS\n",
      "  Encoding: UTF-8 (UTF-8)\n",
      "  Process: 20896@Eilas-MacBook-Air.local [172.20.20.20]\n",
      "  CPUs: 8 - Mem: 16 GB (317.2 MB) - Swap: 4 GB (1.5 GB)\n",
      "Feb-07 15:45:46.467 [main] DEBUG nextflow.Session - Work-dir: gs://eila-nextflow-bucket/work/ATAC [Mac OS X]\n",
      "Feb-07 15:45:46.487 [main] DEBUG nextflow.executor.ExecutorFactory - Extension executors providers=[GoogleLifeSciencesExecutor, GoogleBatchExecutor]\n",
      "Feb-07 15:45:46.496 [main] DEBUG nextflow.Session - Observer factory: DefaultObserverFactory\n",
      "Feb-07 15:45:46.499 [main] DEBUG nextflow.Session - Observer factory: WaveFactory\n",
      "Feb-07 15:45:46.499 [main] DEBUG io.seqera.wave.plugin.WaveFactory - Detected Fusion enabled -- Enabling bundle project resources -- Disabling upload of remote bin directory\n",
      "Feb-07 15:45:46.512 [main] DEBUG i.s.wave.plugin.config.WaveConfig - Wave strategy not specified - using default: [container, dockerfile, conda, spack]\n",
      "Feb-07 15:45:46.517 [main] DEBUG io.seqera.wave.plugin.WaveClient - Wave config: WaveConfig(enabled:true, endpoint:https://wave.seqera.io, containerConfigUrl:[], tokensCacheMaxDuration:30m, condaOpts:CondaOpts(mambaImage=mambaorg/micromamba:1.5.1; basePackages=conda-forge::procps-ng, commands=null), spackOpts:SpackOpts(basePackages=null, commands=null), strategy:[container, dockerfile, conda, spack], bundleProjectResources:true, buildRepository:null, cacheRepository:null, retryOpts:RetryOpts(delay:450ms, maxDelay:1m 30s, maxAttempts:10, jitter:0.25), httpClientOpts:HttpOpts(), freezeMode:null)\n",
      "Feb-07 15:45:46.616 [main] DEBUG nextflow.Session - Observer factory: TowerFactory\n",
      "Feb-07 15:45:46.651 [main] DEBUG nextflow.cache.CacheFactory - Using Nextflow cache factory: nextflow.cache.DefaultCacheFactory\n",
      "Feb-07 15:45:46.657 [main] DEBUG nextflow.util.CustomThreadPool - Creating default thread pool > poolSize: 9; maxThreads: 1000\n",
      "Feb-07 15:45:46.722 [main] DEBUG nextflow.Session - Session start\n",
      "Feb-07 15:45:46.730 [main] DEBUG io.seqera.tower.plugin.TowerClient - Creating Tower observer -- endpoint=https://api.tower.nf; requestInterval=1s; aliveInterval=1m; maxRetries=5; backOffBase=3; backOffDelay=250\n",
      "Feb-07 15:45:47.658 [main] DEBUG nextflow.script.ScriptRunner - > Launching execution\n",
      "Feb-07 15:45:48.437 [main] INFO  nextflow.script.BaseScript - /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/local_atac_fastqs_files.tsv\n",
      "Feb-07 15:45:48.438 [main] INFO  nextflow.script.BaseScript - eyJ0eXAiOiJKV1QiLCJraWQiOiJXN05OOldMSlQ6SjVSSzpMN1RMOlQ3TDc6M1ZYNjpKRU9VOjY0NFI6VTNJWDo1S1oyOjdaQ0s6RlBUSCIsImFsZyI6IlJTMjU2In0.eyJhY2Nlc3MiOnsic2NvcGUiOlsidmlldyIsImRvd25sb2FkIl0sIm9pZGNfY2xhaW1zIjp7fX0sInRva2VuX3R5cGUiOiJQRVJTT05BTF9BQ0NFU1NfVE9LRU4iLCJpc3MiOiJodHRwczovL3JlcG8tcHJvZC5wcm9kLnNhZ2ViYXNlLm9yZy9hdXRoL3YxIiwiYXVkIjoiMCIsIm5iZiI6MTY5NzgxODkxNiwiaWF0IjoxNjk3ODE4OTE2LCJqdGkiOiI0MTIzIiwic3ViIjoiMzQ3NDU3MSJ9.n0cugYMAVzYZ8mMMc7CjcqqEV7FB-K-oKUDT45SvM6KmbCpb_v3lZ2gg3xXi1tubWH0__wpnbJ7cYrGEtOyt4vXAPkMiEtZFGqS0MFKp_h-mHHH4pnACSQQYEy1zdyqFn9WRE7ipjWg4yYSbq84N7J3gzi8VKmiOAmuIGKaJaS0snkUjtJEVQH-0vZqWs1nKgVRWIZv31Bo8vuDzWpBrXD-zRT_uQW9l_I9fU50o1AqyFhNe_gbty_Z6imb8VT33wMXiDgJh15lG_M70PXU15v3egDXLgw1dGj32kj1nKanfqBVqbMSzgzPAG037p3YH_Q556BLceFBBb5Jaej5JVA\n",
      "Feb-07 15:45:48.486 [main] INFO  nextflow.script.BaseScript - after genome_fasta_ch\n",
      "Feb-07 15:45:48.513 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:zcat` matches labels `zcat` for process with name run_zcat\n",
      "Feb-07 15:45:48.516 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-07 15:45:48.516 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-07 15:45:48.517 [main] DEBUG nextflow.executor.Executor - [warm up] executor > google-batch (fusion enabled)\n",
      "Feb-07 15:45:48.519 [main] DEBUG n.processor.TaskPollingMonitor - Creating task monitor for executor 'google-batch' > capacity: 1000; pollInterval: 10s; dumpInterval: 5m \n",
      "Feb-07 15:45:48.521 [main] DEBUG n.processor.TaskPollingMonitor - >>> barrier register (monitor: google-batch)\n",
      "Feb-07 15:45:48.524 [main] DEBUG nextflow.cloud.google.GoogleOpts - Google auth via application credentials file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/hai-gcp-genomic-beb9207a40b8.json\n",
      "Feb-07 15:45:48.527 [main] DEBUG n.c.google.batch.GoogleBatchExecutor - [GOOGLE BATCH] Executor config=BatchConfig[googleOpts=GoogleOpts(projectId:hai-gcp-genomic, credsFile:/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/hai-gcp-genomic-beb9207a40b8.json, location:us-central1, enableRequesterPaysBuckets:false, httpConnectTimeout:1m, httpReadTimeout:1m, credentials:ServiceAccountCredentials{clientId=111364348320962814542, clientEmail=eila-batch@hai-gcp-genomic.iam.gserviceaccount.com, privateKeyId=beb9207a40b8b2085c7f48c3fcf8c2342180d471, transportFactoryClassName=com.google.auth.oauth2.OAuth2Utils$DefaultHttpTransportFactory, tokenServerUri=https://oauth2.googleapis.com/token, scopes=[https://www.googleapis.com/auth/cloud-platform], defaultScopes=[], serviceAccountUser=null, quotaProjectId=null, lifetime=3600, useJwtAccessWithScope=false, defaultRetriesEnabled=true})\n",
      "Feb-07 15:45:48.537 [main] DEBUG n.c.google.batch.client.BatchClient - [GOOGLE BATCH] Creating service client with config credentials\n",
      "Feb-07 15:45:49.063 [main] INFO  nextflow.script.BaseScript - after run_zcat \n",
      "Feb-07 15:45:49.069 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:gunzip` matches labels `gunzip` for process with name run_whitelist_gunzip\n",
      "Feb-07 15:45:49.070 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-07 15:45:49.070 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-07 15:45:49.072 [main] INFO  nextflow.script.BaseScript - after run_whitelist_gunzip\n",
      "Feb-07 15:45:49.073 [main] INFO  nextflow.script.BaseScript - after load chrompa mapping debug\n",
      "Feb-07 15:45:49.075 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:pool_fragments` matches labels `pool_fragments` for process with name run_add_subpool_prefix_to_fragment_table\n",
      "Feb-07 15:45:49.076 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-07 15:45:49.076 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-07 15:45:49.078 [main] INFO  nextflow.script.BaseScript - before call bgzip\n",
      "Feb-07 15:45:49.079 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:bgzip` matches labels `bgzip` for process with name run_bgzip_on_chromap_fragments_output\n",
      "Feb-07 15:45:49.079 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-07 15:45:49.079 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-07 15:45:49.081 [main] INFO  nextflow.script.BaseScript - after call bgzip\n",
      "Feb-07 15:45:49.081 [main] INFO  nextflow.script.BaseScript - before call run_tabix_chromap with tabix shell script\n",
      "Feb-07 15:45:49.082 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:tabix` matches labels `tabix` for process with name run_tabix_chromap\n",
      "Feb-07 15:45:49.082 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-07 15:45:49.082 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-07 15:45:49.083 [main] INFO  nextflow.script.BaseScript - after call run_tabix_chromap\n",
      "Feb-07 15:45:49.083 [main] INFO  nextflow.script.BaseScript - before call run_merge_chromap_logs\n",
      "Feb-07 15:45:49.085 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:merge_log` matches labels `merge_log` for process with name run_merge_chromap_logs\n",
      "Feb-07 15:45:49.085 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-07 15:45:49.085 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-07 15:45:49.087 [main] INFO  nextflow.script.BaseScript - after call run_merge_chromap_logs\n",
      "Feb-07 15:45:49.087 [main] INFO  nextflow.script.BaseScript - before run_add_subpool_prefix_to_summary_file\n",
      "Feb-07 15:45:49.088 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:pool_dictionary` matches labels `pool_dictionary` for process with name run_process_conversion_to_barcode\n",
      "Feb-07 15:45:49.089 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-07 15:45:49.089 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-07 15:45:49.091 [main] INFO  nextflow.script.BaseScript - after run_add_subpool_prefix_to_summary_file\n",
      "Feb-07 15:45:49.091 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:pool_filter` matches labels `pool_filter` for process with name run_filter_aligned_fragments\n",
      "Feb-07 15:45:49.092 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-07 15:45:49.092 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-07 15:45:49.095 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:tabix_singelston` matches labels `tabix_singelston` for process with name run_tabix_no_singleton\n",
      "Feb-07 15:45:49.096 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-07 15:45:49.096 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-07 15:45:49.097 [main] INFO  nextflow.script.BaseScript - before call run_calculate_tss_enrichment_bulk\n",
      "Feb-07 15:45:49.098 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:tss_bulk` matches labels `tss_bulk` for process with name run_calculate_tss_enrichment_bulk\n",
      "Feb-07 15:45:49.099 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-07 15:45:49.099 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-07 15:45:49.100 [main] INFO  nextflow.script.BaseScript - after call run_calculate_tss_enrichment_bulk\n",
      "Feb-07 15:45:49.101 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:gzip` matches labels `gzip` for process with name run_gzip_on_genes_gtf\n",
      "Feb-07 15:45:49.102 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-07 15:45:49.102 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-07 15:45:49.103 [main] INFO  nextflow.script.BaseScript - after run_gzip_on_genes_gtf\n",
      "Feb-07 15:45:49.103 [main] INFO  nextflow.script.BaseScript - before call run_calculate_tss_enrichment_snapatac2\n",
      "Feb-07 15:45:49.103 [main] INFO  nextflow.script.BaseScript - after call run_calculate_tss_enrichment_snapatac2\n",
      "Feb-07 15:45:49.104 [main] INFO  nextflow.script.BaseScript - before run_generate_insert_size_histogram_data\n",
      "Feb-07 15:45:49.105 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:zcat_insert` matches labels `zcat_insert` for process with name run_generate_insert_size_histogram_data\n",
      "Feb-07 15:45:49.105 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-07 15:45:49.105 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-07 15:45:49.106 [main] INFO  nextflow.script.BaseScript - after run_generate_insert_size_histogram_data\n",
      "Feb-07 15:45:49.106 [main] INFO  nextflow.script.BaseScript - before call run_generate_insert_size_plot\n",
      "Feb-07 15:45:49.109 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Feb-07 15:45:49.109 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Feb-07 15:45:49.110 [main] INFO  nextflow.script.BaseScript - after call run_generate_insert_size_plot\n",
      "Feb-07 15:45:49.110 [main] INFO  nextflow.script.BaseScript - before run_barcode_metadata_stats\n",
      "Feb-07 15:45:49.110 [main] INFO  nextflow.script.BaseScript - after run_barcode_metadata_stats\n",
      "Feb-07 15:45:49.110 [main] INFO  nextflow.script.BaseScript - before call run_atac_barcode_rank_plot\n",
      "Feb-07 15:45:49.110 [main] INFO  nextflow.script.BaseScript - after run_atac_barcode_rank_plot\n",
      "Feb-07 15:45:49.111 [main] DEBUG nextflow.Session - Workflow process names [dsl2]: run_bgzip_on_chromap_fragments_output, run_seqspec_print, run_calculate_tss_enrichment_snapatac2, run_process_conversion_to_barcode, run_barcode_metadata_stats, run_tabix_chromap, run_calculate_tss_enrichment_bulk, run_generate_insert_size_plot, run_seqspec_test, run_tabix_no_singleton, run_chromap_map_to_idx, run_atac_merge_barcode_metadata, run_filter_aligned_fragments, run_seqspec_modify_rna, run_atac_barcode_rank_plot, run_gzip_on_genes_gtf, run_whitelist_gunzip, run_merge_chromap_logs, run_create_chromap_idx, run_generate_insert_size_histogram_data, run_seqspec_check, run_zcat, run_seqspec_index_rna_kb, run_seqspec_modify_atac, run_chromap_test, run_tabix_filtered_fragments, run_add_subpool_prefix_to_fragment_table, run_downloadFiles\n",
      "Feb-07 15:45:49.458 [main] INFO  io.seqera.tower.plugin.TowerClient - Monitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/4YUdmqfnbjPozx\n",
      "Feb-07 15:45:49.458 [main] DEBUG nextflow.Session - Igniting dataflow network (22)\n",
      "Feb-07 15:45:49.462 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_zcat\n",
      "Feb-07 15:45:49.463 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_whitelist_gunzip\n",
      "Feb-07 15:45:49.463 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_add_subpool_prefix_to_fragment_table\n",
      "Feb-07 15:45:49.463 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_bgzip_on_chromap_fragments_output\n",
      "Feb-07 15:45:49.464 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_tabix_chromap\n",
      "Feb-07 15:45:49.464 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_merge_chromap_logs\n",
      "Feb-07 15:45:49.464 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_process_conversion_to_barcode\n",
      "Feb-07 15:45:49.464 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_filter_aligned_fragments\n",
      "Feb-07 15:45:49.464 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_tabix_no_singleton\n",
      "Feb-07 15:45:49.464 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_calculate_tss_enrichment_bulk\n",
      "Feb-07 15:45:49.465 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_gzip_on_genes_gtf\n",
      "Feb-07 15:45:49.465 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_generate_insert_size_histogram_data\n",
      "Feb-07 15:45:49.465 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_generate_insert_size_plot\n",
      "Feb-07 15:45:49.465 [main] DEBUG nextflow.script.ScriptRunner - Parsed script files:\n",
      "  Script_87a669c832149f5a: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_barcode_subpool.nf\n",
      "  Script_fec0f9da3ae38823: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_tss.nf\n",
      "  Script_985c81f2bb4fd175: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_atac_generate_metadata_annoation.nf\n",
      "  Script_67e45395c241a45f: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_chromap_utils.nf\n",
      "  Script_9d4639c8cd883c70: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_download_url_files.nf\n",
      "  Script_bc539eb5853c51b8: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_gunzip.nf\n",
      "  Script_d4029f31196c8525: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nf_wf_fg_singlecell_atac.nf\n",
      "  Script_6e882383f934962a: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_atac_generate_insert_size_plot.nf\n",
      "  Script_280f31ae5ae5ed4d: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_tabix.nf\n",
      "  Script_5df81d0ae421efa0: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_bgzip.nf\n",
      "  Script_e112646bcdb899f0: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_seqspec_utils.nf\n",
      "  Script_8081eedda99eaa43: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_merge_logs.nf\n",
      "  Script_48127624ae4a6640: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_zcat.nf\n",
      "  Script_7f43a272cfe3ae1b: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_generate_merged_barcode_metadata.nf\n",
      "  Script_34d6d3540f17eeb9: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_gzip.nf\n",
      "Feb-07 15:45:49.466 [main] DEBUG nextflow.script.ScriptRunner - > Awaiting termination \n",
      "Feb-07 15:45:49.466 [main] DEBUG nextflow.Session - Session await\n",
      "Feb-07 15:45:50.483 [FileTransfer-1] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/7f/c06fc55cbf9bab0ee4c3d2a86b8bd6/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\n",
      "Feb-07 15:45:50.493 [Actor Thread 3] DEBUG i.s.wave.plugin.config.WaveConfig - Wave strategy not specified - using default: [container, dockerfile, conda, spack]\n",
      "Feb-07 15:45:50.493 [Actor Thread 3] DEBUG io.seqera.wave.plugin.WaveClient - Wave config: WaveConfig(enabled:true, endpoint:https://wave.seqera.io, containerConfigUrl:[], tokensCacheMaxDuration:30m, condaOpts:CondaOpts(mambaImage=mambaorg/micromamba:1.5.1; basePackages=conda-forge::procps-ng, commands=null), spackOpts:SpackOpts(basePackages=null, commands=null), strategy:[container, dockerfile, conda, spack], bundleProjectResources:true, buildRepository:null, cacheRepository:null, retryOpts:RetryOpts(delay:450ms, maxDelay:1m 30s, maxAttempts:10, jitter:0.25), httpClientOpts:HttpOpts(), freezeMode:null)\n",
      "Feb-07 15:45:50.497 [Actor Thread 3] DEBUG io.seqera.wave.plugin.WaveClient - Wave assets bundle bin resources: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/bin\n",
      "Feb-07 15:45:50.687 [Actor Thread 3] DEBUG io.seqera.wave.plugin.WaveClient - Wave request container config: https://fusionfs.seqera.io/releases/v2.2-amd64.json\n",
      "Feb-07 15:45:50.878 [FileTransfer-2] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/genes.gtf at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/eb/6412854cc6c199859cdfbe280f86c9/genes.gtf\n",
      "Feb-07 15:45:51.102 [Actor Thread 3] DEBUG io.seqera.wave.plugin.WaveClient - Wave container config response: [200] {\n",
      "  \"entrypoint\": [ \"/usr/bin/fusion\" ],\n",
      "  \"layers\": [\n",
      "    {\n",
      "      \"location\": \"https://fusionfs.seqera.io/releases/pkg/2/2/12/fusion-amd64.tar.gz\",\n",
      "      \"gzipDigest\": \"sha256:6ccfd8f431db2874e82e27d61b6d31b95d7dddce7e810c5f8c95438a0d03c066\",\n",
      "      \"gzipSize\": 30262575,\n",
      "      \"tarDigest\": \"sha256:60391bebe0cc5d48b0875854f0c93c0d0adc25b5007fdb1b2efc2a431cfeef6e\",\n",
      "      \"skipHashing\": true\n",
      "    }\n",
      "  ]\n",
      "}\n",
      "\n",
      "Feb-07 15:45:51.173 [Actor Thread 8] DEBUG io.seqera.wave.plugin.WaveClient - Wave request: https://wave.seqera.io/container-token; attempt=1 - request: SubmitContainerTokenRequest(towerAccessToken:eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz, towerRefreshToken:null, towerWorkspaceId:247507587970818, towerEndpoint:https://api.tower.nf, workflowId:null, containerImage:eilalan/utils_env:latest, containerFile:null, containerConfig:ContainerConfig(entrypoint:[/usr/bin/fusion], cmd:null, env:null, workingDir:null, layers:[ContainerLayer[location=data:H4sIAAAAAAAA/+19bXvb...; tarDigest=sha256:d4a33c89bf487eb106fd5ec72ee517a9bc603ec15673059b84eb7688fe5ac9cc; gzipDigest=sha256:aa8d94321b98c31848c4f6f6caa54879385123d552fa0688f75a4debf5ee5b52; gzipSize=11292], ContainerLayer[location=https://fusionfs.seqera.io/releases/pkg/2/2/12/fusion-amd64.tar.gz; tarDigest=sha256:60391bebe0cc5d48b0875854f0c93c0d0adc25b5007fdb1b2efc2a431cfeef6e; gzipDigest=sha256:6ccfd8f431db2874e82e27d61b6d31b95d7dddce7e810c5f8c95438a0d03c066; gzipSize=30262575]]), condaFile:null, spackFile:null, containerPlatform:linux/amd64, buildRepository:null, cacheRepository:null, timestamp:2024-02-07T15:45:51.171868-05:00, fingerprint:5cdc7e72317ec605b2d1994c130829b6, freeze:false, format:null, dryRun:false)\n",
      "Feb-07 15:45:51.977 [Actor Thread 8] DEBUG io.seqera.wave.plugin.WaveClient - Wave response: statusCode=200; body={\"containerToken\":\"5ac4fe63c099\",\"targetImage\":\"wave.seqera.io/wt/5ac4fe63c099/eilalan/utils_env:latest\",\"expiration\":\"2024-02-09T08:45:51.965796878Z\",\"containerImage\":\"docker.io/eilalan/utils_env:latest\"}\n",
      "Feb-07 15:45:52.515 [Actor Thread 3] INFO  nextflow.processor.TaskProcessor - [3f/f09a4d] Cached process > run_zcat\n",
      "Feb-07 15:45:52.584 [Actor Thread 8] INFO  nextflow.processor.TaskProcessor - [67/a1b7db] Cached process > run_gzip_on_genes_gtf (1)\n",
      "Feb-07 15:45:52.652 [FileTransfer-3] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/6f/e3ee295a7f5364c39365c48e8bd0b0/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\n",
      "Feb-07 15:45:52.831 [FileTransfer-7] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/d5/12b6921f983d2326fa1e91570a96e7/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\n",
      "Feb-07 15:45:52.856 [FileTransfer-5] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/4a/7b5d74b3a4903977d7316c5f02ef84/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\n",
      "Feb-07 15:45:52.859 [FileTransfer-4] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/c5/cc9149d8d21bb423337888a9dc77ea/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\n",
      "Feb-07 15:45:52.859 [FileTransfer-6] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/dc/65201d13e12dffd8779bbc935604a6/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\n",
      "Feb-07 15:45:52.932 [FileTransfer-3] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/chromap_out_debug/BMMC-single-donor.atac.align.barcode.summary.csv at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/83/230e2c5cb965e17c572d46066ce07f/BMMC-single-donor.atac.align.barcode.summary.csv\n",
      "Feb-07 15:45:52.940 [FileTransfer-2] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/chromap_out_debug/ATAC_BMMC-single-donor.atac.filter.fragments.tsv at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/12/a7aa6d9e76c7234da48d36951f96c6/ATAC_BMMC-single-donor.atac.filter.fragments.tsv\n",
      "Feb-07 15:45:52.941 [FileTransfer-8] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/2d/0e59fd7f9797eaf684de4abc52eb46/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\n",
      "Feb-07 15:45:52.942 [FileTransfer-1] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/na1.na at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/38/b73884d031cf00d6089d843ba0f965/na1.na\n",
      "Feb-07 15:45:52.944 [FileTransfer-10] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/sai_192_whitelist.txt at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/09/7ba70bbeac00bcd6547ad0c09655c7/sai_192_whitelist.txt\n",
      "Feb-07 15:45:52.945 [FileTransfer-9] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/na.na at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/f4/bab8ffeff41eecffb2ddcf013b4121/na.na\n",
      "Feb-07 15:45:54.123 [Actor Thread 7] INFO  nextflow.processor.TaskProcessor - [15/0ac619] Cached process > run_whitelist_gunzip (1)\n",
      "Feb-07 15:45:54.484 [Actor Thread 10] INFO  nextflow.processor.TaskProcessor - [59/d22a1d] Cached process > run_add_subpool_prefix_to_fragment_table (1)\n",
      "Feb-07 15:45:54.767 [FileTransfer-7] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/chromap_out_debug/BMMC-single-donor.atac.align.k4.hg38.log.txt at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/ad/37cb0426d11a8313b27a0af2761113/BMMC-single-donor.atac.align.k4.hg38.log.txt\n",
      "Feb-07 15:45:55.267 [Actor Thread 5] INFO  nextflow.processor.TaskProcessor - [a9/95c478] Cached process > run_merge_chromap_logs (1)\n",
      "Feb-07 15:45:55.778 [Actor Thread 9] INFO  nextflow.processor.TaskProcessor - [3c/fca9ef] Cached process > run_process_conversion_to_barcode (1)\n",
      "Feb-07 15:45:56.396 [Actor Thread 5] INFO  nextflow.processor.TaskProcessor - [3b/05be46] Cached process > run_filter_aligned_fragments (1)\n",
      "Feb-07 15:45:56.826 [Actor Thread 9] INFO  nextflow.processor.TaskProcessor - [b3/94bc73] Cached process > run_tabix_no_singleton (1)\n",
      "Feb-07 15:45:57.023 [FileTransfer-5] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/hg38.synapse_transcripts.TSS.bed at gs://eila-nextflow-bucket/work/ATAC/stage-ab09407f-2f8c-43e1-855a-210d7286a110/0c/437c3f4b510f46db73da468060710f/hg38.synapse_transcripts.TSS.bed\n",
      "Feb-07 15:45:57.033 [Actor Thread 10] DEBUG io.seqera.wave.plugin.WaveClient - Wave request: https://wave.seqera.io/container-token; attempt=1 - request: SubmitContainerTokenRequest(towerAccessToken:eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz, towerRefreshToken:null, towerWorkspaceId:247507587970818, towerEndpoint:https://api.tower.nf, workflowId:null, containerImage:eilalan/tss_env:latest, containerFile:null, containerConfig:ContainerConfig(entrypoint:[/usr/bin/fusion], cmd:null, env:null, workingDir:null, layers:[ContainerLayer[location=data:H4sIAAAAAAAA/+19bXvb...; tarDigest=sha256:d4a33c89bf487eb106fd5ec72ee517a9bc603ec15673059b84eb7688fe5ac9cc; gzipDigest=sha256:aa8d94321b98c31848c4f6f6caa54879385123d552fa0688f75a4debf5ee5b52; gzipSize=11292], ContainerLayer[location=https://fusionfs.seqera.io/releases/pkg/2/2/12/fusion-amd64.tar.gz; tarDigest=sha256:60391bebe0cc5d48b0875854f0c93c0d0adc25b5007fdb1b2efc2a431cfeef6e; gzipDigest=sha256:6ccfd8f431db2874e82e27d61b6d31b95d7dddce7e810c5f8c95438a0d03c066; gzipSize=30262575]]), condaFile:null, spackFile:null, containerPlatform:linux/amd64, buildRepository:null, cacheRepository:null, timestamp:2024-02-07T15:45:57.032695-05:00, fingerprint:8f04bb44a233006c7e7c93d2701bcdbb, freeze:false, format:null, dryRun:false)\n",
      "Feb-07 15:45:57.210 [Actor Thread 10] DEBUG io.seqera.wave.plugin.WaveClient - Wave response: statusCode=200; body={\"containerToken\":\"ca967caac0db\",\"targetImage\":\"wave.seqera.io/wt/ca967caac0db/eilalan/tss_env:latest\",\"expiration\":\"2024-02-09T08:45:57.198983276Z\",\"containerImage\":\"docker.io/eilalan/tss_env:latest\"}\n",
      "Feb-07 15:45:57.310 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_bgzip_on_chromap_fragments_output (1)` submitted > job=nf-70cf6256-1707338754932; uid=nf-70cf6256-170733-61cb9f1c-d513-44420; work-dir=gs://eila-nextflow-bucket/work/ATAC/70/cf6256904289609bf995f0b60064ce\n",
      "Feb-07 15:45:57.310 [Task submitter] INFO  nextflow.Session - [70/cf6256] Submitted process > run_bgzip_on_chromap_fragments_output (1)\n",
      "Feb-07 15:45:58.673 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_calculate_tss_enrichment_bulk (1)` - adding local volume as fusion scratch: DiskResource(request:375 GB, type:local-ssd)\n",
      "Feb-07 15:45:59.033 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_calculate_tss_enrichment_bulk (1)` submitted > job=nf-8a8f19f8-1707338758220; uid=nf-8a8f19f8-170733-0a4642d1-e8d0-485c0; work-dir=gs://eila-nextflow-bucket/work/ATAC/8a/8f19f86c2ac97bd9ca2b62ae1266a2\n",
      "Feb-07 15:45:59.034 [Task submitter] INFO  nextflow.Session - [8a/8f19f8] Submitted process > run_calculate_tss_enrichment_bulk (1)\n",
      "Feb-07 15:50:08.898 [Task monitor] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_calculate_tss_enrichment_bulk (1)` - terminated job=nf-8a8f19f8-1707338758220; state=FAILED\n",
      "Feb-07 15:50:10.496 [Task monitor] DEBUG n.processor.TaskPollingMonitor - Task completed > TaskHandler[id: 10; name: run_calculate_tss_enrichment_bulk (1); status: COMPLETED; exit: 1; error: -; workDir: gs://eila-nextflow-bucket/work/ATAC/8a/8f19f86c2ac97bd9ca2b62ae1266a2]\n",
      "Feb-07 15:50:10.502 [Task monitor] DEBUG nextflow.processor.TaskProcessor - Handling unexpected condition for\n",
      "  task: name=run_calculate_tss_enrichment_bulk (1); work-dir=gs://eila-nextflow-bucket/work/ATAC/8a/8f19f86c2ac97bd9ca2b62ae1266a2\n",
      "  error [nextflow.exception.ProcessFailedException]: Process `run_calculate_tss_enrichment_bulk (1)` terminated with an error exit status (1)\n",
      "Feb-07 15:50:10.913 [Task monitor] ERROR nextflow.processor.TaskProcessor - Error executing process > 'run_calculate_tss_enrichment_bulk (1)'\n",
      "\n",
      "Caused by:\n",
      "  Process `run_calculate_tss_enrichment_bulk (1)` terminated with an error exit status (1)\n",
      "\n",
      "Command executed:\n",
      "\n",
      "  echo 'start run_calculate_tss_enrichment'\n",
      "  echo 'calculation_script is compute_tss_enrichment_bulk.py. check that it is: compute_tss_script.py'\n",
      "  echo 'tbi_fragments: no-singleton.bed.gz.tbi'\n",
      "  echo 'tss_bases_flank: 2000'\n",
      "  echo 'tss_col_strand_info: 4'\n",
      "  echo 'smoothing_window_size: 20'\n",
      "  python3 /usr/local/bin/compute_tss_enrichment_bulk.py no-singleton.bed.gz.tbi -e 2000 -s 4 -p 16 --prefix BMMC-single-donor --regions hg38.synapse_transcripts.TSS.bed\n",
      "  echo 'finished run_calculate_tss_enrichment'\n",
      "\n",
      "Command exit status:\n",
      "  1\n",
      "\n",
      "Command output:\n",
      "  start run_calculate_tss_enrichment\n",
      "  calculation_script is compute_tss_enrichment_bulk.py. check that it is: compute_tss_script.py\n",
      "  tbi_fragments: no-singleton.bed.gz.tbi\n",
      "  tss_bases_flank: 2000\n",
      "  tss_col_strand_info: 4\n",
      "  smoothing_window_size: 20\n",
      "  Command-line arguments: Namespace(e=2000, p=16, prefix='BMMC-single-donor', regions='hg38.synapse_transcripts.TSS.bed', s=4, tabix='no-singleton.bed.gz.tbi', window=20)\n",
      "\n",
      "Command error:\n",
      "  Traceback (most recent call last):\n",
      "    File \"/usr/local/bin/compute_tss_enrichment_bulk.py\", line 108, in <module>\n",
      "      count_args = _prepare_args_for_counting(args.tabix, regions_list, args.e, args.p)\n",
      "  NameError: name '_prepare_args_for_counting' is not defined\n",
      "\n",
      "Work dir:\n",
      "  gs://eila-nextflow-bucket/work/ATAC/8a/8f19f86c2ac97bd9ca2b62ae1266a2\n",
      "\n",
      "Tip: you can replicate the issue by changing to the process work dir and entering the command `bash .command.run`\n",
      "Feb-07 15:50:11.068 [Task monitor] DEBUG nextflow.Session - Session aborted -- Cause: Process `run_calculate_tss_enrichment_bulk (1)` terminated with an error exit status (1)\n",
      "Feb-07 15:50:11.092 [Task monitor] DEBUG nextflow.Session - The following nodes are still active:\n",
      "[process] run_tabix_chromap\n",
      "  status=ACTIVE\n",
      "  port 0: (value) bound ; channel: tabix_script\n",
      "  port 1: (queue) OPEN  ; channel: tabix_fragments\n",
      "  port 2: (cntrl) -     ; channel: $\n",
      "\n",
      "[process] run_generate_insert_size_histogram_data\n",
      "  status=ACTIVE\n",
      "  port 0: (value) bound ; channel: script_name\n",
      "  port 1: (queue) OPEN  ; channel: in_chromap_bzip_fragments_tsv\n",
      "  port 2: (cntrl) -     ; channel: $\n",
      "\n",
      "[process] run_generate_insert_size_plot\n",
      "  status=ACTIVE\n",
      "  port 0: (value) bound ; channel: plot_script\n",
      "  port 1: (queue) OPEN  ; channel: histogram_file\n",
      "  port 2: (value) bound ; channel: pkr_id\n",
      "  port 3: (cntrl) -     ; channel: $\n",
      "\n",
      "Feb-07 15:50:11.320 [main] DEBUG nextflow.Session - Session await > all processes finished\n",
      "Feb-07 15:50:11.321 [main] DEBUG nextflow.Session - Session await > all barriers passed\n",
      "Feb-07 15:50:11.322 [Task monitor] DEBUG n.processor.TaskPollingMonitor - <<< barrier arrives (monitor: google-batch) - terminating tasks monitor poll loop\n",
      "Feb-07 15:50:11.488 [main] WARN  n.processor.TaskPollingMonitor - Killing running tasks (1)\n",
      "Feb-07 15:50:11.502 [main] DEBUG n.trace.WorkflowStatsObserver - Workflow completed > WorkflowStats[succeededCount=0; failedCount=1; ignoredCount=0; cachedCount=8; pendingCount=0; submittedCount=0; runningCount=0; retriesCount=0; abortedCount=1; succeedDuration=0ms; failedDuration=1.6s; cachedDuration=3m 28s;loadCpus=0; loadMemory=0; peakRunning=2; peakCpus=2; peakMemory=0; ]\n",
      "Feb-07 15:50:11.885 [main] WARN  i.s.tower.plugin.TowerJsonGenerator - Tower request field `workflow.errorMessage` exceeds expected size | offending value: `Traceback (most recent call last):\n",
      "  File \"/usr/local/bin/compute_tss_enrichment_bulk.py\", line 108, in <module>\n",
      "    count_args = _prepare_args_for_counting(args.tabix, regions_list, args.e, args.p)\n",
      "NameError: name '_prepare_args_for_counting' is not defined`, size: 258 (max: 255)\n",
      "Feb-07 15:50:12.337 [main] DEBUG nextflow.cache.CacheDB - Closing CacheDB done\n",
      "Feb-07 15:50:12.338 [main] INFO  org.pf4j.AbstractPluginManager - Stop plugin 'nf-wave@1.0.1'\n",
      "Feb-07 15:50:12.338 [main] DEBUG nextflow.plugin.BasePlugin - Plugin stopped nf-wave\n",
      "Feb-07 15:50:12.338 [main] INFO  org.pf4j.AbstractPluginManager - Stop plugin 'nf-tower@1.6.3'\n",
      "Feb-07 15:50:12.338 [main] DEBUG nextflow.plugin.BasePlugin - Plugin stopped nf-tower\n",
      "Feb-07 15:50:12.338 [main] INFO  org.pf4j.AbstractPluginManager - Stop plugin 'nf-google@1.8.3'\n",
      "Feb-07 15:50:12.338 [main] DEBUG nextflow.plugin.BasePlugin - Plugin stopped nf-google\n",
      "Feb-07 15:50:12.387 [main] DEBUG nextflow.script.ScriptRunner - > Execution complete -- Goodbye\n"
     ]
    }
   ],
   "source": [
    "!cat .nextflow.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 3415,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3416,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3417,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !ls ../../nf_data/test_atac_pipeline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3418,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !gsutil cp gs://eila-nextflow-bucket/test_atac_pipeline/sai_192_whitelist.txt ../../nf_data/test_atac_pipeline/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3420,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ServiceException: 401 Anonymous caller does not have storage.objects.get access to the Google Cloud Storage object. Permission 'storage.objects.get' denied on resource (or it may not exist).\n"
     ]
    }
   ],
   "source": [
    "!gsutil cp gs://eila-nextflow-bucket/work/ATAC/b3/94bc73c6d43de41d17d75cac4fbda1/no-singleton.bed.gz.tbi /Users/eilaarich-landkof-stanford/Downloads/no-singleton.bed.gz.tbi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
