{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# start docker deamon in terminal\n",
    "# !sudo docker login\n",
    "# OR - becuase the jupyther with allow root\n",
    "# !docker login"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "TODO:\n",
    "\n",
    "when running with cloud. \n",
    "if you dont need to keep the execution logs, change the cloud settings to keep it for a short while\n",
    "Check that the cache if you need to set the cache to false (especially in plot process)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/hai-gcp-genomic-beb9207a40b8.json\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "json_path = os.path.join(os.path.dirname(os.getcwd()),\"hai-gcp-genomic-beb9207a40b8.json\")\n",
    "!ls $json_path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Activated service account credentials for: [eila-batch@hai-gcp-genomic.iam.gserviceaccount.com]\n"
     ]
    }
   ],
   "source": [
    "# # in the command line. same env\n",
    "import os\n",
    "json_path = os.path.join(os.path.dirname(os.getcwd()),\"hai-gcp-genomic-beb9207a40b8.json\")\n",
    "os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = json_path\n",
    "!gcloud auth activate-service-account eila-batch@hai-gcp-genomic.iam.gserviceaccount.com --key-file=$GOOGLE_APPLICATION_CREDENTIALS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                  Credentialed Accounts\n",
      "ACTIVE  ACCOUNT\n",
      "*       eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\n",
      "        eila@stanford.edu\n",
      "\n",
      "To set the active account, run:\n",
      "    $ gcloud config set account `ACCOUNT`\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'hai-gcp-genomic'"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "!gcloud auth list\n",
    "project_name = !gcloud config get-value project\n",
    "project_name = project_name[0]\n",
    "project_name\n",
    "# gcloud config set account\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "os.environ[\"TOWER_ACCESS_TOKEN\"] = 'eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz'\n",
    "os.environ[\"TOWER_WORKSPACE_ID\"]='247507587970818'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "247507587970818\n"
     ]
    }
   ],
   "source": [
    "!echo $TOWER_ACCESS_TOKEN\n",
    "!echo $TOWER_WORKSPACE_ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/bin'"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# !chmod +x ./bin/scrna_qc_metrics.py\n",
    "import os\n",
    "bin_fld = os.path.join(os.getcwd(),\"bin\")\n",
    "bin_fld"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "!chmod +x {os.path.join(bin_fld, \"scrna_plot_atac_qc_metrics.R\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"run_tabix.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"plot_insert_size_hist.py\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"generate_barcode_metadata.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"create_sample_filtered_fragment_file.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"compute_tss_script.py\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"barcode_rank_functions.R\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"atac_barcode_and_subpool.sh\")}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# # TODO: For a new user: run these commands with your service account for the execution. I am using eila-batch@...\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/iam.serviceAccountUser\"\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/batch.jobsEditor\"\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/logging.viewer\"\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !nextflow -self-update"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  Version: 23.10.1 build 5891\n",
      "  Created: 12-01-2024 22:01 UTC (14:01 PDT)\n",
      "  System: Mac OS X 14.2.1\n",
      "  Runtime: Groovy 3.0.19 on OpenJDK 64-Bit Server VM 17.0.8+7-LTS\n",
      "  Encoding: UTF-8 (UTF-8)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!nextflow info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "\n",
    "# Get the current working directory\n",
    "current_directory = os.getcwd()\n",
    "\n",
    "# Navigate two folders above\n",
    "two_folders_above = os.path.abspath(os.path.join(current_directory, '..', '..'))\n",
    "\n",
    "nf_data_base_path = os.path.join(two_folders_above,'nf_data')\n",
    "# Print the result\n",
    "print(nf_data_base_path)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data'"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nf_data_base_path\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "nf_data_project=\"test_atac_pipeline\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/local_atac_fastqs_files.tsv'"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.path.exists(os.path.join(nf_data_base_path,nf_data_project,\"local_atac_fastqs_files.tsv\"))\n",
    "os.path.join(nf_data_base_path,nf_data_project,\"local_atac_fastqs_files.tsv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "\n",
    "# Write the configuration file\n",
    "with open(\"nextflow_atac.config\", \"w\") as config_file:\n",
    "    config_file.write(f\"\"\"\n",
    "manifest {{\n",
    "  description = 'A nexflow pipeline for fg single cell atac' \n",
    "  mainScript = 'nf_wf_fg_singlecell_atac.nf'\n",
    "}}\n",
    "\n",
    "profiles {{\n",
    "  google_batch_profile {{\n",
    "    workDir = 'gs://eila-nextflow-bucket/work/ATAC'\n",
    "    includeConfig 'conf/docker-google-batch.config'\n",
    "    docker.enabled = true\n",
    "    wave.enabled = true\n",
    "    fusion.enabled = true\n",
    "    process.scratch = false\n",
    "    executor = 'google-batch'\n",
    "    docker.envWhitelist=['ENV_SYNAPSE_TOKEN','TOWER_ACCESS_TOKEN']\n",
    "    google.project = '{project_name}'\n",
    "    google.location = 'us-central1'\n",
    "    google.batch.spot = true\n",
    "    google.batch.serviceAccountEmail='eila-batch@hai-gcp-genomic.iam.gserviceaccount.com'\n",
    "  }}\n",
    "}}\n",
    "\n",
    "// Define ENV_SYNAPSE_TOKEN as a parameter\n",
    "params.FASTQS_SPEC_CH = '{nf_data_base_path}/{nf_data_project}/local_atac_fastqs_files.tsv'\n",
    "// CHROMAP\n",
    "// 6a\n",
    "params.CHROMAP_IDX='gs://eila-nextflow-bucket/test_atac_pipeline/ref.index'\n",
    "// 6b:\n",
    "// FASTA file\n",
    "params.CHROMAP_READ_LENGTH = 2000\n",
    "params.CHROMAP_BC_ERROR_THRESHOLD = 2\n",
    "params.CHROMAP_BC_PROBABILITY_THRESHOLD = 0.9\n",
    "params.CHROMAP_READ_FORMAT = 'bc:0:-1,r1:0:-1,r2:0:-1'\n",
    "params.CHROMAP_DROP_REPETITIVE_READS = 4\n",
    "params.CHROMAP_QUALITY_THRESHOLD = 0\n",
    "\n",
    "// CHROMAP DEBUG: load output files \n",
    "params.CHROMAP_GENOME_REFERENCE_FASTA='{nf_data_base_path}/{nf_data_project}/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz'\n",
    "params.chromap_filter_fragments_tsv ='{nf_data_base_path}/{nf_data_project}/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.atac.filter.fragments.tsv'\n",
    "params.barcode_summary_csv ='{nf_data_base_path}/{nf_data_project}/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.atac.align.barcode.summary.csv'\n",
    "\n",
    "// Other steps\n",
    "\n",
    "params.GENOME_GZ_GTF='http://ftp.ensembl.org/pub/release-109/gtf/homo_sapiens/Homo_sapiens.GRCh38.109.gtf.gz'\n",
    "params.ENV_SYNAPSE_TOKEN = \"nextflow secrets get SYNAPSE\".execute().text.trim()\n",
    "\n",
    "\n",
    "params.MAPQ_THRESHOLD = false\n",
    "params.BARCODE_TRANSLATE = 'path'\n",
    "params.ATAC_BARCODE_AND_POOL_SCRIPT='atac_barcode_and_subpool.sh'\n",
    "params.ATAC_FRAGMENTS_CUTOFF = 4\n",
    "params.RUN_TABIX_SCRIPT='run_tabix.sh'\n",
    "params.ATAC_TSS_BASES_FLANK = 2000\n",
    "params.ATAC_TSS_COL_WITH_STRANDS_INFO = 4\n",
    "params.ATAC_TSS_SMOOTHING_WINDOW_SIZE = 20\n",
    "params.ATAC_TSS_REGION_BED_FILE='{nf_data_base_path}nf_wf_fg_singlecell/tss_regions.bed'\n",
    "params.ATAC_TSS_CALCULATION_SCRIPT='compute_tss_script.py'\n",
    "\n",
    "\n",
    "params.SC_ATAC_QC_PLOT_SCRIPT='scrna_plot_atac_qc_metrics.R'\n",
    "params.SC_ATAC_QC_PLOT_HELPER_SCRIPT='barcode_rank_functions.R'\n",
    "params.SC_ATAC_QC_BARCODE_METADATA_FILE='{nf_data_base_path}nf_wf_fg_singlecell/barcode_metadata.txt'\n",
    "params.SC_ATAC_QC_FRAGMENT_CUTOFF=4\n",
    "params.SC_ATAC_QC_BARCODE_OUTPUT_FILE='sc_atac_qc_barcode_output_file.png'\n",
    "    \n",
    "params.SC_ATAC_BARCODE_METADATA_SCRIPT='generate_barcode_metadata.sh'\n",
    "params.SC_ATAC_BARCODE_METADATA_FILTERED_BARCODE_STATS='{nf_data_base_path}nf_wf_fg_singlecell/filtered_barcode_stats.csv'\n",
    "params.SC_ATAC_BARCODE_METADATA_TSS_ENRICHMENT_BARCODE_STATS='{nf_data_base_path}nf_wf_fg_singlecell/tss_enrichment_barcode_stats.csv'\n",
    "\n",
    "params.SC_ATAC_GENERATE_BARCODE_RANK_PLOT_SCRIPT='plot_insert_size_hist.py'\n",
    "params.SC_ATAC_GENERATE_BARCODE_RANK_PLOT_PKR='PKR123'\n",
    "\n",
    "params.SC_JOINT_CELL_BARCODE_PLOT_SCRTIP_PY='joint_cell_plotting.py'\n",
    "params.SC_JOINT_CELL_BARCODE_PLOT_SCRTIP_R='joint_cell_plotting_density.R'\n",
    "\"\"\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !rm -r /Users/eilaarich-landkof-stanford/Documents/Code/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/work\n",
    "# !rm -r /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/work\n",
    "# !gsutil rm -r gs://eila-nextflow-bucket/work/ATAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !rm .nextflow*\n",
    "\n",
    "# !ls /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/local_atac_fastqs_files.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "N E X T F L O W  ~  version 23.10.1\n",
      "Launching `nf_wf_fg_singlecell_atac.nf` [fervent_khorana] DSL2 - revision: 5d629d5acd\n",
      "/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/local_atac_fastqs_files.tsv\n",
      "eyJ0eXAiOiJKV1QiLCJraWQiOiJXN05OOldMSlQ6SjVSSzpMN1RMOlQ3TDc6M1ZYNjpKRU9VOjY0NFI6VTNJWDo1S1oyOjdaQ0s6RlBUSCIsImFsZyI6IlJTMjU2In0.eyJhY2Nlc3MiOnsic2NvcGUiOlsidmlldyIsImRvd25sb2FkIl0sIm9pZGNfY2xhaW1zIjp7fX0sInRva2VuX3R5cGUiOiJQRVJTT05BTF9BQ0NFU1NfVE9LRU4iLCJpc3MiOiJodHRwczovL3JlcG8tcHJvZC5wcm9kLnNhZ2ViYXNlLm9yZy9hdXRoL3YxIiwiYXVkIjoiMCIsIm5iZiI6MTY5NzgxODkxNiwiaWF0IjoxNjk3ODE4OTE2LCJqdGkiOiI0MTIzIiwic3ViIjoiMzQ3NDU3MSJ9.n0cugYMAVzYZ8mMMc7CjcqqEV7FB-K-oKUDT45SvM6KmbCpb_v3lZ2gg3xXi1tubWH0__wpnbJ7cYrGEtOyt4vXAPkMiEtZFGqS0MFKp_h-mHHH4pnACSQQYEy1zdyqFn9WRE7ipjWg4yYSbq84N7J3gzi8VKmiOAmuIGKaJaS0snkUjtJEVQH-0vZqWs1nKgVRWIZv31Bo8vuDzWpBrXD-zRT_uQW9l_I9fU50o1AqyFhNe_gbty_Z6imb8VT33wMXiDgJh15lG_M70PXU15v3egDXLgw1dGj32kj1nKanfqBVqbMSzgzPAG037p3YH_Q556BLceFBBb5Jaej5JVA\n",
      "after genome_fasta_ch\n",
      "[-        ] process > run_zcat                     -\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip         -\u001b[K\n",
      "[-        ] process > run_chromap_map_to_idx       -\u001b[K\n",
      "[-        ] process > run_bgzip                    -\u001b[K\n",
      "[-        ] process > run_tabix_chromap            -\u001b[K\n",
      "[-        ] process > run_merge_logs               -\u001b[K\n",
      "[-        ] process > run_filter_fragments         -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "The value na is present in subpoolValues. NO subpool\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix with tabix shell script\u001b[K\n",
      "after call run_tabix\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "after call run_tabix for filtered fragments\u001b[K\n",
      "before call run_calculate_tss_enrichment\u001b[K\n",
      "after regions_ch\u001b[K\n",
      "after call run_calculate_tss_enrichment\u001b[K\n",
      "before call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "after call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "before call run_generate_barcode_metadata\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "\u001b[30A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/5ubgGvXtfIdH0Y\u001b[22;39m\u001b[K\n",
      "[-        ] process > run_zcat                     -\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip         -\u001b[K\n",
      "[-        ] process > run_chromap_map_to_idx       -\u001b[K\n",
      "[-        ] process > run_bgzip                    -\u001b[K\n",
      "[-        ] process > run_tabix_chromap            -\u001b[K\n",
      "[-        ] process > run_merge_logs               -\u001b[K\n",
      "[-        ] process > run_filter_fragments         -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "The value na is present in subpoolValues. NO subpool\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix with tabix shell script\u001b[K\n",
      "after call run_tabix\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "after call run_tabix for filtered fragments\u001b[K\n",
      "before call run_calculate_tss_enrichment\u001b[K\n",
      "after regions_ch\u001b[K\n",
      "after call run_calculate_tss_enrichment\u001b[K\n",
      "before call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "after call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "before call run_generate_barcode_metadata\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "\u001b[31A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/5ubgGvXtfIdH0Y\u001b[22;39m\u001b[K\n",
      "[-        ] process > run_zcat                     -\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip         -\u001b[K\n",
      "[-        ] process > run_chromap_map_to_idx       -\u001b[K\n",
      "[-        ] process > run_bgzip                    -\u001b[K\n",
      "[-        ] process > run_tabix_chromap            -\u001b[K\n",
      "[-        ] process > run_merge_logs               -\u001b[K\n",
      "[-        ] process > run_filter_fragments         -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "The value na is present in subpoolValues. NO subpool\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix with tabix shell script\u001b[K\n",
      "after call run_tabix\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "after call run_tabix for filtered fragments\u001b[K\n",
      "before call run_calculate_tss_enrichment\u001b[K\n",
      "after regions_ch\u001b[K\n",
      "after call run_calculate_tss_enrichment\u001b[K\n",
      "before call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "after call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "before call run_generate_barcode_metadata\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "\u001b[32A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/5ubgGvXtfIdH0Y\u001b[22;39m\u001b[K\n",
      "[-        ] process > run_zcat                     -\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip         -\u001b[K\n",
      "[-        ] process > run_chromap_map_to_idx       -\u001b[K\n",
      "[-        ] process > run_bgzip                    -\u001b[K\n",
      "[-        ] process > run_tabix_chromap            -\u001b[K\n",
      "[-        ] process > run_merge_logs               -\u001b[K\n",
      "[-        ] process > run_filter_fragments         -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "The value na is present in subpoolValues. NO subpool\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix with tabix shell script\u001b[K\n",
      "after call run_tabix\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "after call run_tabix for filtered fragments\u001b[K\n",
      "before call run_calculate_tss_enrichment\u001b[K\n",
      "after regions_ch\u001b[K\n",
      "after call run_calculate_tss_enrichment\u001b[K\n",
      "before call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "after call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "before call run_generate_barcode_metadata\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "\u001b[33A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/5ubgGvXtfIdH0Y\u001b[22;39m\u001b[K\n",
      "[-        ] process > run_zcat                     -\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip         -\u001b[K\n",
      "[-        ] process > run_chromap_map_to_idx       -\u001b[K\n",
      "[-        ] process > run_bgzip                    -\u001b[K\n",
      "[-        ] process > run_tabix_chromap            -\u001b[K\n",
      "[-        ] process > run_merge_logs               -\u001b[K\n",
      "[-        ] process > run_filter_fragments         -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "The value na is present in subpoolValues. NO subpool\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix with tabix shell script\u001b[K\n",
      "after call run_tabix\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "after call run_tabix for filtered fragments\u001b[K\n",
      "before call run_calculate_tss_enrichment\u001b[K\n",
      "after regions_ch\u001b[K\n",
      "after call run_calculate_tss_enrichment\u001b[K\n",
      "before call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "after call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "before call run_generate_barcode_metadata\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "\u001b[34A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/5ubgGvXtfIdH0Y\u001b[22;39m\u001b[K\n",
      "[-        ] process > run_zcat                     -\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip         -\u001b[K\n",
      "[-        ] process > run_chromap_map_to_idx       -\u001b[K\n",
      "[-        ] process > run_bgzip                    -\u001b[K\n",
      "[-        ] process > run_tabix_chromap            -\u001b[K\n",
      "[-        ] process > run_merge_logs               -\u001b[K\n",
      "[-        ] process > run_filter_fragments         -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "The value na is present in subpoolValues. NO subpool\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix with tabix shell script\u001b[K\n",
      "after call run_tabix\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "after call run_tabix for filtered fragments\u001b[K\n",
      "before call run_calculate_tss_enrichment\u001b[K\n",
      "after regions_ch\u001b[K\n",
      "after call run_calculate_tss_enrichment\u001b[K\n",
      "before call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "after call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "before call run_generate_barcode_metadata\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "\u001b[35A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/5ubgGvXtfIdH0Y\u001b[22;39m\u001b[K\n",
      "[-        ] process > run_zcat                     -\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip         -\u001b[K\n",
      "[-        ] process > run_chromap_map_to_idx       -\u001b[K\n",
      "[-        ] process > run_bgzip                    -\u001b[K\n",
      "[-        ] process > run_tabix_chromap            -\u001b[K\n",
      "[-        ] process > run_merge_logs               -\u001b[K\n",
      "[-        ] process > run_filter_fragments         -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "The value na is present in subpoolValues. NO subpool\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix with tabix shell script\u001b[K\n",
      "after call run_tabix\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "after call run_tabix for filtered fragments\u001b[K\n",
      "before call run_calculate_tss_enrichment\u001b[K\n",
      "after regions_ch\u001b[K\n",
      "after call run_calculate_tss_enrichment\u001b[K\n",
      "before call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "after call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "before call run_generate_barcode_metadata\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "\u001b[36A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/5ubgGvXtfIdH0Y\u001b[22;39m\u001b[K\n",
      "[-        ] process > run_zcat                     -\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip         -\u001b[K\n",
      "[-        ] process > run_chromap_map_to_idx       -\u001b[K\n",
      "[-        ] process > run_bgzip                    -\u001b[K\n",
      "[-        ] process > run_tabix_chromap            -\u001b[K\n",
      "[-        ] process > run_merge_logs               -\u001b[K\n",
      "[-        ] process > run_filter_fragments         -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "The value na is present in subpoolValues. NO subpool\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix with tabix shell script\u001b[K\n",
      "after call run_tabix\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "after call run_tabix for filtered fragments\u001b[K\n",
      "before call run_calculate_tss_enrichment\u001b[K\n",
      "after regions_ch\u001b[K\n",
      "after call run_calculate_tss_enrichment\u001b[K\n",
      "before call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "after call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "before call run_generate_barcode_metadata\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "\u001b[37A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/5ubgGvXtfIdH0Y\u001b[22;39m\u001b[K\n",
      "[-        ] process > run_zcat                     -\u001b[K\n",
      "[-        ] process > run_whitelist_gunzip         -\u001b[K\n",
      "[-        ] process > run_chromap_map_to_idx       -\u001b[K\n",
      "[-        ] process > run_bgzip                    -\u001b[K\n",
      "[-        ] process > run_tabix_chromap            -\u001b[K\n",
      "[-        ] process > run_merge_logs               -\u001b[K\n",
      "[-        ] process > run_filter_fragments         -\u001b[K\n",
      "[-        ] process > run_tabix_filtered_fragments -\u001b[K\n",
      "after run_zcat \u001b[K\n",
      "after run_whitelist_gunzip\u001b[K\n",
      "finished run_chromap_map_to_idx\u001b[K\n",
      "The value na is present in subpoolValues. NO subpool\u001b[K\n",
      "before call bgzip\u001b[K\n",
      "after call bgzip\u001b[K\n",
      "before call run_tabix with tabix shell script\u001b[K\n",
      "after call run_tabix\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_merge_logs\u001b[K\n",
      "before call run_merge_logs\u001b[K\n",
      "after call run_filter_fragments\u001b[K\n",
      "before call run_tabix for the second time with filtered_fragment_file_out\u001b[K\n",
      "after call run_tabix for filtered fragments\u001b[K\n",
      "before call run_calculate_tss_enrichment\u001b[K\n",
      "after regions_ch\u001b[K\n",
      "after call run_calculate_tss_enrichment\u001b[K\n",
      "before call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "after call run_scrna_atac_plot_qc_metrics\u001b[K\n",
      "before call run_generate_barcode_metadata\u001b[K\n",
      "before call run_generate_barcode_rank_plot\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz\u001b[K\n",
      "Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz\u001b[K\n"
     ]
    }
   ],
   "source": [
    "!nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_atac.config -profile google_batch_profile -process.echo -with-tower -resume\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_vm.config -profile google_batch_profile\n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_vm.config -profile google_batch_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_vm_anvil.config -profile google_batch_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_local.config -profile google_batch_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_local.config -profile docker_profile\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Jan-31 12:43:02.428 [main] DEBUG nextflow.cli.Launcher - $> nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_atac.config -profile google_batch_profile -process.echo -with-tower -resume\n",
      "Jan-31 12:43:02.498 [main] INFO  nextflow.cli.CmdRun - N E X T F L O W  ~  version 23.10.1\n",
      "Jan-31 12:43:02.507 [main] DEBUG nextflow.plugin.PluginsFacade - Setting up plugin manager > mode=prod; embedded=false; plugins-dir=/Users/eilaarich-landkof-stanford/.nextflow/plugins; core-plugins: nf-amazon@2.1.4,nf-azure@1.3.3,nf-cloudcache@0.3.0,nf-codecommit@0.1.5,nf-console@1.0.6,nf-ga4gh@1.1.0,nf-google@1.8.3,nf-tower@1.6.3,nf-wave@1.0.1\n",
      "Jan-31 12:43:02.511 [main] INFO  o.pf4j.DefaultPluginStatusProvider - Enabled plugins: []\n",
      "Jan-31 12:43:02.512 [main] INFO  o.pf4j.DefaultPluginStatusProvider - Disabled plugins: []\n",
      "Jan-31 12:43:02.513 [main] INFO  org.pf4j.DefaultPluginManager - PF4J version 3.4.1 in 'deployment' mode\n",
      "Jan-31 12:43:02.518 [main] INFO  org.pf4j.AbstractPluginManager - No plugins\n",
      "Jan-31 12:43:02.527 [main] DEBUG nextflow.config.ConfigBuilder - User config file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Jan-31 12:43:02.527 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Jan-31 12:43:02.538 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `google_batch_profile`\n",
      "Jan-31 12:43:03.654 [main] DEBUG nextflow.config.ConfigBuilder - Available config profiles: [google_batch_profile]\n",
      "Jan-31 12:43:03.670 [main] DEBUG nextflow.cli.CmdRun - Applied DSL=2 by global default\n",
      "Jan-31 12:43:03.681 [main] INFO  nextflow.cli.CmdRun - Launching `nf_wf_fg_singlecell_atac.nf` [modest_mandelbrot] DSL2 - revision: eb7f6aba5e\n",
      "Jan-31 12:43:03.681 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins default=[nf-google@1.8.3]\n",
      "Jan-31 12:43:03.682 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins resolved requirement=[nf-google@1.8.3, nf-tower@1.6.3, nf-wave@1.0.1]\n",
      "Jan-31 12:43:03.682 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-google version: 1.8.3\n",
      "Jan-31 12:43:03.692 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-google@1.8.3' resolved\n",
      "Jan-31 12:43:03.693 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-google@1.8.3'\n",
      "Jan-31 12:43:03.725 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-google@1.8.3\n",
      "Jan-31 12:43:03.726 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-tower version: 1.6.3\n",
      "Jan-31 12:43:03.728 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-tower@1.6.3' resolved\n",
      "Jan-31 12:43:03.728 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-tower@1.6.3'\n",
      "Jan-31 12:43:03.731 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-tower@1.6.3\n",
      "Jan-31 12:43:03.731 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-wave version: 1.0.1\n",
      "Jan-31 12:43:03.734 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-wave@1.0.1' resolved\n",
      "Jan-31 12:43:03.734 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-wave@1.0.1'\n",
      "Jan-31 12:43:03.739 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-wave@1.0.1\n",
      "Jan-31 12:43:03.746 [main] DEBUG n.secret.LocalSecretsProvider - Secrets store: /Users/eilaarich-landkof-stanford/.nextflow/secrets/store.json\n",
      "Jan-31 12:43:03.747 [main] DEBUG nextflow.secret.SecretsLoader - Discovered secrets providers: [nextflow.secret.LocalSecretsProvider@d2291de] - activable => nextflow.secret.LocalSecretsProvider@d2291de\n",
      "Jan-31 12:43:03.786 [main] DEBUG nextflow.Session - Session UUID: 2946dea8-a84e-4812-9c1d-6d0f65982913\n",
      "Jan-31 12:43:03.787 [main] DEBUG nextflow.Session - Run name: modest_mandelbrot\n",
      "Jan-31 12:43:03.787 [main] DEBUG nextflow.Session - Executor pool size: 8\n",
      "Jan-31 12:43:03.881 [main] DEBUG nextflow.file.FilePorter - File porter settings maxRetries=3; maxTransfers=50; pollTimeout=null\n",
      "Jan-31 12:43:03.883 [main] DEBUG nextflow.util.ThreadPoolBuilder - Creating thread pool 'FileTransfer' minSize=10; maxSize=24; workQueue=LinkedBlockingQueue[10000]; allowCoreThreadTimeout=false\n",
      "Jan-31 12:43:03.885 [main] DEBUG nextflow.config.ConfigBuilder - User config file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Jan-31 12:43:03.885 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Jan-31 12:43:03.885 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `google_batch_profile`\n",
      "Jan-31 12:43:04.617 [main] DEBUG nextflow.config.ConfigBuilder - Available config profiles: [google_batch_profile]\n",
      "Jan-31 12:43:04.862 [main] DEBUG nextflow.cli.CmdRun - \n",
      "  Version: 23.10.1 build 5891\n",
      "  Created: 12-01-2024 22:01 UTC (14:01 PDT)\n",
      "  System: Mac OS X 14.2.1\n",
      "  Runtime: Groovy 3.0.19 on OpenJDK 64-Bit Server VM 17.0.8+7-LTS\n",
      "  Encoding: UTF-8 (UTF-8)\n",
      "  Process: 98136@Eilas-MacBook-Air.local [127.0.0.1]\n",
      "  CPUs: 8 - Mem: 16 GB (159.5 MB) - Swap: 1 GB (798.4 MB)\n",
      "Jan-31 12:43:04.880 [main] DEBUG nextflow.Session - Work-dir: gs://eila-nextflow-bucket/work/ATAC [Mac OS X]\n",
      "Jan-31 12:43:04.894 [main] DEBUG nextflow.executor.ExecutorFactory - Extension executors providers=[GoogleLifeSciencesExecutor, GoogleBatchExecutor]\n",
      "Jan-31 12:43:04.900 [main] DEBUG nextflow.Session - Observer factory: DefaultObserverFactory\n",
      "Jan-31 12:43:04.901 [main] DEBUG nextflow.Session - Observer factory: WaveFactory\n",
      "Jan-31 12:43:04.901 [main] DEBUG io.seqera.wave.plugin.WaveFactory - Detected Fusion enabled -- Enabling bundle project resources -- Disabling upload of remote bin directory\n",
      "Jan-31 12:43:04.909 [main] DEBUG i.s.wave.plugin.config.WaveConfig - Wave strategy not specified - using default: [container, dockerfile, conda, spack]\n",
      "Jan-31 12:43:04.912 [main] DEBUG io.seqera.wave.plugin.WaveClient - Wave config: WaveConfig(enabled:true, endpoint:https://wave.seqera.io, containerConfigUrl:[], tokensCacheMaxDuration:30m, condaOpts:CondaOpts(mambaImage=mambaorg/micromamba:1.5.1; basePackages=conda-forge::procps-ng, commands=null), spackOpts:SpackOpts(basePackages=null, commands=null), strategy:[container, dockerfile, conda, spack], bundleProjectResources:true, buildRepository:null, cacheRepository:null, retryOpts:RetryOpts(delay:450ms, maxDelay:1m 30s, maxAttempts:10, jitter:0.25), httpClientOpts:HttpOpts(), freezeMode:null)\n",
      "Jan-31 12:43:04.967 [main] DEBUG nextflow.Session - Observer factory: TowerFactory\n",
      "Jan-31 12:43:04.993 [main] DEBUG nextflow.cache.CacheFactory - Using Nextflow cache factory: nextflow.cache.DefaultCacheFactory\n",
      "Jan-31 12:43:04.998 [main] DEBUG nextflow.util.CustomThreadPool - Creating default thread pool > poolSize: 9; maxThreads: 1000\n",
      "Jan-31 12:43:05.030 [main] DEBUG nextflow.Session - Session start\n",
      "Jan-31 12:43:05.035 [main] DEBUG io.seqera.tower.plugin.TowerClient - Creating Tower observer -- endpoint=https://api.tower.nf; requestInterval=1s; aliveInterval=1m; maxRetries=5; backOffBase=3; backOffDelay=250\n",
      "Jan-31 12:43:06.198 [main] DEBUG nextflow.script.ScriptRunner - > Launching execution\n",
      "Jan-31 12:43:06.618 [main] INFO  nextflow.script.BaseScript - /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/local_atac_fastqs_files.tsv\n",
      "Jan-31 12:43:06.618 [main] INFO  nextflow.script.BaseScript - eyJ0eXAiOiJKV1QiLCJraWQiOiJXN05OOldMSlQ6SjVSSzpMN1RMOlQ3TDc6M1ZYNjpKRU9VOjY0NFI6VTNJWDo1S1oyOjdaQ0s6RlBUSCIsImFsZyI6IlJTMjU2In0.eyJhY2Nlc3MiOnsic2NvcGUiOlsidmlldyIsImRvd25sb2FkIl0sIm9pZGNfY2xhaW1zIjp7fX0sInRva2VuX3R5cGUiOiJQRVJTT05BTF9BQ0NFU1NfVE9LRU4iLCJpc3MiOiJodHRwczovL3JlcG8tcHJvZC5wcm9kLnNhZ2ViYXNlLm9yZy9hdXRoL3YxIiwiYXVkIjoiMCIsIm5iZiI6MTY5NzgxODkxNiwiaWF0IjoxNjk3ODE4OTE2LCJqdGkiOiI0MTIzIiwic3ViIjoiMzQ3NDU3MSJ9.n0cugYMAVzYZ8mMMc7CjcqqEV7FB-K-oKUDT45SvM6KmbCpb_v3lZ2gg3xXi1tubWH0__wpnbJ7cYrGEtOyt4vXAPkMiEtZFGqS0MFKp_h-mHHH4pnACSQQYEy1zdyqFn9WRE7ipjWg4yYSbq84N7J3gzi8VKmiOAmuIGKaJaS0snkUjtJEVQH-0vZqWs1nKgVRWIZv31Bo8vuDzWpBrXD-zRT_uQW9l_I9fU50o1AqyFhNe_gbty_Z6imb8VT33wMXiDgJh15lG_M70PXU15v3egDXLgw1dGj32kj1nKanfqBVqbMSzgzPAG037p3YH_Q556BLceFBBb5Jaej5JVA\n",
      "Jan-31 12:43:06.652 [main] INFO  nextflow.script.BaseScript - after genome_fasta_ch\n",
      "Jan-31 12:43:06.668 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:zcat` matches labels `zcat` for process with name run_zcat\n",
      "Jan-31 12:43:06.670 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Jan-31 12:43:06.670 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Jan-31 12:43:06.671 [main] DEBUG nextflow.executor.Executor - [warm up] executor > google-batch (fusion enabled)\n",
      "Jan-31 12:43:06.672 [main] DEBUG n.processor.TaskPollingMonitor - Creating task monitor for executor 'google-batch' > capacity: 1000; pollInterval: 10s; dumpInterval: 5m \n",
      "Jan-31 12:43:06.673 [main] DEBUG n.processor.TaskPollingMonitor - >>> barrier register (monitor: google-batch)\n",
      "Jan-31 12:43:06.676 [main] DEBUG nextflow.cloud.google.GoogleOpts - Google auth via application credentials file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/hai-gcp-genomic-beb9207a40b8.json\n",
      "Jan-31 12:43:06.678 [main] DEBUG n.c.google.batch.GoogleBatchExecutor - [GOOGLE BATCH] Executor config=BatchConfig[googleOpts=GoogleOpts(projectId:hai-gcp-genomic, credsFile:/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/hai-gcp-genomic-beb9207a40b8.json, location:us-central1, enableRequesterPaysBuckets:false, httpConnectTimeout:1m, httpReadTimeout:1m, credentials:ServiceAccountCredentials{clientId=111364348320962814542, clientEmail=eila-batch@hai-gcp-genomic.iam.gserviceaccount.com, privateKeyId=beb9207a40b8b2085c7f48c3fcf8c2342180d471, transportFactoryClassName=com.google.auth.oauth2.OAuth2Utils$DefaultHttpTransportFactory, tokenServerUri=https://oauth2.googleapis.com/token, scopes=[https://www.googleapis.com/auth/cloud-platform], defaultScopes=[], serviceAccountUser=null, quotaProjectId=null, lifetime=3600, useJwtAccessWithScope=false, defaultRetriesEnabled=true})\n",
      "Jan-31 12:43:06.685 [main] DEBUG n.c.google.batch.client.BatchClient - [GOOGLE BATCH] Creating service client with config credentials\n",
      "Jan-31 12:43:07.165 [main] INFO  nextflow.script.BaseScript - after run_zcat \n",
      "Jan-31 12:43:07.168 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:gunzip` matches labels `gunzip` for process with name run_whitelist_gunzip\n",
      "Jan-31 12:43:07.169 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Jan-31 12:43:07.169 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Jan-31 12:43:07.170 [main] INFO  nextflow.script.BaseScript - after run_whitelist_gunzip\n",
      "Jan-31 12:43:07.172 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:chromap_map_idx` matches labels `chromap_map_idx` for process with name run_chromap_map_to_idx\n",
      "Jan-31 12:43:07.172 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Jan-31 12:43:07.172 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Jan-31 12:43:07.174 [main] INFO  nextflow.script.BaseScript - finished run_chromap_map_to_idx\n",
      "Jan-31 12:43:07.175 [main] INFO  nextflow.script.BaseScript - The value na is present in subpoolValues. NO subpool\n",
      "Jan-31 12:43:07.175 [main] INFO  nextflow.script.BaseScript - before call bgzip\n",
      "Jan-31 12:43:07.175 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:bgzip` matches labels `bgzip` for process with name run_bgzip\n",
      "Jan-31 12:43:07.176 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Jan-31 12:43:07.176 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Jan-31 12:43:07.176 [main] INFO  nextflow.script.BaseScript - after call bgzip\n",
      "Jan-31 12:43:07.176 [main] INFO  nextflow.script.BaseScript - before call run_tabix with tabix shell script\n",
      "Jan-31 12:43:07.177 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:tabix` matches labels `tabix` for process with name run_tabix\n",
      "Jan-31 12:43:07.178 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Jan-31 12:43:07.178 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Jan-31 12:43:07.178 [main] INFO  nextflow.script.BaseScript - after call run_tabix\n",
      "Jan-31 12:43:07.178 [main] INFO  nextflow.script.BaseScript - before call run_merge_logs\n",
      "Jan-31 12:43:07.179 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:merge_log` matches labels `merge_log` for process with name run_merge_logs\n",
      "Jan-31 12:43:07.179 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Jan-31 12:43:07.179 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Jan-31 12:43:07.180 [main] INFO  nextflow.script.BaseScript - after call run_merge_logs\n",
      "Jan-31 12:43:07.180 [main] INFO  nextflow.script.BaseScript - before call run_merge_logs\n",
      "Jan-31 12:43:07.180 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:filter_fragments` matches labels `filter_fragments` for process with name run_filter_fragments\n",
      "Jan-31 12:43:07.181 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Jan-31 12:43:07.181 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Jan-31 12:43:07.182 [main] INFO  nextflow.script.BaseScript - after call run_filter_fragments\n",
      "Jan-31 12:43:07.182 [main] INFO  nextflow.script.BaseScript - before call run_tabix for the second time with filtered_fragment_file_out\n",
      "Jan-31 12:43:07.182 [main] INFO  nextflow.script.BaseScript - before call run_calculate_tss_enrichment\n",
      "Jan-31 12:43:07.182 [main] INFO  nextflow.script.BaseScript - after regions_ch\n",
      "Jan-31 12:43:07.182 [main] INFO  nextflow.script.BaseScript - after call run_calculate_tss_enrichment\n",
      "Jan-31 12:43:07.182 [main] INFO  nextflow.script.BaseScript - before call run_scrna_atac_plot_qc_metrics\n",
      "Jan-31 12:43:07.182 [main] INFO  nextflow.script.BaseScript - after call run_scrna_atac_plot_qc_metrics\n",
      "Jan-31 12:43:07.182 [main] INFO  nextflow.script.BaseScript - before call run_generate_barcode_metadata\n",
      "Jan-31 12:43:07.182 [main] INFO  nextflow.script.BaseScript - before call run_generate_barcode_rank_plot\n",
      "Jan-31 12:43:07.183 [main] DEBUG nextflow.Session - Workflow process names [dsl2]: run_add_pool_prefix, run_seqspec_modify_rna, run_merge_logs, run_seqspec_print, run_atac_barcode_rank_plot, run_bgzip, run_calculate_tss_enrichment, run_atac_barcode_metadata, run_tabix, run_filter_fragments, run_whitelist_gunzip, run_create_chromap_idx, run_seqspec_check, run_zcat, run_seqspec_test, run_seqspec_index_rna_kb, run_scrna_atac_plot_qc_metrics, run_seqspec_modify_atac, run_chromap_test, run_tabix_filtered_fragments, run_downloadFiles, run_chromap_map_to_idx\n",
      "Jan-31 12:43:07.619 [main] INFO  io.seqera.tower.plugin.TowerClient - Monitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/36vLx63vJR4a9y\n",
      "Jan-31 12:43:07.620 [main] DEBUG nextflow.Session - Igniting dataflow network (10)\n",
      "Jan-31 12:43:07.631 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_zcat\n",
      "Jan-31 12:43:07.632 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_whitelist_gunzip\n",
      "Jan-31 12:43:07.633 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_chromap_map_to_idx\n",
      "Jan-31 12:43:07.633 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_bgzip\n",
      "Jan-31 12:43:07.633 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_tabix\n",
      "Jan-31 12:43:07.633 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_merge_logs\n",
      "Jan-31 12:43:07.634 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_filter_fragments\n",
      "Jan-31 12:43:07.635 [main] DEBUG nextflow.script.ScriptRunner - Parsed script files:\n",
      "  Script_f6e231f5a61c42a9: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_filter_fragments.nf\n",
      "  Script_0de7f532c8b3da2c: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_bgzip.nf\n",
      "  Script_87d8d9fb99b69787: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_atac_barcode_rank_plot.nf\n",
      "  Script_cc895528cea919f6: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_zcat.nf\n",
      "  Script_ae8414c0634a8df3: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_tss.nf\n",
      "  Script_3fd3ab31f542f906: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_gunzip.nf\n",
      "  Script_9d4639c8cd883c70: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_download_url_files.nf\n",
      "  Script_9c2ce54dcf0b5c06: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_merge_logs.nf\n",
      "  Script_3cd14cf2699c764c: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nf_wf_fg_singlecell_atac.nf\n",
      "  Script_ea7cdb417e42517f: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_generate_barcode_metadata.nf\n",
      "  Script_e112646bcdb899f0: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_seqspec_utils.nf\n",
      "  Script_3d78df279f932afb: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_atac_qc_plots.nf\n",
      "  Script_caef5739a3ac2dec: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_tabix.nf\n",
      "  Script_d683fb4654d4632a: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_chromap_utils.nf\n",
      "Jan-31 12:43:07.635 [main] DEBUG nextflow.script.ScriptRunner - > Awaiting termination \n",
      "Jan-31 12:43:07.635 [main] DEBUG nextflow.Session - Session await\n",
      "Jan-31 12:43:09.307 [FileTransfer-1] DEBUG nextflow.file.FilePorter - Copying foreign file /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz to work dir: gs://eila-nextflow-bucket/work/ATAC/stage-2946dea8-a84e-4812-9c1d-6d0f65982913/85/0617d2729d181a62442f7ac2166894/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\n",
      "Jan-31 12:43:11.006 [Actor Thread 4] INFO  nextflow.file.FilePorter - Staging foreign file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\n",
      "Jan-31 12:43:17.564 [Actor Thread 8] DEBUG nextflow.processor.TaskProcessor - Process run_whitelist_gunzip > collision check staging file names: [BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz:1, BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz:1, na.na:2, BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz:1, BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz:1, BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz:1, BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz:1, sai_192_whitelist.txt:1]\n",
      "Jan-31 12:43:17.574 [Actor Thread 8] DEBUG nextflow.processor.TaskProcessor - Handling unexpected condition for\n",
      "  task: name=run_whitelist_gunzip (1); work-dir=null\n",
      "  error [nextflow.exception.ProcessUnrecoverableException]: Process `run_whitelist_gunzip` input file name collision -- There are multiple input files for each of the following file names: na.na\n",
      "Jan-31 12:43:17.587 [Actor Thread 8] ERROR nextflow.processor.TaskProcessor - Error executing process > 'run_whitelist_gunzip (1)'\n",
      "\n",
      "Caused by:\n",
      "  Process `run_whitelist_gunzip` input file name collision -- There are multiple input files for each of the following file names: na.na\n",
      "\n",
      "\n",
      "Tip: you can replicate the issue by changing to the process work dir and entering the command `bash .command.run`\n",
      "Jan-31 12:43:17.589 [Actor Thread 8] DEBUG nextflow.Session - Session aborted -- Cause: Process `run_whitelist_gunzip` input file name collision -- There are multiple input files for each of the following file names: na.na\n",
      "Jan-31 12:43:17.597 [Actor Thread 8] DEBUG nextflow.Session - The following nodes are still active:\n",
      "[process] run_zcat\n",
      "  status=ACTIVE\n",
      "  port 0: (value) bound ; channel: zcat_file\n",
      "  port 1: (cntrl) -     ; channel: $\n",
      "\n",
      "[process] run_whitelist_gunzip\n",
      "  status=ACTIVE\n",
      "  port 0: (queue) closed; channel: -\n",
      "  port 1: (cntrl) -     ; channel: $\n",
      "\n",
      "[process] run_chromap_map_to_idx\n",
      "  status=ACTIVE\n",
      "  port 0: (queue) closed; channel: -\n",
      "  port 1: (value) bound ; channel: genome_fasta\n",
      "  port 2: (value) bound ; channel: genome_chromap_idx\n",
      "  port 3: (value) bound ; channel: CHROMAP_QUALITY_THRESHOLD\n",
      "  port 4: (value) bound ; channel: CHROMAP_DROP_REPETITIVE_READS\n",
      "  port 5: (value) bound ; channel: CHROMAP_READ_FORMAT\n",
      "  port 6: (value) bound ; channel: CHROMAP_BC_PROBABILITY_THRESHOLD\n",
      "  port 7: (value) bound ; channel: CHROMAP_BC_ERROR_THRESHOLD\n",
      "  port 8: (value) bound ; channel: CHROMAP_READ_LENGTH\n",
      "  port 9: (value) bound ; channel: CPUS_TO_USE\n",
      "  port 10: (cntrl) -     ; channel: $\n",
      "\n",
      "[process] run_bgzip\n",
      "  status=ACTIVE\n",
      "  port 0: (queue) OPEN  ; channel: fragment_file\n",
      "  port 1: (cntrl) -     ; channel: $\n",
      "\n",
      "[process] run_tabix\n",
      "  status=ACTIVE\n",
      "  port 0: (value) bound ; channel: tabix_script\n",
      "  port 1: (queue) OPEN  ; channel: tabix_fragments\n",
      "  port 2: (cntrl) -     ; channel: $\n",
      "\n",
      "[process] run_merge_logs\n",
      "  status=ACTIVE\n",
      "  port 0: (queue) OPEN  ; channel: chromap_alignment_log\n",
      "  port 1: (queue) OPEN  ; channel: chromap_barcode_log\n",
      "  port 2: (cntrl) -     ; channel: $\n",
      "\n",
      "[process] run_filter_fragments\n",
      "  status=ACTIVE\n",
      "  port 0: (value) bound ; channel: atac_and_pool_bash_script\n",
      "  port 1: (queue) OPEN  ; channel: fragments_file\n",
      "  port 2: (value) bound ; channel: barcode_conversion_dict_file\n",
      "  port 3: (value) bound ; channel: subpool_in\n",
      "  port 4: (queue) OPEN  ; channel: barcode_summary\n",
      "  port 5: (value) bound ; channel: fragments_cutoff\n",
      "  port 6: (cntrl) -     ; channel: $\n",
      "\n",
      "Jan-31 12:43:17.599 [Task monitor] DEBUG n.processor.TaskPollingMonitor - <<< barrier arrives (monitor: google-batch) - terminating tasks monitor poll loop\n",
      "Jan-31 12:43:17.599 [main] DEBUG nextflow.Session - Session await > all processes finished\n",
      "Jan-31 12:43:17.599 [main] DEBUG nextflow.Session - Session await > all barriers passed\n",
      "Jan-31 12:43:17.599 [Actor Thread 6] DEBUG nextflow.processor.TaskProcessor - Handling unexpected condition for\n",
      "  task: name=run_zcat; work-dir=null\n",
      "  error [java.lang.InterruptedException]: java.lang.InterruptedException\n",
      "Jan-31 12:43:17.600 [Actor Thread 9] DEBUG nextflow.processor.TaskProcessor - Handling unexpected condition for\n",
      "  task: name=run_chromap_map_to_idx; work-dir=null\n",
      "  error [java.lang.InterruptedException]: java.lang.InterruptedException\n",
      "Jan-31 12:43:17.600 [Actor Thread 5] DEBUG nextflow.processor.TaskProcessor - Handling unexpected condition for\n",
      "  task: name=run_whitelist_gunzip; work-dir=null\n",
      "  error [java.lang.InterruptedException]: java.lang.InterruptedException\n",
      "Jan-31 12:43:17.602 [main] DEBUG n.trace.WorkflowStatsObserver - Workflow completed > WorkflowStats[succeededCount=0; failedCount=0; ignoredCount=0; cachedCount=0; pendingCount=0; submittedCount=0; runningCount=0; retriesCount=0; abortedCount=0; succeedDuration=0ms; failedDuration=0ms; cachedDuration=0ms;loadCpus=0; loadMemory=0; peakRunning=0; peakCpus=0; peakMemory=0; ]\n",
      "Jan-31 12:43:17.820 [Actor Thread 2] DEBUG nextflow.processor.TaskProcessor - Process run_chromap_map_to_idx > collision check staging file names: [BMMC_single_donor_ATAC_L001_R2_corrected_trimmed.fastq.gz:1, BMMC_single_donor_ATAC_L002_R1_corrected_barcode.fastq.gz:1, ref.index:1, na.na:2, BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.gz:1, BMMC_single_donor_ATAC_L002_R2_corrected_trimmed.fastq.gz:1, BMMC_single_donor_ATAC_L001_R1_corrected_barcode.fastq.gz:1, BMMC_single_donor_ATAC_L002_R1_corrected_trimmed.fastq.gz:1, sai_192_whitelist.txt:1, GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz:1]\n",
      "Jan-31 12:43:17.821 [Actor Thread 2] DEBUG nextflow.processor.TaskProcessor - Handling unexpected condition for\n",
      "  task: name=run_chromap_map_to_idx (1); work-dir=null\n",
      "  error [nextflow.exception.ProcessUnrecoverableException]: Process `run_chromap_map_to_idx` input file name collision -- There are multiple input files for each of the following file names: na.na\n",
      "Jan-31 12:43:18.663 [main] DEBUG nextflow.cache.CacheDB - Closing CacheDB done\n",
      "Jan-31 12:43:18.663 [main] INFO  org.pf4j.AbstractPluginManager - Stop plugin 'nf-wave@1.0.1'\n",
      "Jan-31 12:43:18.663 [main] DEBUG nextflow.plugin.BasePlugin - Plugin stopped nf-wave\n",
      "Jan-31 12:43:18.663 [main] INFO  org.pf4j.AbstractPluginManager - Stop plugin 'nf-tower@1.6.3'\n",
      "Jan-31 12:43:18.663 [main] DEBUG nextflow.plugin.BasePlugin - Plugin stopped nf-tower\n",
      "Jan-31 12:43:18.663 [main] INFO  org.pf4j.AbstractPluginManager - Stop plugin 'nf-google@1.8.3'\n",
      "Jan-31 12:43:18.663 [main] DEBUG nextflow.plugin.BasePlugin - Plugin stopped nf-google\n",
      "Jan-31 12:43:18.699 [main] DEBUG nextflow.script.ScriptRunner - > Execution complete -- Goodbye\n"
     ]
    }
   ],
   "source": [
    "!cat .nextflow.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !ls ../../nf_data/test_atac_pipeline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !gsutil cp gs://eila-nextflow-bucket/test_atac_pipeline/sai_192_whitelist.txt ../../nf_data/test_atac_pipeline/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
