{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 391,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# start docker deamon in terminal\n",
    "# !sudo docker login\n",
    "# OR - becuase the jupyther with allow root\n",
    "# !docker login"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "TODO:\n",
    "\n",
    "when running with cloud. \n",
    "if you dont need to keep the execution logs, change the cloud settings to keep it for a short while\n",
    "Check that the cache if you need to set the cache to false (especially in plot process)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 392,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 393,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/hai-gcp-genomic-beb9207a40b8.json\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "json_path = os.path.join(os.path.dirname(os.getcwd()),\"hai-gcp-genomic-beb9207a40b8.json\")\n",
    "!ls $json_path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 394,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Activated service account credentials for: [eila-batch@hai-gcp-genomic.iam.gserviceaccount.com]\n"
     ]
    }
   ],
   "source": [
    "# # in the command line. same env\n",
    "import os\n",
    "json_path = os.path.join(os.path.dirname(os.getcwd()),\"hai-gcp-genomic-beb9207a40b8.json\")\n",
    "os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = json_path\n",
    "!gcloud auth activate-service-account eila-batch@hai-gcp-genomic.iam.gserviceaccount.com --key-file=$GOOGLE_APPLICATION_CREDENTIALS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 395,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                  Credentialed Accounts\n",
      "ACTIVE  ACCOUNT\n",
      "*       eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\n",
      "        eila@stanford.edu\n",
      "\n",
      "To set the active account, run:\n",
      "    $ gcloud config set account `ACCOUNT`\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'hai-gcp-genomic'"
      ]
     },
     "execution_count": 395,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "!gcloud auth list\n",
    "project_name = !gcloud config get-value project\n",
    "project_name = project_name[0]\n",
    "project_name\n",
    "# gcloud config set account\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 396,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "os.environ[\"TOWER_ACCESS_TOKEN\"] = 'eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz'\n",
    "os.environ[\"TOWER_WORKSPACE_ID\"]='247507587970818'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 397,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz\n",
      "247507587970818\n"
     ]
    }
   ],
   "source": [
    "!echo $TOWER_ACCESS_TOKEN\n",
    "!echo $TOWER_WORKSPACE_ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 398,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 399,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/bin'"
      ]
     },
     "execution_count": 399,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# !chmod +x ./bin/scrna_qc_metrics.py\n",
    "import os\n",
    "bin_fld = os.path.join(os.getcwd(),\"bin\")\n",
    "bin_fld"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 400,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "!chmod +x {os.path.join(bin_fld, \"scrna_plot_atac_qc_metrics.R\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"run_tabix.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"plot_insert_size_hist.py\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"generate_barcode_metadata.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"create_sample_filtered_fragment_file.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"compute_tss_enrichment_bulk.py\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"barcode_rank_functions.R\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"plot_insert_size_hist.py\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"compute_tss_enrichment_snapatac2.py\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"add_subpool_to_fragments_table.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"add_subpool_to_summary_file.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"process_dictionary_conversion.sh\")}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 401,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# # TODO: For a new user: run these commands with your service account for the execution. I am using eila-batch@...\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/iam.serviceAccountUser\"\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/batch.jobsEditor\"\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/logging.viewer\"\n",
    "\n",
    "# !gcloud projects add-iam-policy-binding hai-gcp-genomic \\\n",
    "# --member=\"serviceAccount:eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\" \\"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 402,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !nextflow -self-update"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 403,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  Version: 23.10.1 build 5891\n",
      "  Created: 12-01-2024 22:01 UTC (14:01 PDT)\n",
      "  System: Mac OS X 14.3\n",
      "  Runtime: Groovy 3.0.19 on OpenJDK 64-Bit Server VM 17.0.8+7-LTS\n",
      "  Encoding: UTF-8 (UTF-8)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!nextflow info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 404,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "\n",
    "# Get the current working directory\n",
    "current_directory = os.getcwd()\n",
    "\n",
    "# Navigate two folders above\n",
    "two_folders_above = os.path.abspath(os.path.join(current_directory, '..', '..'))\n",
    "\n",
    "nf_data_base_path = os.path.join(two_folders_above,'nf_data')\n",
    "# Print the result\n",
    "print(nf_data_base_path)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 405,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data'"
      ]
     },
     "execution_count": 405,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nf_data_base_path\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 406,
   "metadata": {},
   "outputs": [],
   "source": [
    "nf_data_project=\"test_atac_pipeline\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 407,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/local_atac_fastqs_files.tsv'"
      ]
     },
     "execution_count": 407,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.path.exists(os.path.join(nf_data_base_path,nf_data_project,\"local_atac_fastqs_files.tsv\"))\n",
    "os.path.join(nf_data_base_path,nf_data_project,\"local_atac_fastqs_files.tsv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 408,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "\n",
    "# Write the configuration file\n",
    "with open(\"nextflow_atac.config\", \"w\") as config_file:\n",
    "    config_file.write(f\"\"\"\n",
    "manifest {{\n",
    "  description = 'A nexflow pipeline for fg single cell atac' \n",
    "  mainScript = 'nf_wf_fg_singlecell_atac.nf'\n",
    "}}\n",
    "\n",
    "profiles {{\n",
    "  google_batch_profile {{\n",
    "    workDir = 'gs://eila-nextflow-bucket/work/ATAC'\n",
    "    includeConfig 'conf/docker-google-batch.config'\n",
    "    docker.enabled = true\n",
    "    wave.enabled = true\n",
    "    fusion.enabled = true\n",
    "    process.scratch = false\n",
    "    executor = 'google-batch'\n",
    "    docker.envWhitelist=['ENV_SYNAPSE_TOKEN','TOWER_ACCESS_TOKEN']\n",
    "    google.project = '{project_name}'\n",
    "    google.location = 'us-central1'\n",
    "    google.batch.spot = true\n",
    "    google.batch.serviceAccountEmail='eila-batch@hai-gcp-genomic.iam.gserviceaccount.com'\n",
    "  }}\n",
    "}}\n",
    "\n",
    "// Define ENV_SYNAPSE_TOKEN as a parameter\n",
    "params.FASTQS_SPEC_CH = '{nf_data_base_path}/{nf_data_project}/local_atac_fastqs_files.tsv'\n",
    "// CHROMAP\n",
    "// 6a\n",
    "params.CHROMAP_IDX='gs://eila-nextflow-bucket/test_atac_pipeline/ref.index'\n",
    "// 6b:\n",
    "// FASTA file\n",
    "params.CHROMAP_READ_LENGTH = 2000\n",
    "params.CHROMAP_BC_ERROR_THRESHOLD = 2\n",
    "params.CHROMAP_BC_PROBABILITY_THRESHOLD = 0.9\n",
    "params.CHROMAP_READ_FORMAT = 'bc:0:-1,r1:0:-1,r2:0:-1'\n",
    "params.CHROMAP_DROP_REPETITIVE_READS = 4\n",
    "params.CHROMAP_QUALITY_THRESHOLD = 0\n",
    "\n",
    "// CHROMAP DEBUG: load output files \n",
    "params.CHROMAP_GENOME_REFERENCE_FASTA='{nf_data_base_path}/{nf_data_project}/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz'\n",
    "params.chromap_filter_fragments_tsv ='{nf_data_base_path}/{nf_data_project}/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.atac.filter.fragments.tsv'\n",
    "params.barcode_summary_csv ='{nf_data_base_path}/{nf_data_project}/BMMC_single_donor_ATAC_L001_R1_corrected_trimmed.fastq.atac.align.barcode.summary.csv'\n",
    "params.TSV_ADD_SUBPOOL_SCRIPT = 'add_subpool_to_fragments_table.sh'\n",
    "params.CSV_ADD_SUBPOOL_SCRIPT = 'add_subpool_to_summary_file.sh'\n",
    "params.POOL_DICTIONARY_CONVERT_SCRIPT = 'process_dictionary_conversion.sh'\n",
    "params.ATAC_FRAGMENTS_MIN_FRAG_CUTOFF=10\n",
    "\n",
    "// Other steps\n",
    "\n",
    "params.GENOME_GZ_GTF='http://ftp.ensembl.org/pub/release-109/gtf/homo_sapiens/Homo_sapiens.GRCh38.109.gtf.gz'\n",
    "params.ENV_SYNAPSE_TOKEN = \"nextflow secrets get SYNAPSE\".execute().text.trim()\n",
    "\n",
    "\n",
    "params.MAPQ_THRESHOLD = false\n",
    "params.BARCODE_TRANSLATE = 'path'\n",
    "params.ATAC_FRAGMENTS_CUTOFF = 4\n",
    "params.RUN_TABIX_SCRIPT='run_tabix.sh'\n",
    "params.ATAC_TSS_BASES_FLANK = 2000\n",
    "params.ATAC_TSS_COL_WITH_STRANDS_INFO = 4\n",
    "params.ATAC_TSS_SMOOTHING_WINDOW_SIZE = 20\n",
    "params.ATAC_TSS_REGION_BED_FILE='{nf_data_base_path}/{nf_data_project}/hg38.synapse_transcripts.TSS.bed'\n",
    "params.ATAC_TSS_BULK_CALCULATION_SCRIPT='compute_tss_enrichment_bulk.py'\n",
    "params.ATAC_TSS_SNAPATAC2_CALCULATION_SCRIPT='compute_tss_enrichment_snapatac2.py'\n",
    "params.ATAC_TSS_SNAPATAC2_GENES_GTF='{nf_data_base_path}/{nf_data_project}/genes.gtf'\n",
    "params.ATAC_TSS_SNAPATAC2_MIN_FRAG_CUTOFF = 10\n",
    "\n",
    "params.ATAC_INSERT_SIZE_PLOT_SCRIPT='plot_insert_size_hist.py'\n",
    "\n",
    "params.SC_ATAC_QC_PLOT_SCRIPT='scrna_plot_atac_qc_metrics.R'\n",
    "params.SC_ATAC_QC_PLOT_HELPER_SCRIPT='barcode_rank_functions.R'\n",
    "params.SC_ATAC_QC_BARCODE_METADATA_FILE='{nf_data_base_path}/{nf_data_project}/barcode_metadata.txt'\n",
    "params.SC_ATAC_QC_FRAGMENT_CUTOFF=10\n",
    "params.SC_ATAC_QC_BARCODE_OUTPUT_FILE='sc_atac_qc_barcode_output_file.png'\n",
    "    \n",
    "params.SC_ATAC_BARCODE_METADATA_SCRIPT='generate_barcode_metadata.sh'\n",
    "params.SC_ATAC_BARCODE_METADATA_FILTERED_BARCODE_STATS='{nf_data_base_path}nf_wf_fg_singlecell/filtered_barcode_stats.csv'\n",
    "params.SC_ATAC_BARCODE_METADATA_TSS_ENRICHMENT_BARCODE_STATS='{nf_data_base_path}nf_wf_fg_singlecell/tss_enrichment_barcode_stats.csv'\n",
    "\n",
    "params.SC_ATAC_GENERATE_BARCODE_RANK_PLOT_SCRIPT='plot_insert_size_hist.py'\n",
    "params.SC_ATAC_GENERATE_BARCODE_RANK_PLOT_PKR='PKR123'\n",
    "\n",
    "params.SC_JOINT_CELL_BARCODE_PLOT_SCRTIP_PY='joint_cell_plotting.py'\n",
    "params.SC_JOINT_CELL_BARCODE_PLOT_SCRTIP_R='joint_cell_plotting_density.R'\n",
    "\"\"\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 409,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !rm -r /Users/eilaarich-landkof-stanford/Documents/Code/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/work\n",
    "# !rm -r /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_atac/work\n",
    "# !gsutil rm -r gs://eila-nextflow-bucket/work/ATAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 410,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !rm .nextflow*\n",
    "\n",
    "# !ls /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/local_atac_fastqs_files.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 411,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "N E X T F L O W  ~  version 23.10.1\n",
      "Launching `nf_wf_fg_singlecell_atac.nf` [agitated_williams] DSL2 - revision: b1fb73e789\n",
      "/Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/local_atac_fastqs_files.tsv\n",
      "eyJ0eXAiOiJKV1QiLCJraWQiOiJXN05OOldMSlQ6SjVSSzpMN1RMOlQ3TDc6M1ZYNjpKRU9VOjY0NFI6VTNJWDo1S1oyOjdaQ0s6RlBUSCIsImFsZyI6IlJTMjU2In0.eyJhY2Nlc3MiOnsic2NvcGUiOlsidmlldyIsImRvd25sb2FkIl0sIm9pZGNfY2xhaW1zIjp7fX0sInRva2VuX3R5cGUiOiJQRVJTT05BTF9BQ0NFU1NfVE9LRU4iLCJpc3MiOiJodHRwczovL3JlcG8tcHJvZC5wcm9kLnNhZ2ViYXNlLm9yZy9hdXRoL3YxIiwiYXVkIjoiMCIsIm5iZiI6MTY5NzgxODkxNiwiaWF0IjoxNjk3ODE4OTE2LCJqdGkiOiI0MTIzIiwic3ViIjoiMzQ3NDU3MSJ9.n0cugYMAVzYZ8mMMc7CjcqqEV7FB-K-oKUDT45SvM6KmbCpb_v3lZ2gg3xXi1tubWH0__wpnbJ7cYrGEtOyt4vXAPkMiEtZFGqS0MFKp_h-mHHH4pnACSQQYEy1zdyqFn9WRE7ipjWg4yYSbq84N7J3gzi8VKmiOAmuIGKaJaS0snkUjtJEVQH-0vZqWs1nKgVRWIZv31Bo8vuDzWpBrXD-zRT_uQW9l_I9fU50o1AqyFhNe_gbty_Z6imb8VT33wMXiDgJh15lG_M70PXU15v3egDXLgw1dGj32kj1nKanfqBVqbMSzgzPAG037p3YH_Q556BLceFBBb5Jaej5JVA\n",
      "Access to 'run_gen_values_from_channel.out' is undefined since the process 'run_gen_values_from_channel' has not been invoked before accessing the output attribute\n",
      "\n",
      " -- Check script 'nf_wf_fg_singlecell_atac.nf' at line: 40 or see '.nextflow.log' file for more details\n"
     ]
    }
   ],
   "source": [
    "!nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_atac.config -profile google_batch_profile -process.echo -with-tower -resume\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_vm.config -profile google_batch_profile\n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_vm.config -profile google_batch_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_vm_anvil.config -profile google_batch_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_local.config -profile google_batch_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_local.config -profile docker_profile\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 412,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Feb-01 20:02:25.930 [main] DEBUG nextflow.cli.Launcher - $> nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_atac.config -profile google_batch_profile -process.echo -with-tower -resume\n",
      "Feb-01 20:02:25.992 [main] INFO  nextflow.cli.CmdRun - N E X T F L O W  ~  version 23.10.1\n",
      "Feb-01 20:02:26.005 [main] DEBUG nextflow.plugin.PluginsFacade - Setting up plugin manager > mode=prod; embedded=false; plugins-dir=/Users/eilaarich-landkof-stanford/.nextflow/plugins; core-plugins: nf-amazon@2.1.4,nf-azure@1.3.3,nf-cloudcache@0.3.0,nf-codecommit@0.1.5,nf-console@1.0.6,nf-ga4gh@1.1.0,nf-google@1.8.3,nf-tower@1.6.3,nf-wave@1.0.1\n",
      "Feb-01 20:02:26.009 [main] INFO  o.pf4j.DefaultPluginStatusProvider - Enabled plugins: []\n",
      "Feb-01 20:02:26.010 [main] INFO  o.pf4j.DefaultPluginStatusProvider - Disabled plugins: []\n",
      "Feb-01 20:02:26.011 [main] INFO  org.pf4j.DefaultPluginManager - PF4J version 3.4.1 in 'deployment' mode\n",
      "Feb-01 20:02:26.015 [main] INFO  org.pf4j.AbstractPluginManager - No plugins\n",
      "Feb-01 20:02:26.023 [main] DEBUG nextflow.config.ConfigBuilder - User config file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Feb-01 20:02:26.023 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Feb-01 20:02:26.034 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `google_batch_profile`\n",
      "Feb-01 20:02:27.080 [main] DEBUG nextflow.config.ConfigBuilder - Available config profiles: [google_batch_profile]\n",
      "Feb-01 20:02:27.096 [main] DEBUG nextflow.cli.CmdRun - Applied DSL=2 by global default\n",
      "Feb-01 20:02:27.104 [main] INFO  nextflow.cli.CmdRun - Launching `nf_wf_fg_singlecell_atac.nf` [agitated_williams] DSL2 - revision: b1fb73e789\n",
      "Feb-01 20:02:27.105 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins default=[nf-google@1.8.3]\n",
      "Feb-01 20:02:27.105 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins resolved requirement=[nf-google@1.8.3, nf-tower@1.6.3, nf-wave@1.0.1]\n",
      "Feb-01 20:02:27.105 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-google version: 1.8.3\n",
      "Feb-01 20:02:27.113 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-google@1.8.3' resolved\n",
      "Feb-01 20:02:27.114 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-google@1.8.3'\n",
      "Feb-01 20:02:27.133 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-google@1.8.3\n",
      "Feb-01 20:02:27.133 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-tower version: 1.6.3\n",
      "Feb-01 20:02:27.135 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-tower@1.6.3' resolved\n",
      "Feb-01 20:02:27.135 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-tower@1.6.3'\n",
      "Feb-01 20:02:27.138 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-tower@1.6.3\n",
      "Feb-01 20:02:27.138 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-wave version: 1.0.1\n",
      "Feb-01 20:02:27.140 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-wave@1.0.1' resolved\n",
      "Feb-01 20:02:27.140 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-wave@1.0.1'\n",
      "Feb-01 20:02:27.143 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-wave@1.0.1\n",
      "Feb-01 20:02:27.149 [main] DEBUG n.secret.LocalSecretsProvider - Secrets store: /Users/eilaarich-landkof-stanford/.nextflow/secrets/store.json\n",
      "Feb-01 20:02:27.151 [main] DEBUG nextflow.secret.SecretsLoader - Discovered secrets providers: [nextflow.secret.LocalSecretsProvider@68a78f3c] - activable => nextflow.secret.LocalSecretsProvider@68a78f3c\n",
      "Feb-01 20:02:27.187 [main] DEBUG nextflow.Session - Session UUID: 0cbd4d64-b857-4807-b564-11778fa2d750\n",
      "Feb-01 20:02:27.187 [main] DEBUG nextflow.Session - Run name: agitated_williams\n",
      "Feb-01 20:02:27.187 [main] DEBUG nextflow.Session - Executor pool size: 8\n",
      "Feb-01 20:02:27.254 [main] DEBUG nextflow.file.FilePorter - File porter settings maxRetries=3; maxTransfers=50; pollTimeout=null\n",
      "Feb-01 20:02:27.255 [main] DEBUG nextflow.util.ThreadPoolBuilder - Creating thread pool 'FileTransfer' minSize=10; maxSize=24; workQueue=LinkedBlockingQueue[10000]; allowCoreThreadTimeout=false\n",
      "Feb-01 20:02:27.257 [main] DEBUG nextflow.config.ConfigBuilder - User config file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Feb-01 20:02:27.257 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nextflow_atac.config\n",
      "Feb-01 20:02:27.257 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `google_batch_profile`\n",
      "Feb-01 20:02:27.926 [main] DEBUG nextflow.config.ConfigBuilder - Available config profiles: [google_batch_profile]\n",
      "Feb-01 20:02:28.079 [main] DEBUG nextflow.cli.CmdRun - \n",
      "  Version: 23.10.1 build 5891\n",
      "  Created: 12-01-2024 22:01 UTC (14:01 PDT)\n",
      "  System: Mac OS X 14.3\n",
      "  Runtime: Groovy 3.0.19 on OpenJDK 64-Bit Server VM 17.0.8+7-LTS\n",
      "  Encoding: UTF-8 (UTF-8)\n",
      "  Process: 67918@Eilas-MacBook-Air.local [10.0.0.212]\n",
      "  CPUs: 8 - Mem: 16 GB (364.1 MB) - Swap: 3 GB (1.6 GB)\n",
      "Feb-01 20:02:28.104 [main] DEBUG nextflow.Session - Work-dir: gs://eila-nextflow-bucket/work/ATAC [Mac OS X]\n",
      "Feb-01 20:02:28.115 [main] DEBUG nextflow.executor.ExecutorFactory - Extension executors providers=[GoogleLifeSciencesExecutor, GoogleBatchExecutor]\n",
      "Feb-01 20:02:28.120 [main] DEBUG nextflow.Session - Observer factory: DefaultObserverFactory\n",
      "Feb-01 20:02:28.121 [main] DEBUG nextflow.Session - Observer factory: WaveFactory\n",
      "Feb-01 20:02:28.121 [main] DEBUG io.seqera.wave.plugin.WaveFactory - Detected Fusion enabled -- Enabling bundle project resources -- Disabling upload of remote bin directory\n",
      "Feb-01 20:02:28.127 [main] DEBUG i.s.wave.plugin.config.WaveConfig - Wave strategy not specified - using default: [container, dockerfile, conda, spack]\n",
      "Feb-01 20:02:28.131 [main] DEBUG io.seqera.wave.plugin.WaveClient - Wave config: WaveConfig(enabled:true, endpoint:https://wave.seqera.io, containerConfigUrl:[], tokensCacheMaxDuration:30m, condaOpts:CondaOpts(mambaImage=mambaorg/micromamba:1.5.1; basePackages=conda-forge::procps-ng, commands=null), spackOpts:SpackOpts(basePackages=null, commands=null), strategy:[container, dockerfile, conda, spack], bundleProjectResources:true, buildRepository:null, cacheRepository:null, retryOpts:RetryOpts(delay:450ms, maxDelay:1m 30s, maxAttempts:10, jitter:0.25), httpClientOpts:HttpOpts(), freezeMode:null)\n",
      "Feb-01 20:02:28.185 [main] DEBUG nextflow.Session - Observer factory: TowerFactory\n",
      "Feb-01 20:02:28.208 [main] DEBUG nextflow.cache.CacheFactory - Using Nextflow cache factory: nextflow.cache.DefaultCacheFactory\n",
      "Feb-01 20:02:28.211 [main] DEBUG nextflow.util.CustomThreadPool - Creating default thread pool > poolSize: 9; maxThreads: 1000\n",
      "Feb-01 20:02:28.256 [main] DEBUG nextflow.Session - Session start\n",
      "Feb-01 20:02:28.261 [main] DEBUG io.seqera.tower.plugin.TowerClient - Creating Tower observer -- endpoint=https://api.tower.nf; requestInterval=1s; aliveInterval=1m; maxRetries=5; backOffBase=3; backOffDelay=250\n",
      "Feb-01 20:02:30.580 [main] DEBUG nextflow.script.ScriptRunner - > Launching execution\n",
      "Feb-01 20:02:31.071 [main] INFO  nextflow.script.BaseScript - /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_data/test_atac_pipeline/local_atac_fastqs_files.tsv\n",
      "Feb-01 20:02:31.071 [main] INFO  nextflow.script.BaseScript - eyJ0eXAiOiJKV1QiLCJraWQiOiJXN05OOldMSlQ6SjVSSzpMN1RMOlQ3TDc6M1ZYNjpKRU9VOjY0NFI6VTNJWDo1S1oyOjdaQ0s6RlBUSCIsImFsZyI6IlJTMjU2In0.eyJhY2Nlc3MiOnsic2NvcGUiOlsidmlldyIsImRvd25sb2FkIl0sIm9pZGNfY2xhaW1zIjp7fX0sInRva2VuX3R5cGUiOiJQRVJTT05BTF9BQ0NFU1NfVE9LRU4iLCJpc3MiOiJodHRwczovL3JlcG8tcHJvZC5wcm9kLnNhZ2ViYXNlLm9yZy9hdXRoL3YxIiwiYXVkIjoiMCIsIm5iZiI6MTY5NzgxODkxNiwiaWF0IjoxNjk3ODE4OTE2LCJqdGkiOiI0MTIzIiwic3ViIjoiMzQ3NDU3MSJ9.n0cugYMAVzYZ8mMMc7CjcqqEV7FB-K-oKUDT45SvM6KmbCpb_v3lZ2gg3xXi1tubWH0__wpnbJ7cYrGEtOyt4vXAPkMiEtZFGqS0MFKp_h-mHHH4pnACSQQYEy1zdyqFn9WRE7ipjWg4yYSbq84N7J3gzi8VKmiOAmuIGKaJaS0snkUjtJEVQH-0vZqWs1nKgVRWIZv31Bo8vuDzWpBrXD-zRT_uQW9l_I9fU50o1AqyFhNe_gbty_Z6imb8VT33wMXiDgJh15lG_M70PXU15v3egDXLgw1dGj32kj1nKanfqBVqbMSzgzPAG037p3YH_Q556BLceFBBb5Jaej5JVA\n",
      "Feb-01 20:02:31.103 [main] DEBUG nextflow.script.ScriptRunner - Parsed script files:\n",
      "  Script_87d8d9fb99b69787: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_atac_barcode_rank_plot.nf\n",
      "  Script_f42273824ffc07cb: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_bgzip.nf\n",
      "  Script_85e7960f7cd2325a: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_tss.nf\n",
      "  Script_cc895528cea919f6: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_zcat.nf\n",
      "  Script_ebde5475a5a0230e: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_get_values.nf\n",
      "  Script_9d4639c8cd883c70: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_download_url_files.nf\n",
      "  Script_9c2ce54dcf0b5c06: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_merge_logs.nf\n",
      "  Script_63fd7a5b62c959a8: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_gzip.nf\n",
      "  Script_bc539eb5853c51b8: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_gunzip.nf\n",
      "  Script_96da3bf7860cf6e0: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_chromap_utils.nf\n",
      "  Script_d97ebc7c7e3deab4: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/nf_wf_fg_singlecell_atac.nf\n",
      "  Script_ea7cdb417e42517f: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_generate_barcode_metadata.nf\n",
      "  Script_280f31ae5ae5ed4d: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_tabix.nf\n",
      "  Script_033b6bf7eb6f27b3: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_barcode_subpool.nf\n",
      "  Script_e112646bcdb899f0: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_seqspec_utils.nf\n",
      "  Script_b8ed272d33e7871d: /Users/eilaarich-landkof-stanford/Documents/Code/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_atac/./../../nf_processes/nf_prcs_atac_generate_metadata_annoation.nf\n",
      "Feb-01 20:02:31.103 [main] DEBUG nextflow.Session - Session aborted -- Cause: Access to 'run_gen_values_from_channel.out' is undefined since the process 'run_gen_values_from_channel' has not been invoked before accessing the output attribute\n",
      "Feb-01 20:02:31.108 [main] DEBUG nextflow.Session - The following nodes are still active:\n",
      "  [operator] splitCsv\n",
      "  [operator] map\n",
      "\n",
      "Feb-01 20:02:31.111 [main] ERROR nextflow.cli.Launcher - Access to 'run_gen_values_from_channel.out' is undefined since the process 'run_gen_values_from_channel' has not been invoked before accessing the output attribute\n",
      "nextflow.exception.ScriptRuntimeException: Access to 'run_gen_values_from_channel.out' is undefined since the process 'run_gen_values_from_channel' has not been invoked before accessing the output attribute\n",
      "\tat nextflow.script.ProcessDef.getOut(ProcessDef.groovy:152)\n",
      "\tat java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n",
      "\tat java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)\n",
      "\tat java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\n",
      "\tat java.base/java.lang.reflect.Method.invoke(Method.java:568)\n",
      "\tat org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:107)\n",
      "\tat groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)\n",
      "\tat org.codehaus.groovy.runtime.metaclass.MethodMetaProperty$GetBeanMethodMetaProperty.getProperty(MethodMetaProperty.java:76)\n",
      "\tat org.codehaus.groovy.runtime.callsite.GetEffectivePogoPropertySite.getProperty(GetEffectivePogoPropertySite.java:85)\n",
      "\tat org.codehaus.groovy.runtime.callsite.AbstractCallSite.callGetProperty(AbstractCallSite.java:329)\n",
      "\tat Script_d97ebc7c7e3deab4$_runScript_closure1$_closure2.doCall(Script_d97ebc7c7e3deab4:40)\n",
      "\tat java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n",
      "\tat java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)\n",
      "\tat java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\n",
      "\tat java.base/java.lang.reflect.Method.invoke(Method.java:568)\n",
      "\tat org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:107)\n",
      "\tat groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)\n",
      "\tat org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:274)\n",
      "\tat groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1030)\n",
      "\tat groovy.lang.Closure.call(Closure.java:427)\n",
      "\tat groovy.lang.Closure.call(Closure.java:406)\n",
      "\tat nextflow.script.WorkflowDef.run0(WorkflowDef.groovy:204)\n",
      "\tat nextflow.script.WorkflowDef.run(WorkflowDef.groovy:188)\n",
      "\tat nextflow.script.BindableDef.invoke_a(BindableDef.groovy:51)\n",
      "\tat nextflow.script.ChainableDef$invoke_a.call(Unknown Source)\n",
      "\tat org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)\n",
      "\tat org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)\n",
      "\tat org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:139)\n",
      "\tat nextflow.script.BaseScript.run0(BaseScript.groovy:183)\n",
      "\tat nextflow.script.BaseScript.run(BaseScript.groovy:192)\n",
      "\tat nextflow.script.ScriptParser.runScript(ScriptParser.groovy:236)\n",
      "\tat nextflow.script.ScriptRunner.run(ScriptRunner.groovy:242)\n",
      "\tat nextflow.script.ScriptRunner.execute(ScriptRunner.groovy:137)\n",
      "\tat nextflow.cli.CmdRun.run(CmdRun.groovy:372)\n",
      "\tat nextflow.cli.Launcher.run(Launcher.groovy:500)\n",
      "\tat nextflow.cli.Launcher.main(Launcher.groovy:672)\n"
     ]
    }
   ],
   "source": [
    "!cat .nextflow.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 413,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 414,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 415,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !ls ../../nf_data/test_atac_pipeline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 416,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !gsutil cp gs://eila-nextflow-bucket/test_atac_pipeline/sai_192_whitelist.txt ../../nf_data/test_atac_pipeline/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
