{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# start docker deamon in terminal\n",
    "# !sudo docker login\n",
    "# OR - becuase the jupyther with allow root\n",
    "# !docker login"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_rna\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/eila/git_igvf/atomic-workflows/nextflow/hai-gcp-genomic-beb9207a40b8.json\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "json_path = os.path.join(os.path.dirname(os.path.dirname(os.getcwd())),\"hai-gcp-genomic-beb9207a40b8.json\")\n",
    "!ls $json_path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Activated service account credentials for: [eila-batch@hai-gcp-genomic.iam.gserviceaccount.com]\n"
     ]
    }
   ],
   "source": [
    "# # in the command line. same env\n",
    "import os\n",
    "json_path = os.path.join(os.path.dirname(os.path.dirname(os.getcwd())),\"hai-gcp-genomic-beb9207a40b8.json\")\n",
    "os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = json_path\n",
    "!gcloud auth activate-service-account eila-batch@hai-gcp-genomic.iam.gserviceaccount.com --key-file=$GOOGLE_APPLICATION_CREDENTIALS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                  Credentialed Accounts\n",
      "ACTIVE  ACCOUNT\n",
      "        886987709752-compute@developer.gserviceaccount.com\n",
      "*       eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\n",
      "        eila@stanford.edu\n",
      "\n",
      "To set the active account, run:\n",
      "    $ gcloud config set account `ACCOUNT`\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'hai-gcp-genomic'"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "!gcloud auth list\n",
    "project_name = !gcloud config get-value project\n",
    "project_name = project_name[0]\n",
    "project_name\n",
    "# gcloud config set account\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "import os\n",
    "os.environ[\"TOWER_ACCESS_TOKEN\"] = 'eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz'\n",
    "os.environ[\"TOWER_WORKSPACE_ID\"]='247507587970818'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz\n"
     ]
    }
   ],
   "source": [
    "!echo $TOWER_ACCESS_TOKEN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "247507587970818\n"
     ]
    }
   ],
   "source": [
    "!echo $TOWER_WORKSPACE_ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# %%bash\n",
    "# export TOWER_ACCESS_TOKEN='eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz'\n",
    "# echo $TOWER_ACCESS_TOKEN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# %%bash\n",
    "# export TOWER_WORKSPACE_ID='247507587970818'\n",
    "# echo $TOWER_WORKSPACE_ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_rna/bin'"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# !chmod +x ./bin/scrna_qc_metrics.py\n",
    "import os\n",
    "bin_fld = os.path.join(os.getcwd(),\"bin\")\n",
    "bin_fld"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "!chmod +x {os.path.join(bin_fld, \"cellatlas_build.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"cellatlas_create_jq_commands_txt.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"rna_kb_ref_with_jq_commands.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"seqspec_retrieve_technology.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"rna_kb_count_execution.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"seqspec_run_modify.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"rna_modify_kb_count_output_barcode_h5.py\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"rna_modify_kb_count_cell_gene_barcode.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"cellatlas_jq_run_info_and_inspect.sh\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"rna_qc_metrics.py\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"rna_plot_qc_metrics.R\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"barcode_rank_functions.R\")}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  Version: 23.10.1 build 5891\n",
      "  Created: 12-01-2024 22:01 UTC \n",
      "  System: Linux 5.10.0-28-cloud-amd64\n",
      "  Runtime: Groovy 3.0.19 on OpenJDK 64-Bit Server VM 11.0.22+7-post-Debian-1deb11u1\n",
      "  Encoding: UTF-8 (UTF-8)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!nextflow info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !sudo nextflow self-update\n",
    "\n",
    "# sometimes you need to allow execution for nextflow\n",
    "# !ls -l /usr/local/bin/nextflow\n",
    "# !sudo chmod +x /usr/local/bin/nextflow\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  Version: 23.10.1 build 5891\n",
      "  Created: 12-01-2024 22:01 UTC \n",
      "  System: Linux 5.10.0-28-cloud-amd64\n",
      "  Runtime: Groovy 3.0.19 on OpenJDK 64-Bit Server VM 11.0.22+7-post-Debian-1deb11u1\n",
      "  Encoding: UTF-8 (UTF-8)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!sudo nextflow info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/eila/git_igvf/atomic-workflows/nextflow/nf_data\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "\n",
    "# Get the current working directory\n",
    "current_directory = os.getcwd()\n",
    "\n",
    "# Navigate two folders above\n",
    "two_folders_above = os.path.abspath(os.path.join(current_directory, '..', '..'))\n",
    "\n",
    "nf_data_base_path = os.path.join(two_folders_above,'nf_data')\n",
    "# Print the result\n",
    "print(nf_data_base_path)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# nf_data_project=\"test_atac_pipeline\"\n",
    "# nf_data_project=\"test_rna_pipeline\"\n",
    "nf_data_project=\"wdl_rna_pipeline\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/eila/git_igvf/atomic-workflows/nextflow/nf_data/wdl_rna_pipeline/vm_rna_input_files.tsv'"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# VM\n",
    "os.path.exists(os.path.join(nf_data_base_path,nf_data_project,\"vm_atac_fastqs_files.tsv\"))\n",
    "os.path.join(nf_data_base_path,nf_data_project,\"vm_rna_input_files.tsv\")\n",
    "# /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/wdl_rna_pipeline/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !ls /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/wdl_rna_pipeline/vm_rna_input_files.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "!cp ../../config_files/docker-google-batch.config ../workflow_fg_single_cell_rna/conf/\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Write the configuration file\n",
    "with open(\"nextflow_rna.config\", \"w\") as config_file:\n",
    "    config_file.write(f\"\"\"\n",
    "manifest {{\n",
    "  description = 'A nextflow pipeline for fg single cell' \n",
    "  mainScript = 'nf_wf_fg_singlecell_rna.nf'\n",
    "}}\n",
    "\n",
    "profiles {{\n",
    "  google_batch_profile {{\n",
    "    workDir = 'gs://eila-nextflow-bucket/work/RNA'\n",
    "    includeConfig 'conf/docker-google-batch.config'\n",
    "    docker.enabled = true\n",
    "    wave.enabled = true\n",
    "    fusion.enabled = true\n",
    "    process.scratch = false\n",
    "    executor = 'google-batch'\n",
    "    docker.envWhitelist=['ENV_SYNAPSE_TOKEN','TOWER_ACCESS_TOKEN']\n",
    "    google.project = '{project_name}'\n",
    "    google.location = 'us-central1'\n",
    "    google.batch.spot = true\n",
    "    google.batch.serviceAccountEmail='eila-batch@hai-gcp-genomic.iam.gserviceaccount.com'\n",
    "  }}\n",
    "}}\n",
    "\n",
    "// Define ENV_SYNAPSE_TOKEN as a parameter\n",
    "params.ORGANISM = 'human'\n",
    "params.FASTQS_SPEC_CH = '{nf_data_base_path}/{nf_data_project}/vm_rna_input_files.tsv'\n",
    "params.GENOME_FASTA_GZ = '{nf_data_base_path}/{nf_data_project}/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz'\n",
    "params.GENES_GTF = '{nf_data_base_path}/{nf_data_project}/genes.gtf'\n",
    "params.CELLATLAS_BUILD_SCRIPT = 'cellatlas_build.sh'\n",
    "params.JQ_CELLATLAS_BUILD_JSON_SCRIPT = 'cellatlas_create_jq_commands_txt.sh'\n",
    "params.KB_REF_WITH_JQ_COMMANDS='rna_kb_ref_with_jq_commands.sh'\n",
    "params.TRANSCRIPTOME_FA='{nf_data_base_path}/{nf_data_project}/transcriptome.fa'\n",
    "\n",
    "params.EXECUTE_SEQSPEC_MODIFY='seqspec_run_modify.sh'\n",
    "params.RETRIEVE_SEQSPEC_TECHNOLOGY='seqspec_retrieve_technology.sh'\n",
    "params.EXECUTE_KB_COUNT_RNA_SCRIPT='rna_kb_count_execution.sh'\n",
    "params.KB_COUNT_CPUS=16\n",
    "//Debug until I have a complete seqspec\n",
    "params.KB_COUNT_TECHNOLGY_FILE_DEBUG='{nf_data_base_path}/{nf_data_project}/technology_file.txt'\n",
    "\n",
    "// STEP 7\n",
    "\n",
    "params.RNA_MODIFY_KB_COUNT_H5AD_WITH_SUBPOOL='rna_modify_kb_count_output_barcode_h5.py'\n",
    "params.RNA_MODIFY_CELL_GENE_WITH_SUBPOOL='rna_modify_kb_count_cell_gene_barcode.sh'\n",
    "// DEBUG\n",
    "params.adata_out_h5ad_debug='{nf_data_base_path}/{nf_data_project}/out/counts_unfiltered/adata.h5ad'\n",
    "params.kb_count_run_info_json_debug='{nf_data_base_path}/{nf_data_project}/out/run_info.json'\n",
    "params.kb_count_inspect_json_debug='{nf_data_base_path}/{nf_data_project}/out/inspect.json'\n",
    "params.cells_x_genes_barcodes_out_debug='{nf_data_base_path}/{nf_data_project}/out/counts_unfiltered/cells_x_genes.barcodes.txt'\n",
    "// STEP 8\n",
    "params.RNA_JQ_CELLATLAS_OUTPUT_TO_LOG='cellatlas_jq_run_info_and_inspect.sh'\n",
    "params.LOG_FILE_NAME='BMMC-single-donor.rna.log.hg38.txt'\n",
    "\n",
    "// STEP 9: QC METRICS\n",
    "params.RNA_QC__METRICS_SCRIPT = 'rna_qc_metrics.py'\n",
    "\n",
    "// STEP 10: RNA QC PLOTS\n",
    "params.RNA_QC_PLOT_SCRIPT = 'rna_plot_qc_metrics.R'\n",
    "params.RNA_QC_PLOT_HELPER_SCRIPT = 'barcode_rank_functions.R'\n",
    "params.UMI_CUTOFF = 2\n",
    "params.GENE_CUTOFF = 3\n",
    "\n",
    "\n",
    "// params.GENOME_FASTA = 'http://ftp.ensembl.org/pub/release-109/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz'\n",
    "// params.GENOME_GZ_GTF = 'http://ftp.ensembl.org/pub/release-109/gtf/homo_sapiens/Homo_sapiens.GRCh38.109.gtf.gz'\n",
    "// params.ENV_SYNAPSE_TOKEN = \"nextflow secrets get SYNAPSE\".execute().text.trim()\n",
    "// params.SYNAPSE_CONFIG = '{nf_data_base_path}/{nf_data_project}nf_wf_fg_singlecell/synapse.config'\n",
    "// params.SYNAPSE_IDS = '{nf_data_base_path}/{nf_data_project}/nf_wf_fg_singlecell/synapse_ids.csv'\n",
    "\n",
    "// params.RNA_DEBUG_QC = '{nf_data_base_path}/{nf_data_project}/nf_workflow/workflow_fg_single_cell_rna/sc_rna_qc_calculate.tsv'\n",
    "\"\"\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !rm -r /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/wdl_rna_pipeline/out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !rm -r /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/wdl_rna_pipeline/out/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !mv /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/test_rna_pipeline/out /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/wdl_rna_pipeline/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !ls /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/wdl_rna_pipeline\n",
    "# file:///home/eila/git_igvf/atomic-workflows/nextflow/nf_data/wdl_rna_pipeline/out/counts_unfiltered/adata.h5ad "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz\n"
     ]
    }
   ],
   "source": [
    "# !rm -rf /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_rna/work\n",
    "# !gsutil -m rm -r gs://eila-nextflow-bucket/work/RNA\n",
    "!echo $TOWER_ACCESS_TOKEN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "N E X T F L O W  ~  version 23.10.1\n",
      "Launching `nf_wf_fg_singlecell_rna.nf` [astonishing_mandelbrot] DSL2 - revision: fefe02afab\n",
      "in the RNA workflow\n",
      "[-        ] process > run_cellatlas_build            -\u001b[K\n",
      "[-        ] process > run_jq_on_cellatlas            -\u001b[K\n",
      "[-        ] process > run_kb_ref_with_jq_commands    -\u001b[K\n",
      "[-        ] process > run_seqspec_modify_rna         -\u001b[K\n",
      "[-        ] process > run_add_subpool_to_rna_kb_c... -\u001b[K\n",
      "[-        ] process > run_jq_on_kb_count_outputs_... -\u001b[K\n",
      "[-        ] process > rna_calculate_qc_metrics       -\u001b[K\n",
      "after run_seqspec_modify_rna\u001b[K\n",
      "Calling run_add_subpool_to_rna_kb_cout_outputs with the following inputs:\u001b[K\n",
      "params.RNA_MODIFY_KB_COUNT_H5AD_WITH_SUBPOOL: rna_modify_kb_count_output_barcode_h5.py\u001b[K\n",
      "params.RNA_MODIFY_CELL_GENE_WITH_SUBPOOL: rna_modify_kb_count_cell_gene_barcode.sh\u001b[K\n",
      "adata_out_h5ad: DataflowVariable(value=null)\u001b[K\n",
      "cells_x_genes_barcodes_out: DataflowVariable(value=null)\u001b[K\n",
      "Calling run_jq_on_kb_count_outputs_to_logs with the following inputs:\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: params.RNA_JQ_CELLATLAS_OUTPUT_TO_LOG: cellatlas_jq_run_info_and_inspect.sh\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: kb_count_run_info_json: DataflowVariable(value=null)\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: kb_count_inspect_json: DataflowVariable(value=null)\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: params.LOG_FILE_NAME: BMMC-single-donor.rna.log.hg38.txt\u001b[K\n",
      "\u001b[19A\n",
      "[-        ] process > run_cellatlas_build            -\u001b[K\n",
      "[-        ] process > run_jq_on_cellatlas            -\u001b[K\n",
      "[-        ] process > run_kb_ref_with_jq_commands    -\u001b[K\n",
      "[-        ] process > run_seqspec_modify_rna         -\u001b[K\n",
      "[-        ] process > run_add_subpool_to_rna_kb_c... -\u001b[K\n",
      "[-        ] process > run_jq_on_kb_count_outputs_... -\u001b[K\n",
      "[-        ] process > rna_calculate_qc_metrics       -\u001b[K\n",
      "[-        ] process > rna_plot_qc_metrics_prcs       -\u001b[K\n",
      "after run_seqspec_modify_rna\u001b[K\n",
      "Calling run_add_subpool_to_rna_kb_cout_outputs with the following inputs:\u001b[K\n",
      "params.RNA_MODIFY_KB_COUNT_H5AD_WITH_SUBPOOL: rna_modify_kb_count_output_barcode_h5.py\u001b[K\n",
      "params.RNA_MODIFY_CELL_GENE_WITH_SUBPOOL: rna_modify_kb_count_cell_gene_barcode.sh\u001b[K\n",
      "adata_out_h5ad: DataflowVariable(value=null)\u001b[K\n",
      "cells_x_genes_barcodes_out: DataflowVariable(value=null)\u001b[K\n",
      "Calling run_jq_on_kb_count_outputs_to_logs with the following inputs:\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: params.RNA_JQ_CELLATLAS_OUTPUT_TO_LOG: cellatlas_jq_run_info_and_inspect.sh\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: kb_count_run_info_json: DataflowVariable(value=null)\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: kb_count_inspect_json: DataflowVariable(value=null)\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: params.LOG_FILE_NAME: BMMC-single-donor.rna.log.hg38.txt\u001b[K\n",
      "\u001b[20A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2hMUZ1brXDrASM\u001b[22;39m\u001b[K\n",
      "[-        ] process > run_cellatlas_build            -\u001b[K\n",
      "[-        ] process > run_jq_on_cellatlas            -\u001b[K\n",
      "[-        ] process > run_kb_ref_with_jq_commands    -\u001b[K\n",
      "[-        ] process > run_seqspec_modify_rna         -\u001b[K\n",
      "[-        ] process > run_add_subpool_to_rna_kb_c... -\u001b[K\n",
      "[-        ] process > run_jq_on_kb_count_outputs_... -\u001b[K\n",
      "[-        ] process > rna_calculate_qc_metrics       -\u001b[K\n",
      "[-        ] process > rna_plot_qc_metrics_prcs       -\u001b[K\n",
      "after run_seqspec_modify_rna\u001b[K\n",
      "Calling run_add_subpool_to_rna_kb_cout_outputs with the following inputs:\u001b[K\n",
      "params.RNA_MODIFY_KB_COUNT_H5AD_WITH_SUBPOOL: rna_modify_kb_count_output_barcode_h5.py\u001b[K\n",
      "params.RNA_MODIFY_CELL_GENE_WITH_SUBPOOL: rna_modify_kb_count_cell_gene_barcode.sh\u001b[K\n",
      "adata_out_h5ad: DataflowVariable(value=null)\u001b[K\n",
      "cells_x_genes_barcodes_out: DataflowVariable(value=null)\u001b[K\n",
      "Calling run_jq_on_kb_count_outputs_to_logs with the following inputs:\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: params.RNA_JQ_CELLATLAS_OUTPUT_TO_LOG: cellatlas_jq_run_info_and_inspect.sh\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: kb_count_run_info_json: DataflowVariable(value=null)\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: kb_count_inspect_json: DataflowVariable(value=null)\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: params.LOG_FILE_NAME: BMMC-single-donor.rna.log.hg38.txt\u001b[K\n",
      "\u001b[21A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2hMUZ1brXDrASM\u001b[22;39m\u001b[K\n",
      "[-        ] process > run_cellatlas_build            -\u001b[K\n",
      "[-        ] process > run_jq_on_cellatlas            -\u001b[K\n",
      "[-        ] process > run_kb_ref_with_jq_commands    -\u001b[K\n",
      "[f4/6d209b] process > run_seqspec_modify_rna (1)     [100%] 1 of 1, cached: 1\u001b[K\n",
      "[-        ] process > run_add_subpool_to_rna_kb_c... -\u001b[K\n",
      "[8f/cb996e] process > run_jq_on_kb_count_outputs_... [100%] 1 of 1, cached: 1\u001b[K\n",
      "[-        ] process > rna_calculate_qc_metrics       -\u001b[K\n",
      "[-        ] process > rna_plot_qc_metrics_prcs       -\u001b[K\n",
      "after run_seqspec_modify_rna\u001b[K\n",
      "Calling run_add_subpool_to_rna_kb_cout_outputs with the following inputs:\u001b[K\n",
      "params.RNA_MODIFY_KB_COUNT_H5AD_WITH_SUBPOOL: rna_modify_kb_count_output_barcode_h5.py\u001b[K\n",
      "params.RNA_MODIFY_CELL_GENE_WITH_SUBPOOL: rna_modify_kb_count_cell_gene_barcode.sh\u001b[K\n",
      "adata_out_h5ad: DataflowVariable(value=null)\u001b[K\n",
      "cells_x_genes_barcodes_out: DataflowVariable(value=null)\u001b[K\n",
      "Calling run_jq_on_kb_count_outputs_to_logs with the following inputs:\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: params.RNA_JQ_CELLATLAS_OUTPUT_TO_LOG: cellatlas_jq_run_info_and_inspect.sh\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: kb_count_run_info_json: DataflowVariable(value=null)\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: kb_count_inspect_json: DataflowVariable(value=null)\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: params.LOG_FILE_NAME: BMMC-single-donor.rna.log.hg38.txt\u001b[K\n",
      "\u001b[21A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2hMUZ1brXDrASM\u001b[22;39m\u001b[K\n",
      "[-        ] process > run_cellatlas_build            -\u001b[K\n",
      "[-        ] process > run_jq_on_cellatlas            -\u001b[K\n",
      "[-        ] process > run_kb_ref_with_jq_commands    -\u001b[K\n",
      "[f4/6d209b] process > run_seqspec_modify_rna (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[f6/c0345a] process > run_add_subpool_to_rna_kb_c... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[8f/cb996e] process > run_jq_on_kb_count_outputs_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > rna_calculate_qc_metrics       -\u001b[K\n",
      "[-        ] process > rna_plot_qc_metrics_prcs       -\u001b[K\n",
      "after run_seqspec_modify_rna\u001b[K\n",
      "Calling run_add_subpool_to_rna_kb_cout_outputs with the following inputs:\u001b[K\n",
      "params.RNA_MODIFY_KB_COUNT_H5AD_WITH_SUBPOOL: rna_modify_kb_count_output_barcode_h5.py\u001b[K\n",
      "params.RNA_MODIFY_CELL_GENE_WITH_SUBPOOL: rna_modify_kb_count_cell_gene_barcode.sh\u001b[K\n",
      "adata_out_h5ad: DataflowVariable(value=null)\u001b[K\n",
      "cells_x_genes_barcodes_out: DataflowVariable(value=null)\u001b[K\n",
      "Calling run_jq_on_kb_count_outputs_to_logs with the following inputs:\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: params.RNA_JQ_CELLATLAS_OUTPUT_TO_LOG: cellatlas_jq_run_info_and_inspect.sh\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: kb_count_run_info_json: DataflowVariable(value=null)\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: kb_count_inspect_json: DataflowVariable(value=null)\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: params.LOG_FILE_NAME: BMMC-single-donor.rna.log.hg38.txt\u001b[K\n",
      "=== Start run_seqspec_modify_rna ===\u001b[K\n",
      " Input Fastq Files:\u001b[K\n",
      " R1: BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\u001b[K\n",
      " R2: BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\u001b[K\n",
      " Specification YAML: bmmc_spec_igvf_rna.yaml\u001b[K\n",
      " Whitelist File: sai_192_whitelist.txt\u001b[K\n",
      " seqspec_rna_region_id is: rna\u001b[K\n",
      "Running command: seqspec modify -m rna -o modrna1.yaml -r \"rna_R1.fastq.gz\" --region-id \"BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\" bmmc_spec_igvf_rna.yaml\u001b[K\n",
      "Running command: seqspec modify -m rna -o modrna2.yaml -r \"rna_R2.fastq.gz\" --region-id \"BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\" modrna1.yaml\u001b[K\n",
      "Running command: seqspec modify -m rna -o \"seqspec_modify_rna_file_names.yaml\" -r \"rna_R2.fastq.gz\" --region-id \"BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\" modrna2.yaml\u001b[K\n",
      "=== Finished seqspec modify rna ===\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_jq_on_kb_count_outputs_to_logs ------\u001b[K\n",
      "Input script_name: cellatlas_jq_run_info_and_inspect.sh\u001b[K\n",
      "Input output_file_name with json : BMMC-single-donor.rna.log.hg38.json\u001b[K\n",
      "Input run_info_json: run_info.json\u001b[K\n",
      "Input inspect_json: inspect.json\u001b[K\n",
      "------ Finished run_jq_on_kb_count_outputs_to_logs ------\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_subpool_to_rna_kb_cout_outputs ------\u001b[K\n",
      "Input subpool_script_modify_h5ad: rna_modify_kb_count_output_barcode_h5.py\u001b[K\n",
      "Input subpool_script_cell_gene: rna_modify_kb_count_cell_gene_barcode.sh\u001b[K\n",
      "Input kb_count_h5ad_file: adata.h5ad\u001b[K\n",
      "Input kb_count_fragments_file: cells_x_genes.barcodes.txt\u001b[K\n",
      "Input fastq1: BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\u001b[K\n",
      "Input fastq2: BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\u001b[K\n",
      "Input spec_yaml: bmmc_spec_igvf_rna.yaml\u001b[K\n",
      "Input whitelist_file: sai_192_whitelist.txt\u001b[K\n",
      "Input seqspec_rna_region_id: rna\u001b[K\n",
      "Input subpool: none\u001b[K\n",
      "------ Function append_subpool_to_h5ad ------\u001b[K\n",
      "Input H5AD file: adata.h5ad\u001b[K\n",
      "Subpool to append: _none\u001b[K\n",
      "Output H5AD file: adata_none.h5ad\u001b[K\n",
      "Copied adata.h5ad to adata_none.h5ad\u001b[K\n",
      "------ Finished append_subpool_to_h5ad ------\u001b[K\n",
      "Output H5AD file: adata_none.h5ad\u001b[K\n",
      "Output Fragments file: cells_x_genes.barcodes_none.txt\u001b[K\n",
      "Input filename: cells_x_genes.barcodes.txt\u001b[K\n",
      "Subpool: none\u001b[K\n",
      "Output filename with subpool: cells_x_genes.barcodes_none.txt\u001b[K\n",
      "File copied to cells_x_genes.barcodes_none.txt\u001b[K\n",
      "Subpool is 'none', skipping sed command.\u001b[K\n",
      "------ Finished run_add_subpool_to_rna_kb_cout_outputs ------\u001b[K\n",
      "\u001b[K\n",
      "\u001b[66A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2hMUZ1brXDrASM\u001b[22;39m\u001b[K\n",
      "[9b/81e0dc] process > run_cellatlas_build (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_jq_on_cellatlas            -\u001b[K\n",
      "[-        ] process > run_kb_ref_with_jq_commands    -\u001b[K\n",
      "[f4/6d209b] process > run_seqspec_modify_rna (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[f6/c0345a] process > run_add_subpool_to_rna_kb_c... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[8f/cb996e] process > run_jq_on_kb_count_outputs_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > rna_calculate_qc_metrics       -\u001b[K\n",
      "[-        ] process > rna_plot_qc_metrics_prcs       -\u001b[K\n",
      "after run_seqspec_modify_rna\u001b[K\n",
      "Calling run_add_subpool_to_rna_kb_cout_outputs with the following inputs:\u001b[K\n",
      "params.RNA_MODIFY_KB_COUNT_H5AD_WITH_SUBPOOL: rna_modify_kb_count_output_barcode_h5.py\u001b[K\n",
      "params.RNA_MODIFY_CELL_GENE_WITH_SUBPOOL: rna_modify_kb_count_cell_gene_barcode.sh\u001b[K\n",
      "adata_out_h5ad: DataflowVariable(value=null)\u001b[K\n",
      "cells_x_genes_barcodes_out: DataflowVariable(value=null)\u001b[K\n",
      "Calling run_jq_on_kb_count_outputs_to_logs with the following inputs:\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: params.RNA_JQ_CELLATLAS_OUTPUT_TO_LOG: cellatlas_jq_run_info_and_inspect.sh\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: kb_count_run_info_json: DataflowVariable(value=null)\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: kb_count_inspect_json: DataflowVariable(value=null)\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: params.LOG_FILE_NAME: BMMC-single-donor.rna.log.hg38.txt\u001b[K\n",
      "=== Start run_seqspec_modify_rna ===\u001b[K\n",
      " Input Fastq Files:\u001b[K\n",
      " R1: BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\u001b[K\n",
      " R2: BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\u001b[K\n",
      " Specification YAML: bmmc_spec_igvf_rna.yaml\u001b[K\n",
      " Whitelist File: sai_192_whitelist.txt\u001b[K\n",
      " seqspec_rna_region_id is: rna\u001b[K\n",
      "Running command: seqspec modify -m rna -o modrna1.yaml -r \"rna_R1.fastq.gz\" --region-id \"BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\" bmmc_spec_igvf_rna.yaml\u001b[K\n",
      "Running command: seqspec modify -m rna -o modrna2.yaml -r \"rna_R2.fastq.gz\" --region-id \"BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\" modrna1.yaml\u001b[K\n",
      "Running command: seqspec modify -m rna -o \"seqspec_modify_rna_file_names.yaml\" -r \"rna_R2.fastq.gz\" --region-id \"BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\" modrna2.yaml\u001b[K\n",
      "=== Finished seqspec modify rna ===\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_jq_on_kb_count_outputs_to_logs ------\u001b[K\n",
      "Input script_name: cellatlas_jq_run_info_and_inspect.sh\u001b[K\n",
      "Input output_file_name with json : BMMC-single-donor.rna.log.hg38.json\u001b[K\n",
      "Input run_info_json: run_info.json\u001b[K\n",
      "Input inspect_json: inspect.json\u001b[K\n",
      "------ Finished run_jq_on_kb_count_outputs_to_logs ------\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_subpool_to_rna_kb_cout_outputs ------\u001b[K\n",
      "Input subpool_script_modify_h5ad: rna_modify_kb_count_output_barcode_h5.py\u001b[K\n",
      "Input subpool_script_cell_gene: rna_modify_kb_count_cell_gene_barcode.sh\u001b[K\n",
      "Input kb_count_h5ad_file: adata.h5ad\u001b[K\n",
      "Input kb_count_fragments_file: cells_x_genes.barcodes.txt\u001b[K\n",
      "Input fastq1: BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\u001b[K\n",
      "Input fastq2: BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\u001b[K\n",
      "Input spec_yaml: bmmc_spec_igvf_rna.yaml\u001b[K\n",
      "Input whitelist_file: sai_192_whitelist.txt\u001b[K\n",
      "Input seqspec_rna_region_id: rna\u001b[K\n",
      "Input subpool: none\u001b[K\n",
      "------ Function append_subpool_to_h5ad ------\u001b[K\n",
      "Input H5AD file: adata.h5ad\u001b[K\n",
      "Subpool to append: _none\u001b[K\n",
      "Output H5AD file: adata_none.h5ad\u001b[K\n",
      "Copied adata.h5ad to adata_none.h5ad\u001b[K\n",
      "------ Finished append_subpool_to_h5ad ------\u001b[K\n",
      "Output H5AD file: adata_none.h5ad\u001b[K\n",
      "Output Fragments file: cells_x_genes.barcodes_none.txt\u001b[K\n",
      "Input filename: cells_x_genes.barcodes.txt\u001b[K\n",
      "Subpool: none\u001b[K\n",
      "Output filename with subpool: cells_x_genes.barcodes_none.txt\u001b[K\n",
      "File copied to cells_x_genes.barcodes_none.txt\u001b[K\n",
      "Subpool is 'none', skipping sed command.\u001b[K\n",
      "------ Finished run_add_subpool_to_rna_kb_cout_outputs ------\u001b[K\n",
      "\u001b[K\n",
      "calling cellatlas build\u001b[K\n",
      "Output Directory: cellatlas_out\u001b[K\n",
      "Mode: rna\u001b[K\n",
      "Spec File: bmmc_spec_igvf_rna.yaml\u001b[K\n",
      "Genome Fasta Gz: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Gene GTF: genes.gtf\u001b[K\n",
      "Fastq1: BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\u001b[K\n",
      "Fastq2: BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\u001b[K\n",
      "Contents of Output Directory:\u001b[K\n",
      "cellatlas_info.json\u001b[K\n",
      "BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\u001b[K\n",
      "BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\u001b[K\n",
      "GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "bmmc_spec_igvf_rna.yaml\u001b[K\n",
      "cellatlas_out\u001b[K\n",
      "genes.gtf\u001b[K\n",
      "sai_192_whitelist.txt\u001b[K\n",
      "\u001b[K\n",
      "\u001b[84A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2hMUZ1brXDrASM\u001b[22;39m\u001b[K\n",
      "[9b/81e0dc] process > run_cellatlas_build (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[78/ace689] process > run_jq_on_cellatlas (1)        [100%] 1 of 1, cached: 1\u001b[K\n",
      "[-        ] process > run_kb_ref_with_jq_commands    -\u001b[K\n",
      "[f4/6d209b] process > run_seqspec_modify_rna (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[f6/c0345a] process > run_add_subpool_to_rna_kb_c... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[8f/cb996e] process > run_jq_on_kb_count_outputs_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > rna_calculate_qc_metrics       -\u001b[K\n",
      "[-        ] process > rna_plot_qc_metrics_prcs       -\u001b[K\n",
      "after run_seqspec_modify_rna\u001b[K\n",
      "Calling run_add_subpool_to_rna_kb_cout_outputs with the following inputs:\u001b[K\n",
      "params.RNA_MODIFY_KB_COUNT_H5AD_WITH_SUBPOOL: rna_modify_kb_count_output_barcode_h5.py\u001b[K\n",
      "params.RNA_MODIFY_CELL_GENE_WITH_SUBPOOL: rna_modify_kb_count_cell_gene_barcode.sh\u001b[K\n",
      "adata_out_h5ad: DataflowVariable(value=null)\u001b[K\n",
      "cells_x_genes_barcodes_out: DataflowVariable(value=null)\u001b[K\n",
      "Calling run_jq_on_kb_count_outputs_to_logs with the following inputs:\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: params.RNA_JQ_CELLATLAS_OUTPUT_TO_LOG: cellatlas_jq_run_info_and_inspect.sh\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: kb_count_run_info_json: DataflowVariable(value=null)\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: kb_count_inspect_json: DataflowVariable(value=null)\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: params.LOG_FILE_NAME: BMMC-single-donor.rna.log.hg38.txt\u001b[K\n",
      "=== Start run_seqspec_modify_rna ===\u001b[K\n",
      " Input Fastq Files:\u001b[K\n",
      " R1: BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\u001b[K\n",
      " R2: BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\u001b[K\n",
      " Specification YAML: bmmc_spec_igvf_rna.yaml\u001b[K\n",
      " Whitelist File: sai_192_whitelist.txt\u001b[K\n",
      " seqspec_rna_region_id is: rna\u001b[K\n",
      "Running command: seqspec modify -m rna -o modrna1.yaml -r \"rna_R1.fastq.gz\" --region-id \"BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\" bmmc_spec_igvf_rna.yaml\u001b[K\n",
      "Running command: seqspec modify -m rna -o modrna2.yaml -r \"rna_R2.fastq.gz\" --region-id \"BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\" modrna1.yaml\u001b[K\n",
      "Running command: seqspec modify -m rna -o \"seqspec_modify_rna_file_names.yaml\" -r \"rna_R2.fastq.gz\" --region-id \"BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\" modrna2.yaml\u001b[K\n",
      "=== Finished seqspec modify rna ===\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_jq_on_kb_count_outputs_to_logs ------\u001b[K\n",
      "Input script_name: cellatlas_jq_run_info_and_inspect.sh\u001b[K\n",
      "Input output_file_name with json : BMMC-single-donor.rna.log.hg38.json\u001b[K\n",
      "Input run_info_json: run_info.json\u001b[K\n",
      "Input inspect_json: inspect.json\u001b[K\n",
      "------ Finished run_jq_on_kb_count_outputs_to_logs ------\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_subpool_to_rna_kb_cout_outputs ------\u001b[K\n",
      "Input subpool_script_modify_h5ad: rna_modify_kb_count_output_barcode_h5.py\u001b[K\n",
      "Input subpool_script_cell_gene: rna_modify_kb_count_cell_gene_barcode.sh\u001b[K\n",
      "Input kb_count_h5ad_file: adata.h5ad\u001b[K\n",
      "Input kb_count_fragments_file: cells_x_genes.barcodes.txt\u001b[K\n",
      "Input fastq1: BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\u001b[K\n",
      "Input fastq2: BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\u001b[K\n",
      "Input spec_yaml: bmmc_spec_igvf_rna.yaml\u001b[K\n",
      "Input whitelist_file: sai_192_whitelist.txt\u001b[K\n",
      "Input seqspec_rna_region_id: rna\u001b[K\n",
      "Input subpool: none\u001b[K\n",
      "------ Function append_subpool_to_h5ad ------\u001b[K\n",
      "Input H5AD file: adata.h5ad\u001b[K\n",
      "Subpool to append: _none\u001b[K\n",
      "Output H5AD file: adata_none.h5ad\u001b[K\n",
      "Copied adata.h5ad to adata_none.h5ad\u001b[K\n",
      "------ Finished append_subpool_to_h5ad ------\u001b[K\n",
      "Output H5AD file: adata_none.h5ad\u001b[K\n",
      "Output Fragments file: cells_x_genes.barcodes_none.txt\u001b[K\n",
      "Input filename: cells_x_genes.barcodes.txt\u001b[K\n",
      "Subpool: none\u001b[K\n",
      "Output filename with subpool: cells_x_genes.barcodes_none.txt\u001b[K\n",
      "File copied to cells_x_genes.barcodes_none.txt\u001b[K\n",
      "Subpool is 'none', skipping sed command.\u001b[K\n",
      "------ Finished run_add_subpool_to_rna_kb_cout_outputs ------\u001b[K\n",
      "\u001b[K\n",
      "calling cellatlas build\u001b[K\n",
      "Output Directory: cellatlas_out\u001b[K\n",
      "Mode: rna\u001b[K\n",
      "Spec File: bmmc_spec_igvf_rna.yaml\u001b[K\n",
      "Genome Fasta Gz: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Gene GTF: genes.gtf\u001b[K\n",
      "Fastq1: BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\u001b[K\n",
      "Fastq2: BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\u001b[K\n",
      "Contents of Output Directory:\u001b[K\n",
      "cellatlas_info.json\u001b[K\n",
      "BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\u001b[K\n",
      "BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\u001b[K\n",
      "GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "bmmc_spec_igvf_rna.yaml\u001b[K\n",
      "cellatlas_out\u001b[K\n",
      "genes.gtf\u001b[K\n",
      "sai_192_whitelist.txt\u001b[K\n",
      "\u001b[K\n",
      "\u001b[84A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2hMUZ1brXDrASM\u001b[22;39m\u001b[K\n",
      "[9b/81e0dc] process > run_cellatlas_build (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[78/ace689] process > run_jq_on_cellatlas (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_kb_ref_with_jq_commands    -\u001b[K\n",
      "[f4/6d209b] process > run_seqspec_modify_rna (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[f6/c0345a] process > run_add_subpool_to_rna_kb_c... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[8f/cb996e] process > run_jq_on_kb_count_outputs_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > rna_calculate_qc_metrics       -\u001b[K\n",
      "[-        ] process > rna_plot_qc_metrics_prcs       -\u001b[K\n",
      "after run_seqspec_modify_rna\u001b[K\n",
      "Calling run_add_subpool_to_rna_kb_cout_outputs with the following inputs:\u001b[K\n",
      "params.RNA_MODIFY_KB_COUNT_H5AD_WITH_SUBPOOL: rna_modify_kb_count_output_barcode_h5.py\u001b[K\n",
      "params.RNA_MODIFY_CELL_GENE_WITH_SUBPOOL: rna_modify_kb_count_cell_gene_barcode.sh\u001b[K\n",
      "adata_out_h5ad: DataflowVariable(value=null)\u001b[K\n",
      "cells_x_genes_barcodes_out: DataflowVariable(value=null)\u001b[K\n",
      "Calling run_jq_on_kb_count_outputs_to_logs with the following inputs:\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: params.RNA_JQ_CELLATLAS_OUTPUT_TO_LOG: cellatlas_jq_run_info_and_inspect.sh\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: kb_count_run_info_json: DataflowVariable(value=null)\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: kb_count_inspect_json: DataflowVariable(value=null)\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: params.LOG_FILE_NAME: BMMC-single-donor.rna.log.hg38.txt\u001b[K\n",
      "=== Start run_seqspec_modify_rna ===\u001b[K\n",
      " Input Fastq Files:\u001b[K\n",
      " R1: BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\u001b[K\n",
      " R2: BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\u001b[K\n",
      " Specification YAML: bmmc_spec_igvf_rna.yaml\u001b[K\n",
      " Whitelist File: sai_192_whitelist.txt\u001b[K\n",
      " seqspec_rna_region_id is: rna\u001b[K\n",
      "Running command: seqspec modify -m rna -o modrna1.yaml -r \"rna_R1.fastq.gz\" --region-id \"BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\" bmmc_spec_igvf_rna.yaml\u001b[K\n",
      "Running command: seqspec modify -m rna -o modrna2.yaml -r \"rna_R2.fastq.gz\" --region-id \"BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\" modrna1.yaml\u001b[K\n",
      "Running command: seqspec modify -m rna -o \"seqspec_modify_rna_file_names.yaml\" -r \"rna_R2.fastq.gz\" --region-id \"BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\" modrna2.yaml\u001b[K\n",
      "=== Finished seqspec modify rna ===\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_jq_on_kb_count_outputs_to_logs ------\u001b[K\n",
      "Input script_name: cellatlas_jq_run_info_and_inspect.sh\u001b[K\n",
      "Input output_file_name with json : BMMC-single-donor.rna.log.hg38.json\u001b[K\n",
      "Input run_info_json: run_info.json\u001b[K\n",
      "Input inspect_json: inspect.json\u001b[K\n",
      "------ Finished run_jq_on_kb_count_outputs_to_logs ------\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_subpool_to_rna_kb_cout_outputs ------\u001b[K\n",
      "Input subpool_script_modify_h5ad: rna_modify_kb_count_output_barcode_h5.py\u001b[K\n",
      "Input subpool_script_cell_gene: rna_modify_kb_count_cell_gene_barcode.sh\u001b[K\n",
      "Input kb_count_h5ad_file: adata.h5ad\u001b[K\n",
      "Input kb_count_fragments_file: cells_x_genes.barcodes.txt\u001b[K\n",
      "Input fastq1: BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\u001b[K\n",
      "Input fastq2: BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\u001b[K\n",
      "Input spec_yaml: bmmc_spec_igvf_rna.yaml\u001b[K\n",
      "Input whitelist_file: sai_192_whitelist.txt\u001b[K\n",
      "Input seqspec_rna_region_id: rna\u001b[K\n",
      "Input subpool: none\u001b[K\n",
      "------ Function append_subpool_to_h5ad ------\u001b[K\n",
      "Input H5AD file: adata.h5ad\u001b[K\n",
      "Subpool to append: _none\u001b[K\n",
      "Output H5AD file: adata_none.h5ad\u001b[K\n",
      "Copied adata.h5ad to adata_none.h5ad\u001b[K\n",
      "------ Finished append_subpool_to_h5ad ------\u001b[K\n",
      "Output H5AD file: adata_none.h5ad\u001b[K\n",
      "Output Fragments file: cells_x_genes.barcodes_none.txt\u001b[K\n",
      "Input filename: cells_x_genes.barcodes.txt\u001b[K\n",
      "Subpool: none\u001b[K\n",
      "Output filename with subpool: cells_x_genes.barcodes_none.txt\u001b[K\n",
      "File copied to cells_x_genes.barcodes_none.txt\u001b[K\n",
      "Subpool is 'none', skipping sed command.\u001b[K\n",
      "------ Finished run_add_subpool_to_rna_kb_cout_outputs ------\u001b[K\n",
      "\u001b[K\n",
      "calling cellatlas build\u001b[K\n",
      "Output Directory: cellatlas_out\u001b[K\n",
      "Mode: rna\u001b[K\n",
      "Spec File: bmmc_spec_igvf_rna.yaml\u001b[K\n",
      "Genome Fasta Gz: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Gene GTF: genes.gtf\u001b[K\n",
      "Fastq1: BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\u001b[K\n",
      "Fastq2: BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\u001b[K\n",
      "Contents of Output Directory:\u001b[K\n",
      "cellatlas_info.json\u001b[K\n",
      "BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\u001b[K\n",
      "BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\u001b[K\n",
      "GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "bmmc_spec_igvf_rna.yaml\u001b[K\n",
      "cellatlas_out\u001b[K\n",
      "genes.gtf\u001b[K\n",
      "sai_192_whitelist.txt\u001b[K\n",
      "\u001b[K\n",
      "calling run_jq_on_cellatlas\u001b[K\n",
      "cellatlas_info.json\u001b[K\n",
      "jq_commands.txt\u001b[K\n",
      "\u001b[K\n",
      "\u001b[88A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2hMUZ1brXDrASM\u001b[22;39m\u001b[K\n",
      "[9b/81e0dc] process > run_cellatlas_build (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[78/ace689] process > run_jq_on_cellatlas (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_kb_ref_with_jq_commands    -\u001b[K\n",
      "[f4/6d209b] process > run_seqspec_modify_rna (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[f6/c0345a] process > run_add_subpool_to_rna_kb_c... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[8f/cb996e] process > run_jq_on_kb_count_outputs_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > rna_calculate_qc_metrics       [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > rna_plot_qc_metrics_prcs       -\u001b[K\n",
      "after run_seqspec_modify_rna\u001b[K\n",
      "Calling run_add_subpool_to_rna_kb_cout_outputs with the following inputs:\u001b[K\n",
      "params.RNA_MODIFY_KB_COUNT_H5AD_WITH_SUBPOOL: rna_modify_kb_count_output_barcode_h5.py\u001b[K\n",
      "params.RNA_MODIFY_CELL_GENE_WITH_SUBPOOL: rna_modify_kb_count_cell_gene_barcode.sh\u001b[K\n",
      "adata_out_h5ad: DataflowVariable(value=null)\u001b[K\n",
      "cells_x_genes_barcodes_out: DataflowVariable(value=null)\u001b[K\n",
      "Calling run_jq_on_kb_count_outputs_to_logs with the following inputs:\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: params.RNA_JQ_CELLATLAS_OUTPUT_TO_LOG: cellatlas_jq_run_info_and_inspect.sh\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: kb_count_run_info_json: DataflowVariable(value=null)\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: kb_count_inspect_json: DataflowVariable(value=null)\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: params.LOG_FILE_NAME: BMMC-single-donor.rna.log.hg38.txt\u001b[K\n",
      "=== Start run_seqspec_modify_rna ===\u001b[K\n",
      " Input Fastq Files:\u001b[K\n",
      " R1: BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\u001b[K\n",
      " R2: BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\u001b[K\n",
      " Specification YAML: bmmc_spec_igvf_rna.yaml\u001b[K\n",
      " Whitelist File: sai_192_whitelist.txt\u001b[K\n",
      " seqspec_rna_region_id is: rna\u001b[K\n",
      "Running command: seqspec modify -m rna -o modrna1.yaml -r \"rna_R1.fastq.gz\" --region-id \"BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\" bmmc_spec_igvf_rna.yaml\u001b[K\n",
      "Running command: seqspec modify -m rna -o modrna2.yaml -r \"rna_R2.fastq.gz\" --region-id \"BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\" modrna1.yaml\u001b[K\n",
      "Running command: seqspec modify -m rna -o \"seqspec_modify_rna_file_names.yaml\" -r \"rna_R2.fastq.gz\" --region-id \"BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\" modrna2.yaml\u001b[K\n",
      "=== Finished seqspec modify rna ===\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_jq_on_kb_count_outputs_to_logs ------\u001b[K\n",
      "Input script_name: cellatlas_jq_run_info_and_inspect.sh\u001b[K\n",
      "Input output_file_name with json : BMMC-single-donor.rna.log.hg38.json\u001b[K\n",
      "Input run_info_json: run_info.json\u001b[K\n",
      "Input inspect_json: inspect.json\u001b[K\n",
      "------ Finished run_jq_on_kb_count_outputs_to_logs ------\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_subpool_to_rna_kb_cout_outputs ------\u001b[K\n",
      "Input subpool_script_modify_h5ad: rna_modify_kb_count_output_barcode_h5.py\u001b[K\n",
      "Input subpool_script_cell_gene: rna_modify_kb_count_cell_gene_barcode.sh\u001b[K\n",
      "Input kb_count_h5ad_file: adata.h5ad\u001b[K\n",
      "Input kb_count_fragments_file: cells_x_genes.barcodes.txt\u001b[K\n",
      "Input fastq1: BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\u001b[K\n",
      "Input fastq2: BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\u001b[K\n",
      "Input spec_yaml: bmmc_spec_igvf_rna.yaml\u001b[K\n",
      "Input whitelist_file: sai_192_whitelist.txt\u001b[K\n",
      "Input seqspec_rna_region_id: rna\u001b[K\n",
      "Input subpool: none\u001b[K\n",
      "------ Function append_subpool_to_h5ad ------\u001b[K\n",
      "Input H5AD file: adata.h5ad\u001b[K\n",
      "Subpool to append: _none\u001b[K\n",
      "Output H5AD file: adata_none.h5ad\u001b[K\n",
      "Copied adata.h5ad to adata_none.h5ad\u001b[K\n",
      "------ Finished append_subpool_to_h5ad ------\u001b[K\n",
      "Output H5AD file: adata_none.h5ad\u001b[K\n",
      "Output Fragments file: cells_x_genes.barcodes_none.txt\u001b[K\n",
      "Input filename: cells_x_genes.barcodes.txt\u001b[K\n",
      "Subpool: none\u001b[K\n",
      "Output filename with subpool: cells_x_genes.barcodes_none.txt\u001b[K\n",
      "File copied to cells_x_genes.barcodes_none.txt\u001b[K\n",
      "Subpool is 'none', skipping sed command.\u001b[K\n",
      "------ Finished run_add_subpool_to_rna_kb_cout_outputs ------\u001b[K\n",
      "\u001b[K\n",
      "calling cellatlas build\u001b[K\n",
      "Output Directory: cellatlas_out\u001b[K\n",
      "Mode: rna\u001b[K\n",
      "Spec File: bmmc_spec_igvf_rna.yaml\u001b[K\n",
      "Genome Fasta Gz: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Gene GTF: genes.gtf\u001b[K\n",
      "Fastq1: BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\u001b[K\n",
      "Fastq2: BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\u001b[K\n",
      "Contents of Output Directory:\u001b[K\n",
      "cellatlas_info.json\u001b[K\n",
      "BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\u001b[K\n",
      "BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\u001b[K\n",
      "GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "bmmc_spec_igvf_rna.yaml\u001b[K\n",
      "cellatlas_out\u001b[K\n",
      "genes.gtf\u001b[K\n",
      "sai_192_whitelist.txt\u001b[K\n",
      "\u001b[K\n",
      "calling run_jq_on_cellatlas\u001b[K\n",
      "cellatlas_info.json\u001b[K\n",
      "jq_commands.txt\u001b[K\n",
      "\u001b[K\n",
      "\u001b[88A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2hMUZ1brXDrASM\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (1)\u001b[K\n",
      "[9b/81e0dc] process > run_cellatlas_build (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[78/ace689] process > run_jq_on_cellatlas (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_kb_ref_with_jq_commands    -\u001b[K\n",
      "[f4/6d209b] process > run_seqspec_modify_rna (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[f6/c0345a] process > run_add_subpool_to_rna_kb_c... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[8f/cb996e] process > run_jq_on_kb_count_outputs_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[09/970c22] process > rna_calculate_qc_metrics (1)   [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > rna_plot_qc_metrics_prcs       -\u001b[K\n",
      "after run_seqspec_modify_rna\u001b[K\n",
      "Calling run_add_subpool_to_rna_kb_cout_outputs with the following inputs:\u001b[K\n",
      "params.RNA_MODIFY_KB_COUNT_H5AD_WITH_SUBPOOL: rna_modify_kb_count_output_barcode_h5.py\u001b[K\n",
      "params.RNA_MODIFY_CELL_GENE_WITH_SUBPOOL: rna_modify_kb_count_cell_gene_barcode.sh\u001b[K\n",
      "adata_out_h5ad: DataflowVariable(value=null)\u001b[K\n",
      "cells_x_genes_barcodes_out: DataflowVariable(value=null)\u001b[K\n",
      "Calling run_jq_on_kb_count_outputs_to_logs with the following inputs:\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: params.RNA_JQ_CELLATLAS_OUTPUT_TO_LOG: cellatlas_jq_run_info_and_inspect.sh\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: kb_count_run_info_json: DataflowVariable(value=null)\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: kb_count_inspect_json: DataflowVariable(value=null)\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: params.LOG_FILE_NAME: BMMC-single-donor.rna.log.hg38.txt\u001b[K\n",
      "=== Start run_seqspec_modify_rna ===\u001b[K\n",
      " Input Fastq Files:\u001b[K\n",
      " R1: BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\u001b[K\n",
      " R2: BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\u001b[K\n",
      " Specification YAML: bmmc_spec_igvf_rna.yaml\u001b[K\n",
      " Whitelist File: sai_192_whitelist.txt\u001b[K\n",
      " seqspec_rna_region_id is: rna\u001b[K\n",
      "Running command: seqspec modify -m rna -o modrna1.yaml -r \"rna_R1.fastq.gz\" --region-id \"BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\" bmmc_spec_igvf_rna.yaml\u001b[K\n",
      "Running command: seqspec modify -m rna -o modrna2.yaml -r \"rna_R2.fastq.gz\" --region-id \"BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\" modrna1.yaml\u001b[K\n",
      "Running command: seqspec modify -m rna -o \"seqspec_modify_rna_file_names.yaml\" -r \"rna_R2.fastq.gz\" --region-id \"BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\" modrna2.yaml\u001b[K\n",
      "=== Finished seqspec modify rna ===\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_jq_on_kb_count_outputs_to_logs ------\u001b[K\n",
      "Input script_name: cellatlas_jq_run_info_and_inspect.sh\u001b[K\n",
      "Input output_file_name with json : BMMC-single-donor.rna.log.hg38.json\u001b[K\n",
      "Input run_info_json: run_info.json\u001b[K\n",
      "Input inspect_json: inspect.json\u001b[K\n",
      "------ Finished run_jq_on_kb_count_outputs_to_logs ------\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_subpool_to_rna_kb_cout_outputs ------\u001b[K\n",
      "Input subpool_script_modify_h5ad: rna_modify_kb_count_output_barcode_h5.py\u001b[K\n",
      "Input subpool_script_cell_gene: rna_modify_kb_count_cell_gene_barcode.sh\u001b[K\n",
      "Input kb_count_h5ad_file: adata.h5ad\u001b[K\n",
      "Input kb_count_fragments_file: cells_x_genes.barcodes.txt\u001b[K\n",
      "Input fastq1: BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\u001b[K\n",
      "Input fastq2: BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\u001b[K\n",
      "Input spec_yaml: bmmc_spec_igvf_rna.yaml\u001b[K\n",
      "Input whitelist_file: sai_192_whitelist.txt\u001b[K\n",
      "Input seqspec_rna_region_id: rna\u001b[K\n",
      "Input subpool: none\u001b[K\n",
      "------ Function append_subpool_to_h5ad ------\u001b[K\n",
      "Input H5AD file: adata.h5ad\u001b[K\n",
      "Subpool to append: _none\u001b[K\n",
      "Output H5AD file: adata_none.h5ad\u001b[K\n",
      "Copied adata.h5ad to adata_none.h5ad\u001b[K\n",
      "------ Finished append_subpool_to_h5ad ------\u001b[K\n",
      "Output H5AD file: adata_none.h5ad\u001b[K\n",
      "Output Fragments file: cells_x_genes.barcodes_none.txt\u001b[K\n",
      "Input filename: cells_x_genes.barcodes.txt\u001b[K\n",
      "Subpool: none\u001b[K\n",
      "Output filename with subpool: cells_x_genes.barcodes_none.txt\u001b[K\n",
      "File copied to cells_x_genes.barcodes_none.txt\u001b[K\n",
      "Subpool is 'none', skipping sed command.\u001b[K\n",
      "------ Finished run_add_subpool_to_rna_kb_cout_outputs ------\u001b[K\n",
      "\u001b[K\n",
      "calling cellatlas build\u001b[K\n",
      "Output Directory: cellatlas_out\u001b[K\n",
      "Mode: rna\u001b[K\n",
      "Spec File: bmmc_spec_igvf_rna.yaml\u001b[K\n",
      "Genome Fasta Gz: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Gene GTF: genes.gtf\u001b[K\n",
      "Fastq1: BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\u001b[K\n",
      "Fastq2: BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\u001b[K\n",
      "Contents of Output Directory:\u001b[K\n",
      "cellatlas_info.json\u001b[K\n",
      "BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\u001b[K\n",
      "BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\u001b[K\n",
      "GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "bmmc_spec_igvf_rna.yaml\u001b[K\n",
      "cellatlas_out\u001b[K\n",
      "genes.gtf\u001b[K\n",
      "sai_192_whitelist.txt\u001b[K\n",
      "\u001b[K\n",
      "calling run_jq_on_cellatlas\u001b[K\n",
      "cellatlas_info.json\u001b[K\n",
      "jq_commands.txt\u001b[K\n",
      "\u001b[K\n",
      "\u001b[89A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2hMUZ1brXDrASM\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (1)\u001b[K\n",
      "[9b/81e0dc] process > run_cellatlas_build (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[78/ace689] process > run_jq_on_cellatlas (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[-        ] process > run_kb_ref_with_jq_commands    [  0%] 0 of 1\u001b[K\n",
      "[f4/6d209b] process > run_seqspec_modify_rna (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[f6/c0345a] process > run_add_subpool_to_rna_kb_c... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[8f/cb996e] process > run_jq_on_kb_count_outputs_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[09/970c22] process > rna_calculate_qc_metrics (1)   [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > rna_plot_qc_metrics_prcs       -\u001b[K\n",
      "after run_seqspec_modify_rna\u001b[K\n",
      "Calling run_add_subpool_to_rna_kb_cout_outputs with the following inputs:\u001b[K\n",
      "params.RNA_MODIFY_KB_COUNT_H5AD_WITH_SUBPOOL: rna_modify_kb_count_output_barcode_h5.py\u001b[K\n",
      "params.RNA_MODIFY_CELL_GENE_WITH_SUBPOOL: rna_modify_kb_count_cell_gene_barcode.sh\u001b[K\n",
      "adata_out_h5ad: DataflowVariable(value=null)\u001b[K\n",
      "cells_x_genes_barcodes_out: DataflowVariable(value=null)\u001b[K\n",
      "Calling run_jq_on_kb_count_outputs_to_logs with the following inputs:\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: params.RNA_JQ_CELLATLAS_OUTPUT_TO_LOG: cellatlas_jq_run_info_and_inspect.sh\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: kb_count_run_info_json: DataflowVariable(value=null)\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: kb_count_inspect_json: DataflowVariable(value=null)\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: params.LOG_FILE_NAME: BMMC-single-donor.rna.log.hg38.txt\u001b[K\n",
      "=== Start run_seqspec_modify_rna ===\u001b[K\n",
      " Input Fastq Files:\u001b[K\n",
      " R1: BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\u001b[K\n",
      " R2: BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\u001b[K\n",
      " Specification YAML: bmmc_spec_igvf_rna.yaml\u001b[K\n",
      " Whitelist File: sai_192_whitelist.txt\u001b[K\n",
      " seqspec_rna_region_id is: rna\u001b[K\n",
      "Running command: seqspec modify -m rna -o modrna1.yaml -r \"rna_R1.fastq.gz\" --region-id \"BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\" bmmc_spec_igvf_rna.yaml\u001b[K\n",
      "Running command: seqspec modify -m rna -o modrna2.yaml -r \"rna_R2.fastq.gz\" --region-id \"BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\" modrna1.yaml\u001b[K\n",
      "Running command: seqspec modify -m rna -o \"seqspec_modify_rna_file_names.yaml\" -r \"rna_R2.fastq.gz\" --region-id \"BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\" modrna2.yaml\u001b[K\n",
      "=== Finished seqspec modify rna ===\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_jq_on_kb_count_outputs_to_logs ------\u001b[K\n",
      "Input script_name: cellatlas_jq_run_info_and_inspect.sh\u001b[K\n",
      "Input output_file_name with json : BMMC-single-donor.rna.log.hg38.json\u001b[K\n",
      "Input run_info_json: run_info.json\u001b[K\n",
      "Input inspect_json: inspect.json\u001b[K\n",
      "------ Finished run_jq_on_kb_count_outputs_to_logs ------\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_subpool_to_rna_kb_cout_outputs ------\u001b[K\n",
      "Input subpool_script_modify_h5ad: rna_modify_kb_count_output_barcode_h5.py\u001b[K\n",
      "Input subpool_script_cell_gene: rna_modify_kb_count_cell_gene_barcode.sh\u001b[K\n",
      "Input kb_count_h5ad_file: adata.h5ad\u001b[K\n",
      "Input kb_count_fragments_file: cells_x_genes.barcodes.txt\u001b[K\n",
      "Input fastq1: BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\u001b[K\n",
      "Input fastq2: BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\u001b[K\n",
      "Input spec_yaml: bmmc_spec_igvf_rna.yaml\u001b[K\n",
      "Input whitelist_file: sai_192_whitelist.txt\u001b[K\n",
      "Input seqspec_rna_region_id: rna\u001b[K\n",
      "Input subpool: none\u001b[K\n",
      "------ Function append_subpool_to_h5ad ------\u001b[K\n",
      "Input H5AD file: adata.h5ad\u001b[K\n",
      "Subpool to append: _none\u001b[K\n",
      "Output H5AD file: adata_none.h5ad\u001b[K\n",
      "Copied adata.h5ad to adata_none.h5ad\u001b[K\n",
      "------ Finished append_subpool_to_h5ad ------\u001b[K\n",
      "Output H5AD file: adata_none.h5ad\u001b[K\n",
      "Output Fragments file: cells_x_genes.barcodes_none.txt\u001b[K\n",
      "Input filename: cells_x_genes.barcodes.txt\u001b[K\n",
      "Subpool: none\u001b[K\n",
      "Output filename with subpool: cells_x_genes.barcodes_none.txt\u001b[K\n",
      "File copied to cells_x_genes.barcodes_none.txt\u001b[K\n",
      "Subpool is 'none', skipping sed command.\u001b[K\n",
      "------ Finished run_add_subpool_to_rna_kb_cout_outputs ------\u001b[K\n",
      "\u001b[K\n",
      "calling cellatlas build\u001b[K\n",
      "Output Directory: cellatlas_out\u001b[K\n",
      "Mode: rna\u001b[K\n",
      "Spec File: bmmc_spec_igvf_rna.yaml\u001b[K\n",
      "Genome Fasta Gz: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Gene GTF: genes.gtf\u001b[K\n",
      "Fastq1: BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\u001b[K\n",
      "Fastq2: BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\u001b[K\n",
      "Contents of Output Directory:\u001b[K\n",
      "cellatlas_info.json\u001b[K\n",
      "BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\u001b[K\n",
      "BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\u001b[K\n",
      "GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "bmmc_spec_igvf_rna.yaml\u001b[K\n",
      "cellatlas_out\u001b[K\n",
      "genes.gtf\u001b[K\n",
      "sai_192_whitelist.txt\u001b[K\n",
      "\u001b[K\n",
      "calling run_jq_on_cellatlas\u001b[K\n",
      "cellatlas_info.json\u001b[K\n",
      "jq_commands.txt\u001b[K\n",
      "\u001b[K\n",
      "\u001b[89A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2hMUZ1brXDrASM\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (2)\u001b[K\n",
      "[9b/81e0dc] process > run_cellatlas_build (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[78/ace689] process > run_jq_on_cellatlas (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[9b/875869] process > run_kb_ref_with_jq_commands... [  0%] 0 of 1\u001b[K\n",
      "[f4/6d209b] process > run_seqspec_modify_rna (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[f6/c0345a] process > run_add_subpool_to_rna_kb_c... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[8f/cb996e] process > run_jq_on_kb_count_outputs_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[09/970c22] process > rna_calculate_qc_metrics (1)   [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > rna_plot_qc_metrics_prcs       -\u001b[K\n",
      "after run_seqspec_modify_rna\u001b[K\n",
      "Calling run_add_subpool_to_rna_kb_cout_outputs with the following inputs:\u001b[K\n",
      "params.RNA_MODIFY_KB_COUNT_H5AD_WITH_SUBPOOL: rna_modify_kb_count_output_barcode_h5.py\u001b[K\n",
      "params.RNA_MODIFY_CELL_GENE_WITH_SUBPOOL: rna_modify_kb_count_cell_gene_barcode.sh\u001b[K\n",
      "adata_out_h5ad: DataflowVariable(value=null)\u001b[K\n",
      "cells_x_genes_barcodes_out: DataflowVariable(value=null)\u001b[K\n",
      "Calling run_jq_on_kb_count_outputs_to_logs with the following inputs:\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: params.RNA_JQ_CELLATLAS_OUTPUT_TO_LOG: cellatlas_jq_run_info_and_inspect.sh\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: kb_count_run_info_json: DataflowVariable(value=null)\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: kb_count_inspect_json: DataflowVariable(value=null)\u001b[K\n",
      "run_jq_on_kb_count_outputs_to_logs: params.LOG_FILE_NAME: BMMC-single-donor.rna.log.hg38.txt\u001b[K\n",
      "=== Start run_seqspec_modify_rna ===\u001b[K\n",
      " Input Fastq Files:\u001b[K\n",
      " R1: BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\u001b[K\n",
      " R2: BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\u001b[K\n",
      " Specification YAML: bmmc_spec_igvf_rna.yaml\u001b[K\n",
      " Whitelist File: sai_192_whitelist.txt\u001b[K\n",
      " seqspec_rna_region_id is: rna\u001b[K\n",
      "Running command: seqspec modify -m rna -o modrna1.yaml -r \"rna_R1.fastq.gz\" --region-id \"BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\" bmmc_spec_igvf_rna.yaml\u001b[K\n",
      "Running command: seqspec modify -m rna -o modrna2.yaml -r \"rna_R2.fastq.gz\" --region-id \"BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\" modrna1.yaml\u001b[K\n",
      "Running command: seqspec modify -m rna -o \"seqspec_modify_rna_file_names.yaml\" -r \"rna_R2.fastq.gz\" --region-id \"BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\" modrna2.yaml\u001b[K\n",
      "=== Finished seqspec modify rna ===\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_jq_on_kb_count_outputs_to_logs ------\u001b[K\n",
      "Input script_name: cellatlas_jq_run_info_and_inspect.sh\u001b[K\n",
      "Input output_file_name with json : BMMC-single-donor.rna.log.hg38.json\u001b[K\n",
      "Input run_info_json: run_info.json\u001b[K\n",
      "Input inspect_json: inspect.json\u001b[K\n",
      "------ Finished run_jq_on_kb_count_outputs_to_logs ------\u001b[K\n",
      "\u001b[K\n",
      "------ Start run_add_subpool_to_rna_kb_cout_outputs ------\u001b[K\n",
      "Input subpool_script_modify_h5ad: rna_modify_kb_count_output_barcode_h5.py\u001b[K\n",
      "Input subpool_script_cell_gene: rna_modify_kb_count_cell_gene_barcode.sh\u001b[K\n",
      "Input kb_count_h5ad_file: adata.h5ad\u001b[K\n",
      "Input kb_count_fragments_file: cells_x_genes.barcodes.txt\u001b[K\n",
      "Input fastq1: BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\u001b[K\n",
      "Input fastq2: BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\u001b[K\n",
      "Input spec_yaml: bmmc_spec_igvf_rna.yaml\u001b[K\n",
      "Input whitelist_file: sai_192_whitelist.txt\u001b[K\n",
      "Input seqspec_rna_region_id: rna\u001b[K\n",
      "Input subpool: none\u001b[K\n",
      "------ Function append_subpool_to_h5ad ------\u001b[K\n",
      "Input H5AD file: adata.h5ad\u001b[K\n",
      "Subpool to append: _none\u001b[K\n",
      "Output H5AD file: adata_none.h5ad\u001b[K\n",
      "Copied adata.h5ad to adata_none.h5ad\u001b[K\n",
      "------ Finished append_subpool_to_h5ad ------\u001b[K\n",
      "Output H5AD file: adata_none.h5ad\u001b[K\n",
      "Output Fragments file: cells_x_genes.barcodes_none.txt\u001b[K\n",
      "Input filename: cells_x_genes.barcodes.txt\u001b[K\n",
      "Subpool: none\u001b[K\n",
      "Output filename with subpool: cells_x_genes.barcodes_none.txt\u001b[K\n",
      "File copied to cells_x_genes.barcodes_none.txt\u001b[K\n",
      "Subpool is 'none', skipping sed command.\u001b[K\n",
      "------ Finished run_add_subpool_to_rna_kb_cout_outputs ------\u001b[K\n",
      "\u001b[K\n",
      "calling cellatlas build\u001b[K\n",
      "Output Directory: cellatlas_out\u001b[K\n",
      "Mode: rna\u001b[K\n",
      "Spec File: bmmc_spec_igvf_rna.yaml\u001b[K\n",
      "Genome Fasta Gz: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Gene GTF: genes.gtf\u001b[K\n",
      "Fastq1: BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\u001b[K\n",
      "Fastq2: BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\u001b[K\n",
      "Contents of Output Directory:\u001b[K\n",
      "cellatlas_info.json\u001b[K\n",
      "BMMC_single_donor_RNA_L001_R1_corrected.fastq.gz\u001b[K\n",
      "BMMC_single_donor_RNA_L001_R2_corrected.fastq.gz\u001b[K\n",
      "GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "bmmc_spec_igvf_rna.yaml\u001b[K\n",
      "cellatlas_out\u001b[K\n",
      "genes.gtf\u001b[K\n",
      "sai_192_whitelist.txt\u001b[K\n",
      "\u001b[K\n",
      "calling run_jq_on_cellatlas\u001b[K\n",
      "cellatlas_info.json\u001b[K\n",
      "jq_commands.txt\u001b[K\n",
      "\u001b[K\n",
      "\u001b[89A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2hMUZ1brXDrASM\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (2)\u001b[K\n",
      "[9b/81e0dc] process > run_cellatlas_build (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[78/ace689] process > run_jq_on_cellatlas (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[9b/875869] process > run_kb_ref_with_jq_commands... [  0%] 0 of 1\u001b[K\n",
      "[f4/6d209b] process > run_seqspec_modify_rna (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[f6/c0345a] process > run_add_subpool_to_rna_kb_c... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[8f/cb996e] process > run_jq_on_kb_count_outputs_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[09/970c22] process > rna_calculate_qc_metrics (1)   [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > rna_plot_qc_metrics_prcs       -\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[89A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2hMUZ1brXDrASM\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (2)\u001b[K\n",
      "[9b/81e0dc] process > run_cellatlas_build (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[78/ace689] process > run_jq_on_cellatlas (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[9b/875869] process > run_kb_ref_with_jq_commands... [  0%] 0 of 1\u001b[K\n",
      "[f4/6d209b] process > run_seqspec_modify_rna (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[f6/c0345a] process > run_add_subpool_to_rna_kb_c... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[8f/cb996e] process > run_jq_on_kb_count_outputs_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[09/970c22] process > rna_calculate_qc_metrics (1)   [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > rna_plot_qc_metrics_prcs       -\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[89A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2hMUZ1brXDrASM\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (2)\u001b[K\n",
      "[9b/81e0dc] process > run_cellatlas_build (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[78/ace689] process > run_jq_on_cellatlas (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[9b/875869] process > run_kb_ref_with_jq_commands... [  0%] 0 of 1\u001b[K\n",
      "[f4/6d209b] process > run_seqspec_modify_rna (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[f6/c0345a] process > run_add_subpool_to_rna_kb_c... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[8f/cb996e] process > run_jq_on_kb_count_outputs_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[09/970c22] process > rna_calculate_qc_metrics (1)   [  0%] 0 of 1 ✔\u001b[K\n",
      "[-        ] process > rna_plot_qc_metrics_prcs       -\u001b[K\n",
      "Conda environment: # conda environments:\u001b[K\n",
      "#\u001b[K\n",
      "base                     /opt/conda\u001b[K\n",
      "rna_atac_plot_qc_env     /opt/conda/envs/rna_atac_plot_qc_env\u001b[K\n",
      "Script path: /usr/local/bin/rna_qc_metrics.py\u001b[K\n",
      "Adata path: adata_none.h5ad\u001b[K\n",
      "Subpool argument: --subpool none\u001b[K\n",
      "Starting rna_calculate_qc_metrics...\u001b[K\n",
      "Result file path: \u001b[K\n",
      "Finished rna_calculate_qc_metrics.\u001b[K\n",
      "\u001b[K\n",
      "\u001b[22A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2hMUZ1brXDrASM\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (2)\u001b[K\n",
      "[9b/81e0dc] process > run_cellatlas_build (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[78/ace689] process > run_jq_on_cellatlas (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[9b/875869] process > run_kb_ref_with_jq_commands... [  0%] 0 of 1\u001b[K\n",
      "[f4/6d209b] process > run_seqspec_modify_rna (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[f6/c0345a] process > run_add_subpool_to_rna_kb_c... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[8f/cb996e] process > run_jq_on_kb_count_outputs_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[09/970c22] process > rna_calculate_qc_metrics (1)   [100%] 1 of 1 ✔\u001b[K\n",
      "[-        ] process > rna_plot_qc_metrics_prcs       [  0%] 0 of 1\u001b[K\n",
      "Conda environment: # conda environments:\u001b[K\n",
      "#\u001b[K\n",
      "base                     /opt/conda\u001b[K\n",
      "rna_atac_plot_qc_env     /opt/conda/envs/rna_atac_plot_qc_env\u001b[K\n",
      "Script path: /usr/local/bin/rna_qc_metrics.py\u001b[K\n",
      "Adata path: adata_none.h5ad\u001b[K\n",
      "Subpool argument: --subpool none\u001b[K\n",
      "Starting rna_calculate_qc_metrics...\u001b[K\n",
      "Result file path: \u001b[K\n",
      "Finished rna_calculate_qc_metrics.\u001b[K\n",
      "\u001b[K\n",
      "\u001b[22A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2hMUZ1brXDrASM\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (3)\u001b[K\n",
      "[9b/81e0dc] process > run_cellatlas_build (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[78/ace689] process > run_jq_on_cellatlas (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[9b/875869] process > run_kb_ref_with_jq_commands... [  0%] 0 of 1\u001b[K\n",
      "[f4/6d209b] process > run_seqspec_modify_rna (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[f6/c0345a] process > run_add_subpool_to_rna_kb_c... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[8f/cb996e] process > run_jq_on_kb_count_outputs_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[09/970c22] process > rna_calculate_qc_metrics (1)   [100%] 1 of 1 ✔\u001b[K\n",
      "[d4/b3712b] process > rna_plot_qc_metrics_prcs (1)   [  0%] 0 of 1\u001b[K\n",
      "Conda environment: # conda environments:\u001b[K\n",
      "#\u001b[K\n",
      "base                     /opt/conda\u001b[K\n",
      "rna_atac_plot_qc_env     /opt/conda/envs/rna_atac_plot_qc_env\u001b[K\n",
      "Script path: /usr/local/bin/rna_qc_metrics.py\u001b[K\n",
      "Adata path: adata_none.h5ad\u001b[K\n",
      "Subpool argument: --subpool none\u001b[K\n",
      "Starting rna_calculate_qc_metrics...\u001b[K\n",
      "Result file path: \u001b[K\n",
      "Finished rna_calculate_qc_metrics.\u001b[K\n",
      "\u001b[K\n",
      "\u001b[22A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2hMUZ1brXDrASM\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (3)\u001b[K\n",
      "[9b/81e0dc] process > run_cellatlas_build (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[78/ace689] process > run_jq_on_cellatlas (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[9b/875869] process > run_kb_ref_with_jq_commands... [  0%] 0 of 1\u001b[K\n",
      "[f4/6d209b] process > run_seqspec_modify_rna (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[f6/c0345a] process > run_add_subpool_to_rna_kb_c... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[8f/cb996e] process > run_jq_on_kb_count_outputs_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[09/970c22] process > rna_calculate_qc_metrics (1)   [100%] 1 of 1 ✔\u001b[K\n",
      "[d4/b3712b] process > rna_plot_qc_metrics_prcs (1)   [  0%] 0 of 1\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[22A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2hMUZ1brXDrASM\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (3)\u001b[K\n",
      "[9b/81e0dc] process > run_cellatlas_build (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[78/ace689] process > run_jq_on_cellatlas (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[9b/875869] process > run_kb_ref_with_jq_commands... [  0%] 0 of 1\u001b[K\n",
      "[f4/6d209b] process > run_seqspec_modify_rna (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[f6/c0345a] process > run_add_subpool_to_rna_kb_c... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[8f/cb996e] process > run_jq_on_kb_count_outputs_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[09/970c22] process > rna_calculate_qc_metrics (1)   [100%] 1 of 1 ✔\u001b[K\n",
      "[d4/b3712b] process > rna_plot_qc_metrics_prcs (1)   [  0%] 0 of 1 ✔\u001b[K\n",
      "Starting rna_plot_qc_metrics_prcs...\u001b[K\n",
      "R QC Plot Script: rna_plot_qc_metrics.R\u001b[K\n",
      "R QC Plot Helper Script: barcode_rank_functions.R\u001b[K\n",
      "RNA QC Metrics TSV: rna_qc_calculate.tsv\u001b[K\n",
      "UMI Cutoff: 2\u001b[K\n",
      "Gene Cutoff: 3\u001b[K\n",
      "UMI Rank Plot All Output: umi_plot_all_BMMC-single-donor.rna.log.hg38.png\u001b[K\n",
      "UMI Rank Plot Top Output: umi_plot_top_BMMC-single-donor.rna.log.hg38.png\u001b[K\n",
      "Gene UMI Plot File Output: gene_umi_BMMC-single-donor.rna.log.hg38.png\u001b[K\n",
      "Script path: /usr/local/bin/rna_plot_qc_metrics.R\u001b[K\n",
      "Executing main script...\u001b[K\n",
      "Found file: /usr/local/bin/barcode_rank_functions.R \u001b[K\n",
      "Hello, this is the main function!\u001b[K\n",
      "Barcode Metadata File: rna_qc_calculate.tsv \u001b[K\n",
      "UMI Cutoff: 2 \u001b[K\n",
      "Gene Cutoff: 3 \u001b[K\n",
      "UMI Rank Plot File: umi_plot_all_BMMC-single-donor.rna.log.hg38.png \u001b[K\n",
      "GENE Rank Plot File: umi_plot_top_BMMC-single-donor.rna.log.hg38.png \u001b[K\n",
      "GENE UMI Rank Plot File: gene_umi_BMMC-single-donor.rna.log.hg38.png \u001b[K\n",
      "current_directory: /fusion/gs/eila-nextflow-bucket/work/RNA/d4/b3712b6c3b867e1ed96b3698c8fd8b \u001b[K\n",
      "Files in the working directory:\u001b[K\n",
      "rna_qc_calculate.tsv\u001b[K\n",
      "PATH: /usr/local/bin:/opt/conda/envs/rna_atac_plot_qc_env/bin:/opt/conda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin \u001b[K\n",
      "Printing RNA QC metrics...\u001b[K\n",
      "print_rna_qc_metrics: Parsed Arguments:\u001b[K\n",
      "print_rna_qc_metrics: Barcode Metadata File: rna_qc_calculate.tsv \u001b[K\n",
      "print_rna_qc_metrics: UMI Cutoff: 2 \u001b[K\n",
      "print_rna_qc_metrics: Gene Cutoff: 3 \u001b[K\n",
      "print_rna_qc_metrics: UMI Rank Plot File: umi_plot_all_BMMC-single-donor.rna.log.hg38.png \u001b[K\n",
      "print_rna_qc_metrics: GENE Rank Plot File: umi_plot_top_BMMC-single-donor.rna.log.hg38.png \u001b[K\n",
      "print_rna_qc_metrics: GENE UMI Rank Plot File: gene_umi_BMMC-single-donor.rna.log.hg38.png \u001b[K\n",
      "print_rna_qc_metrics: Reading barcode metadata from file: rna_qc_calculate.tsv \u001b[K\n",
      "print_rna_qc_metrics: calling get_elbow_knee_points \u001b[K\n",
      "[1] \"Returning sliced vectors\"\u001b[K\n",
      "[1] \"Returning sliced vectors\"\u001b[K\n",
      "null device \u001b[K\n",
      "          1 \u001b[K\n",
      "Finished rna_plot_qc_metrics_prcs.\u001b[K\n",
      "\u001b[K\n",
      "\u001b[50A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2hMUZ1brXDrASM\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (3)\u001b[K\n",
      "[9b/81e0dc] process > run_cellatlas_build (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[78/ace689] process > run_jq_on_cellatlas (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[9b/875869] process > run_kb_ref_with_jq_commands... [  0%] 0 of 1\u001b[K\n",
      "[f4/6d209b] process > run_seqspec_modify_rna (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[f6/c0345a] process > run_add_subpool_to_rna_kb_c... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[8f/cb996e] process > run_jq_on_kb_count_outputs_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[09/970c22] process > rna_calculate_qc_metrics (1)   [100%] 1 of 1 ✔\u001b[K\n",
      "[d4/b3712b] process > rna_plot_qc_metrics_prcs (1)   [100%] 1 of 1 ✔\u001b[K\n",
      "Starting rna_plot_qc_metrics_prcs...\u001b[K\n",
      "R QC Plot Script: rna_plot_qc_metrics.R\u001b[K\n",
      "R QC Plot Helper Script: barcode_rank_functions.R\u001b[K\n",
      "RNA QC Metrics TSV: rna_qc_calculate.tsv\u001b[K\n",
      "UMI Cutoff: 2\u001b[K\n",
      "Gene Cutoff: 3\u001b[K\n",
      "UMI Rank Plot All Output: umi_plot_all_BMMC-single-donor.rna.log.hg38.png\u001b[K\n",
      "UMI Rank Plot Top Output: umi_plot_top_BMMC-single-donor.rna.log.hg38.png\u001b[K\n",
      "Gene UMI Plot File Output: gene_umi_BMMC-single-donor.rna.log.hg38.png\u001b[K\n",
      "Script path: /usr/local/bin/rna_plot_qc_metrics.R\u001b[K\n",
      "Executing main script...\u001b[K\n",
      "Found file: /usr/local/bin/barcode_rank_functions.R \u001b[K\n",
      "Hello, this is the main function!\u001b[K\n",
      "Barcode Metadata File: rna_qc_calculate.tsv \u001b[K\n",
      "UMI Cutoff: 2 \u001b[K\n",
      "Gene Cutoff: 3 \u001b[K\n",
      "UMI Rank Plot File: umi_plot_all_BMMC-single-donor.rna.log.hg38.png \u001b[K\n",
      "GENE Rank Plot File: umi_plot_top_BMMC-single-donor.rna.log.hg38.png \u001b[K\n",
      "GENE UMI Rank Plot File: gene_umi_BMMC-single-donor.rna.log.hg38.png \u001b[K\n",
      "current_directory: /fusion/gs/eila-nextflow-bucket/work/RNA/d4/b3712b6c3b867e1ed96b3698c8fd8b \u001b[K\n",
      "Files in the working directory:\u001b[K\n",
      "rna_qc_calculate.tsv\u001b[K\n",
      "PATH: /usr/local/bin:/opt/conda/envs/rna_atac_plot_qc_env/bin:/opt/conda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin \u001b[K\n",
      "Printing RNA QC metrics...\u001b[K\n",
      "print_rna_qc_metrics: Parsed Arguments:\u001b[K\n",
      "print_rna_qc_metrics: Barcode Metadata File: rna_qc_calculate.tsv \u001b[K\n",
      "print_rna_qc_metrics: UMI Cutoff: 2 \u001b[K\n",
      "print_rna_qc_metrics: Gene Cutoff: 3 \u001b[K\n",
      "print_rna_qc_metrics: UMI Rank Plot File: umi_plot_all_BMMC-single-donor.rna.log.hg38.png \u001b[K\n",
      "print_rna_qc_metrics: GENE Rank Plot File: umi_plot_top_BMMC-single-donor.rna.log.hg38.png \u001b[K\n",
      "print_rna_qc_metrics: GENE UMI Rank Plot File: gene_umi_BMMC-single-donor.rna.log.hg38.png \u001b[K\n",
      "print_rna_qc_metrics: Reading barcode metadata from file: rna_qc_calculate.tsv \u001b[K\n",
      "print_rna_qc_metrics: calling get_elbow_knee_points \u001b[K\n",
      "[1] \"Returning sliced vectors\"\u001b[K\n",
      "[1] \"Returning sliced vectors\"\u001b[K\n",
      "null device \u001b[K\n",
      "          1 \u001b[K\n",
      "Finished rna_plot_qc_metrics_prcs.\u001b[K\n",
      "\u001b[K\n",
      "\u001b[50A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2hMUZ1brXDrASM\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (3)\u001b[K\n",
      "[9b/81e0dc] process > run_cellatlas_build (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[78/ace689] process > run_jq_on_cellatlas (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[9b/875869] process > run_kb_ref_with_jq_commands... [  0%] 0 of 1 ✔\u001b[K\n",
      "[f4/6d209b] process > run_seqspec_modify_rna (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[f6/c0345a] process > run_add_subpool_to_rna_kb_c... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[8f/cb996e] process > run_jq_on_kb_count_outputs_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[09/970c22] process > rna_calculate_qc_metrics (1)   [100%] 1 of 1 ✔\u001b[K\n",
      "[d4/b3712b] process > rna_plot_qc_metrics_prcs (1)   [100%] 1 of 1 ✔\u001b[K\n",
      "start run_kb_ref_with_jq_commands\u001b[K\n",
      "Index File: index.idx\u001b[K\n",
      "T2G File: t2g.txt\u001b[K\n",
      "Transcriptome File: transcriptome.fa\u001b[K\n",
      "Genome Fasta: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Gene GTF: genes.gtf\u001b[K\n",
      "JQ Commands File: jq_commands.txt\u001b[K\n",
      "finished run_kb_ref_with_jq_commands\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[59A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2hMUZ1brXDrASM\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (3)\u001b[K\n",
      "[9b/81e0dc] process > run_cellatlas_build (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[78/ace689] process > run_jq_on_cellatlas (1)        [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[9b/875869] process > run_kb_ref_with_jq_commands... [100%] 1 of 1 ✔\u001b[K\n",
      "[f4/6d209b] process > run_seqspec_modify_rna (1)     [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[f6/c0345a] process > run_add_subpool_to_rna_kb_c... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[8f/cb996e] process > run_jq_on_kb_count_outputs_... [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[09/970c22] process > rna_calculate_qc_metrics (1)   [100%] 1 of 1 ✔\u001b[K\n",
      "[d4/b3712b] process > rna_plot_qc_metrics_prcs (1)   [100%] 1 of 1 ✔\u001b[K\n",
      "start run_kb_ref_with_jq_commands\u001b[K\n",
      "Index File: index.idx\u001b[K\n",
      "T2G File: t2g.txt\u001b[K\n",
      "Transcriptome File: transcriptome.fa\u001b[K\n",
      "Genome Fasta: GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz\u001b[K\n",
      "Gene GTF: genes.gtf\u001b[K\n",
      "JQ Commands File: jq_commands.txt\u001b[K\n",
      "finished run_kb_ref_with_jq_commands\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[32;1mCompleted at: 08-Mar-2024 05:26:47\n",
      "Duration    : 22m 4s\n",
      "CPU hours   : 0.4 (1.2% cached)\n",
      "Succeeded   : 3\n",
      "Cached      : 5\n",
      "\u001b[22;39m\u001b[K\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_rna.config -profile google_batch_profile -with-tower -resume\n",
    "# -with-dag\n",
    "\n",
    "# -resume \n",
    "# 8af1ba30-6606-4dfd-a851-c6d0b74bdd43\n",
    "# 8af1ba30-6606-4dfd-a851-c6d0b74bdd43\n",
    "# 8af1ba30-6606-4dfd-a851-c6d0b74bdd43\n",
    "\n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_vm.config -profile docker_local -resume\n",
    "\n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_vm.config -profile google_batch_profile  -with-tower\n",
    "\n",
    "# docker_local\n",
    "\n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_vm.config -profile google_batch_profile\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow.config -profile docker_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_vm.config -profile google_batch_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_vm_anvil.config -profile google_batch_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_local.config -profile google_batch_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_local.config -profile docker_profile"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_rna\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Mar-08 02:46:30.918 [main] DEBUG nextflow.cli.Launcher - $> nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_rna.config -profile google_batch_profile -with-tower -resume\n",
      "Mar-08 02:46:31.156 [main] INFO  nextflow.cli.CmdRun - N E X T F L O W  ~  version 23.10.1\n",
      "Mar-08 02:46:31.192 [main] DEBUG nextflow.plugin.PluginsFacade - Setting up plugin manager > mode=prod; embedded=false; plugins-dir=/root/.nextflow/plugins; core-plugins: nf-amazon@2.1.4,nf-azure@1.3.3,nf-cloudcache@0.3.0,nf-codecommit@0.1.5,nf-console@1.0.6,nf-ga4gh@1.1.0,nf-google@1.8.3,nf-tower@1.6.3,nf-wave@1.0.1\n",
      "Mar-08 02:46:31.206 [main] INFO  o.pf4j.DefaultPluginStatusProvider - Enabled plugins: []\n",
      "Mar-08 02:46:31.208 [main] INFO  o.pf4j.DefaultPluginStatusProvider - Disabled plugins: []\n",
      "Mar-08 02:46:31.214 [main] INFO  org.pf4j.DefaultPluginManager - PF4J version 3.4.1 in 'deployment' mode\n",
      "Mar-08 02:46:31.230 [main] INFO  org.pf4j.AbstractPluginManager - No plugins\n",
      "Mar-08 02:46:31.264 [main] DEBUG nextflow.config.ConfigBuilder - User config file: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_rna/nextflow_rna.config\n",
      "Mar-08 02:46:31.265 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_rna/nextflow_rna.config\n",
      "Mar-08 02:46:31.333 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `google_batch_profile`\n",
      "Mar-08 02:46:31.786 [main] ERROR nextflow.cli.Launcher - Unable to parse config file: '/home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_rna/nextflow_rna.config'\n",
      "\n",
      "  Compile failed for sources FixedSetSources[name='/groovy/script/Script5E292565B0432F3C3737BD95DCE48EB3/_nf_config_5b6902a0']. Cause: org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed:\n",
      "  /groovy/script/Script5E292565B0432F3C3737BD95DCE48EB3/_nf_config_5b6902a0: 7: Unexpected input: '{' @ line 7, column 9.\n",
      "     process {\n",
      "             ^\n",
      "  \n",
      "  1 error\n",
      "\n",
      "ch.artecat.grengine.except.CompileException: Compile failed for sources FixedSetSources[name='/groovy/script/Script5E292565B0432F3C3737BD95DCE48EB3/_nf_config_5b6902a0']. Cause: org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed:\n",
      "/groovy/script/Script5E292565B0432F3C3737BD95DCE48EB3/_nf_config_5b6902a0: 7: Unexpected input: '{' @ line 7, column 9.\n",
      "   process {\n",
      "           ^\n",
      "\n",
      "1 error\n",
      "\n",
      "\tat ch.artecat.grengine.code.groovy.DefaultGroovyCompiler.compile(DefaultGroovyCompiler.java:258)\n",
      "\tat ch.artecat.grengine.load.DefaultTopCodeCache.getUpToDateCode(DefaultTopCodeCache.java:80)\n",
      "\tat ch.artecat.grengine.load.LayeredClassLoader.getTopLoader(LayeredClassLoader.java:251)\n",
      "\tat ch.artecat.grengine.load.LayeredClassLoader.loadMainClass(LayeredClassLoader.java:202)\n",
      "\tat ch.artecat.grengine.engine.LayeredEngine.loadMainClass(LayeredEngine.java:198)\n",
      "\tat ch.artecat.grengine.Grengine.loadMainClass(Grengine.java:679)\n",
      "\tat ch.artecat.grengine.BaseGrengine.load(BaseGrengine.java:538)\n",
      "\tat ch.artecat.grengine.BaseGrengine.load(BaseGrengine.java:421)\n",
      "\tat ch.artecat.grengine.BaseGrengine$load.call(Unknown Source)\n",
      "\tat org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)\n",
      "\tat org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)\n",
      "\tat org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:148)\n",
      "\tat nextflow.config.ConfigParser.loadScript(ConfigParser.groovy:222)\n",
      "\tat java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n",
      "\tat java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)\n",
      "\tat java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\n",
      "\tat java.base/java.lang.reflect.Method.invoke(Method.java:566)\n",
      "\tat org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:48)\n",
      "\tat org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:189)\n",
      "\tat org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:57)\n",
      "\tat org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:51)\n",
      "\tat org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:171)\n",
      "\tat org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:185)\n",
      "\tat nextflow.config.ConfigParser.parse(ConfigParser.groovy:303)\n",
      "\tat nextflow.config.ConfigParser$parse$3.call(Unknown Source)\n",
      "\tat org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)\n",
      "\tat org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)\n",
      "\tat org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:139)\n",
      "\tat nextflow.config.ConfigBuilder.parse0(ConfigBuilder.groovy:434)\n",
      "\tat java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n",
      "\tat java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)\n",
      "\tat java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\n",
      "\tat java.base/java.lang.reflect.Method.invoke(Method.java:566)\n",
      "\tat org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:48)\n",
      "\tat org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:189)\n",
      "\tat org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:57)\n",
      "\tat org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:51)\n",
      "\tat org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:171)\n",
      "\tat org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:194)\n",
      "\tat nextflow.config.ConfigBuilder.merge0(ConfigBuilder.groovy:420)\n",
      "\tat java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n",
      "\tat java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)\n",
      "\tat java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\n",
      "\tat java.base/java.lang.reflect.Method.invoke(Method.java:566)\n",
      "\tat org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:48)\n",
      "\tat org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:189)\n",
      "\tat org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:57)\n",
      "\tat org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:51)\n",
      "\tat org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:171)\n",
      "\tat org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:203)\n",
      "\tat nextflow.config.ConfigBuilder.buildConfig0(ConfigBuilder.groovy:370)\n",
      "\tat java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n",
      "\tat java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)\n",
      "\tat java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\n",
      "\tat java.base/java.lang.reflect.Method.invoke(Method.java:566)\n",
      "\tat org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:48)\n",
      "\tat org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:189)\n",
      "\tat org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:57)\n",
      "\tat org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:51)\n",
      "\tat org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:171)\n",
      "\tat org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:194)\n",
      "\tat nextflow.config.ConfigBuilder.buildGivenFiles(ConfigBuilder.groovy:315)\n",
      "\tat java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n",
      "\tat java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)\n",
      "\tat java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\n",
      "\tat java.base/java.lang.reflect.Method.invoke(Method.java:566)\n",
      "\tat org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:48)\n",
      "\tat org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:189)\n",
      "\tat org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:57)\n",
      "\tat org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:51)\n",
      "\tat org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:171)\n",
      "\tat org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:185)\n",
      "\tat nextflow.config.ConfigBuilder.buildConfigObject(ConfigBuilder.groovy:806)\n",
      "\tat nextflow.config.ConfigBuilder.build(ConfigBuilder.groovy:819)\n",
      "\tat nextflow.cli.CmdRun.run(CmdRun.groovy:324)\n",
      "\tat nextflow.cli.Launcher.run(Launcher.groovy:500)\n",
      "\tat nextflow.cli.Launcher.main(Launcher.groovy:672)\n",
      "Caused by: org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed:\n",
      "/groovy/script/Script5E292565B0432F3C3737BD95DCE48EB3/_nf_config_5b6902a0: 7: Unexpected input: '{' @ line 7, column 9.\n",
      "   process {\n",
      "           ^\n",
      "\n",
      "1 error\n",
      "\n",
      "\tat org.codehaus.groovy.control.ErrorCollector.failIfErrors(ErrorCollector.java:292)\n",
      "\tat org.codehaus.groovy.control.ErrorCollector.addFatalError(ErrorCollector.java:148)\n",
      "\tat org.apache.groovy.parser.antlr4.AstBuilder.collectSyntaxError(AstBuilder.java:4321)\n",
      "\tat org.apache.groovy.parser.antlr4.AstBuilder.access$000(AstBuilder.java:176)\n",
      "\tat org.apache.groovy.parser.antlr4.AstBuilder$1.syntaxError(AstBuilder.java:4332)\n",
      "\tat groovyjarjarantlr4.v4.runtime.ProxyErrorListener.syntaxError(ProxyErrorListener.java:44)\n",
      "\tat groovyjarjarantlr4.v4.runtime.Parser.notifyErrorListeners(Parser.java:543)\n",
      "\tat groovyjarjarantlr4.v4.runtime.DefaultErrorStrategy.notifyErrorListeners(DefaultErrorStrategy.java:154)\n",
      "\tat org.apache.groovy.parser.antlr4.internal.DescriptiveErrorStrategy.reportInputMismatch(DescriptiveErrorStrategy.java:103)\n",
      "\tat org.apache.groovy.parser.antlr4.internal.DescriptiveErrorStrategy.recover(DescriptiveErrorStrategy.java:55)\n",
      "\tat org.apache.groovy.parser.antlr4.internal.DescriptiveErrorStrategy.recoverInline(DescriptiveErrorStrategy.java:68)\n",
      "\tat groovyjarjarantlr4.v4.runtime.Parser.match(Parser.java:213)\n",
      "\tat org.apache.groovy.parser.antlr4.GroovyParser.compilationUnit(GroovyParser.java:362)\n",
      "\tat org.apache.groovy.parser.antlr4.AstBuilder.buildCST(AstBuilder.java:250)\n",
      "\tat org.apache.groovy.parser.antlr4.AstBuilder.buildCST(AstBuilder.java:228)\n",
      "\tat org.apache.groovy.parser.antlr4.AstBuilder.buildAST(AstBuilder.java:269)\n",
      "\tat org.apache.groovy.parser.antlr4.Antlr4ParserPlugin.buildAST(Antlr4ParserPlugin.java:58)\n",
      "\tat org.codehaus.groovy.control.SourceUnit.buildAST(SourceUnit.java:256)\n",
      "\tat java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)\n",
      "\tat java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)\n",
      "\tat java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658)\n",
      "\tat org.codehaus.groovy.control.CompilationUnit.buildASTs(CompilationUnit.java:667)\n",
      "\tat org.codehaus.groovy.control.CompilationUnit.compile(CompilationUnit.java:633)\n",
      "\tat ch.artecat.grengine.code.groovy.DefaultGroovyCompiler.compile(DefaultGroovyCompiler.java:229)\n",
      "\t... 76 common frames omitted\n"
     ]
    }
   ],
   "source": [
    "!cat .nextflow.log "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2024-03-08 05:26:49.563275\n"
     ]
    }
   ],
   "source": [
    "import datetime\n",
    "now = datetime.datetime.now()\n",
    "print(now)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_vm.config -profile google_batch_profile -with-tower -resume \n",
    "# 8af1ba30-6606-4dfd-a851-c6d0b74bdd43\n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_vm.config -profile google_batch_profile -resume 886dc4ef-ff60-46d0-aecb-8f1ab56296a3\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_vm.config -profile google_batch_profile -resume\n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_vm.config -profile docker_local\n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_vm.config -profile google_batch_profile -resume\n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_vm.config -profile google_batch_profile -resume\n",
    "\n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_vm.config -profile docker_local -resume\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_vm.config -profile google_batch_profile -resume\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow.config -profile docker_profile -resume\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_vm.config -profile google_batch_profile -resume\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_local.config -profile google_batch_profile -resume\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow_vm.config -profile google_batch_profile -resume\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow.config -profile docker_profile -resume\n",
    "# !nextflow run nf_wf_fg_singlecell_rna.nf -c nextflow.config -profile google_batch_profile -resume"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !nextflow log\n",
    "# use the session id with the resume"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !cat .nextflow.log "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
