{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# start docker deamon in terminal\n",
    "# !sudo docker login\n",
    "# OR - becuase the jupyther with allow root\n",
    "# !docker login"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "TODO:\n",
    "\n",
    "when running with cloud. \n",
    "if you dont need to keep the execution logs, change the cloud settings to keep it for a short while\n",
    "Check that the cache if you need to set the cache to false (especially in plot process)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_joint\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/eila/git_igvf/atomic-workflows/nextflow/hai-gcp-genomic-beb9207a40b8.json\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "json_path = os.path.join(os.path.dirname(os.path.dirname(os.getcwd())),\"hai-gcp-genomic-beb9207a40b8.json\")\n",
    "!ls $json_path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Activated service account credentials for: [eila-batch@hai-gcp-genomic.iam.gserviceaccount.com]\n"
     ]
    }
   ],
   "source": [
    "# # in the command line. same env\n",
    "import os\n",
    "json_path = os.path.join(os.path.dirname(os.path.dirname(os.getcwd())),\"hai-gcp-genomic-beb9207a40b8.json\")\n",
    "os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = json_path\n",
    "!gcloud auth activate-service-account eila-batch@hai-gcp-genomic.iam.gserviceaccount.com --key-file=$GOOGLE_APPLICATION_CREDENTIALS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                  Credentialed Accounts\n",
      "ACTIVE  ACCOUNT\n",
      "        886987709752-compute@developer.gserviceaccount.com\n",
      "*       eila-batch@hai-gcp-genomic.iam.gserviceaccount.com\n",
      "        eila@stanford.edu\n",
      "\n",
      "To set the active account, run:\n",
      "    $ gcloud config set account `ACCOUNT`\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'hai-gcp-genomic'"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "!gcloud auth list\n",
    "project_name = !gcloud config get-value project\n",
    "project_name = project_name[0]\n",
    "project_name\n",
    "# gcloud config set account"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "os.environ[\"TOWER_ACCESS_TOKEN\"] = 'eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz'\n",
    "os.environ[\"TOWER_WORKSPACE_ID\"]='247507587970818'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz\n",
      "247507587970818\n"
     ]
    }
   ],
   "source": [
    "!echo $TOWER_ACCESS_TOKEN\n",
    "!echo $TOWER_WORKSPACE_ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_joint/bin'"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os\n",
    "bin_fld = os.path.join(os.getcwd(),\"bin\")\n",
    "bin_fld"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "!cp /home/eila/git_igvf/atomic-workflows/bin_files/* /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_joint/bin/\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "!chmod +x {os.path.join(bin_fld, \"joint_cell_plotting.py\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"joint_cell_plotting_density.R\")}\n",
    "!chmod +x {os.path.join(bin_fld, \"write_html.py\")}\n",
    "# !chmod +x {os.path.join(bin_fld, \"create_list.sh\")}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# in the command line. based on:\n",
    "# https://nextflow.io/blog/2023/nextflow-with-gbatch.html\n",
    "# gcloud services enable batch.googleapis.com compute.googleapis.com logging.googleapis.com\n",
    "# gcloud iam service-accounts create eila-batch\n",
    "# gcloud iam service-accounts list \n",
    "\n",
    "\n",
    "# gcloud projects add-iam-policy-binding terra-addd4c49 \\\n",
    "# --member=\"serviceAccount:pet-26869426946125b815f6b@terra-addd4c49.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/iam.serviceAccountUser\"\n",
    "\n",
    "# $ gcloud projects add-iam-policy-binding terra-addd4c49 \\\n",
    "# --member=\"serviceAccount:pet-26869426946125b815f6b@terra-addd4c49.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/batch.jobsEditor\"\n",
    "\n",
    "# $ gcloud projects add-iam-policy-binding terra-addd4c49 \\\n",
    "# --member=\"serviceAccount:pet-26869426946125b815f6b@terra-addd4c49.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/logging.viewer\"\n",
    "\n",
    "# $ gcloud projects add-iam-policy-binding terra-addd4c49 \\\n",
    "# --member=\"serviceAccount:pet-26869426946125b815f6b@terra-addd4c49.iam.gserviceaccount.com\" \\\n",
    "# --role=\"roles/storage.admin\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !nextflow -self-update"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  Version: 23.10.1 build 5891\n",
      "  Created: 12-01-2024 22:01 UTC \n",
      "  System: Linux 5.10.0-28-cloud-amd64\n",
      "  Runtime: Groovy 3.0.19 on OpenJDK 64-Bit Server VM 11.0.22+7-post-Debian-1deb11u1\n",
      "  Encoding: UTF-8 (UTF-8)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!nextflow info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/eila/git_igvf/atomic-workflows/nextflow/nf_data\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "\n",
    "# Get the current working directory\n",
    "current_directory = os.getcwd()\n",
    "\n",
    "# Navigate two folders above\n",
    "two_folders_above = os.path.abspath(os.path.join(current_directory, '..', '..'))\n",
    "\n",
    "nf_data_base_path = os.path.join(two_folders_above,'nf_data')\n",
    "# Print the result\n",
    "print(nf_data_base_path)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/eila/git_igvf/atomic-workflows/nextflow/nf_data'"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nf_data_base_path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# nf_data_project=\"test_atac_pipeline\"\n",
    "nf_data_project=\"test_joint_pipeline\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "!cp ../../config_files/docker-google-batch.config ../workflow_fg_single_cell_joint/conf/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# Write the configuration file\n",
    "with open(\"nextflow_joint.config\", \"w\") as config_file:\n",
    "    config_file.write(f\"\"\"\n",
    "manifest {{\n",
    "  description = 'A nexflow pipeline for fg single cell atac' \n",
    "  mainScript = 'nf_wf_fg_singlecell_joint.nf'\n",
    "}}\n",
    "\n",
    "profiles {{\n",
    "  google_batch_profile {{\n",
    "    workDir = 'gs://eila-nextflow-bucket/work/JOINT'\n",
    "    includeConfig 'conf/docker-google-batch.config'\n",
    "    docker.enabled = true\n",
    "    wave.enabled = true\n",
    "    fusion.enabled = true\n",
    "    process.scratch = false\n",
    "    executor = 'google-batch'\n",
    "    docker.envWhitelist=['ENV_SYNAPSE_TOKEN','TOWER_ACCESS_TOKEN']\n",
    "    google.project = '{project_name}'\n",
    "    google.location = 'us-central1'\n",
    "    google.batch.spot = true\n",
    "    google.batch.serviceAccountEmail='eila-batch@hai-gcp-genomic.iam.gserviceaccount.com'\n",
    "  }}\n",
    "}}\n",
    "\n",
    "// Define ENV_SYNAPSE_TOKEN as a parameter\n",
    "params.SC_JOINT_CELL_BARCODE_PLOT_SCRIPT_PY='joint_cell_plotting.py'\n",
    "\n",
    "//Minimum number of UMIs/fragments required for a cell to be plotted\n",
    "params.SC_JOINT_CELL_BARCODE_PLOT_REMOVE_LOW_YIELDING_CELLS=10\n",
    "params.SC_JOINT_CELL_BARCODE_PLOT_METADATA_FILES='{nf_data_base_path}/{nf_data_project}/vm_joint_files.tsv'\n",
    "params.SC_JOINT_CELL_BARCODE_PLOT_MIN_UMIS=100\n",
    "params.SC_JOINT_CELL_BARCODE_PLOT_MIN_GENES=200\n",
    "params.SC_JOINT_CELL_BARCODE_PLOT_MIN_TSS=4\n",
    "\n",
    "//Cutoff for minimum number of ATAC fragments\n",
    "params.SC_JOINT_CELL_BARCODE_PLOT_MIN_FRAGS=100\n",
    "params.SC_JOINT_CELL_BARCODE_PLOT_PKR='BMMC-single-donor'\n",
    "params.SC_JOINT_CELL_BARCODE_PLOT_SCRIPT_R='joint_cell_plotting_density.R'\n",
    "\n",
    "//params.SC_JOINT_CELL_BARCODE_PLOT_HTML_LIST_SCRIPT='create_list.sh'\n",
    "params.SC_JOINT_CELL_BARCODE_PLOT_HTML_SCRIPT='write_html.py'\n",
    "params.SC_JOINT_CELL_BARCODE_PLOT_HTML_FILES='{nf_data_base_path}/{nf_data_project}/vm_joint_html.tsv'\n",
    "\n",
    "\n",
    "\n",
    "\"\"\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_joint\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !rm -r /Users/eilaarich-landkof-stanford/Documents/Code/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_joint/work\n",
    "# !rm -r /home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_workflow/workflow_fg_single_cell_joint/work\n",
    "# !gsutil rm -r gs://eila-nextflow-bucket/work/JOINT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !rm .nextflow*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# !ls nextflow.config"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "N E X T F L O W  ~  version 23.10.1\n",
      "Launching `nf_wf_fg_singlecell_joint.nf` [kickass_hugle] DSL2 - revision: cc5ee23a93\n",
      "Before calling run_joint_barcode_cell_plotting\n",
      "[-        ] process > run_joint_barcode_cell_plot... -\u001b[K\n",
      "[-        ] process > run_joint_cell_plotting_den... -\u001b[K\n",
      "[-        ] process > run_joint_metrics_html_report  -\u001b[K\n",
      "After calling run_joint_barcode_cell_plotting\u001b[K\n",
      "Before calling run_joint_cell_plotting_density\u001b[K\n",
      "After calling run_joint_cell_plotting_density\u001b[K\n",
      "Before run_joint_metrics_html_report\u001b[K\n",
      "After run_joint_metrics_html_report\u001b[K\n",
      "\u001b[9A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2THExQVsV0zhfU\u001b[22;39m\u001b[K\n",
      "[-        ] process > run_joint_barcode_cell_plot... -\u001b[K\n",
      "[-        ] process > run_joint_cell_plotting_den... -\u001b[K\n",
      "[-        ] process > run_joint_metrics_html_report  -\u001b[K\n",
      "After calling run_joint_barcode_cell_plotting\u001b[K\n",
      "Before calling run_joint_cell_plotting_density\u001b[K\n",
      "After calling run_joint_cell_plotting_density\u001b[K\n",
      "Before run_joint_metrics_html_report\u001b[K\n",
      "After run_joint_metrics_html_report\u001b[K\n",
      "\u001b[10A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2THExQVsV0zhfU\u001b[22;39m\u001b[K\n",
      "[-        ] process > run_joint_barcode_cell_plot... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_joint_cell_plotting_den... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_joint_metrics_html_report  -\u001b[K\n",
      "After calling run_joint_barcode_cell_plotting\u001b[K\n",
      "Before calling run_joint_cell_plotting_density\u001b[K\n",
      "After calling run_joint_cell_plotting_density\u001b[K\n",
      "Before run_joint_metrics_html_report\u001b[K\n",
      "After run_joint_metrics_html_report\u001b[K\n",
      "\u001b[10A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2THExQVsV0zhfU\u001b[22;39m\u001b[K\n",
      "[-        ] process > run_joint_barcode_cell_plot... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_joint_cell_plotting_den... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_joint_metrics_html_report  [  0%] 0 of 1\u001b[K\n",
      "After calling run_joint_barcode_cell_plotting\u001b[K\n",
      "Before calling run_joint_cell_plotting_density\u001b[K\n",
      "After calling run_joint_cell_plotting_density\u001b[K\n",
      "Before run_joint_metrics_html_report\u001b[K\n",
      "After run_joint_metrics_html_report\u001b[K\n",
      "\u001b[10A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2THExQVsV0zhfU\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (1)\u001b[K\n",
      "[-        ] process > run_joint_barcode_cell_plot... [  0%] 0 of 1\u001b[K\n",
      "[ab/5a4ffe] process > run_joint_cell_plotting_den... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_joint_metrics_html_report  [  0%] 0 of 1\u001b[K\n",
      "After calling run_joint_barcode_cell_plotting\u001b[K\n",
      "Before calling run_joint_cell_plotting_density\u001b[K\n",
      "After calling run_joint_cell_plotting_density\u001b[K\n",
      "Before run_joint_metrics_html_report\u001b[K\n",
      "After run_joint_metrics_html_report\u001b[K\n",
      "\u001b[11A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2THExQVsV0zhfU\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (2)\u001b[K\n",
      "[ec/97ff9f] process > run_joint_barcode_cell_plot... [  0%] 0 of 1\u001b[K\n",
      "[ab/5a4ffe] process > run_joint_cell_plotting_den... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > run_joint_metrics_html_report  [  0%] 0 of 1\u001b[K\n",
      "After calling run_joint_barcode_cell_plotting\u001b[K\n",
      "Before calling run_joint_cell_plotting_density\u001b[K\n",
      "After calling run_joint_cell_plotting_density\u001b[K\n",
      "Before run_joint_metrics_html_report\u001b[K\n",
      "After run_joint_metrics_html_report\u001b[K\n",
      "\u001b[11A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2THExQVsV0zhfU\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (3)\u001b[K\n",
      "[ec/97ff9f] process > run_joint_barcode_cell_plot... [  0%] 0 of 1\u001b[K\n",
      "[ab/5a4ffe] process > run_joint_cell_plotting_den... [  0%] 0 of 1\u001b[K\n",
      "[c8/8eb037] process > run_joint_metrics_html_repo... [  0%] 0 of 1\u001b[K\n",
      "After calling run_joint_barcode_cell_plotting\u001b[K\n",
      "Before calling run_joint_cell_plotting_density\u001b[K\n",
      "After calling run_joint_cell_plotting_density\u001b[K\n",
      "Before run_joint_metrics_html_report\u001b[K\n",
      "After run_joint_metrics_html_report\u001b[K\n",
      "\u001b[11A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2THExQVsV0zhfU\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (3)\u001b[K\n",
      "[ec/97ff9f] process > run_joint_barcode_cell_plot... [  0%] 0 of 1\u001b[K\n",
      "[ab/5a4ffe] process > run_joint_cell_plotting_den... [  0%] 0 of 1\u001b[K\n",
      "[c8/8eb037] process > run_joint_metrics_html_repo... [  0%] 0 of 1\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[11A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2THExQVsV0zhfU\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (3)\u001b[K\n",
      "[ec/97ff9f] process > run_joint_barcode_cell_plot... [  0%] 0 of 1\u001b[K\n",
      "[ab/5a4ffe] process > run_joint_cell_plotting_den... [  0%] 0 of 1\u001b[K\n",
      "[c8/8eb037] process > run_joint_metrics_html_repo... [  0%] 0 of 1 ✔\u001b[K\n",
      "Starting run_joint_metrics_html_report...\u001b[K\n",
      "Script path: /usr/local/bin/write_html.py\u001b[K\n",
      "RNA Metrics: BMMC-single-donor.rna.log.hg38.txt\u001b[K\n",
      "ATAC Metrics: BMMC-single-donor_qc_metrics.csv\u001b[K\n",
      "image_files: is BMMC-single-donor.hg38.joint.qc.plot.png BMMC-single-donor.hg38.joint.density.plot.png BMMC-single-donor.qc.rna.hg38.umi.barcode.rank.plot.png BMMC-single-donor.qc.rna.hg38.gene.barcode.rank.plot.png BMMC-single-donor.qc.rna.hg38.gene.umi.scatter.plot.png BMMC-single-donor.atac.qc.hist.hg38.log.png BMMC-single-donor.atac.qc.hg38.tss_enrichment_bulk.png\u001b[K\n",
      "Content of image_files.txt:\u001b[K\n",
      "BMMC-single-donor.hg38.joint.qc.plot.png\u001b[K\n",
      "BMMC-single-donor.hg38.joint.density.plot.png\u001b[K\n",
      "BMMC-single-donor.qc.rna.hg38.umi.barcode.rank.plot.png\u001b[K\n",
      "BMMC-single-donor.qc.rna.hg38.gene.barcode.rank.plot.png\u001b[K\n",
      "BMMC-single-donor.qc.rna.hg38.gene.umi.scatter.plot.png\u001b[K\n",
      "BMMC-single-donor.atac.qc.hist.hg38.log.png\u001b[K\n",
      "BMMC-single-donor.atac.qc.hg38.tss_enrichment_bulk.png\u001b[K\n",
      "start of printing summary stats loop\u001b[K\n",
      "stat is Number_of_reads,770560126\u001b[K\n",
      "stat is Number_of_mapped_reads,717898494\u001b[K\n",
      "stat is Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "stat is Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "stat is Number_of_candidates,3413147335\u001b[K\n",
      "stat is Number_of_mappings,717898494\u001b[K\n",
      "stat is Number_of_uni-mappings,660005178\u001b[K\n",
      "stat is Number_of_multi-mappings,57893316\u001b[K\n",
      "stat is Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "stat is Number_of_corrected_barcodes,0\u001b[K\n",
      "stat is Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "stat is uni-mappings,260650161\u001b[K\n",
      "stat is multi-mappings,14666503\u001b[K\n",
      "stat is total,275316664\u001b[K\n",
      "stat is percentage_duplicates, 15.7\u001b[K\n",
      "stat is n_targets,0\u001b[K\n",
      "stat is n_bootstraps,0\u001b[K\n",
      "stat is n_processed,755403267\u001b[K\n",
      "stat is n_pseudoaligned,261490978\u001b[K\n",
      "stat is n_unique,109691397\u001b[K\n",
      "stat is p_pseudoaligned,34.6\u001b[K\n",
      "stat is p_unique,14.5\u001b[K\n",
      "stat is numRecords,170631458\u001b[K\n",
      "stat is numReads,270162228\u001b[K\n",
      "stat is numBarcodes,6898982\u001b[K\n",
      "stat is medianReadsPerBarcode,11\u001b[K\n",
      "stat is meanReadsPerBarcode,39.159724\u001b[K\n",
      "stat is numUMIs,1007199\u001b[K\n",
      "stat is numBarcodeUMIs,155719426\u001b[K\n",
      "stat is medianUMIsPerBarcode,8\u001b[K\n",
      "stat is meanUMIsPerBarcode,22.571363\u001b[K\n",
      "stat is gtRecords,102490306\u001b[K\n",
      "stat is numBarcodesOnWhitelist,6898982\u001b[K\n",
      "stat is percentageBarcodesOnWhitelist,100\u001b[K\n",
      "stat is numReadsOnWhitelist,270162228\u001b[K\n",
      "stat is percentageReadsOnWhitelist,100\u001b[K\n",
      "start of printing logs loop\u001b[K\n",
      "start of qc vals text loop\u001b[K\n",
      "Finished run_joint_metrics_html_report.\u001b[K\n",
      "\u001b[K\n",
      "\u001b[60A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2THExQVsV0zhfU\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (3)\u001b[K\n",
      "[ec/97ff9f] process > run_joint_barcode_cell_plot... [  0%] 0 of 1\u001b[K\n",
      "[ab/5a4ffe] process > run_joint_cell_plotting_den... [  0%] 0 of 1\u001b[K\n",
      "[c8/8eb037] process > run_joint_metrics_html_repo... [100%] 1 of 1 ✔\u001b[K\n",
      "Starting run_joint_metrics_html_report...\u001b[K\n",
      "Script path: /usr/local/bin/write_html.py\u001b[K\n",
      "RNA Metrics: BMMC-single-donor.rna.log.hg38.txt\u001b[K\n",
      "ATAC Metrics: BMMC-single-donor_qc_metrics.csv\u001b[K\n",
      "image_files: is BMMC-single-donor.hg38.joint.qc.plot.png BMMC-single-donor.hg38.joint.density.plot.png BMMC-single-donor.qc.rna.hg38.umi.barcode.rank.plot.png BMMC-single-donor.qc.rna.hg38.gene.barcode.rank.plot.png BMMC-single-donor.qc.rna.hg38.gene.umi.scatter.plot.png BMMC-single-donor.atac.qc.hist.hg38.log.png BMMC-single-donor.atac.qc.hg38.tss_enrichment_bulk.png\u001b[K\n",
      "Content of image_files.txt:\u001b[K\n",
      "BMMC-single-donor.hg38.joint.qc.plot.png\u001b[K\n",
      "BMMC-single-donor.hg38.joint.density.plot.png\u001b[K\n",
      "BMMC-single-donor.qc.rna.hg38.umi.barcode.rank.plot.png\u001b[K\n",
      "BMMC-single-donor.qc.rna.hg38.gene.barcode.rank.plot.png\u001b[K\n",
      "BMMC-single-donor.qc.rna.hg38.gene.umi.scatter.plot.png\u001b[K\n",
      "BMMC-single-donor.atac.qc.hist.hg38.log.png\u001b[K\n",
      "BMMC-single-donor.atac.qc.hg38.tss_enrichment_bulk.png\u001b[K\n",
      "start of printing summary stats loop\u001b[K\n",
      "stat is Number_of_reads,770560126\u001b[K\n",
      "stat is Number_of_mapped_reads,717898494\u001b[K\n",
      "stat is Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "stat is Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "stat is Number_of_candidates,3413147335\u001b[K\n",
      "stat is Number_of_mappings,717898494\u001b[K\n",
      "stat is Number_of_uni-mappings,660005178\u001b[K\n",
      "stat is Number_of_multi-mappings,57893316\u001b[K\n",
      "stat is Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "stat is Number_of_corrected_barcodes,0\u001b[K\n",
      "stat is Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "stat is uni-mappings,260650161\u001b[K\n",
      "stat is multi-mappings,14666503\u001b[K\n",
      "stat is total,275316664\u001b[K\n",
      "stat is percentage_duplicates, 15.7\u001b[K\n",
      "stat is n_targets,0\u001b[K\n",
      "stat is n_bootstraps,0\u001b[K\n",
      "stat is n_processed,755403267\u001b[K\n",
      "stat is n_pseudoaligned,261490978\u001b[K\n",
      "stat is n_unique,109691397\u001b[K\n",
      "stat is p_pseudoaligned,34.6\u001b[K\n",
      "stat is p_unique,14.5\u001b[K\n",
      "stat is numRecords,170631458\u001b[K\n",
      "stat is numReads,270162228\u001b[K\n",
      "stat is numBarcodes,6898982\u001b[K\n",
      "stat is medianReadsPerBarcode,11\u001b[K\n",
      "stat is meanReadsPerBarcode,39.159724\u001b[K\n",
      "stat is numUMIs,1007199\u001b[K\n",
      "stat is numBarcodeUMIs,155719426\u001b[K\n",
      "stat is medianUMIsPerBarcode,8\u001b[K\n",
      "stat is meanUMIsPerBarcode,22.571363\u001b[K\n",
      "stat is gtRecords,102490306\u001b[K\n",
      "stat is numBarcodesOnWhitelist,6898982\u001b[K\n",
      "stat is percentageBarcodesOnWhitelist,100\u001b[K\n",
      "stat is numReadsOnWhitelist,270162228\u001b[K\n",
      "stat is percentageReadsOnWhitelist,100\u001b[K\n",
      "start of printing logs loop\u001b[K\n",
      "start of qc vals text loop\u001b[K\n",
      "Finished run_joint_metrics_html_report.\u001b[K\n",
      "\u001b[K\n",
      "\u001b[60A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2THExQVsV0zhfU\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (3)\u001b[K\n",
      "[ec/97ff9f] process > run_joint_barcode_cell_plot... [  0%] 0 of 1\u001b[K\n",
      "[ab/5a4ffe] process > run_joint_cell_plotting_den... [  0%] 0 of 1\u001b[K\n",
      "[c8/8eb037] process > run_joint_metrics_html_repo... [100%] 1 of 1 ✔\u001b[K\n",
      "Starting run_joint_metrics_html_report...\u001b[K\n",
      "Script path: /usr/local/bin/write_html.py\u001b[K\n",
      "RNA Metrics: BMMC-single-donor.rna.log.hg38.txt\u001b[K\n",
      "ATAC Metrics: BMMC-single-donor_qc_metrics.csv\u001b[K\n",
      "image_files: is BMMC-single-donor.hg38.joint.qc.plot.png BMMC-single-donor.hg38.joint.density.plot.png BMMC-single-donor.qc.rna.hg38.umi.barcode.rank.plot.png BMMC-single-donor.qc.rna.hg38.gene.barcode.rank.plot.png BMMC-single-donor.qc.rna.hg38.gene.umi.scatter.plot.png BMMC-single-donor.atac.qc.hist.hg38.log.png BMMC-single-donor.atac.qc.hg38.tss_enrichment_bulk.png\u001b[K\n",
      "Content of image_files.txt:\u001b[K\n",
      "BMMC-single-donor.hg38.joint.qc.plot.png\u001b[K\n",
      "BMMC-single-donor.hg38.joint.density.plot.png\u001b[K\n",
      "BMMC-single-donor.qc.rna.hg38.umi.barcode.rank.plot.png\u001b[K\n",
      "BMMC-single-donor.qc.rna.hg38.gene.barcode.rank.plot.png\u001b[K\n",
      "BMMC-single-donor.qc.rna.hg38.gene.umi.scatter.plot.png\u001b[K\n",
      "BMMC-single-donor.atac.qc.hist.hg38.log.png\u001b[K\n",
      "BMMC-single-donor.atac.qc.hg38.tss_enrichment_bulk.png\u001b[K\n",
      "start of printing summary stats loop\u001b[K\n",
      "stat is Number_of_reads,770560126\u001b[K\n",
      "stat is Number_of_mapped_reads,717898494\u001b[K\n",
      "stat is Number_of_uniquely_mapped_reads,660005178\u001b[K\n",
      "stat is Number_of_reads_have_multi-mappings,57893316\u001b[K\n",
      "stat is Number_of_candidates,3413147335\u001b[K\n",
      "stat is Number_of_mappings,717898494\u001b[K\n",
      "stat is Number_of_uni-mappings,660005178\u001b[K\n",
      "stat is Number_of_multi-mappings,57893316\u001b[K\n",
      "stat is Number_of_barcodes_in_whitelist,385280063\u001b[K\n",
      "stat is Number_of_corrected_barcodes,0\u001b[K\n",
      "stat is Number_of_output_mappings_(passed_filters),275316664\u001b[K\n",
      "stat is uni-mappings,260650161\u001b[K\n",
      "stat is multi-mappings,14666503\u001b[K\n",
      "stat is total,275316664\u001b[K\n",
      "stat is percentage_duplicates, 15.7\u001b[K\n",
      "stat is n_targets,0\u001b[K\n",
      "stat is n_bootstraps,0\u001b[K\n",
      "stat is n_processed,755403267\u001b[K\n",
      "stat is n_pseudoaligned,261490978\u001b[K\n",
      "stat is n_unique,109691397\u001b[K\n",
      "stat is p_pseudoaligned,34.6\u001b[K\n",
      "stat is p_unique,14.5\u001b[K\n",
      "stat is numRecords,170631458\u001b[K\n",
      "stat is numReads,270162228\u001b[K\n",
      "stat is numBarcodes,6898982\u001b[K\n",
      "stat is medianReadsPerBarcode,11\u001b[K\n",
      "stat is meanReadsPerBarcode,39.159724\u001b[K\n",
      "stat is numUMIs,1007199\u001b[K\n",
      "stat is numBarcodeUMIs,155719426\u001b[K\n",
      "stat is medianUMIsPerBarcode,8\u001b[K\n",
      "stat is meanUMIsPerBarcode,22.571363\u001b[K\n",
      "stat is gtRecords,102490306\u001b[K\n",
      "stat is numBarcodesOnWhitelist,6898982\u001b[K\n",
      "stat is percentageBarcodesOnWhitelist,100\u001b[K\n",
      "stat is numReadsOnWhitelist,270162228\u001b[K\n",
      "stat is percentageReadsOnWhitelist,100\u001b[K\n",
      "start of printing logs loop\u001b[K\n",
      "start of qc vals text loop\u001b[K\n",
      "Finished run_joint_metrics_html_report.\u001b[K\n",
      "\u001b[K\n",
      "\u001b[60A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2THExQVsV0zhfU\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (3)\u001b[K\n",
      "[ec/97ff9f] process > run_joint_barcode_cell_plot... [  0%] 0 of 1\u001b[K\n",
      "[ab/5a4ffe] process > run_joint_cell_plotting_den... [  0%] 0 of 1\u001b[K\n",
      "[c8/8eb037] process > run_joint_metrics_html_repo... [100%] 1 of 1 ✔\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[60A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2THExQVsV0zhfU\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (3)\u001b[K\n",
      "[ec/97ff9f] process > run_joint_barcode_cell_plot... [  0%] 0 of 1\u001b[K\n",
      "[ab/5a4ffe] process > run_joint_cell_plotting_den... [  0%] 0 of 1 ✔\u001b[K\n",
      "[c8/8eb037] process > run_joint_metrics_html_repo... [100%] 1 of 1 ✔\u001b[K\n",
      "Starting run_joint_cell_plotting_density...\u001b[K\n",
      "Script path: /usr/local/bin/joint_cell_plotting_density.R\u001b[K\n",
      "Running R script...\u001b[K\n",
      "[1] \"All arguments: /opt/conda/envs/joint_plot_r_env/lib/R/bin/exec/R --no-echo --no-restore --file=/usr/local/bin/joint_cell_plotting_density.R --args BMMC-single-donor BMMC-single-donor.joint.barcode.metadata.hg38.csv joint_density_plot.png\"\u001b[K\n",
      "[1] \"pkr: BMMC-single-donor\"\u001b[K\n",
      "[1] \"barcode_metadata_file: BMMC-single-donor.joint.barcode.metadata.hg38.csv\"\u001b[K\n",
      "[1] \"plot_file: joint_density_plot.png\"\u001b[K\n",
      "[1] \"TODO: uncomment generate_density_plot when I have real data\"\u001b[K\n",
      "null device \u001b[K\n",
      "          1 \u001b[K\n",
      "Finished run_joint_cell_plotting_density.\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[60A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2THExQVsV0zhfU\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (3)\u001b[K\n",
      "[ec/97ff9f] process > run_joint_barcode_cell_plot... [  0%] 0 of 1\u001b[K\n",
      "[ab/5a4ffe] process > run_joint_cell_plotting_den... [100%] 1 of 1 ✔\u001b[K\n",
      "[c8/8eb037] process > run_joint_metrics_html_repo... [100%] 1 of 1 ✔\u001b[K\n",
      "Starting run_joint_cell_plotting_density...\u001b[K\n",
      "Script path: /usr/local/bin/joint_cell_plotting_density.R\u001b[K\n",
      "Running R script...\u001b[K\n",
      "[1] \"All arguments: /opt/conda/envs/joint_plot_r_env/lib/R/bin/exec/R --no-echo --no-restore --file=/usr/local/bin/joint_cell_plotting_density.R --args BMMC-single-donor BMMC-single-donor.joint.barcode.metadata.hg38.csv joint_density_plot.png\"\u001b[K\n",
      "[1] \"pkr: BMMC-single-donor\"\u001b[K\n",
      "[1] \"barcode_metadata_file: BMMC-single-donor.joint.barcode.metadata.hg38.csv\"\u001b[K\n",
      "[1] \"plot_file: joint_density_plot.png\"\u001b[K\n",
      "[1] \"TODO: uncomment generate_density_plot when I have real data\"\u001b[K\n",
      "null device \u001b[K\n",
      "          1 \u001b[K\n",
      "Finished run_joint_cell_plotting_density.\u001b[K\n",
      "\u001b[K\n",
      "\u001b[18A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2THExQVsV0zhfU\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (3)\u001b[K\n",
      "[ec/97ff9f] process > run_joint_barcode_cell_plot... [  0%] 0 of 1 ✔\u001b[K\n",
      "[ab/5a4ffe] process > run_joint_cell_plotting_den... [100%] 1 of 1 ✔\u001b[K\n",
      "[c8/8eb037] process > run_joint_metrics_html_repo... [100%] 1 of 1 ✔\u001b[K\n",
      "Starting run_joint_barcode_cell_plotting...\u001b[K\n",
      "Script path: /usr/local/bin/joint_cell_plotting.py\u001b[K\n",
      "Parameters:\u001b[K\n",
      "- pkr: BMMC-single-donor\u001b[K\n",
      "- rna_metrics_file: BMMC-single-donor.qc.rna.hg38.barcode.metadata.tsv\u001b[K\n",
      "- atac_metrics_file: BMMC-single-donor.atac.qc.hg38.tss_enrichment_barcode_stats.tsv\u001b[K\n",
      "- remove_low_yielding_cells: 10\u001b[K\n",
      "- barcode_metadata_file: BMMC-single-donor.joint.barcode.metadata.hg38.csv\u001b[K\n",
      "- min_umis: 100\u001b[K\n",
      "- min_genes: 200\u001b[K\n",
      "- min_tss: 4\u001b[K\n",
      "- min_frags: 100\u001b[K\n",
      "Finished run_joint_barcode_cell_plotting.\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[32A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2THExQVsV0zhfU\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (3)\u001b[K\n",
      "[ec/97ff9f] process > run_joint_barcode_cell_plot... [100%] 1 of 1 ✔\u001b[K\n",
      "[ab/5a4ffe] process > run_joint_cell_plotting_den... [100%] 1 of 1 ✔\u001b[K\n",
      "[c8/8eb037] process > run_joint_metrics_html_repo... [100%] 1 of 1 ✔\u001b[K\n",
      "Starting run_joint_barcode_cell_plotting...\u001b[K\n",
      "Script path: /usr/local/bin/joint_cell_plotting.py\u001b[K\n",
      "Parameters:\u001b[K\n",
      "- pkr: BMMC-single-donor\u001b[K\n",
      "- rna_metrics_file: BMMC-single-donor.qc.rna.hg38.barcode.metadata.tsv\u001b[K\n",
      "- atac_metrics_file: BMMC-single-donor.atac.qc.hg38.tss_enrichment_barcode_stats.tsv\u001b[K\n",
      "- remove_low_yielding_cells: 10\u001b[K\n",
      "- barcode_metadata_file: BMMC-single-donor.joint.barcode.metadata.hg38.csv\u001b[K\n",
      "- min_umis: 100\u001b[K\n",
      "- min_genes: 200\u001b[K\n",
      "- min_tss: 4\u001b[K\n",
      "- min_frags: 100\u001b[K\n",
      "Finished run_joint_barcode_cell_plotting.\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "\u001b[32A\n",
      "\u001b[32;1mMonitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2THExQVsV0zhfU\u001b[22;39m\u001b[K\n",
      "executor >  google-batch (fusion enabled) (3)\u001b[K\n",
      "[ec/97ff9f] process > run_joint_barcode_cell_plot... [100%] 1 of 1 ✔\u001b[K\n",
      "[ab/5a4ffe] process > run_joint_cell_plotting_den... [100%] 1 of 1 ✔\u001b[K\n",
      "[c8/8eb037] process > run_joint_metrics_html_repo... [100%] 1 of 1 ✔\u001b[K\n",
      "Starting run_joint_barcode_cell_plotting...\u001b[K\n",
      "Script path: /usr/local/bin/joint_cell_plotting.py\u001b[K\n",
      "Parameters:\u001b[K\n",
      "- pkr: BMMC-single-donor\u001b[K\n",
      "- rna_metrics_file: BMMC-single-donor.qc.rna.hg38.barcode.metadata.tsv\u001b[K\n",
      "- atac_metrics_file: BMMC-single-donor.atac.qc.hg38.tss_enrichment_barcode_stats.tsv\u001b[K\n",
      "- remove_low_yielding_cells: 10\u001b[K\n",
      "- barcode_metadata_file: BMMC-single-donor.joint.barcode.metadata.hg38.csv\u001b[K\n",
      "- min_umis: 100\u001b[K\n",
      "- min_genes: 200\u001b[K\n",
      "- min_tss: 4\u001b[K\n",
      "- min_frags: 100\u001b[K\n",
      "Finished run_joint_barcode_cell_plotting.\u001b[K\n",
      "\u001b[K\n",
      "\u001b[32;1mCompleted at: 02-Apr-2024 03:10:07\n",
      "Duration    : 5m 33s\n",
      "CPU hours   : 0.1\n",
      "Succeeded   : 3\n",
      "\u001b[22;39m\u001b[K\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!nextflow run nf_wf_fg_singlecell_joint.nf -c nextflow_joint.config -profile google_batch_profile -with-tower -resume \n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_joint.nf -c nextflow_joint.config -profile google_batch_profile -with-tower -resume\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_vm.config -profile google_batch_profile\n",
    "\n",
    "\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_vm.config -profile google_batch_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_vm_anvil.config -profile google_batch_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_local.config -profile google_batch_profile\n",
    "# !nextflow run nf_wf_fg_singlecell_atac.nf -c nextflow_local.config -profile docker_profile\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Apr-02 03:04:30.600 [main] DEBUG nextflow.cli.Launcher - $> nextflow run nf_wf_fg_singlecell_joint.nf -c nextflow_joint.config -profile google_batch_profile -with-tower -resume\n",
      "Apr-02 03:04:30.848 [main] INFO  nextflow.cli.CmdRun - N E X T F L O W  ~  version 23.10.1\n",
      "Apr-02 03:04:30.883 [main] DEBUG nextflow.plugin.PluginsFacade - Setting up plugin manager > mode=prod; embedded=false; plugins-dir=/home/eila/.nextflow/plugins; core-plugins: nf-amazon@2.1.4,nf-azure@1.3.3,nf-cloudcache@0.3.0,nf-codecommit@0.1.5,nf-console@1.0.6,nf-ga4gh@1.1.0,nf-google@1.8.3,nf-tower@1.6.3,nf-wave@1.0.1\n",
      "Apr-02 03:04:30.900 [main] INFO  o.pf4j.DefaultPluginStatusProvider - Enabled plugins: []\n",
      "Apr-02 03:04:30.902 [main] INFO  o.pf4j.DefaultPluginStatusProvider - Disabled plugins: []\n",
      "Apr-02 03:04:30.908 [main] INFO  org.pf4j.DefaultPluginManager - PF4J version 3.4.1 in 'deployment' mode\n",
      "Apr-02 03:04:30.925 [main] INFO  org.pf4j.AbstractPluginManager - No plugins\n",
      "Apr-02 03:04:30.953 [main] DEBUG nextflow.config.ConfigBuilder - User config file: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_joint/nextflow_joint.config\n",
      "Apr-02 03:04:30.954 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_joint/nextflow_joint.config\n",
      "Apr-02 03:04:31.025 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `google_batch_profile`\n",
      "Apr-02 03:04:32.790 [main] DEBUG nextflow.config.ConfigBuilder - Available config profiles: [google_batch_profile]\n",
      "Apr-02 03:04:32.850 [main] DEBUG nextflow.cli.CmdRun - Applied DSL=2 by global default\n",
      "Apr-02 03:04:32.882 [main] INFO  nextflow.cli.CmdRun - Launching `nf_wf_fg_singlecell_joint.nf` [kickass_hugle] DSL2 - revision: cc5ee23a93\n",
      "Apr-02 03:04:32.883 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins default=[nf-google@1.8.3]\n",
      "Apr-02 03:04:32.885 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins resolved requirement=[nf-google@1.8.3, nf-tower@1.6.3, nf-wave@1.0.1]\n",
      "Apr-02 03:04:32.885 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-google version: 1.8.3\n",
      "Apr-02 03:04:32.909 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-google@1.8.3' resolved\n",
      "Apr-02 03:04:32.910 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-google@1.8.3'\n",
      "Apr-02 03:04:33.030 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-google@1.8.3\n",
      "Apr-02 03:04:33.031 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-tower version: 1.6.3\n",
      "Apr-02 03:04:33.034 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-tower@1.6.3' resolved\n",
      "Apr-02 03:04:33.035 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-tower@1.6.3'\n",
      "Apr-02 03:04:33.051 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-tower@1.6.3\n",
      "Apr-02 03:04:33.051 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-wave version: 1.0.1\n",
      "Apr-02 03:04:33.057 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-wave@1.0.1' resolved\n",
      "Apr-02 03:04:33.057 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-wave@1.0.1'\n",
      "Apr-02 03:04:33.068 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-wave@1.0.1\n",
      "Apr-02 03:04:33.093 [main] DEBUG n.secret.LocalSecretsProvider - Secrets store: /home/eila/.nextflow/secrets/store.json\n",
      "Apr-02 03:04:33.099 [main] DEBUG nextflow.secret.SecretsLoader - Discovered secrets providers: [nextflow.secret.LocalSecretsProvider@18371d89] - activable => nextflow.secret.LocalSecretsProvider@18371d89\n",
      "Apr-02 03:04:33.193 [main] DEBUG nextflow.Session - Session UUID: b275f460-223c-48c1-a5ae-8ee4e32eaaa2\n",
      "Apr-02 03:04:33.193 [main] DEBUG nextflow.Session - Run name: kickass_hugle\n",
      "Apr-02 03:04:33.194 [main] DEBUG nextflow.Session - Executor pool size: 4\n",
      "Apr-02 03:04:33.476 [main] DEBUG nextflow.file.FilePorter - File porter settings maxRetries=3; maxTransfers=50; pollTimeout=null\n",
      "Apr-02 03:04:33.484 [main] DEBUG nextflow.util.ThreadPoolBuilder - Creating thread pool 'FileTransfer' minSize=10; maxSize=12; workQueue=LinkedBlockingQueue[10000]; allowCoreThreadTimeout=false\n",
      "Apr-02 03:04:33.493 [main] DEBUG nextflow.config.ConfigBuilder - User config file: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_joint/nextflow_joint.config\n",
      "Apr-02 03:04:33.494 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_joint/nextflow_joint.config\n",
      "Apr-02 03:04:33.495 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `google_batch_profile`\n",
      "Apr-02 03:04:33.804 [main] DEBUG nextflow.config.ConfigBuilder - Available config profiles: [google_batch_profile]\n",
      "Apr-02 03:04:33.859 [main] DEBUG nextflow.cli.CmdRun - \n",
      "  Version: 23.10.1 build 5891\n",
      "  Created: 12-01-2024 22:01 UTC \n",
      "  System: Linux 5.10.0-28-cloud-amd64\n",
      "  Runtime: Groovy 3.0.19 on OpenJDK 64-Bit Server VM 11.0.22+7-post-Debian-1deb11u1\n",
      "  Encoding: UTF-8 (UTF-8)\n",
      "  Process: 161881@eila-vm [10.128.0.63]\n",
      "  CPUs: 4 - Mem: 14.6 GB (1.3 GB) - Swap: 0 (0)\n",
      "Apr-02 03:04:33.927 [main] DEBUG nextflow.file.FileHelper - Can't check if specified path is NFS (1): gs://eila-nextflow-bucket/work/JOINT\n",
      "\n",
      "Apr-02 03:04:33.928 [main] DEBUG nextflow.Session - Work-dir: gs://eila-nextflow-bucket/work/JOINT [null]\n",
      "Apr-02 03:04:33.975 [main] DEBUG nextflow.executor.ExecutorFactory - Extension executors providers=[GoogleLifeSciencesExecutor, GoogleBatchExecutor]\n",
      "Apr-02 03:04:34.002 [main] DEBUG nextflow.Session - Observer factory: DefaultObserverFactory\n",
      "Apr-02 03:04:34.006 [main] DEBUG nextflow.Session - Observer factory: WaveFactory\n",
      "Apr-02 03:04:34.006 [main] DEBUG io.seqera.wave.plugin.WaveFactory - Detected Fusion enabled -- Enabling bundle project resources -- Disabling upload of remote bin directory\n",
      "Apr-02 03:04:34.037 [main] DEBUG i.s.wave.plugin.config.WaveConfig - Wave strategy not specified - using default: [container, dockerfile, conda, spack]\n",
      "Apr-02 03:04:34.107 [main] DEBUG io.seqera.wave.plugin.WaveClient - Wave config: WaveConfig(enabled:true, endpoint:https://wave.seqera.io, containerConfigUrl:[], tokensCacheMaxDuration:30m, condaOpts:CondaOpts(mambaImage=mambaorg/micromamba:1.5.1; basePackages=conda-forge::procps-ng, commands=null), spackOpts:SpackOpts(basePackages=null, commands=null), strategy:[container, dockerfile, conda, spack], bundleProjectResources:true, buildRepository:null, cacheRepository:null, retryOpts:RetryOpts(delay:450ms, maxDelay:1m 30s, maxAttempts:10, jitter:0.25), httpClientOpts:HttpOpts(), freezeMode:null)\n",
      "Apr-02 03:04:34.296 [main] DEBUG nextflow.Session - Observer factory: TowerFactory\n",
      "Apr-02 03:04:34.429 [main] DEBUG nextflow.cache.CacheFactory - Using Nextflow cache factory: nextflow.cache.DefaultCacheFactory\n",
      "Apr-02 03:04:34.498 [main] DEBUG nextflow.util.CustomThreadPool - Creating default thread pool > poolSize: 5; maxThreads: 1000\n",
      "Apr-02 03:04:34.700 [main] DEBUG nextflow.Session - Session start\n",
      "Apr-02 03:04:34.718 [main] DEBUG io.seqera.tower.plugin.TowerClient - Creating Tower observer -- endpoint=https://api.tower.nf; requestInterval=1s; aliveInterval=1m; maxRetries=5; backOffBase=3; backOffDelay=250\n",
      "Apr-02 03:04:36.014 [main] DEBUG nextflow.script.ScriptRunner - > Launching execution\n",
      "Apr-02 03:04:36.542 [main] INFO  nextflow.script.BaseScript - Before calling run_joint_barcode_cell_plotting\n",
      "Apr-02 03:04:36.778 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:joint_barcode_metadata_py` matches labels `joint_barcode_metadata_py` for process with name run_joint_barcode_cell_plotting\n",
      "Apr-02 03:04:36.788 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Apr-02 03:04:36.788 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Apr-02 03:04:36.791 [main] DEBUG nextflow.executor.Executor - [warm up] executor > google-batch (fusion enabled)\n",
      "Apr-02 03:04:36.796 [main] DEBUG n.processor.TaskPollingMonitor - Creating task monitor for executor 'google-batch' > capacity: 1000; pollInterval: 10s; dumpInterval: 5m \n",
      "Apr-02 03:04:36.799 [main] DEBUG n.processor.TaskPollingMonitor - >>> barrier register (monitor: google-batch)\n",
      "Apr-02 03:04:36.806 [main] DEBUG nextflow.cloud.google.GoogleOpts - Google auth via application credentials file: /home/eila/git_igvf/atomic-workflows/nextflow/hai-gcp-genomic-beb9207a40b8.json\n",
      "Apr-02 03:04:36.814 [main] DEBUG n.c.google.batch.GoogleBatchExecutor - [GOOGLE BATCH] Executor config=BatchConfig[googleOpts=GoogleOpts(projectId:hai-gcp-genomic, credsFile:/home/eila/git_igvf/atomic-workflows/nextflow/hai-gcp-genomic-beb9207a40b8.json, location:us-central1, enableRequesterPaysBuckets:false, httpConnectTimeout:1m, httpReadTimeout:1m, credentials:ServiceAccountCredentials{clientId=111364348320962814542, clientEmail=eila-batch@hai-gcp-genomic.iam.gserviceaccount.com, privateKeyId=beb9207a40b8b2085c7f48c3fcf8c2342180d471, transportFactoryClassName=com.google.auth.oauth2.OAuth2Utils$DefaultHttpTransportFactory, tokenServerUri=https://oauth2.googleapis.com/token, scopes=[https://www.googleapis.com/auth/cloud-platform], defaultScopes=[], serviceAccountUser=null, quotaProjectId=null, lifetime=3600, useJwtAccessWithScope=false, defaultRetriesEnabled=true})\n",
      "Apr-02 03:04:36.835 [main] DEBUG n.c.google.batch.client.BatchClient - [GOOGLE BATCH] Creating service client with config credentials\n",
      "Apr-02 03:04:37.431 [main] INFO  nextflow.script.BaseScript - After calling run_joint_barcode_cell_plotting\n",
      "Apr-02 03:04:37.438 [main] INFO  nextflow.script.BaseScript - Before calling run_joint_cell_plotting_density\n",
      "Apr-02 03:04:37.441 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:joint_barcode_metadata_r` matches labels `joint_barcode_metadata_r` for process with name run_joint_cell_plotting_density\n",
      "Apr-02 03:04:37.444 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Apr-02 03:04:37.444 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Apr-02 03:04:37.450 [main] INFO  nextflow.script.BaseScript - After calling run_joint_cell_plotting_density\n",
      "Apr-02 03:04:37.458 [main] INFO  nextflow.script.BaseScript - Before run_joint_metrics_html_report\n",
      "Apr-02 03:04:37.461 [main] DEBUG nextflow.script.ProcessConfig - Config settings `withLabel:joint_metrics_html_report` matches labels `joint_metrics_html_report` for process with name run_joint_metrics_html_report\n",
      "Apr-02 03:04:37.463 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch\n",
      "Apr-02 03:04:37.463 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'\n",
      "Apr-02 03:04:37.470 [main] INFO  nextflow.script.BaseScript - After run_joint_metrics_html_report\n",
      "Apr-02 03:04:37.471 [main] DEBUG nextflow.Session - Workflow process names [dsl2]: run_joint_metrics_html_report, run_joint_cell_plotting_density, run_joint_barcode_cell_plotting\n",
      "Apr-02 03:04:38.002 [main] INFO  io.seqera.tower.plugin.TowerClient - Monitor the execution with Nextflow Tower using this URL: https://tower.nf/orgs/ort_eila_org/workspaces/ort_eila_wrkspc/watch/2THExQVsV0zhfU\n",
      "Apr-02 03:04:38.003 [main] DEBUG nextflow.Session - Igniting dataflow network (5)\n",
      "Apr-02 03:04:38.013 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_joint_barcode_cell_plotting\n",
      "Apr-02 03:04:38.014 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_joint_cell_plotting_density\n",
      "Apr-02 03:04:38.015 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > run_joint_metrics_html_report\n",
      "Apr-02 03:04:38.016 [main] DEBUG nextflow.script.ScriptRunner - Parsed script files:\n",
      "  Script_f81eeb9dba1a1cad: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_joint/./../../nf_processes/nf_prcs_joint_barcode_cell_plotting.nf\n",
      "  Script_b02cf6bea2ec2c3e: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_joint/./../../nf_processes/nf_prcs_joint_barcode_cell_plotting_density.nf\n",
      "  Script_d219189e2fbdd6f4: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_joint/nf_wf_fg_singlecell_joint.nf\n",
      "  Script_e28a1820cc42e11a: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_joint/./../../nf_processes/nf_prcs_joint_metrics_html_report.nf\n",
      "Apr-02 03:04:38.016 [main] DEBUG nextflow.script.ScriptRunner - > Awaiting termination \n",
      "Apr-02 03:04:38.016 [main] DEBUG nextflow.Session - Session await\n",
      "Apr-02 03:04:40.701 [FileTransfer-2] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/test_joint_pipeline/BMMC-single-donor.atac.qc.hg38.tss_enrichment_barcode_stats.tsv at gs://eila-nextflow-bucket/work/JOINT/stage-b275f460-223c-48c1-a5ae-8ee4e32eaaa2/31/38fa4c5b590adbdc9fc9da3502c486/BMMC-single-donor.atac.qc.hg38.tss_enrichment_barcode_stats.tsv\n",
      "Apr-02 03:04:40.701 [FileTransfer-3] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/test_joint_pipeline/BMMC-single-donor.joint.barcode.metadata.hg38.csv at gs://eila-nextflow-bucket/work/JOINT/stage-b275f460-223c-48c1-a5ae-8ee4e32eaaa2/8b/5377942960f9dd4f890632b9581664/BMMC-single-donor.joint.barcode.metadata.hg38.csv\n",
      "Apr-02 03:04:40.711 [FileTransfer-1] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/test_joint_pipeline/BMMC-single-donor.qc.rna.hg38.barcode.metadata.tsv at gs://eila-nextflow-bucket/work/JOINT/stage-b275f460-223c-48c1-a5ae-8ee4e32eaaa2/6e/561499cf9683a86c0cfa1cbd46df14/BMMC-single-donor.qc.rna.hg38.barcode.metadata.tsv\n",
      "Apr-02 03:04:40.738 [Actor Thread 1] DEBUG i.s.wave.plugin.config.WaveConfig - Wave strategy not specified - using default: [container, dockerfile, conda, spack]\n",
      "Apr-02 03:04:40.739 [Actor Thread 1] DEBUG io.seqera.wave.plugin.WaveClient - Wave config: WaveConfig(enabled:true, endpoint:https://wave.seqera.io, containerConfigUrl:[], tokensCacheMaxDuration:30m, condaOpts:CondaOpts(mambaImage=mambaorg/micromamba:1.5.1; basePackages=conda-forge::procps-ng, commands=null), spackOpts:SpackOpts(basePackages=null, commands=null), strategy:[container, dockerfile, conda, spack], bundleProjectResources:true, buildRepository:null, cacheRepository:null, retryOpts:RetryOpts(delay:450ms, maxDelay:1m 30s, maxAttempts:10, jitter:0.25), httpClientOpts:HttpOpts(), freezeMode:null)\n",
      "Apr-02 03:04:40.752 [Actor Thread 1] DEBUG io.seqera.wave.plugin.WaveClient - Wave assets bundle bin resources: /home/eila/git_igvf/atomic-workflows/nextflow/nf_workflow/workflow_fg_single_cell_joint/bin\n",
      "Apr-02 03:04:40.845 [Actor Thread 1] DEBUG io.seqera.wave.plugin.WaveClient - Wave request container config: https://fusionfs.seqera.io/releases/v2.2-amd64.json\n",
      "Apr-02 03:04:41.737 [Actor Thread 1] DEBUG io.seqera.wave.plugin.WaveClient - Wave container config response: [200] {\n",
      "  \"entrypoint\": [ \"/usr/bin/fusion\" ],\n",
      "  \"layers\": [\n",
      "    {\n",
      "      \"location\": \"https://fusionfs.seqera.io/releases/pkg/2/2/13/fusion-amd64.tar.gz\",\n",
      "      \"gzipDigest\": \"sha256:f90ba6148e1a4d16763d09ad6fa1562d4b254a01e01958672f4a2e9a59028fe7\",\n",
      "      \"gzipSize\": 31322345,\n",
      "      \"tarDigest\": \"sha256:15dff8049a534f29c8b238e90aa9213e0df1ab5280a161cf8e0f47e1f6c56fa9\",\n",
      "      \"skipHashing\": true\n",
      "    }\n",
      "  ]\n",
      "}\n",
      "\n",
      "Apr-02 03:04:41.912 [Actor Thread 1] DEBUG io.seqera.wave.plugin.WaveClient - Wave request: https://wave.seqera.io/container-token; attempt=1 - request: SubmitContainerTokenRequest(towerAccessToken:eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz, towerRefreshToken:null, towerWorkspaceId:247507587970818, towerEndpoint:https://api.tower.nf, workflowId:null, containerImage:eilalan/joint_plot_r_env:latest, containerFile:null, containerConfig:ContainerConfig(entrypoint:[/usr/bin/fusion], cmd:null, env:null, workingDir:null, layers:[ContainerLayer[location=data:H4sIAAAAAAAA/+x963ob...; tarDigest=sha256:b7a5e038b08e1c6bf46c494d00314006892179f01291f2099a3dcbbf12e4aa1d; gzipDigest=sha256:c07f3f42d6df08eb1d22c6ffc60ec57fcbfb6321314ed79bd8511d5044140fe7; gzipSize=33791], ContainerLayer[location=https://fusionfs.seqera.io/releases/pkg/2/2/13/fusion-amd64.tar.gz; tarDigest=sha256:15dff8049a534f29c8b238e90aa9213e0df1ab5280a161cf8e0f47e1f6c56fa9; gzipDigest=sha256:f90ba6148e1a4d16763d09ad6fa1562d4b254a01e01958672f4a2e9a59028fe7; gzipSize=31322345]]), condaFile:null, spackFile:null, containerPlatform:linux/amd64, buildRepository:null, cacheRepository:null, timestamp:2024-04-02T03:04:41.909635Z, fingerprint:1f3afeb7326dce285ae187337a51713e, freeze:false, format:null, dryRun:false)\n",
      "Apr-02 03:04:41.914 [Actor Thread 4] DEBUG io.seqera.wave.plugin.WaveClient - Wave request: https://wave.seqera.io/container-token; attempt=1 - request: SubmitContainerTokenRequest(towerAccessToken:eyJ0aWQiOiA4Mjk3fS5jNDNmYzA4NTg2MjAxNDQzOTAwYmUzNWEzZTY0MGMyZTc0OWE3M2Mz, towerRefreshToken:null, towerWorkspaceId:247507587970818, towerEndpoint:https://api.tower.nf, workflowId:null, containerImage:eilalan/rank_plot_env:latest, containerFile:null, containerConfig:ContainerConfig(entrypoint:[/usr/bin/fusion], cmd:null, env:null, workingDir:null, layers:[ContainerLayer[location=data:H4sIAAAAAAAA/+x963ob...; tarDigest=sha256:b7a5e038b08e1c6bf46c494d00314006892179f01291f2099a3dcbbf12e4aa1d; gzipDigest=sha256:c07f3f42d6df08eb1d22c6ffc60ec57fcbfb6321314ed79bd8511d5044140fe7; gzipSize=33791], ContainerLayer[location=https://fusionfs.seqera.io/releases/pkg/2/2/13/fusion-amd64.tar.gz; tarDigest=sha256:15dff8049a534f29c8b238e90aa9213e0df1ab5280a161cf8e0f47e1f6c56fa9; gzipDigest=sha256:f90ba6148e1a4d16763d09ad6fa1562d4b254a01e01958672f4a2e9a59028fe7; gzipSize=31322345]]), condaFile:null, spackFile:null, containerPlatform:linux/amd64, buildRepository:null, cacheRepository:null, timestamp:2024-04-02T03:04:41.909636Z, fingerprint:e77d3ab46d21e8799c1b5820025b50dc, freeze:false, format:null, dryRun:false)\n",
      "Apr-02 03:04:42.588 [FileTransfer-5] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/test_joint_pipeline/BMMC-single-donor_qc_metrics.csv at gs://eila-nextflow-bucket/work/JOINT/stage-b275f460-223c-48c1-a5ae-8ee4e32eaaa2/c0/6c6d39386f1765267b422aa61dd08e/BMMC-single-donor_qc_metrics.csv\n",
      "Apr-02 03:04:42.627 [FileTransfer-3] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/test_joint_pipeline/BMMC-single-donor.atac.qc.hist.hg38.log.png at gs://eila-nextflow-bucket/work/JOINT/stage-b275f460-223c-48c1-a5ae-8ee4e32eaaa2/1a/3f1838c4f0227efd28a3bc21043c5c/BMMC-single-donor.atac.qc.hist.hg38.log.png\n",
      "Apr-02 03:04:42.645 [FileTransfer-6] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/test_joint_pipeline/BMMC-single-donor.hg38.joint.qc.plot.png at gs://eila-nextflow-bucket/work/JOINT/stage-b275f460-223c-48c1-a5ae-8ee4e32eaaa2/5f/9d385f1ab5f3cd3e510c2123839561/BMMC-single-donor.hg38.joint.qc.plot.png\n",
      "Apr-02 03:04:42.647 [FileTransfer-4] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/test_joint_pipeline/BMMC-single-donor.rna.log.hg38.txt at gs://eila-nextflow-bucket/work/JOINT/stage-b275f460-223c-48c1-a5ae-8ee4e32eaaa2/9a/80f53f87ababd01da37a891171f8f2/BMMC-single-donor.rna.log.hg38.txt\n",
      "Apr-02 03:04:42.651 [FileTransfer-2] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/test_joint_pipeline/BMMC-single-donor.atac.qc.hg38.tss_enrichment_bulk.png at gs://eila-nextflow-bucket/work/JOINT/stage-b275f460-223c-48c1-a5ae-8ee4e32eaaa2/4e/0d012c77dc49cdeccb2ed638fb5fef/BMMC-single-donor.atac.qc.hg38.tss_enrichment_bulk.png\n",
      "Apr-02 03:04:42.653 [FileTransfer-7] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/test_joint_pipeline/BMMC-single-donor.hg38.joint.density.plot.png at gs://eila-nextflow-bucket/work/JOINT/stage-b275f460-223c-48c1-a5ae-8ee4e32eaaa2/1b/f33ca662eececb1624369d212e22ad/BMMC-single-donor.hg38.joint.density.plot.png\n",
      "Apr-02 03:04:42.654 [Actor Thread 1] DEBUG io.seqera.wave.plugin.WaveClient - Wave response: statusCode=200; body={\"containerToken\":\"cff27314fdb2\",\"targetImage\":\"wave.seqera.io/wt/cff27314fdb2/eilalan/joint_plot_r_env:latest\",\"expiration\":\"2024-04-03T15:04:42.602005873Z\",\"containerImage\":\"docker.io/eilalan/joint_plot_r_env:latest\"}\n",
      "Apr-02 03:04:42.657 [Actor Thread 4] DEBUG io.seqera.wave.plugin.WaveClient - Wave response: statusCode=200; body={\"containerToken\":\"9f3e11cf9aa5\",\"targetImage\":\"wave.seqera.io/wt/9f3e11cf9aa5/eilalan/rank_plot_env:latest\",\"expiration\":\"2024-04-03T15:04:42.604488520Z\",\"containerImage\":\"docker.io/eilalan/rank_plot_env:latest\"}\n",
      "Apr-02 03:04:42.748 [FileTransfer-5] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/test_joint_pipeline/BMMC-single-donor.atac.align.k4.hg38.log.txt at gs://eila-nextflow-bucket/work/JOINT/stage-b275f460-223c-48c1-a5ae-8ee4e32eaaa2/25/9d22cff88993a8cfa78b17d6a8bc58/BMMC-single-donor.atac.align.k4.hg38.log.txt\n",
      "Apr-02 03:04:42.805 [FileTransfer-8] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/test_joint_pipeline/BMMC-single-donor.qc.rna.hg38.umi.barcode.rank.plot.png at gs://eila-nextflow-bucket/work/JOINT/stage-b275f460-223c-48c1-a5ae-8ee4e32eaaa2/9c/d41952ccc79d033e8bf62d03ea1e16/BMMC-single-donor.qc.rna.hg38.umi.barcode.rank.plot.png\n",
      "Apr-02 03:04:42.808 [FileTransfer-10] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/test_joint_pipeline/BMMC-single-donor.qc.rna.hg38.gene.umi.scatter.plot.png at gs://eila-nextflow-bucket/work/JOINT/stage-b275f460-223c-48c1-a5ae-8ee4e32eaaa2/c7/7d06a71390cdb596f215a04b38614d/BMMC-single-donor.qc.rna.hg38.gene.umi.scatter.plot.png\n",
      "Apr-02 03:04:42.812 [FileTransfer-1] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/test_joint_pipeline/run_info.json at gs://eila-nextflow-bucket/work/JOINT/stage-b275f460-223c-48c1-a5ae-8ee4e32eaaa2/6d/606aef9e672a0beb029445cb179738/run_info.json\n",
      "Apr-02 03:04:42.822 [FileTransfer-9] DEBUG nextflow.file.FilePorter - Local cache found for foreign file /home/eila/git_igvf/atomic-workflows/nextflow/nf_data/test_joint_pipeline/BMMC-single-donor.qc.rna.hg38.gene.barcode.rank.plot.png at gs://eila-nextflow-bucket/work/JOINT/stage-b275f460-223c-48c1-a5ae-8ee4e32eaaa2/7e/05be833c2bac736b2163800c3a85d8/BMMC-single-donor.qc.rna.hg38.gene.barcode.rank.plot.png\n",
      "Apr-02 03:04:44.390 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_joint_cell_plotting_density (1)` - adding local volume as fusion scratch: DiskResource(request:375 GB, type:local-ssd)\n",
      "Apr-02 03:04:46.117 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_joint_cell_plotting_density (1)` submitted > job=nf-ab5a4ffe-1712027083610; uid=nf-ab5a4ffe-171202-979b5e32-b52d-4cae0; work-dir=gs://eila-nextflow-bucket/work/JOINT/ab/5a4ffeb95254b787461c2b6c72c048\n",
      "Apr-02 03:04:46.117 [Task submitter] INFO  nextflow.Session - [ab/5a4ffe] Submitted process > run_joint_cell_plotting_density (1)\n",
      "Apr-02 03:04:46.652 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_joint_barcode_cell_plotting (1)` - adding local volume as fusion scratch: DiskResource(request:375 GB, type:local-ssd)\n",
      "Apr-02 03:04:46.870 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_joint_barcode_cell_plotting (1)` submitted > job=nf-ec97ff9f-1712027083610; uid=nf-ec97ff9f-171202-70d3a26f-8caf-4cd10; work-dir=gs://eila-nextflow-bucket/work/JOINT/ec/97ff9f89f486df2fd72315c2398f10\n",
      "Apr-02 03:04:46.871 [Task submitter] INFO  nextflow.Session - [ec/97ff9f] Submitted process > run_joint_barcode_cell_plotting (1)\n",
      "Apr-02 03:04:47.388 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_joint_metrics_html_report (1)` - adding local volume as fusion scratch: DiskResource(request:375 GB, type:local-ssd)\n",
      "Apr-02 03:04:47.561 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_joint_metrics_html_report (1)` submitted > job=nf-c88eb037-1712027084207; uid=nf-c88eb037-171202-00b90826-a682-427c0; work-dir=gs://eila-nextflow-bucket/work/JOINT/c8/8eb037a78ce3c6309338343e95a07b\n",
      "Apr-02 03:04:47.562 [Task submitter] INFO  nextflow.Session - [c8/8eb037] Submitted process > run_joint_metrics_html_report (1)\n",
      "Apr-02 03:07:26.985 [Task monitor] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_joint_metrics_html_report (1)` - terminated job=nf-c88eb037-1712027084207; state=SUCCEEDED\n",
      "Apr-02 03:07:27.315 [Task monitor] DEBUG n.processor.TaskPollingMonitor - Task completed > TaskHandler[id: 3; name: run_joint_metrics_html_report (1); status: COMPLETED; exit: 0; error: -; workDir: gs://eila-nextflow-bucket/work/JOINT/c8/8eb037a78ce3c6309338343e95a07b]\n",
      "Apr-02 03:08:36.865 [Task monitor] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_joint_cell_plotting_density (1)` - terminated job=nf-ab5a4ffe-1712027083610; state=SUCCEEDED\n",
      "Apr-02 03:08:37.187 [Task monitor] DEBUG n.processor.TaskPollingMonitor - Task completed > TaskHandler[id: 1; name: run_joint_cell_plotting_density (1); status: COMPLETED; exit: 0; error: -; workDir: gs://eila-nextflow-bucket/work/JOINT/ab/5a4ffeb95254b787461c2b6c72c048]\n",
      "Apr-02 03:09:46.815 [Task monitor] DEBUG n.processor.TaskPollingMonitor - !! executor google-batch > tasks to be completed: 1 -- submitted tasks are shown below\n",
      "~> TaskHandler[id: 2; name: run_joint_barcode_cell_plotting (1); status: RUNNING; exit: -; error: -; workDir: gs://eila-nextflow-bucket/work/JOINT/ec/97ff9f89f486df2fd72315c2398f10]\n",
      "Apr-02 03:10:06.859 [Task monitor] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Process `run_joint_barcode_cell_plotting (1)` - terminated job=nf-ec97ff9f-1712027083610; state=SUCCEEDED\n",
      "Apr-02 03:10:07.058 [Task monitor] DEBUG n.processor.TaskPollingMonitor - Task completed > TaskHandler[id: 2; name: run_joint_barcode_cell_plotting (1); status: COMPLETED; exit: 0; error: -; workDir: gs://eila-nextflow-bucket/work/JOINT/ec/97ff9f89f486df2fd72315c2398f10]\n",
      "Apr-02 03:10:07.516 [main] DEBUG nextflow.Session - Session await > all processes finished\n",
      "Apr-02 03:10:07.743 [Task monitor] DEBUG n.processor.TaskPollingMonitor - <<< barrier arrives (monitor: google-batch) - terminating tasks monitor poll loop\n",
      "Apr-02 03:10:07.744 [main] DEBUG nextflow.Session - Session await > all barriers passed\n",
      "Apr-02 03:10:07.758 [main] DEBUG n.trace.WorkflowStatsObserver - Workflow completed > WorkflowStats[succeededCount=3; failedCount=0; ignoredCount=0; cachedCount=0; pendingCount=0; submittedCount=0; runningCount=0; retriesCount=0; abortedCount=0; succeedDuration=3m 13s; failedDuration=0ms; cachedDuration=0ms;loadCpus=0; loadMemory=0; peakRunning=2; peakCpus=2; peakMemory=0; ]\n",
      "Apr-02 03:10:08.451 [main] DEBUG nextflow.cache.CacheDB - Closing CacheDB done\n",
      "Apr-02 03:10:08.451 [main] INFO  org.pf4j.AbstractPluginManager - Stop plugin 'nf-wave@1.0.1'\n",
      "Apr-02 03:10:08.452 [main] DEBUG nextflow.plugin.BasePlugin - Plugin stopped nf-wave\n",
      "Apr-02 03:10:08.452 [main] INFO  org.pf4j.AbstractPluginManager - Stop plugin 'nf-tower@1.6.3'\n",
      "Apr-02 03:10:08.452 [main] DEBUG nextflow.plugin.BasePlugin - Plugin stopped nf-tower\n",
      "Apr-02 03:10:08.452 [main] INFO  org.pf4j.AbstractPluginManager - Stop plugin 'nf-google@1.8.3'\n",
      "Apr-02 03:10:08.452 [main] DEBUG nextflow.plugin.BasePlugin - Plugin stopped nf-google\n",
      "Apr-02 03:10:08.494 [main] DEBUG nextflow.util.ThreadPoolManager - Thread pool 'FileTransfer' shutdown completed (hard=false)\n",
      "Apr-02 03:10:08.496 [main] DEBUG nextflow.script.ScriptRunner - > Execution complete -- Goodbye\n"
     ]
    }
   ],
   "source": [
    "!cat .nextflow.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# nf_data_base_path = '/home/eila/git_igvf/cellatlas-scrna-seq-pipeline/nextflow/nf_data/'\n",
    "\n",
    "# # Write the configuration file\n",
    "# with open(\"nextflow_vm.config\", \"w\") as config_file:\n",
    "#     config_file.write(f\"\"\"\n",
    "# manifest {{\n",
    "#   description = 'A nexflow pipeline for fg single cell joint qc' \n",
    "#   mainScript = 'nf_wf_fg_singlecell_joint.nf'\n",
    "# }}\n",
    "\n",
    "# profiles {{\n",
    "#   google_batch_profile {{\n",
    "#     workDir = 'gs://eila-nextflow-bucket/work/ATAC'\n",
    "#     includeConfig 'conf/docker-google-batch.config'\n",
    "#     docker.enabled = true\n",
    "#     docker.envWhitelist=['ENV_SYNAPSE_TOKEN']\n",
    "#     executor = 'google-batch'\n",
    "#     google.project = 'hai-gcp-genomic'\n",
    "#     google.location = 'us-central1'\n",
    "#     google.batch.spot = false\n",
    "#     google.batch.serviceAccountEmail='eila-batch@hai-gcp-genomic.iam.gserviceaccount.com'\n",
    "#   }}   \n",
    "# }}\n",
    "\n",
    "# // Define ENV_SYNAPSE_TOKEN as a parameter\n",
    "# params.SC_JOINT_CELL_BARCODE_PLOT_SCRIPT_PY='joint_cell_plotting.py'\n",
    "# params.SC_JOINT_CELL_BARCODE_PLOT_RNA_METRICS='{nf_data_base_path}nf_wf_fg_singlecell/na.na'\n",
    "# params.SC_JOINT_CELL_BARCODE_PLOT_ATAC_METRICS='{nf_data_base_path}nf_wf_fg_singlecell/na1.na'\n",
    "# //Minimum number of UMIs/fragments required for a cell to be plotted\n",
    "# params.SC_JOINT_CELL_BARCODE_PLOT_REMOVE_LOW_YIELDING_CELLS=5\n",
    "# params.SC_JOINT_CELL_BARCODE_PLOT_METADATA_FILE='{nf_data_base_path}nf_wf_fg_singlecell/na2.na'\n",
    "# params.SC_JOINT_CELL_BARCODE_PLOT_MIN_UMIS=5\n",
    "# params.SC_JOINT_CELL_BARCODE_PLOT_MIN_GENES=5\n",
    "# params.SC_JOINT_CELL_BARCODE_PLOT_MIN_TSS=5\n",
    "# //Cutoff for minimum number of ATAC fragments\n",
    "# params.SC_JOINT_CELL_BARCODE_PLOT_MIN_FRAGS=5\n",
    "# params.SC_JOINT_CELL_BARCODE_PLOT_PKR='PKR123'\n",
    "# params.SC_JOINT_CELL_BARCODE_PLOT_SCRIPT_R='joint_cell_plotting_density.R'\n",
    "\n",
    "# params.SC_JOINT_CELL_BARCODE_PLOT_HTML_SCRIPT='write_html.py'\n",
    "# params.SC_JOINT_CELL_BARCODE_PLOT_HTML_ATAC_METRICS='{nf_data_base_path}nf_wf_fg_singlecell/na.na'\n",
    "# params.SC_JOINT_CELL_BARCODE_PLOT_HTML_RNA_METRICS='{nf_data_base_path}nf_wf_fg_singlecell/na1.na'\n",
    "# params.SC_JOINT_CELL_BARCODE_PLOT_HTML_IMAGE_FILES='{nf_data_base_path}nf_wf_fg_singlecell/na2.na'\n",
    "# params.SC_JOINT_CELL_BARCODE_PLOT_HTML_LOG_FILES='{nf_data_base_path}nf_wf_fg_singlecell/na3.na'\n",
    "# \"\"\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
